---
id: 2yuhautdd3mtmj0ikxas2ty
title: Sgd Expression Studies
desc: ''
updated: 1739991774133
created: 1739991774133
---
# SGD Expression Studies


This document contains README files from all expression studies in SGD.


## Table of Contents


- [DeRisi_1997_PMID_9381177](#derisi_1997_pmid_9381177)

- [Cho_1998_PMID_9702192](#cho_1998_pmid_9702192)

- [Chu_1998_PMID_9784122](#chu_1998_pmid_9784122)

- [Marton_1998_PMID_9809554](#marton_1998_pmid_9809554)

- [Spellman_1998_PMID_9843569](#spellman_1998_pmid_9843569)

- [Ferea_1999_PMID_10449761](#ferea_1999_pmid_10449761)

- [Galitski_1999_PMID_10398601](#galitski_1999_pmid_10398601)

- [Hardwick_1999_PMID_10611304](#hardwick_1999_pmid_10611304)

- [Madhani_1999_PMID_10535956](#madhani_1999_pmid_10535956)

- [Wyrick_1999_PMID_10586882](#wyrick_1999_pmid_10586882)

- [Bernstein_2000_PMID_11095743](#bernstein_2000_pmid_11095743)

- [Gasch_2000_PMID_11102521](#gasch_2000_pmid_11102521)

- [Gross_2000_PMID_10922376](#gross_2000_pmid_10922376)

- [Hughes_2000_PMID_10929718](#hughes_2000_pmid_10929718)

- [Lee_2000_PMID_12760044](#lee_2000_pmid_12760044)

- [Lopez_2000_PMID_10940042](#lopez_2000_pmid_10940042)

- [Lyons_2000_PMID_10884426](#lyons_2000_pmid_10884426)

- [Ogawa_2000_PMID_11102525](#ogawa_2000_pmid_11102525)

- [Primig_2000_PMID_11101837](#primig_2000_pmid_11101837)

- [Roberts_2000_PMID_10657304](#roberts_2000_pmid_10657304)

- [Robertson_2000_PMID_10811893](#robertson_2000_pmid_10811893)

- [Sudarsanam_2000_PMID_10725359](#sudarsanam_2000_pmid_10725359)

- [Travers_2000_PMID_10847680](#travers_2000_pmid_10847680)

- [Yun_2000_PMID_10744769](#yun_2000_pmid_10744769)

- [Zhu_2000_PMID_10894548](#zhu_2000_pmid_10894548)

- [Angus-Hill_2001_PMID_11336698](#angus-hill_2001_pmid_11336698)

- [Backhus_2001_PMID_12702356](#backhus_2001_pmid_12702356)

- [Baetz_2001_PMID_11533240](#baetz_2001_pmid_11533240)

- [Bedalov_2001_PMID_11752457](#bedalov_2001_pmid_11752457)

- [Carmel-Harel_2001_PMID_11169101](#carmel-harel_2001_pmid_11169101)

- [Carroll_2001_PMID_11675494](#carroll_2001_pmid_11675494)

- [Causton_2001_PMID_11179418](#causton_2001_pmid_11179418)

- [Epstein_2001_PMID_11179416](#epstein_2001_pmid_11179416)

- [Gasch_2001_PMID_11598186](#gasch_2001_pmid_11598186)

- [Ideker_2001_PMID_11340206](#ideker_2001_pmid_11340206)

- [Keller_2001_PMID_11504737](#keller_2001_pmid_11504737)

- [Kuhn_2001_PMID_11154278](#kuhn_2001_pmid_11154278)

- [Protchenko_2001_PMID_11673473](#protchenko_2001_pmid_11673473)

- [Chitikila_2002_PMID_12419230](#chitikila_2002_pmid_12419230)

- [Clark_2002_PMID_11988574](#clark_2002_pmid_11988574)

- [Cohen_2002_PMID_12006656](#cohen_2002_pmid_12006656)

- [Fleming_2002_PMID_11830665](#fleming_2002_pmid_11830665)

- [Huang_2002_PMID_12077337](#huang_2002_pmid_12077337)

- [Kitagawa_2002_PMID_12269742](#kitagawa_2002_pmid_12269742)

- [Olesen_2002_PMID_12702272](#olesen_2002_pmid_12702272)

- [Pramila_2002_PMID_12464633](#pramila_2002_pmid_12464633)

- [Smith_2002_PMID_12135984](#smith_2002_pmid_12135984)

- [Wang_2002_PMID_11972065](#wang_2002_pmid_11972065)

- [Williams_2002_PMID_12370439](#williams_2002_pmid_12370439)

- [Yoshimoto_2002_PMID_12058033](#yoshimoto_2002_pmid_12058033)

- [Bro_2003_PMID_12791685](#bro_2003_pmid_12791685)

- [Bulik_2003_PMID_14555471](#bulik_2003_pmid_14555471)

- [Duvel_2003_PMID_12820961](#duvel_2003_pmid_12820961)

- [Fry_2003_PMID_12875747](#fry_2003_pmid_12875747)

- [Jones_2003_PMID_14570984](#jones_2003_pmid_14570984)

- [Kitagawa_2003_PMID_12854720](#kitagawa_2003_pmid_12854720)

- [Meneghini_2003_PMID_12628191](#meneghini_2003_pmid_12628191)

- [Segal_2003_PMID_12740579](#segal_2003_pmid_12740579)

- [Belli_2004_PMID_14722110](#belli_2004_pmid_14722110)

- [Cullen_2004_PMID_15256499](#cullen_2004_pmid_15256499)

- [Daran-Lapujade_2004_PMID_14630934](#daran-lapujade_2004_pmid_14630934)

- [De_Nadal_2004_PMID_14737171](#de_nadal_2004_pmid_14737171)

- [Garcia_2004_PMID_14739279](#garcia_2004_pmid_14739279)

- [Haugen_2004_PMID_15575969](#haugen_2004_pmid_15575969)

- [Huang_2004_PMID_15539461](#huang_2004_pmid_15539461)

- [Huisinga_2004_PMID_14992726](#huisinga_2004_pmid_14992726)

- [Jansen_2004_PMID_15066785](#jansen_2004_pmid_15066785)

- [Jin_2004_PMID_15528549](#jin_2004_pmid_15528549)

- [Klevecz_2004_PMID_14734811](#klevecz_2004_pmid_14734811)

- [Leber_2004_PMID_15314654](#leber_2004_pmid_15314654)

- [Martin_2004_PMID_15476558](#martin_2004_pmid_15476558)

- [Miyake_2004_PMID_15192094](#miyake_2004_pmid_15192094)

- [Mizuguchi_2004_PMID_14645854](#mizuguchi_2004_pmid_14645854)

- [Mnaimneh_2004_PMID_15242642](#mnaimneh_2004_pmid_15242642)

- [O'Rourke_2004_PMID_14595107](#o'rourke_2004_pmid_14595107)

- [Orlandi_2004_PMID_14623890](#orlandi_2004_pmid_14623890)

- [Pitkanen_2004_PMID_15520001](#pitkanen_2004_pmid_15520001)

- [Prinz_2004_PMID_14993204](#prinz_2004_pmid_14993204)

- [Rodriguez-Navarro_2004_PMID_14718168](#rodriguez-navarro_2004_pmid_14718168)

- [Sabet_2004_PMID_15456858](#sabet_2004_pmid_15456858)

- [Saldanha_2004_PMID_15240820](#saldanha_2004_pmid_15240820)

- [Sapra_2004_PMID_15452114](#sapra_2004_pmid_15452114)

- [Schawalder_2004_PMID_15616569](#schawalder_2004_pmid_15616569)

- [Shakoury-Elizeh_2004_PMID_14668481](#shakoury-elizeh_2004_pmid_14668481)

- [Shapira_2004_PMID_15371544](#shapira_2004_pmid_15371544)

- [van_Attikum_2004_PMID_15607975](#van_attikum_2004_pmid_15607975)

- [Aguilera_2005_PMID_15780657](#aguilera_2005_pmid_15780657)

- [Boer_2005_PMID_15949974](#boer_2005_pmid_15949974)

- [Brauer_2005_PMID_15758028](#brauer_2005_pmid_15758028)

- [Brem_2005_PMID_15659551](#brem_2005_pmid_15659551)

- [Caba_2005_PMID_15878181](#caba_2005_pmid_15878181)

- [Dion_2005_PMID_15795371](#dion_2005_pmid_15795371)

- [Eriksson_2005_PMID_16199888](#eriksson_2005_pmid_16199888)

- [Gardner_2005_PMID_15797381](#gardner_2005_pmid_15797381)

- [Gardner_2005_PMID_15988024](#gardner_2005_pmid_15988024)

- [Ghazal_2005_PMID_15798187](#ghazal_2005_pmid_15798187)

- [Iwahashi_2005_PMID_15876434](#iwahashi_2005_pmid_15876434)

- [Jansen_2005_PMID_15870473](#jansen_2005_pmid_15870473)

- [Kitagawa_2005_PMID_16377885](#kitagawa_2005_pmid_16377885)

- [Lai_2005_PMID_15870279](#lai_2005_pmid_15870279)

- [Lee_2005_PMID_15989963](#lee_2005_pmid_15989963)

- [Lu_2005_PMID_16208474](#lu_2005_pmid_16208474)

- [Matsumoto_2005_PMID_16209719](#matsumoto_2005_pmid_16209719)

- [Penheiter_2005_PMID_16246724](#penheiter_2005_pmid_16246724)

- [Ronald_2005_PMID_15687292](#ronald_2005_pmid_15687292)

- [Rosaleny_2005_PMID_16023114](#rosaleny_2005_pmid_16023114)

- [Rudra_2005_PMID_15692568](#rudra_2005_pmid_15692568)

- [Singh_2005_PMID_16332871](#singh_2005_pmid_16332871)

- [Tai_2005_PMID_15496405](#tai_2005_pmid_15496405)

- [Takagi_2005_PMID_15837426](#takagi_2005_pmid_15837426)

- [Tamaki_2005_PMID_15743410](#tamaki_2005_pmid_15743410)

- [Tu_2005_PMID_16254148](#tu_2005_pmid_16254148)

- [Wyers_2005_PMID_15935759](#wyers_2005_pmid_15935759)

- [Yamamoto_2005_PMID_15647283](#yamamoto_2005_pmid_15647283)

- [Ye_2005_PMID_15808514](#ye_2005_pmid_15808514)

- [Yeang_2005_PMID_15998451](#yeang_2005_pmid_15998451)

- [Alper_2006_PMID_17158319](#alper_2006_pmid_17158319)

- [Anderson_2006_PMID_16896209](#anderson_2006_pmid_16896209)

- [Angell_2006_PMID_17185230](#angell_2006_pmid_17185230)

- [Aragon_2006_PMID_16507144](#aragon_2006_pmid_16507144)

- [Arigo_2006_PMID_16973436](#arigo_2006_pmid_16973436)

- [Auld_2006_PMID_16543154](#auld_2006_pmid_16543154)

- [Benton_2006_PMID_17140446](#benton_2006_pmid_17140446)

- [Boccazzi_2006_PMID_16739953](#boccazzi_2006_pmid_16739953)

- [Carter_2006_PMID_16533914](#carter_2006_pmid_16533914)

- [Chen_2006_PMID_16397762](#chen_2006_pmid_16397762)

- [Chua_2006_PMID_16880382](#chua_2006_pmid_16880382)

- [Crisp_2006_PMID_16437160](#crisp_2006_pmid_16437160)

- [Davis_2006_PMID_16484372](#davis_2006_pmid_16484372)

- [Dubacq_2006_PMID_16328372](#dubacq_2006_pmid_16328372)

- [Friedlander_2006_PMID_16542486](#friedlander_2006_pmid_16542486)

- [Fry_2006_PMID_17163986](#fry_2006_pmid_17163986)

- [Gonzalez_2006_PMID_16973600](#gonzalez_2006_pmid_16973600)

- [Guan_2006_PMID_17166056](#guan_2006_pmid_17166056)

- [Guo_2006_PMID_16122766](#guo_2006_pmid_16122766)

- [Hashikawa_2006_PMID_16361698](#hashikawa_2006_pmid_16361698)

- [Irvin_2006_PMID_16407318](#irvin_2006_pmid_16407318)

- [Johanson_2006_PMID_16440349](#johanson_2006_pmid_16440349)

- [Kelly_2006_PMID_16720269](#kelly_2006_pmid_16720269)

- [Kresnowati_2006_PMID_16969341](#kresnowati_2006_pmid_16969341)

- [Kuranda_2006_PMID_16925551](#kuranda_2006_pmid_16925551)

- [Lai_2006_PMID_16963631](#lai_2006_pmid_16963631)

- [Landry_2006_PMID_16427747](#landry_2006_pmid_16427747)

- [Li_2006_PMID_17043222](#li_2006_pmid_17043222)

- [Malagon_2006_PMID_16510790](#malagon_2006_pmid_16510790)

- [Mendiratta_2006_PMID_16415340](#mendiratta_2006_pmid_16415340)

- [Oberstrass_2006_PMID_16429156](#oberstrass_2006_pmid_16429156)

- [Parra_2006_PMID_16648479](#parra_2006_pmid_16648479)

- [Pramila_2006_PMID_16912276](#pramila_2006_pmid_16912276)

- [Reinke_2006_PMID_16923813](#reinke_2006_pmid_16923813)

- [Roberts_2006_PMID_16741729](#roberts_2006_pmid_16741729)

- [Ronen_2006_PMID_16381818](#ronen_2006_pmid_16381818)

- [Scherens_2006_PMID_16879428](#scherens_2006_pmid_16879428)

- [Schrader_2006_PMID_16987817](#schrader_2006_pmid_16987817)

- [Simons_2006_PMID_16870766](#simons_2006_pmid_16870766)

- [Takahashi_2006_PMID_16857587](#takahashi_2006_pmid_16857587)

- [Tanaka_2006_PMID_16943074](#tanaka_2006_pmid_16943074)

- [Yu_2006_PMID_16461773](#yu_2006_pmid_16461773)

- [Abbott_2007_PMID_17484738](#abbott_2007_pmid_17484738)

- [Abe_2007_PMID_17910955](#abe_2007_pmid_17910955)

- [Barbara_2007_PMID_17124610](#barbara_2007_pmid_17124610)

- [Boer_2007_PMID_17419774](#boer_2007_pmid_17419774)

- [Carter_2007_PMID_17389876](#carter_2007_pmid_17389876)

- [Daran-Lapujade_2007_PMID_17898166](#daran-lapujade_2007_pmid_17898166)

- [De_Nicola_2007_PMID_17933919](#de_nicola_2007_pmid_17933919)

- [Demae_2007_PMID_17986087](#demae_2007_pmid_17986087)

- [Dettman_2007_PMID_17538619](#dettman_2007_pmid_17538619)

- [Dong_2007_PMID_17317628](#dong_2007_pmid_17317628)

- [Hickman_2007_PMID_17785431](#hickman_2007_pmid_17785431)

- [Hu_2007_PMID_17417638](#hu_2007_pmid_17417638)

- [Huisinga_2007_PMID_17407552](#huisinga_2007_pmid_17407552)

- [Iwahashi_2007_PMID_17408496](#iwahashi_2007_pmid_17408496)

- [Iwahashi_2007_PMID_18075112](#iwahashi_2007_pmid_18075112)

- [James_2007_PMID_17660549](#james_2007_pmid_17660549)

- [Jin_2007_PMID_17664279](#jin_2007_pmid_17664279)

- [Johansson_2007_PMID_18087042](#johansson_2007_pmid_18087042)

- [Joseph-Strauss_2007_PMID_17999778](#joseph-strauss_2007_pmid_17999778)

- [Knijnenburg_2007_PMID_17241460](#knijnenburg_2007_pmid_17241460)

- [Komili_2007_PMID_17981122](#komili_2007_pmid_17981122)

- [Kugou_2007_PMID_17396017](#kugou_2007_pmid_17396017)

- [Landry_2007_PMID_17525304](#landry_2007_pmid_17525304)

- [Levy_2007_PMID_17327914](#levy_2007_pmid_17327914)

- [Liko_2007_PMID_17616518](#liko_2007_pmid_17616518)

- [Medintz_2007_PMID_17700863](#medintz_2007_pmid_17700863)

- [Menacho-Marquez_2007_PMID_17890903](#menacho-marquez_2007_pmid_17890903)

- [Mendes-Ferreira_2007_PMID_17601813](#mendes-ferreira_2007_pmid_17601813)

- [Mutiu_2007_PMID_17447102](#mutiu_2007_pmid_17447102)

- [Mutiu_2007_PMID_17660562](#mutiu_2007_pmid_17660562)

- [Parra_2007_PMID_17724083](#parra_2007_pmid_17724083)

- [Sheehan_2007_PMID_17201921](#sheehan_2007_pmid_17201921)

- [Smith_2007_PMID_17551510](#smith_2007_pmid_17551510)

- [Stern_2007_PMID_17453047](#stern_2007_pmid_17453047)

- [Tai_2007_PMID_17928405](#tai_2007_pmid_17928405)

- [Thorsen_2007_PMID_17327492](#thorsen_2007_pmid_17327492)

- [Tompa_2007_PMID_17179083](#tompa_2007_pmid_17179083)

- [Torres_2007_PMID_17702937](#torres_2007_pmid_17702937)

- [Urban_2007_PMID_17560372](#urban_2007_pmid_17560372)

- [Vemuri_2007_PMID_17287356](#vemuri_2007_pmid_17287356)

- [Venkatasubrahmanyam_2007_PMID_17925448](#venkatasubrahmanyam_2007_pmid_17925448)

- [Wiesenberger_2007_PMID_17337637](#wiesenberger_2007_pmid_17337637)

- [Yarragudi_2007_PMID_17158163](#yarragudi_2007_pmid_17158163)

- [Abbott_2008_PMID_18676708](#abbott_2008_pmid_18676708)

- [Agarwal_2008_PMID_18156292](#agarwal_2008_pmid_18156292)

- [Aragon_2008_PMID_18199684](#aragon_2008_pmid_18199684)

- [Baerends_2008_PMID_18378663](#baerends_2008_pmid_18378663)

- [Berry_2008_PMID_18753408](#berry_2008_pmid_18753408)

- [Brauer_2008_PMID_17959824](#brauer_2008_pmid_17959824)

- [Brown_2008_PMID_18422925](#brown_2008_pmid_18422925)

- [Capaldi_2008_PMID_18931682](#capaldi_2008_pmid_18931682)

- [Chan_2008_PMID_18708563](#chan_2008_pmid_18708563)

- [Chechik_2008_PMID_18953355](#chechik_2008_pmid_18953355)

- [Cheung_2008_PMID_18998772](#cheung_2008_pmid_18998772)

- [Cipollina_2008_PMID_18174152](#cipollina_2008_pmid_18174152)

- [Cipollina_2008_PMID_18524923](#cipollina_2008_pmid_18524923)

- [Erlich_2008_PMID_19005567](#erlich_2008_pmid_19005567)

- [Gibson_2008_PMID_18373683](#gibson_2008_pmid_18373683)

- [Gregori_2008_PMID_18621731](#gregori_2008_pmid_18621731)

- [Gresham_2008_PMID_19079573](#gresham_2008_pmid_19079573)

- [Grund_2008_PMID_18762579](#grund_2008_pmid_18762579)

- [Hazelwood_2008_PMID_18281432](#hazelwood_2008_pmid_18281432)

- [Kao_2008_PMID_19029899](#kao_2008_pmid_19029899)

- [Kaplan_2008_PMID_19023413](#kaplan_2008_pmid_19023413)

- [Klockow_2008_PMID_18455824](#klockow_2008_pmid_18455824)

- [Kundaje_2008_PMID_19008939](#kundaje_2008_pmid_19008939)

- [Kvitek_2008_PMID_18927628](#kvitek_2008_pmid_18927628)

- [Lai_2008_PMID_19105839](#lai_2008_pmid_19105839)

- [Lee_2008_PMID_18679056](#lee_2008_pmid_18679056)

- [Levy_2008_PMID_18687885](#levy_2008_pmid_18687885)

- [Marks_2008_PMID_18215224](#marks_2008_pmid_18215224)

- [Molina-Navarro_2008_PMID_18424442](#molina-navarro_2008_pmid_18424442)

- [Nag_2008_PMID_18508805](#nag_2008_pmid_18508805)

- [Orlando_2008_PMID_18463633](#orlando_2008_pmid_18463633)

- [Pan_2008_PMID_18366703](#pan_2008_pmid_18366703)

- [Pelechano_2008_PMID_17914747](#pelechano_2008_pmid_17914747)

- [Protchenko_2008_PMID_18326586](#protchenko_2008_pmid_18326586)

- [Puig_2008_PMID_18522836](#puig_2008_pmid_18522836)

- [Ro_2008_PMID_18983675](#ro_2008_pmid_18983675)

- [Rojas_2008_PMID_18667070](#rojas_2008_pmid_18667070)

- [Salusjarvi_2008_PMID_18533012](#salusjarvi_2008_pmid_18533012)

- [Shalem_2008_PMID_18854817](#shalem_2008_pmid_18854817)

- [Shirra_2008_PMID_18955495](#shirra_2008_pmid_18955495)

- [Shivaswamy_2008_PMID_18212068](#shivaswamy_2008_pmid_18212068)

- [Smith_2008_PMID_18416601](#smith_2008_pmid_18416601)

- [Tirosh_2008_PMID_18197176](#tirosh_2008_pmid_18197176)

- [Trott_2008_PMID_18199679](#trott_2008_pmid_18199679)

- [Willis_2008_PMID_18604275](#willis_2008_pmid_18604275)

- [Wu_2008_PMID_18673560](#wu_2008_pmid_18673560)

- [Yiu_2008_PMID_18245757](#yiu_2008_pmid_18245757)

- [Zhu_2008_PMID_18552845](#zhu_2008_pmid_18552845)

- [van_den_Brink_2008_PMID_18304306](#van_den_brink_2008_pmid_18304306)

- [Alejandro-Osorio_2009_PMID_19470158](#alejandro-osorio_2009_pmid_19470158)

- [Anderson_2009_PMID_19273689](#anderson_2009_pmid_19273689)

- [Azzouz_2009_PMID_19155328](#azzouz_2009_pmid_19155328)

- [Borde_2009_PMID_19078966](#borde_2009_pmid_19078966)

- [Bradley_2009_PMID_19180179](#bradley_2009_pmid_19180179)

- [Falbo_2009_PMID_19855395](#falbo_2009_pmid_19855395)

- [Garcia_2009_PMID_19234305](#garcia_2009_pmid_19234305)

- [Holbein_2009_PMID_19324962](#holbein_2009_pmid_19324962)

- [Kim_2009_PMID_19456872](#kim_2009_pmid_19456872)

- [Knijnenburg_2009_PMID_19173729](#knijnenburg_2009_pmid_19173729)

- [Moxley_2009_PMID_19346491](#moxley_2009_pmid_19346491)

- [Rodriguez-Quinones_2009_PMID_19426543](#rodriguez-quinones_2009_pmid_19426543)

- [Rossouw_2009_PMID_19711068](#rossouw_2009_pmid_19711068)

- [Sprouse_2009_PMID_19098311](#sprouse_2009_pmid_19098311)

- [Zhu_2009_PMID_19158363](#zhu_2009_pmid_19158363)

- [Adomas_2010_PMID_20716356](#adomas_2010_pmid_20716356)

- [Aghajan_2010_PMID_20581845](#aghajan_2010_pmid_20581845)

- [Amit_2010_PMID_20600117](#amit_2010_pmid_20600117)

- [Berthelet_2010_PMID_20439772](#berthelet_2010_pmid_20439772)

- [Canelas_2010_PMID_21266995](#canelas_2010_pmid_21266995)

- [Dowell_2010_PMID_20844014](#dowell_2010_pmid_20844014)

- [Fendt_2010_PMID_21119627](#fendt_2010_pmid_21119627)

- [Leadsham_2010_PMID_21108829](#leadsham_2010_pmid_21108829)

- [Li_2010_PMID_20439498](#li_2010_pmid_20439498)

- [Lopez-Garcia_2010_PMID_21078184](#lopez-garcia_2010_pmid_21078184)

- [Ma_2010_PMID_21106074](#ma_2010_pmid_21106074)

- [Mendes-Ferreira_2010_PMID_20952643](#mendes-ferreira_2010_pmid_20952643)

- [Morillo-Huesca_2010_PMID_20711347](#morillo-huesca_2010_pmid_20711347)

- [Munding_2010_PMID_21123654](#munding_2010_pmid_21123654)

- [Otero_2010_PMID_21176163](#otero_2010_pmid_21176163)

- [Papini_2010_PMID_20815084](#papini_2010_pmid_20815084)

- [Perez-Valle_2010_PMID_20952580](#perez-valle_2010_pmid_20952580)

- [Rivera-Ruiz_2010_PMID_21126371](#rivera-ruiz_2010_pmid_21126371)

- [Rodriguez-Colman_2010_PMID_20847055](#rodriguez-colman_2010_pmid_20847055)

- [Sahi_2010_PMID_19822657](#sahi_2010_pmid_19822657)

- [Shakoury-Elizeh_2010_PMID_20231268](#shakoury-elizeh_2010_pmid_20231268)

- [Simola_2010_PMID_20969771](#simola_2010_pmid_20969771)

- [Stanley_2010_PMID_20393705](#stanley_2010_pmid_20393705)

- [Stanley_2010_PMID_20661734](#stanley_2010_pmid_20661734)

- [Stefanini_2010_PMID_20501664](#stefanini_2010_pmid_20501664)

- [Talarek_2010_PMID_20471941](#talarek_2010_pmid_20471941)

- [Verwaal_2010_PMID_20632327](#verwaal_2010_pmid_20632327)

- [Wenger_2010_PMID_20485559](#wenger_2010_pmid_20485559)

- [Wisselink_2010_PMID_20816840](#wisselink_2010_pmid_20816840)

- [Xu_2010_PMID_20228058](#xu_2010_pmid_20228058)

- [Yasokawa_2010_PMID_19499198](#yasokawa_2010_pmid_19499198)

- [Zhang_2010_PMID_20545859](#zhang_2010_pmid_20545859)

- [Zheng_2010_PMID_20237471](#zheng_2010_pmid_20237471)

- [van_Wageningen_2010_PMID_21145464](#van_wageningen_2010_pmid_21145464)

- [Ambroset_2011_PMID_22384338](#ambroset_2011_pmid_22384338)

- [Aparicio_2011_PMID_22087310](#aparicio_2011_pmid_22087310)

- [Arribere_2011_PMID_22152478](#arribere_2011_pmid_22152478)

- [Boender_2011_PMID_21803078](#boender_2011_pmid_21803078)

- [Boender_2011_PMID_22093745](#boender_2011_pmid_22093745)

- [Burrill_2011_PMID_21363961](#burrill_2011_pmid_21363961)

- [Casado_2011_PMID_21749328](#casado_2011_pmid_21749328)

- [Celona_2011_PMID_21738444](#celona_2011_pmid_21738444)

- [Chiu_2011_PMID_21147769](#chiu_2011_pmid_21147769)

- [Cocklin_2011_PMID_21196523](#cocklin_2011_pmid_21196523)

- [Dori-Bachash_2011_PMID_21811398](#dori-bachash_2011_pmid_21811398)

- [Dymond_2011_PMID_21918511](#dymond_2011_pmid_21918511)

- [Edskes_2011_PMID_21368275](#edskes_2011_pmid_21368275)

- [Eser_2011_PMID_21855792](#eser_2011_pmid_21855792)

- [Fowler_2011_PMID_21085730](#fowler_2011_pmid_21085730)

- [Gil_2011_PMID_21842488](#gil_2011_pmid_21842488)

- [Gomez-Gonzalez_2011_PMID_21701562](#gomez-gonzalez_2011_pmid_21701562)

- [Gonzalez-Aguilera_2011_PMID_21680710](#gonzalez-aguilera_2011_pmid_21680710)

- [Hong_2011_PMID_21715660](#hong_2011_pmid_21715660)

- [Jimeno_2011_PMID_21149575](#jimeno_2011_pmid_21149575)

- [Lenstra_2011_PMID_21596317](#lenstra_2011_pmid_21596317)

- [Levy_2011_PMID_22158820](#levy_2011_pmid_22158820)

- [Livas_2011_PMID_21827659](#livas_2011_pmid_21827659)

- [Ouyang_2011_PMID_20971184](#ouyang_2011_pmid_20971184)

- [Park_2011_PMID_21269496](#park_2011_pmid_21269496)

- [Parreiras_2011_PMID_21856932](#parreiras_2011_pmid_21856932)

- [Ragni_2011_PMID_21320323](#ragni_2011_pmid_21320323)

- [Rintala_2011_PMID_21348598](#rintala_2011_pmid_21348598)

- [Rossmann_2011_PMID_21474074](#rossmann_2011_pmid_21474074)

- [Ruenwai_2011_PMID_21184438](#ruenwai_2011_pmid_21184438)

- [Sen_2011_PMID_21835776](#sen_2011_pmid_21835776)

- [Shalem_2011_PMID_21931566](#shalem_2011_pmid_21931566)

- [Suzuki_2011_PMID_21648421](#suzuki_2011_pmid_21648421)

- [Takahashi_2011_PMID_21474073](#takahashi_2011_pmid_21474073)

- [Tauber_2011_PMID_21044956](#tauber_2011_pmid_21044956)

- [Thibault_2011_PMID_22143797](#thibault_2011_pmid_22143797)

- [Tu_2011_PMID_21469193](#tu_2011_pmid_21469193)

- [Uluisik_2011_PMID_22114689](#uluisik_2011_pmid_22114689)

- [Verzijlbergen_2011_PMID_21998594](#verzijlbergen_2011_pmid_21998594)

- [Wan_2011_PMID_21763693](#wan_2011_pmid_21763693)

- [Wenger_2011_PMID_21829391](#wenger_2011_pmid_21829391)

- [Zhang_2011_PMID_22068328](#zhang_2011_pmid_22068328)

- [Zhou_2011_PMID_21700227](#zhou_2011_pmid_21700227)

- [de_Kok_2011_PMID_21684346](#de_kok_2011_pmid_21684346)

- [van_Leeuwen_2011_PMID_21724882](#van_leeuwen_2011_pmid_21724882)

- [Albulescu_2012_PMID_22479188](#albulescu_2012_pmid_22479188)

- [Ansari_2012_PMID_21971086](#ansari_2012_pmid_21971086)

- [Apweiler_2012_PMID_22697265](#apweiler_2012_pmid_22697265)

- [Arana_2012_PMID_23079308](#arana_2012_pmid_23079308)

- [Barreto_2012_PMID_23039231](#barreto_2012_pmid_23039231)

- [Bester_2012_PMID_22384390](#bester_2012_pmid_22384390)

- [Bose_2012_PMID_22719263](#bose_2012_pmid_22719263)

- [Busti_2012_PMID_21964263](#busti_2012_pmid_21964263)

- [Cain_2012_PMID_22095082](#cain_2012_pmid_22095082)

- [Cap_2012_PMID_22560924](#cap_2012_pmid_22560924)

- [Carter_2012_PMID_23071457](#carter_2012_pmid_23071457)

- [Casamayor_2012_PMID_22372618](#casamayor_2012_pmid_22372618)

- [Celton_2012_PMID_22805527](#celton_2012_pmid_22805527)

- [Chin_2012_PMID_22289124](#chin_2012_pmid_22289124)

- [Chumnanpuen_2012_PMID_22622761](#chumnanpuen_2012_pmid_22622761)

- [Costelloe_2012_PMID_22960744](#costelloe_2012_pmid_22960744)

- [Dominissini_2012_PMID_22575960](#dominissini_2012_pmid_22575960)

- [Flom_2012_PMID_22461145](#flom_2012_pmid_22461145)

- [Garcia-Rodriguez_2012_PMID_22493290](#garcia-rodriguez_2012_pmid_22493290)

- [Geijer_2012_PMID_23066959](#geijer_2012_pmid_23066959)

- [Guan_2012_PMID_22851651](#guan_2012_pmid_22851651)

- [Haitani_2012_PMID_22841865](#haitani_2012_pmid_22841865)

- [Hao_2012_PMID_22179789](#hao_2012_pmid_22179789)

- [Hong_2012_PMID_22904057](#hong_2012_pmid_22904057)

- [Jaime_2012_PMID_22727066](#jaime_2012_pmid_22727066)

- [Joshi_2012_PMID_22718977](#joshi_2012_pmid_22718977)

- [Lanza_2012_PMID_22558379](#lanza_2012_pmid_22558379)

- [Lewis_2012_PMID_23275883](#lewis_2012_pmid_23275883)

- [Lin_2012_PMID_22308403](#lin_2012_pmid_22308403)

- [Marquina_2012_PMID_22750472](#marquina_2012_pmid_22750472)

- [Nadal-Ribelles_2012_PMID_23158682](#nadal-ribelles_2012_pmid_23158682)

- [Navlakha_2012_PMID_22916002](#navlakha_2012_pmid_22916002)

- [Niles_2012_PMID_22307609](#niles_2012_pmid_22307609)

- [Ochi_2012_PMID_22679107](#ochi_2012_pmid_22679107)

- [Oromendia_2012_PMID_23222101](#oromendia_2012_pmid_23222101)

- [Orozco_2012_PMID_22327582](#orozco_2012_pmid_22327582)

- [Pedersen_2012_PMID_23284296](#pedersen_2012_pmid_23284296)

- [Penacho_2012_PMID_22065482](#penacho_2012_pmid_22065482)

- [Prunuske_2012_PMID_22203981](#prunuske_2012_pmid_22203981)

- [Radman-Livaja_2012_PMID_22807688](#radman-livaja_2012_pmid_22807688)

- [Rupwate_2012_PMID_22673525](#rupwate_2012_pmid_22673525)

- [Sanz_2012_PMID_22621902](#sanz_2012_pmid_22621902)

- [Sheltzer_2012_PMID_22802626](#sheltzer_2012_pmid_22802626)

- [Simmons_Kovacs_2012_PMID_22306294](#simmons_kovacs_2012_pmid_22306294)

- [Spedale_2012_PMID_21976730](#spedale_2012_pmid_21976730)

- [Suzuki_2012_PMID_22897823](#suzuki_2012_pmid_22897823)

- [Thibault_2012_PMID_23000174](#thibault_2012_pmid_23000174)

- [Travesa_2012_PMID_22333915](#travesa_2012_pmid_22333915)

- [Tyo_2012_PMID_22380681](#tyo_2012_pmid_22380681)

- [Vizoso-Vazquez_2012_PMID_22189861](#vizoso-vazquez_2012_pmid_22189861)

- [Voordeckers_2012_PMID_22882838](#voordeckers_2012_pmid_22882838)

- [Wuttke_2012_PMID_22912585](#wuttke_2012_pmid_22912585)

- [Yona_2012_PMID_23197825](#yona_2012_pmid_23197825)

- [de_Castro_2012_PMID_23092287](#de_castro_2012_pmid_23092287)

- [de_Kok_2012_PMID_22257278](#de_kok_2012_pmid_22257278)

- [Alkim_2013_PMID_23864114](#alkim_2013_pmid_23864114)

- [Almario_2013_PMID_23613173](#almario_2013_pmid_23613173)

- [Aow_2013_PMID_23558744](#aow_2013_pmid_23558744)

- [Aristizabal_2013_PMID_24009531](#aristizabal_2013_pmid_24009531)

- [Ask_2013_PMID_24341320](#ask_2013_pmid_24341320)

- [Batista-Nascimento_2013_PMID_23295455](#batista-nascimento_2013_pmid_23295455)

- [Bessonov_2013_PMID_24130853](#bessonov_2013_pmid_24130853)

- [Braberg_2013_PMID_23932120](#braberg_2013_pmid_23932120)

- [Braun_2013_PMID_23207903](#braun_2013_pmid_23207903)

- [Brion_2013_PMID_24094006](#brion_2013_pmid_24094006)

- [Brown_2013_PMID_23360418](#brown_2013_pmid_23360418)

- [Buhligen_2013_PMID_24060829](#buhligen_2013_pmid_24060829)

- [Byeon_2013_PMID_23779104](#byeon_2013_pmid_23779104)

- [Cai_2013_PMID_24386402](#cai_2013_pmid_24386402)

- [Chang_2013_PMID_23660188](#chang_2013_pmid_23660188)

- [Chang_2013_PMID_24051094](#chang_2013_pmid_24051094)

- [Chen_2013_PMID_23124522](#chen_2013_pmid_23124522)

- [Chu_2013_PMID_23979602](#chu_2013_pmid_23979602)

- [Chumnanpuen_2013_PMID_24456840](#chumnanpuen_2013_pmid_24456840)

- [Cook_2013_PMID_23826764](#cook_2013_pmid_23826764)

- [Garcia-Oliver_2013_PMID_23599000](#garcia-oliver_2013_pmid_23599000)

- [Geijer_2013_PMID_23758973](#geijer_2013_pmid_23758973)

- [Gitter_2013_PMID_23064748](#gitter_2013_pmid_23064748)

- [Gonzalez_2013_PMID_23704987](#gonzalez_2013_pmid_23704987)

- [Hong_2013_PMID_23376593](#hong_2013_pmid_23376593)

- [Jaehnig_2013_PMID_23810556](#jaehnig_2013_pmid_23810556)

- [Judeh_2013_PMID_23268448](#judeh_2013_pmid_23268448)

- [Kucukgoze_2013_PMID_23992612](#kucukgoze_2013_pmid_23992612)

- [Lai_2013_PMID_23893744](#lai_2013_pmid_23893744)

- [Larsson_2013_PMID_23991176](#larsson_2013_pmid_23991176)

- [Lenstra_2013_PMID_24324601](#lenstra_2013_pmid_24324601)

- [Liang_2013_PMID_23985319](#liang_2013_pmid_23985319)

- [Ling_2013_PMID_23826995](#ling_2013_pmid_23826995)

- [Liu_2013_PMID_23392765](#liu_2013_pmid_23392765)

- [Liu_2013_PMID_23435897](#liu_2013_pmid_23435897)

- [Liu_2013_PMID_23725375](#liu_2013_pmid_23725375)

- [Lopez-Serra_2013_PMID_23219725](#lopez-serra_2013_pmid_23219725)

- [Mendes_2013_PMID_23872557](#mendes_2013_pmid_23872557)

- [Montefusco_2013_PMID_24170935](#montefusco_2013_pmid_24170935)

- [Moreno-Cermeno_2013_PMID_24100161](#moreno-cermeno_2013_pmid_24100161)

- [Ng_2013_PMID_24037263](#ng_2013_pmid_24037263)

- [O'Doherty_2013_PMID_24074273](#o'doherty_2013_pmid_24074273)

- [Park_2013_PMID_24349523](#park_2013_pmid_24349523)

- [Rachfall_2013_PMID_23071099](#rachfall_2013_pmid_23071099)

- [Reiter_2013_PMID_23275436](#reiter_2013_pmid_23275436)

- [Roy_2013_PMID_23640720](#roy_2013_pmid_23640720)

- [Santos-Pereira_2013_PMID_24240235](#santos-pereira_2013_pmid_24240235)

- [Schuller_2013_PMID_23576094](#schuller_2013_pmid_23576094)

- [Sha_2013_PMID_24073228](#sha_2013_pmid_24073228)

- [Suzuki_2013_PMID_23325148](#suzuki_2013_pmid_23325148)

- [Tesniere_2013_PMID_23658613](#tesniere_2013_pmid_23658613)

- [Van_de_Vosse_2013_PMID_23452847](#van_de_vosse_2013_pmid_23452847)

- [Worley_2013_PMID_23643537](#worley_2013_pmid_23643537)

- [Xue-Franzen_2013_PMID_23865462](#xue-franzen_2013_pmid_23865462)

- [van_de_Pasch_2013_PMID_23785440](#van_de_pasch_2013_pmid_23785440)

- [Bisschops_2014_PMID_24487068](#bisschops_2014_pmid_24487068)

- [Bloom-Ackermann_2014_PMID_24453985](#bloom-ackermann_2014_pmid_24453985)

- [Brice_2014_PMID_24334661](#brice_2014_pmid_24334661)

- [Bristow_2014_PMID_25200947](#bristow_2014_pmid_25200947)

- [Chasman_2014_PMID_25411400](#chasman_2014_pmid_25411400)

- [Chiasson_2014_PMID_24707045](#chiasson_2014_pmid_24707045)

- [Drozdova_2014_PMID_25227157](#drozdova_2014_pmid_25227157)

- [Gaillard_2014_PMID_24603480](#gaillard_2014_pmid_24603480)

- [Gamero_2014_PMID_24854353](#gamero_2014_pmid_24854353)

- [Gil_2014_PMID_24835131](#gil_2014_pmid_24835131)

- [Gopinath_2014_PMID_25316598](#gopinath_2014_pmid_25316598)

- [Hackmann_2014_PMID_24452287](#hackmann_2014_pmid_24452287)

- [Hasin_2014_PMID_24628813](#hasin_2014_pmid_24628813)

- [Hasunuma_2014_PMID_23916856](#hasunuma_2014_pmid_23916856)

- [Hebly_2014_PMID_24814792](#hebly_2014_pmid_24814792)

- [Herrera-Moyano_2014_PMID_24636987](#herrera-moyano_2014_pmid_24636987)

- [Hou_2014_PMID_24237754](#hou_2014_pmid_24237754)

- [Hughes_Hallett_2014_PMID_25085507](#hughes_hallett_2014_pmid_25085507)

- [Johnson_2014_PMID_24830393](#johnson_2014_pmid_24830393)

- [Kamei_2014_PMID_25294875](#kamei_2014_pmid_25294875)

- [Kang_2014_PMID_24206186](#kang_2014_pmid_24206186)

- [Kemmeren_2014_PMID_24766815](#kemmeren_2014_pmid_24766815)

- [Kozak_2014_PMID_24269999](#kozak_2014_pmid_24269999)

- [Landry_2014_PMID_24714560](#landry_2014_pmid_24714560)

- [Lewis_2014_PMID_24970865](#lewis_2014_pmid_24970865)

- [Licursi_2014_PMID_24164706](#licursi_2014_pmid_24164706)

- [Lindgren_2014_PMID_25483075](#lindgren_2014_pmid_25483075)

- [Lopez-Serra_2014_PMID_25173104](#lopez-serra_2014_pmid_25173104)

- [Lu_2014_PMID_24631914](#lu_2014_pmid_24631914)

- [Macadangdang_2014_PMID_24939988](#macadangdang_2014_pmid_24939988)

- [Merrill_2014_PMID_24938287](#merrill_2014_pmid_24938287)

- [Miller-Fleming_2014_PMID_24706893](#miller-fleming_2014_pmid_24706893)

- [Mojzita_2014_PMID_25192596](#mojzita_2014_pmid_25192596)

- [Nagarajan_2014_PMID_24706810](#nagarajan_2014_pmid_24706810)

- [Nakamura_2014_PMID_25177541](#nakamura_2014_pmid_25177541)

- [Nasuno_2014_PMID_25423296](#nasuno_2014_pmid_25423296)

- [O'Duibhir_2014_PMID_24952590](#o'duibhir_2014_pmid_24952590)

- [Reyes_2014_PMID_24262517](#reyes_2014_pmid_24262517)

- [Rodriguez-Lombardero_2014_PMID_24763424](#rodriguez-lombardero_2014_pmid_24763424)

- [Romagnoli_2014_PMID_24733517](#romagnoli_2014_pmid_24733517)

- [Saint_2014_PMID_24550006](#saint_2014_pmid_24550006)

- [Santos-Pereira_2014_PMID_25294824](#santos-pereira_2014_pmid_25294824)

- [Slavov_2014_PMID_24767987](#slavov_2014_pmid_24767987)

- [Stovicek_2014_PMID_24533484](#stovicek_2014_pmid_24533484)

- [Tsang_2014_PMID_24647101](#tsang_2014_pmid_24647101)

- [Wang_2014_PMID_24682482](#wang_2014_pmid_24682482)

- [Yamane-Sando_2014_PMID_24510621](#yamane-sando_2014_pmid_24510621)

- [Barbosa_2015_PMID_25884705](#barbosa_2015_pmid_25884705)

- [Barbosa_2015_PMID_26314747](#barbosa_2015_pmid_26314747)

- [Buck_2015_PMID_25759377](#buck_2015_pmid_25759377)

- [Canadell_2015_PMID_25425491](#canadell_2015_pmid_25425491)

- [Cary_2015_PMID_26392412](#cary_2015_pmid_26392412)

- [Chujo_2015_PMID_25304510](#chujo_2015_pmid_25304510)

- [Crosas_2015_PMID_25715111](#crosas_2015_pmid_25715111)

- [Felipe-Abrio_2015_PMID_25452497](#felipe-abrio_2015_pmid_25452497)

- [Fernandez-Acero_2015_PMID_26261079](#fernandez-acero_2015_pmid_26261079)

- [Gamero_2015_PMID_26336982](#gamero_2015_pmid_26336982)

- [Garca-Rodrguez_2015_PMID_25713143](#garca-rodrguez_2015_pmid_25713143)

- [Golla_2015_PMID_25919230](#golla_2015_pmid_25919230)

- [Gomar-Alba_2015_PMID_26048894](#gomar-alba_2015_pmid_26048894)

- [Goossens_2015_PMID_25873380](#goossens_2015_pmid_25873380)

- [Hodges_2015_PMID_25971662](#hodges_2015_pmid_25971662)

- [Hoffman_2015_PMID_25609572](#hoffman_2015_pmid_25609572)

- [Jung_2015_PMID_26602082](#jung_2015_pmid_26602082)

- [Khakhina_2015_PMID_25724885](#khakhina_2015_pmid_25724885)

- [Klecker_2015_PMID_26669658](#klecker_2015_pmid_26669658)

- [Lopez-Malo_2015_PMID_26194190](#lopez-malo_2015_pmid_26194190)

- [Mith_2015_PMID_25755096](#mith_2015_pmid_25755096)

- [Noble_2015_PMID_25947166](#noble_2015_pmid_25947166)

- [Oliveira_2015_PMID_25888284](#oliveira_2015_pmid_25888284)

- [Pastor-Flores_2015_PMID_26391581](#pastor-flores_2015_pmid_26391581)

- [Renaud-Young_2015_PMID_25701288](#renaud-young_2015_pmid_25701288)

- [Robinson_2015_PMID_26130709](#robinson_2015_pmid_26130709)

- [Sameith_2015_PMID_26700642](#sameith_2015_pmid_26700642)

- [Schosserer_2015_PMID_25635753](#schosserer_2015_pmid_25635753)

- [Suzuki_2015_PMID_25609182](#suzuki_2015_pmid_25609182)

- [Thierry_2015_PMID_25635677](#thierry_2015_pmid_25635677)

- [Vos_2015_PMID_26369953](#vos_2015_pmid_26369953)

- [Wanichthanarak_2015_PMID_26079307](#wanichthanarak_2015_pmid_26079307)

- [Zhang_2015_PMID_25897127](#zhang_2015_pmid_25897127)

- [Zhou_2015_PMID_26150418](#zhou_2015_pmid_26150418)

- [Zhu_2015_PMID_26138482](#zhu_2015_pmid_26138482)

- [Alff-Tuomala_2016_PMID_26454869](#alff-tuomala_2016_pmid_26454869)

- [Belanger_2016_PMID_26888869](#belanger_2016_pmid_26888869)

- [Dorfel_2016_PMID_27668839](#dorfel_2016_pmid_27668839)

- [Garcia-Martinez_2016_PMID_26717982](#garcia-martinez_2016_pmid_26717982)

- [Garcia-Martinez_2016_PMID_27648972](#garcia-martinez_2016_pmid_27648972)

- [Garrido-Godino_2016_PMID_27001033](#garrido-godino_2016_pmid_27001033)

- [Gavalda_2016_PMID_27035147](#gavalda_2016_pmid_27035147)

- [Golla_2016_PMID_27734932](#golla_2016_pmid_27734932)

- [Hinkle_2016_PMID_26606276](#hinkle_2016_pmid_26606276)

- [Kugler_2016_PMID_27245696](#kugler_2016_pmid_27245696)

- [Lu_2016_PMID_27812096](#lu_2016_pmid_27812096)

- [Marsit_2016_PMID_26549518](#marsit_2016_pmid_26549518)

- [Navarro-Tapia_2016_PMID_26925053](#navarro-tapia_2016_pmid_26925053)

- [Oeser_2016_PMID_26800527](#oeser_2016_pmid_26800527)

- [Ogawa_2016_PMID_27698120](#ogawa_2016_pmid_27698120)

- [Rollero_2016_PMID_26861624](#rollero_2016_pmid_26861624)

- [Sariki_2016_PMID_27718307](#sariki_2016_pmid_27718307)

- [Tarrant_2016_PMID_27807005](#tarrant_2016_pmid_27807005)

- [Yadav_2016_PMID_27222810](#yadav_2016_pmid_27222810)

- [Zeng_2016_PMID_27485516](#zeng_2016_pmid_27485516)

- [Baptista_2017_PMID_28918903](#baptista_2017_pmid_28918903)

- [Celaj_2017_PMID_28705884](#celaj_2017_pmid_28705884)

- [Chen_2017_PMID_27387517](#chen_2017_pmid_27387517)

- [Chen_2017_PMID_28798664](#chen_2017_pmid_28798664)

- [Chirichella_2017_PMID_28092690](#chirichella_2017_pmid_28092690)

- [Cho_2017_PMID_28934013](#cho_2017_pmid_28934013)

- [Choi_2017_PMID_28329151](#choi_2017_pmid_28329151)

- [Deshpande_2017_PMID_28435838](#deshpande_2017_pmid_28435838)

- [Driscoll_2017_PMID_28560281](#driscoll_2017_pmid_28560281)

- [Duc_2017_PMID_28922393](#duc_2017_pmid_28922393)

- [Golla_2017_PMID_29228701](#golla_2017_pmid_29228701)

- [Klopf_2017_PMID_28025392](#klopf_2017_pmid_28025392)

- [Plateau_2017_PMID_28303947](#plateau_2017_pmid_28303947)

- [Schiavone_2017_PMID_29085340](#schiavone_2017_pmid_29085340)

- [Tai_2017_PMID_28495531](#tai_2017_pmid_28495531)

- [Warfield_2017_PMID_28918900](#warfield_2017_pmid_28918900)

- [Yeh_2017_PMID_28934473](#yeh_2017_pmid_28934473)

- [Zeng_2017_PMID_28004152](#zeng_2017_pmid_28004152)

- [Zhu_2017_PMID_28499955](#zhu_2017_pmid_28499955)

- [de_Jonge_2017_PMID_27979920](#de_jonge_2017_pmid_27979920)

- [Chia_2018_PMID_30206180](#chia_2018_pmid_30206180)

- [Gil_2018_PMID_29611082](#gil_2018_pmid_29611082)

- [Ahmed_2019_PMID_31144305](#ahmed_2019_pmid_31144305)

- [Blasco-Moreno_2019_PMID_30899024](#blasco-moreno_2019_pmid_30899024)

- [Cho_2019_PMID_30668223](#cho_2019_pmid_30668223)

- [Farrugia_2019_PMID_30992471](#farrugia_2019_pmid_30992471)

- [Gaillard_2019_PMID_30715474](#gaillard_2019_pmid_30715474)

- [Hopfler_2019_PMID_31015336](#hopfler_2019_pmid_31015336)

- [Kunkel_2019_PMID_31395866](#kunkel_2019_pmid_31395866)

- [Lien_2019_PMID_30973945](#lien_2019_pmid_30973945)

- [Moreno_2019_PMID_31439886](#moreno_2019_pmid_31439886)

- [Pagano_2019_PMID_30986968](#pagano_2019_pmid_30986968)

- [Romero_2019_PMID_31394264](#romero_2019_pmid_31394264)

- [Sunyer-Figueres_2020_PMID_33019712](#sunyer-figueres_2020_pmid_33019712)

- [Bao_2021_PMID_34520050](#bao_2021_pmid_34520050)

- [Garcia-Martinez_2021_PMID_33258404](#garcia-martinez_2021_pmid_33258404)

- [Klim_2021_PMID_34039270](#klim_2021_pmid_34039270)

- [Alhaj_Sulaiman_2022_PMID_35705668](#alhaj_sulaiman_2022_pmid_35705668)

- [Chattopadhyay_2022_PMID_36418294](#chattopadhyay_2022_pmid_36418294)

- [Godillot_2022_PMID_35273585](#godillot_2022_pmid_35273585)

- [Jimenez-Gutierrez_2022_PMID_36232332](#jimenez-gutierrez_2022_pmid_36232332)

- [Lu_2022_PMID_36066422](#lu_2022_pmid_36066422)

- [Moreno_2022_PMID_35017613](#moreno_2022_pmid_35017613)

- [Revilleza_2022_PMID_35522675](#revilleza_2022_pmid_35522675)

- [Wu_2022_PMID_36419096](#wu_2022_pmid_36419096)

- [Kamata_2023_PMID_37189367](#kamata_2023_pmid_37189367)

- [Park_2023_PMID_37639444](#park_2023_pmid_37639444)


---


# DeRisi_1997_PMID_9381177

Citation: DeRisi JL, Iyer VR, Brown PO. 1997. Exploring the metabolic and genetic control of gene 
expression on a genomic scale.

Full Description: DNA microarrays containing virtually every gene of Saccharomyces cerevisiae were used to 
carry out a comprehensive investigation of the temporal program of gene expression 
accompanying the metabolic shift from fermentation to respiration. The expression profiles
observed for genes with known metabolic functions pointed to features of the metabolic 
reprogramming that occur during the diauxic shift, and the expression patterns of many 
previously uncharacterized genes provided clues to their possible functions. The same DNA 
microarrays were also used to identify genes whose expression was affected by deletion of 
the transcriptional co-repressor TUP1 or overexpression of the transcriptional activator 
YAP1. These results demonstrate the feasibility and utility of this approach to genomewide
exploration of gene expression patterns.

PMID: 9381177

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.diauxic.pcl	Expression during the diauxic shift, 20.5 hr time course	7	carbon utilization|diauxic shift|fermentation|respiration	2

File last modified: Thu Apr 25 2024

---


# Cho_1998_PMID_9702192

Citation: Cho RJ, Campbell MJ, Winzeler EA, Steinmetz L, Conway A, Wodicka L, Wolfsberg TG, 
Gabrielian AE, Landsman D, Lockhart DJ, Davis RW. 1998. A genome-wide transcriptional 
analysis of the mitotic cell cycle.

Full Description: Cell-cycle arrested yeast cells are synchronized by being held at a restrictive 
temperature and then released at t=0. RNA expression via Affymetrix chips is measured 
every 10 minutes until t=160. Data up to t=40 reflects temperature-induced as well as cell
-cycle effects.

PMID: 9702192

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Cho98.filter.flt.knn.avg.div.log.pcl	Mitotic cell cycle	17	mitotic cell cycle	2

File last modified: Thu Apr 25 2024

---


# Chu_1998_PMID_9784122

Citation: Chu S, DeRisi J, Eisen M, Mulholland J, Botstein D, Brown PO, Herskowitz I. 1998. The 
transcriptional program of sporulation in budding yeast.

Full Description: Samples taken during sporulation up to 11 hours.

PMID: 9784122

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Chu98.flt.knn.avg.pcl	Expression during sporulation, 11 hr time course	7	sporulation	2

File last modified: Thu Apr 25 2024

---


# Marton_1998_PMID_9809554

Citation: Marton MJ, DeRisi JL, Bennett HA, Iyer VR, Meyer MR, Roberts CJ, Stoughton R, Burchard J, 
Slade D, Dai H, Bassett DE Jr, Hartwell LH, Brown PO, Friend SH. 1998. Drug target 
validation and identification of secondary drug target effects using DNA microarrays.

Full Description: Treatment of yeast mutant strains defective in calcineurin, immunophilins or other genes 
with the immunosuppressants cyclosporin A or FK506.

PMID: 9809554

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Marton98.log.flt.knn.avg.pcl	Immunosuppressant response	7	chemical stimulus|protein dephosphorylation|signaling	2

File last modified: Thu Apr 25 2024

---


# Spellman_1998_PMID_9843569

Citation: Spellman PT, Sherlock G, Zhang MQ, Iyer VR, Anders K, Eisen MB, Brown PO, Botstein D, 
Futcher B. 1998. Comprehensive identification of cell cycle-regulated genes of the yeast 
Saccharomyces cerevisiae by microarray hybridization.

Full Description: Culture synchronized in telophase by arrest of cdc15 temperature-sensitive mutant. Samples
taken during the course of almost three full cell cycles.

PMID: 9843569

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Spellman98_cdc15.flt.knn.avg.pcl	Cell cycle, cdc15 block-release	25	mitotic cell cycle	2

File last modified: Thu Apr 25 2024
2010.Spellman98_elutriation.flt.knn.avg.pcl	Cell cycle, elutriation	14	mitotic cell cycle	2

File last modified: Thu Apr 25 2024
2010.Spellman98_alphaFactor.flt.knn.avg.pcl	Cell cycle, alpha-factor block-release	16	mitotic cell cycle	2

File last modified: Thu Apr 25 2024

---


# Ferea_1999_PMID_10449761

Citation: Ferea TL, Botstein D, Brown PO, Rosenzweig RF. 1999. Systematic changes in gene expression
patterns following adaptive evolution in yeast.

Full Description: Culturing a population of Saccharomyces cerevisiae for many generations under conditions 
to which it is not optimally adapted selects for fitter genetic variants. This simple 
experimental design provides a tractable model of adaptive evolution under natural 
selection. Beginning with a clonal, founding population, independently evolved strains 
were obtained from three independent cultures after continuous aerobic growth in glucose-
limited chemostats for more than 250 generations. DNA microarrays were used to compare 
genome-wide patterns of gene expression in the evolved strains and the parental strain. 
Several hundred genes were found to have significantly altered expression in the evolved 
strains. Many of these genes showed similar alterations in their expression in all three 
evolved strains. Genes with altered expression in the three evolved strains included genes
involved in glycolysis, the tricarboxylic acid cycle, oxidative phosphorylation, and 
metabolite transport. These results are consistent with physiological observations and 
indicate that increased fitness is acquired by altering regulation of central metabolism 
such that less glucose is fermented and more glucose is completely oxidized.

PMID: 10449761

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.evolution.pcl	Systematic changes in gene expression patterns following adaptive evolution in yeast.	4	evolution	2

File last modified: Thu Apr 25 2024

---


# Galitski_1999_PMID_10398601

Citation: Galitski T, Saldanha AJ, Styles CA, Lander ES, Fink GR. 1999. Ploidy regulation of gene 
expression.

Full Description: Microarray-based gene expression analysis identified genes showing ploidy-dependent 
expression in isogenic Saccharomyces cerevisiae strains that varied in ploidy from haploid
to tetraploid. These genes were induced or repressed in proportion to the number of 
chromosome sets, regardless of the mating type. Ploidy-dependent repression of some G1 
cyclins can explain the greater cell size associated with higher ploidies, and suggests 
ploidy-dependent modifications of cell cycle progression. Moreover, ploidy regulation of 
the FLO11 gene had direct consequences for yeast development.

PMID: 10398601

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.ploidy.pcl	Ploidy regulation of gene expression	11	cell morphogenesis|mitotic cell cycle	2

File last modified: Thu Apr 25 2024

---


# Hardwick_1999_PMID_10611304

Citation: Hardwick JS, Kuruvilla FG, Tong JK, Shamji AF, Schreiber SL. 1999. Rapamycin-modulated 
transcription defines the subset of nutrient-sensitive signaling pathways directly 
controlled by the Tor proteins.

Full Description: We measured the immediate transcriptional response of yeast grown in rich media and 
treated with rapamycin to investigate the direct effects of Tor proteins on nutrient-
sensitive signaling pathways.

PMID: 10611304

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Hardwick00.flt.knn.avg.pcl	Rapamycin exposure	14	amino acid utilization|chemical stimulus|signaling	2

File last modified: Thu Apr 25 2024

---


# Madhani_1999_PMID_10535956

Citation: Madhani HD, Galitski T, Lander ES, Fink GR. 1999. Effectors of a developmental mitogen-
activated protein kinase cascade revealed by expression signatures of signaling mutants.

Full Description: Transcript levels in strains in which MAPK signaling was altered genetically.

PMID: 10535956

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Madhani99.filter.flt.knn.avg.div.log.pcl	MAPK mutants	11	filamentous growth|signaling	2

File last modified: Thu Apr 25 2024

---


# Wyrick_1999_PMID_10586882

Citation: Wyrick JJ, Holstege FC, Jennings EG, Causton HC, Shore D, Grunstein M, Lander ES, Young RA,
 1999. Chromosomal landscape of nucleosome-dependent gene expression and silencing in 
yeast.

Full Description: Eukaryotic genomes are packaged into nucleosomes, which are thought to repress gene 
expression generally. Repression is particularly evident at yeast telomeres, where genes 
within the telomeric heterochromatin appear to be silenced by the histone-binding silent 
information regulator (SIR) complex (Sir2, Sir3, Sir4) and Rap1 (refs 4-10). Here, to 
investigate how nucleosomes and silencing factors influence global gene expression, we use
high-density arrays to study the effects of depleting nucleosomal histones and silencing 
factors in yeast. Reducing nucleosome content by depleting histone H4 caused increased 
expression of 15% of genes and reduced expression of 10% of genes, but it had little 
effect on expression of the majority (75%) of yeast genes. Telomere-proximal genes were 
found to be de-repressed over regions extending 20 kilobases from the telomeres, well 
beyond the extent of Sir protein binding and the effects of loss of Sir function. These 
results indicate that histones make Sir-independent contributions to telomeric silencing, 
and that the role of histones located elsewhere in chromosomes is gene specific rather 
than generally repressive.

PMID: 10586882

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.histone.pcl	Expression in response to histone depletion	7	chromatin organization	2

File last modified: Thu Apr 25 2024

---


# Bernstein_2000_PMID_11095743

Citation: Bernstein BE, Tong JK, Schreiber SL. 2000. Genomewide studies of histone deacetylase 
function in yeast.

Full Description: Transcription profiles of the yeast deletion strains rpd3, sin3, sap30, ume6, hda1, hos2, 
and hos3.

PMID: 11095743

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Bernstein00_HDACrpd3sin3hda1.log.flt.knn.avg.pcl	Histone deacetylase (<i>rpd3</i>/ <i>sin3</i>/ <i>hda1</i> deletions)	6	chromatin organization|histone modification	2

File last modified: Thu Apr 25 2024
2010.Bernstein00_HDACsin3sap30ume6hda1hos2hos3.filter.flt.knn.avg.div.log.pcl	Histone deacetylase (<i>sin3</i>/ <i>sap30</i>/ <i>ume6</i>/ <i>hda1</i>/ <i>hos2</i>/ <i>hos3</i> deletions)	7	chromatin organization|histone modification	2

File last modified: Thu Apr 25 2024
2010.Bernstein00_TSA.filter.flt.knn.avg.div.log.pcl	Trichostatin A treatment time course	5	chromatin organization|histone modification	2

File last modified: Thu Apr 25 2024

---


# Gasch_2000_PMID_11102521

Citation: Gasch AP, Spellman PT, Kao CM, Carmel-Harel O, Eisen MB, Storz G, Botstein D, Brown PO. 
2000. Genomic expression programs in the response of yeast cells to environmental changes.

Full Description: Samples grown in YPD supplemented with 1 M sorbitol and collected at several time points 
up to 60 minutes.

PMID: 11102521

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Gasch00_hyper-osmotic.flt.knn.avg.pcl	Response to hyperosmotic shock	6	osmotic stress|stress	2

File last modified: Thu Apr 25 2024
2010.Gasch00_hypo-osmotic.flt.knn.avg.pcl	Response to hypoosmotic shock	5	osmotic stress|stress	2

File last modified: Thu Apr 25 2024
2010.Gasch00_menadione.flt.knn.avg.pcl	Response to menadione; 160 min time course	9	chemical stimulus|oxidative stress|stress	2

File last modified: Thu Apr 25 2024
2010.Gasch00_Ndepletion.flt.knn.avg.pcl	Response to nitrogen depletion, 3 day time course	9	nitrogen utilization|starvation	2

File last modified: Thu Apr 25 2024
2010.Gasch00_stationaryPhase(y12).flt.knn.avg.pcl	Response to stationary phase; 13 days (y13)	10	stationary phase entry|stress	2

File last modified: Thu Apr 25 2024
2010.Gasch00_stationaryPhase(y14).flt.knn.avg.pcl	Response to stationary phase; 5 days (y14)	9	stationary phase entry|stress	2

File last modified: Thu Apr 25 2024
2010.Gasch00_steadyState(y13).flt.knn.avg.pcl	Response to various temperatures at steady state (y13)	5	heat shock|response to unfolded protein|stress	2

File last modified: Thu Apr 25 2024
2010.Gasch00_steadyState(y14).flt.knn.avg.pcl	Response to various temperatures at steady state (y14)	8	heat shock|response to unfolded protein|stress	2

File last modified: Thu Apr 25 2024
2010.Gasch00_HS29-33.flt.knn.avg.pcl	Heat shock 29C to 33C, 20 min; 90 min time course	4	heat shock|response to unfolded protein|stress	2

File last modified: Thu Apr 25 2024
2010.Gasch00_HS30-37.flt.knn.avg.pcl	Heat shock 29C to 33C, 20 min; 60 min time course	5	heat shock|response to unfolded protein|stress	2

File last modified: Thu Apr 25 2024
2010.Gasch00_HS37-25.flt.knn.avg.pcl	Heat shock 37C to 25C, 20 min; 90 min time course	5	heat shock|response to unfolded protein|stress	2

File last modified: Thu Apr 25 2024
2010.Gasch00_HSmild.flt.knn.avg.pcl	Mild heat shock 29C to 33C with sorbitol (5, 10 and 30 min)	6	heat shock|response to unfolded protein|stress	2

File last modified: Thu Apr 25 2024
2010.Gasch00_HSto37.flt.knn.avg.pcl	Heat shock from various temperatures to 37C	6	heat shock|response to unfolded protein|stress	2

File last modified: Thu Apr 25 2024
2010.Gasch00_DTT(y13).flt.knn.avg.pcl	Response to dithiothreitol; 3 hr time course (y13)	8	chemical stimulus|response to unfolded protein|stress	2

File last modified: Thu Apr 25 2024
2010.Gasch00_DTT(y14).flt.knn.avg.pcl	Response to dithiothreitol; 8 hr time course (y14)	7	chemical stimulus|response to unfolded protein|stress	2

File last modified: Thu Apr 25 2024
2010.Gasch00_HOtimeCourse.flt.knn.avg.pcl	Response to hydrogen peroxide; 160 min time course	9	chemical stimulus|oxidative stress|stress	2

File last modified: Thu Apr 25 2024
2010.Gasch00_HS25-37.flt.knn.avg.pcl	Heat shock 25C to 37C, 20 min; 80 min time course	8	heat shock|response to unfolded protein|stress	2

File last modified: Thu Apr 25 2024
2010.Gasch00_carbonSources.flt.knn.avg.pcl	Response to different carbon sources	6	carbon utilization|stress	2

File last modified: Thu Apr 25 2024
2010.Gasch00_diamideTreatment.flt.knn.avg.pcl	Response to diamide; 90 min time course	8	chemical stimulus|oxidative stress|stress	2

File last modified: Thu Apr 25 2024
2010.Gasch00_adenineStarvation.flt.knn.avg.pcl	Response to amino acid and adenine starvation; 6 hr time course	5	amino acid utilization|starvation|stress	2

File last modified: Thu Apr 25 2024

---


# Gross_2000_PMID_10922376

Citation: Gross C, Kelleher M, Iyer VR, Brown PO, Winge DR. 2000. Identification of the copper 
regulon in Saccharomyces cerevisiae by DNA microarrays.

Full Description: Growth conditions of excess copper or copper deficiency, regulated by <i>Mac1</i> or <i>
Ace1</i> transcriptional activators.

PMID: 10922376

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Gross00.flt.knn.avg.pcl	Copper regulon	6	cellular ion homeostasis|metal or metalloid ion stress	2

File last modified: Thu Apr 25 2024

---


# Hughes_2000_PMID_10929718

Citation: Hughes TR, Marton MJ, Jones AR, Roberts CJ, Stoughton R, Armour CD, Bennett HA, Coffey E, 
Dai H, He YD, Kidd MJ, King AM, Meyer MR, Slade D, Lum PY, Stepaniants SB, Shoemaker DD, 
Gachotte D, Chakraburtty K, Simon J, Bard M, Friend SH. 2000. Functional discovery via a 
compendium of expression profiles.

Full Description: 300 knockout mutants create a compendium of expression profiles.

PMID: 10929718

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Hughes00.flt.knn.avg.pcl	Diverse knockout mutants	300	chemical stimulus	2

File last modified: Thu Apr 25 2024

---


# Lee_2000_PMID_12760044

Citation: Lee SE, Pellicioli A, Demeter J, Vaze MP, Gasch AP, Malkova A, Brown PO, Botstein D, 
Stearns T, Foiani M, Haber JE. 2000. Arrest, adaptation, and recovery following a 
chromosome double-strand break in Saccharomyces cerevisiae.

Full Description: _.

PMID: 12760044

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.HO_endonuclease.pcl	Response to creating a single unrepaired DSB by HO endonuclease in nocodazole-arrested cells	8	DNA damage stimulus	2

File last modified: Thu Apr 25 2024

---


# Lopez_2000_PMID_10940042

Citation: Lopez MC, Baker HV. 2000. Understanding the growth phenotype of the yeast gcr1 mutant in 
terms of global genomic expression patterns.

Full Description: The genomic expression patterns of wild-type and <i>gcr1</i> mutant yeast growing on 
various media, with and without glucose.

PMID: 10940042

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Lopez00.filter.flt.knn.avg.div.log.pcl	<i>gcr1</i> mutant, glucose exposure	17	carbon utilization|fermentation|respiration	2

File last modified: Thu Apr 25 2024

---


# Lyons_2000_PMID_10884426

Citation: Lyons TJ, Gasch AP, Gaither LA, Botstein D, Brown PO, Eide DJ. 2000. Genome-wide 
characterization of the Zap1p zinc-responsive regulon in yeast.

Full Description: The <i>Zap1p</i> transcription factor senses cellular zinc status and increases expression
of its target genes in response to zinc deficiency. Previously known <i>Zap1p</i>-
regulated genes encode the <i>Zrt1p</i>, <i>Zrt2p</i>, and <i>Zrt3p</i> zinc transporter 
genes and <i>Zap1p</i> itself. To allow the characterization of additional genes in yeast 
important for zinc homeostasis, a systematic study of gene expression on the genome-wide 
scale was used to identify other <i>Zap1p</i> target genes. Using a combination of DNA 
microarrays and a computer-assisted analysis of shared motifs in the promoters of 
similarly regulated genes, we identified 46 genes that are potentially regulated by <i>
Zap1p</i>. <i>Zap1p</i>-regulated expression of seven of these newly identified target 
genes was confirmed independently by using lacZ reporter fusions, suggesting that many of 
the remaining candidate genes are also <i>Zap1p</i> targets. Our studies demonstrate the 
efficacy of this combined approach to define the regulon of a specific eukaryotic 
transcription factor.

PMID: 10884426

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Lyons00.flt.knn.avg.pcl	Expression in response to varying zinc levels in wild type and zap1 null mutant	9	cellular ion homeostasis|metal or metalloid ion stress	2

File last modified: Thu Apr 25 2024

---


# Ogawa_2000_PMID_11102525

Citation: Ogawa N, DeRisi J, Brown PO. 2000. New components of a system for phosphate accumulation 
and polyphosphate metabolism in Saccharomyces cerevisiae revealed by genomic expression 
analysis.

Full Description: Modifying phosphate levels through mutation (<i>pho81c</i>, <i>pho85</i>, <i>pho80</i>, <i
>pho4c</i>) or chemical manipulation.

PMID: 11102525

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Ogawa00_set1.flt.knn.avg.pcl	Effect of modifying phosphate levels through mutation (PHO81c, pho85, pho80, PHO4c)	5	phosphorus utilization|signaling	2

File last modified: Thu Apr 25 2024
2010.Ogawa00_set2.flt.knn.avg.pcl	Effect of low vs high phosphate containing medium	3	phosphorus utilization|starvation	2

File last modified: Thu Apr 25 2024

---


# Primig_2000_PMID_11101837

Citation: Primig M, Williams RM, Winzeler EA, Tevzadze GG, Conway AR, Hwang SY, Davis RW, Esposito 
RE. 2000. The core meiotic transcriptome in budding yeasts.

Full Description: Meiotic expression patterns of two yeast strains, SK1 and W303, that display distinct 
kinetics and efficiencies of sporulation.

PMID: 11101837

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Primig00.filter.flt.knn.avg.div.log.pcl	Sporulation of two strains	24	sporulation	2

File last modified: Thu Apr 25 2024

---


# Roberts_2000_PMID_10657304

Citation: Roberts CJ, Nelson B, Marton MJ, Stoughton R, Meyer MR, Bennett HA, He YD, Dai H, Walker 
WL, Hughes TR, Tyers M, Boone C, Friend SH. 2000. Signaling and circuitry of multiple MAPK,
pathways revealed by a matrix of global gene expression profiles.

Full Description: Genome-wide transcript profiling was used to monitor signal transduction during yeast 
pheromone response. Genetic manipulations allowed analysis of changes in gene expression 
underlying pheromone signaling, cell cycle control, and polarized morphogenesis. A two-
dimensional hierarchical clustered matrix, covering 383 of the most highly regulated genes
, was constructed from 46 diverse experimental conditions. Diagnostic subsets of 
coexpressed genes reflected signaling activity, cross talk, and overlap of multiple 
mitogen-activated protein kinase (MAPK) pathways. Analysis of the profiles specified by 
two different MAPKs-Fus3p and Kss1p-revealed functional overlap of the filamentous growth 
and mating responses. Global transcript analysis reflects biological responses associated 
with the activation and perturbation of signal transduction pathways.

PMID: 10657304

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.alpha_time.pcl	Expression in response to alpha-factor (over time)	7	cell morphogenesis|mating|signaling	2

File last modified: Thu Apr 25 2024
2010.alpha_conc.pcl	Expression in response to alpha-factor (various concentrations)	8	cell morphogenesis|mating|signaling	2

File last modified: Thu Apr 25 2024

---


# Robertson_2000_PMID_10811893

Citation: Robertson LS, Causton HC, Young RA, Fink GR. 2000. The yeast A kinases differentially 
regulate iron uptake and respiratory function.

Full Description: Study of <i>TPK1</i> (alternate name for YJL164C), <i>TPK2</i> (alternate name for YPL203W
), and <i>TPK3</i> (YRL166C) mutants.

PMID: 10811893

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Robertson00.filter.flt.knn.avg.div.log.pcl	<i>TPK1</i>, <i>TPK2</i>, <i>TPK3</i> mutants	12	protein phosphorylation|signaling	2

File last modified: Thu Apr 25 2024

---


# Sudarsanam_2000_PMID_10725359

Citation: Sudarsanam P, Iyer VR, Brown PO, Winston F. 2000. Whole-genome expression analysis of snf/
swi mutants of Saccharomyces cerevisiae.

Full Description: The Saccharomyces cerevisiae Snf/Swi complex has been previously demonstrated to control 
transcription and chromatin structure of particular genes in vivo and to remodel 
nucleosomes in vitro. We have performed whole-genome expression analysis, using DNA 
microarrays, to study mutants deleted for a gene encoding one conserved (Snf2) or one 
unconserved (Swi1) Snf/Swi component. This analysis was performed on cells grown in both 
rich and minimal media. The microarray results, combined with Northern blot, computational
, and genetic analyses, show that snf2Delta and swi1Delta mutations cause similar effects 
on mRNA levels, that Snf/Swi controls some genes differently in rich and minimal media, 
and that Snf/Swi control is exerted at the level of individual genes rather than over 
larger chromosomal domains. In addition, this work shows that Snf/Swi controls mRNA levels
of MATalpha-specific genes, likely via controlling transcription of the regulators 
MATalpha1 and MCM1. Finally, we provide evidence that Snf/Swi acts both as an activator 
and as a repressor of transcription, and that neither mode of control is an indirect 
effect of the other.

PMID: 10725359

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.snf_swi_mutants.pcl	Expression analysis of snf/swi mutants	12	chromatin organization	2

File last modified: Thu Apr 25 2024

---


# Travers_2000_PMID_10847680

Citation: Travers KJ, Patil CK, Wodicka L, Lockhart DJ, Weissman JS, Walter P. 2000. Functional and 
genomic analyses reveal an essential coordination between the unfolded protein response 
and ER-associated degradation.

Full Description: The unfolded protein response (UPR) regulates gene expression in response to stress in the
endoplasmic reticulum (ER). We determined the transcriptional scope of the UPR using DNA 
microarrays. Rather than regulating only ER-resident chaperones and phospholipid 
biosynthesis, as anticipated from earlier work, the UPR affects multiple ER and secretory 
pathway functions. Studies of UPR targets engaged in ER-associated protein degradation (
ERAD) reveal an intimate coordination between these responses: efficient ERAD requires an 
intact UPR, and UPR induction increases ERAD capacity. Conversely, loss of ERAD leads to 
constitutive UPR induction. Finally, simultaneous loss of ERAD and the UPR greatly 
decreases cell viability. Thus, the UPR and ERAD are dynamic responses required for the 
coordinated disposal of misfolded proteins even in the absence of acute stress.

PMID: 10847680

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.unfolded_protein_response.pcl	Expression during the unfolded protein response	10	chemical stimulus|response to unfolded protein|stress	2

File last modified: Thu Apr 25 2024

---


# Yun_2000_PMID_10744769

Citation: Yun CW, Ferea T, Rashford J, Ardon O, Brown PO, Botstein D, Kaplan J, Philpott CC. 2000. 
Desferrioxamine-mediated iron uptake in Saccharomyces cerevisiae. Evidence for two 
pathways of iron uptake.

Full Description: In the yeast Saccharomyces cerevisiae, uptake of iron is largely regulated by the 
transcription factor <i>Aft1</i>. cDNA microarrays were used to identify new iron and <i>
AFT1</i>-regulated genes.

PMID: 10744769

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Yun00.flt.knn.avg.pcl	Iron uptake	4	cellular ion homeostasis|metal or metalloid ion stress	2

File last modified: Thu Apr 25 2024

---


# Zhu_2000_PMID_10894548

Citation: Zhu G, Spellman PT, Volpe T, Brown PO, Botstein D, Davis TN, Futcher B. 2000. Two yeast 
forkhead genes regulate the cell cycle and pseudohyphal growth.

Full Description: There are about 800 genes in Saccharomyces cerevisiae whose transcription is cell-cycle 
regulated. Some of these form clusters of co-regulated genes. The 'CLB2' cluster contains 
33 genes whose transcription peaks early in mitosis, including CLB1, CLB2, SWI5, ACE2, 
CDC5, CDC20 and other genes important for mitosis. Here we find that the genes in this 
cluster lose their cell cycle regulation in a mutant that lacks two forkhead transcription
factors, Fkh1 and Fkh2. Fkh2 protein is associated with the promoters of CLB2, SWI5 and 
other genes of the cluster. These results indicate that Fkh proteins are transcription 
factors for the CLB2 cluster. The fkh1 fkh2 mutant also displays aberrant regulation of 
the 'SIC1' cluster, whose member genes are expressed in the M-G1 interval and are involved
in mitotic exit. This aberrant regulation may be due to aberrant expression of the 
transcription factors Swi5 and Ace2, which are members of the CLB2 cluster and controllers
of the SIC1 cluster. Thus, a cascade of transcription factors operates late in the cell 
cycle. Finally, the fkh1 fkh2 mutant displays a constitutive pseudohyphal morphology, 
indicating that Fkh1 and Fkh2 may help control the switch to this mode of growth.

PMID: 10894548

GEO ID: GSE3182

PCL filename	short description	# conditions	tags	# channels
GSE3182_setA_family.pcl	Two yeast forkhead genes regulate the cell cycle and pseudohyphal growth.	26	cell cycle regulation|filamentous growth	2

File last modified: Thu Apr 25 2024

---


# Angus-Hill_2001_PMID_11336698

Citation: Angus-Hill ML, Schlichter A, Roberts D, Erdjument-Bromage H, Tempst P, Cairns BR. 2001. A 
Rsc3/Rsc30 zinc cluster dimer reveals novel roles for the chromatin remodeler RSC in gene 
expression and cell cycle control.

Full Description: DNA microarray analysis with <i>rsc3</i> or <i>rsc30</i> mutants reveals different effects
on the expression levels of ribosomal protein genes and cell wall genes.

PMID: 11336698

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Angus01.flt.knn.avg.pcl	<i>rsc3</i>/<i>rsc30</i> knockouts	8	chromatin organization	2

File last modified: Thu Apr 25 2024

---


# Backhus_2001_PMID_12702356

Citation: Backhus LE, DeRisi J, Bisson LF. 2001. Functional genomic analysis of a commercial wine 
strain of Saccharomyces cerevisiae under differing nitrogen conditions.

Full Description: DNA microarray analysis was used to profile gene expression in a commercial isolate of 
Saccharomyces cerevisiae grown in a synthetic grape juice medium under conditions 
mimicking a natural environment for yeast: High-sugar and variable nitrogen conditions. 
The high nitrogen condition displayed elevated levels of expression of genes involved in 
biosynthesis of macromolecular precursors across the time course as compared to low-
nitrogen. In contrast, expression of genes involved in translation and oxidative carbon 
metabolism were increased in the low-nitrogen condition, suggesting that respiration is 
more nitrogen-conserving than fermentation. Several genes under glucose repression control
were induced in low-nitrogen in spite of very high (17%) external glucose concentrations, 
but there was no general relief of glucose repression. Expression of many stress response 
genes was elevated in stationary phase. Some of these genes were expressed regardless of 
the nitrogen concentration while others were found at higher levels only under high 
nitrogen conditions. A few genes, FSP2, RGS2, AQY1, YFL030W, were expressed more strongly 
with nitrogen limitation as compared to other conditions.

PMID: 12702356

GEO ID: GSE4887

PCL filename	short description	# conditions	tags	# channels
GSE4887_setA_family.pcl	Functional genomic analysis of a commercial wine strain of Saccharomyces cerevisiae under differing nitrogen conditions.	6	nitrogen utilization|starvation	2

File last modified: Thu Apr 25 2024

---


# Baetz_2001_PMID_11533240

Citation: Baetz K, Moffat J, Haynes J, Chang M, Andrews B. 2001. Transcriptional coregulation by the
cell integrity mitogen-activated protein kinase Slt2 and the cell cycle regulator Swi4.

Full Description: Transcriptional coregulation by the cell integrity mitogen-activated protein kinase <i>
Slt2</i> and the cell cycle regulator <i>Swi4</i

>.

PMID: 11533240

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Baetz01.flt.knn.avg.pcl	<i>slt2</i>/ <i>swi4</i>/ <i>swi6</i>/ <i>bck1</i> knockouts	5	mitotic cell cycle|signaling	2

File last modified: Thu Apr 25 2024

---


# Bedalov_2001_PMID_11752457

Citation: Bedalov A, Gatbonton T, Irvine WP, Gottschling DE, Simon JA. 2001. Identification of a 
small molecule inhibitor of Sir2p.

Full Description: Identification of a small molecule inhibitor of <i>Sir2p</i

>.

PMID: 11752457

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Bedalov01.flt.knn.avg.pcl	Splitomicin exposure and <i>sir2</i> mutants	7	chemical stimulus|chromatin organization|histone modification	2

File last modified: Thu Apr 25 2024

---


# Carmel-Harel_2001_PMID_11169101

Citation: Carmel-Harel O, Stearman R, Gasch AP, Botstein D, Brown PO, Storz G. 2001. Role of 
thioredoxin reductase in the Yap1p-dependent response to oxidative stress in Saccharomyces
cerevisiae.

Full Description: Role of thioredoxin reductase in the <i>Yap1p</i>-dependent response to oxidative stress 
in Saccharomyces cerevisiae.

PMID: 11169101

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.CarmelHarel01.flt.knn.avg.pcl	H2O2 exposure to wt and Delta<i>trr1</i> knockout	15	chemical stimulus|oxidative stress|stress	2

File last modified: Thu Apr 25 2024

---


# Carroll_2001_PMID_11675494

Citation: Carroll AS, Bishop AC, DeRisi JL, Shokat KM, O'Shea EK. 2001. Chemical inhibition of the 
Pho85 cyclin-dependent kinase reveals a role in the environmental stress response.

Full Description: We used a chemical genetic approach that enabled us to control Pho85 kinase activity with 
a cell-permeable inhibitor and whole genome transcript profiling.

PMID: 11675494

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Carroll01.log.flt.knn.avg.pcl	<i>pho85</i> inhibition	12	protein phosphorylation|stress	2

File last modified: Thu Apr 25 2024

---


# Causton_2001_PMID_11179418

Citation: Causton HC, Ren B, Koh SS, Harbison CT, Kanin E, Jennings EG, Lee TI, True HL, Lander ES, 
Young RA. 2001. Remodeling of yeast genome expression in response to environmental changes.

Full Description: Response to various changes in extracellular environment, including changes in temperature
, oxidation, nutrients, pH, and osmolarity.

PMID: 11179418

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Causton01_peroxide.filter.flt.knn.avg.div.log.pcl	Peroxide response	7	chemical stimulus|oxidative stress|stress	2

File last modified: Thu Apr 25 2024
2010.Causton01_sorbitol.filter.flt.knn.avg.div.log.pcl	Sorbitol response	6	osmotic stress|stress	2

File last modified: Thu Apr 25 2024
2010.Causton01_acid.filter.flt.knn.avg.div.log.pcl	Acid response	11	chemical stimulus|stress	2

File last modified: Thu Apr 25 2024
2010.Causton01_alkali.filter.flt.knn.avg.div.log.pcl	Alkali response	8	chemical stimulus|stress	2

File last modified: Thu Apr 25 2024
2010.Causton01_heat.filter.flt.knn.avg.div.log.pcl	Heat response	7	heat shock|response to unfolded protein|stress	2

File last modified: Thu Apr 25 2024
2010.Causton01_NaCl.filter.flt.knn.avg.div.log.pcl	NaCl response	6	osmotic stress|stress	2

File last modified: Thu Apr 25 2024

---


# Epstein_2001_PMID_11179416

Citation: Epstein CB, Waddle JA, Hale W 4th, Dave V, Thornton J, Macatee TL, Garner HR, Butow RA. 
2001. Genome-wide responses to mitochondrial dysfunction.

Full Description: We used DNA microarrays to characterize the transcriptional responses to different 
mitochondrial perturbations.

PMID: 11179416

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Epstein00.flt.knn.avg.pcl	Mitochondrial dysfunction	11	carbon utilization|respiration	2

File last modified: Thu Apr 25 2024

---


# Gasch_2001_PMID_11598186

Citation: Gasch AP, Huang M, Metzner S, Botstein D, Elledge SJ, Brown PO. 2001. Genomic expression 
responses to DNA-damaging agents and the regulatory role of the yeast ATR homolog Mec1p.

Full Description: Eukaryotic cells respond to DNA damage by arresting the cell cycle and modulating gene 
expression to ensure efficient DNA repair. The human ATR kinase and its homolog in yeast, 
MEC1, play central roles in transducing the damage signal. To characterize the role of the
Mec1 pathway in modulating the cellular response to DNA damage, we used DNA microarrays to
observe genomic expression in Saccharomyces cerevisiae responding to two different DNA-
damaging agents. We compared the genome-wide expression patterns of wild-type cells and 
mutants defective in Mec1 signaling, including mec1, dun1, and crt1 mutants, under normal 
growth conditions and in response to the methylating-agent methylmethane sulfonate (MMS) 
and ionizing radiation. Here, we present a comparative analysis of wild-type and mutant 
cells responding to these DNA-damaging agents, and identify specific features of the gene 
expression responses that are dependent on the Mec1 pathway. Among the hundreds of genes 
whose expression was affected by Mec1p, one set of genes appears to represent an MEC1-
dependent expression signature of DNA damage. Other aspects of the genomic responses were 
independent of Mec1p, and likely independent of DNA damage, suggesting the pleiotropic 
effects of MMS and ionizing radiation. The complete data set as well as supplemental 
materials is available at http://www-genome.stanford.edu/mec1.

PMID: 11598186

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.DNAdamage.pcl	Expression in response to DNA-damaging agents and role of the Mec1 signaling pathway (mec1, dun1 and crt1)	52	DNA damage stimulus|radiation|signaling|stress	2

File last modified: Thu Apr 25 2024

---


# Ideker_2001_PMID_11340206

Citation: Ideker T, Thorsson V, Ranish JA, Christmas R, Buhler J, Eng JK, Bumgarner R, Goodlett DR, 
Aebersold R, Hood L. 2001. Integrated genomic and proteomic analyses of a systematically 
perturbed metabolic network.

Full Description: Systematic perturbations of the yeast galactose-utilization pathway.

PMID: 11340206

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Ideker01.flt.knn.avg.pcl	GAL mutants	21	carbon utilization	2

File last modified: Thu Apr 25 2024

---


# Keller_2001_PMID_11504737

Citation: Keller G, Ray E, Brown PO, Winge DR. 2001. Haa1, a protein homologous to the copper-
regulated transcription factor Ace1, is a novel transcriptional activator.

Full Description: Microarray experiments were carried out to determine whether <i>Haa1</i> is a 
transcription factor and to identify any potential target genes. <i>Haa1</i> was found to 
be a transcriptional activator of a set of genes encoding membrane stress proteins.

PMID: 11504737

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Keller01.flt.knn.avg.pcl	<i>Haa1</i> analysis	4	metal or metalloid ion stress|starvation|stress|transcription	2

File last modified: Thu Apr 25 2024

---


# Kuhn_2001_PMID_11154278

Citation: Kuhn KM, DeRisi JL, Brown PO, Sarnow P. 2001. Global and specific translational regulation
in the genomic response of Saccharomyces cerevisiae to a rapid transfer from a fermentable
to a nonfermentable carbon source.

Full Description: The global gene expression program that accompanies the adaptation of Saccharomyces 
cerevisiae to an abrupt transfer from a fermentable to a nonfermentable carbon source was 
characterized by using a cDNA microarray to monitor the relative abundances and polysomal 
distributions of mRNAs.

PMID: 11154278

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Kuhn01.flt.knn.avg.pcl	Carbon source shift	3	carbon utilization|fermentation|respiration	2

File last modified: Thu Apr 25 2024

---


# Protchenko_2001_PMID_11673473

Citation: Protchenko O, Ferea T, Rashford J, Tiedeman J, Brown PO, Botstein D, Philpott CC. 2001. 
Three cell wall mannoproteins facilitate the uptake of iron in Saccharomyces cerevisiae.

Full Description: Analysis of iron-regulated gene expression in Saccharomyces cerevisiae using cDNA 
microarrays has identified three putative cell wall proteins that are directly regulated 
by <i>Aft1p</i>, the major iron-dependent transcription factor in yeast.

PMID: 11673473

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Protchenko01.flt.knn.avg.pcl	Iron concentration and <i>AFT1</i> overexpression	4	metal or metalloid ion stress	2

File last modified: Thu Apr 25 2024

---


# Chitikila_2002_PMID_12419230

Citation: Chitikila C, Huisinga KL, Irvin JD, Basehoar AD, Pugh BF. 2002. Interplay of TBP 
inhibitors in global transcriptional control.

Full Description: Here, we use DNA microarrays to explore the genome-wide interplay of several TBP-
interacting inhibitors in the yeast Saccharomyces cerevisiae.

PMID: 12419230

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Chitikila02.flt.knn.avg.pcl	TBP inhibition	20	transcription	2

File last modified: Thu Apr 25 2024

---


# Clark_2002_PMID_11988574

Citation: Clark TA, Sugnet CW, Ares M Jr. 2002. Genomewide analysis of mRNA processing in yeast 
using splicing-specific microarrays.

Full Description: Hybridiztion of <i>brr1</i>, <i>cus2</i>, <i>dbr1</i>, <i>ecm2</i>, <i>gcr3</i>, <i>hsp104
</i>, <i>msl1</i>, <i>mud1</i>, <i>mud13</i>, <i>mud2</i>, <i>nam8</i>, <i>prp17</i>, <i>
prp18</i>, <i>snu17</i>, <i>snu40</i>, <i>snu66</i>, <i>upf3</i> deletion mutants.

PMID: 11988574

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Clark02_orig.flt.knn.avg.pcl	Deletion of mRNA processing and splicing factors	17	mRNA processing	2

File last modified: Thu Apr 25 2024

---


# Cohen_2002_PMID_12006656

Citation: Cohen BA, Pilpel Y, Mitra RD, Church GM. 2002. Discrimination between paralogs using 
microarray analysis: application to the Yap1p and Yap2p transcriptional networks.

Full Description: There is a separate summary series for each strain and treatment combination. These 
include wild type control, with peroxide and with cadmium, <i>yap1</i> knockout control 
and with peroxide and with cadmium, <i>yap2</i> knockout control and with peroxide and 
with cadmium, double knockouts control and with peroxide, wild type and fold change 
numbers from wild type control for each strain and treatment.

PMID: 12006656

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Cohen02.filter.flt.knn.avg.div.log.pcl	<i>yap1</i> and <i>yap2</i> knockouts with peroxide and cadmium added	11	chemical stimulus|oxidative stress|stress|transcription	2

File last modified: Thu Apr 25 2024

---


# Fleming_2002_PMID_11830665

Citation: Fleming JA, Lightcap ES, Sadis S, Thoroddsen V, Bulawa CE, Blackman RK. 2002. 
Complementary whole-genome technologies reveal the cellular response to proteasome 
inhibition by PS-341.

Full Description: Whole-genome technologies in Saccharomyces cerevisiae to determine the cellular impact of 
the proteasome inhibitor PS-341.

PMID: 11830665

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Fleming99.log.flt.knn.avg.pcl	Proteasome inhibition with exposure to PS-341	30	chemical stimulus|proteolysis	2

File last modified: Thu Apr 25 2024

---


# Huang_2002_PMID_12077337

Citation: Huang D, Moffat J, Andrews B. 2002. Dissection of a complex phenotype by functional 
genomics reveals roles for the yeast cyclin-dependent protein kinase Pho85 in stress 
adaptation and cell integrity.

Full Description: DNA microarrays to examine the genome-wide transcriptional consequences of deleting <i>
PHO85</i> or members of the <i>Pho85</i> cyclin family.

PMID: 12077337

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Huang02.flt.knn.avg.pcl	<i>pho85</i> related knockouts	20	protein phosphorylation	2

File last modified: Thu Apr 25 2024

---


# Kitagawa_2002_PMID_12269742

Citation: Kitagawa E, Takahashi J, Momose Y, Iwahashi H. 2002. Effects of the pesticide thiuram: 
genome-wide screening of indicator genes by yeast DNA microarray.

Full Description: Although there have been studies on the toxicity of the pesticide thiuram, the present 
study is the first one to attempt to integrate a whole genomic response using microarray 
technology. From the DNA microarray experiment it was found that exposure to thiuram led 
to alterations of gene expression in yeast cells and that many genes involved in 
detoxification and stress response were highly induced. The induced genes were classified 
according to the MIPS yeast database. The induction of genes concerned with folding and 
proteolysis reflects the protein denaturing and degradation effects of the thiuram 
treatment The induction of genes involved in redox and defense against reaction oxygen 
species also suggests that thiuram has other effects, such as oxidative stress. Genes 
classified for carbohydrate metabolism and energy were also highly induced, and these gene
products may play the role of providing the energy for the detoxification mechanism. In 
addition, in view of the induction of some genes involved in DNA repair, thiuram 
potentially causes DNA damage. Therefore, as stated in previous reports, thiuram is a 
potential positive toxic chemical. On the other hand, YKL071W, YCR102C, YLR303W, and 
YLL057C were selected based on the result of a DNA microarray experiment and used for the 
promoter activity assay. Thiuram treatment affected the promoter of these genes, 
indicating that this technique could be used for the selection of biomarker candidates.

PMID: 12269742

GEO ID: GSE9336

PCL filename	short description	# conditions	tags	# channels
GSE9336_setA_family.pcl	Effects of the pesticide thiuram: genome-wide screening of indicator genes by yeast DNA microarray.	9	chemical stimulus	2

File last modified: Thu Apr 25 2024

---


# Olesen_2002_PMID_12702272

Citation: Olesen K, Felding T, Gjermansen C, Hansen J. 2002. The dynamics of the Saccharomyces 
carlsbergensis brewing yeast transcriptome during a production-scale lager beer 
fermentation.

Full Description: Expression of brewing yeast during production scale lager beer fermentation.  Samples at 
time points up to 264 hours.

PMID: 12702272

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Olesen02.filter.flt.knn.avg.div.log.pcl	Fermentation time course	12	carbon utilization|fermentation	2

File last modified: Thu Apr 25 2024

---


# Pramila_2002_PMID_12464633

Citation: Pramila T, Miles S, GuhaThakurta D, Jemiolo D, Breeden LL. 2002. Conserved homeodomain 
proteins interact with MADS box protein Mcm1 to restrict ECB-dependent transcription to 
the M/G1 phase of the cell cycle.

Full Description: Two homeodomain proteins, Yox1 and Yhp1, act as repressors at early cell cycle boxes (ECBs
) to restrict their activity to the M/G1 phase of the cell cycle in budding yeast. These 
proteins bind to Mcm1 and to a typical homeodomain binding site. The expression of Yox1 is
periodic and directly correlated with its binding to, and repression of, ECB activity. The
absence of Yox1 and Yhp1 or the constitutive expression of Yox1 leads to the loss of cell-
cycle regulation of ECB activity. Therefore, the cell-cycle-regulated expression of these 
repressors defines the interval of ECB-dependent transcription. Twenty-eight genes, 
including MCM2-7, CDC6, SWI4, CLN3, and a number of genes required during late M phase 
have been identified that are coordinately regulated by this pathway.

PMID: 12464633

GEO ID: GSE5283

PCL filename	short description	# conditions	tags	# channels
GSE5283_setA_family.pcl	Cell cycle-time course of a yox1 yhp1 double mutant, alpha factor synchronized	13	mitotic cell cycle	2

File last modified: Thu Apr 25 2024
GSE3635_setA_family.pcl	Cell cycle-time course, alpha factor synchronized	13	mitotic cell cycle	2

File last modified: Thu Apr 25 2024

---


# Smith_2002_PMID_12135984

Citation: Smith JJ, Marelli M, Christmas RH, Vizeacoumar FJ, Dilworth DJ, Ideker T, Galitski T, 
Dimitrov K, Rachubinski RA, Aitchison JD. 2002. Transcriptome profiling to identify genes 
involved in peroxisome assembly and function.

Full Description: Yeast cells were induced to proliferate peroxisomes, and microarray transcriptional 
profiling was used to identify PEX genes encoding peroxins involved in peroxisome assembly
and genes involved in peroxisome function. Clustering algorithms identified 224 genes with
expression profiles similar to those of genes encoding peroxisomal proteins and genes 
involved in peroxisome biogenesis. Several previously uncharacterized genes were 
identified, two of which, YPL112c and YOR084w, encode proteins of the peroxisomal membrane
and matrix, respectively. Ypl112p, renamed Pex25p, is a novel peroxin required for the 
regulation of peroxisome size and maintenance. These studies demonstrate the utility of 
comparative gene profiling as an alternative to functional assays to identify genes with 
roles in peroxisome biogenesis.

PMID: 12135984

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.peroxisome.pcl	Expression in response to peroxisome induction and repression	8	carbon utilization|lipid metabolism	2

File last modified: Thu Apr 25 2024

---


# Wang_2002_PMID_11972065

Citation: Wang Y, Liu CL, Storey JD, Tibshirani RJ, Herschlag D, Brown PO. 2002. Precision and 
functional specificity in mRNA decay.

Full Description: Posttranscriptional processing of mRNA is an integral component of the gene expression 
program. By using DNA microarrays, we precisely measured the decay of each yeast mRNA, 
after thermal inactivation of a temperature-sensitive RNA polymerase II. The half-lives 
varied widely, ranging from approximately 3 min to more than 90 min. We found no simple 
correlation between mRNA half-lives and ORF size, codon bias, ribosome density, or 
abundance. However, the decay rates of mRNAs encoding groups of proteins that act together
in stoichiometric complexes were generally closely matched, and other evidence pointed to 
a more general relationship between physiological function and mRNA turnover rates. The 
results provide strong evidence that precise control of the decay of each mRNA is a 
fundamental feature of the gene expression program in yeast.

PMID: 11972065

GEO ID: GSE2986

PCL filename	short description	# conditions	tags	# channels
GSE2986_setA_family.pcl	Precision and functional specificity in mRNA decay.	58	RNA catabolism	2

File last modified: Thu Apr 25 2024

---


# Williams_2002_PMID_12370439

Citation: Williams RM, Primig M, Washburn BK, Winzeler EA, Bellis M, Sarrauste de Menthiere C, Davis
RW, Esposito RE. 2002. The Ume6 regulon coordinates metabolic and meiotic gene expression 
in yeast.

Full Description: wt, <i>ume6</i> deletion diploids during vegetative growth in glucose and acetate.

PMID: 12370439

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Williams02.filter.flt.knn.avg.div.log.pcl	<i>Ume6</i> regulon	8	carbon utilization|sporulation	2

File last modified: Thu Apr 25 2024

---


# Yoshimoto_2002_PMID_12058033

Citation: Yoshimoto H, Saltsman K, Gasch AP, Li HX, Ogawa N, Botstein D, Brown PO, Cyert MS. 2002. 
Genome-wide analysis of gene expression regulated by the calcineurin/Crz1p signaling 
pathway in Saccharomyces cerevisiae.

Full Description: The Ca(2+)/calmodulin-dependent protein phosphatase, calcineurin, is activated by specific
environmental conditions, including exposure to Ca(2+) and Na(+), and induces gene 
expression by regulating the <i>Crz1p/Tcn1p</i> transcription factor. We used DNA 
microarrays to perform a comprehensive analysis of calcineurin/<i>Crz1p</i>-dependent gene
expression following addition of Ca(2+) (200 mm) or Na(+) (0.8 m) to yeast.

PMID: 12058033

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Yoshimoto02_Ca.flt.knn.avg.pcl	Response to Ca(2+) in calcineurin/Crz1 pathway regulated gene expression	24	cellular ion homeostasis|chemical stimulus|signaling	2

File last modified: Thu Apr 25 2024
2010.Yoshimoto02_Na.flt.knn.avg.pcl	Response to Na(+) in calcineurin/Crz1 pathway regulated gene expression	16	cellular ion homeostasis|chemical stimulus|signaling	2

File last modified: Thu Apr 25 2024

---


# Bro_2003_PMID_12791685

Citation: Bro C, Regenberg B, Lagniel G, Labarre J, Montero-Lomeli M, Nielsen J. 2003. 
Transcriptional, proteomic, and metabolic responses to lithium in galactose-grown yeast 
cells.

Full Description: Wild type CEN.PK113-7D grown with 20 g/L galactose. LiCl (10 mM, therapeutically relevant
) added and samples analyzed at 0, 20, 40, 60 and 140 minutes after LiCl addition. Lithium
inhibits phosphoglucomutase, affecting galactose uptake and growth.

PMID: 12791685

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Bro03.filter.flt.knn.avg.div.log.pcl	Lithium response	7	chemical stimulus|metal or metalloid ion stress	2

File last modified: Thu Apr 25 2024

---


# Bulik_2003_PMID_14555471

Citation: Bulik DA, Olczak M, Lucero HA, Osmond BC, Robbins PW, Specht CA. 2003. Chitin synthesis in
Saccharomyces cerevisiae in response to supplementation of growth medium with glucosamine 
and cell wall stress.

Full Description: wt and <i>fks1</i> mutant exposed to chitin-inducing glucosamine for 2-hours or continuous
steady state.

PMID: 14555471

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Bulik03.filter.flt.knn.avg.div.log.pcl	Chitin synthesis	11	cell wall organization|chemical stimulus|stress	2

File last modified: Thu Apr 25 2024

---


# Duvel_2003_PMID_12820961

Citation: Duvel K, Santhanam A, Garrett S, Schneper L, Broach JR. 2003. Multiple roles of Tap42 in 
mediating rapamycin-induced transcriptional changes in yeast.

Full Description: We have examined the role of the essential protein <i>Tap42</i> in transcriptional 
regulation by Tor.

PMID: 12820961

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Duvel03_delayedRap.flt.knn.avg.pcl	Post heat shock, delayed rapamycin exposure time course	20	chemical stimulus|heat shock|protein dephosphorylation|signaling	2

File last modified: Thu Apr 25 2024
2010.Duvel03_immediateRap.flt.knn.avg.pcl	Post heat shock, immediate rapamycin exposure time course	10	chemical stimulus|heat shock|protein dephosphorylation|signaling	2

File last modified: Thu Apr 25 2024

---


# Fry_2003_PMID_12875747

Citation: Fry RC, Sambandan TG, Rha C. 2003. DNA damage and stress transcripts in Saccharomyces 
cerevisiae mutant sgs1.

Full Description: wt & <i>sgs1</i> null, w/ or w/o MMS (DNA damaging agent

).

PMID: 12875747

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Fry03.filter.flt.knn.avg.div.log.pcl	Aging in yeast	8	DNA damage stimulus|stress	2

File last modified: Thu Apr 25 2024

---


# Jones_2003_PMID_14570984

Citation: Jones DL, Petty J, Hoyle DC, Hayes A, Ragni E, Popolo L, Oliver SG, Stateva LI. 2003. 
Transcriptome profiling of a Saccharomyces cerevisiae mutant with a constitutively 
activated Ras/cAMP pathway.

Full Description: wt, <i>pde2Delta</i> deletion mutant.

PMID: 14570984

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Jones03_orig.flt.knn.avg.pcl	Ras/cAMP signal transduction pathway	5	signaling	2

File last modified: Thu Apr 25 2024

---


# Kitagawa_2003_PMID_12854720

Citation: Kitagawa E, Momose Y, Iwahashi H. 2003. Correlation of the structures of agricultural 
fungicides to gene expression in Saccharomyces cerevisiae upon exposure to toxic doses.

Full Description: Correlations between the chemical structures of agricultural fungicides and mRNA 
expression levels following exposure of Saccharomyces cerevisiae to toxic doses of thiuram
, zineb, maneb, TPN, and PCP were examined. Structurally, thiuram, zineb, and maneb are 
dithiocarbamate fungicides, whereas TPN and PCP are not. To characterize chemical toxicity
, genes expression was classified according to the functional groups used by the MIPS 
database. However, no correlations between the classification scheme and chemical 
structures were found. Hierarchical clustering of gene expression profiles was performed 
to characterize the effects of the five chemicals. According to this analysis the 
similarity of gene expression profiles depended on the similarity of chemical structures. 
These results suggest that DNA microarray technology has potential for predicting the 
major chemicals which will cause environmental toxicity and will provide information on 
new biomonitoring methods.

PMID: 12854720

GEO ID: GSE9337

PCL filename	short description	# conditions	tags	# channels
GSE9337_setA_family.pcl	Correlation of the structures of agricultural fungicides to gene expression in Saccharomyces cerevisiae upon exposure to toxic doses.	15	chemical stimulus	2

File last modified: Thu Apr 25 2024

---


# Meneghini_2003_PMID_12628191

Citation: Meneghini MD, Wu M, Madhani HD. 2003. Conserved histone variant H2A.Z protects euchromatin
from the ectopic spread of silent heterochromatin.

Full Description: Boundary elements hinder the spread of heterochromatin, yet these sites do not fully 
account for the preservation of adjacent euchromatin. Histone variant H2A.Z (Htz1 in yeast
) replaces conventional H2A in many nucleosomes. Microarray analysis revealed that HTZ1-
activated genes cluster near telomeres. The reduced expression of most of these genes in 
htz1Delta cells was reversed by the deletion of SIR2 (sir2Delta) suggesting that H2A.Z 
antagonizes telomeric silencing. Other Htz1-activated genes flank the silent HMR mating-
type locus. Their requirement for Htz1 can be bypassed by sir2Delta or by a deletion 
encompassing the silencing nucleation sites in HMR. In htz1Delta cells, Sir2 and Sir3 
spread into flanking euchromatic regions, producing changes in histone H4 acetylation and 
H3 4-methylation indicative of ectopic heterochromatin formation. Htz1 is enriched in 
these euchromatic regions and acts synergistically with a boundary element to prevent the 
spread of heterochromatin. Thus, euchromatin and heterochromatin each contains components 
that antagonize switching to the opposite chromatin state.

PMID: 12628191

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.swr1_htz1_hmr_sir2.pcl	Gene regulation by HTZ1, SWR1 and SIR2	24	chromatin organization	2

File last modified: Thu Apr 25 2024

---


# Segal_2003_PMID_12740579

Citation: Segal E, Shapira M, Regev A, Pe'er D, Botstein D, Koller D, Friedman N. 2003. Module 
networks: identifying regulatory modules and their condition-specific regulators from gene
expression data.

Full Description: wt and <i>kin82</i> mutant strains after heat shock from 25C to 37C, samples at time 
points up to 1 hour.

PMID: 12740579

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Segal03_HSkin82.flt.knn.avg.pcl	Heat Shock, <i>kin82</i> mutant	10	heat shock|protein phosphorylation|stress	2

File last modified: Thu Apr 25 2024
2010.Segal03_hypo-osmoticPpt1.flt.knn.avg.pcl	Hypo-osmotic shock, <i>ppt1</i> mutant	10	osmotic stress|protein dephosphorylation|stress	2

File last modified: Thu Apr 25 2024
2010.Segal03_stationaryPhaseYPL230W.flt.knn.avg.pcl	Stationary phase, ypl230w mutant	12	stationary phase entry	2

File last modified: Thu Apr 25 2024

---


# Belli_2004_PMID_14722110

Citation: Belli G, Molina MM, Garcia-Martinez J, Perez-Ortin JE, Herrero E. 2004. Saccharomyces 
cerevisiae glutaredoxin 5-deficient cells subjected to continuous oxidizing conditions are
affected in the expression of specific sets of genes.

Full Description: Transcriptome analysis of glutaredoxin 5-deficient (<i>grx5</i>) mutant, a model for 
continuous moderate oxidative stress. Respiratory petite (<i>pet117</i>) mutants and wild 
type also examined.

PMID: 14722110

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Belli04.filter.flt.knn.avg.div.log.pcl	Oxidative stress and glutaredoxin 5-deficient mutant	9	oxidative stress|stress	2

File last modified: Thu Apr 25 2024

---


# Cullen_2004_PMID_15256499

Citation: Cullen PJ, Sabbagh W Jr, Graham E, Irick MM, van Olden EK, Neal C, Delrow J, Bardwell L, 
Sprague GF Jr. 2004. A signaling mucin at the head of the Cdc42- and MAPK-dependent 
filamentous growth pathway in yeast.

Full Description: Signaling molecules such as Cdc42 and mitogen-activated protein kinases (MAPKs) can 
function in multiple pathways in the same cell. Here, we propose one mechanism by which 
such factors may be directed to function in a particular pathway such that a specific 
response is elicited. Using genomic approaches, we identify a new component of the Cdc42- 
and MAPK-dependent signaling pathway that regulates filamentous growth (FG) in yeast. This
factor, called Msb2, is a FG-pathway-specific factor that promotes differential activation
of the MAPK for the FG pathway, Kss1. Msb2 is localized to polarized sites on the cell 
surface and interacts with Cdc42 and with the osmosensor for the high osmolarity glycerol 
response (HOG) pathway, Sho1. Msb2 is glycosylated and is a member of the mucin family, 
proteins that in mammalian cells promote disease resistance and contribute to metastasis 
in cancer cells. Remarkably, loss of the mucin domain of Msb2 causes hyperactivity of the 
FG pathway, demonstrating an inhibitory role for mucin domains in MAPK pathway activation
. Taken together, our data suggest that Msb2 is a signaling mucin that interacts with 
general components, such as Cdc42 and Sho1, to promote their function in the FG pathway.

PMID: 15256499

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.glycosylation.pcl	Effects of glycosylation defects on gene expression	2	filamentous growth|protein glycosylation|signaling	2

File last modified: Thu Apr 25 2024

---


# Daran-Lapujade_2004_PMID_14630934

Citation: Daran-Lapujade P, Jansen ML, Daran JM, van Gulik W, de Winde JH, Pronk JT. 2004. Role of 
transcriptional regulation in controlling fluxes in central carbon metabolism of 
Saccharomyces cerevisiae. A chemostat culture study.

Full Description: In contrast to batch cultivation, chemostat cultivation allows the identification of 
carbon source responses without interference by carbon-catabolite repression, accumulation
of toxic products, and differences in specific growth rate. This study focuses on the 
yeast Saccharomyces cerevisiae, grown in aerobic, carbon-limited chemostat cultures. 
Genome-wide transcript levels and in vivo fluxes were compared for growth on two sugars, 
glucose and maltose, and for two C2-compounds, ethanol and acetate. In contrast to 
previous reports on batch cultures, few genes (180 genes) responded to changes of the 
carbon source by a changed transcript level. Very few transcript levels were changed when 
glucose as the growth-limiting nutrient was compared with maltose (33 transcripts), or 
when acetate was compared with ethanol (16 transcripts). Although metabolic flux analysis 
using a stoichiometric model revealed major changes in the central carbon metabolism, only
117 genes exhibited a significantly different transcript level when sugars and C2-
compounds were provided as the growth-limiting nutrient. Despite the extensive knowledge 
on carbon source regulation in yeast, many of the carbon source-responsive genes encoded 
proteins with unknown or incompletely characterized biological functions. In silico 
promoter analysis of carbon source-responsive genes confirmed the involvement of several 
known transcriptional regulators and suggested the involvement of additional regulators. 
Transcripts involved in the glyoxylate cycle and gluconeogenesis showed a good correlation
with in vivo fluxes. This correlation was, however, not observed for other important 
pathways, including the pentose-phosphate pathway, tricarboxylic acid cycle, and, in 
particular, glycolysis. These results indicate that in vivo fluxes in the central carbon 
metabolism of S. cerevisiae grown in steadystate, carbon-limited chemostat cultures are 
controlled to a large extent via post-transcriptional mechanisms.

PMID: 14630934

GEO ID: GSE8895

PCL filename	short description	# conditions	tags	# channels
GSE8895_setA_family.pcl	Role of transcriptional regulation in controlling fluxes in central carbon metabolism of Saccharomyces cerevisiae. A chemostat culture study.	12	carbon utilization|fermentation|respiration	1

File last modified: Thu Apr 25 2024

---


# De_Nadal_2004_PMID_14737171

Citation: De Nadal E, Zapater M, Alepuz PM, Sumoy L, Mas G, Posas F. 2004. The MAPK Hog1 recruits 
Rpd3 histone deacetylase to activate osmoresponsive genes.

Full Description: Comparison of wild type and <i>rpd3-</i> S. cerevisiae cells in response to osmotic stress
. Wild type (MATalpha <i>ura3</i> <i>leu2</i> <i>his3</i>) and <i>rpd3-</i> (MATalpha <i>
ura3</i> <i>leu2</i> <i>his3</i> <i>rpd3</i>::KANMX) were grown to OD660=1 in YPD media 
and then subjected or not to a brief osmotic stress (5 or 20 min / 0.4 M NaCl). Cells were
collected by centrifugation and total RNA was extracted as described.

PMID: 14737171

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.deNadal04.flt.knn.avg.pcl	Osmotic stress	12	histone modification|osmotic stress|stress	2

File last modified: Thu Apr 25 2024

---


# Garcia_2004_PMID_14739279

Citation: Garcia R, Bermejo C, Grau C, Perez R, Rodriguez-Pena JM, Francois J, Nombela C, Arroyo J. 
2004. The global transcriptional response to transient cell wall damage in Saccharomyces 
cerevisiae and its regulation by the cell integrity signaling pathway.

Full Description: In the yeast Saccharomyces cerevisiae, environmental stress conditions that damage the 
cell wall lead to activation of the so-called compensatory mechanism, aimed at preserving 
cell integrity through a remodeling of this extracellular matrix. Here we used DNA 
microarrays to investigate the molecular basis of this response to two agents that induce 
transient cell wall damage; namely Congo Red and Zymolyase. Treatment of the cells with 
these two agents elicited the up-regulation of 132 and 101 genes respectively, the main 
functional groups among them being involved in cell wall construction and metabolism. The 
main response does not occur until hours after exposure to the cell wall-perturbing agent
. In some cases, this response was transient, but more sustained in others, especially in 
the case of the genes involved in cell wall remodeling. Clustering of these data together 
with those from the response to constitutive cell wall damage, revealed the existence of a
cluster of co-regulated genes that was strongly induced under all conditions assayed. 
Those  genes induced by cell wall damage showed an enrichment in DNA binding motifs for  
Rlm1p, Crz1p, SBF (Swi4p/Swi6p), Msn2p/Msn4p, Ste12p, and Tec1p transcription factors, 
suggesting a complex regulation of this response together with the possible involvement of
several signaling pathways. With the exception of PHO89 and FKS2, none of the genes 
induced by Congo Red was up-regulated in a slt2 strain. Moreover, characterization of the 
transcriptional response to Congo Red in a rlm1 mutant strain revealed that only a few 
genes (i.e. PHO89, FKS2, YLR042C, and CHA1) were induced at least partially independently 
of the transcription factor Rlm1p, the rest being totally dependent on this transcription 
factor for their activation. Our findings consistently demonstrate  that the cell 
integrity signaling pathway regulates the cell wall damage compensatory response, mainly 
through transcriptional activation mediated by Rlm1p.

PMID: 14739279

GEO ID: GSE961.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE961.final.pcl	The global transcriptional response to transient cell wall damage in Saccharomyces cerevisiae and its regulation by the cell integrity signaling pathway.	4	stress|chemical stimulus|cell wall organization	2

File last modified: Thu Apr 25 2024
GSE964.final.pcl	The global transcriptional response to transient cell wall damage in Saccharomyces cerevisiae and its regulation by the cell integrity signaling pathway.	4	stress|chemical stimulus|cell wall organization	2

File last modified: Thu Apr 25 2024
GSE960.final.pcl	The global transcriptional response to transient cell wall damage in Saccharomyces cerevisiae and its regulation by the cell integrity signaling pathway.	2	stress|chemical stimulus|cell wall organization	2

File last modified: Thu Apr 25 2024

---


# Haugen_2004_PMID_15575969

Citation: Haugen AC, Kelley R, Collins JB, Tucker CJ, Deng C, Afshari CA, Brown JM, Ideker T, Van 
Houten B. 2004. Integrating phenotypic and expression profiles to map arsenic-response 
networks.

Full Description: BACKGROUND: Arsenic is a nonmutagenic carcinogen affecting millions of people. The 
cellular impact of this metalloid in Saccharomyces cerevisiae was determined by profiling 
global gene expression and sensitivity phenotypes. These data were then mapped to a 
metabolic network composed of all known biochemical reactions in yeast, as well as the 
yeast network of 20,985 protein-protein/protein-DNA interactions. RESULTS: While the 
expression data unveiled no significant nodes in the metabolic network, the regulatory 
network revealed several important nodes as centers of arsenic-induced activity. The 
highest-scoring proteins included Fhl1, Msn2, Msn4, Yap1, Cad1 (Yap2), Pre1, Hsf1 and 
Met31. Contrary to the gene-expression analyses, the phenotypic-profiling data mapped to 
the metabolic network. The two significant metabolic networks unveiled were shikimate, and
serine, threonine and glutamate biosynthesis. We also carried out transcriptional 
profiling of specific deletion strains, confirming that the transcription factors Yap1, 
Arr1 (Yap8), and Rpn4 strongly mediate the cell's adaptation to arsenic-induced stress but
that Cad1 has negligible impact. CONCLUSIONS: By integrating phenotypic and 
transcriptional profiling and mapping the data onto the metabolic and regulatory networks
, we have shown that arsenic is likely to channel sulfur into glutathione for 
detoxification, leads to indirect oxidative stress by depleting glutathione pools, and 
alters protein turnover via arsenation of sulfhydryl groups on proteins. Furthermore, we 
show that phenotypically sensitive pathways are upstream of differentially expressed ones
, indicating that transcriptional and phenotypic profiling implicate distinct, but related
, pathways.

PMID: 15575969

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.arsenic_time_dose.pcl	Expression in response to arsenic (time and dose)	4	chemical stimulus|metal or metalloid ion stress	2

File last modified: Thu Apr 25 2024
2010.arsenic_deletions.pcl	Role of transcription factors in adaptation to arsenic-induced stress	13	chemical stimulus|metal or metalloid ion stress	2

File last modified: Thu Apr 25 2024

---


# Huang_2004_PMID_15539461

Citation: Huang J, Zhu H, Haggarty SJ, Spring DR, Hwang H, Jin F, Snyder M, Schreiber SL. 2004. 
Finding new components of the target of rapamycin (TOR) signaling network through chemical
genetics and proteome chips.

Full Description: The TOR (target of rapamycin) proteins play important roles in nutrient signaling in 
eukaryotic cells. Rapamycin treatment induces a state reminiscent of the nutrient 
starvation response, often resulting in growth inhibition. Using a chemical genetic 
modifier screen, we identified two classes of small molecules, small-molecule inhibitors 
of rapamycin (SMIRs) and small-molecule enhancers of rapamycin (SMERs), that suppress and 
augment, respectively, rapamycin's effect in the yeast Saccharomyces cerevisiae. Probing 
proteome chips with biotinylated SMIRs revealed putative intracellular target proteins, 
including Tep1p, a homolog of the mammalian PTEN (phosphatase and tensin homologue deleted
on chromosome 10) tumor suppressor, and Ybr077cp (Nir1p), a protein of previously unknown 
function that we show to be a component of the TOR signaling network. Both SMIR target 
proteins are associated with PI(3,4)P2, suggesting a mechanism of regulation of the TOR 
pathway involving phosphatidylinositides. Our results illustrate the combined use of 
chemical genetics and proteomics in biological discovery and map a path for creating 
useful therapeutics for treating human diseases involving the TOR pathway, such as 
diabetes and cancer.

PMID: 15539461

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.rapamycin.pcl	Expression in the presence of small-molecule inhibitors of rapamycin (SMIRs)	10	chemical stimulus|signaling	2

File last modified: Thu Apr 25 2024

---


# Huisinga_2004_PMID_14992726

Citation: Huisinga KL, Pugh BF. 2004. A genome-wide housekeeping role for TFIID and a highly 
regulated stress-related role for SAGA in Saccharomyces cerevisiae.

Full Description: TFIID and SAGA share a common set of TAFs, regulate chromatin, and deliver TBP to 
promoters. Here we examine their relationship within the context of the Saccharomyces 
cerevisiae genome-wide regulatory network. We find that while TFIID and SAGA make 
overlapping contributions to the expression of all genes, TFIID function predominates at 
approximately 90% and SAGA at approximately 10% of the measurable genome. Strikingly, SAGA
-dominated genes are largely stress induced and TAF independent, and are downregulated by 
the coordinate action of a variety  of chromatin, TBP, and RNA polymerase II regulators. 
In contrast, the TFIID-dominated class is less regulated, but is highly dependent upon 
TAFs, including those shared between TFIID and SAGA. These two distinct modes of 
transcription regulation might reflect the need to balance inducible stress responses with
the steady output of housekeeping genes.

PMID: 14992726

GEO ID: GSE885.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE885.final.pcl	A genome-wide housekeeping role for TFIID and a highly regulated stress-related role for SAGA in Saccharomyces cerevisiae.	30	stress|transcriptional regulation	2

File last modified: Thu Apr 25 2024

---


# Jansen_2004_PMID_15066785

Citation: Jansen ML, Daran-Lapujade P, de Winde JH, Piper MD, Pronk JT. 2004. Prolonged maltose-
limited cultivation of Saccharomyces cerevisiae selects for cells with improved maltose 
affinity and hypersensitivity.

Full Description: Prolonged cultivation (>25 generations) of Saccharomyces cerevisiae in aerobic, maltose-
limited chemostat cultures led to profound physiological changes. Maltose hypersensitivity
was observed when cells from prolonged cultivations were suddenly exposed to excess 
maltose. This substrate hypersensitivity was evident from massive cell lysis and loss of 
viability. During prolonged cultivation at a fixed specific growth rate, the affinity for 
the growth-limiting nutrient (i.e., maltose) increased, as evident from a decreasing 
residual maltose concentration. Furthermore, the capacity of maltose-dependent proton 
uptake increased up to 2.5-fold during prolonged cultivation. Genome-wide transcriptome 
analysis showed that the increased maltose transport capacity was not primarily due to 
increased transcript levels of maltose-permease genes upon prolonged cultivation. We 
propose that selection for improved substrate affinity (ratio of maximum substrate 
consumption rate and substrate saturation constant) in maltose-limited cultures leads to 
selection for cells with an increased capacity for maltose uptake. At the same time, the 
accumulative nature of maltose-proton symport in S. cerevisiae leads to unrestricted 
uptake when maltose-adapted cells are exposed to a substrate excess. These changes were 
retained after isolation of individual cell lines from the chemostat cultures and 
nonselective cultivation, indicating that mutations were involved. The observed trade-off 
between substrate affinity and substrate tolerance may be relevant for metabolic 
engineering and strain selection for utilization of substrates that are taken up by proton
symport.

PMID: 15066785

GEO ID: GSE8897

PCL filename	short description	# conditions	tags	# channels
GSE8897_setA_family.pcl	Prolonged maltose-limited cultivation of Saccharomyces cerevisiae selects for cells with improved maltose affinity and hypersensitivity.	5	evolution	1

File last modified: Thu Apr 25 2024

---


# Jin_2004_PMID_15528549

Citation: Jin YS, Laplaza JM, Jeffries TW. 2004. Saccharomyces cerevisiae engineered for xylose 
metabolism exhibits a respiratory response.

Full Description: Strain YSX3 engineered for D-xylose utilization and respiration-deficient mutant FPL-YSX3P
were grown under full aeration or oxygen limitation, with glucose or xylose as a carbon 
source.

PMID: 15528549

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Jin04.filter.flt.knn.avg.div.log.pcl	Xylose metabolism	6	carbon utilization|oxygen level alteration|respiration	2

File last modified: Thu Apr 25 2024

---


# Klevecz_2004_PMID_14734811

Citation: Klevecz RR, Bolen J, Forrest G, Murray DB. 2004. A genomewide oscillation in transcription
gates DNA replication and cell cycle.

Full Description: Microarray analysis from a yeast continuous synchrony culture system shows a genomewide 
oscillation in transcription. Maximums in transcript levels occur at three nearly equally 
spaced intervals in this approximately 40-min cycle of respiration and reduction. Two 
temporal clusters (4,679 of 5,329) are maximally expressed during the reductive phase of 
the cycle, whereas a third cluster (650) is maximally expressed during the respiratory 
phase. Transcription is organized functionally into redox-state superclusters with genes 
known to be important in respiration or reduction being synthesized in opposite phases of 
the cycle. The transcriptional cycle gates synchronous bursts in DNA replication in a 
constant fraction of the population at 40-min intervals. Restriction of DNA synthesis to 
the reductive phase of the cycle may be an evolutionarily important mechanism for reducing
oxidative damage to DNA during replication.

PMID: 14734811

GEO ID: GSE2583

PCL filename	short description	# conditions	tags	# channels
GSE2583_setA_family.pcl	A genomewide oscillation in transcription gates DNA replication and cell cycle.	32	carbon utilization|mitotic cell cycle|oxygen level alteration|respiration	1

File last modified: Thu Apr 25 2024

---


# Leber_2004_PMID_15314654

Citation: Leber JH, Bernales S, Walter P. 2004. IRE1-independent gain control of the unfolded 
protein response.

Full Description: Strains expressing <i>HAC1</i> grown at 30C or shifted from 23C to 37C, and treated with 
6mM DTT or tunicamycin for 60 minutes.  DTT and TM activate the unfolded protein response
, which is regulated by <i>Hac1p</i

>..

PMID: 15314654

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Leber04.flt.knn.avg.pcl	Unfolded protein response and<i>HAC1</i> transcription	13	chemical stimulus|response to unfolded protein|stress	1

File last modified: Thu Apr 25 2024

---


# Martin_2004_PMID_15476558

Citation: Martin DE, Demougin P, Hall MN, Bellis M. 2004. Rank Difference Analysis of Microarrays (
RDAM), a novel approach to statistical analysis of microarray expression profiling data.

Full Description: Comparison of total transcription profiles for temperature-sensitive <i>TOR2</i> mutant 
strain SH121 to its isogenic wild type counterpart SH100.

PMID: 15476558

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Martin04.filter.flt.knn.avg.div.log.pcl	<i>TOR2</i>-controlled transcription	12	protein phosphorylation|signaling	2

File last modified: Thu Apr 25 2024

---


# Miyake_2004_PMID_15192094

Citation: Miyake T, Reese J, Loch CM, Auble DT, Li R. 2004. Genome-wide analysis of ARS (
autonomously replicating sequence) binding factor 1 (Abf1p)-mediated transcriptional 
regulation in Saccharomyces cerevisiae.

Full Description: <i>abf1-1</i> mutant and wt cells grown at 30C, raised to 36C for 45 min.  Four repeats of
the same experiement.

PMID: 15192094

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Miyake04.flt.knn.avg.pcl	<i>abf1-1</i> mutant at 36C	4	transcription	2

File last modified: Thu Apr 25 2024

---


# Mizuguchi_2004_PMID_14645854

Citation: Mizuguchi G, Shen X, Landry J, Wu WH, Sen S, Wu C. 2004. ATP-driven exchange of histone 
H2AZ variant catalyzed by SWR1 chromatin remodeling complex.

Full Description: The conserved histone variant H2AZ has an important role in the regulation of gene 
expression and the establishment of a buffer to the spread of silent heterochromatin. How 
histone variants such as H2AZ are incorporated into nucleosomes has been obscure. We have 
found that Swr1, a Swi2/Snf2-related adenosine triphosphatase, is the catalytic core of a 
multisubunit, histone-variant exchanger that efficiently replaces conventional histone H2A
with histone H2AZ in nucleosome arrays. Swr1 is required for the deposition of histone 
H2AZ at specific chromosome locations in vivo, and Swr1 and H2AZ commonly regulate a 
subset of yeast genes. These findings define a previously unknown role for the adenosine 
triphosphate-dependent chromatin remodeling machinery.

PMID: 14645854

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.swr1_htz1_ino80.pcl	Gene regulation by Swr1, Htz1, and Ino80	3	chromatin organization	2

File last modified: Thu Apr 25 2024

---


# Mnaimneh_2004_PMID_15242642

Citation: Mnaimneh S, Davierwala AP, Haynes J, Moffat J, Peng WT, Zhang W, Yang X, Pootoolal J, Chua
G, Lopez A, Trochesset M, Morse D, Krogan NJ, Hiley SL, Li Z, Morris Q, Grigull J, 
Mitsakakis N, Roberts CJ, Greenblatt JF, Boone C, Kaiser CA, Andrews BJ, Hughes TR. 2004. 
Exploration of essential gene functions via titratable promoter alleles.

Full Description: Nearly 20% of yeast genes are required for viability, hindering genetic analysis with 
knockouts. We created promoter-shutoff strains for over two-thirds of all essential yeast 
genes and subjected them to morphological analysis, size profiling, drug sensitivity 
screening, and microarray expression profiling. We then used this compendium of data to 
ask which phenotypic features characterized different functional classes and used these to
infer potential functions for uncharacterized genes. We identified genes involved in 
ribosome biogenesis (HAS1, URB1, and URB2), protein secretion (SEC39), mitochondrial 
import (MIM1), and tRNA charging (GSN1). In addition, apparent negative feedback 
transcriptional regulation of both ribosome biogenesis and the proteasome was observed. We
furthermore show that these strains are compatible with automated genetic analysis. This 
study underscores the importance of analyzing mutant phenotypes and provides a resource to
complement the yeast knockout collection.

PMID: 15242642

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.EssentialGenes.pcl	Exploration of essential gene functions via titratable promoter alleles	215	chemical stimulus|stress	2

File last modified: Thu Apr 25 2024

---


# O'Rourke_2004_PMID_14595107

Citation: O'Rourke SM, Herskowitz I. 2004. Unique and redundant roles for HOG MAPK pathway 
components as revealed by whole-genome expression analysis.

Full Description: Response to elevated osmolarity.

PMID: 14595107

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.ORourke03.flt.knn.avg.pcl	HOG MAPK pathway	133	osmotic stress|protein phosphorylation|signaling|stress	2

File last modified: Thu Apr 25 2024

---


# Orlandi_2004_PMID_14623890

Citation: Orlandi I, Bettiga M, Alberghina L, Vai M. 2004. Transcriptional profiling of ubp10 null 
mutant reveals altered subtelomeric gene expression and insurgence of oxidative stress 
response.

Full Description: Comparison of wt, <i>ubp10</i> null, <i>sir4</i> null, and <i>ubp10</i> <i>sir4</i> double
null mutants.

PMID: 14623890

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Orlandi04.filter.flt.knn.avg.div.log.pcl	Deubiquitinating enzyme <i>UBP10</i> inactivation	4	stress|ubiquitin or ULP modification	2

File last modified: Thu Apr 25 2024

---


# Pitkanen_2004_PMID_15520001

Citation: Pitkanen JP, Torma A, Alff S, Huopaniemi L, Mattila P, Renkonen R. 2004. Excess mannose 
limits the growth of phosphomannose isomerase PMI40 deletion strain of Saccharomyces 
cerevisiae.

Full Description: Expression profiling of a phosphomannose isomerase <i>pmi40</i> deletion strain growing on
media with various mannose concentrations up to 5 g/liter. Results provide insight into 
the cause of slow growth exhibited by the <i>pmi40</i> deletion strain in media with 
excess mannose.

PMID: 15520001

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Pitkanen04.filter.flt.knn.avg.div.log.pcl	Phosphomannose isomerase <i>pmi40</i> deletion strain response to excess mannose	15	carbon utilization|protein glycosylation	2

File last modified: Thu Apr 25 2024

---


# Prinz_2004_PMID_14993204

Citation: Prinz S, Avila-Campillo I, Aldridge C, Srinivasan A, Dimitrov K, Siegel AF, Galitski T. 
2004. Control of yeast filamentous-form growth by modules in an integrated molecular 
network.

Full Description: wt diploid cells shifted from yeast-form growth in SHAD liquid with plentiful glucose and 
ammonium to filamentous-form growth on SLAD agar with low ammonium.  Samples collected 
hourly for 10 hours.

PMID: 14993204

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Prinz04.flt.knn.avg.pcl	Filamentous-form growth on solid media time course	10	filamentous growth	2

File last modified: Thu Apr 25 2024

---


# Rodriguez-Navarro_2004_PMID_14718168

Citation: Rodriguez-Navarro S, Fischer T, Luo MJ, Antunez O, Brettschneider S, Lechner J, Perez-
Ortin JE, Reed R, Hurt E. 2004. Sus1, a functional component of the SAGA histone acetylase
complex and the nuclear pore-associated mRNA export machinery.

Full Description: wt, <i>sus1</i> deletion mutant.

PMID: 14718168

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.RodriguezNavarro04.filter.flt.knn.avg.div.log.pcl	<i>sus1</i> mutant	6	histone modification	2

File last modified: Thu Apr 25 2024

---


# Sabet_2004_PMID_15456858

Citation: Sabet N, Volo S, Yu C, Madigan JP, Morse RH. 2004. Genome-wide analysis of the 
relationship between transcriptional regulation by Rpd3p and the histone H3 and H4 amino 
termini in budding yeast.

Full Description: Analysis of gene regulation by expression profiling of single mutants for histone 
deacetylase <i>RPD3</i>, histones H3 and H4, and double mutants for either <i>RPD3</i> and
H3 or <i>RPD3</i> and H4. Histone mutants defective in amino terminal tails, which are 
acetylated and deacetylated in transcription regulation.

PMID: 15456858

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Sabet04.filter.flt.knn.avg.div.log.pcl	Histone deacetylase <i>RPD3</i> deletion and histone mutations	18	histone modification	2

File last modified: Thu Apr 25 2024

---


# Saldanha_2004_PMID_15240820

Citation: Saldanha AJ, Brauer MJ, Botstein D. 2004. Nutritional homeostasis in batch and steady-
state culture of yeast.

Full Description: Physiological response to limitation by Uracil in batch and steady-state (chemostat) 
cultures of S. cerevisiae.

PMID: 15240820

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Saldanha04_UracilBatchChem.flt.knn.avg.pcl	Limitation by Uracil	20	nutrient utilization	2

File last modified: Thu Apr 25 2024
2010.Saldanha04_UraSulPhoLeuComp.flt.knn.avg.pcl	Comparison of limitation by Ura, Sul, Pho, and Leu	24	amino acid utilization|nutrient utilization|phosphorus utilization|sulfur utilization	2

File last modified: Thu Apr 25 2024
2010.Saldanha04_SulfateBatchChem.flt.knn.avg.pcl	Limitation by Sulfate	21	sulfur utilization	2

File last modified: Thu Apr 25 2024
2010.Saldanha04_LeucineBatchChem.flt.knn.avg.pcl	Limitation by Leucine	29	amino acid utilization	2

File last modified: Thu Apr 25 2024
2010.Saldanha04_PhosphateBatchChem.flt.knn.avg.pcl	Limitation by Phosphate	30	phosphorus utilization	2

File last modified: Thu Apr 25 2024

---


# Sapra_2004_PMID_15452114

Citation: Sapra AK, Arava Y, Khandelia P, Vijayraghavan U. 2004. Genome-wide analysis of pre-mRNA 
splicing: intron features govern the requirement for the second-step factor, Prp17 in 
Saccharomyces cerevisiae and Schizosaccharomyces pombe.

Full Description: wt, <i>prp17</i> null, <i>prp17-1</i>, <i>prp22-1</i> at time points following a 
temperature shift from permissive 23C to restrictive 37C.

PMID: 15452114

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Sapra04.flt.knn.avg.pcl	Pre-mRNA splicing factor mutants at restrictive temperature time course	24	mRNA processing	2

File last modified: Thu Apr 25 2024

---


# Schawalder_2004_PMID_15616569

Citation: Schawalder SB, Kabani M, Howald I, Choudhury U, Werner M, Shore D. 2004. Growth-regulated 
recruitment of the essential yeast ribosomal protein gene activator Ifh1.

Full Description: Analysis of overexpression of essential ribosomal protein activator <i>IFH1</i>. Cells 
engineered to express <i>IFH1</i> from a galactose inducible promoter. Expression examined
at various time points up to 60 minutes following galactose addition. Results indicate 
that <i>IFH1</i> regulates ribosomal protein genes.

PMID: 15616569

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Schawalder04.filter.flt.knn.avg.div.log.pcl	<i>IFH1</i> overexpression: time course	24	transcription	2

File last modified: Thu Apr 25 2024

---


# Shakoury-Elizeh_2004_PMID_14668481

Citation: Shakoury-Elizeh M, Tiedeman J, Rashford J, Ferea T, Demeter J, Garcia E, Rolfes R, Brown 
PO, Botstein D, Philpott CC. 2004. Transcriptional remodeling in response to iron 
deprivation in Saccharomyces cerevisiae.

Full Description: Saccharomyces cerevisiae responds to depletion of iron in the environment.

PMID: 14668481

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.ShakouryElizeh04.flt.knn.avg.pcl	Iron deprivation	6	metal or metalloid ion stress|starvation	2

File last modified: Thu Apr 25 2024

---


# Shapira_2004_PMID_15371544

Citation: Shapira M, Segal E, Botstein D. 2004. Disruption of yeast forkhead-associated cell cycle 
transcription by oxidative stress.

Full Description: Oxidative stress caused by Menadione or Hydrogen peroxide in synchronized Saccharomyces 
cerevisiae cultures. Alpha factor synchronized cultures (0.2-0.4 OD), treated at the 
beginning of S phase (25 min after release from G1 arrest) with either 2 mM Menadione (MD
) or 0.24 mM Hydrogen peroxide (HP), show cell cycle effects. Cells treated with MD 
arrested at G1. Cells treated with HP delayed at S and then, after removal of HP at 135 
minutes , continued the cell cycle, only to arrest at G2/M. Growth was carried out in 30C 
with shaking (295 rpm

)..

PMID: 15371544

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Shapira04.flt.knn.avg.pcl	Oxidative stress responses	70	chemical stimulus|mitotic cell cycle|oxidative stress|stress	2

File last modified: Thu Apr 25 2024

---


# van_Attikum_2004_PMID_15607975

Citation: van Attikum H, Fritsch O, Hohn B, Gasser SM. 2004. Recruitment of the INO80 complex by H2A,
phosphorylation links ATP-dependent chromatin remodeling with DNA double-strand break 
repair.

Full Description: The budding yeast INO80 complex is a conserved ATP-dependent nucleosome remodeler 
containing actin-related proteins Arp5 and Arp8. Strains lacking INO80, ARP5, or ARP8 have
defects in transcription. Here we show that these mutants are hypersensitive to DNA 
damaging agents and to double-strand breaks (DSBs) induced by the HO endonuclease. The 
checkpoint response and most transcriptional modulation associated with induction of DNA 
damage are unaffected by these mutations. Using chromatin immunoprecipitation we show that
Ino80, Arp5, and Arp8 are recruited to an HO-induced DSB, where a phosphorylated form of 
H2A accumulates. Recruitment of Ino80 is compromised in cells lacking the H2A 
phosphoacceptor S129. Finally, we demonstrate that conversion of the DSB into ssDNA is 
compromised in arp8 and H2A mutants, which are both deficient for INO80 activity at the 
site of damage. These results implicate INO80-mediated chromatin remodeling directly at 
DSBs, where it appears to facilitate processing of the lesion.

PMID: 15607975

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Ino80Arp8.pcl	Gene regulation by Ino80 and Arp8	2	chromatin organization	2

File last modified: Thu Apr 25 2024
2010.MMSresponse.pcl	Gene regulation by Ino80 and Arp8 in response to MMS	3	chromatin organization|DNA damage stimulus	2

File last modified: Thu Apr 25 2024

---


# Aguilera_2005_PMID_15780657

Citation: Aguilera J, Petit T, de Winde JH, Pronk JT. 2005. Physiological and genome-wide 
transcriptional responses of Saccharomyces cerevisiae to high carbon dioxide 
concentrations.

Full Description: Physiological effects of carbon dioxide and impact on genome-wide transcript profiles were
analysed in chemostat cultures of Saccharomyces cerevisiae. In anaerobic, glucose-limited 
chemostat cultures grown at atmospheric pressure, cultivation under CO(2)-saturated 
conditions had only a marginal (<10%) impact on the biomass yield. Conversely, a 25% 
decrease of the biomass yield was found in aerobic, glucose-limited chemostat cultures 
aerated with a mixture of 79% CO(2) and 21% O(2). This observation indicated that 
respiratory metabolism is more sensitive to CO(2) than fermentative metabolism. Consistent
with the more pronounced physiological effects of CO(2) in respiratory cultures, the 
number of CO(2)-responsive transcripts was higher in aerobic cultures than in anaerobic 
cultures. Many genes involved in mitochondrial functions showed a transcriptional response
to elevated CO(2) concentrations. This is consistent with an uncoupling effect of CO(2) 
and/or intracellular bicarbonate on the mitochondrial inner membrane. Other transcripts 
that showed a significant transcriptional response to elevated CO(2) included NCE103 (
probably encoding carbonic anhydrase), PCK1 (encoding PEP carboxykinase) and members of 
the IMD gene family (encoding isozymes of inosine monophosphate dehydrogenase

)..

PMID: 15780657

GEO ID: GSE8900

PCL filename	short description	# conditions	tags	# channels
GSE8900_setA_family.pcl	Physiological and genome-wide transcriptional responses of Saccharomyces cerevisiae to high carbon dioxide concentrations.	18	carbon utilization|chemical stimulus|nitrogen utilization|respiration	1

File last modified: Thu Apr 25 2024

---


# Boer_2005_PMID_15949974

Citation: Boer VM, Daran JM, Almering MJ, de Winde JH, Pronk JT. 2005. Contribution of the 
Saccharomyces cerevisiae transcriptional regulator Leu3p to physiology and gene expression
in nitrogen- and carbon-limited chemostat cultures.

Full Description: Analysis of <i>leu3</i> mutant grown in either limited ethanol or limited ammonium media
. <i>Leu3p</i> regulates a gene involved in nitrogen assimilation and six genes involved 
in branched chain amino acid metabolism. Results provide insight into the role of <i>Leu3p
</i> in gene regulation.

PMID: 15949974

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Boer05.filter.flt.knn.avg.div.log.pcl	<i>leu3</i> mutant expression profiles	12	carbon utilization|nitrogen utilization|transcription	2

File last modified: Thu Apr 25 2024

---


# Brauer_2005_PMID_15758028

Citation: Brauer MJ, Saldanha AJ, Dolinski K, Botstein D. 2005. Homeostatic adjustment and metabolic
remodeling in glucose-limited yeast cultures.

Full Description: Time course of batch growth. Reference (channel 1) was a culture grown in MD medium with 2
.4 g/L glucose, with 5 slpm air-flow, stirring at 400 rpm and a constant 300C temperature
, dilution 0.25 volumes/hour. The experimental (channel 2) time course samples were grown 
in batch for the indicated time. The Low-D chemostat vs. High-D chemostat hybridization's 
channel 2 sample was grown in a chemostat, with a dilution rate of 0.05 volumes/hour; it 
is most similar to the time course samples taken between 8 and 9 hours.

PMID: 15758028

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Brauer05_batch1.flt.knn.avg.pcl	Diauxic shift time course (Batch1)	13	carbon utilization|diauxic shift|fermentation|respiration	2

File last modified: Thu Apr 25 2024
2010.Brauer05_batch2.flt.knn.avg.pcl	Diauxic shift time course (Batch2)	7	carbon utilization|diauxic shift|fermentation|respiration	2

File last modified: Thu Apr 25 2024

---


# Brem_2005_PMID_15659551

Citation: Brem RB, Kruglyak L. 2005. The landscape of genetic complexity across 5,700 gene 
expression traits in yeast.

Full Description: Expression profiling of parental strains and 112 haploid progenies from a cross of strain 
BY4716 and the wild wine strain RM11-1a. Results used to find linkage between gene 
expression levels, which are treated as quantitative traits, and genetic markers.

PMID: 15659551

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Brem05_orig.flt.knn.avg.pcl	Genetic variation in gene expression among parents and progenies	131	QTLs	2

File last modified: Thu Apr 25 2024

---


# Caba_2005_PMID_15878181

Citation: Caba E, Dickinson DA, Warnes GR, Aubrecht J. 2005. Differentiating mechanisms of toxicity 
using global gene expression analysis in Saccharomyces cerevisiae.

Full Description: BY4730 cells treated with DNA-reactive compounds cisplatin, methyl methanesulfonate (MMS
), and bleomycin to induce genotoxic stress. Effect of genotoxic compounds compared to 
cytotoxic compounds, NaCl and ethanol.

PMID: 15878181

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Caba05.filter.flt.knn.avg.div.log.pcl	Genotoxic stress	24	DNA damage stimulus|osmotic stress|stress	1

File last modified: Thu Apr 25 2024

---


# Dion_2005_PMID_15795371

Citation: Dion MF, Altschuler SJ, Wu LF, Rando OJ. 2005. Genomic characterization reveals a simple 
histone H4 acetylation code.

Full Description: The histone code hypothesis holds that covalent posttranslational modifications of histone
tails are interpreted by the cell to yield a rich combinatorial transcriptional output. 
This hypothesis has been the subject of active debate in the literature. Here, we 
investigated the combinatorial complexity of the acetylation code at the four lysine 
residues of the histone H4 tail in budding yeast. We constructed yeast strains carrying 
all 15 possible combinations of mutations among lysines 5, 8, 12, and 16 to arginine in 
the histone H4 tail, mimicking positively charged, unacetylated lysine states, and 
characterized the resulting genome-wide changes in gene expression by using DNA 
microarrays. Only the lysine 16 mutation had specific transcriptional consequences 
independent of the mutational state of the other lysines (affecting approximately 100 
genes). In contrast, for lysines 5, 8, and 12, expression changes were due to nonspecific
, cumulative effects seen as increased transcription correlating with an increase in the 
total number of mutations (affecting approximately 1,200 genes). Thus, acetylation of 
histone H4 is interpreted by two mechanisms: a specific mechanism for lysine 16 and a 
nonspecific, cumulative mechanism for lysines 5, 8, and 12.

PMID: 15795371

GEO ID: GSE2324

PCL filename	short description	# conditions	tags	# channels
GSE2324_setA_family.pcl	Genomic characterization reveals a simple histone H4 acetylation code.	55	histone modification	2

File last modified: Thu Apr 25 2024

---


# Eriksson_2005_PMID_16199888

Citation: Eriksson PR, Mendiratta G, McLaughlin NB, Wolfsberg TG, Marino-Ramirez L, Pompa TA, 
Jainerin M, Landsman D, Shen CH, Clark DJ. 2005. Global regulation by the yeast Spt10 
protein is mediated through chromatin structure and the histone upstream activating 
sequence elements.

Full Description: Expression profiling of mutants lacking <i>Spt10p</i>. <i>Spt10p</i> binds the upstream 
activating sequences in all major core histone promoters. Results provide insight into the
mechanism by which <i>Spt10p</i> acts as a global regulator of transcription.

PMID: 16199888

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Eriksson05.filter.flt.knn.avg.div.log.pcl	<i>spt10</i> global transcription regulator null mutant	6	chromatin organization|histone modification	2

File last modified: Thu Apr 25 2024

---


# Gardner_2005_PMID_15797381

Citation: Gardner RG, Nelson ZW, Gottschling DE. 2005. Degradation-mediated protein quality control 
in the nucleus.

Full Description: Protein quality control degradation systems rid the cell of aberrant proteins, preventing 
detrimental effects on normal cellular function. Although such systems have been 
identified in most subcellular compartments, none have been found in the nucleus. Here, we
report the discovery of such a system in Saccharomyces cerevisiae. It is defined by San1p
, a ubiquitin-protein ligase that, in conjunction with the ubiquitin-conjugating enzymes 
Cdc34p and Ubc1p, targets four distinct mutant nuclear proteins for ubiquitination and 
destruction by the proteasome. San1p has exquisite specificity for aberrant proteins and 
does not target the wild-type versions of its mutant substrates. San1p is nuclear 
localized and requires nuclear localization for function. Loss of SAN1 results in a 
chronic stress response, underscoring its role of protein quality control in the cell. We 
propose that San1p-mediated degradation acts as the last line of proteolytic defense 
against the deleterious accumulation of aberrant proteins in the nucleus and that 
analogous systems exist in other eukaryotes.

PMID: 15797381

GEO ID: GSE2159

PCL filename	short description	# conditions	tags	# channels
GSE2159_setA_family.pcl	Degradation-mediated protein quality control in the nucleus.	12	nitrogen utilization|proteolysis|ubiquitin or ULP modification	2

File last modified: Thu Apr 25 2024

---


# Gardner_2005_PMID_15988024

Citation: Gardner RG, Nelson ZW, Gottschling DE. 2005. Ubp10/Dot4p regulates the persistence of 
ubiquitinated histone H2B: distinct roles in telomeric silencing and general chromatin.

Full Description: We previously discovered that the ubiquitin protease Ubp10/Dot4p is important for 
telomeric silencing through its interaction with Sir4p. However, the mechanism of Ubp10p 
action was unknown. We now provide evidence that Ubp10p removes ubiquitin from histone H2B
; cells with UBP10 deleted have increased steady-state levels of H2B ubiquitination. As a 
consequence, ubp10delta cells also have increased steady-state levels of histone H3 Lys4 
and Lys79 methylation. Consistent with its role in silencing, Ubp10p is preferentially 
localized to silent chromatin where its ubiquitin protease activity maintains low levels 
of H3 Lys4 and Lys79 methylation to allow optimal Sir protein binding to telomeres and 
global telomeric silencing. The ubiquitin protease Ubp8p has also been shown to remove 
ubiquitin from H2B, and ubp8delta cells have increased steady-state levels of H2B 
ubiquitination similar to those in ubp10delta cells. Unlike ubp10delta cells, however, 
ubp8delta cells do not have increased steady-state levels of H3 Lys4 and Lys79 methylation
, nor is telomeric silencing affected. Despite their separate functions in silencing and 
SAGA-mediated transcription, respectively, deletion of both UBP10 and UBP8 results in a 
synergistic increase in the steady-state levels of H2B ubiquitination and in the number of
genes with altered expression, indicating that Ubp10p and Ubp8p likely overlap in some of 
their target chromatin regions. We propose that Ubp10p and Ubp8p are the only ubiquitin 
proteases that normally remove monoubiquitin from histone H2B and, while there are regions
of the genome to which each is specifically targeted, both combine to regulate the global 
balance of H2B ubiquitination.

PMID: 15988024

GEO ID: GSE2330

PCL filename	short description	# conditions	tags	# channels
GSE2330_setA_family.pcl	ubp10 C371S/dot4-1 and ubp10-delta94-250/dot4-5 ubiquitin protease mutants	24	histone modification|ubiquitin or ULP modification	2

File last modified: Thu Apr 25 2024
GSE2329_setA_family.pcl	ubp10/dot4, ubp8 single and ubp10/dot4 ubp8 double null ubiquitin protease mutants	18	histone modification|ubiquitin or ULP modification	2

File last modified: Thu Apr 25 2024

---


# Ghazal_2005_PMID_15798187

Citation: Ghazal G, Ge D, Gervais-Bird J, Gagnon J, Abou Elela S. 2005. Genome-wide prediction and 
analysis of yeast RNase III-dependent snoRNA processing signals.

Full Description: In Saccharomyces cerevisiae, the maturation of both pre-rRNA and pre-small nucleolar RNAs
(pre-snoRNAs) involves common factors, thereby providing a potential mechanism for the 
coregulation of snoRNA and rRNA synthesis. In this study, we examined the global impact of
the double-stranded-RNA-specific RNase Rnt1p, which is required for pre-rRNA processing, 
on the maturation of all known  snoRNAs. In silico searches for Rnt1p cleavage signals, 
and genome-wide analysis  of the Rnt1p-dependent expression profile, identified seven new 
Rnt1p substrates. Interestingly, two of the newly identified Rnt1p-dependent snoRNAs, 
snR39 and snR59, are located in the introns of the ribosomal protein genes RPL7A and RPL7B
. In vitro and in vivo experiments indicated that snR39 is normally processed from  the 
lariat of RPL7A, suggesting that the expressions of RPL7A and snR39 are linked. In 
contrast, snR59 is produced by a direct cleavage of the RPL7B pre-mRNA, indicating that a 
single pre-mRNA transcript cannot be spliced to produce a mature RPL7B mRNA and processed 
by Rnt1p to produce a mature snR59 simultaneously. The results presented here reveal a new
role of yeast RNase III in the processing of intron-encoded snoRNAs that permits 
independent regulation of the host mRNA and its associated snoRNA.

PMID: 15798187

GEO ID: GSE65111.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE65111.final.pcl	Genome-wide prediction and analysis of yeast RNase III-dependent snoRNA processing signals.	8	mRNA processing	1

File last modified: Thu Apr 25 2024

---


# Iwahashi_2005_PMID_15876434

Citation: Iwahashi H, Odani M, Ishidou E, Kitagawa E. 2005. Adaptation of Saccharomyces cerevisiae 
to high hydrostatic pressure causing growth inhibition.

Full Description: Genome-wide mRNA expression profiles of Saccharomyces cerevisiae growing under hydrostatic
pressure were characterized. We selected a hydrostatic pressure of 30 MPa at 25 degrees C 
because yeast cells were able to grow under these conditions, while cell size and 
complexity were increased after decompression. Functional characterization of pressure-
induced genes suggests that genes involved in protein metabolism and membrane metabolism 
were induced. The response to 30 MPa was significantly different from that observed under 
lethal conditions because protein degradation was not activated under 30 MPa pressure. 
Strongly induced genes those that contribute to membrane metabolism and which are also 
induced by  detergents, oils, and membrane stabilizers.

PMID: 15876434

GEO ID: GSE2526.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE2526.final.pcl	Adaptation of Saccharomyces cerevisiae to high hydrostatic pressure causing growth inhibition.	8	stress	2

File last modified: Thu Apr 25 2024

---


# Jansen_2005_PMID_15870473

Citation: Jansen ML, Diderich JA, Mashego M, Hassane A, de Winde JH, Daran-Lapujade P, Pronk JT. 
2005. Prolonged selection in aerobic, glucose-limited chemostat cultures of Saccharomyces 
cerevisiae causes a partial loss of glycolytic capacity.

Full Description: Prolonged cultivation of Saccharomyces cerevisiae in aerobic, glucose-limited chemostat 
cultures (dilution rate, 0.10 h(-1)) resulted in a progressive decrease of the residual 
glucose concentration (from 20 to 8 mg l(-1) after 200 generations). This increase in the 
affinity for glucose was accompanied by a fivefold decrease of fermentative capacity, and 
changes in cellular morphology. These phenotypic changes were retained when single-cell 
isolates from prolonged cultures were used to inoculate fresh chemostat cultures, 
indicating that genetic changes were involved. Kinetic analysis of glucose transport in an
'evolved' strain revealed a decreased Km, while Vmax was slightly increased relative to 
the parental strain. Apparently, fermentative capacity in the evolved strain was not 
controlled by glucose uptake. Instead, enzyme assays in cell extracts of the evolved 
strain revealed strongly decreased capacities of enzymes in the lower part of glycolysis. 
This decrease was corroborated by genome-wide transcriptome analysis using DNA microarrays
. In aerobic batch cultures on 20 g glucose l(-1), the specific growth rate of the evolved
strain was lower than that of the parental strain (0.28 and 0.37 h(-1), respectively). 
Instead of the characteristic instantaneous production of ethanol that is observed when 
aerobic, glucose-limited cultures of wild-type S. cerevisiae are exposed to excess glucose
, the evolved strain exhibited a delay of approximately 90 min before aerobic ethanol 
formation set in. This study demonstrates that the effects of selection in glucose-limited
chemostat cultures extend beyond glucose-transport kinetics. Although extensive 
physiological analysis offered insight into the underlying cellular processes, the 
evolutionary 'driving force' for several of the observed changes remains to be elucidated.

PMID: 15870473

GEO ID: GSE8898

PCL filename	short description	# conditions	tags	# channels
GSE8898_setA_family.pcl	Prolonged selection in aerobic, glucose-limited chemostat cultures of Saccharomyces cerevisiae causes a partial loss of glycolytic capacity.	6	evolution|fermentation|respiration	1

File last modified: Thu Apr 25 2024

---


# Kitagawa_2005_PMID_16377885

Citation: Kitagawa E, Akama K, Iwahashi H. 2005. Effects of iodine on global gene expression in 
Saccharomyces cerevisiae.

Full Description: It is well documented that iodine kills microorganisms with a broad spectrum, but a 
systematic study of its mechanism of action has not yet been reported. Here we found the 
action of iodine on gene expression level, using the yeast Saccharomyces cerevisiae with a
DNA microarray. It was found that, like antimicrobial activity, iodine causes an immediate
and dose-dependent (0.5 mM, 0.75 mM and 1 mM) transcriptional alteration in yeast cells. 
The effects of iodine continued after the first immediate response. Genes for c-compound 
and carbohydrate metabolism, for energy, and for cell rescue were continuously up-
regulated. On the other hand, genes related to protein fate were induced especially at 0.5
h. The gene expression profile at 0.5 h was significantly different from that of a longer 
iodine exposed condition. The main reaction at 0.5 h after iodine addition might be due to
oxidative toxicity, and the profile at 0.5 h was similar to that of an agricultural 
bactericide.

PMID: 16377885

GEO ID: GSE9401

PCL filename	short description	# conditions	tags	# channels
GSE9401_setA_family.pcl	Effects of iodine on global gene expression in Saccharomyces cerevisiae.	27	chemical stimulus	2

File last modified: Thu Apr 25 2024

---


# Lai_2005_PMID_15870279

Citation: Lai LC, Kosorukoff AL, Burke PV, Kwast KE. 2005. Dynamical remodeling of the transcriptome
during short-term anaerobiosis in Saccharomyces cerevisiae: differential response and role
of Msn2 and/or Msn4 and  other factors in galactose and glucose media.

Full Description: In contrast to previous steady-state analyses of the O(2)-responsive transcriptome, here 
we examined the dynamics of the response to short-term anaerobiosis (2 generations) in 
both catabolite-repressed (glucose) and derepressed (galactose) cells, assessed the 
specific role that Msn2 and Msn4 play in mediating the response, and identified gene 
networks using a novel clustering  approach. Upon shifting cells to anaerobic conditions 
in galactose medium, there  was an acute ( approximately 10 min) yet transient (<45 min) 
induction of Msn2- and/or Msn4-regulated genes associated with the remodeling of reserve 
energy and  catabolic pathways during the switch from mixed respiro-fermentative to 
strictly  fermentative growth. Concomitantly, MCB- and SCB-regulated networks associated 
with the G(1)/S transition of the cell cycle were transiently down-regulated along with 
rRNA processing genes containing PAC and RRPE motifs. Remarkably, none of these gene 
networks were differentially expressed when cells were shifted in glucose, suggesting that
a metabolically derived signal arising from the abrupt cessation of respiration, rather 
than O(2) deprivation per se, elicits this stress response. By approximately 0.2 
generation of anaerobiosis in both media, more chronic, heme-dependent effects were 
observed, including the down-regulation of Hap1-regulated networks, derepression of Rox1-
regulated networks, and activation of Upc2-regulated ones. Changes in these networks 
result in the functional remodeling of the cell wall, sterol and sphingolipid metabolism, 
and dissimilatory pathways required for long-term anaerobiosis. Overall, this study 
reveals that the acute withdrawal of oxygen can invoke a metabolic state-dependent stress 
response but that acclimatization to oxygen deprivation  is a relatively slow process 
involving complex changes primarily in heme-regulated gene networks.

PMID: 15870279

GEO ID: GSE1879.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE1879.final.pcl	Dynamical remodeling of the transcriptome during short-term anaerobiosis in Saccharomyces cerevisiae: differential response and role of Msn2 and/or Msn4 and  other factors in galactose and glucose media.	45	carbon utilization|oxygen level alteration	2

File last modified: Thu Apr 25 2024

---


# Lee_2005_PMID_15989963

Citation: Lee A, Henras AK, Chanfreau G. 2005. Multiple RNA surveillance pathways limit aberrant 
expression of iron uptake mRNAs and prevent iron toxicity in S. cerevisiae.

Full Description: Expression profiling of isogenic strains lacking <i>Rnt1p</i>, a member of the RNase III 
family of double-stranded RNA endonucleases. Results identify potential <i>Rnt1p</i> mRNA 
targets.

PMID: 15989963

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Lee05.filter.flt.knn.avg.div.log.pcl	<i>rnt1</i> null mutant expression profile	9	RNA catabolism	2

File last modified: Thu Apr 25 2024

---


# Lu_2005_PMID_16208474

Citation: Lu L, Roberts GG, Oszust C, Hudson AP. 2005. The YJR127C/ZMS1 gene product is involved in 
glycerol-based respiratory growth of the yeast Saccharomyces cerevisiae.

Full Description: A putative yeast mitochondrial upstream activating sequence (UAS) was used in a one-hybrid
screening procedure that identified the YJR127C ORF on chromosome X. This gene was 
previously designated ZMS1 and is listed as a transcription factor  on the SGD website. 
Real time RT-PCR assays showed that expression of YJR127C/ZMS1 was glucose-repressible, 
and a deletion mutant for the gene showed a growth defect on glycerol-based but not on 
glucose- or ethanol-based medium. Real time RT-PCR analyses identified severely attenuated
transcript levels from GUT1 and GUT2 to be the source of that growth defect, the products 
of GUT1 and GUT2 are required for glycerol utilization. mRNA levels from a large group of 
mitochondria- and respiration-related nuclear genes also were shown to be attenuated in 
the deletion mutant. Importantly, transcript levels from the mitochondrial OLI1 gene, 
which has an associated organellar UAS, were attenuated  in the DeltaYJR127C mutant during
glycerol-based growth, but those from COX3 (OXI2), which lacks an associated mitochondrial
UAS, were not. Transcriptome analysis of the glycerol-grown deletion mutant showed that 
genes in several metabolic and other categories are affected by loss of this gene product
, including protein transport, signal transduction, and others. Thus, the product of 
YJR127C/ZMS1 is involved in transcriptional control for genes in both cellular genetic 
compartments, many of which specify products required for glycerol-based  growth, 
respiration, and other functions.

PMID: 16208474

GEO ID: GSE29530.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE29530.final.pcl	The YJR127C/ZMS1 gene product is involved in glycerol-based respiratory growth of the yeast Saccharomyces cerevisiae.	5	respiration	1

File last modified: Thu Apr 25 2024

---


# Matsumoto_2005_PMID_16209719

Citation: Matsumoto R, Akama K, Rakwal R, Iwahashi H. 2005. The stress response against denatured 
proteins in the deletion of cytosolic chaperones SSA1/2 is different from heat-shock 
response in Saccharomyces cerevisiae.

Full Description: BACKGROUND: A yeast strain lacking the two genes SSA1 and SSA2, which encode cytosolic 
molecular chaperones, acquires thermotolerance as well as the mild heat-shocked wild-type 
yeast strain. We investigated the genomic response at the level of mRNA expression to the 
deletion of SSA1/2 in comparison with the mild heat-shocked wild-type using cDNA 
microarray. RESULTS: Yeast cDNA microarray analysis revealed that genes involved in the 
stress response, including molecular chaperones, were up-regulated in a similar manner in 
both the ssa1/2 deletion mutant and the mild heat-shocked wild-type. Genes involved in 
protein synthesis were up-regulated in the ssa1/2 deletion mutant, but were markedly 
suppressed in the mild heat-shocked wild-type. The genes involved in ubiquitin-proteasome 
protein degradation were also up-regulated in the ssa1/2 deletion mutant, whereas the 
unfolded protein response (UPR) genes were highly expressed in the mild heat-shocked wild-
type. RT-PCR confirmed that the genes regulating protein synthesis and cytosolic protein 
degradation were up-regulated in the ssa1/2 deletion mutant. At the translational level, 
more ubiquitinated proteins and proteasomes were detected in the ssa1/2 deletion mutant, 
than in the wild-type, confirming that ubiquitin-proteasome protein degradation was up-
regulated by the deletion of SSA1/2. CONCLUSION: These results suggest that the mechanism 
for rescue of denatured proteins in the ssa1/2 deletion mutant is different from that in 
the mild heat-shocked wild-type: Activated protein synthesis in the ssa1/2 deletion mutant
supplies a deficiency of proteins by their degradation, whereas mild heat-shock induces 
UPR.

PMID: 16209719

GEO ID: GSE3316

PCL filename	short description	# conditions	tags	# channels
GSE3316_setA_family.pcl	The stress response against denatured proteins in the deletion of cytosolic chaperones SSA1/2 is different from heat-shock response in Saccharomyces cerevisiae.	12	heat shock|response to unfolded protein|stress	2

File last modified: Thu Apr 25 2024

---


# Penheiter_2005_PMID_16246724

Citation: Penheiter KL, Washburn TM, Porter SE, Hoffman MG, Jaehning JA. 2005. A posttranscriptional
role for the yeast Paf1-RNA polymerase II complex is revealed by identification of primary
targets.

Full Description: The yeast Paf1 complex (Paf1C: Paf1, Cdc73, Ctr9, Rtf1, and Leo1) is associated with RNA 
Polymerase II (Pol II) at promoters and coding regions of transcriptionally active genes, 
but transcript abundance for only a small subset of genes is altered by loss of Paf1. By 
using conditional and null alleles of PAF1 and microarrays, we determined the identity of 
both primary and secondary targets of the Paf1C. Neither primary nor secondary Paf1C 
target promoters were responsive to loss of Paf1. Instead, Paf1 loss altered poly(A) site 
utilization of primary target genes SDA1 and MAK21, resulting in increased abundance of 3
'-extended mRNAs. The 3'-extended MAK21 RNA is sensitive to nonsense-mediated decay (NMD
), as revealed by its increased abundance in the absence of Upf1. Therefore, although the 
Paf1C is associated with Pol II at initiation and during elongation, these critical Paf1-
dependent changes in transcript abundance are due to alterations in posttranscriptional 
processing.

PMID: 16246724

GEO ID: GSE3200

PCL filename	short description	# conditions	tags	# channels
GSE3200_set1_family.pcl	Expression profiles of isogenic wild type, paf1 and ctr9 null mutants	5	transcription	1

File last modified: Thu Apr 25 2024
GSE3200_set0_family.pcl	Expression profiles after repressible PAF1 shutoff; 8 hr time course	6	transcription	1

File last modified: Thu Apr 25 2024

---


# Ronald_2005_PMID_15687292

Citation: Ronald J, Akey JM, Whittle J, Smith EN, Yvert G, Kruglyak L. 2005. Simultaneous genotyping
gene-expression measurement, and detection of allele-specific expression with 
oligonucleotide arrays.

Full Description: Oligonucleotide microarrays provide a high-throughput method for exploring genomes. In 
addition to their utility for gene-expression analysis, oligonucleotide-expression arrays 
have also been used to perform genotyping on genomic DNA. Here, we show that in segregants
from a cross between two unrelated  strains of Saccharomyces cerevisiae, high-quality 
genotype data can also be obtained when mRNA is hybridized to an oligonucleotide-
expression array. We were  able to identify and genotype nearly 1000 polymorphisms at an 
error rate close to 3% in segregants and at an error rate of 7% in diploid strains, a 
performance comparable to methods using genomic DNA. In addition, we demonstrate how 
simultaneous genotyping and gene-expression profiling can reveal cis-regulatory variation 
by screening hundreds of genes for allele-specific expression. With this method, we 
discovered 70 ORFs with evidence for preferential expression of one allele in a diploid 
hybrid of two S. cerevisiae strains.

PMID: 15687292

GEO ID: GSE1975.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE1975.final.pcl	Simultaneous genotyping, gene-expression measurement, and detection of allele-specific expression with oligonucleotide arrays.	28	genome variation	1

File last modified: Thu Apr 25 2024

---


# Rosaleny_2005_PMID_16023114

Citation: Rosaleny LE, Antunez O, Ruiz-Garcia AB, Perez-Ortin JE, Tordera V. 2005. Yeast HAT1 and 
HAT2 deletions have different life-span and transcriptome phenotypes.

Full Description: HAT-B is a yeast histone acetyltransferase composed of Hat1, Hat2 and Hif1 proteins. We 
demonstrate that a hat2 mutant or a hat1hat2 double mutant, but not a hat1 mutant, have an
extended life-span. Transcriptome analysis shows that the single hat mutants are not very 
different from wild type. However, the comparison of the hat1 and hat2 transcriptomes 
shows that they are different. The hat1hat2 double mutant shows a transcriptional 
phenotype similar to that of the hat1 mutant but strongly enhanced. These results indicate
that Hat2p could have additional functions in the cell to those of Hat1p.

PMID: 16023114

GEO ID: GSE2434

PCL filename	short description	# conditions	tags	# channels
GSE2434_setA_family.pcl	Yeast HAT1 and HAT2 deletions have different life-span and transcriptome phenotypes.	15	histone modification	1

File last modified: Thu Apr 25 2024

---


# Rudra_2005_PMID_15692568

Citation: Rudra D, Zhao Y, Warner JR. 2005. Central role of Ifh1p-Fhl1p interaction in the synthesis
of yeast ribosomal proteins.

Full Description: Expression profiling of <i>fhl1</i> single deletion mutant and <i>ifh1</i> <i>fhl1</i> 
double deletion mutant. Mutants generated from W303 strain. Results indicate that <i>Ifh1p
</i> and <i>Fhl1p</i> function together to regulate the transcription of ribosomal protein
genes.

PMID: 15692568

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Rudra05.filter.flt.knn.avg.div.log.pcl	<i>fhl1</i> and <i>ifh1</i> deletion mutants	6	transcription	1

File last modified: Thu Apr 25 2024

---


# Singh_2005_PMID_16332871

Citation: Singh J, Kumar D, Ramakrishnan N, Singhal V, Jervis J, Garst JF, Slaughter SM, DeSantis AM,
Potts M, Helm RF. 2005. Transcriptional response of Saccharomyces cerevisiae to 
desiccation and rehydration.

Full Description: A transcriptional analysis of the response of Saccharomyces cerevisiae strain BY4743 to 
controlled air-drying (desiccation) and subsequent rehydration under minimal glucose 
conditions was performed. Expression of genes involved in fatty acid oxidation and the 
glyoxylate cycle was observed to increase during drying and remained in this state during 
the rehydration phase. When the BY4743 expression profile for the dried sample was 
compared to that of a commercially prepared dry active yeast, strikingly similar 
expression changes were observed. The fact that these two samples, dried by different 
means, possessed very similar transcriptional profiles supports the hypothesis that the 
response to desiccation is a coordinated event independent of the particular conditions 
involved in water removal. Similarities between stationary-phase-essential genes and those
upregulated during desiccation were also noted, suggesting commonalities in different 
routes to reduced metabolic states. Trends in extracellular and intracellular glucose and 
trehalose levels suggested that the cells were in a holding pattern during the rehydration
phase, a concept that was reinforced by cell cycle analyses. Application of a 
redescription mining algorithm suggested that sulfur metabolism is important for cell 
survival during desiccation and rehydration.

PMID: 16332871

GEO ID: GSE1313

PCL filename	short description	# conditions	tags	# channels
GSE1313_setA_family.pcl	Response to desiccation and subsequent rehydration under minimal glucose conditions; experiment 3	24	stress	1

File last modified: Thu Apr 25 2024
GSE1312_setA_family.pcl	Response to desiccation and subsequent rehydration under minimal glucose conditions; experiment 2	21	stress	1

File last modified: Thu Apr 25 2024
GSE1311_setA_family.pcl	Response to desiccation and subsequent rehydration under minimal glucose conditions; experiment 1	21	stress	1

File last modified: Thu Apr 25 2024

---


# Tai_2005_PMID_15496405

Citation: Tai SL, Boer VM, Daran-Lapujade P, Walsh MC, de Winde JH, Daran JM, Pronk JT. 2005. Two-
dimensional transcriptome analysis in chemostat cultures. Combinatorial effects of oxygen 
availability and macronutrient limitation in Saccharomyces cerevisiae.

Full Description: Expression profiling of MATa CEN.PK113-7D chemostat cultures grown either aerobically or 
anaerobically in media limited for either glucose, nitrogen, phosphorous, or sulfur. 
Results provide insight into the interaction between oxygen and nutrient responsive 
pathways.

PMID: 15496405

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Tai05.filter.flt.knn.avg.div.log.pcl	Nutrient limitation under aerobic and anaerobic conditions	24	carbon utilization|fermentation|nitrogen utilization|oxygen level alteration|phosphorus utilization|respiration|sulfur utilization	2

File last modified: Thu Apr 25 2024

---


# Takagi_2005_PMID_15837426

Citation: Takagi Y, Masuda CA, Chang WH, Komori H, Wang D, Hunter T, Joazeiro CA, Kornberg RD. 2005
 Ubiquitin ligase activity of TFIIH and the transcriptional response to DNA damage.

Full Description: Analysis of <i>Ssl1</i> mutant for a subunit of TFIIH treated with methyl methanesulfonate
(MMS). <i>Ssl1</i> mutant contains C406A substitution and is sensitive to DNA damaging 
agents such as MMS.

PMID: 15837426

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Takagi05.filter.flt.knn.avg.div.log.pcl	<i>Ssl1</i> mutant for a subunit of TFIIH response to methyl methanesulfonate	12	DNA damage stimulus|transcription|ubiquitin or ULP modification	2

File last modified: Thu Apr 25 2024

---


# Tamaki_2005_PMID_15743410

Citation: Tamaki H, Yun CW, Mizutani T, Tsuzuki T, Takagi Y, Shinozaki M, Kodama Y, Shirahige K, 
Kumagai H. 2005. Glucose-dependent cell size is regulated by a G protein-coupled receptor 
system in yeast Saccharomyces cerevisiae.

Full Description: In the yeast, Saccharomyces cerevisiae, cell size is affected by the kind of carbon source
in the medium. Here, we present evidence that the Gpr1 receptor and Gpa2 Galpha subunit 
are required for both maintenance and modulation of cell size in response to glucose. In 
the presence of glucose, mutants lacking GPR1 or GPA2  gene showed smaller cells than the 
wild-type strain. Physiological studies revealed that protein synthesis rate was reduced 
in the mutant strains indicating that reduced growth rate, while the level of mRNAs for 
CLN1, 2 and 3 was not affected in all strains. Gene chip analysis also revealed a down-
regulation in the expression of genes related to biosynthesis of not only protein but also
other cellular component in the mutant strains. We also show that GPR1 and GPA2 are 
required for a rapid increase in cell size in response to glucose. Wild-type  cells grown 
in ethanol quickly increased in size by addition of glucose, while little change was 
observed in the mutant strains, in which glucose-dependent cell cycle arrest caused by 
CLN1 repression was somewhat alleviated. Our study indicates that the yeast G-protein 
coupled receptor system consisting of Gpr1 and Gpa2 regulates cell size by affecting both 
growth rate and cell division.

PMID: 15743410

GEO ID: GSE970.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE970.final.pcl	Glucose-dependent cell size is regulated by a G protein-coupled receptor system in yeast Saccharomyces cerevisiae.	6	cell morphogenesis	1

File last modified: Thu Apr 25 2024

---


# Tu_2005_PMID_16254148

Citation: Tu BP, Kudlicki A, Rowicka M, McKnight SL. 2005. Logic of the yeast metabolic cycle: 
temporal compartmentalization of cellular processes.

Full Description: Budding yeast grown under continuous, nutrient-limited conditions exhibit robust, highly 
periodic cycles in the form of respiratory bursts. Microarray studies reveal that over 
half of the yeast genome is expressed periodically during these metabolic cycles. Genes 
encoding proteins having a common function exhibit similar temporal expression patterns, 
and genes specifying functions associated with energy and metabolism tend to be expressed 
with exceptionally robust periodicity. Essential cellular and metabolic events occur in 
synchrony with the metabolic cycle, demonstrating that key processes in a simple 
eukaryotic cell are compartmentalized in time.

PMID: 16254148

GEO ID: GSE3431

PCL filename	short description	# conditions	tags	# channels
GSE3431_setA_family.pcl	Logic of the yeast metabolic cycle: temporal compartmentalization of cellular processes.	36	carbon utilization|respiration	2

File last modified: Thu Apr 25 2024

---


# Wyers_2005_PMID_15935759

Citation: Wyers F, Rougemaille M, Badis G, Rousselle JC, Dufour ME, Boulay J, Regnault B, Devaux F, 
Namane A, Seraphin B, Libri D, Jacquier A. 2005. Cryptic pol II transcripts are degraded 
by a nuclear quality control pathway involving a new poly(A) polymerase.

Full Description: Since detection of an RNA molecule is the major criterion to define transcriptional 
activity, the fraction of the genome that is expressed is generally considered to parallel
the complexity of the transcriptome. We show here that several supposedly silent 
intergenic regions in the genome of S. cerevisiae are actually transcribed by RNA 
polymerase II, suggesting that the expressed fraction of the genome is higher than 
anticipated. Surprisingly, however, RNAs originating from these regions are rapidly 
degraded by the combined action of the exosome and a new poly(A) polymerase activity that 
is defined by the Trf4 protein and one of two RNA binding proteins, Air1p or Air2p. We 
show that such a polyadenylation-assisted degradation mechanism is also responsible for 
the degradation of several Pol I and Pol III transcripts. Our data strongly support the 
existence of a posttranscriptional quality control mechanism limiting inappropriate 
expression of genetic information.

PMID: 15935759

GEO ID: GSE2579.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE2579.final.pcl	Cryptic pol II transcripts are degraded by a nuclear quality control pathway involving a new poly(A) polymerase.	6	RNA processing and metabolism	1

File last modified: Thu Apr 25 2024

---


# Yamamoto_2005_PMID_15647283

Citation: Yamamoto A, Mizukami Y, Sakurai H. 2005. Identification of a novel class of target genes 
and a novel type of binding sequence of heat shock transcription factor in Saccharomyces 
cerevisiae.

Full Description: Analysis of a heat shock transcription factor 1 (<i>HSF1</i>) temperature sensitive mutant
strain subjected to heat stress at 33 degrees C. <i>HSF1</i> mutant contains an arginine 
to serine and a phenylalanine to serine substitution at residues 206 and 256 respectively
. Results identify novel targets of <i>HSF1</i

>..

PMID: 15647283

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Yamamoto05.filter.flt.knn.avg.div.log.pcl	Heat shock transcription factor 1 mutant response to heat stress	4	heat shock|transcription	2

File last modified: Thu Apr 25 2024

---


# Ye_2005_PMID_15808514

Citation: Ye J, Ai X, Eugeni EE, Zhang L, Carpenter LR, Jelinek MA, Freitas MA, Parthun MR. 2005. 
Histone H4 lysine 91 acetylation a core domain modification associated with chromatin 
assembly.

Full Description: The acetylation of the NH2-terminal tail of histone H4 by type B histone 
acetyltransferases (HATs) is involved in the process of chromatin assembly. Histone H4 
associated with a nuclear type B HAT complex contains modifications in its globular core 
domain as well. In particular, acetylation was found at lysine  91. A mutation that alters
this residue, which lies in the interface between histone H3/H4 tetramers and H2A/H2B 
dimers, confers phenotypes consistent with defects in chromatin assembly such as 
sensitivity to DNA damaging agents and derepression and alteration of silent chromatin 
structure. In addition, this mutation destabilizes the histone octamer, leading to defects
in chromatin structure. These results indicate an important role for histone modifications
outside the NH2-tail domains in the processes of chromatin assembly, DNA repair,  and 
transcriptional silencing.

PMID: 15808514

GEO ID: GSE2349.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE2349.final.pcl	Histone H4 lysine 91 acetylation a core domain modification associated with chromatin assembly.	6	histone modification|chromatin organization	1

File last modified: Thu Apr 25 2024

---


# Yeang_2005_PMID_15998451

Citation: Yeang CH, Mak HC, McCuine S, Workman C, Jaakkola T, Ideker T. 2005. Validation and 
refinement of gene-regulatory pathways on a network of physical interactions.

Full Description: Study expression effects of single transcription factor deletions.

PMID: 15998451

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.Yeang05.flt.knn.avg.pcl	Transcription factor deletions	7	transcription	2

File last modified: Thu Apr 25 2024

---


# Alper_2006_PMID_17158319

Citation: Alper H, Moxley J, Nevoigt E, Fink GR, Stephanopoulos G. 2006. Engineering yeast 
transcription machinery for improved ethanol tolerance and production.

Full Description: Global transcription machinery engineering (gTME) is an approach for reprogramming gene 
transcription to elicit cellular phenotypes important for technological applications. Here
we show the application of gTME to Saccharomyces cerevisiae for improved glucose/ethanol 
tolerance, a key trait for many biofuels programs. Mutagenesis of the transcription factor
Spt15p and selection led to dominant mutations that conferred increased tolerance and more
efficient glucose conversion to ethanol. The desired phenotype results from the combined 
effect of three separate mutations in the SPT15 gene [serine substituted for phenylalanine
(Phe(177)Ser) and, similarly, Tyr(195)His, and Lys(218)Arg]. Thus, gTME can provide a 
route to complex phenotypes that are not readily accessible by traditional methods.

PMID: 17158319

GEO ID: GSE5185

PCL filename	short description	# conditions	tags	# channels
GSE5185_setA_family.pcl	Engineering yeast transcription machinery for improved ethanol tolerance and production.	12	stress|transcription	1

File last modified: Thu Apr 25 2024

---


# Anderson_2006_PMID_16896209

Citation: Anderson JB, Ricker N, Sirjusingh C. 2006. Antagonism between two mechanisms of antifungal
drug resistance.

Full Description: This study tested for interaction between two independently evolved mechanisms of 
fluconazole resistance in Saccharomyces cerevisiae. One set of strains was from a 400-
generation evolution experiment, during which the concentration of fluconazole was 
increased from 16 to 256 microg/ml in four increments. At 100 generations, populations 
became fixed for resistance mutations in either of two transcriptional regulators, PDR1 or
PDR3. At 400 generations, replicate populations became fixed for another resistance 
mutation in UNK1, an unmapped gene further increasing resistance. Another genotype used in
this study came from a population placed initially in 128 microg/ml of fluconazole; this 
environment selects for resistance through loss of function at ERG3, resulting in altered 
sterol metabolism. Mutant strains carrying PDR1(r) or PDR3(r) were crossed with the erg3(r
) mutant strain, and the doubly mutant, haploid offspring were identified. The double-
mutant strains grew less well than the parent strains at all concentrations of fluconazole
tested. In genome-wide assays of gene expression, several ABC transporter genes that were 
overexpressed in one parent and several ERG genes that were overexpressed in the other 
parent were also overexpressed in the double mutants. Of the 43 genes that were 
consistently overexpressed in the PDR1(r) parents at generation 100, however, 31 were not 
consistently overexpressed in the double mutants. Of these 31 genes, 30 were also not 
consistently overexpressed after a further 300 generations of evolution in the PDR1(r) 
parent populations. The two independently evolved mechanisms of fluconazole resistance are
strongly antagonistic to one another.

PMID: 16896209

GEO ID: GSE4261

PCL filename	short description	# conditions	tags	# channels
GSE4261_setA_family.pcl	Antagonism between two mechanisms of antifungal drug resistance.	26	chemical stimulus|evolution	2

File last modified: Thu Apr 25 2024

---


# Angell_2006_PMID_17185230

Citation: Angell S, Bench BJ, Williams H, Watanabe CM. 2006. Pyocyanin isolated from a marine 
microbial population: synergistic production between two distinct bacterial species and 
mode of action.

Full Description: Marine microbial populations collected from the Hawaiian Islands were screened for 
antimicrobial activity. A blue metabolite was identified from mixed cell cultures, but 
production was not evident in pure cultures. Experiments designed to probe the synergistic
role of the microorganisms are presented. Full characterization of the blue natural 
product, pyocyanin, is provided including corrections made to 1H and 13C-NMR assignments 
of the molecule misreported in the chemical literature and yeast transcriptome analysis. 
The transcriptional effects were consistent with the compound's purported role as an 
inducer of oxidative stress and damage and illustrates the overall potential of the method
to reveal the primary biological/cellular effects of a natural product. The experiments 
outlined here might serve as a general paradigm for identification of natural products 
arising from microbial communities and investigation of their respective interactions.

PMID: 17185230

GEO ID: GSE6185

PCL filename	short description	# conditions	tags	# channels
GSE6185_setA_family.pcl	Pyocyanin isolated from a marine microbial population: synergistic production between two distinct bacterial species and mode of action.	6	chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Aragon_2006_PMID_16507144

Citation: Aragon AD, Quinones GA, Thomas EV, Roy S, Werner-Washburne M. 2006. Release of extraction-
resistant mRNA in stationary phase Saccharomyces cerevisiae produces a massive increase in
transcript abundance in response to stress.

Full Description: BACKGROUND: As carbon sources are exhausted, Saccharomyces cerevisiae cells exhibit 
reduced metabolic activity and cultures enter the stationary phase. We asked whether cells
in stationary phase cultures respond to additional stress at the level of transcript 
abundance. RESULTS: Microarrays were used to quantify changes in transcript abundance in 
cells from stationary phase cultures in response to stress. More than 800 mRNAs increased 
in abundance by one minute after oxidative stress. A significant number of these mRNAs 
encode proteins involved in stress responses. We tested whether mRNA increases were due to
new transcription, rapid poly-adenylation of message (which would not be detected by 
microarrays), or potential release of mature mRNA present in the cell but resistant to 
extraction during RNA isolation. Examination of the response to oxidative stress in an RNA
polymerase II mutant, rpb1-1, suggested that new transcription was not required. 
Quantitative RT-PCR analysis of a subset of these transcripts further suggested that the 
transcripts present in isolated total RNA from stationary phase cultures were 
polyadenylated. In contrast, over 2,000 transcripts increased after protease treatment of 
cell-free lysates from stationary phase but not exponentially growing cultures. Different 
subsets of transcripts were released by oxidative stress and temperature upshift, 
suggesting that mRNA release is stress-specific. CONCLUSIONS: Cells in stationary phase 
cultures contain a large number of extraction-resistant mRNAs in a protease-labile, 
rapidly releasable form. The transcript release appears to be stress-specific. We 
hypothesize that these transcripts are associated with P-bodies.

PMID: 16507144

GEO ID: GSE3729

PCL filename	short description	# conditions	tags	# channels
GSE3729_set6_family.pcl	Response of stationary phase cells to oxidative stress; 35 min time course, 1 min intervals	78	chemical stimulus|oxidative stress|stationary phase maintenance|stress	1

File last modified: Thu Apr 25 2024
GSE3729_set5_family.pcl	Effect of various proteases on RNA extraction in oxidatively stressed stationary phase cells	28	chemical stimulus|oxidative stress|stationary phase maintenance|stress	1

File last modified: Thu Apr 25 2024
GSE3729_set4_family.pcl	Response of stationary phase cells to oxidative stress after RNAPII (rpb1-1) inactivation	12	chemical stimulus|oxidative stress|stationary phase maintenance|stress	1

File last modified: Thu Apr 25 2024
GSE3729_set3_family.pcl	Effect of hot phenol extraction vs proteinase K treatment during RNA isolation in stationary phase cells	6	stationary phase maintenance	1

File last modified: Thu Apr 25 2024
GSE3729_set2_family.pcl	Response of stationary phase cells to temperature upshift; 8 hour time course	32	heat shock|stationary phase maintenance|stress	1

File last modified: Thu Apr 25 2024
GSE3729_set1_family.pcl	Response of stationary phase cells to oxidative stress; 8 hour time course, 30 min intervals	38	chemical stimulus|oxidative stress|stationary phase maintenance|stress	1

File last modified: Thu Apr 25 2024
GSE3729_set0_family.pcl	Response of stationary phase cells to oxidative stress; 3 min time course, 10 sec intervals	25	chemical stimulus|oxidative stress|stationary phase maintenance|stress	1

File last modified: Thu Apr 25 2024

---


# Arigo_2006_PMID_16973436

Citation: Arigo JT, Eyler DE, Carroll KL, Corden JL. 2006. Termination of cryptic unstable 
transcripts is directed by yeast RNA-binding proteins Nrd1 and Nab3.

Full Description: Studies of yeast transcription have revealed the widespread distribution of intergenic RNA
polymerase II transcripts. These cryptic unstable transcripts (CUTs) are rapidly degraded 
by the nuclear exosome. Yeast RNA binding proteins Nrd1 and Nab3 direct termination of sn/
snoRNAs and recently have also been implicated in premature transcription termination of 
the NRD1 gene. In this paper, we show that Nrd1 and Nab3 are required for transcription 
termination of CUTs. In nrd1 and nab3 mutants, we observe 3'-extended transcripts 
originating from CUT promoters but failing to terminate through the Nrd1- and Nab3-
directed pathway. Nrd1 and Nab3 colocalize to regions of the genome expressing antisense 
CUTs, and these transcripts require yeast nuclear exosome and TRAMP components for 
degradation. Dissection of a CUT terminator reveals a minimal element sufficient for Nrd1
- and Nab3-directed termination. These results suggest that transcription termination of 
CUTs directed by Nrd1 and Nab3 is a prerequisite for rapid degradation by the nuclear 
exosome.

PMID: 16973436

GEO ID: GSE4657.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE4657.final.pcl	Termination of cryptic unstable transcripts is directed by yeast RNA-binding proteins Nrd1 and Nab3.	4	transcriptional regulation	1

File last modified: Thu Apr 25 2024

---


# Auld_2006_PMID_16543154

Citation: Auld KL, Brown CR, Casolari JM, Komili S, Silver PA. 2006. Genomic association of the 
proteasome demonstrates overlapping gene regulatory activity with transcription factor 
substrates.

Full Description: The proteasome can regulate transcription through proteolytic processing of transcription 
factors and via gene locus binding, but few targets of proteasomal regulation have been 
identified. Using genome-wide location analysis and transcriptional profiling in 
Saccharomyces cerevisiae, we have established which genes are bound and regulated by the 
proteasome and by Spt23 and Mga2, transcription factors activated by the proteasome. We 
observed proteasome association with gene sets that are highly transcribed, controlled by 
the mating type loci, and involved in lipid metabolism. At ribosomal protein (RP) genes, 
proteasome and RNA polymerase II (RNA Pol II) binding was enriched in a proteasome mutant
, indicating a role for the proteasome in dissociating elongation complexes. The genomic 
occupancies of Spt23 and Mga2 overlapped significantly with the genes bound by the 
proteasome. Finally, the proteasome acts in two distinct ways, one dependent and one 
independent of Spt23/Mga2 cleavage, providing evidence for cooperative gene regulation by 
the proteasome and its substrates.

PMID: 16543154

GEO ID: GSE4272

PCL filename	short description	# conditions	tags	# channels
GSE4272_set0_family.pcl	Genomic association of the proteasome demonstrates overlapping gene regulatory activity with transcription factor substrates.	17	proteolysis|transcription	2

File last modified: Thu Apr 25 2024

---


# Benton_2006_PMID_17140446

Citation: Benton MG, Somasundaram S, Glasner JD, Palecek SP. 2006. Analyzing the dose-dependence of 
the Saccharomyces cerevisiae global transcriptional response to methyl methanesulfonate 
and ionizing radiation.

Full Description: BACKGROUND: One of the most crucial tasks for a cell to ensure its long term survival is 
preserving the integrity of its genetic heritage via maintenance of DNA structure and 
sequence. While the DNA damage response in the yeast Saccharomyces cerevisiae, a model 
eukaryotic organism, has been extensively studied, much remains to be elucidated about how
the organism senses and responds to different types and doses of DNA damage. We have 
measured the global transcriptional response of S. cerevisiae to multiple doses of two 
representative DNA damaging agents, methyl methanesulfonate (MMS) and gamma radiation. 
RESULTS: Hierarchical clustering of genes with a statistically significant change in 
transcription illustrated the differences in the cellular responses to MMS and gamma 
radiation. Overall, MMS produced a larger transcriptional response than gamma radiation, 
and many of the genes modulated in response to MMS are involved in protein and 
translational regulation. Several clusters of coregulated genes whose responses varied 
with DNA damaging agent dose were identified. Perhaps the most interesting cluster 
contained four genes exhibiting biphasic induction in response to MMS dose. All of the 
genes (DUN1, RNR2, RNR4, and HUG1) are involved in the Mec1p kinase pathway known to 
respond to MMS, presumably due to stalled DNA replication forks. The biphasic responses of
these genes suggest that the pathway is induced at lower levels as MMS dose increases. The
genes in this cluster with a threefold or greater transcriptional response to gamma 
radiation all showed an increased induction with increasing gamma radiation dosage. 
CONCLUSION: Analyzing genome-wide transcriptional changes to multiple doses of external 
stresses enabled the identification of cellular responses that are modulated by magnitude 
of the stress, providing insights into how a cell deals with genotoxicity.

PMID: 17140446

GEO ID: GSE6018

PCL filename	short description	# conditions	tags	# channels
GSE6018_set1_family.pcl	Dose response to gamma radiation; 3 replicates	13	radiation	1

File last modified: Thu Apr 25 2024
GSE6018_set0_family.pcl	Dose response to methyl methanesulfonate (MMS); 3 replicates	12	DNA damage stimulus	1

File last modified: Thu Apr 25 2024

---


# Boccazzi_2006_PMID_16739953

Citation: Boccazzi P, Zhang Z, Kurosawa K, Szita N, Bhattacharya S, Jensen KF, Sinskey AJ. 2006. 
Differential gene expression profiles and real-time measurements of growth parameters in 
Saccharomyces cerevisiae grown in microliter-scale bioreactors equipped with internal 
stirring.

Full Description: Combining real-time growth kinetics measurements with global gene expression analysis of 
microbial cultures is of significant value for high-throughput biological research. We 
have performed differential gene expression analysis in the eukaryotic model Saccharomyces
cerevisiae grown in galactose and glucose media in 150 muL bioreactors equipped with 
sensors for in situ and real-time measurements of optical density (OD), pH, and dissolved 
oxygen (DO). The microbioreactors were fabricated from poly(dimethylsiloxane) (PDMS) and 
poly(methyl methacrylate) (PMMA) and equipped with internal magnetic ministirrers and 
evaporation compensation by water replacement. In galactose-grown cells, the  core genes 
of the GAL operon GAL2, GAL1, GAL7, and GAL10 were upregulated at least 100-fold relative 
to glucose-grown cells. These differential gene expression levels were similar to those 
observed in large-scale culture vessels.  The increasing rate at which complete genomic 
sequences of microorganisms are becoming available offers an unprecedented opportunity for
comparative investigations of these organisms. Our results from S. cerevisiae cultures 
grown  in instrumented microbioreactors show that it is possible to integrate high-
throughput studies of growth physiology with global gene expression analysis of 
microorganisms.

PMID: 16739953

GEO ID: GSE2605.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE2605.final.pcl	Differential gene expression profiles and real-time measurements of growth parameters in Saccharomyces cerevisiae grown in microliter-scale bioreactors equipped with internal stirring.	4	carbon utilization	1

File last modified: Thu Apr 25 2024

---


# Carter_2006_PMID_16533914

Citation: Carter GW, Rupp S, Fink GR, Galitski T. 2006. Disentangling information flow in the Ras-
cAMP signaling network.

Full Description: The perturbation of signal-transduction molecules elicits genomic-expression effects that 
are typically neither restricted to a small set of genes nor uniform. Instead there are 
broad, varied, and complex changes in expression across the genome. These observations 
suggest that signal transduction is not mediated by isolated pathways of information flow 
to distinct groups of genes in  the genome. Rather, multiple entangled paths of 
information flow influence overlapping sets of genes. Using the Ras-cAMP pathway in 
Saccharomyces cerevisiae as a model system, we perturbed key pathway elements and 
collected genomic-expression data. Singular value decomposition was applied to separate 
the genome-wide transcriptional response into weighted expression components exhibited by 
overlapping groups of genes. Molecular interaction data were integrated to connect gene 
groups to perturbed signaling elements. The resulting  series of linked subnetworks maps 
multiple putative pathways of information flow  through a dense signaling network, and 
provides a set of testable hypotheses for  complex gene-expression effects across the 
genome.

PMID: 16533914

GEO ID: GSE2927.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE2927.final.pcl	Disentangling information flow in the Ras-cAMP signaling network.	20	signaling	1

File last modified: Thu Apr 25 2024

---


# Chen_2006_PMID_16397762

Citation: Chen JC, Powers T. 2006. Coordinate regulation of multiple and distinct biosynthetic 
pathways by TOR and PKA kinases in S. cerevisiae.

Full Description: The target of rapamycin (TOR) signaling pathway is an essential regulator of cell growth 
in eukaryotic cells. In Saccharomyces cerevisiae, TOR controls the expression of many 
genes involved in a wide array of distinct nutrient-responsive metabolic pathways. By 
exploring the TOR pathway under different growth conditions, we have identified novel TOR-
regulated genes, including genes required for branched-chain amino acid biosynthesis as 
well as lysine biosynthesis (LYS genes). We show that TOR-dependent control of LYS gene 
expression occurs independently from previously identified LYS gene regulators and is 
instead coupled to cAMP-regulated protein kinase A (PKA). Additional genome-wide 
expression analyses reveal that TOR and PKA coregulate LYS gene expression in a pattern 
that is remarkably similar to genes within the ribosomal protein and Ribi regulon genes 
required for ribosome biogenesis. Moreover, this pattern of coregulation is distinct from 
other clusters of TOR/PKA coregulated genes, which includes genes involved in fermentation
as well as aerobic respiration, suggesting that control of gene expression by TOR and PKA 
involves multiple modes of crosstalk. Our results underscore how multiple signaling 
pathways, general growth conditions, as well as the availability of specific nutrients 
contribute to the maintenance of appropriate patterns of gene activity in yeast.

PMID: 16397762

GEO ID: GSE3160

PCL filename	short description	# conditions	tags	# channels
GSE3160_setA_family.pcl	Coordinate regulation of multiple and distinct biosynthetic pathways by TOR and PKA kinases in S. cerevisiae.	12	amino acid utilization|protein phosphorylation|signaling|starvation	2

File last modified: Thu Apr 25 2024

---


# Chua_2006_PMID_16880382

Citation: Chua G, Morris QD, Sopko R, Robinson MD, Ryan O, Chan ET, Frey BJ, Andrews BJ, Boone C, 
Hughes TR. 2006. Identifying transcription factor functions and targets by phenotypic 
activation.

Full Description: Mapping transcriptional regulatory networks is difficult because many transcription 
factors (TFs) are activated only under specific conditions. We describe a generic strategy
for identifying genes and pathways induced by individual TFs that does not require 
knowledge of their normal activation cues. Microarray analysis of 55 yeast TFs that caused
a growth phenotype when overexpressed showed that the majority caused increased transcript
levels of genes in specific physiological categories, suggesting a mechanism for growth 
inhibition. Induced genes typically included established targets and genes with consensus 
promoter motifs, if known, indicating that these data are useful for identifying potential
new target genes and binding sites. We identified the sequence 5'-TCACGCAA as a binding 
sequence for Hms1p, a TF that positively regulates pseudohyphal growth and previously had 
no known motif. The general strategy outlined here presents a straightforward approach to 
discovery of TF activities and mapping targets that could be adapted to any organism with 
transgenic technology.

PMID: 16880382

GEO ID: GSE5499

PCL filename	short description	# conditions	tags	# channels
GSE5499_set1_family.pcl	Response to deletion of 51 non-essential transcription factors; 2 replicates	102	transcription	2

File last modified: Thu Apr 25 2024
GSE5499_set0_family.pcl	Response to overexpression of 55 different transcription factors; 2 replicates	168	transcription	2

File last modified: Thu Apr 25 2024

---


# Crisp_2006_PMID_16437160

Citation: Crisp RJ, Adkins EM, Kimmel E, Kaplan J. 2006. Recruitment of Tup1p and Cti6p regulates 
heme-deficient expression of Aft1p target genes.

Full Description: In the budding yeast Saccharomyces cerevisiae, transcription of genes encoding for the 
high-affinity iron (FET3, FTR1) and copper (CTR1) transporters does not occur in the 
absence of heme. We show that the Aft1p binding region of the FET3 promoter or the Mac1p 
binding region of the CTR1 promoter is necessary and sufficient to mediate heme-deficient 
repression. Transcription is repressed in the absence of heme, and a genetic screen 
identified Tup1p and Hda1p as being required for transcriptional repression. In contrast 
to FET3 and CTR1, Aft1p target genes ARN1 and FIT1 are transcribed in the absence of heme
. A 14 bp sequence in the ARN1 promoter is necessary and sufficient to permit 
transcription in the absence of heme. Transcription in the absence of heme required the 
presence of Cti6p to overcome the effect of Tup1p, and Cti6p was recruited to the ARN1 
promoter in the absence of heme. We hypothesize that transcription of the siderophore 
transporter ARN1 permits yeast to accumulate iron in the absence of oxygen and to deny 
iron to competing organisms.

PMID: 16437160

GEO ID: GSE3470.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE3470.final.pcl	Recruitment of Tup1p and Cti6p regulates heme-deficient expression of Aft1p target genes.	4	metal or metalloid ion stress	1

File last modified: Thu Apr 25 2024

---


# Davis_2006_PMID_16484372

Citation: Davis CA, Ares M Jr. 2006. Accumulation of unstable promoter-associated transcripts upon 
loss of the nuclear exosome subunit Rrp6p in Saccharomyces cerevisiae.

Full Description: Mutations in RRP6 result in the accumulation of aberrant polyadenylated transcripts from 
small nucleolar RNA genes. We exploited this observation to search for novel noncoding RNA
genes in the yeast genome. When RNA from rrp6Delta yeast is compared with wild-type on 
whole-genome microarrays, numerous intergenic loci exhibit an increased mutant/wild type 
signal ratio. Among these loci, we found one encoding a new C/D box small nucleolar RNA, 
as well as a surprising number that gave rise to heterogeneous Trf4p-polyadenylated RNAs 
with lengths of  approximately 250-500 nt. This class of RNAs is not easily detected in 
wild-type  cells and appears associated with promoters. Fine mapping of several such 
transcripts shows they originate near known promoter elements but do not usually  extend 
far enough to act as mRNAs, and may regulate the transcription of downstream mRNAs. Rather
than being uninformative transcriptional noise, we hypothesize that these transcripts 
reflect important features of RNA polymerase activity at the promoter. This activity is 
normally undetectable in wild-type cells because the transcripts are somehow distinguished
from true mRNAs and are degraded in an Rrp6p-dependent fashion in the nucleus.

PMID: 16484372

GEO ID: GSE3813.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE3813.final.pcl	Accumulation of unstable promoter-associated transcripts upon loss of the nuclear exosome subunit Rrp6p in Saccharomyces cerevisiae.	1	mRNA processing	2

File last modified: Thu Apr 25 2024

---


# Dubacq_2006_PMID_16328372

Citation: Dubacq C, Chevalier A, Courbeyrette R, Petat C, Gidrol X, Mann C. 2006. Role of the iron 
mobilization and oxidative stress regulons in the genomic response of yeast to hydroxyurea.

Full Description: Hydroxyurea (HU) is a specific inhibitor of ribonucleotide reductase and thus impairs dNTP
synthesis and DNA replication. The long-term transcriptional response of yeast cells to 
hydroxyurea was investigated using DNA microarrays containing all yeast coding sequences. 
We show that the redox-responsive Yap regulon and the iron-mobilization Aft regulon are 
activated in yeast cells treated with HU. Yap1 accumulates in the nucleus in response to 
HU, but HU activation of the Yap regulon was only partially dependent on Yap1 and 
yap1Delta mutants were not hypersensitive to HU. In contrast, deletion of the AFT1 and 
AFT2 transcription factor genes blocked the HU activation of a subset of the Aft regulon 
and the aft1Delta aft2Delta double mutant was hypersensitive to HU in an iron-suppressible
manner. These results highlight the importance of the redox and iron mobilization regulons
in the cellular response to HU.

PMID: 16328372

GEO ID: GSE1914

PCL filename	short description	# conditions	tags	# channels
GSE1914_setA_family.pcl	Mid-term response of wild type, yap1, aft1 and aft1 aft2 null mutants to hydroxyurea	24	DNA damage stimulus|stress	2

File last modified: Thu Apr 25 2024
GSE1915_setA_family.pcl	Long-term response of wild type cells to hydroxyurea	16	DNA damage stimulus|stress	2

File last modified: Thu Apr 25 2024

---


# Friedlander_2006_PMID_16542486

Citation: Friedlander G, Joseph-Strauss D, Carmi M, Zenvirth D, Simchen G, Barkai N. 2006. 
Modulation of the transcription regulatory program in yeast cells committed to sporulation.

Full Description: BACKGROUND: Meiosis in budding yeast is coupled to the process of sporulation, where the 
four haploid nuclei are packaged into a gamete. This differentiation process is 
characterized by a point of transition, termed commitment, when it becomes independent of 
the environment. Not much is known about the mechanisms underlying commitment, but it is 
often assumed that positive feedback loops stabilize the underlying gene-expression 
cascade. RESULTS: We describe the gene-expression program of committed cells. Sporulating 
cells were transferred back to growth medium at different stages of the process, and their
transcription response was characterized. Most sporulation-induced genes were immediately 
downregulated upon transfer, even in committed cells that continued to sporulate. Focusing
on the metabolic-related transcription response, we observed that pre-committed cells, as 
well as mature spores, responded to the transfer to growth medium in essentially the same 
way that vegetative cells responded to glucose. In contrast, committed cells elicited a 
dramatically different response. CONCLUSION: Our results suggest that cells ensure 
commitment to sporulation not by stabilizing the process, but by modulating their gene-
expression program in an active manner. This unique transcriptional program may optimize 
sporulation in an environment-specific manner.

PMID: 16542486

GEO ID: GSE3820

PCL filename	short description	# conditions	tags	# channels
GSE3820_setA_family.pcl	Sporulation time course (2-9 hrs) and return to growth in YPA for 20, 80, and 140 min; 6.4K cDNA array	27	nutrient utilization|sporulation	2

File last modified: Thu Apr 25 2024
GSE3816_set1_family.pcl	Return to growth from sporulating cultures (2-10 hrs) into YPA for 20 and 40 min; 6.4K cDNA array	27	nutrient utilization|sporulation	2

File last modified: Thu Apr 25 2024
GSE3816_set0_family.pcl	Sporulation time course; 2-10 hr time course, 1 hr intervals; 6.4K cDNA array	9	sporulation	2

File last modified: Thu Apr 25 2024
GSE3815_set1_family.pcl	Return to growth from sporulating cultures (2-9 hrs) into YPA for 20 and 40 min; 16K oligo array	26	nutrient utilization|sporulation	2

File last modified: Thu Apr 25 2024
GSE3815_set0_family.pcl	Sporulation time course; 2-9 hr time course, 1 hr intervals; 16K oligo array	8	sporulation	2

File last modified: Thu Apr 25 2024
GSE3814_set0_family.pcl	Sporulation time course; 2-9 hr time course, 1 hr intervals; 6.4K cDNA array	8	sporulation	2

File last modified: Thu Apr 25 2024
GSE3814_set1_family.pcl	Return to growth from sporulating cultures (2-9 hrs) into YPA for 20 and 40 min; 6.4K cDNA array	24	nutrient utilization|sporulation	2

File last modified: Thu Apr 25 2024

---


# Fry_2006_PMID_17163986

Citation: Fry RC, DeMott MS, Cosgrove JP, Begley TJ, Samson LD, Dedon PC. 2006. The DNA-damage 
signature in Saccharomyces cerevisiae is associated with single-strand breaks in DNA.

Full Description: BACKGROUND: Upon exposure to agents that damage DNA, Saccharomyces cerevisiae undergo 
widespread reprogramming of gene expression. Such a vast response may be due not only to 
damage to DNA but also damage to proteins, RNA, and lipids. Here the transcriptional 
response of S. cerevisiae specifically induced by DNA damage was discerned by exposing S. 
cerevisiae to a panel of three radiomimetic enediyne antibiotics (calicheamicin gamma1I, 
esperamicin A1 and neocarzinostatin) that bind specifically to DNA and generate varying 
proportions of single- and double-strand DNA breaks. The genome-wide responses were 
compared to those induced by the non-selective oxidant gamma-radiation. RESULTS: Given 
well-controlled exposures that resulted in similar and minimal cell death (~20-25%) across
all conditions, the extent of gene expression modulation was markedly different depending 
on treatment with the enediynes or gamma-radiation. Exposure to gamma-radiation resulted 
in more extensive transcriptional changes classified both by the number of genes modulated
and the magnitude of change. Common biological responses were identified between the 
enediynes and gamma-radiation, with the induction of DNA repair and stress response genes
, and the repression of ribosomal biogenesis genes. Despite these common responses, a 
fraction of the response induced by gamma radiation was repressed by the enediynes and 
vise versa, suggesting that the enediyne response is not entirely radiomimetic. Regression
analysis identified 55 transcripts with gene expression induction associated both with 
double- or single-strand break formation. The S. cerevisiae DNA damage signature genes as 
defined by Gasch et al. 1 were enriched among regulated transcripts associated with single
-strand breaks, while genes involved in cell cycle regulation were associated with double-
strand breaks. CONCLUSION: Dissection of the transcriptional response in yeast that is 
specifically signaled by DNA strand breaks has identified that single-strand breaks 
provide the signal for activation of transcripts encoding proteins involved in the DNA 
damage signature in S. cerevisiae, and double-strand breaks signal changes in cell cycle 
regulation genes.

PMID: 17163986

GEO ID: GSE5301

PCL filename	short description	# conditions	tags	# channels
GSE5301_setA_family.pcl	The DNA-damage signature in Saccharomyces cerevisiae is associated with single-strand breaks in DNA.	18	DNA damage stimulus|radiation	1

File last modified: Thu Apr 25 2024

---


# Gonzalez_2006_PMID_16973600

Citation: Gonzalez A, Ruiz A, Serrano R, Arino J, Casamayor A. 2006. Transcriptional profiling of 
the protein phosphatase 2C family in yeast provides insights into the unique functional 
roles of Ptc1.

Full Description: Type 2C protein phosphatases are encoded in Saccharomyces cerevisiae by several related 
genes (PTC1-5 and PTC7). To gain insight into the functions attributable to specific 
members of this gene family, we have investigated the transcriptional profiles of ptc1-5 
mutants. Two main patterns were obtained as follows: the one generated by the ptc1 
mutation and the one resulting from the lack of Ptc2-5. ptc4 and ptc5 profiles were quite 
similar, whereas that of ptc2 was less related to this group. Mutation of PTC1 resulted in
increased expression of numerous genes that are also induced by cell wall damage, such as 
YKL161c, SED1, or CRH1, as well as in higher amounts of active Slt2 mitogen-activated 
protein kinase, indicating that lack of the phosphatase activates the cell wall integrity 
pathway. ptc1 cells were even more sensitive than slt2 mutants to a number of cell wall-
damaging agents, and both mutations had additive effects. The sensitivity of ptc1 cells 
was not dependent on Hog1. Besides these phenotypes, we observed that calcineurin was 
hyperactivated in ptc1 cells, which were also highly sensitive to calcium ions, heavy 
metals, and alkaline pH, and exhibited a random haploid budding pattern. Remarkably, many 
of these traits are found in certain mutants with impaired vacuolar function. As ptc1 
cells also display fragmented vacuoles, we hypothesized that lack of Ptc1 would primarily 
cause vacuolar malfunction, from which other phenotypes would derive. In agreement with 
this scenario, overexpression of VPS73, a gene of unknown function involved in vacuolar 
protein sorting, largely rescues not only vacuolar fragmentation but also sensitivity to 
cell wall damage, high calcium, alkaline pH, as well as other ptc1-specific phenotypes.

PMID: 16973600

GEO ID: GSE6801

PCL filename	short description	# conditions	tags	# channels
GSE6801_setA_family.pcl	Transcriptional profiling of the protein phosphatase 2C family in yeast provides insights into the unique functional roles of Ptc1.	10	protein dephosphorylation|signaling	2

File last modified: Thu Apr 25 2024

---


# Guan_2006_PMID_17166056

Citation: Guan Q, Zheng W, Tang S, Liu X, Zinkel RA, Tsui KW, Yandell BS, Culbertson MR. 2006. 
Impact of nonsense-mediated mRNA decay on the global expression profile of budding yeast.

Full Description: Nonsense-mediated mRNA decay (NMD) is a eukaryotic mechanism of RNA surveillance that 
selectively eliminates aberrant transcripts coding for potentially deleterious proteins. 
NMD also functions in the normal repertoire of gene expression. In Saccharomyces 
cerevisiae, hundreds of endogenous RNA Polymerase II transcripts achieve steady-state 
levels that depend on NMD. For some, the decay rate is directly influenced by NMD (direct 
targets). For others, abundance is NMD-sensitive but without any effect on the decay rate
(indirect targets). To distinguish between direct and indirect targets, total RNA from 
wild-type (Nmd(+)) and mutant (Nmd(-)) strains was probed with high-density arrays across 
a 1-h time window following transcription inhibition. Statistical models were developed to
describe the kinetics of RNA decay. 45% +/- 5% of RNAs targeted by NMD were predicted to 
be direct targets with altered decay rates in Nmd(-) strains. Parallel experiments using 
conventional methods were conducted to empirically test predictions from the global 
experiment. The results show that the global assay reliably distinguished direct versus 
indirect targets. Different types of targets were investigated, including transcripts 
containing adjacent, disabled open reading frames, upstream open reading frames, and those
prone to out-of-frame initiation of translation. Known targeting mechanisms fail to 
account for all of the direct targets of NMD, suggesting that additional targeting 
mechanisms remain to be elucidated. 30% of the protein-coding targets of NMD fell into two
broadly defined functional themes: those affecting chromosome structure and behavior and 
those affecting cell surface dynamics. Overall, the results provide a preview for how 
expression profiles in multi-cellular eukaryotes might be impacted by NMD. Furthermore, 
the methods for analyzing decay rates on a global scale offer a blueprint for new ways to 
study mRNA decay pathways in any organism where cultured cell lines are available.

PMID: 17166056

GEO ID: GSE3076

PCL filename	short description	# conditions	tags	# channels
GSE3076_setA_family.pcl	Impact of nonsense-mediated mRNA decay on the global expression profile of budding yeast.	96	chemical stimulus|RNA catabolism	1

File last modified: Thu Apr 25 2024

---


# Guo_2006_PMID_16122766

Citation: Guo Y, Breeden LL, Fan W, Zhao LP, Eaton DL, Zarbl H. 2006. Analysis of cellular responses
to aflatoxin B(1) in yeast expressing human cytochrome P450 1A2 using cDNA microarrays.

Full Description: Aflatoxin B1 (AFB(1)) is a potent human hepatotoxin and hepatocarcinogen produced by the 
mold Aspergillus flavus. In human, AFB(1) is bioactivated by cytochrome P450 (CYP450) 
enzymes, primarily CYP1A2, to the genotoxic epoxide that forms N(7)-guanine DNA adducts. 
To characterize the transcriptional responses to genotoxic insults from AFB(1), a strain 
of Saccharomyces cerevisiae engineered to express human CYP1A2 was exposed to doses of AFB
(1) that resulted in minimal lethality, but substantial genotoxicity. Flow cytometric 
analysis demonstrated a dose and time dependent S phase delay under the same treatment 
conditions, indicating a checkpoint response to DNA damage. Replicate cDNA microarray 
analyses of AFB(1) treated cells showed that about 200 genes were significantly affected 
by the exposure. The genes activated by AFB(1)-treatment included RAD51, DUN1 and other 
members of the DNA damage response signature reported in a previous study with 
methylmethane sulfonate and ionizing radiation [A.P. Gasch, M. Huang, S. Metzner, D. 
Botstein, S.J. Elledge, P.O. Brown, Genomic expression responses to DNA-damaging agents 
and the regulatory role of the yeast ATR homolog Mec1p, Mol. Biol. Cell 12 (2001) 2987-
3003]. However, unlike previous studies using highly cytotoxic doses, environmental stress
response genes [A.P. Gasch, P.T. Spellman, C.M. Kao, O. Carmel-Harel, M.B. Eisen, G. Storz
, D. Botstein, P.O. Brown, Genomic expression programs in the response of yeast cells to 
environmental changes, Mol. Biol. Cell 11 (2000) 4241-4257] were largely unaffected by our
dosing regimen. About half of the transcripts affected are also known to be cell cycle 
regulated. The most strongly repressed transcripts were those encoding the histone genes 
and a group of genes that are cell cycle regulated and peak in M phase and early G1. These
include most of the known daughter-specific genes. The rapid and coordinated repression of
histones and M/G1-specific transcripts cannot be explained by cell cycle arrest, and 
suggested that there are additional signaling pathways that directly repress these genes 
in cells under genotoxic stress.

PMID: 16122766

GEO ID: GSE2420

PCL filename	short description	# conditions	tags	# channels
GSE2420_setA_family.pcl	Response to aflatoxin B1; 2 hr time course	11	chemical stimulus	2

File last modified: Thu Apr 25 2024
GSE2419_setA_family.pcl	Response to aflatoxin B1; dose response at 2 hrs	10	chemical stimulus	2

File last modified: Thu Apr 25 2024

---


# Hashikawa_2006_PMID_16361698

Citation: Hashikawa N, Mizukami Y, Imazu H, Sakurai H. 2006. Mutated yeast heat shock transcription 
factor activates transcription independently of hyperphosphorylation.

Full Description: The homotrimeric heat shock transcription factor (HSF) binds to the heat shock element of 
target genes and regulates transcription in response to various stresses. The Hsf1 protein
of Saccharomyces cerevisiae is extensively phosphorylated upon heat shock; a modification 
that is under positive regulation  by its C-terminal regulatory domain (CTM). 
Hyperphosphorylation has been implicated in gene-specific transcriptional activation. Here
, we surveyed genes whose heat shock response is reduced by a CTM mutation. The CTM is 
indispensable  for transcription via heat shock elements bound by a single Hsf1 trimer but
is dispensable for transcription via heat shock elements bound by Hsf1 trimers in a  
cooperative manner. Intragenic mutations located within or near the wing region of the 
winged helix-turn-helix DNA-binding domain suppress the temperature-sensitive growth 
phenotype associated with the CTM mutation and enable Hsf1 to activate transcription 
independently of hyperphosphorylation. Deletion of the wing partially restores the 
transcriptional defects of the unphosphorylated Hsf1. These results demonstrate a 
functional link between hyperphosphorylation and the wing region and suggest that this 
modification is involved in a conformational change of a single Hsf1 trimer to an active 
form.

PMID: 16361698

GEO ID: GSE3361.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE3361.final.pcl	Mutated yeast heat shock transcription factor activates transcription independently of hyperphosphorylation.	4	heat shock|protein phosphorylation	1

File last modified: Thu Apr 25 2024

---


# Irvin_2006_PMID_16407318

Citation: Irvin JD, Pugh BF. 2006. Genome-wide transcriptional dependence on TAF1 functional domains.

Full Description: Transcription factor IID (TFIID) plays a central role in regulating the expression of most
eukaryotic genes. Of the 14 TBP-associated factor (TAF) subunits that compose TFIID, TAF1 
is one of the largest and most functionally diverse. Yeast TAF1 can be divided into four 
regions including a putative histone acetyltransferase domain and TBP, TAF, and promoter 
binding domains. Establishing the importance of each region in gene expression through 
deletion analysis has been hampered by the cellular requirement of TAF1 for viability. To 
circumvent this limitation we introduced galactose-inducible deletion derivatives of 
previously defined functional regions of TAF1 into a temperature-sensitive taf1ts2 yeast 
strain. After galactose induction of the TAF1 mutants and temperature-induced elimination 
of the resident Taf1ts2 protein, we examined the properties and phenotypes of the mutants
, including their impact on genome-wide transcription. Virtually all TAF1-dependent genes
, which comprise approximately 90% of the yeast genome, displayed a strong dependence upon
all regions of TAF1 that were tested. This finding might reflect the need for each region 
of TAF1 to stabilize TAF1 against degradation or may indicate that all TAF1-dependent 
genes require the many activities of TAF1. Paradoxically, deletion of the region of TAF1 
that is important for promoter binding interfered with the expression of many genes that 
are normally TFIID-independent/SAGA (Spt-Ada-Gcn5-acetyltransferase)-dominated, suggesting
that this region normally prevents TAF1 (TFIID) from interfering with the expression of 
SAGA-regulated genes.

PMID: 16407318

GEO ID: GSE3006

PCL filename	short description	# conditions	tags	# channels
GSE3006_setA_family.pcl	Genome-wide transcriptional dependence on TAF1 functional domains.	14	transcription	2

File last modified: Thu Apr 25 2024

---


# Johanson_2006_PMID_16440349

Citation: Johanson K, Allen PL, Gonzalez-Villalobos RA, Baker CB, D'Elia R, Hammond TG. 2006. Gene 
expression and survival changes in Saccharomyces cerevisiae during suspension culture.

Full Description: This study explores the connection between changes in gene expression and the genes that 
determine strain survival during suspension culture, using the model eukaryotic organism, 
Saccharomyces cerevisiae. The Saccharomyces cerevisiae homozygous diploid deletion pool (
HDDP), and the BY4743 parental strain were grown for 18 h in a rotating wall vessel (RWV
), a suspension culture device optimized to minimize the delivered shear. In addition to 
the reduced shear conditions, the RWVs were also placed in a static position or in a 
shaker in order to change the amount of shear stress on the cells. Using simple linear 
regression, it was found that there were 140 differentially expressed genes for which >70
% of the variation can be explained by shear stress alone. A significant number of these 
genes are involved in catalytic activity. In the HDDP, shear stress was associated with 
significant survival changes in 15 deletion strains (R(2>) > 0.7) Interestingly, both 
analyses uncovered changes in the ribosomal protein machinery. Comparing the changes in 
gene expression and strain survival under the different shear conditions allows for the 
insights into the molecular mechanisms behind the cells response to shear stress. This in 
turn can provide information for the optimization of suspension culture.

PMID: 16440349

GEO ID: GSE2818

PCL filename	short description	# conditions	tags	# channels
GSE2818_set0_family.pcl	Gene expression and survival changes in Saccharomyces cerevisiae during suspension culture.	6	stress	1

File last modified: Thu Apr 25 2024

---


# Kelly_2006_PMID_16720269

Citation: Kelly GT, Liu C, Smith R 3rd, Coleman RS, Watanabe CM. 2006. Cellular effects induced by 
the antitumor agent azinomycin B.

Full Description: Studies on the mechanism of action of the antitumor agent azinomycin B in vitro suggest 
that the drug elicits its lethal effects by the formation of interstrand crosslinks within
the major groove of DNA. Here, we demonstrate the biological effects of the drug in vivo. 
Fluorescence imaging revealed localization of azinomycin B in the nuclear region of yeast
. Moreover, experiments with oligonucleotide microarrays examined the effects of the drug 
across the yeast transcriptome. The results demonstrated a robust DNA damage response that
supports the proposed role of the drug as a covalent DNA modifying agent. RT-PCR analysis 
validated the gene changes, and flow cytometry of azinomycin-treated yeast cells 
demonstrated a phenotypic S phase shift consistent with transcriptional effects.

PMID: 16720269

GEO ID: GSE4311

PCL filename	short description	# conditions	tags	# channels
GSE4311_setA_family.pcl	Cellular effects induced by the antitumor agent azinomycin B.	6	chemical stimulus|DNA damage stimulus	1

File last modified: Thu Apr 25 2024

---


# Kresnowati_2006_PMID_16969341

Citation: Kresnowati MT, van Winden WA, Almering MJ, ten Pierick A, Ras C, Knijnenburg TA, Daran-
Lapujade P, Pronk JT, Heijnen JJ, Daran JM. 2006. When transcriptome meets metabolome: 
fast cellular responses of yeast to sudden relief of glucose limitation.

Full Description: Within the first 5 min after a sudden relief from glucose limitation, Saccharomyces 
cerevisiae exhibited fast changes of intracellular metabolite levels and a major 
transcriptional reprogramming. Integration of transcriptome and metabolome data revealed 
tight relationships between the changes at these two levels. Transcriptome as well as 
metabolite changes reflected a major investment in two processes: adaptation from fully 
respiratory to respiro-fermentative metabolism and preparation for growth acceleration. At
the metabolite level, a severe drop of the AXP pools directly after glucose addition was 
not accompanied by any of the other three NXP. To counterbalance this loss, purine 
biosynthesis and salvage pathways were transcriptionally upregulated in a concerted manner
, reflecting a sudden increase of the purine demand. The short-term dynamics of the 
transcriptome revealed a remarkably fast decrease in the average half-life of 
downregulated genes. This acceleration of mRNA decay can be interpreted both as an 
additional nucleotide salvage pathway and an additional level of glucose-induced 
regulation of gene expression.

PMID: 16969341

GEO ID: GSE3821

PCL filename	short description	# conditions	tags	# channels
GSE3821_setA_family.pcl	When transcriptome meets metabolome: fast cellular responses of yeast to sudden relief of glucose limitation.	16	carbon utilization	1

File last modified: Thu Apr 25 2024

---


# Kuranda_2006_PMID_16925551

Citation: Kuranda K, Leberre V, Sokol S, Palamarczyk G, Francois J. 2006. Investigating the caffeine
effects in the yeast Saccharomyces cerevisiae brings new insights into the connection 
between TOR, PKC and Ras/cAMP signalling pathways.

Full Description: Caffeine is a natural purine analogue that elicits pleiotropic effects leading ultimately 
to cell's death by a largely uncharacterized mechanism. Previous works have shown that 
this drug induces a rapid phosphorylation of the Mpk1p, the final mitogen-activated 
protein (MAP) kinase of the Pkc1p-mediated cell integrity pathway. In this work, we showed
that this phosphorylation did not necessitate the main cell wall sensors Wsc1p and Mid2p, 
but was abolished upon deletion of ROM2 encoding a GDP/GTP exchange factor of Rho1p. We 
also showed that the caffeine-induced phosphorylation of Mpk1p was accompanied by a 
negligible activation of its main downstream target, the Rlm1p transcription factor. This 
result was consolidated by the finding that the loss of RLM1 had no consequence on the 
increased resistance of caffeine-treated cells to zymolyase, indicating that the cell wall
modification caused by this drug is largely independent of transcriptional activation of 
Rlm1p-regulated genes. Additionally, the transcriptional programme elicited by caffeine 
resembled that of rapamycin, a potent inhibitor of the TOR1/2 kinases. Consistent with 
this analysis, we found that the caffeine-induced phosphorylation of Mpk1p was lost in a 
tor1Delta mutant. Moreover, a tor1Delta mutant was, like mutants defective in components 
of the Pkc1p-Mpk1p cascade, highly sensitive to caffeine. However, the hypersensitivity of
a tor1 null mutant to this drug was rescued neither by sorbitol nor by adenine, which was 
found to outcompete caffeine effects specially on mutants in the PKC pathway. Altogether, 
these data indicated that Tor1 kinase is a target of caffeine, whose inhibition 
incidentally activates the Pkc1p-Mpk1p cascade, and that the caffeine-dependent phenotypes
are largely dependent on inhibition of Tor1p-regulated cellular functions. Finally, we 
found that caffeine provoked, in a Rom2p-dependent manner, a transient drop in 
intracellular levels of cAMP, that was followed by change in expression of genes 
implicated in Ras/cAMP pathway. This result may pose Rom2p as a mediator in the interplay 
between Tor1p and the Ras/cAMP pathway.

PMID: 16925551

GEO ID: GSE4049

PCL filename	short description	# conditions	tags	# channels
GSE4049_setA_family.pcl	Investigating the caffeine effects in the yeast Saccharomyces cerevisiae brings new insights into the connection between TOR, PKC and Ras/cAMP signalling pathways.	36	cell wall organization|chemical stimulus|signaling	2

File last modified: Thu Apr 25 2024

---


# Lai_2006_PMID_16963631

Citation: Lai LC, Kosorukoff AL, Burke PV, Kwast KE. 2006. Metabolic-state-dependent remodeling of 
the transcriptome in response to anoxia and subsequent reoxygenation in Saccharomyces 
cerevisiae.

Full Description: We conducted a comprehensive genomic analysis of the temporal response of yeast to 
anaerobiosis (six generations) and subsequent aerobic recovery ( approximately 2 
generations) to reveal metabolic-state (galactose versus glucose)-dependent differences in
gene network activity and function. Analysis of variance showed that far fewer genes 
responded (raw P value of <or=10(-8)) to the O(2) shifts in glucose (1,603 genes) than in 
galactose (2,388 genes). Gene network analysis reveals that this difference is due largely
to the failure of stress-activated networks controlled by Msn2/4, Fhl1, MCB, SCB, PAC, and
RRPE to transiently respond to the shift to anaerobiosis in glucose as they did in 
galactose. After approximately 1 generation of anaerobiosis, the response was similar in 
both media, beginning with the deactivation of Hap1 and Hap2/3/4/5 networks involved in 
mitochondrial functions and the concomitant derepression of Rox1-regulated networks for 
carbohydrate catabolism and redox regulation and ending (>or=2 generations) with the 
activation of Upc2- and Mot3-regulated networks involved in sterol and cell wall 
homeostasis. The response to reoxygenation was rapid (<5 min) and similar in both media, 
dominated by Yap1 networks involved in oxidative stress/redox regulation and the 
concomitant activation of heme-regulated ones. Our analyses revealed extensive networks of
genes subject to combinatorial regulation by both heme-dependent (e.g., Hap1, Hap2/3/4/5, 
Rox1, Mot3, and Upc2) and heme-independent (e.g., Yap1, Skn7, and Puf3) factors under 
these conditions. We also uncover novel functions for several cis-regulatory sites and 
trans-acting factors and define functional regulons involved in the physiological 
acclimatization to changes in oxygen availability.

PMID: 16963631

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.glucose_anoxia_reoxygenation.pcl	Expression in response to anoxia (0 - 6 generations) and subsequent reoxygenation (6 - 7.6 generations) in glucose	24	carbon utilization|oxygen level alteration|stress	2

File last modified: Thu Apr 25 2024
2010.galactose_anoxia_reoxygenation.pcl	Expression in response to anoxia (0 - 6 generations) and subsequent reoxygenation (6 - 7.6 generations) in galactose	24	carbon utilization|oxygen level alteration|stress	2

File last modified: Thu Apr 25 2024

---


# Landry_2006_PMID_16427747

Citation: Landry CR, Oh J, Hartl DL, Cavalieri D. 2006. Genome-wide scan reveals that genetic 
variation for transcriptional plasticity in yeast is biased towards multi-copy and 
dispensable genes.

Full Description: One of the most important aspects of the evolution of development and physiology is the 
interplay between gene expression and the environment, by which traits become altered in 
response to environmental triggers. This feature is known as phenotypic plasticity. When 
different genotypes show different levels of plasticity for a trait, then they show 
genotype-by-environment interaction, or GEI. It is now clear that gene expression plays an
important role in organismic-level phenotypic plasticity, but we know very little about 
whether gene expression itself is subject to genetic variation for phenotypic plasticity (
GEI). Given that gene regulation is likely to have evolved to respond to environmental 
changes, it is of central importance to understand how environmental and genetic variation
interact to produce variation in gene expression. Here we investigate genetic variation 
for phenotypic plasticity in the yeast transcriptome for the whole genome. Six strains of 
Saccharomyces cerevisiae were grown in four different environments representing a 
continuum of rich and poor natural conditions. Using DNA-microarray data and an ANOVA 
analysis with a stringent criterion of significance, we found significant genetic 
variation for transcriptional plasticity (GEI) among strains for approximately 5% of the 
genes in the genome. There are about twice as many genes that show genetic variation for 
phenotypic plasticity as show genetic variation in transcription level independent of the 
environment. We also found that genes with genetic variation for plasticity were less 
likely to be essential and were significantly biased towards genes that have paralogs.

PMID: 16427747

GEO ID: GSE3021

PCL filename	short description	# conditions	tags	# channels
GSE3021_setA_family.pcl	Genome-wide scan reveals that genetic variation for transcriptional plasticity in yeast is biased towards multi-copy and dispensable genes.	91	carbon utilization|fermentation|nitrogen utilization|oxygen level alteration	2

File last modified: Thu Apr 25 2024

---


# Li_2006_PMID_17043222

Citation: Li CM, Klevecz RR. 2006. A rapid genome-scale response of the transcriptional oscillator 
to perturbation reveals a period-doubling path to phenotypic change.

Full Description: Perturbation of the gated-synchrony system in yeast with phenelzine, an antidepressant 
drug used in the treatment of affective disorders in humans, leads to a rapid lengthening 
in the period of the genome-wide transcriptional oscillation. The effect is a concerted, 
genome-scale change in expression that is first seen in genes maximally expressed in the 
late-reductive phase of the cycle, doubling the length of the reductive phase within two 
cycles after treatment. Clustering of genes based on their temporal patterns of expression
yielded just three super clusters whose trajectories through time could then be mapped 
into a simple 3D figure. In contrast to transcripts in the late-reductive phase, most 
transcripts do not show transients in expression relative to others in their temporal 
cluster but change their period in a concerted fashion. Mapping the trajectories of the 
transcripts into low-dimensional surfaces that can be represented by simple systems of 
differential equations provides a readily testable model of the dynamic architecture of 
phenotype. In this system, period doubling may be a preferred pathway for phenotypic 
change. As a practical matter, low-amplitude, genome-wide oscillations, a ubiquitous but 
often unrecognized attribute of phenotype, could be a source of seemingly intractable 
biological noise in microarray studies.

PMID: 17043222

GEO ID: GSE9302

PCL filename	short description	# conditions	tags	# channels
GSE9302_setA_family.pcl	A rapid genome-scale response of the transcriptional oscillator to perturbation reveals a period-doubling path to phenotypic change.	48	carbon utilization|chemical stimulus|oxygen level alteration|respiration	1

File last modified: Thu Apr 25 2024

---


# Malagon_2006_PMID_16510790

Citation: Malagon F, Kireeva ML, Shafer BK, Lubkowska L, Kashlev M, Strathern JN. 2006. Mutations in
the Saccharomyces cerevisiae RPB1 gene conferring hypersensitivity to 6-azauracil.

Full Description: RNA polymerase II (RNAPII) in eukaryotic cells drives transcription of most messenger RNAs
. RNAPII core enzyme is composed of 12 polypeptides where Rpb1 is the largest subunit. To 
further understand the mechanisms of RNAPII transcription, we isolated and characterized 
novel point mutants of RPB1 that are sensitive to the nucleotide-depleting drug 6-
azauracil (6AU). In this work we reisolated the rpo21-24/rpb1-E1230K allele, which reduces
the interaction of RNAPII-TFIIS, and identified five new point mutations in RPB1 that 
cause hypersensitivity to 6AU. The novel mutants affect highly conserved residues of Rpb1 
and have differential genetic and biochemical effects. Three of the mutations affect the 
lid and rudder, two small loops suggested by structural studies to play a central role in 
the separation of the RNA-DNA hybrids. Most interestingly, two mutations affecting the 
catalytic center (rpb1-N488D) and the homology box G (rpb1-E1103G) have strong opposite 
effects on the intrinsic in vitro polymerization rate of RNAPII. Moreover, the synthetic 
interactions of these mutants with soh1, spt4, and dst1 suggest differential in vivo 
effects.

PMID: 16510790

GEO ID: GSE2831

PCL filename	short description	# conditions	tags	# channels
GSE2831_setA_family.pcl	Mutations in the Saccharomyces cerevisiae RPB1 gene conferring hypersensitivity to 6-azauracil.	14	chemical stimulus|transcription	1

File last modified: Thu Apr 25 2024

---


# Mendiratta_2006_PMID_16415340

Citation: Mendiratta G, Eriksson PR, Shen CH, Clark DJ. 2006. The DNA-binding domain of the yeast 
Spt10p activator includes a zinc finger that is homologous to foamy virus integrase.

Full Description: The yeast SPT10 gene encodes a putative histone acetyltransferase that binds specifically 
to pairs of upstream activating sequence (UAS) elements found only in the histone gene 
promoters. Here, we demonstrate that the DNA-binding domain of Spt10p is located between 
residues 283 and 396 and includes a His(2)-Cys(2) zinc finger. The binding of Spt10p to 
the histone UAS is zinc-dependent and is disabled by a zinc finger mutation (C388S). The 
isolated DNA-binding domain binds to single histone UAS elements with high affinity. In 
contrast, full-length Spt10p binds with high affinity only to pairs of UAS elements with 
very strong positive cooperativity and is unable to bind to a single UAS element. This 
implies the presence of a blocking domain in full-length Spt10p, which forces it to search
for a pair of UAS elements. Chromatin immunoprecipitation experiments indicate that, 
unlike wild-type Spt10p, the C388S protein does not bind to the promoter of the gene 
encoding histone H2A (HTA1) in vivo. The C388S mutant has a phenotype similar to that of 
the spt10Delta mutant: poor growth and global aberrations in gene expression. Thus, the 
C388S mutation disables the DNA-binding function of Spt10p in vitro and in vivo. The zinc 
finger of Spt10p is homologous to that of foamy virus integrase, perhaps suggesting that 
this integrase is also a sequence-specific DNA-binding protein.

PMID: 16415340

GEO ID: GSE3354

PCL filename	short description	# conditions	tags	# channels
GSE3354_setA_family.pcl	The DNA-binding domain of the yeast Spt10p activator includes a zinc finger that is homologous to foamy virus integrase.	9	transcription	1

File last modified: Thu Apr 25 2024

---


# Oberstrass_2006_PMID_16429156

Citation: Oberstrass FC, Lee A, Stefl R, Janis M, Chanfreau G, Allain FH. 2006. Shape-specific 
recognition in the structure of the Vts1p SAM domain with RNA.

Full Description: Although the abundant sterile alpha motif (SAM) domain was originally classified as a 
protein-protein interaction domain, it has recently been shown that certain SAM domains 
have the ability to bind RNA, defining a new type of post-transcriptional gene regulator. 
To further understand the function of SAM-RNA recognition, we determined the solution 
structures of the SAM domain of the Saccharomyces cerevisiae Vts1p (Vts1p-SAM) and the 
Smaug response element (SRE) stem-loop RNA as a complex and in isolation. The structures 
show that Vts1p-SAM recognizes predominantly the shape of the SRE rather than its sequence
, with the exception of a G located at the tip of the pentaloop. Using microarray gene 
profiling, we identified several genes in S. cerevisiae that seem to be regulated by Vts1p
and contain one or more copies of the SRE.

PMID: 16429156

GEO ID: GSE3859

PCL filename	short description	# conditions	tags	# channels
GSE3859_setA_family.pcl	Shape-specific recognition in the structure of the Vts1p SAM domain with RNA.	5	RNA catabolism	1

File last modified: Thu Apr 25 2024

---


# Parra_2006_PMID_16648479

Citation: Parra MA, Kerr D, Fahy D, Pouchnik DJ, Wyrick JJ. 2006. Deciphering the roles of the 
histone H2B N-terminal domain in genome-wide transcription.

Full Description: Histone N-terminal domains are frequent targets of posttranslational modifications. 
Multiple acetylated lysine residues have been identified in the N-terminal domain of H2B (
K6, K11, K16, K17, K21, and K22), but little is known about how these modifications 
regulate transcription. We systematically mutated the N-terminal domain of histone H2B, 
both at known sites of lysine acetylation and elsewhere, and characterized the resulting 
changes in genome-wide expression in each mutant strain. Our results indicate that known 
sites of lysine acetylation in this domain are required for gene-specific transcriptional 
activation. However, the entire H2B N-terminal domain is principally required for the 
transcriptional repression of a large subset of the yeast genome. We find that the histone
H2B repression (HBR) domain, comprised of residues 30 to 37, is necessary and sufficient 
for this repression. Many of the genes repressed by the HBR domain are located adjacent to
telomeres or function in vitamin and carbohydrate metabolism. Deletion of the HBR domain 
also confers an increased sensitivity to DNA damage by UV irradiation. We mapped the 
critical residues in the HBR domain required for its repression function. Finally, 
comparisons of these data with previous studies reveal that a surprising number of genes 
are coregulated by the N-terminal domains of histone H2B, H3, and H4.

PMID: 16648479

GEO ID: GSE3806

PCL filename	short description	# conditions	tags	# channels
GSE3806_setA_family.pcl	Deciphering the roles of the histone H2B N-terminal domain in genome-wide transcription.	20	histone modification	1

File last modified: Thu Apr 25 2024

---


# Pramila_2006_PMID_16912276

Citation: Pramila T, Wu W, Miles S, Noble WS, Breeden LL. 2006. The Forkhead transcription factor 
Hcm1 regulates chromosome segregation genes and fills the S-phase gap in the 
transcriptional circuitry of the cell cycle.

Full Description: Transcription patterns shift dramatically as cells transit from one phase of the cell 
cycle to another. To better define this transcriptional circuitry, we collected new 
microarray data across the cell cycle of budding yeast. The combined analysis of these 
data with three other cell cycle data sets identifies hundreds of new highly periodic 
transcripts and provides a weighted average peak time for each transcript. Using these 
data and phylogenetic comparisons of promoter sequences, we have identified a late S-phase
-specific promoter element. This element is the binding site for the forkhead protein Hcm1
, which is required for its cell cycle-specific activity. Among the cell cycle-regulated 
genes that contain conserved Hcm1-binding sites, there is a significant enrichment of 
genes involved in chromosome segregation, spindle dynamics, and budding. This may explain 
why Hcm1 mutants show 10-fold elevated rates of chromosome loss and require the spindle 
checkpoint for viability. Hcm1 also induces the M-phase-specific transcription factors 
FKH1, FKH2, and NDD1, and two cell cycle-specific transcriptional repressors, WHI5 and 
YHP1. As such, Hcm1 fills a significant gap in our understanding of the transcriptional 
circuitry that underlies the cell cycle.

PMID: 16912276

GEO ID: GSE4987

PCL filename	short description	# conditions	tags	# channels
GSE4987_setA_family.pcl	The Forkhead transcription factor Hcm1 regulates chromosome segregation genes and fills the S-phase gap in the transcriptional circuitry of the cell cycle.	50	mitotic cell cycle	2

File last modified: Thu Apr 25 2024

---


# Reinke_2006_PMID_16923813

Citation: Reinke A, Chen JC, Aronova S, Powers T. 2006. Caffeine targets TOR complex I and provides 
evidence for a regulatory link between the FRB and kinase domains of Tor1p.

Full Description: The target of rapamycin (TOR) kinase is an important regulator of growth in eukaryotic 
cells. In budding yeast, Tor1p and Tor2p function as part of two distinct protein 
complexes, TORC1 and TORC2, where TORC1 is specifically inhibited by the antibiotic 
rapamycin. Significant insight into TORC1 function has been obtained using rapamycin as a 
specific small molecule inhibitor of TOR activity. Here we show that caffeine acts as a 
distinct and novel small molecule inhibitor of TORC1: (i) deleting components specific to 
TORC1 but not TORC2 renders cells hypersensitive to caffeine; (ii) rapamycin and caffeine 
display remarkably similar effects on global gene expression; and (iii) mutations were 
isolated in Tor1p, a component specific to TORC1, that confers significant caffeine 
resistance both in vivo and in vitro. Strongest resistance requires two simultaneous 
mutations in TOR1, the first at either one of two highly conserved positions within the 
FRB (rapamycin binding) domain and a second at a highly conserved position within the ATP 
binding pocket of the kinase domain. Biochemical and genetic analyses of these mutant 
forms of Tor1p support a model wherein functional interactions between the FRB and kinase 
domains, as well as between the FRB domain and the TORC1 component Kog1p, regulate TOR 
activity as well as contribute to the mechanism of caffeine resistance.

PMID: 16923813

GEO ID: GSE4584

PCL filename	short description	# conditions	tags	# channels
GSE4584_set1_family.pcl	Dose response to caffeine and response in different strain backgrounds	6	chemical stimulus|signaling	2

File last modified: Thu Apr 25 2024
GSE4586_setA_family.pcl	Dose response to caffeine on wildtype versus a caffeine resistant tor1 mutant	6	chemical stimulus|signaling	2

File last modified: Thu Apr 25 2024
GSE4584_set0_family.pcl	Dose response to rapamycin and response in different strain backgrounds	5	chemical stimulus|signaling	2

File last modified: Thu Apr 25 2024

---


# Roberts_2006_PMID_16741729

Citation: Roberts GG, Hudson AP. 2006. Transcriptome profiling of Saccharomyces cerevisiae during a 
transition from fermentative to glycerol-based respiratory growth reveals extensive 
metabolic and structural remodeling.

Full Description: Transcriptome analyses using a wild-type strain of Saccharomyces cerevisiae were performed
to assess the overall pattern of gene expression during the transition from glucose-based 
fermentative to glycerol-based respiratory growth. These experiments revealed a complex 
suite of metabolic and structural changes associated with the adaptation process. 
Alterations in gene expression leading to remodeling of various membrane transport systems
and the cortical actin cytoskeleton were observed. Transition to respiratory growth was 
accompanied by alterations in transcript patterns demonstrating not only a general stress 
response, as seen in earlier studies, but also the oxidative and osmotic stress responses
. In some contrast to earlier studies, these experiments identified modulation of 
expression for many genes specifying transcription factors during the transition to 
glycerol-based growth. Importantly and unexpectedly, an ordered series of changes was seen
in transcript levels from genes encoding components of the TFIID, SAGA (Spt-Ada-Gcn5-
Acetyltransferase), and SLIK (Saga LIKe) complexes and all three RNA polymerases, 
suggesting a modulation of structure for the basal transcriptional machinery during 
adaptation to respiratory growth. In concert with data given in earlier studies, the 
results presented here highlight important aspects of metabolic and other adaptations to 
respiratory growth in yeast that are common to utilization of multiple carbon sources. 
Importantly, they also identify aspects specific to adaptation of this organism to growth 
on glycerol as sole carbon source.

PMID: 16741729

GEO ID: N/A

PCL filename	short description	# conditions	tags	# channels
2010.glucose_glycerol_transition.pcl	Transition from fermentative to glycerol-based respiratory growth	5	carbon utilization|fermentation|respiration	2

File last modified: Thu Apr 25 2024

---


# Ronen_2006_PMID_16381818

Citation: Ronen M, Botstein D. 2006. Transcriptional response of steady-state yeast cultures to 
transient perturbations in carbon source.

Full Description: To understand the dynamics of transcriptional response to changing environments, well 
defined, easily controlled, and short-term perturbation experiments were undertaken. We 
subjected steady-state cultures of Saccharomyces cerevisiae in chemostats growing on 
limiting galactose to two different size pulses of glucose, well known to be a preferred 
carbon source. Although these pulses were not large enough to change growth rates or cell 
size, approximately 25% of the genes changed their expression at least 2-fold. Using DNA 
microarrays to estimate mRNA abundance, we found a number of distinguishable patterns of 
transcriptional response among the many genes whose expression changed. Many of these 
genes were already known to be regulated by particular transcription factors; we estimated
five potentially relevant transcription factor activities from the observed changes in 
gene expression (i.e., Mig1p, Gal4p, Cat8p, Rgt1p, Adr1p, and Rcs1p). With these estimates
, for two regulatory circuits involving interaction among multiple regulators we could 
generate dynamical models that quantitatively account for the observed transcriptional 
responses to the transient perturbations.

PMID: 16381818

GEO ID: GSE4158

PCL filename	short description	# conditions	tags	# channels
GSE4158_set0_family.pcl	Transcriptional response of steady-state yeast cultures to transient perturbations in carbon source.	26	carbon utilization	2

File last modified: Thu Apr 25 2024

---


# Scherens_2006_PMID_16879428

Citation: Scherens B, Feller A, Vierendeels F, Messenguy F, Dubois E. 2006. Identification of direct
and indirect targets of the Gln3 and Gat1 activators by transcriptional profiling in 
response to nitrogen availability in the short and long term.

Full Description: Nitrogen catabolite repression (NCR) consists in the specific inhibition of 
transcriptional activation of genes encoding the permeases and catabolic enzymes needed to
degrade poor nitrogen sources. Under nitrogen limitation or rapamycin treatment, NCR genes
are activated by Gln3 or Gat1, or by both factors. To compare the sets of genes responding
to rapamycin or to nitrogen limitation, we used DNA microarrays to establishing the 
expression profiles of a wild type strain, and of a double gln3Delta-gat1Delta strain, 
grown on glutamine, after addition of rapamycin, on proline, or after a shift from 
glutamine to proline. Analysis of microarray data revealed 392 genes whose expression was 
dependent on the nitrogen source quality. 91 genes were activated in a GATA factor-
dependent manner in all growth conditions, suggesting a direct role of Gln3 and Gat1 in 
their expression. Other genes were only transiently up-regulated (stress-responsive genes
) or down-regulated (genes encoding ribosomal proteins and translational factors) upon 
nitrogen limitation, and this regulation was delayed in a gln3Delta-gat1Delta strain. 
Repression of amino acid and nucleotide biosynthetic genes after a nitrogen shift did not 
depend on Gcn4. Several transporter genes were repressed as a consequence of enhanced 
levels of NCR-responsive permeases present at the plasma membrane.

PMID: 16879428

GEO ID: GSE2891

PCL filename	short description	# conditions	tags	# channels
GSE2891_setA_family.pcl	Identification of direct and indirect targets of the Gln3 and Gat1 activators by transcriptional profiling in response to nitrogen availability in the short and long term.	18	chemical stimulus|nitrogen utilization|transcription	2

File last modified: Thu Apr 25 2024

---


# Schrader_2006_PMID_16987817

Citation: Schrader R, Young C, Kozian D, Hoffmann R, Lottspeich F. 2006. Temperature-sensitive eIF5A,
mutant accumulates transcripts targeted to the nonsense-mediated decay pathway.

Full Description: The highly conserved protein eIF5A found in Archaea and all eukaryotes uniquely contains 
the posttranslationally formed amino acid hypusine. Despite being essential the functions 
of this protein and its modification remain unclear. To gain more insight into these 
functions temperature-sensitive mutants of the human EIF5A1 were characterized in the 
yeast Saccharomyces cerevisiae. Expression of the point mutated form V81G in a DeltaeIF5A 
strain of yeast led to a strongly temperature-sensitive phenotype and to a significantly 
reduced protein level at restrictive temperature. The mutant showed accumulation of a 
subset of mRNAs that was also observed in nonsense-mediated decay (NMD)-deficient yeast 
strains. After short incubation at restrictive temperature the mutant exhibited increased 
half-lives of the intron containing CYH2 pre-mRNA and mature transcripts of NMD-dependent 
genes. Reduced telomere silencing and shortening was detected in the V81G mutant further 
supporting similarities to NMD-deficient strains. Our data suggest that eIF5A mediates 
important cellular processes like cell viability and senescence through its effects on the
stability of certain mRNAs.

PMID: 16987817

GEO ID: GSE5290

PCL filename	short description	# conditions	tags	# channels
GSE5290_setA_family.pcl	Temperature-sensitive eIF5A mutant accumulates transcripts targeted to the nonsense-mediated decay pathway.	8	RNA catabolism|translational regulation	1

File last modified: Thu Apr 25 2024

---


# Simons_2006_PMID_16870766

Citation: Simons V, Morrissey JP, Latijnhouwers M, Csukai M, Cleaver A, Yarrow C, Osbourn A. 2006. 
Dual effects of plant steroidal alkaloids on Saccharomyces cerevisiae.

Full Description: Many plant species accumulate sterols and triterpenes as antimicrobial glycosides. These 
secondary metabolites (saponins) provide built-in chemical protection against pest and 
pathogen attack and can also influence induced defense responses. In addition, they have a
variety of important pharmacological properties, including anticancer activity. The 
biological mechanisms underpinning the varied and diverse effects of saponins on microbes
, plants, and animals are only poorly understood despite the ecological and pharmaceutical
importance of this major class of plant secondary metabolites. Here we have exploited 
budding yeast (Saccharomyces cerevisiae) to investigate the effects of saponins on 
eukaryotic cells. The tomato steroidal glycoalkaloid alpha-tomatine has antifungal 
activity towards yeast, and this activity is associated with membrane permeabilization. 
Removal of a single sugar from the tetrasaccharide chain of alpha-tomatine results in a 
substantial reduction in antimicrobial activity. Surprisingly, the complete loss of sugars
leads to enhanced antifungal activity. Experiments with alpha-tomatine and its aglycone 
tomatidine indicate that the mode of action of tomatidine towards yeast is distinct from 
that of alpha-tomatine and does not involve membrane permeabilization. Investigation of 
the effects of tomatidine on yeast by gene expression and sterol analysis indicate that 
tomatidine inhibits ergosterol biosynthesis. Tomatidine-treated cells accumulate 
zymosterol rather than ergosterol, which is consistent with inhibition of the sterol C(24
) methyltransferase Erg6p. However, erg6 and erg3 mutants (but not erg2 mutants) have 
enhanced resistance to tomatidine, suggesting a complex interaction of erg mutations, 
sterol content, and tomatidine resistance.

PMID: 16870766

GEO ID: GSE4669

PCL filename	short description	# conditions	tags	# channels
GSE4669_setA_family.pcl	Dual effects of plant steroidal alkaloids on Saccharomyces cerevisiae.	12	chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Takahashi_2006_PMID_16857587

Citation: Takahashi H, McCaffery JM, Irizarry RA, Boeke JD. 2006. Nucleocytosolic acetyl-coenzyme a 
synthetase is required for histone acetylation  and global transcription.

Full Description: Metabolic enzymes rarely regulate informational processes like gene expression. Yeast 
acetyl-CoA synthetases (Acs1p and 2p) are exceptional, as they are important not only for 
carbon metabolism but also are shown here to supply the acetyl-CoA for histone acetylation
by histone acetyltransferases (HATs). acs2-Ts  mutants exhibit global histone 
deacetylation, transcriptional defects, and synthetic growth defects with HAT mutants at 
high temperatures. In glycerol with  ethanol, Acs1p is an alternate acetyl-CoA source for 
HATs. Rapid deacetylation after Acs2p inactivation suggests nuclear acetyl-CoA synthesis 
is rate limiting for histone acetylation. Different histone lysines exhibit distinct 
deacetylation rates, with N-terminal tail lysines deacetylated rapidly and H3 lysine 56 
slowly. Yeast mitochondrial and nucleocytosolic acetyl-CoA pools are biochemically 
isolated. Thus, acetyl-CoA metabolism is directly linked to chromatin regulation  and may 
affect diverse cellular processes in which acetylation and metabolism intersect, such as 
disease states and aging.

PMID: 16857587

GEO ID: GSE2806.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE2806.final.pcl	Nucleocytosolic acetyl-coenzyme a synthetase is required for histone acetylation  and global transcription.	2	histone modification	1

File last modified: Thu Apr 25 2024

---


# Tanaka_2006_PMID_16943074

Citation: Tanaka F, Ando A, Nakamura T, Takagi H, Shima J. 2006. Functional genomic analysis of 
commercial baker's yeast during initial stages of model dough-fermentation.

Full Description: Gene expression profiles of baker's yeast during initial dough-fermentation were 
investigated using liquid fermentation (LF) media to obtain insights at the molecular 
level into rapid adaptation mechanisms of baker's yeast. Results showed that onset of 
fermentation caused drastic changes in gene expression profiles within 15 min. Genes 
involved in the tricarboxylic acid (TCA) cycle were down-regulated and genes involved in 
glycolysis were up-regulated, indicating a metabolic shift from respiration to 
fermentation. Genes involved in ethanol production (PDC genes and ADH1), in glycerol 
synthesis (GPD1 and HOR2), and in low-affinity hexose transporters (HXT1 and HXT3) were up
-regulated at the beginning of model dough-fermentation. Among genes up-regulated at 15 
min, several genes classified as transcription were down-regulated within 30 min. These 
down-regulated genes are involved in messenger RNA splicing and ribosomal protein 
biogenesis and in transcriptional regulator (SRB8, MIG1). In contrast, genes involved in 
amino acid metabolism and in vitamin metabolism, such as arginine biosynthesis, riboflavin
biosynthesis, and thiamin biosynthesis, were subsequently up-regulated after 30 min. 
Interestingly, the genes involved in the unfolded protein response (UPR) pathway were also
subsequently up-regulated. Our study presents the first overall description of the 
transcriptional response of baker's yeast during dough-fermentation, and will thus help 
clarify genomic responses to various stresses during commercial fermentation processes.

PMID: 16943074

GEO ID: GSE3043

PCL filename	short description	# conditions	tags	# channels
GSE3043_setA_family.pcl	Functional genomic analysis of commercial baker's yeast during initial stages of model dough-fermentation.	6	carbon utilization|fermentation	1

File last modified: Thu Apr 25 2024

---


# Yu_2006_PMID_16461773

Citation: Yu C, Palumbo MJ, Lawrence CE, Morse RH. 2006. Contribution of the histone H3 and H4 amino
termini to Gcn4p- and Gcn5p-mediated transcription in yeast.

Full Description: Histone amino termini are post-translationally modified by both transcriptional 
coactivators and corepressors, but the extent to which the relevant histone modifications 
contribute to gene expression, and the mechanisms by which they do so, are incompletely 
understood. To address this issue, we have examined the contributions of the histone H3 
and H4 amino termini, and of the coactivator and histone acetyltransferase Gcn5p, to 
activation of a small group of Gcn4p-activated genes. The histone H3 tail exerts a modest
(about 2-fold) but significant effect on activation that correlates with a requirement for
Gcn5p and is distributed over multiple lysine residues. The H4 tail also plays a positive 
role in activation of some of those genes tested, but this does not correlate as closely 
with Gcn5p coactivation. Microarray experiments did not reveal a close correspondence 
between those genes activated by Gcn4p and genes requiring the H3 or H4 tail, and analysis
of published microarray data indicates that Gcn4p-regulated genes are not in general 
strongly dependent on Gcn5p. However, a large fraction of genes activated by Gcn4p were 
found to be repressed by the H3 and H4 amino termini under non-inducing conditions, 
indicating that one role for Gcn4p is to overcome repression mediated by the histone tails.

PMID: 16461773

GEO ID: GSE4135

PCL filename	short description	# conditions	tags	# channels
GSE4135_setA_family.pcl	Contribution of the histone H3 and H4 amino termini to Gcn4p- and Gcn5p-mediated transcription in yeast.	14	histone modification	1

File last modified: Thu Apr 25 2024

---


# Abbott_2007_PMID_17484738

Citation: Abbott DA, Knijnenburg TA, de Poorter LM, Reinders MJ, Pronk JT, van Maris AJ. 2007. 
Generic and specific transcriptional responses to different weak organic acids in 
anaerobic chemostat cultures of Saccharomyces cerevisiae.

Full Description: Transcriptional responses to four weak organic acids (benzoate, sorbate, acetate  and 
propionate) were investigated in anaerobic, glucose-limited chemostat cultures of 
Saccharomyces cerevisiae. To enable quantitative comparison of the responses to the acids
, their concentrations were chosen such that they caused a  50% decrease of the biomass 
yield on glucose. The concentration of each acid required to achieve this yield was 
negatively correlated with membrane affinity.  Microarray analysis revealed that each acid
caused hundreds of transcripts to change by over twofold relative to reference cultures 
without added organic acids. However, only 14 genes were consistently upregulated in 
response to all acids. The moderately lipophilic compounds benzoate and sorbate and, to a 
lesser  extent, the less lipophilic acids acetate and propionate showed overlapping 
transcriptional responses. Statistical analysis for overrepresented functional categories 
and upstream regulatory elements indicated that responses to the strongly lipophilic acids
were focused on genes related to the cell wall, while acetate and propionate had a 
stronger impact on membrane-associated transport processes. The fact that S. cerevisiae 
exhibits a minimal generic transcriptional response to weak organic acids along with 
extensive specific responses is relevant for interpreting and controlling weak acid 
toxicity in food products and in industrial fermentation processes.

PMID: 17484738

GEO ID: GSE5926

PCL filename	short description	# conditions	tags	# channels
GSE5926_s12.final.pcl	Generic and specific transcriptional responses to different weak organic acids in anaerobic chemostat cultures of Saccharomyces cerevisiae.	12	stress|chemical stimulus|fermentation	1

File last modified: Thu Apr 25 2024

---


# Abe_2007_PMID_17910955

Citation: Abe F. 2007. Induction of DAN/TIR yeast cell wall mannoprotein genes in response to high 
hydrostatic pressure and low temperature.

Full Description: Global transcriptional profiles of Saccharomyces cerevisiae were studied following changes
in growth conditions to high hydrostatic pressure and low temperature. These profiles were
quantitatively very similar, encompassing 561 co-upregulated genes and 161 co-
downregulated genes. In particular, expression of the DAN/TIR cell wall mannoprotein genes
, which are generally expressed under hypoxia, were markedly upregulated by high pressure 
and low temperature, suggesting the overlapping regulatory networks of transcription. In 
support of the role of mannoproteins in cell wall integrity, cells acquired resistance 
against treatment with SDS, Zymolyase and lethal levels of high pressure when preincubated
under high pressure and low temperature.

PMID: 17910955

GEO ID: GSE9136.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE9136.final.pcl	Induction of DAN/TIR yeast cell wall mannoprotein genes in response to high hydrostatic pressure and low temperature.	3	heat shock|stress	1

File last modified: Thu Apr 25 2024

---


# Barbara_2007_PMID_17124610

Citation: Barbara KE, Haley TM, Willis KA, Santangelo GM. 2007. The transcription factor Gcr1 
stimulates cell growth by participating in nutrient-responsive gene expression on a global
level.

Full Description: Transcriptomic reprogramming is critical to the coordination between growth and cell cycle
progression in response to changing extracellular conditions. In Saccharomyces cerevisiae
, the transcription factor Gcr1 contributes to this coordination by supporting maximum 
expression of G1 cyclins in addition to regulating both glucose-induced and glucose-
repressed genes. We report here the comprehensive genome-wide expression profiling of 
gcr1Delta cells. Our data show that reduced expression of ribosomal protein genes in 
gcr1Delta cells is detectable both 20 min after glucose addition and in steady-state 
cultures of raffinose-grown cells, showing that this defect is not the result of slow 
growth or growth on a repressing sugar. However, the large cell phenotype of the gcr1Delta
mutant occurs only in the presence of repressing sugars. GCR1 deletion also results in 
aberrant derepression of numerous glucose repressed loci; glucose-grown gcr1Delta cells 
actively respire, demonstrating that this global alteration in transcription corresponds 
to significant changes at the physiological level. These data offer an insight into the 
coordination of growth and cell division by providing an integrated view of the 
transcriptomic, phenotypic, and metabolic consequences of GCR1 deletion.

PMID: 17124610

GEO ID: GSE5027

PCL filename	short description	# conditions	tags	# channels
GSE5027_setA_family.pcl	The transcription factor Gcr1 stimulates cell growth by participating in nutrient-responsive gene expression on a global level.	12	carbon utilization|fermentation	2

File last modified: Thu Apr 25 2024

---


# Boer_2007_PMID_17419774

Citation: Boer VM, Tai SL, Vuralhan Z, Arifin Y, Walsh MC, Piper MD, de Winde JH, Pronk JT, Daran JM,
 2007. Transcriptional responses of Saccharomyces cerevisiae to preferred and nonpreferred
nitrogen sources in glucose-limited chemostat cultures.

Full Description: Aerobic, glucose-limited chemostat cultures of Saccharomyces cerevisiae grown with six 
different nitrogen sources were subjected to transcriptome analysis. The use of chemostats
enabled an analysis of nitrogen-source-dependent transcriptional regulation at a fixed 
specific growth rate. A selection of preferred (ammonium and asparagine) and nonpreferred
(leucine, phenylalanine, methionine and proline) nitrogen sources was investigated. For 
each nitrogen source, distinct sets of genes were induced or repressed relative to the 
other five nitrogen sources. In total, 131 such 'signature transcripts' were identified in
this study. In addition to signature transcripts, genes were identified that showed a 
transcriptional coresponse to two or more of the six nitrogen sources. For example, 33 
genes were transcriptionally upregulated in leucine-grown, phenylalanine-grown and 
methionine-grown cultures; this was partly attributed to the involvement of common enzymes
in the dissimilation of these amino acids. In addition to specific transcriptional 
responses elicited by individual nitrogen sources, their impact on global regulatory 
mechanisms such as nitrogen catabolite repression (NCR) were monitored. NCR-sensitive gene
expression in the chemostat cultures showed that ammonium and asparagine were 'rich' 
nitrogen sources. By this criterion, leucine, proline and methionine were 'poor' nitrogen 
sources, and phenylalanine showed an 'intermediate' NCR response.

PMID: 17419774

GEO ID: GSE6405

PCL filename	short description	# conditions	tags	# channels
GSE6405_setA_family.pcl	Transcriptional responses of Saccharomyces cerevisiae to preferred and nonpreferred nitrogen sources in glucose-limited chemostat cultures.	15	nitrogen utilization	1

File last modified: Thu Apr 25 2024

---


# Carter_2007_PMID_17389876

Citation: Carter GW, Prinz S, Neou C, Shelby JP, Marzolf B, Thorsson V, Galitski T. 2007. Prediction
of phenotype and gene expression for combinations of mutations.

Full Description: Molecular interactions provide paths for information flows. Genetic interactions reveal 
active information flows and reflect their functional consequences. We integrated these 
complementary data types to model the transcription network controlling cell 
differentiation in yeast. Genetic interactions were inferred from linear decomposition of 
gene expression data and were used to direct the construction of a molecular interaction 
network mediating these genetic effects. This network included both known and novel 
regulatory influences, and predicted genetic interactions. For corresponding combinations 
of mutations, the network model predicted quantitative gene expression profiles and 
precise phenotypic effects. Multiple predictions were tested and verified.

PMID: 17389876

GEO ID: GSE5938

PCL filename	short description	# conditions	tags	# channels
GSE5938_setA_family.pcl	Prediction of phenotype and gene expression for combinations of mutations.	90	filamentous growth	1

File last modified: Thu Apr 25 2024

---


# Daran-Lapujade_2007_PMID_17898166

Citation: Daran-Lapujade P, Rossell S, van Gulik WM, Luttik MA, de Groot MJ, Slijper M, Heck AJ, 
Daran JM, de Winde JH, Westerhoff HV, Pronk JT, Bakker BM. 2007. The fluxes through 
glycolytic enzymes in Saccharomyces cerevisiae are predominantly regulated at 
posttranscriptional levels.

Full Description: Metabolic fluxes may be regulated hierarchically, e.g., by changes of gene expression that
adjust enzyme capacities (V(max)) and/or metabolically by interactions of enzymes with 
substrates, products, or allosteric effectors. In the present study, a method is developed
to dissect the hierarchical regulation into contributions by transcription, translation, 
protein degradation, and posttranslational modification. The method was applied to the 
regulation of fluxes through individual glycolytic enzymes when the yeast Saccharomyces 
cerevisiae was confronted with the absence of oxygen and the presence of benzoic acid 
depleting its ATP. Metabolic regulation largely contributed to the approximately 10-fold 
change in flux through the glycolytic enzymes. This contribution varied from 50 to 80%, 
depending on the glycolytic step and the cultivation condition tested. Within the 50-20% 
hierarchical regulation of fluxes, transcription played a minor role, whereas regulation 
of protein synthesis or degradation was the most important. These also contributed to 75-
100% of the regulation of protein levels.

PMID: 17898166

GEO ID: GSE9232

PCL filename	short description	# conditions	tags	# channels
GSE9232_setA_family.pcl	The fluxes through glycolytic enzymes in Saccharomyces cerevisiae are predominantly regulated at posttranscriptional levels.	9	carbon utilization|fermentation|oxygen level alteration|respiration	1

File last modified: Thu Apr 25 2024

---


# De_Nicola_2007_PMID_17933919

Citation: De Nicola R, Hazelwood LA, De Hulster EA, Walsh MC, Knijnenburg TA, Reinders MJ, Walker GM,
Pronk JT, Daran JM, Daran-Lapujade P. 2007. Physiological and transcriptional responses of
Saccharomyces cerevisiae to zinc limitation in chemostat cultures.

Full Description: Transcriptional responses of the yeast Saccharomyces cerevisiae to Zn availability were 
investigated at a fixed specific growth rate under limiting and abundant Zn concentrations
in chemostat culture. To investigate the context dependency of this transcriptional 
response and eliminate growth rate-dependent variations in transcription, yeast was grown 
under several chemostat regimens, resulting in various carbon (glucose), nitrogen (
ammonium), zinc, and oxygen supplies. A robust set of genes that responded consistently to
Zn limitation was identified, and the set enabled the definition of the Zn-specific Zap1p 
regulon, comprised of 26 genes and characterized by a broader zinc-responsive element 
consensus (MHHAACCBYNMRGGT) than so far described. Most surprising was the Zn-dependent 
regulation of genes involved in storage carbohydrate metabolism. Their concerted down-
regulation was physiologically relevant as revealed by a substantial decrease in glycogen 
and trehalose cellular content under Zn limitation. An unexpectedly large number of genes 
were synergistically or antagonistically regulated by oxygen and Zn availability. This 
combinatorial regulation suggested a more prominent involvement of Zn in mitochondrial 
biogenesis and function than hitherto identified.

PMID: 17933919

GEO ID: GSE8089

PCL filename	short description	# conditions	tags	# channels
GSE8089_setA_family.pcl	Response of yeast to nitrogen limitation in a chemostat culture under aerobic or anaerobic conditions	6	nitrogen utilization	1

File last modified: Thu Apr 25 2024
GSE8035_setA_family.pcl	Response of yeast to zinc limitation in a chemostat culture under aerobic or anaerobic conditions	6	metal or metalloid ion stress|nutrient utilization	1

File last modified: Thu Apr 25 2024

---


# Demae_2007_PMID_17986087

Citation: Demae M, Murata Y, Hisano M, Haitani Y, Shima J, Takagi H. 2007. Overexpression of two 
transcriptional factors, Kin28 and Pog1, suppresses the stress sensitivity caused by the 
rsp5 mutation in Saccharomyces cerevisiae.

Full Description: Rsp5 is an essential and multi-functional E3 ubiquitin ligase in Saccharomyces cerevisiae
. The Ala401Glu rsp5 mutant, which is hypersensitive to various stresses, was isolated 
previously. To understand the function of Rsp5 in stress responses, suppressor genes whose
overexpression allows rsp5(A401E) cells to grow at a high temperature were screened. The 
KIN28 and POG1 genes, encoding a subunit of the transcription factor TFIIH and a putative 
transcriptional activator, respectively, were identified as multicopy suppressors of not 
only high temperature but also LiCl stresses. The overexpression of Kin28 and Pog1 in rsp5
(A401E) cells led to an increase in the transcriptional level of some stress  proteins 
when exposed to a temperature up-shift. Based on DNA microarray analysis under LiCl stress
, it appears that the transcriptional level of some proteasome components is slightly 
increased in rsp5(A401E) cells overexpressing Kin28 or Pog1. These results suggest that 
the overexpression of Kin28 and Pog1 enhances the protein refolding and degradation 
pathways in rsp5(A401E) cells.

PMID: 17986087

GEO ID: GSE8729.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE8729.final.pcl	Overexpression of two transcriptional factors, Kin28 and Pog1, suppresses the stress sensitivity caused by the rsp5 mutation in Saccharomyces cerevisiae.	4	heat shock	1

File last modified: Thu Apr 25 2024

---


# Dettman_2007_PMID_17538619

Citation: Dettman JR, Sirjusingh C, Kohn LM, Anderson JB. 2007. Incipient speciation by divergent 
adaptation and antagonistic epistasis in yeast.

Full Description: Establishing the conditions that promote the evolution of reproductive isolation and 
speciation has long been a goal in evolutionary biology. In ecological speciation, 
reproductive isolation between populations evolves as a by-product of divergent selection 
and the resulting environment-specific adaptations. The leading genetic model of 
reproductive isolation predicts that hybrid inferiority is caused by antagonistic 
epistasis between incompatible alleles at interacting loci. The fundamental link between 
divergent adaptation and reproductive isolation through genetic incompatibilities has been
predicted, but has not been directly demonstrated experimentally. Here we empirically 
tested key predictions of speciation theory by evolving the initial stages of speciation 
in experimental populations of the yeast Saccharomyces cerevisiae. After replicate 
populations adapted to two divergent environments, we consistently observed the evolution 
of two forms of postzygotic isolation in hybrids: reduced rate of mitotic reproduction and
reduced efficiency of meiotic reproduction. This divergent selection resulted in greater 
reproductive isolation than parallel selection, as predicted by the ecological speciation 
theory. Our experimental system allowed controlled comparison of the relative importance 
of ecological and genetic isolation, and we demonstrated that hybrid inferiority can be 
ecological and/or genetic in basis. Overall, our results show that adaptation to divergent
environments promotes the evolution of reproductive isolation through antagonistic 
epistasis, providing evidence of a plausible common avenue to speciation and adaptive 
radiation in nature.

PMID: 17538619

GEO ID: GSE6870

PCL filename	short description	# conditions	tags	# channels
GSE6870_setA_family.pcl	Incipient speciation by divergent adaptation and antagonistic epistasis in yeast.	8	evolution	2

File last modified: Thu Apr 25 2024

---


# Dong_2007_PMID_17317628

Citation: Dong S, Li C, Zenklusen D, Singer RH, Jacobson A, He F. 2007. YRA1 autoregulation requires
nuclear export and cytoplasmic Edc3p-mediated degradation of its pre-mRNA.

Full Description: Autoregulatory loops often provide precise control of the level of expression of specific 
genes that encode key regulatory proteins. Here we have defined a pathway by which Yra1p, 
a yeast mRNA export factor, controls its own expression. We show that YRA1 exon 1 
sequences in cis and Yra1p in trans inhibit YRA1 pre-mRNA splicing and commit the pre-mRNA
to nuclear export. Mex67p and Crm1p jointly promote YRA1 pre-mRNA export, and once in the 
cytoplasm, the pre-mRNA is degraded by a 5' to 3' decay mechanism that is dependent on the
decapping activator Edc3p and on specific sequences in the YRA1 intron. These results 
illustrate how common steps in the nuclear processing, export, and degradation of a 
transcript can be uniquely combined to control the expression of a specific gene and 
suggest that Edc3p-mediated decay may have additional regulatory functions in eukaryotic 
cells.

PMID: 17317628

GEO ID: GSE6647

PCL filename	short description	# conditions	tags	# channels
GSE6647_setA_family.pcl	YRA1 autoregulation requires nuclear export and cytoplasmic Edc3p-mediated degradation of its pre-mRNA.	10	RNA catabolism	1

File last modified: Thu Apr 25 2024

---


# Hickman_2007_PMID_17785431

Citation: Hickman MJ, Winston F. 2007. Heme levels switch the function of Hap1 of Saccharomyces 
cerevisiae between transcriptional activator and transcriptional repressor.

Full Description: Changes in oxygen levels cause widespread changes in gene expression in organisms ranging 
from bacteria to humans. In Saccharomyces cerevisiae, this response is mediated in part by
Hap1, originally identified as a heme-dependent transcriptional activator that functions 
during aerobic growth. We show here that Hap1 also plays a significant and direct role 
under hypoxic conditions, not as an activator, but as a repressor. The repressive activity
of Hap1 controls several genes, including three ERG genes required for ergosterol 
biosynthesis. Chromatin immunoprecipitation experiments showed that Hap1 binds to the ERG 
gene promoters, while additional experiments showed that the corepressor Tup1/Ssn6 is 
recruited by Hap1 and is also required for repression. Furthermore, mutational analysis 
demonstrated that conserved Hap1 binding sites in the ERG5 5' regulatory region are 
required for repression. The switch of Hap1 from acting as a hypoxic repressor to an 
aerobic activator is determined by heme, which is synthesized only in the presence of 
oxygen. The ability of Hap1 to function as a ligand-dependent repressor and activator is a
property shared with mammalian nuclear hormone receptors and likely allows greater 
transcriptional control by Hap1 in response to changing oxygen levels.

PMID: 17785431

GEO ID: GSE8613

PCL filename	short description	# conditions	tags	# channels
GSE8613_setA_family.pcl	Heme levels switch the function of Hap1 of Saccharomyces cerevisiae between transcriptional activator and transcriptional repressor.	6	cofactor metabolism|oxygen level alteration	2

File last modified: Thu Apr 25 2024

---


# Hu_2007_PMID_17417638

Citation: Hu Z, Killion PJ, Iyer VR. 2007. Genetic reconstruction of a functional transcriptional 
regulatory network.

Full Description: Although global analyses of transcription factor binding provide one view of potential 
transcriptional regulatory networks, regulation also occurs at levels distinct from 
transcription factor binding. Here, we use a genetic approach to identify targets of 
transcription factors in yeast and reconstruct a functional regulatory network. First, we 
profiled transcriptional responses in S. cerevisiae strains with individual deletions of 
263 transcription factors. Then we used directed-weighted graph modeling and regulatory 
epistasis analysis to identify indirect regulatory relationships between these 
transcription factors, and from this we reconstructed a functional transcriptional 
regulatory network. The enrichment of promoter motifs and Gene Ontology annotations 
provide insight into the biological functions of the transcription factors.

PMID: 17417638

GEO ID: GSE4654

PCL filename	short description	# conditions	tags	# channels
GSE4654_setA_family.pcl	Genetic reconstruction of a functional transcriptional regulatory network.	269	transcription	2

File last modified: Thu Apr 25 2024

---


# Huisinga_2007_PMID_17407552

Citation: Huisinga KL, Pugh BF. 2007. A TATA binding protein regulatory network that governs 
transcription complex assembly.

Full Description: BACKGROUND: Eukaryotic genes are controlled by proteins that assemble stepwise into a 
transcription complex. How the individual biochemically defined assembly steps are 
coordinated and applied throughout a genome is largely unknown. Here, we model and 
experimentally test a portion of the assembly process involving the regulation of the TATA
binding protein (TBP) throughout the yeast genome. RESULTS: Biochemical knowledge was used
to formulate a series of coupled TBP regulatory reactions involving TFIID, SAGA, NC2, Mot1
, and promoter DNA. The reactions were then linked to basic segments of the transcription 
cycle and modeled computationally. A single framework was employed, allowing the 
contribution of specific steps to vary from gene to gene. Promoter binding and 
transcriptional output were measured genome-wide using ChIP-chip and expression microarray
assays. Mutagenesis was used to test the framework by shutting down specific parts of the 
network. CONCLUSION: The model accounts for the regulation of TBP at most 
transcriptionally active promoters and provides a conceptual tool for interpreting genome-
wide data sets. The findings further demonstrate the interconnections of TBP regulation on
a genome-wide scale.

PMID: 17407552

GEO ID: GSE7385

PCL filename	short description	# conditions	tags	# channels
GSE7385_setA_family.pcl	A TATA binding protein regulatory network that governs transcription complex assembly.	149	transcription	2

File last modified: Thu Apr 25 2024

---


# Iwahashi_2007_PMID_17408496

Citation: Iwahashi H, Kitagawa E, Suzuki Y, Ueda Y, Ishizawa YH, Nobumasa H, Kuboki Y, Hosoda H, 
Iwahashi Y. 2007. Evaluation of toxicity of the mycotoxin citrinin using yeast ORF DNA 
microarray and Oligo DNA microarray.

Full Description: BACKGROUND: Mycotoxins are fungal secondary metabolites commonly present in feed and food
, and are widely regarded as hazardous contaminants. Citrinin, one of the very well known 
mycotoxins that was first isolated from Penicillium citrinum, is produced by more than 10 
kinds of fungi, and is possibly spread all over the world. However, the information on the
action mechanism of the toxin is limited. Thus, we investigated the citrinin-induced 
genomic response for evaluating its toxicity. RESULTS: Citrinin inhibited growth of yeast 
cells at a concentration higher than 100 ppm. We monitored the citrinin-induced mRNA 
expression profiles in yeast using the ORF DNA microarray and Oligo DNA microarray, and 
the expression profiles were compared with those of the other stress-inducing agents. 
Results obtained from both microarray experiments clustered together, but were different 
from those of the mycotoxin patulin. The oxidative stress response genes--AADs, FLR1, OYE3
, GRE2, and MET17--were significantly induced. In the functional category, expression of 
genes involved in metabolism, cell rescue, defense and virulence, and energy were 
significantly activated. In the category of metabolism, genes involved in the glutathione 
synthesis pathway were activated, and in the category of cell rescue, defense and 
virulence, the ABC transporter genes were induced. To alleviate the induced stress, these 
cells might pump out the citrinin after modification with glutathione. While, the citrinin
treatment did not induce the genes involved in the DNA repair. CONCLUSION: Results from 
both microarray studies suggest that citrinin treatment induced oxidative stress in yeast 
cells. The genotoxicity was less severe than the patulin, suggesting that citrinin is less
toxic than patulin. The reproducibility of the expression profiles was much better with 
the Oligo DNA microarray. However, the Oligo DNA microarray did not completely overcome 
cross hybridization.

PMID: 17408496

GEO ID: GSE6118

PCL filename	short description	# conditions	tags	# channels
GSE6118_setA_family.pcl	Evaluation of toxicity of the mycotoxin citrinin using yeast ORF DNA microarray and Oligo DNA microarray.	15	chemical stimulus	2

File last modified: Thu Apr 25 2024

---


# Iwahashi_2007_PMID_18075112

Citation: Iwahashi H, Ishidou E, Kitagawa E, Momose Y. 2007. Combined cadmium and thiuram show 
synergistic toxicity and induce mitochondrial petite mutants.

Full Description: A bioassay is a system for monitoring toxicity of chemicals in the environment via the 
biological responses of experimental organisms. These responses can be detected by 
analysis of genome-wide changes in mRNA expression levels using DNA microarrays. We 
applied this system for evaluation of synergistic toxicity by cadmium and thiuram, as this
combination showed mutual growth inhibition in yeast. Hierarchical cluster analysis using 
mRNA expression profiles suggested the response of yeast to this combination is similar to
that seen with cadmium treatment alone. Functional characterization of genes induced by 
this combinational treatment also suggests that the toxicity of cadmium was enhanced. This
toxicity was observed as the damage to mitochondrial functions, which were not observed 
with either cadmium or thiuram treatments alone. The potential toxicity to mitochondria by
this combinational treatment was confirmed as the result of deletion of mitochondria with 
petite colony formation frequency. We could evaluate synergistic toxicity by cadmium and 
thiuram and show the possible use of transcriptome bioassay for synergistic toxicity.

PMID: 18075112

GEO ID: GSE8718

PCL filename	short description	# conditions	tags	# channels
GSE8718_setA_family.pcl	Combined cadmium and thiuram show synergistic toxicity and induce mitochondrial petite mutants.	9	chemical stimulus	2

File last modified: Thu Apr 25 2024

---


# James_2007_PMID_17660549

Citation: James N, Landrieux E, Collart MA. 2007. A SAGA-independent function of SPT3 mediates 
transcriptional deregulation in a mutant of the Ccr4-not complex in Saccharomyces 
cerevisiae.

Full Description: The conserved multi-subunit Ccr4-Not complex regulates gene expression in diverse ways. In
this work, we characterize the suppression of temperature sensitivity associated with a 
mutation in the gene encoding the scaffold subunit of the Ccr4-Not complex, NOT1, by the 
deletion of SPT3. We determine that the deletion of SPT3, but not the deletion of genes 
encoding other subunits of the SAGA complex, globally suppresses transcriptional defects 
of not1-2. We find that transcriptional activation in not1-2 is associated with increased 
binding of TFIID and SAGA at promoters of upregulated genes, and this is suppressed by the
deletion of SPT3. Interestingly, Spt3p-dependent activation of transcription occurs in 
not1-2 even if the SAGA complex is disrupted by the deletion of SPT7 that encodes a 
subunit of SAGA required for its integrity. Consistent with a SAGA-independent function of
Spt3p, the deletion of SPT3 displays synthetic phenotypes when combined with a deletion of
SPT7. Taken together, our results provide a new view of the Spt3 protein by identifying a 
SAGA-independent function of this protein that is functionally linked to the Ccr4-Not 
complex.

PMID: 17660549

GEO ID: GSE9432

PCL filename	short description	# conditions	tags	# channels
GSE9432_setA_family.pcl	A SAGA-independent function of SPT3 mediates transcriptional deregulation in a mutant of the Ccr4-not complex in Saccharomyces cerevisiae.	8	transcription	1

File last modified: Thu Apr 25 2024

---


# Jin_2007_PMID_17664279

Citation: Jin Y, Rodriguez AM, Stanton JD, Kitazono AA, Wyrick JJ. 2007. Simultaneous mutation of 
methylated lysine residues in histone H3 causes enhanced gene silencing, cell cycle 
defects, and cell lethality in Saccharomyces cerevisiae.

Full Description: The methylation of specific lysine residues in histone H3 is integral to transcription 
regulation; however, little is known about how combinations of methylated lysine residues 
act in concert to regulate genome-wide transcription. We have systematically mutated 
methylated histone lysine residues in yeast and found that the triple mutation of H3K4, 
H3K36, and H3K79 to arginine (H3 K4,36,79R) is lethal. The histone H3 K4,36,79R mutant 
causes a mitotic cell cycle delay and a progressive transcription defect that initiates in
telomere regions and then spreads into the chromosome. This effect is mediated by the 
silent information regulator (SIR) silencing complex, as we observe increased binding of 
the SIR complex to genomic regions adjacent to yeast telomeres in the H3 K4,36,79R mutant 
and deletion of SIR2, SIR3, or SIR4 rescues the lethal phenotype. Curiously, a yeast 
strain in which the histone methyltransferase genes are simultaneously deleted is viable. 
Indeed, deletion of the histone methyltransferase genes can suppress the H3 K4,36,79R 
lethal phenotype. These and other data suggest that the cause of lethality may in part be 
due to the association of histone methyltransferase enzymes with a histone substrate that 
cannot be methylated.

PMID: 17664279

GEO ID: GSE6331

PCL filename	short description	# conditions	tags	# channels
GSE6331_setA_family.pcl	Simultaneous mutation of methylated lysine residues in histone H3 causes enhanced gene silencing, cell cycle defects, and cell lethality in Saccharomyces cerevisiae.	20	histone modification	1

File last modified: Thu Apr 25 2024

---


# Johansson_2007_PMID_18087042

Citation: Johansson MJ, He F, Spatrick P, Li C, Jacobson A. 2007. Association of yeast Upf1p with 
direct substrates of the NMD pathway.

Full Description: Nonsense-mediated mRNA decay (NMD) is a surveillance mechanism that detects and degrades 
transcripts containing premature translation termination codons. Gene expression profiling
experiments have shown that inactivation of the NMD pathway leads to the accumulation of 
both aberrant, nonsense-containing mRNAs, and many apparently wild-type transcripts. Such 
increases in transcript steady-state levels could arise from direct changes in the 
respective mRNA half-lives, or indirectly, as a consequence of the stabilization of 
transcripts encoding specific regulatory proteins. Here, we distinguished direct from 
indirect substrates by virtue of their association with the Saccharomyces cerevisiae Upf1 
protein. Analyses of this dataset, and its comparison to the sets of transcripts that 
respectively increase or decrease in abundance when NMD is either inactivated or 
reactivated, indicate that the number of direct NMD substrates is larger than previously 
thought and that low abundance, alternatively transcribed mRNAs, i.e., mRNAs whose 5' ends
are derived from previously unannotated 5' flanking sequences, comprise a significant 
class of direct substrates. Using thiamine metabolism as an example, we also show that 
apparent NMD-regulated cellular pathways may actually reflect the detection of low-
abundance alternative transcripts under conditions where a pathway is repressed.

PMID: 18087042

GEO ID: GSE9482

PCL filename	short description	# conditions	tags	# channels
GSE9482_setA_family.pcl	Association of yeast Upf1p with direct substrates of the NMD pathway.	40	RNA catabolism	1

File last modified: Thu Apr 25 2024

---


# Joseph-Strauss_2007_PMID_17999778

Citation: Joseph-Strauss D, Zenvirth D, Simchen G, Barkai N. 2007. Spore germination in 
Saccharomyces cerevisiae: global gene expression patterns and cell cycle landmarks.

Full Description: BACKGROUND: Spore germination in the yeast Saccharomyces cerevisiae is a process in which 
non-dividing haploid spores re-enter the mitotic cell cycle and resume vegetative growth. 
To study the signals and pathways underlying spore germination we examined the global 
changes in gene expression and followed cell-cycle and germination markers during this 
process. RESULTS: We find that the germination process can be divided into two distinct 
stages. During the first stage, the induced spores respond only to glucose. The 
transcription program during this stage recapitulates the general transcription response 
of yeast cells to glucose. Only during the second phase are the cells able to sense and 
respond to other nutritional components in the environment. Components of the mitotic 
machinery are involved in spore germination but in a distinct pattern. In contrast to the 
mitotic cell cycle, growth-related events during germination are not coordinated with 
nuclear events and are separately regulated. Thus, genes that are co-induced during G1/S 
of the mitotic cell cycle, the dynamics of the septin Cdc10 and the kinetics of 
accumulation of the cyclin Clb2 all exhibit distinct patterns of regulation during spore 
germination, which allow the separation of cell growth from nuclear events. CONCLUSION: 
Taken together, genome-wide expression profiling enables us to follow the progression of 
spore germination, thus dividing this process into two major stages, and to identify 
germination-specific regulation of components of the mitotic cell cycle machinery.

PMID: 17999778

GEO ID: GSE7393

PCL filename	short description	# conditions	tags	# channels
GSE7393_set5_family.pcl	Spore germination in rich medium following pre-incubation in nitrogen; 4 hr time course	7	nitrogen utilization|nutrient utilization|sporulation	2

File last modified: Thu Apr 25 2024
GSE7393_set4_family.pcl	Spore germination in rich medium following pre-incubation in glucose; 4 hr time course	8	carbon utilization|nutrient utilization|sporulation	2

File last modified: Thu Apr 25 2024
GSE7393_set3_family.pcl	Spore incubation in nitrogen; 6 hr time course	11	nitrogen utilization|sporulation	2

File last modified: Thu Apr 25 2024
GSE7393_set2_family.pcl	Spore incubation in glucose; 6 hr time course	11	carbon utilization|sporulation	2

File last modified: Thu Apr 25 2024
GSE7393_set1_family.pcl	Spore germination in rich medium; 40 min time course	4	nutrient utilization|sporulation	2

File last modified: Thu Apr 25 2024
GSE7393_set0_family.pcl	Spore germination in rich medium; 7.5 hr time course	28	nutrient utilization|sporulation	2

File last modified: Thu Apr 25 2024

---


# Knijnenburg_2007_PMID_17241460

Citation: Knijnenburg TA, de Winde JH, Daran JM, Daran-Lapujade P, Pronk JT, Reinders MJ, Wessels LF,
 2007. Exploiting combinatorial cultivation conditions to infer transcriptional regulation.

Full Description: BACKGROUND: Regulatory networks often employ the model that attributes changes in gene 
expression levels, as observed across different cellular conditions, to changes in the 
activity of transcription factors (TFs). Although the actual conditions that trigger a 
change in TF activity should form an integral part of the generated regulatory network, 
they are usually lacking. This is due to the fact that the large heterogeneity in the 
employed conditions and the continuous changes in environmental parameters in the often 
used shake-flask cultures, prevent the unambiguous modeling of the cultivation conditions 
within the computational framework. RESULTS: We designed an experimental setup that allows
us to explicitly model the cultivation conditions and use these to infer the activity of 
TFs. The yeast Saccharomyces cerevisiae was cultivated under four different nutrient 
limitations in both aerobic and anaerobic chemostat cultures. In the chemostats, 
environmental and growth parameters are accurately controlled. Consequently, the measured 
transcriptional response can be directly correlated with changes in the limited nutrient 
or oxygen concentration. We devised a tailor-made computational approach that exploits the
systematic setup of the cultivation conditions in order to identify the individual and 
combined effects of nutrient limitations and oxygen availability on expression behavior 
and TF activity. CONCLUSION: Incorporating the actual growth conditions when inferring 
regulatory relationships provides detailed insight in the functionality of the TFs that 
are triggered by changes in the employed cultivation conditions. For example, our results 
confirm the established role of TF Hap4 in both aerobic regulation and glucose 
derepression. Among the numerous inferred condition-specific regulatory associations 
between gene sets and TFs, also many novel putative regulatory mechanisms, such as the 
possible role of Tye7 in sulfur metabolism, were identified.

PMID: 17241460

GEO ID: GSE4807

PCL filename	short description	# conditions	tags	# channels
GSE4807_setA_family.pcl	Exploiting combinatorial cultivation conditions to infer transcriptional regulation.	30	carbon utilization|fermentation|nitrogen utilization|oxygen level alteration|phosphorus utilization|respiration|sulfur utilization	1

File last modified: Thu Apr 25 2024

---


# Komili_2007_PMID_17981122

Citation: Komili S, Farny NG, Roth FP, Silver PA. 2007. Functional specificity among ribosomal 
proteins regulates gene expression.

Full Description: Duplicated genes escape gene loss by conferring a dosage benefit or evolving diverged 
functions. The yeast Saccharomyces cerevisiae contains many duplicated genes encoding 
ribosomal proteins. Prior studies have suggested that these duplicated proteins are 
functionally redundant and affect cellular processes in proportion to their expression. In
contrast, through studies of ASH1 mRNA in yeast, we demonstrate paralog-specific 
requirements for the translation of localized mRNAs. Intriguingly, these paralog-specific 
effects are limited to a distinct subset of duplicated ribosomal proteins. Moreover, 
transcriptional and phenotypic profiling of cells lacking specific ribosomal proteins 
reveals differences between the functional roles of ribosomal protein paralogs that extend
beyond effects on mRNA localization. Finally, we show that ribosomal protein paralogs 
exhibit differential requirements for assembly and localization. Together, our data 
indicate complex specialization of ribosomal proteins for specific cellular processes and 
support the existence of a ribosomal code.

PMID: 17981122

GEO ID: GSE8761

PCL filename	short description	# conditions	tags	# channels
GSE8761_setA_family.pcl	Functional specificity among ribosomal proteins regulates gene expression.	24	translational regulation	1

File last modified: Thu Apr 25 2024

---


# Kugou_2007_PMID_17396017

Citation: Kugou K, Sasanuma H, Matsumoto K, Shirahige K, Ohta K. 2007. Mre11 mediates gene 
regulation in yeast spore development.

Full Description: Mre11, together with Rad50 and Xrs2/NBS, plays pivotal roles in homologous recombination, 
repair of DNA double strand breaks (DSBs), activation of damage-induced checkpoint, and 
telomere maintenance. Here we demonstrate that the absence of Mre11 in yeast causes 
specific effects on regulation of a class of meiotic genes for spore development. Using 
DNA microarray assays to analyze yeast mutants defective for meiotic DSB formation, we 
revealed that the meiotic expression profile in the mre11Delta cells was generally 
unaffected when compared to the one in the wild-type strain, although the activation of 
about 90 meiotic genes were severely and specifically impaired in early meiosis. These 
defects were confirmed by northern and lacZ reporter gene assays. Interestingly, a 
substantial portion of the severely affected genes includes genes responsible for spore 
wall biogenesis, the defects of which may account for the fragile spore wall phenotype of 
the mre11Delta strain. The transcriptional deficiency was not observed in other DSB 
mutants such as rad50Delta, xrs2Delta, spo11Delta, and spo11Y135F, suggesting the 
transcriptional defect in mre11Delta is due to neither lack of meiotic DSB formation, nor 
disintegrity of Mre11-Rad50-Xrs2 complex. In addition, the deficiency of mre11Delta in 
gene activation was not alleviated by the deletion of RAD24. Therefore, it is unlikely 
that DNA damage checkpoint activation by mre11Delta caused transcriptional deficiency. We 
also found that a C-terminus DNA binding domain truncation mutant (mre11DeltaC49), which 
has meiosis-specific defects, exhibited transcriptional defects as observed in mre11Delta
, whereas an N-terminal phosphoesterase mutant (mre11D16A) does not. Taken together, we 
propose that Mre11 is involved in the regulation of a specific class of genes during spore
development through its C-terminus domain.

PMID: 17396017

GEO ID: GSE6620

PCL filename	short description	# conditions	tags	# channels
GSE6620_setA_family.pcl	Mre11 mediates gene regulation in yeast spore development.	8	sporulation	1

File last modified: Thu Apr 25 2024

---


# Landry_2007_PMID_17525304

Citation: Landry CR, Lemos B, Rifkin SA, Dickinson WJ, Hartl DL. 2007. Genetic properties 
influencing the evolvability of gene expression.

Full Description: Identifying the properties of gene networks that influence their evolution is a 
fundamental research goal. However, modes of evolution cannot be inferred solely from the 
distribution of natural variation, because selection interacts with demography and 
mutation rates to shape polymorphism and divergence. We estimated the effects of naturally
occurring mutations on gene expression while minimizing the effect of natural selection. 
We demonstrate that sensitivity of gene expression to mutations increases with both 
increasing trans-mutational target size and the presence of a TATA box. Genes with greater
sensitivity to mutations are also more sensitive to systematic environmental perturbations
and stochastic noise. These results provide a mechanistic basis for gene expression 
evolvability that can serve as a foundation for realistic models of regulatory evolution.

PMID: 17525304

GEO ID: GSE7537

PCL filename	short description	# conditions	tags	# channels
GSE7537_setA_family.pcl	Genetic properties influencing the evolvability of gene expression.	10	evolution	2

File last modified: Thu Apr 25 2024

---


# Levy_2007_PMID_17327914

Citation: Levy S, Ihmels J, Carmi M, Weinberger A, Friedlander G, Barkai N. 2007. Strategy of 
transcription regulation in the budding yeast.

Full Description: Cells must adjust their gene expression in order to compete in a constantly changing 
environment. Two alternative strategies could in principle ensure optimal coordination of 
gene expression with physiological requirements. First, characters of the internal 
physiological state, such as growth rate, metabolite levels, or energy availability, could
be feedback to tune gene expression. Second, internal needs could be inferred from the 
external environment, using evolutionary-tuned signaling pathways. Coordination of 
ribosomal biogenesis with the requirement for protein synthesis is of particular 
importance, since cells devote a large fraction of their biosynthetic capacity for 
ribosomal biogenesis. To define the relative contribution of internal vs. external sensing
to the regulation of ribosomal biogenesis gene expression in yeast, we subjected S. 
cerevisiae cells to conditions which decoupled the actual vs. environmentally-expected 
growth rate. Gene expression followed the environmental signal according to the expected, 
but not the actual, growth rate. Simultaneous monitoring of gene expression and growth 
rate in continuous cultures further confirmed that ribosome biogenesis genes responded 
rapidly to changes in the environments but were oblivious to longer-term changes in growth
rate. Our results suggest that the capacity to anticipate and prepare for environmentally-
mediated changes in cell growth presented a major selection force during yeast evolution.

PMID: 17327914

GEO ID: GSE6302

PCL filename	short description	# conditions	tags	# channels
GSE6302_set10_family.pcl	Response to histidine addition in a histidine-limited chemostat; time course	8	amino acid utilization|nutrient utilization	2

File last modified: Thu Apr 25 2024
GSE6302_set09_family.pcl	Response to clotrimazole in a histidine-limited chemostat; time course	10	chemical stimulus	2

File last modified: Thu Apr 25 2024
GSE6302_set08_family.pcl	Response to NaCl in a histidine-limited chemostat; time course	8	osmotic stress|stress	2

File last modified: Thu Apr 25 2024
GSE6302_set07_family.pcl	Response to heat shock in a histidine-limited chemostat; time course	10	heat shock|stress	2

File last modified: Thu Apr 25 2024
GSE6302_set06_family.pcl	Response to DTT in a histidine-limited chemostat; time course	7	chemical stimulus|response to unfolded protein|stress	2

File last modified: Thu Apr 25 2024
GSE6302_set05_family.pcl	Response to glucose addition in a glucose-limited chemostat; time course	8	carbon utilization	2

File last modified: Thu Apr 25 2024
GSE6302_set04_family.pcl	Response to H2O2 in a glucose-limited chemostat; time course	8	chemical stimulus|oxidative stress|stress	2

File last modified: Thu Apr 25 2024
GSE6302_set03_family.pcl	Response to high osmolarity (NaCl) in a glucose-limited chemostat; time course	8	osmotic stress|stress	2

File last modified: Thu Apr 25 2024
GSE6302_set02_family.pcl	Response to heat shock in a glucose-limited chemostat; time course	8	heat shock|stress	2

File last modified: Thu Apr 25 2024
GSE6302_set01_family.pcl	Response to DTT in a glucose-limited chemostat; time course	8	chemical stimulus|response to unfolded protein|stress	2

File last modified: Thu Apr 25 2024
GSE6302_set00_family.pcl	Effect of glycerol vs glucose on growth of adh1 null mutants; includes time course	12	carbon utilization|fermentation|respiration	2

File last modified: Thu Apr 25 2024

---


# Liko_2007_PMID_17616518

Citation: Liko D, Slattery MG, Heideman W. 2007. Stb3 binds to ribosomal RNA processing element 
motifs that control transcriptional responses to growth in Saccharomyces cerevisiae.

Full Description: Transfer of quiescent Saccharomyces cerevisiae cells to fresh medium rapidly induces 
hundreds of genes needed for growth. A large subset of these genes is regulated via a DNA 
sequence motif known as the ribosomal RNA processing element (RRPE). However, no RRPE-
binding proteins have been identified. We screened a panel of 6144 glutathione S-
transferase-open reading frame fusions for RRPE-binding proteins and identified Stb3 as a 
specific RRPE-binding protein, both in vitro and in vivo. Chromatin immunoprecipitation 
experiments showed that glucose increases Stb3 binding to RRPE-containing promoters. 
Microarray experiments demonstrated that the loss of Stb3 inhibits the transcriptional 
response to fresh glucose, especially for genes with RRPE motifs. However, these 
experiments also showed that not all genes containing RRPEs were dependent on Stb3 for 
expression. Overall our data support a model in which Stb3 plays an important but not 
exclusive role in the transcriptional response to growth conditions.

PMID: 17616518

GEO ID: GSE8379

PCL filename	short description	# conditions	tags	# channels
GSE8379_setA_family.pcl	Stb3 binds to ribosomal RNA processing element motifs that control transcriptional responses to growth in Saccharomyces cerevisiae.	8	carbon utilization	1

File last modified: Thu Apr 25 2024

---


# Medintz_2007_PMID_17700863

Citation: Medintz IL, Vora GJ, Rahbar AM, Thach DC. 2007. Transcript and proteomic analyses of wild-
type and gpa2 mutant Saccharomyces cerevisiae strains suggest a role for glycolytic carbon
source sensing in pseudohyphal differentiation.

Full Description: In response to limited nitrogen and abundant carbon sources, diploid Saccharomyces 
cerevisiae strains undergo a filamentous transition in cell growth as part of pseudohyphal
differentiation. Use of the disaccharide maltose as the principal carbon source, in 
contrast to the preferred nutrient monosaccharide glucose, has been shown to induce a 
hyper-filamentous growth phenotype in a strain deficient for GPA2 which codes for a Galpha
protein component that interacts with the glucose-sensing receptor Gpr1p to regulate 
filamentous growth. In this report, we compare the global transcript and proteomic 
profiles of wild-type and Gpa2p deficient diploid yeast strains grown on both rich and 
nitrogen starved maltose media. We find that deletion of GPA2 results in significantly 
different transcript and protein profiles when switching from rich to nitrogen starvation 
media. The results are discussed with a focus on the genes associated with carbon 
utilization, or regulation thereof, and a model for the contribution of carbon sensing/
metabolism-based signal transduction to pseudohyphal differentiation is proposed.

PMID: 17700863

GEO ID: GSE7820

PCL filename	short description	# conditions	tags	# channels
GSE7820_setA_family.pcl	Response of gpa2 null mutants and wildtype cells to growth on nitrogen-rich maltose medium and after shifting to N-starved maltose medium	12	filamentous growth|nitrogen utilization|signaling|starvation	1

File last modified: Thu Apr 25 2024
GSE7820_set0_family.pcl	Response of gpa2 null mutants and wildtype cells to growth on nitrogen-rich maltose medium (YPM)	6	filamentous growth|nitrogen utilization|signaling	1

File last modified: Thu Apr 25 2024
GSE7820_set1_family.pcl	Response of gpa2 null mutants and wildtype cells after shifting from nitrogen-rich to N-starved maltose medium	6	filamentous growth|nitrogen utilization|signaling|starvation	1

File last modified: Thu Apr 25 2024

---


# Menacho-Marquez_2007_PMID_17890903

Citation: Menacho-Marquez M, Perez-Valle J, Arino J, Gadea J, Murguia JR. 2007. Gcn2p regulates a G1
/S cell cycle checkpoint in response to DNA damage.

Full Description: _.

PMID: 17890903

GEO ID: GSE8111

PCL filename	short description	# conditions	tags	# channels
GSE8111_setA_family.pcl	Gcn2p regulates a G1/S cell cycle checkpoint in response to DNA damage.	4	amino acid utilization|cell cycle regulation	2

File last modified: Thu Apr 25 2024

---


# Mendes-Ferreira_2007_PMID_17601813

Citation: Mendes-Ferreira A, del Olmo M, Garcia-Martinez J, Jimenez-Marti E, Leao C, Mendes-Faia A, 
Perez-Ortin JE. 2007. Saccharomyces cerevisiae signature genes for predicting nitrogen 
deficiency during alcoholic fermentation.

Full Description: Genome-wide analysis of the wine yeast strain Saccharomyces cerevisiae PYCC4072 identified
36 genes highly expressed under conditions of low or absent nitrogen in comparison with a 
nitrogen-replete condition. Reverse transcription-PCR analysis for four of these 
transcripts with this strain and its validation with another wine yeast strain underlines 
the usefulness of these signature genes for predicting nitrogen deficiency and therefore 
the diagnosis of wine stuck/sluggish fermentations.

PMID: 17601813

GEO ID: GSE5842

PCL filename	short description	# conditions	tags	# channels
GSE5842_set0_family.pcl	Saccharomyces cerevisiae signature genes for predicting nitrogen deficiency during alcoholic fermentation.	33	nitrogen utilization|starvation	1

File last modified: Thu Apr 25 2024

---


# Mutiu_2007_PMID_17447102

Citation: Mutiu AI, Hoke SM, Genereaux J, Liang G, Brandl CJ. 2007. The role of histone 
ubiquitylation and deubiquitylation in gene expression as determined by the analysis of an
HTB1(K123R) Saccharomyces cerevisiae strain.

Full Description: In Saccharomyces cerevisiae histone H2B is ubiquitylated at lysine 123 in a process 
requiring the E2-ubiquitin conjugase, Rad6. We have analyzed gene expression in a strain 
containing a variant of histone H2B with lysine 123 converted to arginine to address the 
mechanisms by which ubiquitylation and deubiquitylation of histone H2B affect gene 
expression. The SAGA complex component, Ubp8, is one of two proteases that remove the 
ubiquitin moiety at lysine 123. We show that changes in gene expression observed upon 
deletion of ubp8 are suppressed by htb1 ( K123R ), which provides genetic evidence that 
Ubp8 alters gene expression through deubiquitylation of histone H2B. Microarray analyses 
of the htb1 ( K123R ) strain show that loss of histone ubiquitylation results in a twofold
or greater change in expression of approximately 1.5% of the protein coding genes with 
approximately 75% of these increasing. For genes in which ubiquitylation represses 
expression, ubiquitylation principally acts through its effects on histone methylation. In
contrast, decreased expression of the CWP1 gene was not paralleled by deletions of 
methyltransferase components and is thus likely independent of methylation. Finally, by 
comparing gene expression changes in the htb1 ( K123R ) strain with those in a strain 
deleted for rad6, we conclude that lysine 123 affects transcription primarily because of 
it being a site of ubiquitylation.

PMID: 17447102

GEO ID: GSE6316

PCL filename	short description	# conditions	tags	# channels
GSE6316_setA_family.pcl	The role of histone ubiquitylation and deubiquitylation in gene expression as determined by the analysis of an HTB1(K123R) Saccharomyces cerevisiae strain.	9	histone modification|ubiquitin or ULP modification	2

File last modified: Thu Apr 25 2024

---


# Mutiu_2007_PMID_17660562

Citation: Mutiu AI, Hoke SM, Genereaux J, Hannam C, MacKenzie K, Jobin-Robitaille O, Guzzo J, Cote J,
Andrews B, Haniford DB, Brandl CJ. 2007. Structure/function analysis of the 
phosphatidylinositol-3-kinase domain of yeast tra1.

Full Description: Tra1 is an essential component of the Saccharomyces cerevisiae SAGA and NuA4 complexes. 
Using targeted mutagenesis, we identified residues within its C-terminal 
phosphatidylinositol-3-kinase (PI3K) domain that are required for function. The phenotypes
of tra1-P3408A, S3463A, and SRR3413-3415AAA included temperature sensitivity and reduced 
growth in media containing 6% ethanol or calcofluor white or depleted of phosphate. These 
alleles resulted in a twofold or greater change in expression of approximately 7% of yeast
genes in rich media and reduced activation of PHO5 and ADH2 promoters. Tra1-SRR3413 
associated with components of both the NuA4 and SAGA complexes and with the Gal4 
transcriptional activation domain similar to wild-type protein. Tra1-SRR3413 was recruited
to the PHO5 promoter in vivo but gave rise to decreased relative amounts of acetylated 
histone H3 and histone H4 at SAGA and NuA4 regulated promoters. Distinct from other 
components of these complexes, tra1-SRR3413 resulted in generation-dependent telomere 
shortening and synthetic slow growth in combination with deletions of a number of genes 
with roles in membrane-related processes. While the tra1 alleles have some phenotypic 
similarities with deletions of SAGA and NuA4 components, their distinct nature may arise 
from the simultaneous alteration of SAGA and NuA4 functions.

PMID: 17660562

GEO ID: GSE6847

PCL filename	short description	# conditions	tags	# channels
GSE6847_setA_family.pcl	Structure/function analysis of the phosphatidylinositol-3-kinase domain of yeast tra1.	6	chromatin organization|histone modification	2

File last modified: Thu Apr 25 2024

---


# Parra_2007_PMID_17724083

Citation: Parra MA, Wyrick JJ. 2007. Regulation of gene transcription by the histone H2A N-terminal 
domain.

Full Description: Histone N-terminal domains play critical roles in regulating chromatin structure and gene 
transcription. Relatively little is known, however, about the role of the histone H2A N-
terminal domain in transcription regulation. We have used DNA microarrays to characterize 
the changes in genome-wide expression caused by mutations in the N-terminal domain of 
histone H2A. Our results indicate that the N-terminal domain of histone H2A functions 
primarily to repress the transcription of a large subset of the Saccharomyces cerevisiae 
genome and that most of the H2A-repressed genes are also repressed by the histone H2B N-
terminal domain. Using the histone H2A microarray data, we selected three reporter genes (
BNA1, BNA2, and GCY1), which we subsequently used to map regions in the H2A N-terminal 
domain responsible for this transcriptional repression. These studies revealed that a 
small subdomain in the H2A N-terminal tail, comprised of residues 16 to 20, is required 
for the transcriptional repression of these reporter genes. Deletion of either the entire 
histone H2A N-terminal domain or just this small subdomain imparts sensitivity to UV 
irradiation. Finally, we show that two residues in this H2A subdomain, serine-17 and 
arginine-18, are specifically required for the transcriptional repression of the BNA2 
reporter gene.

PMID: 17724083

GEO ID: GSE7338

PCL filename	short description	# conditions	tags	# channels
GSE7338_setA_family.pcl	Expression in a histone H2A N-terminal K4, K7 acetylation defective mutant compared to wildtype; 3 replicates	6	histone modification	1

File last modified: Thu Apr 25 2024
GSE7337_setA_family.pcl	Expression in a histone H2A N-terminal deletion mutant compared to wildtype; 3 replicates	6	histone modification	1

File last modified: Thu Apr 25 2024

---


# Sheehan_2007_PMID_17201921

Citation: Sheehan KB, McInnerney K, Purevdorj-Gage B, Altenburg SD, Hyman LE. 2007. Yeast genomic 
expression patterns in response to low-shear modeled microgravity.

Full Description: UNLABELLED: The low-shear microgravity environment, modeled by rotating suspension culture
bioreactors called high aspect ratio vessels (HARVs), allows investigation in ground-based
studies of the effects of microgravity on eukaryotic cells and provides insights into the 
impact of space flight on cellular physiology. We have previously demonstrated that low-
shear modeled microgravity (LSMMG) causes significant phenotypic changes of a select group
of Saccharomyces cerevisiae genes associated with the establishment of cell polarity, 
bipolar budding, and cell separation. However, the mechanisms cells utilize to sense and 
respond to microgravity and the fundamental gene expression  changes that occur are 
largely unknown. In this study, we examined the global transcriptional response of yeast 
cells grown under LSMMG conditions using DNA microarray analysis in order to determine if 
exposure to LSMMG results in changes in gene expression. RESULTS: LSMMG differentially 
changed the expression of a significant number of genes (1372) when yeast cells were 
cultured for either five generations or twenty-five generations in HARVs, as compared to 
cells grown under identical conditions in normal gravity. We identified genes in cell wall
integrity signaling pathways containing MAP kinase cascades that may provide clues to 
novel physiological responses of eukaryotic cells to the external stress of a low-shear 
modeled microgravity environment. A comparison of the microgravity response to other 
environmental stress response (ESR) genes showed that 26% of the genes that respond 
significantly to LSMMG are involved in a general environmental stress response, while 74% 
of the genes may represent a unique transcriptional response to microgravity. In addition
, we found changes in genes  involved in budding, cell polarity establishment, and cell 
separation that validate our hypothesis that phenotypic changes observed in cells grown in
microgravity are reflected in genome-wide changes. This study documents a considerable 
response to yeast cell growth in low-shear modeled microgravity that is evident, at least 
in part, by changes in gene expression. Notably, we identified genes that are involved in 
cell signaling pathways that allow cells to detect environmental changes, to respond 
within the cell, and to change accordingly, as well as genes of unknown function that may 
have a unique transcriptional response to microgravity. We also uncovered significant 
changes in the expression of many genes involved in cell polarization and bud formation 
that correlate well with the phenotypic effects observed in yeast cells when grown under 
similar conditions. These results are noteworthy as they have implications for human space
flight.

PMID: 17201921

GEO ID: GSE4136.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE4136.final.pcl	Yeast genomic expression patterns in response to low-shear modeled microgravity.	12	stress	1

File last modified: Thu Apr 25 2024

---


# Smith_2007_PMID_17551510

Citation: Smith JJ, Ramsey SA, Marelli M, Marzolf B, Hwang D, Saleem RA, Rachubinski RA, Aitchison 
JD. 2007. Transcriptional responses to fatty acid are coordinated by combinatorial control.

Full Description: In transcriptional regulatory networks, the coincident binding of a combination of factors
to regulate a gene implies the existence of complex mechanisms to control both the gene 
expression profile and specificity of the response. Unraveling this complexity is a major 
challenge to biologists. Here, a novel network topology-based clustering approach was 
applied to condition-specific genome-wide chromatin localization and expression data to 
characterize a dynamic transcriptional regulatory network responsive to the fatty acid 
oleate. A network of four (predicted) regulators of the response (Oaf1p, Pip2p, Adr1p and 
Oaf3p) was investigated. By analyzing trends in the network structure, we found that two 
groups of multi-input motifs form in response to oleate, each controlling distinct 
functional classes of genes. This functionality is contributed in part by Oaf1p, which is 
a component of both types of multi-input motifs and has two different regulatory 
activities depending on its binding context. The dynamic cooperation between Oaf1p and 
Pip2p appears to temporally synchronize the two different responses. Together, these data 
suggest a network mechanism involving dynamic combinatorial control for coordinating 
transcriptional responses.

PMID: 17551510

GEO ID: GSE5862

PCL filename	short description	# conditions	tags	# channels
GSE5862_set0_family.pcl	Transcriptional responses to fatty acid are coordinated by combinatorial control.	20	carbon utilization|fermentation|lipid metabolism|respiration|stress	2

File last modified: Thu Apr 25 2024

---


# Stern_2007_PMID_17453047

Citation: Stern S, Dror T, Stolovicki E, Brenner N, Braun E. 2007. Genome-wide transcriptional 
plasticity underlies cellular adaptation to novel challenge.

Full Description: Cells adjust their transcriptional state to accommodate environmental and genetic 
perturbations. An open question is to what extent transcriptional response to 
perturbations has been specifically selected along evolution. To test the possibility that
transcriptional reprogramming does not need to be 'pre-designed' to lead to an adaptive 
metabolic state on physiological timescales, we confronted yeast cells with a novel 
challenge they had not previously encountered. We rewired the genome by recruiting an 
essential gene, HIS3, from the histidine biosynthesis pathway to a foreign regulatory 
system, the GAL network responsible for galactose utilization. Switching medium to glucose
in a chemostat caused repression of the essential gene and presented the cells with a 
severe challenge to which they adapted over approximately 10 generations. Using genome-
wide expression arrays, we show here that a global transcriptional reprogramming (>1200 
genes) underlies the adaptation. A large fraction of the responding genes is 
nonreproducible in repeated experiments. These results show that a nonspecific 
transcriptional response reflecting the natural plasticity of the regulatory network 
supports adaptation of cells to novel challenges.

PMID: 17453047

GEO ID: GSE6450

PCL filename	short description	# conditions	tags	# channels
GSE6450_setA_family.pcl	Genome-wide transcriptional plasticity underlies cellular adaptation to novel challenge.	22	evolution	2

File last modified: Thu Apr 25 2024

---


# Tai_2007_PMID_17928405

Citation: Tai SL, Daran-Lapujade P, Walsh MC, Pronk JT, Daran JM. 2007. Acclimation of Saccharomyces
cerevisiae to low temperature: a chemostat-based transcriptome analysis.

Full Description: Effects of suboptimal temperatures on transcriptional regulation in yeast have been 
extensively studied in batch cultures. To eliminate indirect effects of specific growth 
rates that are inherent to batch-cultivation studies, genome-wide transcriptional 
responses to low temperatures were analyzed in steady-state chemostats, grown at a fixed 
specific growth rate (0.03 h(-1)). Although in vivo  metabolic fluxes were essentially the
same in cultures grown at 12 and at 30 degrees C, concentrations of the growth-limiting 
nutrients (glucose or ammonia) were higher at 12 degrees C. This difference was reflected 
by transcript levels of genes that encode transporters for the growth-limiting nutrients. 
Several transcriptional responses to low temperature occurred under both nutrient-
limitation regimes. Increased transcription of ribosome-biogenesis genes emphasized the 
importance of adapting protein-synthesis capacity to low temperature. In contrast to 
observations in cold-shock and batch-culture studies, transcript levels of environmental 
stress response genes were reduced at 12 degrees C. Transcription of trehalose-
biosynthesis genes and intracellular trehalose levels indicated that, in contrast to its 
role in cold-shock adaptation, trehalose is not involved in steady-state low-temperature 
adaptation. Comparison of the chemostat-based transcriptome data with literature data 
revealed large differences between transcriptional reprogramming during long-term low-
temperature acclimation and the transcriptional responses to a rapid transition to low 
temperature.

PMID: 17928405

GEO ID: GSE6190.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE6190.final.pcl	Acclimation of Saccharomyces cerevisiae to low temperature: a chemostat-based transcriptome analysis.	12	heat shock	1

File last modified: Thu Apr 25 2024

---


# Thorsen_2007_PMID_17327492

Citation: Thorsen M, Lagniel G, Kristiansson E, Junot C, Nerman O, Labarre J, Tamas MJ. 2007. 
Quantitative transcriptome, proteome, and sulfur metabolite profiling of the Saccharomyces
cerevisiae response to arsenite.

Full Description: Arsenic is ubiquitously present in nature, and various mechanisms have evolved enabling 
cells to evade toxicity and acquire tolerance. Herein, we explored how Saccharomyces 
cerevisiae (budding yeast) respond to trivalent arsenic (arsenite) by quantitative 
transcriptome, proteome, and sulfur metabolite profiling. Arsenite exposure affected 
transcription of genes encoding functions related to protein biosynthesis, arsenic 
detoxification, oxidative stress defense, redox maintenance, and proteolytic activity. 
Importantly, we observed that nearly all components of the sulfate assimilation and 
glutathione biosynthesis pathways were induced at both gene and protein levels. Kinetic 
metabolic profiling evidenced a significant increase in the pools of sulfur metabolites as
well as elevated cellular glutathione levels. Moreover, the flux in the sulfur 
assimilation pathway as well as the glutathione synthesis rate strongly increased with a 
concomitant reduction of sulfur incorporation into proteins. By combining comparative 
genomics and molecular analyses, we pinpointed transcription factors that mediate the core
of the transcriptional response to arsenite. Taken together, our data reveal that arsenite
-exposed cells channel a large part of assimilated sulfur into glutathione biosynthesis, 
and we provide evidence that the transcriptional regulators Yap1p and Met4p control this 
response in concert.

PMID: 17327492

GEO ID: GSE6129

PCL filename	short description	# conditions	tags	# channels
GSE6129_set3_family.pcl	Response to 0.2 mM arsenite (AsIII) induced stress in met4 null vs wildtype cells at 1 hr; 6 replicates	6	chemical stimulus|metal or metalloid ion stress|stress	2

File last modified: Thu Apr 25 2024
GSE6129_set2_family.pcl	Response to 0.2 mM or 1 mM arsenite (AsIII) induced stress in yap1 null vs wildtype cells at hr (3 replicates)	6	chemical stimulus|metal or metalloid ion stress|stress	2

File last modified: Thu Apr 25 2024
GSE6129_set1_family.pcl	Response to 0.2 mM or 1 mM arsenite (AsIII) induced stress at various time points; 3 replicates	12	chemical stimulus|metal or metalloid ion stress|stress	2

File last modified: Thu Apr 25 2024
GSE6129_set0_family.pcl	Response to 1.0 mM arsenite (AsIII) induced stress vs unstressed cells at 1 hr; 6 replicates	6	chemical stimulus|metal or metalloid ion stress|stress	2

File last modified: Thu Apr 25 2024
GSE6068_setA_family.pcl	Response to 1.0 mM arsenite (AsIII) induced stress vs unstressed cells at various time points; multiple replicates	15	chemical stimulus|metal or metalloid ion stress|stress	2

File last modified: Thu Apr 25 2024

---


# Tompa_2007_PMID_17179083

Citation: Tompa R, Madhani HD. 2007. Histone H3 lysine 36 methylation antagonizes silencing in 
Saccharomyces cerevisiae independently of the Rpd3S histone deacetylase complex.

Full Description: In yeast, methylation of histone H3 on lysine 36 (H3-K36) is catalyzed by the NSD1 
leukemia oncoprotein homolog Set2. The histone deacetylase complex Rpd3S is recruited to 
chromatin via binding of the chromodomain protein Eaf3 to methylated H3-K36 to prevent 
erroneous transcription initiation. Here we identify a distinct function for H3-K36 
methylation. We used random mutagenesis of histones H3 and H4 followed by a reporter-based
screen to identify residues necessary to prevent the ectopic spread of silencing from the 
silent mating-type locus HMRa into flanking euchromatin. Mutations in H3-K36 or deletion 
of SET2 caused ectopic silencing of a heterochromatin-adjacent reporter. Transcriptional 
profiling revealed that telomere-proximal genes are enriched for those that display 
decreased expression in a set2Delta strain. Deletion of SIR4 rescued the expression defect
of 26 of 37 telomere-proximal genes with reduced expression in set2Delta cells, implying 
that H3-K36 methylation prevents the spread of telomeric silencing. Indeed, Sir3 spreads 
from heterochromatin into neighboring euchromatin in set2Delta cells. Furthermore, genetic
experiments demonstrated that cells lacking the Rpd3S-specific subunits Eaf3 or Rco1 did 
not display the anti-silencing phenotype of mutations in SET2 or H3-K36. Thus, antagonism 
of silencing is independent of the only known effector of this conserved histone 
modification.

PMID: 17179083

GEO ID: GSE4934

PCL filename	short description	# conditions	tags	# channels
GSE4934_setA_family.pcl	Histone H3 lysine 36 methylation antagonizes silencing in Saccharomyces cerevisiae independently of the Rpd3S histone deacetylase complex.	8	chromatin organization|histone modification	2

File last modified: Thu Apr 25 2024

---


# Torres_2007_PMID_17702937

Citation: Torres EM, Sokolsky T, Tucker CM, Chan LY, Boselli M, Dunham MJ, Amon A. 2007. Effects of 
aneuploidy on cellular physiology and cell division in haploid yeast.

Full Description: Aneuploidy is a condition frequently found in tumor cells, but its effect on cellular 
physiology is not known. We have characterized one aspect of aneuploidy: the gain of extra
chromosomes. We created a collection of haploid yeast strains that each bear an extra copy
of one or more of almost all of the yeast chromosomes. Their characterization revealed 
that aneuploid strains share a number of phenotypes, including defects in cell cycle 
progression, increased glucose uptake, and increased sensitivity to conditions interfering
with protein synthesis and protein folding. These phenotypes were observed only in strains
carrying additional yeast genes, which indicates that they reflect the consequences of 
additional protein production as well as the resulting imbalances in cellular protein 
composition. We conclude that aneuploidy causes not only a proliferative disadvantage but 
also a set of phenotypes that is independent of the identity of the individual extra 
chromosomes.

PMID: 17702937

GEO ID: GSE7812

PCL filename	short description	# conditions	tags	# channels
GSE7812_set1_family.pcl	Effects of aneuploidy in strains disomic for individual chromosomes; chemostat, phosphate-limiting conditions	18	ploidy|phosphorus utilization	2

File last modified: Thu Apr 25 2024
GSE7812_set0_family.pcl	Effects of aneuploidy in strains disomic for individual chromosomes; batch culture, 2 replicates	44	ploidy	2

File last modified: Thu Apr 25 2024

---


# Urban_2007_PMID_17560372

Citation: Urban J, Soulard A, Huber A, Lippman S, Mukhopadhyay D, Deloche O, Wanke V, Anrather D, 
Ammerer G, Riezman H, Broach JR, De Virgilio C, Hall MN, Loewith R. 2007. Sch9 is a major 
target of TORC1 in Saccharomyces cerevisiae.

Full Description: The Target of Rapamycin (TOR) protein is a Ser/Thr kinase that functions in two distinct 
multiprotein complexes: TORC1 and TORC2. These conserved complexes regulate many different
aspects of cell growth in response to intracellular and extracellular cues. Here we report
that the AGC kinase Sch9 is a substrate of yeast TORC1. Six amino acids in the C terminus 
of Sch9 are directly phosphorylated by TORC1. Phosphorylation of these residues is lost 
upon rapamycin treatment as well as carbon or nitrogen starvation and transiently reduced 
following application of osmotic, oxidative, or thermal stress. TORC1-dependent 
phosphorylation is required for Sch9 activity, and replacement of residues phosphorylated 
by TORC1 with Asp/Glu renders Sch9 activity TORC1 independent. Sch9 is required for TORC1 
to properly regulate ribosome biogenesis, translation initiation, and entry into G0 phase
, but not expression of Gln3-dependent genes. Our results suggest that Sch9 functions 
analogously to the mammalian TORC1 substrate S6K1 rather than the mTORC2 substrate PKB/Akt.

PMID: 17560372

GEO ID: GSE7660

PCL filename	short description	# conditions	tags	# channels
GSE7660_set1_family.pcl	Response to rapamycin in cells carrying a CEN based copy of SCH9; 3 hr time course	5	chemical stimulus|protein phosphorylation|signaling|stress	2

File last modified: Thu Apr 25 2024
GSE7660_set2_family.pcl	Response to rapamycin in CEN based copy of sch9-2D3E, a TORC1 phosphorylation site mutant; 3 hr time course	6	chemical stimulus|protein phosphorylation|signaling|stress	2

File last modified: Thu Apr 25 2024
GSE7660_set0_family.pcl	Response to rapamycin in wildtype cells carrying a CEN plasmid; 3 hr time course	8	chemical stimulus|signaling	2

File last modified: Thu Apr 25 2024

---


# Vemuri_2007_PMID_17287356

Citation: Vemuri GN, Eiteman MA, McEwen JE, Olsson L, Nielsen J. 2007. Increasing NADH oxidation 
reduces overflow metabolism in Saccharomyces cerevisiae.

Full Description: Respiratory metabolism plays an important role in energy production in the form of ATP in 
all aerobically growing cells. However, a limitation in respiratory capacity results in 
overflow metabolism, leading to the formation of byproducts, a phenomenon known as 
overflow metabolism or the Crabtree effect. The yeast Saccharomyces cerevisiae has served 
as an important model organism for studying the Crabtree effect. When subjected to 
increasing glycolytic fluxes under aerobic conditions, there is a threshold value of the 
glucose uptake rate at which the metabolism shifts from purely respiratory to mixed 
respiratory and fermentative. It is well known that glucose repression of respiratory 
pathways occurs at high glycolytic fluxes, resulting in a decrease in respiratory capacity
. Despite many years of detailed studies on this subject, it is not known whether the 
onset of the Crabtree effect is due to limited respiratory capacity or is caused by 
glucose-mediated repression of respiration. When respiration in S. cerevisiae was 
increased by introducing a heterologous alternative oxidase, we observed reduced aerobic 
ethanol formation. In contrast, increasing nonrespiratory NADH oxidation by overexpression
of a water-forming NADH oxidase reduced aerobic glycerol formation. The metabolic response
to elevated alternative oxidase occurred predominantly in the mitochondria, whereas NADH 
oxidase affected genes that catalyze cytosolic reactions. Moreover, NADH oxidase restored 
the deficiency of cytosolic NADH dehydrogenases in S. cerevisiae. These results indicate 
that NADH oxidase localizes in the cytosol, whereas alternative oxidase is directed to the
mitochondria.

PMID: 17287356

GEO ID: GSE6277

PCL filename	short description	# conditions	tags	# channels
GSE6277_setA_family.pcl	Increasing NADH oxidation reduces overflow metabolism in Saccharomyces cerevisiae.	6	carbon utilization|respiration	1

File last modified: Thu Apr 25 2024

---


# Venkatasubrahmanyam_2007_PMID_17925448

Citation: Venkatasubrahmanyam S, Hwang WW, Meneghini MD, Tong AH, Madhani HD. 2007. Genome-wide, as 
opposed to local, antisilencing is mediated redundantly by the euchromatic factors Set1 
and H2A.Z.

Full Description: In Saccharomyces cerevisiae, several nonessential mechanisms including histone variant H2A
.Z deposition and transcription-associated histone H3 methylation antagonize the local 
spread of Sir-dependent silent chromatin into adjacent euchromatic regions. However, it is
unclear how and where these factors cooperate. To probe this question, we performed 
systematic genetic array screens for gene deletions that cause a synthetic growth defect 
in an htz1Delta mutant but not in an htz1Delta sir3Delta double mutant. Of the four genes 
identified, three, SET1, SWD1, and SWD3, encode components of the Set1 complex, which 
catalyzes the methylation of histone H3 on lysine 4 (H3-K4), a highly conserved 
modification that occurs in the coding sequences of transcribed genes. Using microarray-
based transcriptional profiling, we find that H2A.Z and Set1 cooperate to prevent Sir-
dependent repression of a large number of genes located across the genome, rather than the
local effects reported previously for the individual mechanisms. This global, redundant 
function appears to be direct: using a DamID chromatin profiling method, we demonstrate 
ectopic association of Sir3 and Sir4 in htz1Delta set1Delta mutants at loci distant from 
silent chromatin domains. Antisilencing mechanisms may therefore cooperate to play a 
considerably broader role in regulating genome-wide transcription than previously thought.

PMID: 17925448

GEO ID: GSE4826

PCL filename	short description	# conditions	tags	# channels
GSE4826_setA_family.pcl	Genome-wide, as opposed to local, antisilencing is mediated redundantly by the euchromatic factors Set1 and H2A.Z.	16	chromatin organization|histone modification	2

File last modified: Thu Apr 25 2024

---


# Wiesenberger_2007_PMID_17337637

Citation: Wiesenberger G, Steinleitner K, Malli R, Graier WF, Vormann J, Schweyen RJ, Stadler JA. 
2007. Mg2+ deprivation elicits rapid Ca2+ uptake and activates Ca2+/calcineurin signaling 
in Saccharomyces cerevisiae.

Full Description: To learn about the cellular processes involved in Mg(2+) homeostasis and the mechanisms 
allowing cells to cope with low Mg(2+) availability, we performed RNA  expression-
profiling experiments and followed changes in gene activity upon Mg(2+) depletion on a 
genome-wide scale. A striking portion of genes up-regulated under Mg(2+) depletion are 
also induced by high Ca(2+) and/or alkalinization. Among the genes significantly up-
regulated by Mg(2+) starvation, Ca(2+) stress, and alkalinization are ENA1 (encoding a P-
type ATPase sodium pump) and PHO89 (encoding a sodium/phosphate cotransporter). We show 
that up-regulation of these  genes is dependent on the calcineurin/Crz1p (calcineurin-
responsive zinc finger protein) signaling pathway. Similarly to Ca(2+) stress, Mg(2+) 
starvation induces translocation of the transcription factor Crz1p from the cytoplasm into
the nucleus. The up-regulation of ENA1 and PHO89 upon Mg(2+) starvation depends on 
extracellular Ca(2+). Using fluorescence resonance energy transfer microscopy, we 
demonstrate that removal of Mg(2+) results in an immediate increase in free cytoplasmic Ca
(2+). This effect is dependent on external Ca(2+). The results presented indicate that Mg(
2+) depletion in yeast cells leads to enhanced cellular Ca(2+) concentrations, which 
activate the Crz1p/calcineurin pathway. We  provide evidence that calcineurin/Crz1p 
signaling is crucial for yeast cells to cope with Mg(2+) depletion stress.

PMID: 17337637

GEO ID: GSE6687.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE6687.final.pcl	Mg2+ deprivation elicits rapid Ca2+ uptake and activates Ca2+/calcineurin signaling in Saccharomyces cerevisiae.	4	metal or metalloid ion utilization	2

File last modified: Thu Apr 25 2024

---


# Yarragudi_2007_PMID_17158163

Citation: Yarragudi A, Parfrey LW, Morse RH. 2007. Genome-wide analysis of transcriptional 
dependence and probable target sites for Abf1 and Rap1 in Saccharomyces cerevisiae.

Full Description: Abf1 and Rap1 are general regulatory factors (GRFs) that contribute to transcriptional 
activation of a large number of genes, as well as to replication, silencing and telomere 
structure in yeast. In spite of their widespread roles in transcription, the scope of 
their functional targets genome-wide has not been previously determined. Here, we use 
microarrays to examine the contribution of these essential GRFs to transcription genome-
wide, by using ts mutants that dissociate from their binding sites at 37 degrees C. We 
then combine this data with published ChIP-chip studies and motif analysis to identify 
probable direct targets for Abf1 and Rap1. We also identify a substantial number of genes 
likely to bind Rap1 or Abf1, but not affected by loss of GRF binding. Interestingly, the 
results strongly suggest that Rap1 can contribute to gene activation from farther upstream
than can Abf1. Also, consistent with previous work, more genes that bind Abf1 are 
unaffected by loss of binding than those that bind Rap1. Finally, we show for several such
genes that the Abf1 C-terminal region, which contains the putative activation domain, is 
not needed to confer this peculiar 'memory effect' that allows continued transcription 
after loss of Abf1 binding.

PMID: 17158163

GEO ID: GSE6073

PCL filename	short description	# conditions	tags	# channels
GSE6073_setA_family.pcl	Genome-wide analysis of transcriptional dependence and probable target sites for Abf1 and Rap1 in Saccharomyces cerevisiae.	12	transcription	1

File last modified: Thu Apr 25 2024

---


# Abbott_2008_PMID_18676708

Citation: Abbott DA, Suir E, van Maris AJ, Pronk JT. 2008. Physiological and transcriptional 
responses to high concentrations of lactic acid in anaerobic chemostat cultures of 
Saccharomyces cerevisiae.

Full Description: Based on the high acid tolerance and the simple nutritional requirements of Saccharomyces 
cerevisiae, engineered strains of this yeast are considered biocatalysts for industrial 
production of high-purity undissociated lactic acid. However, high concentrations of 
lactic acid are toxic to S. cerevisiae, thus limiting its growth and product formation. 
Physiological and transcriptional responses to high concentrations of lactic acid were 
studied in anaerobic, glucose-limited chemostat cultures grown at different pH values and 
lactic acid concentrations, resulting in a 50% decrease in the biomass yield. At pH 5, the
yield decrease was caused mostly by osmotically induced glycerol production and not by the
classic weak-acid action, as was observed at pH 3. Cultures grown at pH 5 with 900 mM 
lactic acid revealed an upregulation of many genes involved in iron homeostasis, 
indicating that iron chelation occurred at high concentrations of dissociated lactic acid
. Chemostat cultivation at pH 3 with 500 mM lactate, resulting in lower anion 
concentrations, showed an alleviation of this iron homeostasis response. Six of the 10 
known targets of the transcriptional regulator Haa1p were strongly upregulated in lactate-
challenged cultures at pH 3 but showed only moderate induction by high lactate 
concentrations at pH 5. Moreover, the haa1Delta mutant exhibited a growth defect at high 
lactic acid concentrations at pH 3. These results indicate that iron homeostasis plays a 
major role in the response of S. cerevisiae to high lactate concentrations, whereas the 
Haa1p regulon is involved primarily in the response to high concentrations of 
undissociated lactic acid.

PMID: 18676708

GEO ID: GSE10066

PCL filename	short description	# conditions	tags	# channels
GSE10066_setA_family.pcl	Physiological and transcriptional responses to high concentrations of lactic acid in anaerobic chemostat cultures of Saccharomyces cerevisiae.	12	chemical stimulus|oxygen level alteration|stress	1

File last modified: Thu Apr 25 2024

---


# Agarwal_2008_PMID_18156292

Citation: Agarwal AK, Xu T, Jacob MR, Feng Q, Lorenz MC, Walker LA, Clark AM. 2008. Role of heme in 
the antifungal activity of the azaoxoaporphine alkaloid sampangine.

Full Description: Sampangine, a plant-derived alkaloid found in the Annonaceae family, exhibits strong 
inhibitory activity against the opportunistic fungal pathogens Candida albicans, 
Cryptococcus neoformans, and Aspergillus fumigatus. In the present study, transcriptional 
profiling experiments coupled with analyses of mutants were performed in an effort to 
elucidate its mechanism of action. Using Saccharomyces cerevisiae as a model organism, we 
show that sampangine produces a transcriptional response indicative of hypoxia, altering 
the expression of genes known to respond to low-oxygen conditions. Several additional 
lines of evidence obtained suggest that these responses could involve effects on heme. 
First, the hem1Delta mutant lacking the first enzyme in the heme biosynthetic pathway 
showed increased sensitivity to sampangine, and exogenously supplied hemin partially 
rescued the inhibitory activity of sampangine in wild-type cells. In addition, 
heterozygous mutants with deletions in genes involved in five out of eight steps in the 
heme biosynthetic pathway showed increased susceptibility to sampangine. Furthermore, 
spectral analyses of pyridine extracts indicated significant accumulation of free 
porphyrins in sampangine-treated cells. Transcriptional profiling experiments were also 
performed with C. albicans to investigate the response of a pathogenic fungal species to 
sampangine. Taking into account the known differences in the physiological responses of C
. albicans and S. cerevisiae to low oxygen, significant correlations were observed between
the two transcription profiles, suggestive of heme-related defects. Our results indicate 
that the antifungal activity of the plant alkaloid sampangine is due, at least in part, to
perturbations in the biosynthesis or metabolism of heme.

PMID: 18156292

GEO ID: GSE10073

PCL filename	short description	# conditions	tags	# channels
GSE10073_setA_family.pcl	Role of heme in the antifungal activity of the azaoxoaporphine alkaloid sampangine.	6	chemical stimulus|cofactor metabolism	1

File last modified: Thu Apr 25 2024

---


# Aragon_2008_PMID_18199684

Citation: Aragon AD, Rodriguez AL, Meirelles O, Roy S, Davidson GS, Tapia PH, Allen C, Joe R, Benn D,
Werner-Washburne M. 2008. Characterization of differentiated quiescent and nonquiescent 
cells in yeast stationary-phase cultures.

Full Description: Cells in glucose-limited Saccharomyces cerevisiae cultures differentiate into quiescent (Q
) and nonquiescent (NQ) fractions before entering stationary phase. To understand this 
differentiation, Q and NQ cells from 101 deletion-mutant strains were tested for viability
and reproductive capacity. Eleven mutants that affected one or both phenotypes in Q or NQ 
fractions were identified. NQ fractions exhibit a high level of petite colonies, and nine 
mutants affecting this phenotype were identified. Microarray analysis revealed >1300 mRNAs
distinguished Q from NQ fractions. Q cell-specific mRNAs encode proteins involved in 
membrane maintenance, oxidative stress response, and signal transduction. NQ-cell mRNAs, 
consistent with apoptosis in these cells, encode proteins involved in Ty-element 
transposition and DNA recombination. More than 2000 protease-released mRNAs were 
identified only in Q cells, consistent with these cells being physiologically poised to 
respond to environmental changes. Our results indicate that Q and NQ cells differentiate 
significantly, with Q cells providing genomic stability and NQ cells providing nutrients 
to Q cells and a regular source of genetic diversity through mutation and transposition. 
These studies are relevant to chronological aging, cell cycle, and genome evolution, and 
they provide insight into complex responses that even simple organisms have to starvation.

PMID: 18199684

GEO ID: GSE8624

PCL filename	short description	# conditions	tags	# channels
GSE8624_set1_family.pcl	Expression in differentiated quiescent and non-quiescent S288C cells from stationary-phase cultures; 10 replicates	20	stationary phase maintenance	2

File last modified: Thu Apr 25 2024
GSE8624_set2_family.pcl	Expression in 101 differentiated quiescent and non-quiescent deletion mutants from stationary phase cultures; some replicates	368	stationary phase maintenance	2

File last modified: Thu Apr 25 2024
GSE8624_set0_family.pcl	Expression in differentiated quiescent and non-quiescent BY4742 cells from stationary-phase cultures; 10 replicates	26	stationary phase maintenance	2

File last modified: Thu Apr 25 2024

---


# Baerends_2008_PMID_18378663

Citation: Baerends RJ, de Hulster E, Geertman JM, Daran JM, van Maris AJ, Veenhuis M, van der Klei 
IJ, Pronk JT. 2008. Engineering and analysis of a Saccharomyces cerevisiae strain that 
uses formaldehyde as an auxiliary substrate.

Full Description: We demonstrated that formaldehyde can be efficiently coutilized by an engineered 
Saccharomyces cerevisiae strain that expresses Hansenula polymorpha genes encoding 
formaldehyde dehydrogenase (FLD1) and formate dehydrogenase (FMD), in contrast to wild-
type strains. Initial chemostat experiments showed that the engineered strain coutilized 
formaldehyde with glucose, but these mixed-substrate cultures failed to reach steady-state
conditions and did not exhibit an increased biomass yield on glucose. Subsequent 
transcriptome analyses of chemostat cultures of the engineered strain, grown on glucose-
formaldehyde mixtures, indicated that the presence of formaldehyde in the feed caused 
biotin limitations. Further transcriptome analysis demonstrated that this biotin 
inactivation was prevented by using separate formaldehyde and vitamin feeds. Using this 
approach, steady-state glucose-limited chemostat cultures were obtained that coutilized 
glucose and formaldehyde. Coutilization of formaldehyde under these conditions resulted in
an enhanced biomass yield of the glucose-limited cultures. The biomass yield was 
quantitatively consistent with the use of formaldehyde as an auxiliary substrate that 
generates NADH and subsequently, via oxidative phosphorylation, ATP. On an electron pair 
basis, the biomass yield increase observed with formaldehyde was larger than that observed
previously for formate, which is tentatively explained by different modes of formate and 
formaldehyde transport in S. cerevisiae.

PMID: 18378663

GEO ID: GSE8902

PCL filename	short description	# conditions	tags	# channels
GSE8902_setA_family.pcl	Engineering and analysis of a Saccharomyces cerevisiae strain that uses formaldehyde as an auxiliary substrate.	8	carbon utilization	1

File last modified: Thu Apr 25 2024

---


# Berry_2008_PMID_18753408

Citation: Berry DB, Gasch AP. 2008. Stress-activated genomic expression changes serve a preparative 
role for impending stress in yeast.

Full Description: Yeast cells respond to stress by mediating condition-specific gene expression changes and 
by mounting a common response to many stresses, called the environmental stress response (
ESR). Giaever et al. previously revealed poor correlation between genes whose expression 
changes in response to acute stress and genes required to survive that stress, raising 
question about the role of stress-activated gene expression. Here we show that gene 
expression changes triggered by a single dose of stress are not required to survive that 
stimulus but rather serve a protective role against future stress. We characterized the 
increased resistance to severe stress in yeast preexposed to mild stress. This acquired 
stress resistance is dependent on protein synthesis during mild-stress treatment and 
requires the general-stress transcription factors Msn2p and/or Msn4p that regulate 
induction of many ESR genes. However, neither protein synthesis nor Msn2/4p is required 
for basal tolerance of a single dose of stress, despite the substantial expression changes
triggered by each condition. Using microarrays, we show that Msn2p and Msn4p play 
nonredundant and condition-specific roles in gene-expression regulation, arguing against a
generic general-stress function. This work highlights the importance of condition-specific
responses in acquired stress resistance and provides new insights into the role of the ESR.

PMID: 18753408

GEO ID: GSE8335

PCL filename	short description	# conditions	tags	# channels
GSE8335_set10_family.pcl	Stress response of an msn4 deletion mutant exposed to NaCl for 45 min; 5 replicates	5	osmotic stress|stress	2

File last modified: Thu Apr 25 2024
GSE8335_set11_family.pcl	Stress response of an msn4 deletion mutant exposed to heat shock for 15 min; 5 replicates	5	heat shock|stress	2

File last modified: Thu Apr 25 2024
GSE8335_set12_family.pcl	Stress response of an msn2 msn4 double-deletion strain exposed to NaCl for 45 min; 5 replicates	5	osmotic stress|stress	2

File last modified: Thu Apr 25 2024
GSE8335_set13_family.pcl	Stress response of an msn2 msn4 double-deletion strain exposed to heat shock for 15 min; 5 replicates	5	heat shock|stress	2

File last modified: Thu Apr 25 2024
GSE8335_set06_family.pcl	Stress response of cells exposed to NaCl for 45 min; 6 replicates	6	osmotic stress|stress	2

File last modified: Thu Apr 25 2024
GSE8335_set07_family.pcl	Stress response of cells exposed to heat shock for 45 min; 5 replicates	5	heat shock|stress	2

File last modified: Thu Apr 25 2024
GSE8335_set08_family.pcl	Stress response of an msn2 deletion mutant exposed to NaCl for 45 min; 5 replicates	5	osmotic stress|stress	2

File last modified: Thu Apr 25 2024
GSE8335_set09_family.pcl	Stress response of an msn2 deletion mutant exposed to heat shock for 15 min; 5 replicates	5	heat shock|stress	2

File last modified: Thu Apr 25 2024
GSE8335_set05_family.pcl	Stress response of an msn2 msn4 double-deletion strain exposed to H2O2 for various times after NaCl pre-exposure	4	oxidative stress|stress	2

File last modified: Thu Apr 25 2024
GSE8335_set03_family.pcl	Stress response of an msn2 msn4 double-deletion strain exposed to NaCl for various times	4	osmotic stress|stress	2

File last modified: Thu Apr 25 2024
GSE8335_set04_family.pcl	Stress response of an msn2 msn4 double-deletion strain exposed to H2O2 for various times	4	oxidative stress|stress	2

File last modified: Thu Apr 25 2024
GSE8335_set02_family.pcl	Stress response of cells exposed to H2O2 for various times after NaCl pre-exposure; 2 replicates	8	oxidative stress|stress	2

File last modified: Thu Apr 25 2024
GSE8335_set01_family.pcl	Stress response of cells exposed to H2O2 for various times; 2 replicates	8	oxidative stress|stress	2

File last modified: Thu Apr 25 2024
GSE8335_set00_family.pcl	Stress response of cells exposed to NaCl for various times; 2 replicates	8	osmotic stress|stress	2

File last modified: Thu Apr 25 2024

---


# Brauer_2008_PMID_17959824

Citation: Brauer MJ, Huttenhower C, Airoldi EM, Rosenstein R, Matese JC, Gresham D, Boer VM, 
Troyanskaya OG, Botstein D. 2008. Coordination of growth rate, cell cycle, stress response
and metabolic activity in yeast.

Full Description: We studied the relationship between growth rate and genome-wide gene expression, cell 
cycle progression, and glucose metabolism in 36 steady-state continuous cultures limited 
by one of six different nutrients (glucose, ammonium, sulfate, phosphate, uracil, or 
leucine). The expression of more than one quarter of all yeast genes is linearly 
correlated with growth rate, independent of the limiting nutrient. The subset of 
negatively growth-correlated genes is most enriched for peroxisomal functions, whereas 
positively correlated genes mainly encode ribosomal functions. Many (not all) genes 
associated with stress response are strongly correlated with growth rate, as are genes 
that are periodically expressed under conditions of metabolic cycling. We confirmed a 
linear relationship between growth rate and the fraction of the cell population in the G0/
G1 cell cycle phase, independent of limiting nutrient. Cultures limited by auxotrophic 
requirements wasted excess glucose, whereas those limited on phosphate, sulfate, or 
ammonia did not; this phenomenon (reminiscent of the Warburg effect in cancer cells) was 
confirmed in batch cultures. Using an aggregate of gene expression values, we predict (in 
both continuous and batch cultures) an instantaneous growth rate. This concept is useful 
in interpreting the system-level connections among growth rate, metabolism, stress, and 
the cell cycle.

PMID: 17959824

GEO ID: GSE8825

PCL filename	short description	# conditions	tags	# channels
GSE8825_set5_family.pcl	Response to uracil limitation in steady-state continuous cultures; six dilution rates	6	nutrient utilization	2

File last modified: Thu Apr 25 2024
GSE8825_set3_family.pcl	Response to sulfate limitation in steady-state continuous cultures; six dilution rates	6	sulfur utilization	2

File last modified: Thu Apr 25 2024
GSE8825_set4_family.pcl	Response to leucine limitation in steady-state continuous cultures; six dilution rates	6	amino acid utilization	2

File last modified: Thu Apr 25 2024
GSE8825_set2_family.pcl	Response to phosphate limitation in steady-state continuous cultures; six dilution rates	6	phosphorus utilization	2

File last modified: Thu Apr 25 2024
GSE8825_set1_family.pcl	Response to ammonium limitation in steady-state continuous cultures; six dilution rates	6	nitrogen utilization	2

File last modified: Thu Apr 25 2024
GSE8825_set0_family.pcl	Response to glucose limitation in steady-state continuous cultures; six dilution rates	6	carbon utilization	2

File last modified: Thu Apr 25 2024

---


# Brown_2008_PMID_18422925

Citation: Brown KM, Landry CR, Hartl DL, Cavalieri D. 2008. Cascading transcriptional effects of a 
naturally occurring frameshift mutation in Saccharomyces cerevisiae.

Full Description: Gene-expression variation in natural populations is widespread, and its phenotypic effects
can be acted upon by natural selection. Only a few naturally segregating genetic 
differences associated with expression variation have been identified at the molecular 
level. We have identified a single nucleotide insertion in a vineyard isolate of 
Saccharomyces cerevisiae that has cascading effects through the gene-expression network. 
This allele is responsible for about 45% (103/230) of the genes that show differential 
gene expression among the homozygous diploid progeny produced by a vineyard isolate. Using
isogenic laboratory strains, we confirm that this allele causes dramatic differences in 
gene-expression levels of key genes involved in amino acid biosynthesis. The mutation is a
frameshift mutation in a mononucleotide run of eight consecutive T's in the coding region 
of the gene SSY1, which encodes a key component of a plasma-membrane sensor of 
extracellular amino acids. The potentially high rate of replication slippage of this 
mononucleotide repeat, combined with its relatively mild effects on growth rate in 
heterozygous genotypes, is sufficient to account for the persistence of this phenotype at 
low frequencies in natural populations.

PMID: 18422925

GEO ID: GSE9875

PCL filename	short description	# conditions	tags	# channels
GSE9875_setA_family.pcl	Cascading transcriptional effects of a naturally occurring frameshift mutation in Saccharomyces cerevisiae.	6	amino acid metabolism	2

File last modified: Thu Apr 25 2024

---


# Capaldi_2008_PMID_18931682

Citation: Capaldi AP, Kaplan T, Liu Y, Habib N, Regev A, Friedman N, O'Shea EK. 2008. Structure and 
function of a transcriptional network activated by the MAPK Hog1.

Full Description: Cells regulate gene expression using a complex network of signaling pathways, 
transcription factors and promoters. To gain insight into the structure and function of 
these networks, we analyzed gene expression in single- and multiple-mutant strains to 
build a quantitative model of the Hog1 MAPK-dependent osmotic stress response in budding 
yeast. Our model reveals that the Hog1 and general stress (Msn2/4) pathways interact, at 
both the signaling and promoter level, to integrate information and create a context-
dependent response. This study lays out a path to identifying and characterizing the role 
of signal integration and processing in other gene regulatory networks.

PMID: 18931682

GEO ID: GSE12270

PCL filename	short description	# conditions	tags	# channels
GSE12270_setA_family.pcl	Structure and function of a transcriptional network activated by the MAPK Hog1.	85	osmotic stress|stress|transcription	2

File last modified: Thu Apr 25 2024

---


# Chan_2008_PMID_18708563

Citation: Chan K, Roth MB. 2008. Anoxia-induced suspended animation in budding yeast as an 
experimental paradigm for studying oxygen-regulated gene expression.

Full Description: A lack of oxygen can force many organisms to enter into recoverable hypometabolic states. 
To better understand how organisms cope with oxygen deprivation, our laboratory previously
had shown that when challenged with anoxia, both the nematode Caenorhabditis elegans and 
embryos of the zebrafish Danio rerio enter into suspended animation, in which all life 
processes that can be observed by light microscopy reversibly halt pending the restoration
of oxygen (P. A. Padilla and M. B. Roth, Proc. Natl. Acad. Sci. USA 98:7331-7335, 2001, 
and P. A. Padilla, T. G. Nystul, R. A. Zager, A. C. Johnson, and M. B. Roth, Mol. Biol. 
Cell 13:1473-1483, 2002). Here, we show that both sporulating and vegetative cells of the 
budding yeast Saccharomyces cerevisiae also enter into a similar state of suspended 
animation when made anoxic on a nonfermentable carbon source. Transcriptional profiling 
using cDNA microarrays and follow-on quantitative real-time PCR analysis revealed a 
relative derepression of aerobic metabolism genes in carbon monoxide (CO)-induced anoxia 
when compared to nitrogen (N(2)) gas-induced anoxia, which is consistent with the known 
oxygen-mimetic effects of CO. We also found that mutants deleted for components of the 
mitochondrial retrograde signaling pathway can tolerate prolonged exposure to CO but not 
to N(2). We conclude that the cellular response to anoxia is dependent on whether the 
anoxic gas is an oxygen mimetic and that the mitochondrial retrograde signaling pathway is
functionally important for mediating this response.

PMID: 18708563

GEO ID: GSE12004

PCL filename	short description	# conditions	tags	# channels
GSE12004_set1_family.pcl	Response of acetate grown cells to nitrogen (N2) gas-induced anoxia, 2 hr time course	5	carbon utilization|oxygen level alteration|respiration|stress	2

File last modified: Thu Apr 25 2024
GSE12004_set0_family.pcl	Response of acetate grown cells to carbon monoxide (CO)-induced anoxia, 2 hr time course	5	carbon utilization|oxygen level alteration|respiration|stress	2

File last modified: Thu Apr 25 2024

---


# Chechik_2008_PMID_18953355

Citation: Chechik G, Oh E, Rando O, Weissman J, Regev A, Koller D. 2008. Activity motifs reveal 
principles of timing in  transcriptional control of the yeast metabolic network.

Full Description: Significant insight about biological networks arises from the study of network motifs--
overly abundant network subgraphs--but such wiring patterns do not specify when and how 
potential routes within a cellular network are used. To address this limitation, we 
introduce activity motifs, which capture patterns in the dynamic use of a network. Using 
this framework to analyze transcription in Saccharomyces cerevisiae metabolism, we find 
that cells use different timing activity motifs to optimize transcription timing in 
response to changing conditions: forward activation to produce metabolic compounds 
efficiently, backward shutoff to rapidly stop production of a detrimental product and 
synchronized activation for co-production of metabolites required for the same reaction. 
Measuring protein abundance over a time course reveals that mRNA timing motifs also occur 
at the protein level. Timing motifs significantly overlap with binding activity motifs, 
where genes in a linear chain have ordered binding affinity to a transcription factor, 
suggesting a mechanism for ordered transcription. Finely timed transcriptional regulation 
is therefore abundant in yeast metabolism, optimizing the organism's adaptation to new 
environmental conditions.

PMID: 18953355

GEO ID: GSE13219

PCL filename	short description	# conditions	tags	# channels
GSE13219_set12_family.pcl	Metabolic shift from SC medium with inositol (ino) to media with ino, without amino acids and Gal as carbon source; 4 hr time course	6	amino acid utilization|carbon utilization|starvation	2

File last modified: Thu Apr 25 2024
GSE13219_set11_family.pcl	Metabolic shift from SC medium with inositol (ino) to media with both ino, all amino acids and Gal as carbon source; 4 hr time course	5	carbon utilization	2

File last modified: Thu Apr 25 2024
GSE13219_set10_family.pcl	Metabolic shift from SC medium with inositol (ino) to media lacking ino but containing all amino acids with Gal as carbon source; 4 hr time course	5	carbon utilization	2

File last modified: Thu Apr 25 2024
GSE13219_set09_family.pcl	Metabolic shift from SC medium with inositol (ino) to media lacking both ino and all amino acids with Gal as carbon source; 4 hr time course	5	amino acid utilization|carbon utilization|starvation	2

File last modified: Thu Apr 25 2024
GSE13219_set08_family.pcl	Metabolic shift from SC medium with inositol (ino) to media with ino, but without amino acids and EtOH as carbon source; 4 hr time course, 2 replicates	8	amino acid utilization|carbon utilization|starvation	2

File last modified: Thu Apr 25 2024
GSE13219_set07_family.pcl	Metabolic shift from SC medium with inositol (ino) to media with both ino, all amino acids and EtOH as carbon source; 4 hr time course	5	carbon utilization	2

File last modified: Thu Apr 25 2024
GSE13219_set06_family.pcl	Metabolic shift from SC medium with inositol (ino) to media lacking ino but containing all amino acids with EtOH as carbon source; 4 hr time course	7	carbon utilization	2

File last modified: Thu Apr 25 2024
GSE13219_set05_family.pcl	Metabolic shift from SC medium with inositol (ino) to media lacking both ino and all amino acids with EtOH as carbon source; 4 hr time course	5	amino acid utilization|carbon utilization|starvation	2

File last modified: Thu Apr 25 2024
GSE13219_set04_family.pcl	Metabolic shift from SC medium with inositol (ino) to media with ino but without all amino acids; 4 hr time course	5	amino acid utilization|starvation	2

File last modified: Thu Apr 25 2024
GSE13219_set03_family.pcl	Metabolic shift from SC medium with inositol (ino) to media with ino but without specific amino acids (ARNCQGKPSDEFTY); 4 hr time course	5	amino acid utilization|starvation	2

File last modified: Thu Apr 25 2024
GSE13219_set02_family.pcl	Metabolic shift from SC medium with inositol (ino) to media with ino but without specific amino acids (LIVWHM); 4 hr time course	5	amino acid utilization|starvation	2

File last modified: Thu Apr 25 2024
GSE13219_set01_family.pcl	Metabolic shift from SC medium with inositol (ino) to media lacking ino but containing all amino acids; 4 hr time course	6	starvation	2

File last modified: Thu Apr 25 2024
GSE13219_set00_family.pcl	Metabolic shift from SC medium with inositol (ino) to media lacking both ino and all amino acids; 4 hr time course	5	amino acid utilization|starvation	2

File last modified: Thu Apr 25 2024

---


# Cheung_2008_PMID_18998772

Citation: Cheung V, Chua G, Batada NN, Landry CR, Michnick SW, Hughes TR, Winston F. 2008. Chromatin
- and transcription-related factors repress transcription from within coding regions 
throughout the Saccharomyces cerevisiae genome.

Full Description: Previous studies in Saccharomyces cerevisiae have demonstrated that cryptic promoters 
within coding regions activate transcription in particular mutants. We have performed a 
comprehensive analysis of cryptic transcription in order to identify factors that normally
repress cryptic promoters, to determine the amount of cryptic transcription genome-wide, 
and to study the potential for expression of genetic information by cryptic transcription
. Our results show that a large number of factors that control chromatin structure and 
transcription are required to repress cryptic transcription from at least 1,000 locations 
across the S. cerevisiae genome. Two results suggest that some cryptic transcripts are 
translated. First, as expected, many cryptic transcripts contain an ATG and an open 
reading frame of at least 100 codons. Second, several cryptic transcripts are translated 
into proteins. Furthermore, a subset of cryptic transcripts tested is transiently induced 
in wild-type cells following a nutritional shift, suggesting a possible physiological role
in response to a change in growth conditions. Taken together, our results demonstrate that
, during normal growth, the global integrity of gene expression is maintained by a wide 
range of factors and suggest that, under altered genetic or physiological conditions, the 
expression of alternative genetic information may occur.

PMID: 18998772

GEO ID: GSE12272

PCL filename	short description	# conditions	tags	# channels
GSE12272_setA_family.pcl	Chromatin- and transcription-related factors repress transcription from within coding regions throughout the Saccharomyces cerevisiae genome.	6	chromatin organization|transcription	2

File last modified: Thu Apr 25 2024

---


# Cipollina_2008_PMID_18174152

Citation: Cipollina C, van den Brink J, Daran-Lapujade P, Pronk JT, Vai M, de Winde JH. 2008. 
Revisiting the role of yeast Sfp1 in ribosome biogenesis and cell size control: a 
chemostat study.

Full Description: Saccharomyces cerevisiae SFP1 promotes transcription of a large cluster of genes involved 
in ribosome biogenesis. During growth in shake flasks, a mutant deleted for SFP1 shows a 
small size phenotype and a reduced growth rate. We characterized the behaviour of an 
sfp1Delta mutant compared to an isogenic reference strain growing in chemostat cultures at
the same specific growth rate. By studying glucose (anaerobic)- and ethanol (aerobic)-
limited cultures we focused specifically on nutrient-dependent effects. Major differences 
in the genome-wide transcriptional profiles were observed during glucose-limited growth. 
In particular, Sfp1 appeared to be involved in the control of ribosome biogenesis but not 
of ribosomal protein gene expression. Flow cytometric analyses revealed size defects for 
the mutant under both growth conditions. Our results suggest that Sfp1 plays a role in 
transcriptional and cell size control, operating at two different levels of the regulatory
network linking growth, metabolism and cell size.

PMID: 18174152

GEO ID: GSE5238

PCL filename	short description	# conditions	tags	# channels
GSE5238_setA_family.pcl	Revisiting the role of yeast Sfp1 in ribosome biogenesis and cell size control: a chemostat study.	12	nutrient utilization|diauxic shift	1

File last modified: Thu Apr 25 2024

---


# Cipollina_2008_PMID_18524923

Citation: Cipollina C, van den Brink J, Daran-Lapujade P, Pronk JT, Porro D, de Winde JH. 2008. 
Saccharomyces cerevisiae SFP1: at the crossroads of central metabolism and ribosome 
biogenesis.

Full Description: Saccharomyces cerevisiae SFP1 is required for nutrient-dependent regulation of ribosome 
biogenesis and cell size. A mutant deleted for SFP1 shows specific traits, including a 
slow growth phenotype, especially when growing on glucose. We recently analysed the 
physiology of an sfp1Delta mutant and its isogenic reference strain in chemostat cultures
. This approach was successful in revealing the effects of nutrients on the activity of 
Sfp1 independent of growth rate-related feedback. In the present work we exposed carbon-
limited cultures of an sfp1Delta mutant and its reference strain to sudden glucose excess
. This allowed us to study the effect of SFP1 deletion on cell physiology when the cells 
are forced to exploit their maximum growth potential; this is similar to what happens in 
shake-flask cultures but with no bias due to growth rate differences. We show that 
nutrients differentially affect the role of Sfp1 in cell-size modulation and in 
transcriptional control. Furthermore, we report that while Sfp1 is necessary for the 
efficient glucose-dependent regulation of ribosome biogenesis genes, it is not required 
for the proper induction of ribosomal protein genes in response to glucose excess. Finally
, our data suggest a role for Sfp1 in the regulation of glycolysis, further underlining 
its involvement in the network that links ribosome biogenesis and cell metabolism.

PMID: 18524923

GEO ID: GSE9644

PCL filename	short description	# conditions	tags	# channels
GSE9644_set1_family.pcl	Steady-state expression in carbon-limited chemostat culture and after an anaerobic glucose pulse in an sfp1 deletion mutant; time course, 2 replicates	13	carbon utilization|fermentation|oxygen level alteration|respiration|transcription	1

File last modified: Thu Apr 25 2024
GSE9644_set0_family.pcl	Steady-state expression in carbon-limited chemostat culture and after an anaerobic glucose pulse; time course, 2 replicates	13	carbon utilization|fermentation|oxygen level alteration|respiration	1

File last modified: Thu Apr 25 2024

---


# Erlich_2008_PMID_19005567

Citation: Erlich RL, Fry RC, Begley TJ, Daee DL, Lahue RS, Samson LD. 2008. Anc1, a protein 
associated with multiple transcription complexes, is involved in postreplication repair 
pathway in S. cerevisiae.

Full Description: Yeast strains lacking Anc1, a member of the YEATS protein family, are sensitive to several
DNA damaging agents. The YEATS family includes two human genes that are common fusion 
partners with MLL in human acute leukemias. Anc1 is a member of seven multi-protein 
complexes involved in transcription, and the damage sensitivity observed in anc1Delta 
cells is mirrored in strains deleted for some other non-essential members of several of 
these complexes. Here we show that ANC1 is in the same epistasis group as SRS2 and RAD5, 
members of the postreplication repair (PRR) pathway, but has additive or synergistic 
interactions with several other members of this pathway. Although PRR is traditionally 
divided into an error-prone and an error-free branch, ANC1 is not epistatic with all 
members of either established branch, and instead defines a new error-free branch of the 
PRR pathway. Like several genes involved in PRR, an intact ANC1 gene significantly 
suppresses spontaneous mutation rates, including the expansion of (CAG)(25) repeats. Anc1'
s role in the PRR pathway, as well as its role in suppressing triplet repeats, point to a 
possible mechanism for a protein of potential medical interest.

PMID: 19005567

GEO ID: GSE12150

PCL filename	short description	# conditions	tags	# channels
GSE12150_setA_family.pcl	Anc1, a protein associated with multiple transcription complexes, is involved in postreplication repair pathway in S. cerevisiae.	12	DNA damage stimulus	1

File last modified: Thu Apr 25 2024

---


# Gibson_2008_PMID_18373683

Citation: Gibson BR, Lawrence SJ, Boulton CA, Box WG, Graham NS, Linforth RS, Smart KA. 2008. The 
oxidative stress response of a lager brewing yeast strain during industrial propagation 
and fermentation.

Full Description: Commercial brewing yeast strains are exposed to a number of potential stresses including 
oxidative stress. The aim of this investigation was to measure the physiological and 
transcriptional changes of yeast cells during full-scale industrial brewing processes with
a view to determining the environmental factors influencing the cell's oxidative stress 
response. Cellular antioxidant levels and genome-wide transcriptional changes were 
monitored throughout an industrial propagation and fermentation. The greatest increase in 
cellular antioxidants and  transcription of antioxidant-encoding genes occurred as the 
rapidly fermentable sugars glucose and fructose were depleted from the growth medium (wort
) and the cell population entered the stationary phase. The data suggest that, contrary to
expectation, the oxidative stress response is not influenced by changes in the dissolved 
oxygen concentration of wort but is initiated as part of a general stress response to 
growth-limiting conditions, even in the absence of oxygen. A mechanism is proposed to 
explain the changes in antioxidant response observed in  yeast during anaerobic 
fermentation. The available data suggest that the yeast cell does not experience oxidative
stress during industrial brewery handling. This information may be taken into 
consideration when setting parameters for industrial brewery fermentation.

PMID: 18373683

GEO ID: GSE9423.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE9423.final.pcl	The oxidative stress response of a lager brewing yeast strain during industrial propagation and fermentation.	17	fermentation|oxidative stress	1

File last modified: Thu Apr 25 2024

---


# Gregori_2008_PMID_18621731

Citation: Gregori C, Schuller C, Frohner IE, Ammerer G, Kuchler K. 2008. Weak organic acids trigger 
conformational changes of the yeast transcription factor War1 in vivo to elicit stress 
adaptation.

Full Description: The Saccharomyces cerevisiae zinc cluster regulator War1 mediates an essential 
transcriptional and adaptive response to weak organic acid stress. Here we investigate the
mechanism of War1 activation upon weak acid stress. We identified several gain-of-function
WAR1 alleles mapping to the central War1 region. These mutations constitutively increase 
levels of the plasma membrane ABC transporter Pdr12, the main War1 target mediating stress
adaptation. Functional analysis of War1 reveals that the central region and its C-terminal
activation domain are required for function. Notably, the native DNA-binding and 
dimerization domains appear dispensable for War1 activity, because they can be replaced by
a LexA DNA-binding domain. Chromatin immunoprecipitation demonstrates elevated promoter  
affinity of activated War1, because its PDR12 promoter association increases upon stress. 
Hyperactive WAR1 alleles have constitutively high PDR12 promoter association. Furthermore
, fluorescence resonance energy transfer of functional CFP-War1-YFP proteins also 
demonstrates conformational changes of stress-activated War1 in vivo. Our results suggest 
a mechanism whereby War1 activation is accompanied by conformational changes enhancing 
promoter association, thus initiating the adaptation process.

PMID: 18621731

GEO ID: GSE8399.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE8399.final.pcl	Weak organic acids trigger conformational changes of the yeast transcription factor War1 in vivo to elicit stress adaptation.	3	chemical stimulus|stress	2

File last modified: Thu Apr 25 2024

---


# Gresham_2008_PMID_19079573

Citation: Gresham D, Desai MM, Tucker CM, Jenq HT, Pai DA, Ward A, DeSevo CG, Botstein D, Dunham MJ,
 2008. The repertoire and dynamics of evolutionary adaptations to controlled nutrient-
limited environments in yeast.

Full Description: The experimental evolution of laboratory populations of microbes provides an opportunity 
to observe the evolutionary dynamics of adaptation in real time. Until very recently, 
however, such studies have been limited by our inability to systematically find mutations 
in evolved organisms. We overcome this limitation by using a variety of DNA microarray-
based techniques to characterize genetic changes -- including point mutations, structural 
changes, and insertion variation -- that resulted from the experimental adaptation of 24 
haploid and diploid cultures of Saccharomyces cerevisiae to growth in either glucose, 
sulfate, or phosphate-limited chemostats for approximately 200 generations. We identified 
frequent genomic amplifications and rearrangements as well as novel retrotransposition 
events associated with adaptation. Global nucleotide variation detection in ten clonal 
isolates identified 32 point mutations. On the basis of mutation frequencies, we infer 
that these mutations and the subsequent dynamics of adaptation are determined by the batch
phase of growth prior to initiation of the continuous phase in the chemostat. We relate 
these genotypic changes to phenotypic outcomes, namely global patterns of gene expression
, and to increases in fitness by 5-50%. We found that the spectrum of available mutations 
in glucose- or phosphate-limited environments combined with the batch phase population 
dynamics early in our experiments allowed several distinct genotypic and phenotypic 
evolutionary pathways in response to these nutrient limitations. By contrast, sulfate-
limited populations were much more constrained in both genotypic and phenotypic outcomes. 
Thus, the reproducibility of evolution varies with specific selective pressures, 
reflecting the constraints inherent in the system-level organization of metabolic 
processes in the cell. We were able to relate some of the observed adaptive mutations (e.g
., transporter gene amplifications) to known features of the relevant metabolic pathways, 
but many of the mutations pointed to genes not previously associated with the relevant 
physiology. Thus, in addition to answering basic mechanistic questions about evolutionary 
mechanisms, our work suggests that experimental evolution can also shed light on the 
function and regulation of individual metabolic pathways.

PMID: 19079573

GEO ID: GSE13435

PCL filename	short description	# conditions	tags	# channels
GSE13435_set0_family.pcl	The repertoire and dynamics of evolutionary adaptations to controlled nutrient-limited environments in yeast.	63	evolution	2

File last modified: Thu Apr 25 2024

---


# Grund_2008_PMID_18762579

Citation: Grund SE, Fischer T, Cabal GG, Antunez O, Perez-Ortin JE, Hurt E. 2008. The inner nuclear 
membrane protein Src1 associates with subtelomeric genes and alters their regulated gene 
expression.

Full Description: Inner nuclear membrane proteins containing a LEM (LAP2, emerin, and MAN1) domain 
participate in different processes, including chromatin organization, gene expression, and
nuclear envelope biogenesis. In this study, we identify a robust genetic interaction 
between transcription export (TREX) factors and yeast Src1, an integral inner nuclear 
membrane protein that is homologous to vertebrate LEM2. DNA macroarray analysis revealed 
that the expression of the phosphate-regulated genes PHO11, PHO12, and PHO84 is up-
regulated in src1Delta cells. Notably, these PHO genes are located in subtelomeric regions
of chromatin and exhibit a perinuclear location in vivo. Src1 spans the nuclear membrane 
twice and exposes its N and C domains with putative DNA-binding motifs to the nucleoplasm
. Genome-wide chromatin immunoprecipitation-on-chip analyses indicated that Src1 is highly
enriched at telomeres and subtelomeric regions of the yeast chromosomes. Our data show 
that the inner nuclear membrane protein Src1 functions at the interface between 
subtelomeric gene expression and TREX-dependent messenger RNA export through the nuclear 
pore complexes.

PMID: 18762579

GEO ID: GSE6370

PCL filename	short description	# conditions	tags	# channels
GSE6370_set0_family.pcl	The inner nuclear membrane protein Src1 associates with subtelomeric genes and alters their regulated gene expression.	6	chromatin organization	1

File last modified: Thu Apr 25 2024

---


# Hazelwood_2008_PMID_18281432

Citation: Hazelwood LA, Daran JM, van Maris AJ, Pronk JT, Dickinson JR. 2008. The Ehrlich pathway 
for fusel alcohol production: a century of research on Saccharomyces cerevisiae metabolism.

Full Description: _.

PMID: 18281432

GEO ID: GSE9590

PCL filename	short description	# conditions	tags	# channels
GSE9590_setA_family.pcl	The Ehrlich pathway for fusel alcohol production: a century of research on Saccharomyces cerevisiae metabolism.	12	amino acid metabolism	1

File last modified: Thu Apr 25 2024

---


# Kao_2008_PMID_19029899

Citation: Kao KC, Sherlock G. 2008. Molecular characterization of clonal interference during 
adaptive evolution in asexual populations of Saccharomyces cerevisiae.

Full Description: The classical model of adaptive evolution in an asexual population postulates that each 
adaptive clone is derived from the one preceding it. However, experimental evidence has 
suggested more complex dynamics, with theory predicting the fixation probability of a 
beneficial mutation as dependent on the mutation rate, population size and the mutation's 
selection coefficient. Clonal interference has been demonstrated in viruses and bacteria 
but not in a eukaryote, and a detailed molecular characterization is lacking. Here we use 
three different fluorescent markers to visualize the dynamics of asexually evolving yeast 
populations. For each adaptive clone within one of our evolving populations, we identified
the underlying mutations, monitored their population frequencies and used microarrays to 
characterize changes in the transcriptome. These results represent the most detailed 
molecular characterization of experimental evolution to date and provide direct 
experimental evidence supporting both the clonal interference and the multiple mutation 
models.

PMID: 19029899

GEO ID: GSE11071

PCL filename	short description	# conditions	tags	# channels
GSE11071_setA_family.pcl	Molecular characterization of clonal interference during adaptive evolution in asexual populations of Saccharomyces cerevisiae.	29	evolution	2

File last modified: Thu Apr 25 2024

---


# Kaplan_2008_PMID_19023413

Citation: Kaplan T, Liu CL, Erkmann JA, Holik J, Grunstein M, Kaufman PD, Friedman N, Rando OJ. 2008
 Cell cycle- and chaperone-mediated regulation of H3K56ac incorporation in yeast.

Full Description: Acetylation of histone H3 lysine 56 is a covalent modification best known as a mark of 
newly replicated chromatin, but it has also been linked to replication-independent histone
replacement. Here, we measured H3K56ac levels at single-nucleosome resolution in 
asynchronously growing yeast cultures, as well as in yeast proceeding synchronously 
through the cell cycle. We developed a quantitative model of H3K56ac kinetics, which shows
that H3K56ac is largely explained by the genomic replication timing and the turnover rate 
of each nucleosome, suggesting that cell cycle profiles of H3K56ac should reveal most 
first-time nucleosome incorporation events. However, since the deacetylases Hst3/4 prevent
use of H3K56ac as a marker for histone deposition during M phase, we also directly 
measured M phase histone replacement rates. We report a global decrease in turnover rates 
during M phase and a further specific decrease in turnover at several early origins of 
replication, which switch from rapidly replaced in G1 phase to stably bound during M phase
. Finally, by measuring H3 replacement in yeast deleted for the H3K56 acetyltransferase 
Rtt109 and its two co-chaperones Asf1 and Vps75, we find evidence that Rtt109 and Asf1 
preferentially enhance histone replacement at rapidly replaced nucleosomes, whereas Vps75 
appears to inhibit histone turnover at those loci. These results provide a broad 
perspective on histone replacement/incorporation throughout the cell cycle and suggest 
that H3K56 acetylation provides a positive-feedback loop by which replacement of a 
nucleosome enhances subsequent replacement at the same location.

PMID: 19023413

GEO ID: GSE12822

PCL filename	short description	# conditions	tags	# channels
GSE12822_setA_family.pcl	Cell cycle- and chaperone-mediated regulation of H3K56ac incorporation in yeast.	34	mitotic cell cycle	2

File last modified: Thu Apr 25 2024

---


# Klockow_2008_PMID_18455824

Citation: Klockow C, Stahl F, Scheper T, Hitzmann B. 2008. In vivo regulation of glucose transporter
genes at glucose concentrations between 0 and 500 mg/L in a wild type of Saccharomyces 
cerevisiae.

Full Description: When the yeast Saccharomyces cerevisiae consumes glucose, the expression of the genes for 
the glucose transport is controlled via signal transduction pathways and sensor molecules
. Most publications describe the behavior of deletion strains while little is published 
about the in vivo regulation of glucose transporters in a wild type of S. cerevisiae. Here
a global gene expression analysis via microarray experiments from cultivations with 
glucose concentrations of 50, 70, 100 and 500 mg/L is presented. This permits the 
observation of the fine-tuning of gene expression in dependency on the glucose 
concentration. We detected indications that the transport system for high glucose 
concentrations is activated at glucose concentrations between 50 and 100 mg/L. The 
regulation of genes coding enzymes for the signal pathways and of those encoding the 
transporters themselves supports this assumption. The expression of sensor-, signal- and 
transporter genes will be discussed in detail. In addition, new information about the 
behavior of the so far little described carriers HXT8, HXT12, HXT13, HXT17 and GAL2 will 
be given. According to our findings, HXT13 is active during starvation. HXT12, HXT17 and 
GAL2 are used at low glucose concentrations. The carrier HXT8 supports the glucose 
transport both during starvation and at low glucose concentrations.

PMID: 18455824

GEO ID: GSE6358

PCL filename	short description	# conditions	tags	# channels
GSE6358_setA_family.pcl	In vivo regulation of glucose transporter genes at glucose concentrations between 0 and 500 mg/L in a wild type of Saccharomyces cerevisiae.	7	carbon utilization|starvation	2

File last modified: Thu Apr 25 2024

---


# Kundaje_2008_PMID_19008939

Citation: Kundaje A, Xin X, Lan C, Lianoglou S, Zhou M, Zhang L, Leslie C. 2008. A predictive model 
of the oxygen and heme regulatory network in yeast.

Full Description: Deciphering gene regulatory mechanisms through the analysis of high-throughput expression 
data is a challenging computational problem. Previous computational studies have used 
large expression datasets in order to resolve fine patterns of coexpression, producing 
clusters or modules of potentially coregulated genes. These methods typically examine 
promoter sequence information, such as DNA motifs or transcription factor occupancy data, 
in a separate step after clustering. We needed an alternative and more integrative 
approach to study the oxygen regulatory network in Saccharomyces cerevisiae using a small 
dataset of perturbation experiments. Mechanisms of oxygen sensing and regulation underlie 
many physiological and pathological processes, and only a handful of oxygen regulators 
have been identified in previous studies. We used a new machine learning algorithm called 
MEDUSA to uncover detailed information about the oxygen regulatory network using genome-
wide expression changes in response to perturbations in the levels of oxygen, heme, Hap1, 
and Co2+. MEDUSA integrates mRNA expression, promoter sequence, and ChIP-chip occupancy 
data to learn a model that accurately predicts the differential expression of target genes
in held-out data. We used a novel margin-based score to extract significant condition-
specific regulators and assemble a global map of the oxygen sensing and regulatory network
. This network includes both known oxygen and heme regulators, such as Hap1, Mga2, Hap4, 
and Upc2, as well as many new candidate regulators. MEDUSA also identified many DNA motifs
that are consistent with previous experimentally identified transcription factor binding 
sites. Because MEDUSA's regulatory program associates regulators to target genes through 
their promoter sequences, we directly tested the predicted regulators for OLE1, a gene 
specifically induced under hypoxia, by experimental analysis of the activity of its 
promoter. In each case, deletion of the candidate regulator resulted in the predicted 
effect on promoter activity, confirming that several novel regulators identified by MEDUSA
are indeed involved in oxygen regulation. MEDUSA can reveal important information from a 
small dataset and generate testable hypotheses for further experimental analysis. 
Supplemental data are included.

PMID: 19008939

GEO ID: GSE8343

PCL filename	short description	# conditions	tags	# channels
GSE8343_setA_family.pcl	A predictive model of the oxygen and heme regulatory network in yeast.	24	chemical stimulus|cofactor metabolism|nutrient utilization|oxygen level alteration|transcription	1

File last modified: Thu Apr 25 2024

---


# Kvitek_2008_PMID_18927628

Citation: Kvitek DJ, Will JL, Gasch AP. 2008. Variations in stress sensitivity and genomic 
expression in diverse S. cerevisiae isolates.

Full Description: Interactions between an organism and its environment can significantly influence 
phenotypic evolution. A first step toward understanding this process is to characterize 
phenotypic diversity within and between populations. We explored the phenotypic variation 
in stress sensitivity and genomic expression in a large panel of Saccharomyces strains 
collected from diverse environments. We measured the sensitivity of 52 strains to 14 
environmental conditions, compared genomic expression in 18 strains, and identified gene 
copy-number variations in six of these isolates. Our results demonstrate a large degree of
phenotypic variation in stress sensitivity and gene expression. Analysis of these datasets
reveals relationships between strains from similar niches, suggests common and unique 
features of yeast habitats, and implicates genes whose variable expression is linked to 
stress resistance. Using a simple metric to suggest cases of selection, we found that 
strains collected from oak exudates are phenotypically more similar than expected based on
their genetic diversity, while sake and vineyard isolates display more diverse phenotypes 
than expected under a neutral model. We also show that the laboratory strain S288c is 
phenotypically distinct from all of the other strains studied here, in terms of stress 
sensitivity, gene expression, Ty copy number, mitochondrial content, and gene-dosage 
control. These results highlight the value of understanding the genetic basis of 
phenotypic variation and raise caution about using laboratory strains for comparative 
genomics.

PMID: 18927628

GEO ID: GSE10269

PCL filename	short description	# conditions	tags	# channels
GSE10269_set0_family.pcl	Variations in stress sensitivity and genomic expression in diverse S. cerevisiae isolates.	9	stress	2

File last modified: Thu Apr 25 2024

---


# Lai_2008_PMID_19105839

Citation: Lai LC, Kissinger MT, Burke PV, Kwast KE. 2008. Comparison of the transcriptomic stress 
response evoked by antimycin A and oxygen deprivation in Saccharomyces cerevisiae.

Full Description: BACKGROUND: Acute changes in environmental parameters (e.g., O2, pH, UV, osmolarity, 
nutrients, etc.) evoke a common transcriptomic response in yeast referred to as the 
environmental stress response (ESR) or common environmental response (CER). Why such a 
diverse array of insults should elicit a common transcriptional response remains enigmatic
. Previous functional analyses of the networks involved have found that, in addition to up
-regulating those for mitigating the specific stressor, the majority appear to be involved
in balancing energetic supply and demand and modulating progression through the cell cycle
. Here we compared functional and regulatory aspects of the stress responses elicited by 
the acute inhibition of respiration with antimycin A and oxygen deprivation under 
catabolite non-repressed (galactose) conditions. RESULTS: Gene network analyses of the 
transcriptomic responses revealed both treatments result in the transient (10 - 60 min) 
down-regulation of MBF- and SBF-regulated networks involved in the G1/S transition of the 
cell cycle as well as Fhl1 and PAC/RRPE-associated networks involved in energetically 
costly programs of ribosomal biogenesis and protein synthesis. Simultaneously, Msn2/4 
networks involved in hexose import/dissimilation, reserve energy regulation, and autophagy
were transiently up-regulated. Interestingly, when cells were treated with antimycin A 
well before experiencing anaerobiosis these networks subsequently failed to respond to 
oxygen deprivation. These results suggest the transient stress response is elicited by the
acute inhibition of respiration and, we postulate, changes in cellular energetics and/or 
the instantaneous growth rate, not oxygen deprivation per se. After a considerable delay
(> or = 1 generation) under anoxia, predictable changes in heme-regulated gene networks (e
.g., Hap1, Hap2/3/4/5, Mot3, Rox1 and Upc2) were observed both in the presence and absence
of antimycin A. CONCLUSION: This study not only differentiates between the gene networks 
that respond to respiratory inhibition and those that respond to oxygen deprivation but 
suggests the function of the ESR or CER is to balance energetic supply/demand and 
coordinate growth with the cell cycle, whether in response to perturbations that disrupt 
catabolic pathways or those that require rapidly up-regulating energetically costly 
programs for combating specific stressors.

PMID: 19105839

GEO ID: GSE3706

PCL filename	short description	# conditions	tags	# channels
GSE3706_setA_family.pcl	Comparison of the transcriptomic stress response evoked by antimycin A and oxygen deprivation in Saccharomyces cerevisiae.	227	carbon utilization|chemical stimulus|respiration|stress	2

File last modified: Thu Apr 25 2024

---


# Lee_2008_PMID_18679056

Citation: Lee YL, Lee CK. 2008. Transcriptional response according to strength of calorie 
restriction in Saccharomyces cerevisiae.

Full Description: To characterize gene expression that is dependent on the strength of calorie restriction (
CR), we obtained transcriptome at different levels of glucose, which is a major energy and
carbon source for budding yeast. To faithfully mimic mammalian CR in yeast culture, we 
reconstituted and grew seeding yeast cells in fresh 2% YPD media before inoculating into 2
%, 1%, 0.5% and 0.25% YPD media to reflect different CR strengths. We collected and 
characterized 160 genes that responded to CR strength based on the rigorous statistical 
analyses of multiple test corrected ANOVA (adjusted p<value 0.1 or raw p value<0.0031) and
Pearson correlation (|r|>0.7). Based on the individual gene studies and the GO Term Finder
analysis of 160 genes, we found that CR dose-dependently and gradually increased 
mitochondrial function at the transcriptional level. Therefore, we suggest these 160 genes
are markers that respond to CR strength and that might be useful in elucidating CR 
mechanisms, especially how stronger CR extends life span more.

PMID: 18679056

GEO ID: GSE9217.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE9217.final.pcl	Transcriptional response according to strength of calorie restriction in Saccharomyces cerevisiae.	12	carbon utilization|cell aging	1

File last modified: Thu Apr 25 2024

---


# Levy_2008_PMID_18687885

Citation: Levy A, Eyal M, Hershkovits G, Salmon-Divon M, Klutstein M, Katcoff DJ. 2008. Yeast linker
histone Hho1p is required for efficient RNA polymerase I processivity and transcriptional 
silencing at the ribosomal DNA.

Full Description: Nucleosome core particles in eukaryotes are linked by a stretch of DNA that is usually 
associated with a linker histone. Here, we show in yeast, that the presence of yeast 
linker histone Hho1p represses expression of a pol II transcribed gene (MET15) embedded in
the rDNA. In vivo deletions of Hho1p sequences showed that the second globular domain is 
sufficient for that repression, whereas the presence of the N terminus is required for its
derepression. In contrast, a run-on assay confirmed by a ChIP experiment showed that Hho1p
is required for maximal pol I processivity during rDNA transcription.  Psoralen 
accessibility experiments indicated that Hho1p is necessary for normal rDNA compaction. 
DNA array expression analysis comparing RNA transcripts in wild-type and hho1 strains 
before and after a heat-shock showed that Hho1p is necessary to achieve wild-type mRNA 
levels of transcripts that encode ribosomal components. Taken together, our results 
suggest that Hho1p is involved in rDNA compaction, and like core histones, is required for
efficient rDNA transcription  by pol I.

PMID: 18687885

GEO ID: GSE9095.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE9095.final.pcl	Yeast linker histone Hho1p is required for efficient RNA polymerase I processivity and transcriptional silencing at the ribosomal DNA.	3	transcription|heat shock|stress	2

File last modified: Thu Apr 25 2024

---


# Marks_2008_PMID_18215224

Citation: Marks VD, Ho Sui SJ, Erasmus D, van der Merwe GK, Brumm J, Wasserman WW, Bryan J, van 
Vuuren HJ. 2008. Dynamics of the yeast transcriptome during wine fermentation reveals a 
novel fermentation stress response.

Full Description: In this study, genome-wide expression analyses were used to study the response of 
Saccharomyces cerevisiae to stress throughout a 15-day wine fermentation. Forty per cent 
of the yeast genome significantly changed expression levels to mediate long-term 
adaptation to fermenting grape must. Among the genes that changed expression levels, a 
group of 223 genes was identified, which was designated as fermentation stress response (
FSR) genes that were dramatically induced at various points during fermentation. FSR genes
sustain high levels of induction up to the final time point and exhibited changes in 
expression levels ranging from four- to 80-fold. The FSR is novel; 62% of the genes 
involved have not been implicated in global stress responses and 28% of the FSR genes have
no functional annotation. Genes involved in respiratory metabolism and gluconeogenesis 
were expressed during fermentation despite the presence of high concentrations of glucose
. Ethanol, rather than nutrient depletion, seems to be responsible for entry of yeast 
cells into the stationary phase.

PMID: 18215224

GEO ID: GSE8536

PCL filename	short description	# conditions	tags	# channels
GSE8536_setA_family.pcl	Dynamics of the yeast transcriptome during wine fermentation reveals a novel fermentation stress response.	21	fermentation|stress	1

File last modified: Thu Apr 25 2024

---


# Molina-Navarro_2008_PMID_18424442

Citation: Molina-Navarro MM, Castells-Roca L, Belli G, Garcia-Martinez J, Marin-Navarro J, Moreno J,
Perez-Ortin JE, Herrero E. 2008. Comprehensive transcriptional analysis of the oxidative 
response in yeast.

Full Description: The oxidative stress response in Saccharomyces cerevisiae has been analyzed by parallel 
determination of mRNA levels and transcription rates for the entire genome. A mathematical
algorithm has been adapted for a dynamic situation such as the response to stress, to 
calculate theoretical mRNA decay rates from the experimental data. Yeast genes have been 
grouped into 25 clusters according to mRNA level and transcription rate kinetics, and 
average mRNA decay rates have been calculated for each cluster. In most of the genes, 
changes in one or both experimentally determined parameters occur during the stress 
response. 24% of the genes are transcriptionally induced without an increase in mRNA 
levels. The lack of parallelism between the evolution of the mRNA amount and transcription
rate predicts changes in mRNA stability during stress. Genes for ribosomal proteins and 
rRNA processing enzymes are abundant among those whose mRNAs are predicted to destabilize
. The number of genes whose mRNAs are predicted to stabilize is lower, although some 
protein folding or proteasomal genes are among the latter. We have confirmed the 
mathematical predictions for several genes pertaining to different clusters by 
experimentally determining mRNA decay rates using the regulatable tetO promoter in 
transcriptional expression conditions not affected by the oxidative stress. This study 
indicates that the oxidative stress response in yeast cells is not only conditioned by 
gene transcription but also by the mRNA decay dynamics and that this complex response may 
be particularly relevant to explain the temporary down-regulation of protein synthesis 
occurring during stress.

PMID: 18424442

GEO ID: GSE9663

PCL filename	short description	# conditions	tags	# channels
GSE9663_set0_family.pcl	Oxidative stress response in t-butyl hydroperoxide treated cultures; transcription rate analysis by genomic run-on methodology, 3 replicates	18	oxidative stress|stress	1

File last modified: Thu Apr 25 2024
GSE9645_set0_family.pcl	Oxidative stress response in t-butyl hydroperoxide treated cultures; total mRNA amount, 3 replicates	18	oxidative stress|stress	1

File last modified: Thu Apr 25 2024

---


# Nag_2008_PMID_18508805

Citation: Nag R, Gong F, Fahy D, Smerdon MJ. 2008. A single amino acid change in histone H4 enhances
UV survival and DNA repair in yeast.

Full Description: Single amino acid changes at specific DNA contacts of histones H3 and H4 generate SWI/SNF-
independent (Sin) mutants in yeast. We have analyzed the effect of the Sin mutation at R45
of histone H4 on cell survival following UV irradiation, nucleotide excision repair (NER) 
and chromatin structure. We find that this mutation renders yeast cells more resistant to 
UV damage and enhances NER at specific chromatin loci. In the transcriptionally silent HML
, repressed GAL10 and the constitutively active RPB2 loci, H4 R45 mutants exhibit enhanced
repair of UV-induced cyclobutane pyrimidine dimers (CPDs) compared to wild-type (wt). 
However, the H4 R45 mutation does not increase the transcription of NER genes, disrupt 
transcriptional silencing of the HML locus or alter repression in the GAL10 locus. We have
further shown that the H4 R45C mutation increases the accessibility of nucleosome DNA in 
chromatin to exogenous nucleases and may expedite nucleosome rearrangements during NER. 
Taken together, our results indicate that the increased repair observed in Sin mutants is 
a direct effect of the altered chromatin landscape caused by the mutation, suggesting that
such subtle changes in the conserved histone residues can influence the accessibility of 
DNA repair factors in chromatin.

PMID: 18508805

GEO ID: GSE11282

PCL filename	short description	# conditions	tags	# channels
GSE11282_setA_family.pcl	A single amino acid change in histone H4 enhances UV survival and DNA repair in yeast.	6	chromatin organization	1

File last modified: Thu Apr 25 2024

---


# Orlando_2008_PMID_18463633

Citation: Orlando DA, Lin CY, Bernard A, Wang JY, Socolar JE, Iversen ES, Hartemink AJ, Haase SB. 
2008. Global control of cell-cycle transcription by coupled CDK and network oscillators.

Full Description: A significant fraction of the Saccharomyces cerevisiae genome is transcribed periodically 
during the cell division cycle, indicating that properly timed gene expression is 
important for regulating cell-cycle events. Genomic analyses of the localization and 
expression dynamics of transcription factors suggest that a network of sequentially 
expressed transcription factors could control the temporal programme of transcription 
during the cell cycle. However, directed studies interrogating small numbers of genes 
indicate that their periodic transcription is governed by the activity of cyclin-dependent
kinases (CDKs). To determine the extent to which the global cell-cycle transcription 
programme is controlled by cyclin-CDK complexes, we examined genome-wide transcription 
dynamics in budding yeast mutant cells that do not express S-phase and mitotic cyclins. 
Here we show that a significant fraction of periodic genes are aberrantly expressed in the
cyclin mutant. Although cells lacking cyclins are blocked at the G1/S border, nearly 70% 
of periodic genes continued to be expressed periodically and on schedule. Our findings 
reveal that although CDKs have a function in the regulation of cell-cycle transcription, 
they are not solely responsible for establishing the global periodic transcription 
programme. We propose that periodic transcription is an emergent property of a 
transcription factor network that can function as a cell-cycle oscillator independently of
, and in tandem with, the CDK oscillator.

PMID: 18463633

GEO ID: GSE8799

PCL filename	short description	# conditions	tags	# channels
GSE8799_setA_family.pcl	Global control of cell-cycle transcription by coupled CDK and network oscillators.	60	cell cycle regulation	1

File last modified: Thu Apr 25 2024

---


# Pan_2008_PMID_18366703

Citation: Pan Z, Agarwal AK, Xu T, Feng Q, Baerson SR, Duke SO, Rimando AM. 2008. Identification of 
molecular pathways affected by pterostilbene, a natural dimethylether analog of 
resveratrol.

Full Description: BACKGROUND: Pterostilbene, a naturally occurring phenolic compound produced by 
agronomically important plant genera such as Vitis and Vacciunium, is a phytoalexin 
exhibiting potent antifungal activity. Additionally, recent studies have demonstrated 
several important pharmacological properties associated with pterostilbene. Despite this, 
a systematic study of the effects of pterostilbene on eukaryotic cells at the molecular 
level has not been previously reported. Thus, the aim of the present study was to identify
the cellular pathways affected by pterostilbene by performing transcript profiling studies
, employing the model yeast Saccharomyces cerevisiae. METHODS: S. cerevisiae strain S288C 
was exposed to pterostilbene at the IC50 concentration (70 muM) for one generation (3 h). 
Transcript profiling experiments were performed on three biological replicate samples 
using the Affymetrix GeneChip Yeast Genome S98 Array. The data were analyzed using the 
statistical methods available in the GeneSifter microarray data analysis system. To 
validate the results, eleven differentially expressed genes were further examined by 
quantitative real-time RT-PCR, and S. cerevisiae mutant strains with deletions in these 
genes were analyzed for altered sensitivity to pterostilbene. RESULTS: Transcript 
profiling studies revealed that pterostilbene exposure significantly down-regulated the 
expression of genes involved in methionine metabolism, while the expression of genes 
involved in mitochondrial functions, drug detoxification, and transcription factor 
activity were significantly up-regulated. Additional analyses revealed that a large number
of genes involved in lipid metabolism were also affected by pterostilbene treatment. 
CONCLUSION: Using transcript profiling, we have identified the cellular pathways targeted 
by pterostilbene, an analog of resveratrol. The observed response in lipid metabolism 
genes is consistent with its known hypolipidemic properties, and the induction of 
mitochondrial genes is consistent with its demonstrated role in apoptosis in human cancer 
cell lines. Furthermore, our data show that pterostilbene has a significant effect on 
methionine metabolism, a previously unreported effect for this compound.

PMID: 18366703

GEO ID: GSE10554

PCL filename	short description	# conditions	tags	# channels
GSE10554_setA_family.pcl	Identification of molecular pathways affected by pterostilbene, a natural dimethylether analog of resveratrol.	6	chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Pelechano_2008_PMID_17914747

Citation: Pelechano V, Perez-Ortin JE. 2008. The transcriptional inhibitor thiolutin blocks mRNA 
degradation in yeast.

Full Description: Thiolutin is commonly used as a general inhibitor of transcription in yeast. It has been 
used to calculate mRNA decay rates by stopping the transcription and then determining the 
relative abundance of individual mRNAs at different times after inhibition. We report here
that thiolutin is also an inhibitor of mRNA degradation, and thus its use can lead to 
miscalculations of mRNA half-lives. The inhibition of mRNA decay seems to affect the mRNA 
degradation pathway without impeding poly(A) shortening, given that the decay rate of 
total poly(A) amount is not reduced by thiolutin. Moreover, the thiolutin-dependent 
inhibition of mRNA degradation has variable effects on different functional groups of 
genes, suggesting that they use various degradation pathways for their mRNAs.

PMID: 17914747

GEO ID: GSE8629

PCL filename	short description	# conditions	tags	# channels
GSE8629_setA_family.pcl	The transcriptional inhibitor thiolutin blocks mRNA degradation in yeast.	6	chemical stimulus|RNA catabolism	1

File last modified: Thu Apr 25 2024

---


# Protchenko_2008_PMID_18326586

Citation: Protchenko O, Shakoury-Elizeh M, Keane P, Storey J, Androphy R, Philpott CC. 2008. Role of
PUG1 in inducible porphyrin and heme transport in Saccharomyces cerevisiae.

Full Description: Unlike pathogenic fungi, the budding yeast Saccharomyces cerevisiae is not efficient at 
using heme as a nutritional source of iron. Here we report that for this yeast, heme 
uptake is induced under conditions of heme starvation. Heme synthesis requires oxygen, and
yeast grown anaerobically exhibited an increased uptake of hemin. Similarly, a strain 
lacking aminolevulinate synthase exhibited a sixfold increase in hemin uptake when grown 
without 2-aminolevulinic acid. We used microarray analysis of cells grown under reduced 
oxygen tension or reduced intracellular heme conditions to identify candidate genes 
involved in heme uptake. Surprisingly, overexpression of PUG1 (protoporphyrin uptake gene 
1) resulted in reduced utilization of exogenous heme by a heme-deficient strain and, 
conversely, increased the utilization of protoporphyrin IX. Pug1p was localized to the 
plasma membrane by indirect immunofluorescence and subcellular fractionation. Strains 
overexpressing PUG1 exhibited decreased accumulation of [(55)Fe]hemin but increased 
accumulation of protoporphyrin IX compared to the wild-type strain. To measure the effect 
of PUG1 overexpression on intracellular heme pools, we used a CYC1-lacZ reporter, which is
activated in the presence of heme, and we monitored the activity of a heme-containing 
metalloreductase, Fre1p, expressed from a constitutive promoter. The data from these 
experiments were consistent with a role for Pug1p in inducible protoporphyrin IX influx 
and heme efflux.

PMID: 18326586

GEO ID: GSE9514

PCL filename	short description	# conditions	tags	# channels
GSE9514_setA_family.pcl	Role of PUG1 in inducible porphyrin and heme transport in Saccharomyces cerevisiae.	8	cofactor metabolism|oxygen level alteration|starvation	1

File last modified: Thu Apr 25 2024

---


# Puig_2008_PMID_18522836

Citation: Puig S, Vergara SV, Thiele DJ. 2008. Cooperation of two mRNA-binding proteins drives 
metabolic adaptation to iron deficiency.

Full Description: Iron (Fe) is an essential cofactor for a wide range of cellular processes. We have 
previously demonstrated in yeast that Cth2 is expressed during Fe deficiency and promotes 
degradation of a battery of mRNAs leading to reprogramming of Fe-dependent metabolism and 
Fe storage. We report here that the Cth2-homologous protein Cth1 is transiently expressed 
during Fe deprivation and participates in the response to Fe deficiency through the 
degradation of mRNAs primarily involved in mitochondrially localized activities including 
respiration and amino acid biosynthesis. In parallel, wild-type cells, but not 
cth1Deltacth2Delta cells, accumulate mRNAs encoding proteins that function in glucose 
import and storage and store high levels of glycogen. In addition, Fe deficiency leads to 
phosphorylation of Snf1, an AMP-activated protein kinase family member required for the 
cellular response to glucose starvation. These studies demonstrate a metabolic 
reprogramming as a consequence of Fe starvation that is dependent on the coordinated 
activities of two mRNA-binding proteins.

PMID: 18522836

GEO ID: GSE11236

PCL filename	short description	# conditions	tags	# channels
GSE11236_setA_family.pcl	Cooperation of two mRNA-binding proteins drives metabolic adaptation to iron deficiency.	9	cellular ion homeostasis|metal or metalloid ion stress|RNA catabolism|starvation	1

File last modified: Thu Apr 25 2024

---


# Ro_2008_PMID_18983675

Citation: Ro DK, Ouellet M, Paradise EM, Burd H, Eng D, Paddon CJ, Newman JD, Keasling JD. 2008. 
Induction of multiple pleiotropic drug resistance genes in yeast engineered to produce an 
increased level of anti-malarial drug precursor, artemisinic acid.

Full Description: BACKGROUND: Due to the global occurrence of multi-drug-resistant malarial parasites (
Plasmodium falciparum), the anti-malarial drug most effective against malaria is 
artemisinin, a natural product (sesquiterpene lactone endoperoxide) extracted from sweet 
wormwood (Artemisia annua). However, artemisinin is in short supply and unaffordable to 
most malaria patients. Artemisinin can be semi-synthesized from its precursor artemisinic 
acid, which can be synthesized from simple sugars using microorganisms genetically 
engineered with genes from A. annua. In order to develop an industrially competent yeast 
strain, detailed analyses of microbial physiology and development of gene expression 
strategies are required. RESULTS: Three plant genes coding for amorphadiene synthase, 
amorphadiene oxidase (AMO or CYP71AV1), and cytochrome P450 reductase, which in concert 
divert carbon flux from farnesyl diphosphate to artemisinic acid, were expressed from a 
single plasmid. The artemisinic acid production in the engineered yeast reached 250 microg
mL(-1) in shake-flask cultures and 1 g L(-1) in bio-reactors with the use of Leu2d 
selection marker and appropriate medium formulation. When plasmid stability was measured, 
the yeast strain synthesizing amorphadiene alone maintained the plasmid in 84% of the 
cells, whereas the yeast strain synthesizing artemisinic acid showed poor plasmid 
stability. Inactivation of AMO by a point-mutation restored the high plasmid stability, 
indicating that the low plasmid stability is not caused by production of the AMO protein 
but by artemisinic acid synthesis or accumulation. Semi-quantitative reverse-transcriptase
(RT)-PCR and quantitative real time-PCR consistently showed that pleiotropic drug 
resistance (PDR) genes, belonging to the family of ATP-Binding Cassette (ABC) transporter
, were massively induced in the yeast strain producing artemisinic acid, relative to the 
yeast strain producing the hydrocarbon amorphadiene alone. Global transcriptional analysis
by yeast microarray further demonstrated that the induction of drug-resistant genes such 
as ABC transporters and major facilitator superfamily (MSF) genes is the primary cellular 
stress-response; in addition, oxidative and osmotic stress responses were observed in the 
engineered yeast. CONCLUSION: The data presented here suggest that the engineered yeast 
producing artemisinic acid suffers oxidative and drug-associated stresses. The use of 
plant-derived transporters and optimizing AMO activity may improve the yield of 
artemisinic acid production in the engineered yeast.

PMID: 18983675

GEO ID: GSE11620

PCL filename	short description	# conditions	tags	# channels
GSE11620_setA_family.pcl	Induction of multiple pleiotropic drug resistance genes in yeast engineered to produce an increased level of anti-malarial drug precursor, artemisinic acid.	14	chemical stimulus	2

File last modified: Thu Apr 25 2024

---


# Rojas_2008_PMID_18667070

Citation: Rojas M, Casado M, Portugal J, Pina B. 2008. Selective inhibition of yeast regulons by 
daunorubicin: a transcriptome-wide analysis.

Full Description: BACKGROUND: The antitumor drug daunorubicin exerts some of its cytotoxic effects  by 
binding to DNA and inhibiting the transcription of different genes. We analysed this 
effect in vivo at the transcriptome level using the budding yeast Saccharomyces cerevisiae
as a model and sublethal (IC40) concentrations of the drug to minimise general toxic 
effects. RESULTS: Daunorubicin affected a minor proportion (14%) of the yeast 
transcriptome, increasing the expression of 195 genes and reducing expression of 280 genes
. Daunorubicin down-regulated genes included essentially all genes involved in the 
glycolytic pathway, the tricarboxylic acid cycle and alcohol metabolism, whereas 
transcription of ribosomal protein genes was not affected or even slightly increased. This
pattern is consistent with a specific inhibition of glucose usage in treated cells, with  
only minor effects on proliferation or other basic cell functions. Analysis of promoters 
of down-regulated genes showed that they belong to a limited number of  transcriptional 
regulatory units (regulons). Consistently, data mining showed that daunorubicin-induced 
changes in expression patterns were similar to those observed in yeast strains deleted for
some transcription factors functionally related to the glycolysis and/or the cAMP 
regulatory pathway, which appeared to be particularly sensitive to daunorubicin. 
CONCLUSION: The effects of daunorubicin treatment on the yeast transcriptome are 
consistent with a model in  which this drug impairs binding of different transcription 
factors by competing for their DNA binding sequences, therefore limiting their 
effectiveness and affecting the corresponding regulatory networks. This proposed mechanism
might have broad therapeutic implications against cancer cells growing under hypoxic 
conditions.

PMID: 18667070

GEO ID: GSE19351.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE19351.final.pcl	Selective inhibition of yeast regulons by daunorubicin: a transcriptome-wide analysis.	9	chemical stimulus|stress	1

File last modified: Thu Apr 25 2024

---


# Salusjarvi_2008_PMID_18533012

Citation: Salusjarvi L, Kankainen M, Soliymani R, Pitkanen JP, Penttila M, Ruohonen L. 2008. 
Regulation of xylose metabolism in recombinant Saccharomyces cerevisiae.

Full Description: ABSTRACT: BACKGROUND: Considerable interest in the bioconversion of lignocellulosic 
biomass into ethanol has led to metabolic engineering of Saccharomyces cerevisiae for 
fermentation of xylose. In the present study, the transcriptome and proteome of 
recombinant, xylose-utilising S. cerevisiae grown in aerobic batch cultures on xylose were
compared with those of glucose-grown cells both in glucose repressed and derepressed 
states. The aim was to study at the genome-wide level how signalling and carbon catabolite
repression differ in cells grown on either glucose or xylose. The more detailed knowledge 
whether xylose is sensed as a fermentable carbon source, capable of catabolite repression 
like glucose, or is rather recognised as a non-fermentable carbon source is important for 
further engineering this yeast for more efficient anaerobic fermentation of xylose. 
RESULTS: Genes encoding respiratory proteins, proteins of the tricarboxylic acid and 
glyoxylate cycles, and gluconeogenesis were only partially repressed by xylose, similar to
the genes encoding their transcriptional regulators HAP4, CAT8 and SIP1-2 and 4. Several 
genes that are repressed via the Snf1p/Mig1p-pathway during growth on glucose had higher 
expression in the cells grown on xylose than in the glucose repressed cells but lower than
in the glucose derepressed cells. The observed expression profiles of the transcription 
repressor RGT1 and its target genes HXT2-3, encoding hexose transporters suggested that 
extracellular xylose was sensed by the glucose sensors Rgt2p and Snf3p. Proteome analyses 
revealed distinct patterns in phosphorylation of hexokinase 2, glucokinase and enolase 
isoenzymes in the xylose- and glucose-grown cells. CONCLUSION: The results indicate that 
the metabolism of yeast growing on xylose corresponds neither to that of fully glucose 
repressed cells nor that of derepressed cells. This may be one of the major reasons for 
the suboptimal fermentation of xylose by recombinant S. cerevisiae strains. 
Phosphorylation of different isoforms of glycolytic enzymes suggests that regulation of 
glycolysis also occurred at a post-translational level, supporting prior findings.

PMID: 18533012

GEO ID: GSE12890

PCL filename	short description	# conditions	tags	# channels
GSE12890_setA_family.pcl	Regulation of xylose metabolism in recombinant Saccharomyces cerevisiae.	15	carbon utilization	1

File last modified: Thu Apr 25 2024

---


# Shalem_2008_PMID_18854817

Citation: Shalem O, Dahan O, Levo M, Martinez MR, Furman I, Segal E, Pilpel Y. 2008. Transient 
transcriptional responses to stress are generated by opposing effects of mRNA production 
and degradation.

Full Description: The state of the transcriptome reflects a balance between mRNA production and degradation
. Yet how these two regulatory arms interact in shaping the kinetics of the transcriptome 
in response to environmental changes is not known. We subjected yeast to two stresses, one
that induces a fast and transient response, and another that triggers a slow enduring 
response. We then used microarrays following transcriptional arrest to measure genome-wide
decay profiles under each condition. We found condition-specific changes in mRNA decay 
rates and coordination between mRNA production and degradation. In the transient response
, most induced genes were surprisingly destabilized, whereas repressed genes were somewhat
stabilized, exhibiting counteraction between production and degradation. This strategy can
reconcile high steady-state level with short response time among induced genes. In 
contrast, the stress that induces the slow response displays the more expected behavior, 
whereby most induced genes are stabilized, and repressed genes are destabilized. Our 
results show genome-wide interplay between mRNA production and degradation, and that 
alternative modes of such interplay determine the kinetics of the transcriptome in 
response to stress.

PMID: 18854817

GEO ID: GSE12222

PCL filename	short description	# conditions	tags	# channels
GSE12222_set5_family.pcl	Genome-wide reference decay profile after RNAPII inactivation (rpb1-1); 30 min time course	7	RNA catabolism	1

File last modified: Thu Apr 25 2024
GSE12222_set4_family.pcl	Genome-wide reference decay profile after RNAPII inactivation (rpb1-1); 1 hr time course	10	RNA catabolism	1

File last modified: Thu Apr 25 2024
GSE12222_set3_family.pcl	Genome-wide decay profile after DNA damage stress and subsequent RNAPII (rpb1-1) inactivation; 1 hr time course	10	DNA damage stimulus|RNA catabolism|stress	1

File last modified: Thu Apr 25 2024
GSE12222_set1_family.pcl	Response to DNA damage stress induced by MMS; 3 hr time course	6	DNA damage stimulus|stress	1

File last modified: Thu Apr 25 2024
GSE12222_set2_family.pcl	Genome-wide decay profile after oxidative stress and subsequent RNAPII (rpb1-1) inactivation; 1 hr time course	10	oxidative stress|RNA catabolism|stress	1

File last modified: Thu Apr 25 2024
GSE12222_set0_family.pcl	Response to oxidative stress induced by H2O2; 3 hr time course	6	oxidative stress|stress	1

File last modified: Thu Apr 25 2024

---


# Shirra_2008_PMID_18955495

Citation: Shirra MK, McCartney RR, Zhang C, Shokat KM, Schmidt MC, Arndt KM. 2008. A chemical 
genomics study identifies Snf1 as a repressor of GCN4 translation.

Full Description: The Saccharomyces cerevisiae Snf1 kinase plays a critical role in recalibrating cellular 
metabolism in response to glucose depletion. Hundreds of genes show changes in expression 
levels when the SNF1 gene is deleted. However, cells can adapt to the absence of a 
specific gene when grown in long term culture. Here we apply a chemical genetic method to 
rapidly and selectively inactivate a modified Snf1 kinase using a pyrazolopyrimidine 
inhibitor. By allowing cells to adjust to a change in carbon source prior to inhibition of
the Snf1 kinase activity, we identified a set of genes whose expression increased when 
Snf1 was inhibited. Prominent in this set are genes that are activated by Gcn4, a 
transcriptional activator of amino acid biosynthetic genes. Deletion of Snf1 increased 
Gcn4 protein levels without affecting its mRNA levels. The increased Gcn4 protein levels 
required the Gcn2 kinase and Gcn20, regulators of GCN4 translation. These data indicate 
that Snf1 functions upstream of Gcn20 to regulate control of GCN4 translation in S. 
cerevisiae.

PMID: 18955495

GEO ID: GSE12061

PCL filename	short description	# conditions	tags	# channels
GSE12061_setA_family.pcl	A chemical genomics study identifies Snf1 as a repressor of GCN4 translation.	4	protein phosphorylation|translational regulation	1

File last modified: Thu Apr 25 2024

---


# Shivaswamy_2008_PMID_18212068

Citation: Shivaswamy S, Iyer VR. 2008. Stress-dependent dynamics of global chromatin remodeling in 
yeast: dual role for SWI/SNF in the heat shock stress response.

Full Description: Although chromatin structure is known to affect transcriptional activity, it is not clear 
how broadly patterns of changes in histone modifications and nucleosome occupancy affect 
the dynamic regulation of transcription in response to perturbations. The identity and 
role of chromatin remodelers that mediate some of these changes are also unclear. Here, we
performed temporal genome-wide analyses of gene expression, nucleosome occupancy, and 
histone H4 acetylation during the response of yeast (Saccharomyces cerevisiae) to 
different stresses and report several findings. First, a large class of predominantly 
ribosomal protein genes, whose transcription was repressed during both heat shock and 
stationary phase, showed strikingly contrasting histone acetylation patterns. Second, the 
SWI/SNF complex was required for normal activation as well as repression of genes during 
heat shock, and loss of SWI/SNF delayed chromatin remodeling at the promoters of activated
genes. Third, Snf2 was recruited to ribosomal protein genes and Hsf1 target genes, and its
occupancy of this large set of genes was altered during heat shock. Our results suggest a 
broad and direct dual role for SWI/SNF in chromatin remodeling, during heat shock 
activation as well as repression, at promoters and coding regions.

PMID: 18212068

GEO ID: GSE7665

PCL filename	short description	# conditions	tags	# channels
GSE7665_set0_family.pcl	Stress-dependent dynamics of global chromatin remodeling in yeast: dual role for SWI/SNF in the heat shock stress response.	19	chromatin organization|heat shock|stationary phase entry|stress	2

File last modified: Thu Apr 25 2024

---


# Smith_2008_PMID_18416601

Citation: Smith EN, Kruglyak L. 2008. Gene-environment interaction in yeast gene expression.

Full Description: The effects of genetic variants on phenotypic traits often depend on environmental and 
physiological conditions, but such gene-environment interactions are poorly understood. 
Recently developed approaches that treat transcript abundances of thousands of genes as 
quantitative traits offer the opportunity to broadly characterize the architecture of gene
-environment interactions. We examined the genetic and molecular basis of variation in 
gene expression between two yeast strains (BY and RM) grown in two different conditions (
glucose and ethanol as carbon sources). We observed that most transcripts vary by strain 
and condition, with 2,996, 3,448, and 2,037 transcripts showing significant strain, 
condition, and strain-condition interaction effects, respectively. We expression profiled 
over 100 segregants derived from a cross between BY and RM in both growth conditions, and 
identified 1,555 linkages for 1,382 transcripts that show significant gene-environment 
interaction. At the locus level, local linkages, which usually correspond to polymorphisms
in cis-regulatory elements, tend to be more stable across conditions, such that they are 
more likely to show the same effect or the same direction of effect across conditions. 
Distant linkages, which usually correspond to polymorphisms influencing trans-acting 
factors, are more condition-dependent, and often show effects in different directions in 
the two conditions. We characterized a locus that influences expression of many growth-
related transcripts, and showed that the majority of the variation is explained by 
polymorphism in the gene IRA2. The RM allele of IRA2 appears to inhibit Ras/PKA signaling 
more strongly than the BY allele, and has undergone a change in selective pressure. Our 
results provide a broad overview of the genetic architecture of gene-environment 
interactions, as well as a detailed molecular example, and lead to key insights into how 
the effects of different classes of regulatory variants are modulated by the environment. 
These observations will guide the design of studies aimed at understanding the genetic 
basis of complex traits.

PMID: 18416601

GEO ID: GSE9376

PCL filename	short description	# conditions	tags	# channels
GSE9376_setA_family.pcl	Gene-environment interaction in yeast gene expression.	246	carbon utilization|evolution	2

File last modified: Thu Apr 25 2024

---


# Tirosh_2008_PMID_18197176

Citation: Tirosh I, Weinberger A, Bezalel D, Kaganovich M, Barkai N. 2008.  On the relation between 
promoter divergence and gene expression evolution.

Full Description: Recent studies have characterized significant differences in the cis-regulatory sequences 
of related organisms, but the impact of these differences on gene expression remains 
largely unexplored. Here, we show that most previously identified differences in 
transcription factor (TF)-binding sequences of yeasts and mammals have no detectable 
effect on gene expression, suggesting that compensatory mechanisms allow promoters to 
rapidly evolve while maintaining a stabilized expression pattern. To examine the impact of
changes in cis-regulatory elements in a more controlled setting, we compared the genes 
induced during mating of three yeast species. This response is governed by a single TF (
STE12), and variations in its predicted binding sites can indeed account for about half of
the observed expression differences. The remaining unexplained differences are correlated 
with the increased divergence of the sequences that flank the binding sites and an 
apparent modulation of chromatin structure. Our analysis emphasizes the flexibility of 
promoter structure, and highlights the interplay between specific binding sites and 
general chromatin structure in the control of gene expression.

PMID: 18197176

GEO ID: GSE7525

PCL filename	short description	# conditions	tags	# channels
GSE7525_set0_family.pcl	On the relation between promoter divergence and gene expression evolution.	6	evolution|mating	2

File last modified: Thu Apr 25 2024

---


# Trott_2008_PMID_18199679

Citation: Trott A, West JD, Klaic L, Westerheide SD, Silverman RB, Morimoto RI, Morano KA. 2008. 
Activation of heat shock and antioxidant responses by the natural product celastrol: 
transcriptional signatures of a thiol-targeted molecule.

Full Description: Stress response pathways allow cells to sense and respond to environmental changes and 
adverse pathophysiological states. Pharmacological modulation of cellular stress pathways 
has implications in the treatment of human diseases, including neurodegenerative disorders
, cardiovascular disease, and cancer. The quinone methide triterpene celastrol, derived 
from a traditional Chinese medicinal herb, has numerous pharmacological properties, and it
is a potent activator of the mammalian heat shock transcription factor HSF1. However, its 
mode of action and spectrum of cellular targets are poorly understood. We show here that 
celastrol activates Hsf1 in Saccharomyces cerevisiae at a similar effective concentration 
seen in mammalian cells. Transcriptional profiling revealed that celastrol treatment 
induces a battery of oxidant defense genes in addition to heat shock genes. Celastrol 
activated the yeast Yap1 oxidant defense  transcription factor via the carboxy-terminal 
redox center that responds to electrophilic compounds. Antioxidant response genes were 
likewise induced in mammalian cells, demonstrating that the activation of two major cell 
stress pathways by celastrol is conserved. We report that celastrol's biological effects, 
including inhibition of glucocorticoid receptor activity, can be blocked by the addition 
of excess free thiol, suggesting a chemical mechanism for biological activity based on 
modification of key reactive thiols by this natural  product.

PMID: 18199679

GEO ID: GSE5608.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE5608.final.pcl	Activation of heat shock and antioxidant responses by the natural product celastrol: transcriptional signatures of a thiol-targeted molecule.	4	heat shock|oxidative stress|stress	2

File last modified: Thu Apr 25 2024

---


# Willis_2008_PMID_18604275

Citation: Willis IM, Chua G, Tong AH, Brost RL, Hughes TR, Boone C, Moir RD. 2008. Genetic 
interactions of MAF1 identify a role for Med20 in transcriptional repression of ribosomal 
protein genes.

Full Description: Transcriptional repression of ribosomal components and tRNAs is coordinately regulated in 
response to a wide variety of environmental stresses. Part of this response involves the 
convergence of different nutritional and stress signaling pathways on Maf1, a protein that
is essential for repressing transcription by RNA polymerase (pol) III in Saccharomyces 
cerevisiae. Here we identify the functions buffering yeast cells that are unable to down-
regulate transcription by RNA pol III. MAF1 genetic interactions identified in screens of 
non-essential gene-deletions and conditionally expressed essential genes reveal a highly 
interconnected network of 64 genes involved in ribosome biogenesis, RNA pol II 
transcription, tRNA modification, ubiquitin-dependent proteolysis and other processes. A 
survey of non-essential MAF1 synthetic sick/lethal (SSL) genes identified six gene-
deletions that are defective in transcriptional repression of ribosomal protein (RP) genes
following rapamycin treatment. This subset of MAF1 SSL genes included MED20 which encodes 
a head module subunit of the RNA pol II Mediator complex. Genetic interactions between 
MAF1 and subunits in each structural module of Mediator were investigated to examine the 
functional relationship between these transcriptional regulators. Gene expression 
profiling identified a prominent and highly selective role for Med20 in the repression of 
RP gene transcription under multiple conditions. In addition, attenuated repression of RP 
genes by rapamycin was observed in a strain deleted for the Mediator tail module subunit 
Med16. The data suggest that Mediator and Maf1 function in parallel pathways to negatively
regulate RP mRNA and tRNA synthesis.

PMID: 18604275

GEO ID: GSE11397

PCL filename	short description	# conditions	tags	# channels
GSE11397_setA_family.pcl	Genetic interactions of MAF1 identify a role for Med20 in transcriptional repression of ribosomal protein genes.	24	carbon utilization|diauxic shift|fermentation|heat shock|oxidative stress|respiration|response to unfolded protein|transcription	2

File last modified: Thu Apr 25 2024

---


# Wu_2008_PMID_18673560

Citation: Wu CY, Bird AJ, Chung LM, Newton MA, Winge DR, Eide DJ. 2008. Differential control of Zap1
-regulated genes in response to zinc deficiency in Saccharomyces cerevisiae.

Full Description: BACKGROUND: The Zap1 transcription factor is a central player in the response of yeast to 
changes in zinc status. We previously used transcriptome profiling with DNA microarrays to
identify 46 potential Zap1 target genes in the yeast genome. In this new study, we used 
complementary methods to identify additional Zap1 target genes. RESULTS: With alternative 
growth conditions for the microarray experiments and a more sensitive motif identification
algorithm, we identified 31 new potential targets of Zap1 activation. Moreover, an 
analysis of the response of Zap1 target genes to a range of zinc concentrations and to 
zinc withdrawal over time demonstrated that these genes respond differently to zinc 
deficiency. Some genes are induced under mild zinc deficiency and act as a first line of 
defense against this stress. First-line defense genes serve to maintain zinc homeostasis 
by increasing zinc uptake, and by mobilizing and conserving intracellular zinc pools. 
Other genes respond only to severe zinc limitation and act as a second line of defense. 
These second-line defense genes allow cells to adapt to conditions of zinc deficiency and 
include genes involved in maintaining secretory pathway and cell wall function, and stress
responses. CONCLUSION: We have identified several new targets of Zap1-mediated regulation
. Furthermore, our results indicate that through the differential regulation of its target
genes, Zap1 prioritizes mechanisms of zinc homeostasis and adaptive responses to zinc 
deficiency.

PMID: 18673560

GEO ID: GSE11983

PCL filename	short description	# conditions	tags	# channels
GSE11983_setA_family.pcl	Differential control of Zap1-regulated genes in response to zinc deficiency in Saccharomyces cerevisiae.	4	cellular ion homeostasis|metal or metalloid ion stress|starvation	2

File last modified: Thu Apr 25 2024

---


# Yiu_2008_PMID_18245757

Citation: Yiu G, McCord A, Wise A, Jindal R, Hardee J, Kuo A, Shimogawa MY, Cahoon L, Wu M, Kloke J,
Hardin J, Mays Hoopes LL. 2008. Pathways change in expression during replicative aging in 
Saccharomyces cerevisiae.

Full Description: Yeast replicative aging is a process resembling replicative aging in mammalian cells. 
During aging, wild-type haploid yeast cells enlarge, become sterile, and undergo nucleolar
enlargement and fragmentation; we sought gene expression changes during the time of these 
phenotypic changes. Gene expression studied via microarrays and quantitative real-time 
reverse-transcription polymerase chain reaction (qPCR) has shown reproducible, 
statistically significant changes in messenger RNA (mRNA) of genes at 12 and 18-20 
generations. Our findings support previously described changes towards aerobic metabolism
, decreased ribosome gene expression, and a partial environmental stress response. Our 
findings include a pseudostationary phase, downregulation of methylation-related 
metabolism, increased nucleotide excision repair-related mRNA, and a strong upregulation 
of many of the regulatory subunits of protein phosphatase I (Glc7). These findings are 
correlated with aging changes in higher organisms as well as with the known involvement of
protein phosphorylation states during yeast aging.

PMID: 18245757

GEO ID: GSE10018

PCL filename	short description	# conditions	tags	# channels
GSE10018_set3_family.pcl	Expression after 8 generations during replicative aging; 6 replicates	6	cell aging	2

File last modified: Thu Apr 25 2024
GSE10018_set2_family.pcl	Expression after 18-20 generations during replicative aging; 7 replicates	7	cell aging	2

File last modified: Thu Apr 25 2024
GSE10018_set1_family.pcl	Expression after 12 generations during replicative aging; 7 replicates	7	cell aging	2

File last modified: Thu Apr 25 2024
GSE10018_set0_family.pcl	Expression after one generation during replicative aging; 7 replicates	7	cell aging	2

File last modified: Thu Apr 25 2024

---


# Zhu_2008_PMID_18552845

Citation: Zhu J, Zhang B, Smith EN, Drees B, Brem RB, Kruglyak L, Bumgarner RE, Schadt EE. 2008. 
Integrating large-scale functional genomic data to dissect the complexity of yeast 
regulatory networks.

Full Description: A key goal of biology is to construct networks that predict complex system behavior. We 
combine multiple types of molecular data, including genotypic, expression, transcription 
factor binding site (TFBS), and protein-protein interaction (PPI) data previously 
generated from a number of yeast experiments, in order to reconstruct causal gene networks
. Networks based on different types of data are compared using metrics devised to assess 
the predictive power of a network. We show that a network reconstructed by integrating 
genotypic, TFBS and PPI data is the most predictive. This network is used to predict 
causal regulators responsible for hot spots of gene expression activity in a segregating 
yeast population. We also show that the network can elucidate the mechanisms by which 
causal regulators give rise to larger-scale changes in gene expression activity. We then 
prospectively validate predictions, providing direct experimental evidence that predictive
networks can be constructed by integrating multiple, appropriate data types.

PMID: 18552845

GEO ID: GSE11111

PCL filename	short description	# conditions	tags	# channels
GSE11111_setA_family.pcl	Integrating large-scale functional genomic data to dissect the complexity of yeast regulatory networks.	9	transcriptional regulation	2

File last modified: Thu Apr 25 2024

---


# van_den_Brink_2008_PMID_18304306

Citation: van den Brink J, Daran-Lapujade P, Pronk JT, de Winde JH. 2008. New insights into the 
Saccharomyces cerevisiae fermentation switch: dynamic transcriptional response to 
anaerobicity and glucose-excess.

Full Description: BACKGROUND: The capacity of respiring cultures of Saccharomyces cerevisiae to immediately 
switch to fast alcoholic fermentation upon a transfer to anaerobic sugar-excess conditions
is a key characteristic of Saccharomyces cerevisiae in many of its industrial applications
. This transition was studied by exposing aerobic glucose-limited chemostat cultures grown
at a low specific growth rate to two simultaneous perturbations: oxygen depletion and 
relief of glucose limitation. RESULTS: The shift towards fully fermentative conditions 
caused a massive transcriptional reprogramming, where one third of all genes within the 
genome were transcribed differentially. The changes in transcript levels were mostly 
driven by relief from glucose-limitation. After an initial strong response to the addition
of glucose, the expression profile of most transcriptionally regulated genes displayed a 
clear switch at 30 minutes. In this respect, a striking difference was observed between 
the transcript profiles of genes encoding ribosomal proteins and those encoding ribosomal 
biogenesis components. Not all regulated genes responded with this binary profile. A group
of 87 genes showed a delayed and steady increase in expression that specifically responded
to anaerobiosis. CONCLUSION: Our study demonstrated that, despite the complexity of this 
multiple-input perturbation, the transcriptional responses could be categorized and 
biologically interpreted. By comparing this study with public datasets representing 
dynamic and steady conditions, 14 up-regulated and 11 down-regulated genes were determined
to be anaerobic specific. Therefore, these can be seen as true signature transcripts for 
anaerobicity under dynamic as well as under steady state conditions.

PMID: 18304306

GEO ID: GSE8187

PCL filename	short description	# conditions	tags	# channels
GSE8187_setA_family.pcl	New insights into the Saccharomyces cerevisiae fermentation switch: dynamic transcriptional response to anaerobicity and glucose-excess.	13	carbon utilization|fermentation|oxygen level alteration|respiration	1

File last modified: Thu Apr 25 2024

---


# Alejandro-Osorio_2009_PMID_19470158

Citation: Alejandro-Osorio AL, Huebert DJ, Porcaro DT, Sonntag ME, Nillasithanukroh S, Will JL, 
Gasch AP. 2009. The histone deacetylase Rpd3p is required for transient changes in genomic
expression in response to stress.

Full Description: BACKGROUND: Yeast responding to stress activate a large gene expression program called the
Environmental Stress Response that consists of approximately 600 repressed genes and 
approximately 300 induced genes. Numerous factors are implicated in regulating subsets of 
Environmental Stress Response genes; however, a complete picture of Environmental Stress 
Response regulation remains unclear. We investigated the role of the histone deacetylase 
Rpd3p, previously linked to the upstream regions of many Environmental Stress Response 
genes, in producing Environmental Stress Response gene expression changes in response to 
stress. RESULTS: We found that the Rpd3-Large complex is required for proper expression of
both induced and repressed Environmental Stress Response genes under multiple  stress 
conditions. Cells lacking RPD3 or the Rpd3-Large subunit PHO23 had a major defect in 
Environmental Stress Response initiation, particularly during the transient phase of 
expression immediately after stress exposure. Chromatin-immunoprecipitation showed a 
direct role for Rpd3-Large at representative genes; however, there were different effects 
on nucleosome occupancy and histone deacetylation at different promoters. Computational 
analysis implicated regulators that may act with Rpd3p at Environmental Stress Response 
genes. We provide genetic and biochemical evidence that Rpd3p is required for binding and 
action of the stress-activated transcription factor Msn2p, although the contribution of 
these factors differs for different genes. CONCLUSIONS: Our results implicate Rpd3p as an 
important co-factor in the Environmental Stress Response regulatory network, and suggest 
the importance of histone modification in producing transient changes in gene expression 
triggered  by stress.

PMID: 19470158

GEO ID: GSE9108.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE9108.final.pcl	The histone deacetylase Rpd3p is required for transient changes in genomic expression in response to stress.	109	histone modification|heat shock	2

File last modified: Thu Apr 25 2024

---


# Anderson_2009_PMID_19273689

Citation: Anderson JB, Sirjusingh C, Syed N, Lafayette S. 2009. Gene expression and evolution of 
antifungal drug resistance.

Full Description: Permanent changes in gene expression result from certain forms of antifungal resistance. 
In this study, we asked whether any changes in gene expression are required for the 
evolution of a drug-resistant phenotype in populations. We examined the changes in gene 
expression resulting from the evolution of resistance in experimental populations of the 
yeast Saccharomyces cerevisiae with two antifungal drugs, fluconazole (FLC) in a previous 
study and amphotericin B (AmB) in this study, in which five populations were subjected to 
increasing concentrations of AmB, from 0.25 to 128 microg/ml in twofold increments. Six 
genes, YGR035C, YOR1, ICT1, GRE2, PDR16, and YPLO88W, were consistently overexpressed with
resistance to AmB reported here and with resistance to FLC involving a mechanism of 
increased efflux reported previously. We then asked if the deletion of these genes 
impaired the ability of populations to evolve resistance to FLC over 108 generations of 
asexual reproduction in 32 and 128 microg/ml FLC, the same conditions under which FLC-
resistant types evolved originally. For each of three deletion strains, YOR1, ICT1, and 
PDR16 strains, extinctions occurred in one of two replicate populations growing in 128 
microg/ml FLC. Each of these three deletion strains was mixed 1:1 with a marked version of
the wild type to measure the relative ability of the deletion strain to adapt over 108 
generations. In these assays, only the PDR16 deletion strain consistently became extinct 
both at 32 and at 128 microg/ml FLC. The deletion of PDR16 reduces the capacity of a 
population to evolve to resistance to FLC.

PMID: 19273689

GEO ID: GSE12055

PCL filename	short description	# conditions	tags	# channels
GSE12055_set0_family.pcl	Gene expression and evolution of antifungal drug resistance.	46	chemical stimulus|evolution	2

File last modified: Thu Apr 25 2024

---


# Azzouz_2009_PMID_19155328

Citation: Azzouz N, Panasenko OO, Deluen C, Hsieh J, Theiler G, Collart MA. 2009. Specific roles for
the Ccr4-Not complex subunits in expression of the genome.

Full Description: In this work we used micro-array experiments to determine the role of each nonessential 
subunit of the conserved Ccr4-Not complex in the control of gene expression in the yeast 
Saccharomyces cerevisiae. The study was performed with cells growing exponentially in high
glucose and with cells grown to glucose depletion. Specific patterns of gene deregulation 
were observed upon deletion of any given subunit, revealing the specificity of each 
subunit's function. Consistently, the purification of the Ccr4-Not complex through Caf40p 
by tandem affinity purification from wild-type cells or cells lacking individual subunits 
of the Ccr4-Not complex revealed that each subunit had a particular impact on complex 
integrity. Furthermore, the micro-arrays revealed that the role of each subunit was 
specific to the growth conditions. From the study of only two different growth conditions
, revealing an impact of the Ccr4-Not complex on more than 85% of all studied genes, we 
can infer that the Ccr4-Not complex is important for expression of most of the yeast 
genome.

PMID: 19155328

GEO ID: GSE10521

PCL filename	short description	# conditions	tags	# channels
GSE10521_setA_family.pcl	Specific roles for the Ccr4-Not complex subunits in expression of the genome.	25	carbon utilization|diauxic shift|fermentation|respiration	1

File last modified: Thu Apr 25 2024

---


# Borde_2009_PMID_19078966

Citation: Borde V, Robine N, Lin W, Bonfils S, Geli V, Nicolas A. 2009. Histone H3 lysine 4 
trimethylation marks meiotic recombination initiation sites.

Full Description: The function of histone modifications in initiating and regulating the chromosomal events 
of the meiotic prophase remains poorly understood. In Saccharomyces cerevisiae, we 
examined the genome-wide localization of histone H3 lysine 4 trimethylation (H3K4me3) 
along meiosis and its relationship to gene expression and position of the programmed 
double-strand breaks (DSBs) that initiate interhomologue recombination, essential to yield
viable haploid gametes. We find that the level of H3K4me3 is constitutively higher close 
to DSB sites, independently of local gene expression levels. Without Set1, the H3K4 
methylase, 84% of the DSB sites exhibit a severely reduced DSB frequency, the reduction 
being quantitatively correlated with the local level of H3K4me3 in wild-type cells. 
Further, we show that this differential histone mark is already established in vegetative 
cells, being higher in DSB-prone regions than in regions with no or little DSB. Taken 
together, our results demonstrate that H3K4me3 is a prominent and preexisting mark of 
active meiotic recombination initiation sites. Novel perspectives to dissect the various 
layers of the controls of meiotic DSB formation are discussed.

PMID: 19078966

GEO ID: GSE10948

PCL filename	short description	# conditions	tags	# channels
GSE10948_setA_family.pcl	Expression during meiosis in a clb5 clb6 double-deletion mutant deficient in meiotic replication; 8 hr time course	7	sporulation	2

File last modified: Thu Apr 25 2024
GSE10947_setA_family.pcl	Expression during meiosis in a spo11-Y135F mutant deficient in meiotic DSB formation; 8 hr time course	7	DNA damage stimulus|sporulation	2

File last modified: Thu Apr 25 2024
GSE10944_setA_family.pcl	Expression during meiosis; 8 hr time course	7	sporulation	2

File last modified: Thu Apr 25 2024

---


# Bradley_2009_PMID_19180179

Citation: Bradley PH, Brauer MJ, Rabinowitz JD, Troyanskaya OG. 2009. Coordinated concentration 
changes of transcripts and metabolites in Saccharomyces cerevisiae.

Full Description: Metabolite concentrations can regulate gene expression, which can in turn regulate 
metabolic activity. The extent to which functionally related transcripts and metabolites 
show similar patterns of concentration changes, however, remains unestablished. We measure
and analyze the metabolomic and transcriptional responses of Saccharomyces cerevisiae to 
carbon and nitrogen starvation. Our analysis demonstrates that transcripts and metabolites
show coordinated response dynamics. Furthermore, metabolites and gene products whose 
concentration profiles are alike tend to participate in related biological processes. To 
identify specific, functionally related genes and metabolites, we develop an approach 
based on Bayesian integration of the joint metabolomic and transcriptomic data. This 
algorithm finds interactions by evaluating transcript-metabolite correlations in light of 
the experimental context in which they occur and the class of metabolite involved. It 
effectively predicts known enzymatic and regulatory relationships, including a gene-
metabolite interaction central to the glycolytic-gluconeogenetic switch. This work 
provides quantitative evidence that functionally related metabolites and transcripts show 
coherent patterns of behavior on the genome scale and lays the groundwork for building 
gene-metabolite interaction networks directly from systems-level data.

PMID: 19180179

GEO ID: GSE11754

PCL filename	short description	# conditions	tags	# channels
GSE11754_set1_family.pcl	Response to nitrogen starvation, 8 hr time course	6	nitrogen utilization|starvation	2

File last modified: Thu Apr 25 2024
GSE11754_set0_family.pcl	Response to carbon starvation, 8 hr time course	6	carbon utilization|starvation	2

File last modified: Thu Apr 25 2024

---


# Falbo_2009_PMID_19855395

Citation: Falbo KB, Alabert C, Katou Y, Wu S, Han J, Wehr T, Xiao J, He X, Zhang Z, Shi Y, Shirahige
K, Pasero P, Shen X. 2009. Involvement of a chromatin remodeling complex in damage 
tolerance during DNA replication.

Full Description: ATP-dependent chromatin remodeling complexes have been shown to participate in DNA 
replication in addition to transcription and DNA repair. However, the mechanisms of their 
involvement in DNA replication remain unclear. Here, we reveal a specific function of the 
yeast INO80 chromatin remodeling complex in the DNA damage tolerance pathways. Whereas 
INO80 is necessary for the resumption of replication at forks stalled by methyl methane 
sulfonate (MMS), it is not required for replication fork collapse after treatment with 
hydroxyurea (HU). Mechanistically, INO80 regulates DNA damage tolerance during replication
through  modulation of PCNA (proliferating cell nuclear antigen) ubiquitination and Rad51-
mediated processing of recombination intermediates at impeded replication forks. Our 
findings establish a mechanistic link between INO80 and DNA damage tolerance pathways, 
indicating that chromatin remodeling is important for accurate DNA replication.

PMID: 19855395

GEO ID: GSE18555.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE18555.final.pcl	Involvement of a chromatin remodeling complex in damage tolerance during DNA replication.	6	DNA damage stimulus|chromatin organization|mitotic cell cycle	2

File last modified: Thu Apr 25 2024

---


# Garcia_2009_PMID_19234305

Citation: Garcia R, Rodriguez-Pena JM, Bermejo C, Nombela C, Arroyo J. 2009. The high osmotic 
response and cell wall integrity pathways cooperate to regulate transcriptional responses 
to zymolyase-induced cell wall stress in Saccharomyces cerevisiae.

Full Description: The adaptation of Saccharomyces cerevisiae to situations in which cell wall integrity is 
seriously compromised mainly involves the cell wall integrity (CWI) pathway. However, in a
recent work ( Bermejo, C., Rodriguez, E., Garcia, R., Rodriguez-Pena, J. M., Rodriguez de 
la Concepcion, M. L., Rivas, C., Arias, P., Nombela, C., Posas, F., and Arroyo, J. (2008) 
Mol. Biol. Cell 19, 1113-1124 ) we have demonstrated the co-participation of the high 
osmotic response (HOG) pathway to ensure yeast survival to cell wall stress mediated by 
zymolyase, which hydrolyzes the beta-1,3 glucan network. Here we have characterized the 
role of both pathways in the regulation of the overall yeast transcriptional responses to 
zymolyase treatment using whole genome expression profiling. A main group of yeast genes 
is dependent on both MAPKs, Slt2 and Hog1, for their induction. The transcriptional 
activation of these genes depends on the MAPKKK Bck1, the transcription factor Rlm1, and 
elements of the sho1 branch of the HOG pathway, but not on the sensors of the CWI pathway
. A second group of genes is dependent on Slt2 but not Hog1 or Pbs2. However, the 
induction of these genes is dependent on upstream elements of the HOG pathway such as Sho1
, Ste50, and Ste11, in accordance with a sequential activation of the HOG and CWI pathways
. Zymolyase also promotes an osmotic-like transcriptional response with the activation of 
a group of genes dependent on elements of the Sho1 branch of HOG pathway but not on Slt2, 
with the induction of many of them dependent on Msn2/4. Additionally, in the absence of 
Hog1, zymolyase induces an alternative response related to mating and filamentation as a 
consequence of the cross-talk between these pathways and the HOG pathway. Finally, in the 
absence of Slt2, zymolyase increases the induction of genes associated with osmotic 
adaptation with respect to the wild type, suggesting an inhibitory effect of the CWI 
pathway over the HOG pathway. These studies clearly reveal the complexity of the signal 
transduction machinery responsible for regulating yeast adaptation responses to cell wall 
stress.

PMID: 19234305

GEO ID: GSE12684

PCL filename	short description	# conditions	tags	# channels
GSE12684_setA_family.pcl	The high osmotic response and cell wall integrity pathways cooperate to regulate transcriptional responses to zymolyase-induced cell wall stress in Saccharomyces cerevisiae.	18	cell wall organization|signaling	2

File last modified: Thu Apr 25 2024

---


# Holbein_2009_PMID_19324962

Citation: Holbein S, Wengi A, Decourty L, Freimoser FM, Jacquier A, Dichtl B. 2009. Cordycepin 
interferes with 3' end formation in yeast independently of its potential to terminate RNA 
chain elongation.

Full Description: Cordycepin (3' deoxyadenosine) is a biologically active compound that, when incorporated 
during RNA synthesis in vitro, provokes chain termination due to the absence of a 3' 
hydroxyl moiety. We were interested in the effects mediated by this drug in vivo and 
analyzed its impact on RNA metabolism of yeast. Our results support the view that 
cordycepin-triphosphate (CoTP) is the toxic component that is limiting cell growth through
inhibition of RNA synthesis. Unexpectedly, cordycepin treatment modulated 3' end 
heterogeneity of ACT1 and ASC1 mRNAs and rapidly induced extended transcripts derived from
CYH2 and NEL025c loci. Moreover, cordycepin ameliorated the growth defects of poly(A) 
polymerase mutants and the pap1-1 mutation neutralized the effects of the drug on gene 
expression. Our observations are consistent with an epistatic relationship between poly(A
) polymerase function and cordycepin action and suggest that a major mode of cordycepin 
activity reduces 3' end formation efficiency independently of its potential to terminate 
RNA chain elongation. Finally, chemical-genetic profiling revealed genome-wide pathways 
linked to cordycepin activity and identified novel genes involved in poly(A) homeostasis.

PMID: 19324962

GEO ID: GSE14619

PCL filename	short description	# conditions	tags	# channels
GSE14619_setA_family.pcl	Cordycepin interferes with 3' end formation in yeast independently of its potential to terminate RNA chain elongation.	9	chemical stimulus|mRNA processing	2

File last modified: Thu Apr 25 2024

---


# Kim_2009_PMID_19456872

Citation: Kim HS, Huh J, Fay JC. 2009. Dissecting the pleiotropic consequences of a quantitative 
trait nucleotide.

Full Description: The downstream consequences of a single quantitative trait polymorphism can provide 
important insight into the molecular basis of a trait. However, the molecular consequences
of a polymorphism may be complex and only a subset of these may influence the trait of 
interest. In natural isolates of Saccharomyces cerevisiae, a nonsynonymous polymorphism in
cystathione beta-synthase (CYS4) causes a deficiency in both cysteine and glutathione that
results in rust-colored colonies and drug-dependent growth defects. Using a single-
nucleotide allele replacement, we characterized the effects of this polymorphism on gene 
expression levels across the genome. To determine whether any of the differentially 
expressed genes are necessary for the production of rust-colored colonies, we screened the
yeast deletion collection for genes that enhance or suppress rust coloration. We found 
that genes in the sulfur assimilation pathway are required for the production of rust 
color but not the drug-sensitivity phenotype. Our results show that a single quantitative 
trait polymorphism can generate a complex set of downstream changes, providing a molecular
basis for pleiotropy.

PMID: 19456872

GEO ID: GSE14102

PCL filename	short description	# conditions	tags	# channels
GSE14102_setA_family.pcl	Dissecting the pleiotropic consequences of a quantitative trait nucleotide.	9	amino acid metabolism	2

File last modified: Thu Apr 25 2024

---


# Knijnenburg_2009_PMID_19173729

Citation: Knijnenburg TA, Daran JM, van den Broek MA, Daran-Lapujade PA, de Winde JH, Pronk JT, 
Reinders MJ, Wessels LF. 2009. Combinatorial effects of environmental parameters on 
transcriptional regulation in Saccharomyces cerevisiae: a quantitative analysis of a 
compendium of chemostat-based transcriptome data.

Full Description: BACKGROUND: Microorganisms adapt their transcriptome by integrating multiple chemical and 
physical signals from their environment. Shake-flask cultivation does not allow precise 
manipulation of individual culture parameters and therefore precludes a quantitative 
analysis of the (combinatorial) influence of these parameters on transcriptional 
regulation. Steady-state chemostat cultures, which do enable accurate control, measurement
and manipulation of individual cultivation parameters (e.g. specific growth rate, 
temperature, identity of the growth-limiting nutrient) appear to provide a promising 
experimental platform for such a combinatorial analysis. RESULTS: A microarray compendium 
of 170 steady-state chemostat cultures of the yeast Saccharomyces cerevisiae is presented 
and analyzed. The 170 microarrays encompass 55 unique conditions, which can be 
characterized by the combined settings of 10 different cultivation parameters. By applying
a regression model to assess the impact of (combinations of) cultivation parameters on the
transcriptome, most S. cerevisiae genes were shown to be influenced by multiple 
cultivation parameters, and in many cases by combinatorial effects of cultivation 
parameters. The inclusion of these combinatorial effects in the regression model led to 
higher explained variance of the gene expression patterns and resulted in higher function 
enrichment in subsequent analysis. We further demonstrate the usefulness of the compendium
and regression analysis for interpretation of shake-flask-based transcriptome studies and 
for guiding functional analysis of (uncharacterized) genes and pathways. CONCLUSION: 
Modeling the combinatorial effects of environmental parameters on the transcriptome is 
crucial for understanding transcriptional regulation. Chemostat cultivation offers a 
powerful tool for such an approach.

PMID: 19173729

GEO ID: GSE11452

PCL filename	short description	# conditions	tags	# channels
GSE11452_setA_family.pcl	Combinatorial effects of environmental parameters on transcriptional regulation in Saccharomyces cerevisiae: a quantitative analysis of a compendium of chemostat-based transcriptome data.	170	carbon utilization|nitrogen utilization|nutrient utilization|phosphorus utilization|sulfur utilization	1

File last modified: Thu Apr 25 2024

---


# Moxley_2009_PMID_19346491

Citation: Moxley JF, Jewett MC, Antoniewicz MR, Villas-Boas SG, Alper H, Wheeler RT, Tong L, 
Hinnebusch AG, Ideker T, Nielsen J, Stephanopoulos G. 2009. Linking high-resolution 
metabolic flux phenotypes and transcriptional regulation  in yeast modulated by the global
regulator Gcn4p.

Full Description: Genome sequencing dramatically increased our ability to understand cellular response to 
perturbation. Integrating system-wide measurements such as gene expression with networks 
of protein-protein interactions and transcription factor binding revealed critical 
insights into cellular behavior. However, the potential of systems biology approaches is 
limited by difficulties in integrating metabolic measurements across the functional levels
of the cell despite their being most closely linked to cellular phenotype. To address this
limitation, we developed a  model-based approach to correlate mRNA and metabolic flux data
that combines information from both interaction network models and flux determination 
models. We started by quantifying 5,764 mRNAs, 54 metabolites, and 83 experimental (13)C-
based reaction fluxes in continuous cultures of yeast under stress in the absence or 
presence of global regulator Gcn4p. Although mRNA expression alone did not directly 
predict metabolic response, this correlation improved through incorporating a network-
based model of amino acid biosynthesis (from r = 0.07 to  0.80 for mRNA-flux agreement). 
The model provides evidence of general biological  principles: rewiring of metabolic flux
(i.e., use of different reaction pathways) by transcriptional regulation and metabolite 
interaction density (i.e., level of  pairwise metabolite-protein interactions) as a key 
biosynthetic control determinant. Furthermore, this model predicted flux rewiring in 
studies of follow-on transcriptional regulators that were experimentally validated with 
additional (13)C-based flux measurements. As a first step in linking metabolic control and
genetic regulatory networks, this model underscores the importance of integrating diverse 
data types in large-scale cellular models. We anticipate that an integrated approach 
focusing on metabolic measurements will facilitate construction of more realistic models 
of cellular regulation for understanding diseases and constructing strains for industrial 
applications.

PMID: 19346491

GEO ID: GSE4709.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE4709.final.pcl	Linking high-resolution metabolic flux phenotypes and transcriptional regulation  in yeast modulated by the global regulator Gcn4p.	6	transcription|stress|nutrient utilization|carbon utilization	1

File last modified: Thu Apr 25 2024
GSE7369.final.pcl	Linking high-resolution metabolic flux phenotypes and transcriptional regulation  in yeast modulated by the global regulator Gcn4p.	4	amino acid utilization	2

File last modified: Thu Apr 25 2024

---


# Rodriguez-Quinones_2009_PMID_19426543

Citation: Rodriguez-Quinones JF, Rodriguez-Medina JR. 2009. Differential gene expression signatures 
for cell wall integrity found in chitin synthase II (chs2Delta) and myosin II (myo1Delta) 
deficient cytokinesis mutants of Saccharomyces cerevisiae.

Full Description: BACKGROUND: Myosin II-dependent contraction of the cytokinetic ring and primary septum 
formation by chitin synthase II are interdependent processes during cytokinesis in 
Saccharomyces cerevisiae. Hence, null mutants of myosin II (myo1Delta) and chitin synthase
II (chs2Delta) share multiple morphological and molecular phenotypes. To understand the 
nature of their interdependent functions, we will seek to identify genes undergoing 
transcriptional regulation in chs2Delta strains and to establish a transcription signature
profile for comparison with myo1Delta strains. RESULTS: A total of 467 genes were commonly
regulated between myo1Delta and chs2Delta mutant strains (p <or= 0.01). Common regulated 
biological process categories identified by Gene Set Enrichment Analysis (GSEA) in both 
gene expression profiles were: protein biosynthesis, RNA processing, and stress response. 
Expression of 17/20 genes in the main transcriptional fingerprint for cell wall stress was
confirmed in the chs2Delta strain versus 5/20 for the myo1Delta strain. One of these genes
, SLT2/MPK1, was up-regulated in both strains and both strains accumulated the 
hyperphosphorylated form of Slt2p thereby confirming that the PKC1 cell wall integrity 
pathway (CWIP) was activated by both mutations. The SLT2/MPK1 gene, essential for 
myo1Delta strains, was not required in the chs2Delta strain. CONCLUSION: Comparison of the
chs2Delta and myo1Delta gene expression profiles revealed similarities in the biological 
process categories that respond to the chs2Delta and myo1Delta gene mutations. This 
supports the view that these mutations affect a common function in cytokinesis. Despite 
their similarities, these mutants exhibited significant differences in expression of the 
main transcriptional fingerprint for cell wall stress and their requirement of the CWIP 
for survival.

PMID: 19426543

GEO ID: GSE5931

PCL filename	short description	# conditions	tags	# channels
GSE5931_setA_family.pcl	Differential gene expression signatures for cell wall integrity found in chitin synthase II (chs2Delta) and myosin II (myo1Delta) deficient cytokinesis mutants of Saccharomyces cerevisiae.	6	signaling|stress	2

File last modified: Thu Apr 25 2024

---


# Rossouw_2009_PMID_19711068

Citation: Rossouw D, Bauer FF. 2009. Comparing the transcriptomes of wine yeast strains: toward 
understanding the interaction between environment and transcriptome during fermentation.

Full Description: System-wide omics approaches have been widely applied to study a limited number of 
laboratory strains of Saccharomyces cerevisiae. More recently, industrial S. cerevisiae 
strains have become the target of such analyses, mainly to improve our understanding of 
biotechnologically relevant phenotypes that cannot be adequately studied in laboratory 
strains. Most of these studies have investigated single strains in a single medium. This 
experimental layout cannot differentiate between generally relevant molecular responses 
and strain- or media-specific features. Here we analyzed the transcriptomes of two 
phenotypically diverging wine yeast strains in two different fermentation media at three 
stages of wine fermentation. The data show that the intersection of transcriptome datasets
from fermentations using either synthetic MS300 (simulated wine must) or real grape must (
Colombard) can help to delineate relevant from noisy changes in gene expression in 
response to experimental factors such as fermentation stage and strain identity. The 
differences in the expression profiles of strains in the different environments also 
provide relevant insights into the transcriptional responses toward specific compositional
features of the media. The data also suggest that MS300 is a representative environment 
for conducting research on wine fermentation and industrially relevant properties of wine 
yeast strains.

PMID: 19711068

GEO ID: GSE11651

PCL filename	short description	# conditions	tags	# channels
GSE11651_setA_family.pcl	Comparing the transcriptomes of wine yeast strains: toward understanding the interaction between environment and transcriptome during fermentation.	43	carbon utilization|fermentation	1

File last modified: Thu Apr 25 2024

---


# Sprouse_2009_PMID_19098311

Citation: Sprouse RO, Wells MN, Auble DT. 2009. TATA-binding protein variants that bypass the 
requirement for Mot1 in vivo.

Full Description: Mot1 is an essential TATA-binding protein (TBP)-associated factor and Snf2/Swi2 ATPase 
that both represses and activates transcription. Biochemical and structural results 
support a model in which ATP binding and hydrolysis induce a conformational change in Mot1
that drives local translocation along DNA, thus removing TBP. Although this activity 
explains transcriptional repression, it does not as easily explain Mot1-mediated 
transcriptional activation, and several different models have been proposed to explain how
Mot1 activates transcription. To better understand the function of Mot1 in yeast cells in 
vivo, particularly with regard to gene activation, TBP mutants were identified that bypass
the requirement for Mot1 in vivo. Although TBP has been extensively mutated and analyzed 
previously, this screen uncovered two novel TBP variants that are unique in their ability 
to bypass the requirement for Mot1. Surprisingly, in vitro analyses reveal that rather 
than having acquired an improved biochemical activity, one of the TBPs was defective for 
interaction with polymerase II preinitiation complex (PIC) components and other regulators
of TBP function. The other mutant was defective for DNA binding in vitro yet was still 
recruited to chromatin in vivo. These results suggest that Mot1-mediated dissociation of 
TBP (or TBP-containing complexes) from chromatin can explain the Mot1 activation mechanism
at some promoters. The results also suggest that PICs can be dynamically unstable and that
appropriate PIC instability is critical for the regulation of transcription in vivo.

PMID: 19098311

GEO ID: GSE12371

PCL filename	short description	# conditions	tags	# channels
GSE12371_setA_family.pcl	TATA-binding protein variants that bypass the requirement for Mot1 in vivo.	6	transcription	2

File last modified: Thu Apr 25 2024

---


# Zhu_2009_PMID_19158363

Citation: Zhu C, Byers KJ, McCord RP, Shi Z, Berger MF, Newburger DE, Saulrieta K, Smith Z, Shah MV,
Radhakrishnan M, Philippakis AA, Hu Y, De Masi F, Pacek M, Rolfs A, Murthy T, Labaer J, 
Bulyk ML. 2009. High-resolution DNA-binding specificity analysis of yeast transcription 
factors.

Full Description: Transcription factors (TFs) regulate the expression of genes through sequence-specific 
interactions with DNA-binding sites. However, despite recent progress in identifying in 
vivo TF binding sites by microarray readout of chromatin immunoprecipitation (ChIP-chip), 
nearly half of all known yeast TFs are of unknown DNA-binding specificities, and many 
additional predicted TFs remain uncharacterized. To address these gaps in our knowledge of
yeast TFs and their cis regulatory sequences, we have determined high-resolution binding 
profiles for 89 known and predicted yeast TFs, over more than 2.3 million gapped and 
ungapped 8-bp sequences (k-mers). We report 50 new or significantly different direct DNA-
binding site motifs for yeast DNA-binding proteins and motifs for eight proteins for which
only a consensus sequence was previously known; in total, this corresponds to over a 50% 
increase in the number of yeast DNA-binding proteins with experimentally determined DNA-
binding specificities. Among other novel regulators, we discovered proteins that bind the 
PAC (Polymerase A and C) motif (GATGAG) and regulate ribosomal RNA (rRNA) transcription 
and processing, core cellular processes that are constituent to ribosome biogenesis. In 
contrast to earlier data types, these comprehensive k-mer binding data permit us to 
consider the regulatory potential of genomic sequence at the individual word level. These 
k-mer data allowed us to reannotate in vivo TF binding targets as direct or indirect and 
to examine TFs' potential effects on gene expression in approximately 1,700 environmental 
and cellular conditions. These approaches could be adapted to identify TFs and cis 
regulatory elements in higher eukaryotes.

PMID: 19158363

GEO ID: GSE13684

PCL filename	short description	# conditions	tags	# channels
GSE13684_setA_family.pcl	High-resolution DNA-binding specificity analysis of yeast transcription factors.	24	heat shock|transcription	1

File last modified: Thu Apr 25 2024

---


# Adomas_2010_PMID_20716356

Citation: Adomas AB, Lopez-Giraldez F, Clark TA, Wang Z, Townsend JP. 2010. Multi-targeted priming 
for genome-wide gene expression assays.

Full Description: BACKGROUND: Complementary approaches to assaying global gene expression are needed to 
assess gene expression in regions that are poorly assayed by current methodologies. A key 
component of nearly all gene expression assays is the reverse transcription of transcribed
sequences that has traditionally been performed by priming the poly-A tails on many of the
transcribed genes in eukaryotes with oligo-dT, or by priming RNA indiscriminately with 
random hexamers. We designed an algorithm to find common sequence motifs that were present
within most protein-coding genes of Saccharomyces cerevisiae and of Neurospora crassa, but
that were not present within their ribosomal RNA or transfer RNA genes. We then 
experimentally tested whether degenerately priming these motifs with multi-targeted 
primers improved the accuracy and completeness of transcriptomic assays. RESULTS: We 
discovered two multi-targeted primers that  would prime a preponderance of genes in the 
genomes of Saccharomyces cerevisiae and Neurospora crassa while avoiding priming ribosomal
RNA or transfer RNA. Examining the response of Saccharomyces cerevisiae to nitrogen 
deficiency and profiling Neurospora crassa early sexual development, we demonstrated that 
using  multi-targeted primers in reverse transcription led to superior performance of 
microarray profiling and next-generation RNA tag sequencing. Priming with multi-targeted 
primers in addition to oligo-dT resulted in higher sensitivity, a  larger number of well-
measured genes and greater power to detect differences in gene expression. CONCLUSIONS: 
Our results provide the most complete and detailed  expression profiles of the yeast 
nitrogen starvation response and N. crassa early sexual development to date. Furthermore, 
our multi-targeting priming methodology  for genome-wide gene expression assays provides 
selective targeting of multiple sequences and counter-selection against undesirable 
sequences, facilitating a more complete and precise assay of the transcribed sequences 
within the genome.

PMID: 20716356

GEO ID: GSE22992.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE22992.final.pcl	Multi-targeted priming for genome-wide gene expression assays.	4	nitrogen utilization|stress	2

File last modified: Thu Apr 25 2024
GSE22936.final.pcl	Multi-targeted priming for genome-wide gene expression assays.	4	nitrogen utilization|stress	2

File last modified: Thu Apr 25 2024

---


# Aghajan_2010_PMID_20581845

Citation: Aghajan M, Jonai N, Flick K, Fu F, Luo M, Cai X, Ouni I, Pierce N, Tang X, Lomenick B, 
Damoiseaux R, Hao R, Del Moral PM, Verma R, Li Y, Li C, Houk KN, Jung ME, Zheng N, Huang L,
Deshaies RJ, Kaiser P, Huang J. 2010. Chemical genetics screen for enhancers of rapamycin 
identifies a specific inhibitor of an SCF family E3 ubiquitin ligase.

Full Description: The target of rapamycin (TOR) plays a central role in eukaryotic cell growth control. With
prevalent hyperactivation of the mammalian TOR (mTOR) pathway in human cancers, strategies
to enhance TOR pathway inhibition are needed. We used a yeast-based screen to identify 
small-molecule enhancers of rapamycin (SMERs) and  discovered an inhibitor (SMER3) of the 
Skp1-Cullin-F-box (SCF)(Met30) ubiquitin ligase, a member of the SCF E3-ligase family, 
which regulates diverse cellular processes including transcription, cell-cycle control and
immune response. We show here that SMER3 inhibits SCF(Met30) in vivo and in vitro, but not
the closely related SCF(Cdc4). Furthermore, we demonstrate that SMER3 diminishes binding 
of the F-box subunit Met30 to the SCF core complex in vivo and show evidence for SMER3 
directly binding to Met30. Our results show that there is no fundamental barrier to 
obtaining specific inhibitors to modulate function of individual SCF complexes.

PMID: 20581845

GEO ID: GSE22269.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE22269.final.pcl	Chemical genetics screen for enhancers of rapamycin identifies a specific inhibitor of an SCF family E3 ubiquitin ligase.	13	chemical stimulus|cell cycle regulation	1

File last modified: Thu Apr 25 2024

---


# Amit_2010_PMID_20600117

Citation: Amit M, Weisberg SJ, Nadler-Holly M, McCormack EA, Feldmesser E, Kaganovich D, Willison KR,
Horovitz A. 2010. Equivalent mutations in the eight subunits of the chaperonin CCT produce
dramatically different cellular and gene expression phenotypes.

Full Description: The eukaryotic cytoplasmic chaperonin-containing TCP-1 (CCT) is a complex formed  by two 
back-to-back stacked hetero-octameric rings that assists the folding of actins, tubulins, 
and other proteins in an ATP-dependent manner. Here, we tested  the significance of the 
hetero-oligomeric nature of CCT in its function by introducing, in each of the eight 
subunits in turn, an identical mutation at a position that is conserved in all the 
subunits and is involved in ATP hydrolysis, in order to establish the extent of '
individuality' of the various subunits. Our  results show that these identical mutations 
lead to dramatically different phenotypes. For example, Saccharomyces cerevisiae yeast 
cells with the mutation in subunit CCT2 display heat sensitivity and cold sensitivity for 
growth, have an excess of actin patches, and are the only strain here generated that is 
pseudo-diploid. By contrast, cells with the mutation in subunit CCT7 are the only ones to 
accumulate juxtanuclear protein aggregates that may reflect an impaired stress response in
this strain. System-level analysis of the strains using RNA microarrays reveals 
connections between CCT and several cellular networks, including ribosome biogenesis and 
TOR2, that help to explain the phenotypic variability observed.

PMID: 20600117

GEO ID: GSE19511.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE19511.final.pcl	Equivalent mutations in the eight subunits of the chaperonin CCT produce dramatically different cellular and gene expression phenotypes.	16	protein structure and folding	1

File last modified: Thu Apr 25 2024

---


# Berthelet_2010_PMID_20439772

Citation: Berthelet S, Usher J, Shulist K, Hamza A, Maltez N, Johnston A, Fong Y, Harris LJ, Baetz K,
 2010. Functional genomics analysis of the Saccharomyces cerevisiae iron responsive 
transcription factor Aft1 reveals iron-independent functions.

Full Description: The Saccharomyces cerevisiae transcription factor Aft1 is activated in iron-deficient 
cells to induce the expression of iron regulon genes, which coordinate the increase of 
iron uptake and remodel cellular metabolism to survive low-iron conditions. In addition, 
Aft1 has been implicated in numerous cellular processes including cell-cycle progression 
and chromosome stability; however, it  is unclear if all cellular effects of Aft1 are 
mediated through iron homeostasis. To further investigate the cellular processes affected 
by Aft1, we identified >70 deletion mutants that are sensitive to perturbations in AFT1 
levels using genome-wide synthetic lethal and synthetic dosage lethal screens. Our genetic
network reveals that Aft1 affects a diverse range of cellular processes, including the 
RIM101 pH pathway, cell-wall stability, DNA damage, protein transport, chromosome 
stability, and mitochondrial function. Surprisingly, only a subset of mutants identified 
are sensitive to extracellular iron fluctuations or  display genetic interactions with 
mutants of iron regulon genes AFT2 or FET3. We  demonstrate that Aft1 works in parallel 
with the RIM101 pH pathway and the role of Aft1 in DNA damage repair is mediated by iron. 
In contrast, through both directed studies and microarray transcriptional profiling, we 
show that the role  of Aft1 in chromosome maintenance and benomyl resistance is 
independent of its iron regulatory role, potentially through a nontranscriptional 
mechanism.

PMID: 20439772

GEO ID: GSE20531.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE20531.final.pcl	Functional genomics analysis of the Saccharomyces cerevisiae iron responsive transcription factor Aft1 reveals iron-independent functions.	12	metal or metalloid ion stress	1

File last modified: Thu Apr 25 2024

---


# Canelas_2010_PMID_21266995

Citation: Canelas AB, Harrison N, Fazio A, Zhang J, Pitkanen JP, van den Brink J, Bakker BM, Bogner 
L, Bouwman J, Castrillo JI, Cankorur A, Chumnanpuen P, Daran-Lapujade P, Dikicioglu D, van
Eunen K, Ewald JC, Heijnen JJ, Kirdar B, Mattila I, Mensonides FI, Niebel A, Penttila M, 
Pronk JT, Reuss M, Salusjarvi L, Sauer U, Sherman D, Siemann-Herzberg M, Westerhoff H, de 
Winde J, Petranovic D, Oliver SG, Workman CT, Zamboni N, Nielsen J. 2010. Integrated 
multilaboratory systems biology reveals differences in protein metabolism between two 
reference yeast strains.

Full Description: The field of systems biology is often held back by difficulties in obtaining comprehensive
, high-quality, quantitative data sets. In this paper, we undertook  an interlaboratory 
effort to generate such a data set for a very large number of  cellular components in the 
yeast Saccharomyces cerevisiae, a widely used model organism that is also used in the 
production of fuels, chemicals, food ingredients and pharmaceuticals. With the current 
focus on biofuels and sustainability, there is much interest in harnessing this species as
a general cell factory. In this study, we characterized two yeast strains, under two 
standard growth conditions. We ensured the high quality of the experimental data  by 
evaluating a wide range of sampling and analytical techniques. Here we show significant 
differences in the maximum specific growth rate and biomass yield between the two strains
. On the basis of the integrated analysis of the high-throughput data, we hypothesize that
differences in phenotype are due to differences in protein metabolism.

PMID: 21266995

GEO ID: GSE24234.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE24234.final.pcl	Integrated multilaboratory systems biology reveals differences in protein metabolism between two reference yeast strains.	12	genome variation|metabolism|transcription	1

File last modified: Thu Apr 25 2024

---


# Dowell_2010_PMID_20844014

Citation: Dowell NL, Sperling AS, Mason MJ, Johnson RC. 2010. Chromatin-dependent binding of the S. 
cerevisiae HMGB protein Nhp6A affects nucleosome dynamics and transcription.

Full Description: The Saccharomyces cerevisiae protein Nhp6A is a model for the abundant and multifunctional
high-mobility group B (HMGB) family of chromatin-associated proteins. Nhp6A binds DNA in 
vitro without sequence specificity and bends DNA sharply, but its role in chromosome 
biology is poorly understood. We show by whole-genome chromatin immunoprecipitation (ChIP
) and high-resolution whole-genome tiling arrays (ChIP-chip) that Nhp6A is localized to 
specific regions of chromosomes that include approximately 23% of RNA polymerase II 
promoters. Nhp6A binding functions to stabilize nucleosomes, particularly at the  
transcription start site of these genes. Both genomic binding and transcript expression 
studies point to functionally related groups of genes that are bound specifically by Nhp6A
and whose transcription is altered by the absence of Nhp6.  Genomic analyses of Nhp6A 
mutants specifically defective in DNA bending reveal a  critical role of DNA bending for 
stabilizing chromatin and coregulation of transcription but not for targeted binding by 
Nhp6A. We conclude that the chromatin environment, not DNA sequence recognition, localizes
Nhp6A binding, and that Nhp6A stabilizes chromatin structure and coregulates transcription.

PMID: 20844014

GEO ID: GSE23607.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE23607.final.pcl	Chromatin-dependent binding of the S. cerevisiae HMGB protein Nhp6A affects nucleosome dynamics and transcription.	8	chromatin organization	2

File last modified: Thu Apr 25 2024

---


# Fendt_2010_PMID_21119627

Citation: Fendt SM, Oliveira AP, Christen S, Picotti P, Dechant RC, Sauer U. 2010. Unraveling 
condition-dependent networks of transcription factors that control metabolic pathway 
activity in yeast.

Full Description: Which transcription factors control the distribution of metabolic fluxes under a  given 
condition? We address this question by systematically quantifying metabolic fluxes in 119 
transcription factor deletion mutants of Saccharomyces cerevisiae under five growth 
conditions. While most knockouts did not affect fluxes, we identified 42 condition-
dependent interactions that were mediated by a total of 23 transcription factors that 
control almost exclusively the cellular decision between respiration and fermentation. 
This relatively sparse, condition-specific  network of active metabolic control contrasts 
with the much larger gene regulation network inferred from expression and DNA binding data
. Based on protein and transcript analyses in key mutants, we identified three enzymes in 
the tricarboxylic acid cycle as the key targets of this transcriptional control.  For the 
transcription factor Gcn4, we demonstrate that this control is mediated through the PKA 
and Snf1 signaling cascade. The discrepancy between flux response predictions, based on 
the known regulatory network architecture and our functional (13)C-data, demonstrates the 
importance of identifying and quantifying the extent to which regulatory effectors alter 
cellular functions.

PMID: 21119627

GEO ID: GSE24056.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE24056.final.pcl	Unraveling condition-dependent networks of transcription factors that control metabolic pathway activity in yeast.	6	carbon utilization	1

File last modified: Thu Apr 25 2024
GSE19569.final.pcl	Unraveling condition-dependent networks of transcription factors that control metabolic pathway activity in yeast.	6	carbon utilization	1

File last modified: Thu Apr 25 2024
GSE24053.final.pcl	Unraveling condition-dependent networks of transcription factors that control metabolic pathway activity in yeast.	6	carbon utilization	1

File last modified: Thu Apr 25 2024
GSE24054.final.pcl	Unraveling condition-dependent networks of transcription factors that control metabolic pathway activity in yeast.	5	carbon utilization	1

File last modified: Thu Apr 25 2024

---


# Leadsham_2010_PMID_21108829

Citation: Leadsham JE, Gourlay CW. 2010. cAMP/PKA signaling balances respiratory activity with 
mitochondria dependent apoptosis via transcriptional regulation.

Full Description: BACKGROUND: Appropriate control of mitochondrial function, morphology and biogenesis are 
crucial determinants of the general health of eukaryotic cells. It is therefore imperative
that we understand the mechanisms that co-ordinate mitochondrial function with 
environmental signaling systems. The regulation of yeast mitochondrial function in 
response to nutritional change can be modulated by PKA activity. Unregulated PKA activity 
can lead to the production of mitochondria that are prone to the production of ROS, and an
apoptotic form of cell death. RESULTS: We present evidence that mitochondria are sensitive
to the level of cAMP/PKA signaling and can respond by modulating levels of respiratory 
activity or committing to self execution. The inappropriate activation of one of  the 
yeast PKA catalytic subunits, Tpk3p, is sufficient to commit cells to an apoptotic death 
through transcriptional changes that promote the production of dysfunctional, ROS 
producing mitochondria. Our data implies that cAMP/PKA regulation of mitochondrial 
function that promotes apoptosis engages the function of multiple transcription factors, 
including HAP4, SOK2 and SCO1. CONCLUSIONS: We propose that in yeast, as is the case in 
mammalian cells, mitochondrial function  and biogenesis are controlled in response to 
environmental change by the concerted regulation of multiple transcription factors. The 
visualization of cAMP/TPK3 induced cell death within yeast colonies supports a model that 
PKA regulation plays a physiological role in coordinating respiratory function and cell 
death with nutritional status in budding yeast.

PMID: 21108829

GEO ID: GSE25541.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE25541.final.pcl	cAMP/PKA signaling balances respiratory activity with mitochondria dependent apoptosis via transcriptional regulation.	8	organelles, biogenesis, structure, and function|oxidative stress	1

File last modified: Thu Apr 25 2024

---


# Li_2010_PMID_20439498

Citation: Li M, Petteys BJ, McClure JM, Valsakumar V, Bekiranov S, Frank EL, Smith JS. 2010. 
Thiamine biosynthesis in Saccharomyces cerevisiae is regulated by the NAD+-dependent 
histone deacetylase Hst1.

Full Description: Genes encoding thiamine biosynthesis enzymes in microorganisms are tightly regulated such 
that low environmental thiamine concentrations activate transcription and high 
concentrations are repressive. We have determined that multiple thiamine (THI) genes in 
Saccharomyces cerevisiae are also regulated by the intracellular NAD(+) concentration via 
the NAD(+)-dependent histone deacetylase (HDAC) Hst1 and, to a lesser extent, Sir2. Both 
of these HDACs associate with a distal region of the affected THI gene promoters that does
not overlap with a previously defined enhancer region bound by the thiamine-responsive 
Thi2/Thi3/Pdc2 transcriptional activators. The specificity of histone H3 and/or H4 
deacetylation carried out by Hst1 and Sir2 at the distal promoter region depends on the 
THI gene being tested. Hst1/Sir2-mediated repression of the THI genes occurs at the level 
of basal expression, thus representing the first set of transcription factors shown to 
actively repress this gene class. Importantly, lowering the NAD(+) concentration and 
inhibiting the Hst1/Sum1 HDAC complex elevated the intracellular thiamine concentration 
due  to increased thiamine biosynthesis and transport, implicating NAD(+) in the control 
of thiamine homeostasis.

PMID: 20439498

GEO ID: GSE18488.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE18488.final.pcl	Thiamine biosynthesis in Saccharomyces cerevisiae is regulated by the NAD+-dependent histone deacetylase Hst1.	14	cofactor metabolism	1

File last modified: Thu Apr 25 2024

---


# Lopez-Garcia_2010_PMID_21078184

Citation: Lopez-Garcia B, Gandia M, Munoz A, Carmona L, Marcos JF. 2010. A genomic approach 
highlights common and diverse effects and determinants of susceptibility on the yeast 
Saccharomyces cerevisiae exposed to distinct antimicrobial peptides.

Full Description: BACKGROUND: The mechanism of action of antimicrobial peptides (AMP) was initially 
correlated with peptide membrane permeation properties. However, recent evidences indicate
that action of a number of AMP is more complex and involves specific interactions at cell 
envelopes or with intracellular targets. In this study, a genomic approach was undertaken 
on the model yeast Saccharomyces cerevisiae to characterize the antifungal effect of two 
unrelated AMP. RESULTS: Two differentiated peptides were used: the synthetic cell-
penetrating PAF26 and the natural cytolytic melittin. Transcriptomic analyses demonstrated
distinctive gene expression changes for each peptide. Quantitative RT-PCR confirmed 
differential expression of selected genes. Gene Ontology (GO) annotation of differential 
gene  lists showed that the unique significant terms shared by treatment with both 
peptides were related to the cell wall (CW). Assays with mutants lacking CW-related genes 
including those of MAPK signaling pathways revealed genes having influence on sensitivity 
to peptides. Fluorescence microscopy and flow cytometry  demonstrated PAF26 interaction 
with cells and internalization that correlated with cell killing in sensitive CW-defective
mutants such as Deltaecm33 or Deltassd1. GO annotation also showed differential responses 
between peptides, which included ribosomal biogenesis, ARG genes from the metabolism of 
amino groups (specifically induced by PAF26), or the reaction to unfolded protein stress. 
Susceptibility of deletion mutants confirmed the involvement of these processes. 
Specifically, mutants lacking ARG genes from the metabolism of arginine pathway were 
markedly more resistant to PAF26 and had a functional CW. In the deletant in the 
arginosuccinate synthetase (ARG1) gene, PAF26 interaction  occurred normally, thus 
uncoupling peptide interaction from cell killing. The previously described involvement of 
the glycosphingolipid gene IPT1 was extended  to the peptides studied here. CONCLUSIONS: 
Reinforcement of CW is a general response common after exposure to distinct AMP, and 
likely contributes to shield  cells from peptide interaction. However, a weakened CW is 
not necessarily indicative of a higher sensitivity to AMP. Additional processes modulate 
susceptibility to specific peptides, exemplified in the involvement of the metabolism of 
amino groups in the case of PAF26. The relevance of the response to unfolded protein 
stress or the sphingolipid biosynthesis, previously reported for other unrelated AMP, was 
also independently confirmed.

PMID: 21078184

GEO ID: GSE25279.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE25279.final.pcl	A genomic approach highlights common and diverse effects and determinants of susceptibility on the yeast Saccharomyces cerevisiae exposed to distinct antimicrobial peptides.	9	stress	1

File last modified: Thu Apr 25 2024

---


# Ma_2010_PMID_21106074

Citation: Ma M, Liu ZL. 2010. Comparative transcriptome profiling analyses during the lag phase 
uncover YAP1, PDR1, PDR3, RPN4, and HSF1 as key regulatory genes in genomic adaptation to 
the lignocellulose derived inhibitor HMF for Saccharomyces cerevisiae.

Full Description: BACKGROUND: The yeast Saccharomyces cerevisiae is able to adapt and in situ detoxify 
lignocellulose derived inhibitors such as furfural and HMF. The length of lag phase for 
cell growth in response to the inhibitor challenge has been used to measure tolerance of 
strain performance. Mechanisms of yeast tolerance at the  genome level remain unknown. 
Using systems biology approach, this study investigated comparative transcriptome 
profiling, metabolic profiling, cell growth response, and gene regulatory interactions of 
yeast strains and selective  gene deletion mutations in response to HMF challenges during 
the lag phase of growth. RESULTS: We identified 365 candidate genes and found at least 3 
significant components involving some of these genes that enable yeast adaptation and 
tolerance to HMF in yeast. First, functional enzyme coding genes such as ARI1, ADH6, ADH7
, and OYE3, as well as gene interactions involved in the biotransformation and inhibitor 
detoxification were the direct driving force to reduce HMF damages in cells. Expressions 
of these genes were regulated by YAP1 and its closely related regulons. Second, a large 
number of PDR genes, mainly regulated by PDR1 and PDR3, were induced during the lag phase 
and the PDR gene family-centered functions, including specific and multiple functions 
involving cellular transport such as TPO1, TPO4, RSB1, PDR5, PDR15, YOR1, and SNQ2, 
promoted cellular adaptation and survival in order to cope with the inhibitor stress. 
Third, expressed genes involving degradation of damaged proteins and protein modifications
such as SHP1 and SSA4, regulated by RPN4, HSF1, and other co-regulators, were necessary 
for yeast cells to survive and adapt the HMF stress. A deletion mutation strain Deltarpn4 
was unable to recover the growth in  the presence of HMF. CONCLUSIONS: Complex gene 
interactions and regulatory networks as well as co-regulations exist in yeast adaptation 
and tolerance to the lignocellulose derived inhibitor HMF. Both induced and repressed 
genes involving  diversified functional categories are accountable for adaptation and 
energy rebalancing in yeast to survive and adapt the HMF stress during the lag phase of  
growth. Transcription factor genes YAP1, PDR1, PDR3, RPN4, and HSF1 appeared to play key 
regulatory rules for global adaptation in the yeast S. cerevisiae.

PMID: 21106074

GEO ID: GSE22939.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE22939.final.pcl	Comparative transcriptome profiling analyses during the lag phase uncover YAP1, PDR1, PDR3, RPN4, and HSF1 as key regulatory genes in genomic adaptation to the lignocellulose derived inhibitor HMF for Saccharomyces cerevisiae.	14	stress|chemical stimulus	2

File last modified: Thu Apr 25 2024

---


# Mendes-Ferreira_2010_PMID_20952643

Citation: Mendes-Ferreira A, Sampaio-Marques B, Barbosa C, Rodrigues F, Costa V, Mendes-Faia A, 
Ludovico P, Leao C. 2010. Accumulation of non-superoxide anion reactive oxygen species 
mediates nitrogen-limited alcoholic fermentation by Saccharomyces cerevisiae.

Full Description: Throughout alcoholic fermentation, nitrogen depletion is one of the most important 
environmental stresses that can negatively affect the yeast metabolic activity and 
ultimately leads to fermentation arrest. Thus, the identification of the underlying 
effects and biomarkers of nitrogen limitation is valuable for controlling, and therefore 
optimizing, alcoholic fermentation. In this study, reactive oxygen species (ROS), plasma 
membrane integrity, and cell cycle were evaluated in a wine strain of Saccharomyces 
cerevisiae during alcoholic fermentation in nitrogen-limiting medium under anaerobic 
conditions. The results  indicated that nitrogen limitation leads to an increase in ROS 
and that the superoxide anion is a minor component of the ROS, but there is increased 
activity of both Sod2p and Cta1p. Associated with these effects was a decrease in plasma 
membrane integrity and a persistent cell cycle arrest at G(0)/G(1) phases. Moreover, under
these conditions it appears that autophagy, evaluated by ATG8 expression, is induced, 
suggesting that this mechanism is essential for cell survival but does not prevent the 
cell cycle arrest observed in slow fermentation. Conversely, nitrogen refeeding allowed 
cells to reenter cell cycle  by decreasing ROS generation and autophagy. Altogether, the 
results provide new insights on the understanding of wine fermentations under nitrogen-
limiting conditions and further indicate that ROS accumulation, evaluated by the 
MitoTracker Red dye CM-H(2)XRos, and plasma membrane integrity could be useful as 
predictive markers of fermentation problems.

PMID: 20952643

GEO ID: GSE5837.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE5837.final.pcl	Accumulation of non-superoxide anion reactive oxygen species mediates nitrogen-limited alcoholic fermentation by Saccharomyces cerevisiae.	15	fermentation|nitrogen utilization	1

File last modified: Thu Apr 25 2024
GSE5835.final.pcl	Accumulation of non-superoxide anion reactive oxygen species mediates nitrogen-limited alcoholic fermentation by Saccharomyces cerevisiae.	14	fermentation|nitrogen utilization	1

File last modified: Thu Apr 25 2024
GSE5836_s9.final.pcl	Accumulation of non-superoxide anion reactive oxygen species mediates nitrogen-limited alcoholic fermentation by Saccharomyces cerevisiae.	9	fermentation|nitrogen utilization	1

File last modified: Thu Apr 25 2024

---


# Morillo-Huesca_2010_PMID_20711347

Citation: Morillo-Huesca M, Clemente-Ruiz M, Andujar E, Prado F. 2010. The SWR1 histone replacement 
complex causes genetic instability and genome-wide transcription misregulation in the 
absence of H2A.Z.

Full Description: The SWR1 complex replaces the canonical histone H2A with the variant H2A.Z (Htz1  in yeast
) at specific chromatin regions. This dynamic alteration in nucleosome structure provides 
a molecular mechanism to regulate transcription, gene silencing, chromosome segregation 
and DNA repair. Here we show that genetic instability, sensitivity to drugs impairing 
different cellular processes and genome-wide transcriptional misregulation in htz1Delta 
can be partially or totally suppressed if SWR1 is not formed (swr1Delta), if it forms but 
cannot bind to chromatin (swc2Delta) or if it binds to chromatin but lacks histone 
replacement activity (swc5Delta and the ATPase-dead swr1-K727G). These results suggest 
that in htz1Delta the nucleosome remodelling activity of SWR1 affects chromatin integrity 
because of an attempt to replace H2A with Htz1 in the absence of the latter. This would 
impair transcription and, either directly or indirectly, other cellular processes. 
Specifically, we show that in htz1Delta, the SWR1 complex causes an accumulation of 
recombinogenic DNA damage by a mechanism dependent on phosphorylation of H2A at Ser129, a 
modification that occurs in response to DNA damage, suggesting that the SWR1 complex 
impairs the repair of spontaneous DNA damage in htz1Delta. In addition, SWR1 causes DSBs 
sensitivity in htz1Delta; consistently, in the absence of Htz1 the SWR1 complex bound near
an endonuclease HO-induced DSB at the mating-type (MAT) locus impairs  DSB-induced 
checkpoint activation. Our results support a stepwise mechanism for the replacement of H2A
with Htz1 and demonstrate that a tight control of this mechanism is essential to regulate 
chromatin dynamics but also to prevent the deleterious consequences of an incomplete 
nucleosome remodelling.

PMID: 20711347

GEO ID: GSE21571

PCL filename	short description	# conditions	tags	# channels
GSE21571_final.pcl	The SWR1 histone replacement complex causes genetic instability and genome-wide transcription misregulation in the absence of H2A.Z.	24	chromatin organization|transcription	1

File last modified: Thu Apr 25 2024

---


# Munding_2010_PMID_21123654

Citation: Munding EM, Igel AH, Shiue L, Dorighi KM, Trevino LR, Ares M Jr. 2010. Integration of a 
splicing regulatory network within the meiotic gene expression program of Saccharomyces 
cerevisiae.

Full Description: Splicing regulatory networks are essential components of eukaryotic gene expression 
programs, yet little is known about how they are integrated with transcriptional 
regulatory networks into coherent gene expression programs. Here  we define the MER1 
splicing regulatory network and examine its role in the gene expression program during 
meiosis in budding yeast. Mer1p splicing factor promotes splicing of just four pre-mRNAs. 
All four Mer1p-responsive genes also require Nam8p for splicing activation by Mer1p; 
however, other genes require Nam8p but not Mer1p, exposing an overlapping meiotic splicing
network controlled  by Nam8p. MER1 mRNA and three of the four Mer1p substrate pre-mRNAs 
are induced by the transcriptional regulator Ume6p. This unusual arrangement delays 
expression of Mer1p-responsive genes relative to other genes under Ume6p control. Products
of Mer1p-responsive genes are required for initiating and completing recombination and for
activation of Ndt80p, the activator of the transcriptional  network required for 
subsequent steps in the program. Thus, the MER1 splicing regulatory network mediates the 
dependent relationship between the UME6 and NDT80 transcriptional regulatory networks in 
the meiotic gene expression program. This  study reveals how splicing regulatory networks 
can be interlaced with transcriptional regulatory networks in eukaryotic gene expression 
programs.

PMID: 21123654

GEO ID: GSE24675

PCL filename	short description	# conditions	tags	# channels
GSE24675_final.pcl	Integration of a splicing regulatory network within the meiotic gene expression program of Saccharomyces cerevisiae.	34	mRNA processing|sporulation	2

File last modified: Thu Apr 25 2024
GSE24681_final.pcl	Integration of a splicing regulatory network within the meiotic gene expression program of Saccharomyces cerevisiae.	14	mRNA processing|sporulation	2

File last modified: Thu Apr 25 2024

---


# Otero_2010_PMID_21176163

Citation: Otero JM, Vongsangnak W, Asadollahi MA, Olivares-Hernandes R, Maury J, Farinelli L, 
Barlocher L, Osteras M, Schalk M, Clark A, Nielsen J. 2010. Whole genome sequencing of 
Saccharomyces cerevisiae: from genotype to phenotype for improved metabolic engineering 
applications.

Full Description: BACKGROUND: The need for rapid and efficient microbial cell factory design and 
construction are possible through the enabling technology, metabolic engineering, which is
now being facilitated by systems biology approaches. Metabolic engineering is often 
complimented by directed evolution, where selective pressure is applied to a partially 
genetically engineered strain to confer a desirable phenotype. The exact genetic 
modification or resulting genotype that leads to the improved phenotype is often not 
identified or understood to enable further metabolic engineering. RESULTS: In this work we
performed whole genome high-throughput sequencing and annotation can be used to identify 
single nucleotide polymorphisms (SNPs) between Saccharomyces cerevisiae strains S288c and 
CEN.PK113-7D. The yeast strain S288c was the first eukaryote sequenced, serving as the 
reference genome for the Saccharomyces Genome Database, while CEN.PK113-7D is a preferred 
laboratory strain for industrial biotechnology research. A total of 13,787 high-quality 
SNPs were detected between both strains  (reference strain: S288c). Considering only 
metabolic genes (782 of 5,596 annotated genes), a total of 219 metabolism specific SNPs 
are distributed across  158 metabolic genes, with 85 of the SNPs being nonsynonymous (e.g
., encoding amino acid modifications). Amongst metabolic SNPs detected, there was pathway 
enrichment in the galactose uptake pathway (GAL1, GAL10) and ergosterol biosynthetic 
pathway (ERG8, ERG9). Physiological characterization confirmed a strong deficiency in 
galactose uptake and metabolism in S288c compared to CEN.PK113-7D, and similarly, 
ergosterol content in CEN.PK113-7D was significantly higher in both glucose and galactose 
supplemented cultivations compared to S288c. Furthermore, DNA microarray profiling of 
S288c and CEN.PK113-7D in both glucose and galactose batch cultures did not provide a 
clear hypothesis for major phenotypes observed, suggesting that genotype to phenotype 
correlations are manifested post-transcriptionally or post-translationally either through 
protein  concentration and/or function. CONCLUSIONS: With an intensifying need for 
microbial cell factories that produce a wide array of target compounds, whole genome high-
throughput sequencing and annotation for SNP detection can aid in better reducing and 
defining the metabolic landscape. This work demonstrates direct correlations between 
genotype and phenotype that provides clear and high-probability of success metabolic 
engineering targets. The genome sequence, annotation, and a SNP viewer of CEN.PK113-7D are
deposited at http://www.sysbio.se/cenpk.

PMID: 21176163

GEO ID: GSE21479.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE21479.final.pcl	Whole genome sequencing of Saccharomyces cerevisiae: from genotype to phenotype for improved metabolic engineering applications.	8	carbon utilization|synthetic biology	1

File last modified: Thu Apr 25 2024

---


# Papini_2010_PMID_20815084

Citation: Papini M, Nookaew I, Scalcinati G, Siewers V, Nielsen J. 2010. Phosphoglycerate mutase 
knock-out mutant Saccharomyces cerevisiae: physiological  investigation and transcriptome 
analysis.

Full Description: The yeast Saccharomyces cerevisiae is able to adapt its metabolism to grow on different 
carbon sources and to shift to non-fermentative growth on C2 or C3 carbon sources (ethanol
, acetate, or glycerol) through the activation of gluconeogenesis. Here, we studied the 
response to the deletion of the glycolytic  and gluconeogenic gene GPM1, encoding for 
phosphoglycerate mutase. It was previously shown that a S. cerevisiae strain with non-
functional copies of GPM1 can only grow when glycerol and ethanol are both present as 
carbon sources, whilst addition of glucose was shown to strongly inhibit growth. It was 
suggested that glycerol is needed to feed gluconeogenesis whilst ethanol is required for 
respiration. Here, we studied the physiological response of the GPM1 knock-out mutant 
through fermentation and transcriptome analysis. Furthermore, we compared  the 
physiological results with those obtained through simulations using a genome-scale 
metabolic model, showing that glycerol is only needed in small amounts for growth. Our 
findings strongly suggest a severely impaired growth ability of the knock-out mutant, 
which presents increased transcript levels of genes involved in the pentose phosphate 
pathway and in the glyoxylate shunt. These results indicate an attempt to compensate for 
the energy imbalance caused by the deletion of the glycolytic/gluconeogenic gene within 
the mutant.

PMID: 20815084

GEO ID: GSE22122

PCL filename	short description	# conditions	tags	# channels
GSE22122_final.pcl	Phosphoglycerate mutase knock-out mutant Saccharomyces cerevisiae: physiological  investigation and transcriptome analysis.	6	carbon utilization|respiration|fermentation	1

File last modified: Thu Apr 25 2024

---


# Perez-Valle_2010_PMID_20952580

Citation: Perez-Valle J, Rothe J, Primo C, Martinez Pastor M, Arino J, Pascual-Ahuir A, Mulet JM, 
Serrano R, Yenush L. 2010. Hal4 and Hal5 protein kinases are required for general control 
of carbon and nitrogen uptake and metabolism.

Full Description: The yeast protein kinases Sat4/Hal4 and Hal5 are required for the plasma membrane 
stability of the K(+) transporter Trk1 and some amino acid and glucose permeases. The 
transcriptomic analysis presented here indicates alterations in the general control of the
metabolism of both nitrogen and carbon. Accordingly, we observed reduced uptake of 
methionine and leucine in the hal4 hal5 mutant. This decrease correlates with activation 
of the Gcn2-Gcn4 pathway, as measured by expression of the lacZ gene under the control of 
the GCN4 promoter. However, with the exception of methionine biosynthetic genes, few amino
acid biosynthetic genes are induced in the hal4 hal5 mutant, whereas several genes 
involved in amino acid catabolism  are repressed. Concerning glucose metabolism, we found 
that this mutant exhibits  derepression of respiratory genes in the presence of glucose, 
leading to an increased activity of mitochondrial enzymes, as measured by succinate 
dehydrogenase (SDH) activity. In addition, the reduced glucose consumption in the hal4 
hal5 mutant correlates with a more acidic intracellular pH and with low activity of the 
plasma membrane H(+)-ATPase. As a compensatory mechanism for the  low glycolytic rate, the
hal4 hal5 mutant overexpresses the HXT4 high-affinity glucose transporter and the 
hexokinase genes. These results indicate that the hal4 hal5 mutant presents defects in the
general control of nitrogen and carbon metabolism, which correlate with reduced transport 
of amino acids and glucose, respectively. A more acidic intracellular pH may contribute to
some defects of this mutant.

PMID: 20952580

GEO ID: GSE22976.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE22976.final.pcl	Hal4 and Hal5 protein kinases are required for general control of carbon and nitrogen uptake and metabolism.	12	oxidative stress	2

File last modified: Thu Apr 25 2024

---


# Rivera-Ruiz_2010_PMID_21126371

Citation: Rivera-Ruiz ME, Rodriguez-Quinones JF, Akamine P, Rodriguez-Medina JR. 2010. Post-
transcriptional regulation in the myo1Delta mutant of Saccharomyces cerevisiae.

Full Description: BACKGROUND: Saccharomyces cerevisiae myosin type II-deficient (myo1Delta) strains remain 
viable and divide, despite the absence of a cytokinetic ring, by activation of the PKC1-
dependent cell wall integrity pathway (CWIP). Since the myo1Delta transcriptional 
fingerprint is a subset of the CWIP fingerprint, the myo1Delta strain may provide a 
simplified paradigm for cell wall stress survival. RESULTS: To explore the post-
transcriptional regulation of the myo1Delta stress response, 1,301 differentially 
regulated ribosome-bound mRNAs were identified by  microarray analysis of which 204 were 
co-regulated by transcription and translation. Four categories of mRNA were significantly 
affected - protein biosynthesis, metabolism, carbohydrate metabolism, and unknown 
functions. Nine genes of the 20 CWIP fingerprint genes were post-transcriptionally 
regulated. Down and up regulation of selected ribosomal protein and cell wall biosynthesis
mRNAs was validated by their distribution in polysomes from wild type and myo1Delta 
strains. Western blot analysis revealed accumulation of the phosphorylated form of 
eukaryotic translation initiation factor 2 (eIF2alpha-P) and a reduction in the steady 
state levels of the translation initiation factor eIF4Gp in myo1Delta strains. Deletion of
GCN2 in myo1Delta abolished eIF2alphap phosphorylation, and showed a severe growth defect
. The presence of P-bodies in myo1Delta strains suggests that the process of mRNA 
sequestration is active, however, the three representative down regulated RP mRNAs, RPS8A
, RPL3 and RPL7B  were present at equivalent levels in Dcp2p-mCh-positive 
immunoprecipitated fractions from myo1Delta and wild type cells. These same RP mRNAs were 
also selectively co-precipitated with eIF2alpha-P in myo1Delta strains. CONCLUSIONS: 
Quantitative analysis of ribosome-associated mRNAs and their polyribosome distributions 
suggests selective regulation of mRNA translation efficiency in myo1Delta strains. 
Inhibition of translation initiation factor eIF2alpha (eIF2alpha-P) in these strains was 
by Gcn2p-dependent phosphorylation. The increase in the levels of eIF2alpha-P; the genetic
interaction between GCN2 and MYO1; and the reduced levels of eIF4Gp suggest that other 
signaling pathways, in  addition to the CWIP, may be important for myo1Delta strain 
survival. Selective co-immunoprecipitation of RP mRNAs with eIF2alpha-P in myo1Delta 
strains suggests a novel mode of translational regulation. These results indicate that 
post-transcriptional control is important in the myo1Delta stress response and possibly 
other stresses in yeast.

PMID: 21126371

GEO ID: GSE20203.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE20203.final.pcl	Post-transcriptional regulation in the myo1Delta mutant of Saccharomyces cerevisiae.	4	stress|cytoskeleton and molecular motors	2

File last modified: Thu Apr 25 2024

---


# Rodriguez-Colman_2010_PMID_20847055

Citation: Rodriguez-Colman MJ, Reverter-Branchat G, Sorolla MA, Tamarit J, Ros J, Cabiscol E. 2010. 
The forkhead transcription factor Hcm1 promotes mitochondrial biogenesis and stress 
resistance in yeast.

Full Description: In Saccharomyces cerevisiae, the forkhead transcription factor Hcm1 is involved in 
chromosome segregation, spindle pole dynamics, and budding. We found that Hcm1 interacts 
with the histone deacetylase Sir2 and shifts from cytoplasm to the nucleus in the G(1)/S 
phase or in response to oxidative stress stimuli. The nuclear localization of Hcm1 depends
on the activity of Sir2 as revealed by activators and inhibitors of the sirtuins and the 
Deltasir2 mutant. Hcm1-overexpressing cells display more mitochondria that can be 
attributed to increased amounts of Abf2, a protein involved in mitochondrial biogenesis. 
These  cells also show higher rates of oxygen consumption and improved resistance to 
oxidative stress that would be explained by increased catalase and Sod2 activities and 
molecular chaperones such as Hsp26, Hsp30, and members of Hsp70 family. Microarray 
analyses also reveal increased expression of genes involved in mitochondrial energy 
pathways and those allowing the transition from the exponential to the stationary phase. 
Taken together, these results describe a new and relevant role of Hcm1 for mitochondrial 
functions, suggesting that this transcription factor would participate in the adaptation 
of cells from fermentative to respiratory metabolism.

PMID: 20847055

GEO ID: GSE20420.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE20420.final.pcl	The forkhead transcription factor Hcm1 promotes mitochondrial biogenesis and stress resistance in yeast.	12	organelles, biogenesis, structure, and function|oxidative stress	2

File last modified: Thu Apr 25 2024

---


# Sahi_2010_PMID_19822657

Citation: Sahi C, Lee T, Inada M, Pleiss JA, Craig EA. 2010. Cwc23, an essential J protein critical 
for pre-mRNA splicing with a dispensable J domain.

Full Description: J proteins are structurally diverse, obligatory cochaperones of Hsp70s, each with a highly
conserved J domain that plays a critical role in the stimulation of Hsp70's ATPase 
activity. The essential protein, Cwc23, is one of 13 J proteins found in the cytosol and/
or nucleus of Saccharomyces cerevisiae. We report that a partial loss-of-function CWC23 
mutant has severe, global defects in pre-mRNA splicing. This mutation leads to 
accumulation of the excised, lariat form of the  intron, as well as unspliced pre-mRNA, 
suggesting a role for Cwc23 in spliceosome disassembly. Such a role is further supported 
by the observation that this mutation results in reduced interaction between Cwc23 and 
Ntr1 (SPP382), a known  component of the disassembly pathway. However, Cwc23 is a very 
atypical J protein. Its J domain, although functional, is dispensable for both cell 
viability and pre-mRNA splicing. Nevertheless, strong genetic interactions were uncovered 
between point mutations encoding alterations in Cwc23's J domain and either Ntr1 or Prp43
, a DExD/H-box helicase essential for spliceosome disassembly. These genetic interactions 
suggest that Hsp70-based chaperone machinery does play a role in the disassembly process. 
Cwc23 provides a unique example of a J protein; its partnership with Hsp70 plays an 
auxiliary, rather than a central, role in its essential cellular function.

PMID: 19822657

GEO ID: GSE18485.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE18485.final.pcl	Cwc23, an essential J protein critical for pre-mRNA splicing with a dispensable J domain.	8	mRNA processing	2

File last modified: Thu Apr 25 2024

---


# Shakoury-Elizeh_2010_PMID_20231268

Citation: Shakoury-Elizeh M, Protchenko O, Berger A, Cox J, Gable K, Dunn TM, Prinz WA, Bard M, 
Philpott CC. 2010. Metabolic response to iron deficiency in Saccharomyces cerevisiae.

Full Description: Iron is an essential cofactor for enzymes involved in numerous cellular processes, yet 
little is known about the impact of iron deficiency on cellular metabolism or iron 
proteins. Previous studies have focused on changes in transcript and proteins levels in 
iron-deficient cells, yet these changes may not reflect changes in transport activity or 
flux through a metabolic pathway. We analyzed the metabolomes and transcriptomes of yeast 
grown in iron-rich and iron-poor media to determine which biosynthetic processes are 
altered when iron availability falls. Iron deficiency led to changes in glucose metabolism
, amino acid biosynthesis, and lipid biosynthesis that were due to deficiencies in 
specific iron-dependent enzymes. Iron-sulfur proteins exhibited loss of iron cofactors, 
yet amino acid synthesis was maintained. Ergosterol and sphingolipid biosynthetic pathways
had blocks at points where heme and diiron enzymes function, whereas Ole1, the essential 
fatty acid desaturase, was resistant to iron depletion. Iron-deficient cells exhibited 
depletion of most iron enzyme activities, but loss of activity during iron deficiency did 
not consistently disrupt metabolism. Amino acid homeostasis was robust, but iron 
deficiency impaired lipid synthesis, altering the properties and functions of cellular 
membranes.

PMID: 20231268

GEO ID: GSE19016.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE19016.final.pcl	Metabolic response to iron deficiency in Saccharomyces cerevisiae.	8	metal or metalloid ion stress	1

File last modified: Thu Apr 25 2024

---


# Simola_2010_PMID_20969771

Citation: Simola DF, Francis C, Sniegowski PD, Kim J. 2010. Heterochronic evolution reveals modular 
timing changes in budding yeast transcriptomes.

Full Description: BACKGROUND: Gene expression is a dynamic trait, and the evolution of gene regulation can 
dramatically alter the timing of gene expression without greatly affecting mean expression
levels. Moreover, modules of co-regulated genes may exhibit coordinated shifts in 
expression timing patterns during evolutionary divergence. Here, we examined transcriptome
evolution in the dynamical context of the budding yeast cell-division cycle, to 
investigate the extent of divergence in expression timing and the regulatory architecture 
underlying timing evolution. RESULTS: Using a custom microarray platform, we obtained 378 
measurements for 6,263 genes over 18 timepoints of the cell-division cycle in nine strains
of S. cerevisiae and one strain of S. paradoxus. Most genes show significant divergence in
expression dynamics at all scales of transcriptome organization, suggesting broad 
potential for timing changes. A model test comparing expression level evolution versus 
timing evolution revealed a better fit with timing evolution for 82% of genes. Analysis of
shared patterns of timing evolution suggests the existence of seven dynamically-autonomous
modules, each of which shows coherent evolutionary timing changes. Analysis of 
transcription factors associated with these gene modules suggests a modular pleiotropic 
source of divergence in expression timing. CONCLUSIONS: We propose that transcriptome 
evolution may generally entail changes in timing (heterochrony) rather than changes in 
levels (heterometry) of expression. Evolution of gene expression dynamics may involve 
modular changes in timing control mediated by module-specific transcription factors. We 
hypothesize that genome-wide gene regulation may utilize a general architecture comprised 
of multiple semi-autonomous event timelines, whose superposition could produce 
combinatorial complexity in timing control patterns.

PMID: 20969771

GEO ID: GSE24771.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE24771.final.pcl	Heterochronic evolution reveals modular timing changes in budding yeast transcriptomes.	173	evolution|transcriptional regulation	2

File last modified: Thu Apr 25 2024
GSE24770.final.pcl	Heterochronic evolution reveals modular timing changes in budding yeast transcriptomes.	23	evolution|transcriptional regulation	2

File last modified: Thu Apr 25 2024

---


# Stanley_2010_PMID_20393705

Citation: Stanley D, Fraser S, Stanley GA, Chambers PJ. 2010. Retrotransposon expression in ethanol-
stressed Saccharomyces cerevisiae.

Full Description: There are five retrotransposon families in Saccharomyces cerevisiae, three (Ty1,  Ty2, and
Ty3) of which are known to be transcriptionally active. Early investigations reported 
yeast retrotransposons to be stress-induced; however, microarray-based studies do not 
report retrotransposition-related Gene Ontology (GO) categories in the ethanol stress 
response of S. cerevisiae. In this study, microarray technology was used to investigate 
the ethanol stress response of S. cerevisiae W303-1A, and the highest stress-induced GO 
categories, based on z-score, were found to be retrotransposition-related, namely, 
Retrotransposition  Nucleocapsid and Transposition, RNA-Mediated. Further investigation, 
involving reanalysis of previously published results on the stress response of S. 
cerevisiae, identified the absence of annotation for retrotransposon genes and associated 
GO categories and their omission during the printing of spotted arrays as two reasons why 
these categories in previous gene expression studies on the ethanol stress response of 
yeast were not reported.

PMID: 20393705

GEO ID: GSE20590.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE20590.final.pcl	Retrotransposon expression in ethanol-stressed Saccharomyces cerevisiae.	6	diauxic shift|stress	2

File last modified: Thu Apr 25 2024

---


# Stanley_2010_PMID_20661734

Citation: Stanley D, Chambers PJ, Stanley GA, Borneman A, Fraser S. 2010. Transcriptional changes 
associated with ethanol tolerance in Saccharomyces cerevisiae.

Full Description: Saccharomyces spp. are widely used for ethanol production; however, fermentation  
productivity is negatively affected by the impact of ethanol accumulation on yeast 
metabolic rate and viability. This study used microarray and statistical two-way ANOVA 
analysis to compare and evaluate gene expression profiles of two previously generated 
ethanol-tolerant mutants, CM1 and SM1, with their parent, Saccharomyces cerevisiae W303-1A
, in the presence and absence of ethanol stress.  Although sharing the same parentage, the
mutants were created differently: SM1 by adaptive evolution involving long-term exposure 
to ethanol stress and CM1 using chemical mutagenesis followed by adaptive evolution-based 
screening. Compared to  the parent, differences in the expression levels of genes 
associated with a number of gene ontology categories in the mutants suggest that their 
improved ethanol stress response is a consequence of increased mitochondrial and NADH 
oxidation activities, stimulating glycolysis and other energy-yielding pathways.  This 
leads to increased activity of energy-demanding processes associated with the production 
of proteins and plasma membrane components, which are necessary for acclimation to ethanol
stress. It is suggested that a key function of the ethanol stress response is restoration 
of the NAD(+)/NADH redox balance, which increases glyceraldehyde-3-phosphate dehydrogenase
activity, and higher glycolytic flux in the ethanol-stressed cell. Both mutants achieved 
this by a constitutive increase in carbon flux in the glycerol pathway as a means of 
increasing NADH oxidation.

PMID: 20661734

GEO ID: GSE21120.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE21120.final.pcl	Transcriptional changes associated with ethanol tolerance in Saccharomyces cerevisiae.	18	fermentation|stress|carbon utilization	2

File last modified: Thu Apr 25 2024

---


# Stefanini_2010_PMID_20501664

Citation: Stefanini I, Trabocchi A, Marchi E, Guarna A, Cavalieri D. 2010. A systems biology 
approach to dissection of the effects of small bicyclic peptidomimetics on a panel of 
saccharomyces cerevisiae mutants.

Full Description: In recent years, an approach called chemical genetics has been adopted in drug  research 
to discover and validate new targets and to identify and optimize leads  by high 
throughput screening. In this work, we tested the ability of a library of small 
peptidomimetics to induce phenotypic effects with functional implications on a panel of 
strains of the budding yeast Saccharomyces cerevisiae, both wild type and mutants, for 
respiratory function and multidrug resistance. Further elucidation of the function of 
these peptidomimetics was assessed by testing the  effects of the compound with the most 
prominent inhibitory activity, 089, on gene expression using DNA microarrays. Pathway 
analysis showed the involvement of such a molecule in inducing oxidative damage through 
alterations in mitochondrial functions. Transcriptional experiments were confirmed by 
increased levels of ROS  and activation of mitochondrial membrane potential. Our results 
demonstrate the influence of a functional HAP1 gene in the performance of S. cerevisiae as
a model system.

PMID: 20501664

GEO ID: GSE19331.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE19331.final.pcl	A systems biology approach to dissection of the effects of small bicyclic peptidomimetics on a panel of saccharomyces cerevisiae mutants.	6	chemical stimulus|respiration	2

File last modified: Thu Apr 25 2024

---


# Talarek_2010_PMID_20471941

Citation: Talarek N, Cameroni E, Jaquenoud M, Luo X, Bontron S, Lippman S, Devgan G, Snyder M, 
Broach JR, De Virgilio C. 2010. Initiation of the TORC1-regulated G0 program requires Igo1
/2, which license specific mRNAs to evade degradation via the 5'-3' mRNA decay pathway.

Full Description: Eukaryotic cell proliferation is controlled by growth factors and essential nutrients, in 
the absence of which cells may enter into a quiescent (G(0)) state. In yeast, nitrogen and
/or carbon limitation causes downregulation of the conserved TORC1 and PKA signaling 
pathways and, consequently, activation of the PAS kinase Rim15, which orchestrates G(0) 
program initiation and ensures proper life span by controlling distal readouts, including 
the expression of specific genes. Here, we report that Rim15 coordinates transcription 
with posttranscriptional mRNA protection by phosphorylating the paralogous Igo1 and Igo2 
proteins. This event, which stimulates Igo proteins to associate with the mRNA decapping 
activator Dhh1, shelters newly expressed mRNAs from degradation via the 5'-3' mRNA decay 
pathway, thereby enabling their proper translation during initiation of the G(0) program. 
These results delineate a likely conserved mechanism by which nutrient limitation leads to
stabilization of specific mRNAs that are critical for cell differentiation and life span.

PMID: 20471941

GEO ID: GSE20539.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE20539.final.pcl	Initiation of the TORC1-regulated G0 program requires Igo1/2, which license specific mRNAs to evade degradation via the 5'-3' mRNA decay pathway.	14	signaling|chemical stimulus|cell cycle regulation|cell aging	2

File last modified: Thu Apr 25 2024

---


# Verwaal_2010_PMID_20632327

Citation: Verwaal R, Jiang Y, Wang J, Daran JM, Sandmann G, van den Berg JA, van Ooyen AJ. 2010. 
Heterologous carotenoid production in Saccharomyces cerevisiae induces the pleiotropic 
drug resistance stress response.

Full Description: To obtain insight into the genome-wide transcriptional response of heterologous carotenoid
production in Saccharomyces cerevisiae, the transcriptome of two different S. cerevisiae 
strains overexpressing carotenogenic genes from the yeast Xanthophyllomyces dendrorhous 
grown in carbon-limited chemostat cultures was analysed. The strains exhibited different 
absolute carotenoid levels as well as different intermediate profiles. These discrepancies
were further sustained by the difference of the transcriptional response exhibited by the 
two strains. Transcriptome analysis of the strain producing high carotenoid levels 
resulted in specific induction of genes involved in pleiotropic drug resistance (PDR). 
These  genes encode ABC-type and major facilitator transporters which are reported to be 
involved in secretion of toxic compounds out of cells. beta-Carotene was found to be 
secreted when sunflower oil was added to the medium of S. cerevisiae cells producing high 
levels of carotenoids, which was not observed when added to X. dendrorhous cells. Deletion
of pdr10, one of the induced ABC transporters, decreased the transformation efficiency of 
a plasmid containing carotenogenic genes. The few transformants that were obtained had 
decreased growth rates and lower carotenoid production levels compared to a pdr5 deletion 
and a reference strain transformed with the same genes. Our results suggest that 
production of high amounts of carotenoids in S. cerevisiae leads to membrane stress, in 
which Pdr10 might play an important role, and a cellular response to secrete carotenoids 
out of the cell.

PMID: 20632327

GEO ID: GSE8451.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE8451.final.pcl	Heterologous carotenoid production in Saccharomyces cerevisiae induces the pleiotropic drug resistance stress response.	14	metabolism|chemical stimulus|stress	1

File last modified: Thu Apr 25 2024

---


# Wenger_2010_PMID_20485559

Citation: Wenger JW, Schwartz K, Sherlock G. 2010. Bulk segregant analysis by high-throughput 
sequencing reveals a novel xylose utilization gene from Saccharomyces cerevisiae.

Full Description: Fermentation of xylose is a fundamental requirement for the efficient production  of 
ethanol from lignocellulosic biomass sources. Although they aggressively ferment hexoses, 
it has long been thought that native Saccharomyces cerevisiae strains cannot grow 
fermentatively or non-fermentatively on xylose. Population surveys have uncovered a few 
naturally occurring strains that are weakly xylose-positive, and some S. cerevisiae have 
been genetically engineered to ferment xylose, but no strain, either natural or engineered
, has yet been reported to ferment xylose as efficiently as glucose. Here, we used a 
medium-throughput screen to identify Saccharomyces strains that can increase in optical 
density when xylose is presented as the sole carbon source. We identified 38 strains that 
have this xylose utilization phenotype, including strains of S. cerevisiae, other sensu 
stricto members, and hybrids between them. All the S. cerevisiae xylose-utilizing strains 
we identified are wine yeasts, and for those  that could produce meiotic progeny, the 
xylose phenotype segregates as a single gene trait. We mapped this gene by Bulk Segregant 
Analysis (BSA) using tiling microarrays and high-throughput sequencing. The gene is a 
putative xylitol dehydrogenase, which we name XDH1, and is located in the subtelomeric 
region of the right end of chromosome XV in a region not present in the S288c reference 
genome. We further characterized the xylose phenotype by performing gene expression 
microarrays and by genetically dissecting the endogenous Saccharomyces xylose pathway. We 
have demonstrated that natural S. cerevisiae yeasts are capable of utilizing xylose as the
sole carbon source, characterized the genetic  basis for this trait as well as the 
endogenous xylose utilization pathway, and demonstrated the feasibility of BSA using high-
throughput sequencing.

PMID: 20485559

GEO ID: GSE19116.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE19116.final.pcl	Bulk segregant analysis by high-throughput sequencing reveals a novel xylose utilization gene from Saccharomyces cerevisiae.	135	carbon utilization|fermentation	2

File last modified: Thu Apr 25 2024

---


# Wisselink_2010_PMID_20816840

Citation: Wisselink HW, Cipollina C, Oud B, Crimi B, Heijnen JJ, Pronk JT, van Maris AJ. 2010. 
Metabolome, transcriptome and metabolic flux analysis of arabinose fermentation by 
engineered Saccharomyces cerevisiae.

Full Description: One of the challenges in strain improvement by evolutionary engineering is to subsequently
determine the molecular basis of the improved properties that were enriched from the 
natural genetic variation during the selective conditions. This study focuses on 
Saccharomyces cerevisiae IMS0002 which, after metabolic and evolutionary engineering, 
ferments the pentose sugar arabinose. Glucose- and arabinose-limited anaerobic chemostat 
cultures of IMS0002 and its non-evolved ancestor were subjected to transcriptome analysis
, intracellular metabolite measurements and metabolic flux analysis. Increased expression 
of the GAL-regulon and deletion of GAL2 in IMS0002 confirmed that the galactose 
transporter is essential for growth on arabinose. Elevated intracellular concentrations of
pentose-phosphate-pathway intermediates and upregulation of TKL2 and YGR043c (encoding 
transketolase and transaldolase isoenzymes) suggested an involvement of these genes in 
flux-controlling reactions in arabinose fermentation. Indeed, deletion of these genes in 
IMS0002 caused a 21% reduction of the maximum specific growth rate on arabinose.

PMID: 20816840

GEO ID: GSE27539.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE27539.final.pcl	Metabolome, transcriptome and metabolic flux analysis of arabinose fermentation by engineered Saccharomyces cerevisiae.	9	synthetic biology|carbon utilization|oxygen level alteration|fermentation	1

File last modified: Thu Apr 25 2024

---


# Xu_2010_PMID_20228058

Citation: Xu T, Shively CA, Jin R, Eckwahl MJ, Dobry CJ, Song Q, Kumar A. 2010. A profile of 
differentially abundant proteins at the yeast cell periphery during  pseudohyphal growth.

Full Description: Yeast filamentous growth is a stress response to conditions of nitrogen deprivation, 
wherein yeast colonies form pseudohyphal filaments of elongated and  connected cells. As 
proteins mediating adhesion and transport are required for this growth transition, we 
expect that the protein complement at the yeast cell periphery plays a critical and 
tightly regulated role in pseudohyphal filamentation. To identify proteins differentially 
abundant at the yeast cell periphery during pseudohyphal growth, we generated quantitative
proteomic profiles of plasma membrane protein preparations under conditions of vegetative 
growth and filamentation. By isobaric tags for relative and absolute quantification 
chemistry and two-dimensional liquid chromatography-tandem mass spectrometry, we profiled 
2463 peptides and 356 proteins, identifying 11 differentially abundant proteins that 
localize to the yeast cell periphery. This  protein set includes Ylr414cp, herein renamed 
Pun1p, a previously uncharacterized protein localized to the plasma membrane compartment 
of Can1. Pun1p abundance is  doubled under conditions of nitrogen stress, and deletion of 
PUN1 abolishes filamentous growth in haploids and diploids; pun1Delta mutants are 
noninvasive, lack surface-spread filamentation, grow slowly, and exhibit impaired cell 
adhesion. Conversely, overexpression of PUN1 results in exaggerated cell elongation under 
conditions of nitrogen stress. PUN1 contributes to yeast nitrogen signaling, as pun1Delta 
mutants misregulate amino acid biosynthetic genes during nitrogen stress. By chromatin 
immunoprecipitation and reverse transcription-PCR, we find that the filamentous growth 
factor Mss11p directly binds the PUN1 promoter and regulates its transcription. In total, 
this study provides the first profile of differential protein abundance during 
pseudohyphal  growth, identifying a previously uncharacterized membrane compartment of 
Can1 protein required for wild-type nitrogen signaling and filamentous growth.

PMID: 20228058

GEO ID: GSE20351.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE20351.final.pcl	A profile of differentially abundant proteins at the yeast cell periphery during  pseudohyphal growth.	6	filamentous growth	1

File last modified: Thu Apr 25 2024

---


# Yasokawa_2010_PMID_19499198

Citation: Yasokawa D, Murata S, Iwahashi Y, Kitagawa E, Nakagawa R, Hashido T, Iwahashi H. 2010. 
Toxicity of methanol and formaldehyde towards Saccharomyces cerevisiae as assessed by DNA 
microarray analysis.

Full Description: To assess the toxicity of the C1 compounds methanol and formaldehyde, gene expression 
profiles of treated baker's yeast were analyzed using DNA microarrays. Among approximately
6,000 open reading frames (ORFs), 314 were repressed and 375  were induced in response to 
methanol. The gene process category energy comprised the greatest number of induced genes 
while protein synthesis comprised the greatest number of repressed genes. Products of 
genes induced by methanol were mainly integral membrane proteins or were localized to the 
plasma membrane. A total of 622 and 610 ORFs were induced or repressed by formaldehyde,  
respectively. More than one-third of the genes found to be strongly repressed by  
formaldehyde belonged to the protein synthesis functional category. Conversely, genes in 
the subcategory of nitrogen, sulfur, and selenium metabolism within metabolism and in the 
category of cell rescue, defense, and virulence were up-regulated by exposure to 
formaldehyde. Our data suggest that membrane structure is a major target of methanol 
toxicity, while proteins were major targets of formaldehyde toxicity.

PMID: 19499198

GEO ID: GSE9231.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE9231.final.pcl	Toxicity of methanol and formaldehyde towards Saccharomyces cerevisiae as assessed by DNA microarray analysis.	5	metal or metalloid ion stress	2

File last modified: Thu Apr 25 2024

---


# Zhang_2010_PMID_20545859

Citation: Zhang J, Olsson L, Nielsen J. 2010. The beta-subunits of the Snf1 kinase in Saccharomyces 
cerevisiae, Gal83 and Sip2, but not Sip1, are redundant in glucose derepression and 
regulation of sterol biosynthesis.

Full Description: The conserved Snf1/AMP-activated protein kinase family is one of the central components in
the nutrient sensing and regulation of the carbon metabolism in eukaryotes. It is also 
involved in several other processes such as stress resistance, invasive growth and ageing
. Snf1 kinase is composed of a catalytic alpha-subunit Snf1, a regulatory gamma-subunit 
Snf4 and one of three possible beta-subunits, Sip1, Sip2 or Gal83. We used a systematic 
approach to study the role of the three beta-subunits by analysing all seven possible 
combinations of beta-subunit deletions together with the reference strain. Previous 
studies showed that the three beta-subunits are redundant for growth on alternative carbon
sources. Here we report that the mutant strain with only SIP1 expressed (sip2Delta 
gal83Delta) could utilize acetate, but neither ethanol nor glycerol, as alternative carbon
source. We also showed that Gal83 is the most important isoform not only for the growth on
non-fermentable carbon sources, but also for regulation of ergosterol biosynthetic genes, 
under glucose-limited condition. Furthermore, we found that Sip2, but not Sip1, can take 
over when Gal83 is deleted, but to a lesser extent. However, Sip1 may be sufficient for 
some other processes such as regulation of the nitrogen metabolism and meiosis.

PMID: 20545859

GEO ID: GSE20591.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE20591.final.pcl	The beta-subunits of the Snf1 kinase in Saccharomyces cerevisiae, Gal83 and Sip2, but not Sip1, are redundant in glucose derepression and regulation of sterol biosynthesis.	24	stress|carbon utilization	1

File last modified: Thu Apr 25 2024

---


# Zheng_2010_PMID_20237471

Citation: Zheng W, Zhao H, Mancera E, Steinmetz LM, Snyder M. 2010. Genetic analysis of variation in
transcription factor binding in yeast.

Full Description: Variation in transcriptional regulation is thought to be a major cause of phenotypic 
diversity. Although widespread differences in gene expression among individuals of a 
species have been observed, studies to examine the variability of transcription factor 
binding on a global scale have not been performed, and thus the extent and underlying 
genetic basis of transcription factor binding diversity is unknown. By mapping differences
in transcription factor binding among individuals, here we present the genetic basis of 
such variation on a genome-wide scale. Whole-genome Ste12-binding profiles were determined
using chromatin immunoprecipitation coupled with DNA sequencing in pheromone-treated cells
of 43 segregants of a cross between two highly diverged yeast strains and their parental 
lines. We identified extensive Ste12-binding variation among individuals, and mapped 
underlying cis- and trans-acting loci responsible for such variation. We showed that most 
transcription factor binding variation is cis-linked, and that many variations are 
associated with polymorphisms residing in the binding motifs of Ste12 as well as those of 
several proposed Ste12 cofactors. We also identified two trans-factors, AMN1 and FLO8, 
that modulate Ste12 binding to promoters of more than ten genes under alpha-factor 
treatment. Neither of these two genes was previously known to regulate Ste12, and we 
suggest that they may be mediators of gene activity and phenotypic diversity. Ste12 
binding strongly correlates with gene expression for more than 200 genes, indicating that 
binding variation is functional. Many of the variable-bound genes are involved in cell 
wall organization and biogenesis. Overall, these studies identified genetic regulators of 
molecular diversity among individuals and provide new insights into mechanisms of gene 
regulation.

PMID: 20237471

GEO ID: GSE19634.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE19634.final.pcl	Genetic analysis of variation in transcription factor binding in yeast.	48	transcription|genome variation	1

File last modified: Thu Apr 25 2024

---


# van_Wageningen_2010_PMID_21145464

Citation: van Wageningen S, Kemmeren P, Lijnzaad P, Margaritis T, Benschop JJ, de Castro IJ, van 
Leenen D, Groot Koerkamp MJ, Ko CW, Miles AJ, Brabers N, Brok MO, Lenstra TL, Fiedler D, 
Fokkens L, Aldecoa R, Apweiler E, Taliadouros V, Sameith K, van de Pasch LA, van Hooff SR,
Bakker LV, Krogan NJ, Snel B, Holstege FC. 2010. Functional overlap and regulatory links 
shape genetic interactions between signaling pathways.

Full Description: To understand relationships between phosphorylation-based signaling pathways, we  analyzed
150 deletion mutants of protein kinases and phosphatases in S. cerevisiae using DNA 
microarrays. Downstream changes in gene expression were treated as a phenotypic readout. 
Double mutants with synthetic genetic interactions were included to investigate genetic 
buffering relationships such as redundancy. Three types of genetic buffering relationships
are identified: mixed  epistasis, complete redundancy, and quantitative redundancy. In 
mixed epistasis,  the most common buffering relationship, different gene sets respond in 
different  epistatic ways. Mixed epistasis arises from pairs of regulators that have only 
partial overlap in function and that are coupled by additional regulatory links such as 
repression of one by the other. Such regulatory modules confer the ability to control 
different combinations of processes depending on condition or  context. These properties 
likely contribute to the evolutionary maintenance of paralogs and indicate a way in which 
signaling pathways connect for multiprocess  control.

PMID: 21145464

GEO ID: GSE25644

PCL filename	short description	# conditions	tags	# channels
GSE25644_s377.final.pcl	Functional overlap and regulatory links shape genetic interactions between signaling pathways.	377	signaling	2

File last modified: Thu Apr 25 2024

---


# Ambroset_2011_PMID_22384338

Citation: Ambroset C, Petit M, Brion C, Sanchez I, Delobel P, Guerin C, Chiapello H, Nicolas P, 
Bigey F, Dequin S, Blondin B. 2011. Deciphering the molecular basis of wine yeast 
fermentation traits using a combined genetic and genomic approach.

Full Description: The genetic basis of the phenotypic diversity of yeast is still poorly understood. Wine 
yeast strains have specific abilities to grow and ferment under  stressful conditions 
compared with other strains, but the genetic basis underlying these traits is unknown. 
Understanding how sequence variation influences such phenotypes is a major challenge to 
address adaptation mechanisms  of wine yeast. We aimed to identify the genetic basis of 
fermentation traits and  gain insight into their relationships with variations in gene 
expression among yeast strains. We combined fermentation trait QTL mapping and expression 
profiling of fermenting cells in a segregating population from a cross between a  wine 
yeast derivative and a laboratory strain. We report the identification of QTL for various 
fermentation traits (fermentation rates, nitrogen utilization, metabolites production) as 
well as expression QTL (eQTL). We found that many transcripts mapped to several eQTL 
hotspots and that two of them overlapped with  QTL for fermentation traits. A QTL 
controlling the maximal fermentation rate and  nitrogen utilization overlapping with an 
eQTL hotspot was dissected. We functionally demonstrated that an allele of the ABZ1 gene, 
localized in the hotspot and involved in p-aminobenzoate biosynthesis, controls the 
fermentation rate through modulation of nitrogen utilization. Our data suggest that the 
laboratory strain harbors a defective ABZ1 allele, which triggers strong metabolic and 
physiological alterations responsible for the generation of the eQTL hotspot. They also 
suggest that a number of gene expression differences result from some alleles that trigger
major physiological disturbances.

PMID: 22384338

GEO ID: GSE26437

PCL filename	short description	# conditions	tags	# channels
GSE26437_final.pcl	Deciphering the molecular basis of wine yeast fermentation traits using a combined genetic and genomic approach.	40	fermentation|nitrogen utilization	2

File last modified: Thu Apr 25 2024

---


# Aparicio_2011_PMID_22087310

Citation: Aparicio F, Aparicio-Sanchis R, Gadea J, Sanchez-Navarro JA, Pallas V, Murguia JR. 2011. A,
plant virus movement protein regulates the Gcn2p kinase in budding yeast.

Full Description: Virus life cycle heavily depends on their ability to command the host machinery in order 
to translate their genomes. Animal viruses have been shown to interfere  with host 
translation machinery by expressing viral proteins that either maintain or inhibit 
eIF2alpha function by phosphorylation. However, this interference mechanism has not been 
described for any plant virus yet. Prunnus necrotic ringspot virus (PNRSV) is a serious 
pathogen of cultivated stone fruit trees. The movement protein (MP) of PNRSV is necessary 
for the cell-to-cell movement of the  virus. By using a yeast-based approach we have found
that over-expression of the  PNRSV MP caused a severe growth defect in yeast cells. cDNA 
microarrays analysis  carried out to characterise at the molecular level the growth 
interference phenotype reported the induction of genes related to amino acid deprivation 
suggesting that expression of MP activates the GCN pathway in yeast cells. Accordingly, 
PNRSV MP triggered activation of the Gcn2p kinase, as judged by increased eIF2alpha 
phosphorylation. Activation of Gcn2p by MP expression required a functional Tor1p kinase, 
since rapamycin treatment alleviated the yeast cell growth defect and blocked eIF2alpha 
phosphorylation triggered by MP expression. Overall, these findings uncover a previously 
uncharacterised function for PNRSV MP viral protein, and point out at Tor1p and Gcn2p 
kinases as candidate susceptibility factors for plant viral infections.

PMID: 22087310

GEO ID: GSE26120.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE26120.final.pcl	A plant virus movement protein regulates the Gcn2p kinase in budding yeast.	4	synthetic biology	2

File last modified: Thu Apr 25 2024
GSE26119.final.pcl	A plant virus movement protein regulates the Gcn2p kinase in budding yeast.	4	synthetic biology	2

File last modified: Thu Apr 25 2024

---


# Arribere_2011_PMID_22152478

Citation: Arribere JA, Doudna JA, Gilbert WV. 2011. Reconsidering movement of eukaryotic mRNAs 
between polysomes and P bodies.

Full Description: Cell survival in changing environments requires appropriate regulation of gene expression
, including posttranscriptional regulatory mechanisms. From reporter gene studies in 
glucose-starved yeast, it was proposed that translationally silenced eukaryotic mRNAs 
accumulate in P bodies and can return to active translation. We present evidence 
contradicting the notion that reversible storage of nontranslating mRNAs is a widespread 
and general phenomenon. First, genome-wide measurements of mRNA abundance, translation, 
and ribosome occupancy after glucose withdrawal show that most mRNAs are depleted from the
cell coincident with their depletion from polysomes. Second, only a limited subpopulation 
of translationally repressed transcripts, comprising fewer than 400 genes, can be 
reactivated for translation upon glucose readdition in the absence  of new transcription. 
This highly selective posttranscriptional regulation could  be a mechanism for cells to 
minimize the energetic costs of reversing gene-regulatory decisions in rapidly changing 
environments by transiently preserving a pool of transcripts whose translation is rate-
limiting for growth.

PMID: 22152478

GEO ID: GSE31220.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE31220.final.pcl	Reconsidering movement of eukaryotic mRNAs between polysomes and P bodies.	8	starvation	2

File last modified: Thu Apr 25 2024

---


# Boender_2011_PMID_21803078

Citation: Boender LG, Almering MJ, Dijk M, van Maris AJ, de Winde JH, Pronk JT, Daran-Lapujade P. 
2011. Extreme calorie restriction and energy source starvation in Saccharomyces cerevisiae
represent distinct physiological states.

Full Description: Cultivation methods used to investigate microbial calorie restriction often result in 
carbon and energy starvation. This study aims to dissect cellular responses to calorie 
restriction and starvation in Saccharomyces cerevisiae by using retentostat cultivation. 
In retentostats, cells are continuously supplied with a small, constant carbon and energy 
supply, sufficient for maintenance of cellular viability and integrity but insufficient 
for growth. When glucose-limited retentostats cultivated under extreme calorie restriction
were subjected to glucose starvation, calorie-restricted and glucose-starved cells were 
found to share characteristics such as increased heat-shock tolerance and expression of 
quiescence-related genes. However, they also displayed strikingly different features. 
While calorie-restricted yeast cultures remained metabolically active and viable for 
prolonged periods of time, glucose starvation resulted in rapid consumption of reserve 
carbohydrates, population heterogeneity  due to appearance of senescent cells and, 
ultimately, loss of viability. Moreover, during starvation, calculated rates of ATP 
synthesis from reserve carbohydrates were 2-3 orders of magnitude lower than steady-state 
ATP-turnover rates calculated under extreme calorie restriction in retentostats. Stringent
reduction of ATP turnover during glucose starvation was accompanied by a strong down-
regulation of genes involved in protein synthesis. These results demonstrate that extreme 
calorie restriction and carbon starvation represent different physiological states in S. 
cerevisiae.

PMID: 21803078

GEO ID: GSE22602.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE22602.final.pcl	Extreme calorie restriction and energy source starvation in Saccharomyces cerevisiae represent distinct physiological states.	10	starvation|carbon utilization	1

File last modified: Thu Apr 25 2024

---


# Boender_2011_PMID_22093745

Citation: Boender LG, van Maris AJ, de Hulster EA, Almering MJ, van der Klei IJ, Veenhuis M, de 
Winde JH, Pronk JT, Daran-Lapujade P. 2011. Cellular responses of Saccharomyces cerevisiae
at near-zero growth rates: transcriptome analysis of anaerobic retentostat cultures.

Full Description: Extremely low specific growth rates (below 0.01 h-1) represent a largely unexplored area 
of microbial physiology. Retentostats enable controlled, energy-limited cultivation at 
near-zero specific growth rates while avoiding starvation. In this study, anaerobic, 
glucose-limited retentostats were used to analyze physiological and genome-wide 
transcriptional responses of Saccharomyces cerevisiae to cultivation at near-zero specific
growth rates. Cultures at near-zero specific growth rates exhibited several 
characteristics previously associated with quiescence, including accumulation of storage 
polymers and an increased expression of genes involved in storage metabolism, autophagy 
and exit from the replicative cell cycle into G0. Analysis of transcriptome data from 
glucose-limited retentostat and chemostat cultures showed, as specific growth rate was 
decreased, quiescence-related transcriptional responses already set in at specific growth 
rates above 0.025 h-1. Many genes involved in mitochondrial processes were specifically 
upregulated at near-zero specific growth rates, possibly reflecting an increased turn-over
of organelles under these conditions. Prolonged (> 2 weeks) cultivation in retentostat 
cultures led to induction of several genes that were previously implicated in 
chronological ageing. These observations stress the need for systematic dissection of 
physiological responses to slow growth, quiescence, ageing and starvation and indicate 
that controlled cultivation systems such as retentostats can contribute to this goal.

PMID: 22093745

GEO ID: GSE22574.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE22574.final.pcl	Cellular responses of Saccharomyces cerevisiae at near-zero growth rates: transcriptome analysis of anaerobic retentostat cultures	11	cell aging|nutrient utilization	1

File last modified: Thu Apr 25 2024

---


# Burrill_2011_PMID_21363961

Citation: Burrill DR, Silver PA. 2011. Synthetic circuit identifies subpopulations with sustained 
memory of DNA damage.

Full Description: Differential responses to stimuli can affect how cells succumb to disease. In yeast, DNA 
damage can create heterogeneous responses. To delineate how a response contributes to a 
cell's future behavior, we constructed a transcription-based memory circuit that detects 
DNA repair to isolate subpopulations with heritable damage responses. Strongly responsive 
cells show multigenerational effects, including growth defects and iron-associated gene 
expression. Less-responsive cells exhibit increased mutation frequencies but resume wild-
type behavior. These two subpopulations remain distinct for multiple generations, 
indicating a transmissible memory of damage. Collectively, this work demonstrates the 
efficacy of using synthetic biology to define how environmental exposure contributes to 
distinct cell fates.

PMID: 21363961

GEO ID: GSE36210.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE36210.final.pcl	Synthetic circuit identifies subpopulations with sustained memory of DNA damage.	3	DNA damage stimulus	2

File last modified: Thu Apr 25 2024

---


# Casado_2011_PMID_21749328

Citation: Casado C, Gonzalez A, Platara M, Ruiz A, Arino J. 2011. The role of the protein kinase A 
pathway in the response to alkaline pH stress in yeast.

Full Description: Exposure of Saccharomyces cerevisiae to alkaline pH provokes a stress condition that 
generates a compensatory reaction. In the present study we examined a possible role for 
the PKA (protein kinase A) pathway in this response. Phenotypic analysis revealed that 
mutations that activate the PKA pathway (ira1 ira2, bcy1)  tend to cause sensitivity to 
alkaline pH, whereas its deactivation enhances tolerance to this stress. We observed that 
alkalinization causes a transient decrease in cAMP, the main regulator of the pathway. 
Alkaline pH causes rapid nuclear localization of the PKA-regulated Msn2 transcription 
factor which, together with Msn4, mediates a general stress response by binding with STRE
(stress response element) sequences in many promoters. Consequently, a synthetic  STRE-
LacZ reporter shows a rapid induction in response to alkaline stress. A msn2 msn4 mutant 
is sensitive to alkaline pH, and transcriptomic analysis reveals that after 10 min of 
alkaline stress, the expression of many induced genes (47%) depends, at least in part, on 
the presence of Msn2 and Msn4. Taken together, these results demonstrate that inhibition 
of the PKA pathway by alkaline pH represents a substantial part of the adaptive response 
to this kind of stress and that this response involves Msn2/Msn4-mediated genome 
expression remodelling. However, the relevance of attenuation of PKA in high pH tolerance 
is probably not restricted to regulation of Msn2 function.

PMID: 21749328

GEO ID: GSE27925

PCL filename	short description	# conditions	tags	# channels
GSE27925_final.pcl	The role of the protein kinase A pathway in the response to alkaline pH stress in yeast.	16	stress|signaling	2

File last modified: Thu Apr 25 2024

---


# Celona_2011_PMID_21738444

Citation: Celona B, Weiner A, Di Felice F, Mancuso FM, Cesarini E, Rossi RL, Gregory L, Baban D, 
Rossetti G, Grianti P, Pagani M, Bonaldi T, Ragoussis J, Friedman N, Camilloni G, Bianchi 
ME, Agresti A. 2011. Substantial histone reduction modulates genomewide nucleosomal 
occupancy and global transcriptional output.

Full Description: The basic unit of genome packaging is the nucleosome, and nucleosomes have long been 
proposed to restrict DNA accessibility both to damage and to transcription.  Nucleosome 
number in cells was considered fixed, but recently aging yeast and mammalian cells were 
shown to contain fewer nucleosomes. We show here that mammalian cells lacking High 
Mobility Group Box 1 protein (HMGB1) contain a reduced amount of core, linker, and variant
histones, and a correspondingly reduced number of nucleosomes, possibly because HMGB1 
facilitates nucleosome assembly. Yeast nhp6 mutants lacking Nhp6a and -b proteins, which 
are related to  HMGB1, also have a reduced amount of histones and fewer nucleosomes. 
Nucleosome limitation in both mammalian and yeast cells increases the sensitivity of DNA 
to  damage, increases transcription globally, and affects the relative expression of  
about 10% of genes. In yeast nhp6 cells the loss of more than one nucleosome in four does 
not affect the location of nucleosomes and their spacing, but nucleosomal occupancy. The 
decrease in nucleosomal occupancy is non-uniform and can be modelled assuming that 
different nucleosomal sites compete for available histones. Sites with a high propensity 
to occupation are almost always packaged into nucleosomes both in wild type and nucleosome
-depleted cells; nucleosomes on  sites with low propensity to occupation are 
disproportionately lost in nucleosome-depleted cells. We suggest that variation in 
nucleosome number, by affecting nucleosomal occupancy both genomewide and gene-
specifically, constitutes a novel layer of epigenetic regulation.

PMID: 21738444

GEO ID: GSE23711

PCL filename	short description	# conditions	tags	# channels
GSE23711_final.pcl	Substantial histone reduction modulates genomewide nucleosomal occupancy and global transcriptional output.	6	chromatin organization	1

File last modified: Thu Apr 25 2024

---


# Chiu_2011_PMID_21147769

Citation: Chiu J, Tactacan CM, Tan SX, Lin RC, Wouters MA, Dawes IW. 2011. Cell cycle sensing of 
oxidative stress in Saccharomyces cerevisiae by oxidation of a specific cysteine residue 
in the transcription factor Swi6p.

Full Description: Yeast cells begin to bud and enter the S phase when growth conditions are favorable during
the G(1) phase. When subjected to some oxidative stresses, cells delay entry at G(1), 
allowing repair of cellular damage. Hence, oxidative stress  sensing is coordinated with 
the regulation of cell cycle. We identified a novel function of the cell cycle regulator 
of Saccharomyces cerevisiae, Swi6p, as a redox sensor through its cysteine residue at 
position 404. When alanine was substituted at this position, the resultant mutant, C404A, 
was sensitive to several reactive oxygen species and oxidants including linoleic acid 
hydroperoxide, the superoxide anion, and diamide. This mutant lost the ability to arrest 
in G(1) phase upon treatment with lipid hydroperoxide. The Cys-404 residue of Swi6p in 
wild-type cells was oxidized to a sulfenic acid when cells were subjected to linoleic acid
hydroperoxide. Mutation of Cys-404 to Ala abolished the down-regulation of expression of 
the G(1) cyclin genes CLN1, CLN2, PCL1, and  PCL2 that occurred when cells of the wild 
type were exposed to the lipid hydroperoxide. In conclusion, oxidative stress signaling 
for cell cycle regulation occurs through oxidation of the G(1)/S-specific transcription 
factor Swi6p and consequently leads to suppression of the expression of G(1) cyclins and a
delay in cells entering the cell cycle.

PMID: 21147769

GEO ID: GSE18334

PCL filename	short description	# conditions	tags	# channels
GSE18334_final.pcl	Cell cycle sensing of oxidative stress in Saccharomyces cerevisiae by oxidation of a specific cysteine residue in the transcription factor Swi6p.	18	oxidative stress|mitotic cell cycle	1

File last modified: Thu Apr 25 2024

---


# Cocklin_2011_PMID_21196523

Citation: Cocklin R, Heyen J, Larry T, Tyers M, Goebl M. 2011. New insight into the role of the 
Cdc34 ubiquitin-conjugating enzyme in cell cycle regulation via Ace2 and Sic1.

Full Description: The Cdc34 ubiquitin-conjugating enzyme plays a central role in progression of the cell 
cycle. Through analysis of the phenotype of a mutant missing a highly conserved sequence 
motif within the catalytic domain of Cdc34, we discovered previously unrecognized levels 
of regulation of the Ace2 transcription factor and the cyclin-dependent protein kinase 
inhibitor Sic1. In cells carrying the Cdc34(tm) mutation, which alters the conserved 
sequence, the cyclin-dependent protein kinase inhibitor Sic1, an SCF(Cdc4) substrate, has 
a shorter half-life, while the cyclin Cln1, an SCF(Grr1) substrate, has a longer half-life
than in wild-type cells. Expression of the SIC1 gene cluster, which is regulated by Swi5  
and Ace2 transcription factors, is induced in CDC34(tm) cells. Levels of Swi5, Ace2, and 
the SCF(Grr1) targets Cln1 and Cln2 are elevated in Cdc34(tm) cells, and loss of Grr1 
causes an increase in Ace2 levels. Sic1 levels are similar in CDC34(tm) ace2Delta and wild
-type cells, explaining a paradoxical increase in the steady-state level of Sic1 protein 
despite its reduced half-life. A screen for mutations that interact with CDC34(tm) 
uncovered novel regulators of Sic1, including genes encoding the polyubiquitin chain 
receptors Rad23 and Rpn10.

PMID: 21196523

GEO ID: GSE8453.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE8453.final.pcl	New insight into the role of the Cdc34 ubiquitin-conjugating enzyme in cell cycle regulation via Ace2 and Sic1.	8	protein modification	1

File last modified: Thu Apr 25 2024

---


# Dori-Bachash_2011_PMID_21811398

Citation: Dori-Bachash M, Shema E, Tirosh I. 2011. Coupled evolution of transcription and mRNA 
degradation.

Full Description: mRNA levels are determined by the balance between transcription and mRNA degradation, and 
while transcription has been extensively studied, very little is known regarding the 
regulation of mRNA degradation and its coordination with transcription. Here we examine 
the evolution of mRNA degradation rates between two closely related yeast species. 
Surprisingly, we find that around half of the  evolutionary changes in mRNA degradation 
were coupled to transcriptional changes  that exert opposite effects on mRNA levels. 
Analysis of mRNA degradation rates in an interspecific hybrid further suggests that 
opposite evolutionary changes in transcription and in mRNA degradation are mechanistically
coupled and were generated by the same individual mutations. Coupled changes are 
associated with divergence of two complexes that were previously implicated both in 
transcription and in mRNA degradation (Rpb4/7 and Ccr4-Not), as well as with sequence 
divergence of transcription factor binding motifs. These results suggest that an  opposite
coupling between the regulation of transcription and that of mRNA degradation has shaped 
the evolution of gene regulation in yeast.

PMID: 21811398

GEO ID: GSE28849GPL13449.sfp.pcl

PCL filename	short description	# conditions	tags	# channels
GSE28849GPL13449.sfp.pcl	Coupled evolution of transcription and mRNA degradation.	8	RNA catabolism	1

File last modified: Thu Apr 25 2024

---


# Dymond_2011_PMID_21918511

Citation: Dymond JS, Richardson SM, Coombes CE, Babatz T, Muller H, Annaluru N, Blake WJ, 
Schwerzmann JW, Dai J, Lindstrom DL, Boeke AC, Gottschling DE, Chandrasegaran S, Bader JS,
Boeke JD. 2011. Synthetic chromosome arms function in yeast and generate phenotypic 
diversity by  design.

Full Description: Recent advances in DNA synthesis technology have enabled the construction of novel genetic
pathways and genomic elements, furthering our understanding of system-level phenomena. The
ability to synthesize large segments of DNA allows the engineering of pathways and genomes
according to arbitrary sets of design principles. Here we describe a synthetic yeast 
genome project, Sc2.0, and the first partially synthetic eukaryotic chromosomes, 
Saccharomyces cerevisiae chromosome synIXR, and semi-synVIL. We defined three design 
principles for a synthetic genome as follows: first, it should result in a (near) wild-
type phenotype and fitness; second, it should lack destabilizing elements such as tRNA 
genes or transposons; and third, it should have genetic flexibility to facilitate future 
studies. The synthetic genome features several systemic modifications complying with the 
design principles, including an inducible evolution system, SCRaMbLE (synthetic chromosome
rearrangement and modification by loxP-mediated evolution). We show the utility of 
SCRaMbLE as a novel method of combinatorial mutagenesis, capable of generating complex 
genotypes and a broad variety of phenotypes. When complete, the fully synthetic genome 
will allow massive restructuring of the yeast genome, and may open the door to a new type 
of combinatorial genetics based entirely on variations in gene content and copy number.

PMID: 21918511

GEO ID: GSE31326

PCL filename	short description	# conditions	tags	# channels
GSE31326_final.pcl	Synthetic chromosome arms function in yeast and generate phenotypic diversity by  design.	15	synthetic biology	1

File last modified: Thu Apr 25 2024

---


# Edskes_2011_PMID_21368275

Citation: Edskes HK, Engel A, McCann LM, Brachmann A, Tsai HF, Wickner RB. 2011. Prion-forming 
ability of Ure2 of yeasts is not evolutionarily conserved.

Full Description: [URE3] is a prion (infectious protein) of the Saccharomyces cerevisiae Ure2p, a regulator 
of nitrogen catabolism. We show that wild S. paradoxus can be infected  with a [URE3] 
prion, supporting the use of S. cerevisiae as a prion test bed. We  find that the Ure2p of
Candida albicans and C. glabrata also regulate nitrogen catabolism. Conservation of amino 
acid sequence within the prion domain of Ure2p  has been proposed as evidence that the [
URE3] prion helps its host. We show that  the C. albicans Ure2p, which does not conserve 
this sequence, can nonetheless form a [URE3] prion in S. cerevisiae, but the C. glabrata 
Ure2p, which does have  the conserved sequence, cannot form [URE3] as judged by its 
performance in S. cerevisiae. These results suggest that the sequence is not conserved to 
preserve  prion forming ability.

PMID: 21368275

GEO ID: GSE26611.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE26611.final.pcl	Prion-forming ability of Ure2 of yeasts is not evolutionarily conserved.	4	prions	2

File last modified: Thu Apr 25 2024

---


# Eser_2011_PMID_21855792

Citation: Eser U, Falleur-Fettig M, Johnson A, Skotheim JM. 2011. Commitment to a cellular 
transition precedes genome-wide transcriptional change.

Full Description: In budding yeast, commitment to cell division corresponds to activating the positive 
feedback loop of G1 cyclins controlled by the transcription factors SBF  and MBF. This 
pair of transcription factors has over 200 targets, implying that cell-cycle commitment 
coincides with genome-wide changes in transcription. Here,  we find that genes within this
regulon have a well-defined distribution of transcriptional activation times. 
Combinatorial use of SBF and MBF results in a logical OR function for gene expression and 
partially explains activation timing. Activation of G1 cyclin expression precedes the 
activation of the bulk of the G1/S regulon, ensuring that commitment to cell division 
occurs before large-scale changes in transcription. Furthermore, we find similar positive 
feedback-first regulation in the yeasts S. bayanus and S. cerevisiae, as well as human 
cells. The widespread use of the feedback-first motif in eukaryotic cell-cycle control,  
implemented by nonorthologous proteins, suggests its frequent deployment at cellular 
transitions.

PMID: 21855792

GEO ID: GSE29893.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE29893.final.pcl	Commitment to a cellular transition precedes genome-wide transcriptional change.	12	mitotic cell cycle	2

File last modified: Thu Apr 25 2024
GSE29892.final.pcl	Commitment to a cellular transition precedes genome-wide transcriptional change.	8	mitotic cell cycle|carbon utilization	2

File last modified: Thu Apr 25 2024
GSE29800.final.pcl	Commitment to a cellular transition precedes genome-wide transcriptional change.	12	mitotic cell cycle	2

File last modified: Thu Apr 25 2024

---


# Fowler_2011_PMID_21085730

Citation: Fowler DM, Cooper SJ, Stephany JJ, Hendon N, Nelson S, Fields S. 2011. Suppression of 
statin effectiveness by copper and zinc in yeast and human cells.

Full Description: Lovastatin and other statins inhibit HMG-CoA reductase, which carries out an early step in
the sterol biosynthesis pathway. Statins lower cholesterol and are  widely prescribed to 
prevent heart disease, but like many drugs, they can interact with nutritionally acquired 
metabolites. To probe these interactions, we explored the effect of a diverse library of 
metabolites on statin effectiveness using a Saccharomyces cerevisiae model. In yeast, 
treatment with lovastatin results in reduced growth. We combined lovastatin with the 
library of metabolites, and found that copper and zinc ions impaired the ability of the 
statin to inhibit yeast growth. Using an integrated genomic and metabolomic approach, we 
found that lovastatin plus metal synergistically upregulated some sterol biosynthesis 
genes. This altered pattern of gene expression resulted in greater flux through the sterol
biosynthesis pathway and an increase in ergosterol levels. Each sterol intermediate level 
was correlated with expression  of the upstream gene. Thus, the ergosterol biosynthetic 
response induced by statin is enhanced by copper and zinc. In cultured mammalian cells, 
these metals  also rescued statin growth inhibition. Because copper and zinc impair the 
ability of statin to reduce sterol biosynthesis, dietary intake of these metals could have
clinical relevance for statin treatment in humans.

PMID: 21085730

GEO ID: GSE25004.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE25004.final.pcl	Suppression of statin effectiveness by copper and zinc in yeast and human cells.	23	chemical stimulus|metal or metalloid ion stress|lipid metabolism	2

File last modified: Thu Apr 25 2024

---


# Gil_2011_PMID_21842488

Citation: Gil FN, Goncalves AC, Jacinto MJ, Becker JD, Viegas CA. 2011. Transcriptional profiling in
Saccharomyces cerevisiae relevant for predicting alachlor mechanisms of toxicity.

Full Description: Alachlor has been a commonly applied herbicide and is a substance of ecotoxicological 
concern. The present study aims to identify molecular biomarkers in the eukaryotic model 
Saccharomyces cerevisiae that can be used to predict potential cytotoxic effects of 
alachlor, while providing new mechanistic clues with possible relevance for experimentally
less accessible eukaryotes. It focuses on genome-wide expression profiling in a yeast 
population in response to two exposure scenarios exerting effects from slight to moderate 
magnitude at phenotypic level. In particular, 100 and 264 genes, respectively, were found 
as differentially expressed on a 2-h exposure of yeast cells to the lowest observed  
effect concentration (110 mg/L) and the 20% inhibitory concentration (200 mg/L) of 
alachlor, in comparison with cells not exposed to the herbicide. The datasets  of alachlor
-responsive genes showed functional enrichment in diverse metabolic, transmembrane 
transport, cell defense, and detoxification categories. In general, the modifications in 
transcript levels of selected candidate biomarkers, assessed by quantitative reverse 
transcriptase polymerase chain reaction, confirmed the microarray data and varied 
consistently with the growth inhibitory effects of alachlor. Approximately 16% of the 
proteins encoded by alachlor-differentially expressed genes were found to share 
significant homology with proteins from ecologically relevant eukaryotic species. The 
biological relevance of these results is discussed in relation to new insights into the 
potential adverse effects of alachlor in health of organisms from ecosystems, particularly
in worst-case situations such as accidental spills or careless storage, usage, and 
disposal.

PMID: 21842488

GEO ID: GSE28677

PCL filename	short description	# conditions	tags	# channels
GSE28677_final.pcl	Transcriptional profiling in Saccharomyces cerevisiae relevant for predicting alachlor mechanisms of toxicity.	12	chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Gomez-Gonzalez_2011_PMID_21701562

Citation: Gomez-Gonzalez B, Garcia-Rubio M, Bermejo R, Gaillard H, Shirahige K, Marin A, Foiani M, 
Aguilera A. 2011. Genome-wide function of THO/TREX in active genes prevents R-loop-
dependent replication obstacles.

Full Description: THO/TREX is a conserved nuclear complex that functions in mRNP biogenesis and prevents 
transcription-associated recombination. Whether or not it has a ubiquitous role in the 
genome is unknown. Chromatin immunoprecipitation (ChIP)-chip studies reveal that the Hpr1 
component of THO and the Sub2 RNA-dependent ATPase have genome-wide distributions at 
active ORFs in yeast. In contrast to RNA polymerase II, evenly distributed from promoter 
to termination regions, THO and Sub2 are absent at promoters and distributed in a gradual 
5' --> 3' gradient. This is accompanied by a genome-wide impact of THO-Sub2 deletions on 
expression of highly expressed, long and high G+C-content genes. Importantly, ChIP-chips 
reveal an over-recruitment of Rrm3 in active genes in THO mutants that is reduced by 
RNaseH1 overexpression. Our work establishes a genome-wide function for THO-Sub2 in 
transcription elongation and mRNP biogenesis that function to prevent the accumulation of 
transcription-mediated replication obstacles, including R-loops.

PMID: 21701562

GEO ID: GSE24802

PCL filename	short description	# conditions	tags	# channels
GSE24802_final.pcl	Genome-wide function of THO/TREX in active genes prevents R-loop-dependent replication obstacles.	10	transcription|mRNA processing	1

File last modified: Thu Apr 25 2024

---


# Gonzalez-Aguilera_2011_PMID_21680710

Citation: Gonzalez-Aguilera C, Tous C, Babiano R, de la Cruz J, Luna R, Aguilera A. 2011. Nab2 
functions in the metabolism of RNA driven by polymerases II and III.

Full Description: Gene expression in eukaryotes is an essential process that includes transcription, RNA 
processing, and export. One important player in this interface is the poly(A)(+)-RNA-
binding protein Nab2, which regulates the mRNA poly(A)(+)-tail length and export. Here we 
show that Nab2 has additional roles during mRNA transcription, tRNA metabolism, and 
ribosomal subunit export. Nab2 is associated with the entire open reading frame of 
actively transcribed RNA polymerase (RNAP) II and III genes. As a consequence, nab2 
mutations confer translation defects that are detected by polysome profiling. Genome-wide 
analysis of expression of a conditional degron nab2 mutant shows that the role of Nab2 in
RNAPII transcription and RNAPIII metabolism is direct. Taken together, our results 
identify novel functions for Nab2 in transcription and metabolism of most types of RNAs, 
indicating that Nab2 function is more ubiquitous than previously anticipated, and that it 
is a central player in the general and coordinated control of gene expression from 
transcription to translation.

PMID: 21680710

GEO ID: GSE26303

PCL filename	short description	# conditions	tags	# channels
GSE26303_final.pcl	Nab2 functions in the metabolism of RNA driven by polymerases II and III.	6	mRNA processing	1

File last modified: Thu Apr 25 2024
GSE19302_final.pcl	Nab2 functions in the metabolism of RNA driven by polymerases II and III.	15	mRNA processing	1

File last modified: Thu Apr 25 2024

---


# Hong_2011_PMID_21715660

Citation: Hong KK, Vongsangnak W, Vemuri GN, Nielsen J. 2011. Unravelling evolutionary strategies of
yeast for improving galactose utilization  through integrated systems level analysis.

Full Description: Identification of the underlying molecular mechanisms for a derived phenotype by  adaptive
evolution is difficult. Here, we performed a systems-level inquiry into  the metabolic 
changes occurring in the yeast Saccharomyces cerevisiae as a result of its adaptive 
evolution to increase its specific growth rate on galactose and related these changes to 
the acquired phenotypic properties. Three evolved mutants (62A, 62B, and 62C) with higher 
specific growth rates and faster specific galactose uptake were isolated. The evolved 
mutants were compared with a reference strain and two engineered strains, SO16 and PGM2, 
which also showed higher galactose uptake rate in previous studies. The profile of 
intermediates in galactose metabolism was similar in evolved and engineered mutants, 
whereas reserve carbohydrates metabolism was specifically elevated in the evolved mutants 
and one evolved strain showed changes in ergosterol biosynthesis. Mutations were  
identified in proteins involved in the global carbon sensing Ras/PKA pathway, which is 
known to regulate the reserve carbohydrates metabolism. We evaluated one of the identified
mutations, RAS2(Tyr112), and this mutation resulted in an increased specific growth rate 
on galactose. These results show that adaptive evolution results in the utilization of 
unpredicted routes to accommodate increased galactose flux in contrast to rationally 
engineered strains. Our study  demonstrates that adaptive evolution represents a valuable 
alternative to rational design in bioengineering of improved strains and, that through 
systems biology, it is possible to identify mutations in evolved strain that can serve as 
unforeseen metabolic engineering targets for improving microbial strains for production of
biofuels and chemicals.

PMID: 21715660

GEO ID: GSE27185

PCL filename	short description	# conditions	tags	# channels
GSE27185_final.pcl	Unravelling evolutionary strategies of yeast for improving galactose utilization  through integrated systems level analysis.	12	carbon utilization|evolution	1

File last modified: Thu Apr 25 2024

---


# Jimeno_2011_PMID_21149575

Citation: Jimeno S, Tous C, Garcia-Rubio ML, Ranes M, Gonzalez-Aguilera C, Marin A, Aguilera A. 2011
 New suppressors of THO mutations identify Thp3 (Ypr045c)-Csn12 as a protein complex 
involved in transcription elongation.

Full Description: Formation of a ribonucleoprotein particle (mRNP) competent for export requires the 
coupling of transcription with mRNA processing and RNA export. A key link between these 
processes is provided by the THO complex. To progress in our understanding of this 
coupling, we have performed a search for suppressors of the transcription defect caused by
the hpr1Delta mutation. This has permitted us to identify mutations in the genes for the 
RNA polymerase II mediator component Med10, the Sch9 protein kinase, and the Ypr045c 
protein. We report a role in transcription elongation for Ypr045c (Thp3) and the Csn12 
component of the COP9 signalosome. Thp3 and Csn12 form a complex that is recruited to 
transcribed genes. Their mutations suppress the gene expression defects of THO complex 
mutants involved in mRNP biogenesis and export and show defects in mRNA accumulation. 
Transcription elongation impairment of thp3Delta mutants is shown by in vivo transcript 
run-on analysis performed in G-less systems. Thp3-Csn12 establishes a novel link between 
transcription and mRNA processing that opens new perspectives on our understanding of gene
expression and reveals novel functions  for a component of the COP9 signalosome. Thp3-
Csn12 also copurifies with ribosomal proteins, which opens the possibility that it has 
other functions in addition to transcription.

PMID: 21149575

GEO ID: GSE18994

PCL filename	short description	# conditions	tags	# channels
GSE18994_final.pcl	New suppressors of THO mutations identify Thp3 (Ypr045c)-Csn12 as a protein complex involved in transcription elongation.	6	transcription	1

File last modified: Thu Apr 25 2024

---


# Lenstra_2011_PMID_21596317

Citation: Lenstra TL, Benschop JJ, Kim T, Schulze JM, Brabers NA, Margaritis T, van de Pasch LA, van
Heesch SA, Brok MO, Groot Koerkamp MJ, Ko CW, van Leenen D, Sameith K, van Hooff SR, 
Lijnzaad P, Kemmeren P, Hentrich T, Kobor MS, Buratowski S, Holstege FC. 2011. The 
specificity and topology of chromatin interaction pathways in yeast.

Full Description: Packaging of DNA into chromatin has a profound impact on gene expression. To understand 
how changes in chromatin influence transcription, we analyzed 165 mutants of chromatin 
machinery components in Saccharomyces cerevisiae. mRNA expression patterns change in 80% 
of mutants, always with specific effects, even  for loss of widespread histone marks. The 
data are assembled into a network of chromatin interaction pathways. The network is 
function based, has a branched, interconnected topology, and lacks strict one-to-one 
relationships between complexes. Chromatin pathways are not separate entities for 
different gene sets,  but share many components. The study evaluates which interactions 
are important for which genes and predicts additional interactions, for example between 
Paf1C and Set3C, as well as a role for Mediator in subtelomeric silencing. The results  
indicate the presence of gene-dependent effects that go beyond context-dependent  binding 
of chromatin factors and provide a framework for understanding how specificity is achieved
through regulating chromatin.

PMID: 21596317

GEO ID: GSE25909

PCL filename	short description	# conditions	tags	# channels
GSE25909_s313_final.pcl	The specificity and topology of chromatin interaction pathways in yeast.	313	chromatin organization	2

File last modified: Thu Apr 25 2024

---


# Levy_2011_PMID_22158820

Citation: Levy S, Kafri M, Carmi M, Barkai N. 2011. The competitive advantage of a dual-transporter 
system.

Full Description: Cells use transporters of different affinities to regulate nutrient influx. When  
nutrients are depleted, low-affinity transporters are replaced by high-affinity ones. High
-affinity transporters are helpful when concentrations of nutrients are low, but the 
advantage of reducing their abundance when nutrients are abundant is less clear. When we 
eliminated such reduced production of the Saccharomyces cerevisiae high-affinity 
transporters for phosphate and zinc, the elapsed time from the initiation of the 
starvation program until the lack of nutrients limited growth was shortened, and recovery 
from starvation was delayed. The latter phenotype was rescued by constitutive activation 
of the starvation program. Dual-transporter systems appear to prolong preparation for 
starvation and to facilitate subsequent recovery, which may optimize sensing of nutrient 
depletion  by integrating internal and external information about nutrient availability.

PMID: 22158820

GEO ID: GSE32067.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE32067.final.pcl	The competitive advantage of a dual-transporter system.	16	phosphorus utilization|metal or metalloid ion utilization	2

File last modified: Thu Apr 25 2024

---


# Livas_2011_PMID_21827659

Citation: Livas D, Almering MJ, Daran JM, Pronk JT, Gancedo JM. 2011. Transcriptional responses to 
glucose in Saccharomyces cerevisiae strains lacking  a functional protein kinase A.

Full Description: BACKGROUND: The pattern of gene transcripts in the yeast Saccharomyces cerevisiae is 
strongly affected by the presence of glucose. An increased activity of protein kinase A (
PKA), triggered by a rise in the intracellular concentration of cAMP, can account for many
of the effects of glucose on transcription. In S. cerevisiae three genes, TPK1, TPK2, and 
TPK3, encode catalytic subunits of PKA. The lack of  viability of tpk1 tpk2 tpk3 triple 
mutants may be suppressed by mutations such as yak1 or msn2/msn4. To investigate the 
requirement for PKA in glucose control of gene expression, we have compared the effects of
glucose on global transcription  in a wild-type strain and in two strains devoid of PKA 
activity, tpk1 tpk2 tpk3 yak1 and tpk1 tpk2 tpk3 msn2 msn4. RESULTS: We have identified 
different classes  of genes that can be induced -or repressed- by glucose in the absence 
of PKA. Representative examples are genes required for glucose utilization and genes 
involved in the metabolism of other carbon sources, respectively. Among the genes 
responding to glucose in strains devoid of PKA some are also controlled by a redundant 
signalling pathway involving PKA activation, while others are not affected when PKA is 
activated through an increase in cAMP concentration. On the  other hand, among genes that 
do not respond to glucose in the absence of PKA, some give a full response to increased 
cAMP levels, even in the absence of glucose, while others appear to require the 
cooperation of different signalling pathways. We show also that, for a number of genes 
controlled by glucose through  a PKA-dependent pathway, the changes in mRNA levels are 
transient. We found that, in cells grown in gluconeogenic conditions, expression of a 
small number of genes, mainly connected with the response to stress, is reduced in the 
strains lacking PKA. CONCLUSIONS: In S. cerevisiae, the transcriptional responses to 
glucose are triggered by a variety of pathways, alone or in combination, in which PKA is 
often involved. Redundant signalling pathways confer a greater robustness  to the response
to glucose, while cooperative pathways provide a greater flexibility.

PMID: 21827659

GEO ID: GSE27541.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE27541.final.pcl	Transcriptional responses to glucose in Saccharomyces cerevisiae strains lacking  a functional protein kinase A.	10	carbon utilization|signaling	1

File last modified: Thu Apr 25 2024

---


# Ouyang_2011_PMID_20971184

Citation: Ouyang X, Tran QT, Goodwin S, Wible RS, Sutter CH, Sutter TR. 2011. Yap1 activation by 
H2O2 or thiol-reactive chemicals elicits distinct adaptive gene responses.

Full Description: The yeast Saccharomyces cerevisiae transcription factor Yap1 mediates an adaptive response
to oxidative stress by regulating protective genes. H(2)O(2) activates Yap1 through the 
Gpx3-mediated formation of a Yap1 Cys303-Cys598 intramolecular disulfide bond. Thiol-
reactive electrophiles can activate Yap1 directly by adduction to cysteine residues in the
C-terminal domain containing Cys598, Cys620, and Cys629. H(2)O(2) and N-ethylmaleimide (
NEM) showed no cross-protection against each other, whereas another thiol-reactive 
chemical, acrolein, elicited Yap1-dependent cross-protection against NEM, but not H(2)O(2
). Either Cys620 or Cys629 was sufficient for activation of Yap1 by NEM or acrolein; 
Cys598 was dispensable for this activation mechanism. To determine whether Yap1 activated 
by H(2)O(2) or thiol-reactive chemicals elicits distinct adaptive gene  responses, 
microarray analysis was performed on the wild-type strain or its isogenic single-deletion 
strain Deltayap1 treated with control buffer, H(2)O(2),  NEM, or acrolein. Sixty-five 
unique H(2)O(2) and 327 NEM and acrolein Yap1-dependent responsive genes were identified. 
Functional analysis using single-gene-deletion yeast strains demonstrated that protection 
was conferred by  CTA1 and CTT1 in the H(2)O(2)-responsive subset and YDR042C in the NEM- 
and acrolein-responsive subset. These findings demonstrate that the distinct mechanisms of
Yap1 activation by H(2)O(2) or thiol-reactive chemicals result in selective expression of 
protective genes.

PMID: 20971184

GEO ID: GSE19213.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE19213.final.pcl	Yap1 activation by H2O2 or thiol-reactive chemicals elicits distinct adaptive gene responses.	16	chemical stimulus|oxidative stress	1

File last modified: Thu Apr 25 2024

---


# Park_2011_PMID_21269496

Citation: Park EH, Zhang F, Warringer J, Sunnerhagen P, Hinnebusch AG. 2011. Depletion of eIF4G from
yeast cells narrows the range of translational efficiencies genome-wide.

Full Description: BACKGROUND: Eukaryotic translation initiation factor 4G (eIF4G) is thought to influence 
the translational efficiencies of cellular mRNAs by its roles in forming an eIF4F-mRNA-
PABP mRNP that is competent for attachment of the 43S preinitiation complex, and in 
scanning through structured 5' UTR sequences. We have tested this hypothesis by 
determining the effects of genetically depleting eIF4G from yeast cells on global 
translational efficiencies (TEs), using gene expression microarrays to measure the 
abundance of mRNA in polysomes relative to  total mRNA for ~5900 genes. RESULTS: Although 
depletion of eIF4G is lethal and reduces protein synthesis by ~75%, it had small effects (
less than a factor of 1.5) on the relative TE of most genes. Within these limits, however
, depleting eIF4G narrowed the range of translational efficiencies genome-wide, with mRNAs
of better than average TE being translated relatively worse, and mRNAs with lower than 
average TE being translated relatively better. Surprisingly, the fraction of mRNAs most 
dependent on eIF4G display an average 5' UTR length at or below the mean for all yeast 
genes. CONCLUSIONS: This finding suggests that eIF4G is more critical for ribosome 
attachment to mRNAs than for scanning long, structured 5' UTRs. Our results also indicate 
that eIF4G, and the closed-loop mRNP it assembles with the m7 G cap- and poly(A)-binding 
factors (eIF4E and PABP), is not essential for translation of most (if not all) mRNAs but 
enhances the differentiation of translational efficiencies genome-wide.

PMID: 21269496

GEO ID: GSE25721.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE25721.final.pcl	Depletion of eIF4G from yeast cells narrows the range of translational efficiencies genome-wide.	18	translational regulation	1

File last modified: Thu Apr 25 2024

---


# Parreiras_2011_PMID_21856932

Citation: Parreiras LS, Kohn LM, Anderson JB. 2011. Cellular effects and epistasis among three 
determinants of adaptation in experimental populations of Saccharomyces cerevisiae.

Full Description: Epistatic interactions in which the phenotypic effect of an allele is conditional on its 
genetic background have been shown to play a central part in various evolutionary 
processes. In a previous study (J. B. Anderson et al., Curr. Biol. 20:1383-1388, 2010; J. 
R. Dettman, C. Sirjusingh, L. M. Kohn, and J. B. Anderson, Nature 447:585-588, 2007), 
beginning with a common ancestor, we identified three  determinants of fitness as mutant 
alleles (each designated with the letter e) that arose in replicate Saccharomyces 
cerevisiae populations propagated in two different environments, a low-glucose and a high-
salt environment. In a low-glucose environment, MDS3e and MKT1e interacted positively to 
confer a fitness advantage. Also, PMA1e from a high-salt environment interacted negatively
with MKT1e in a low-glucose environment, an example of a Dobzhansky-Muller incompatibility
that confers reproductive isolation. Here we showed that the negative interaction between 
PMA1e and MKT1e is mediated by alterations in intracellular pH, while the positive 
interaction between MDS3e and MKT1e is mediated by changes in gene expression affecting 
glucose transporter genes. We specifically addressed the evolutionary significance of the 
positive interaction  by showing that the presence of the MDS3 mutation is a necessary 
condition for the spread and fixation of the new mutations at the identical site in MKT1. 
The expected mutations in MKT1 rose to high frequencies in two of three experimental  
populations carrying MDS3e but not in any of three populations carrying the ancestral 
allele. These data show how positive and negative epistasis can contribute to adaptation 
and reproductive isolation.

PMID: 21856932

GEO ID: GSE29353.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE29353.final.pcl	Cellular effects and epistasis among three determinants of adaptation in experimental populations of Saccharomyces cerevisiae.	10	evolution|carbon utilization	1

File last modified: Thu Apr 25 2024

---


# Ragni_2011_PMID_21320323

Citation: Ragni E, Piberger H, Neupert C, Garcia-Cantalejo J, Popolo L, Arroyo J, Aebi M, Strahl S. 
2011. The genetic interaction network of CCW12, a Saccharomyces cerevisiae gene required 
for cell wall integrity during budding and formation of mating projections.

Full Description: BACKGROUND: Mannoproteins construct the outer cover of the fungal cell wall. The  
covalently linked cell wall protein Ccw12p is an abundant mannoprotein. It is considered 
as crucial structural cell wall component since in baker's yeast the lack of CCW12 results
in severe cell wall damage and reduced mating efficiency. RESULTS: In order to explore the
function of CCW12, we performed a Synthetic Genetic Analysis (SGA) and identified genes 
that are essential in the absence of  CCW12. The resulting interaction network identified 
21 genes involved in cell wall integrity, chitin synthesis, cell polarity, vesicular 
transport and endocytosis. Among those are PFD1, WHI3, SRN2, PAC10, FEN1 and YDR417C, 
which have not been related to cell wall integrity before. We correlated our results with 
genetic interaction networks of genes involved in glucan and chitin synthesis. A core of 
genes essential to maintain cell integrity in response to cell wall stress was identified
. In addition, we performed a large-scale transcriptional analysis and compared the 
transcriptional changes observed in mutant ccw12Delta with transcriptomes from studies 
investigating responses to constitutive or acute cell wall damage. We identified a set of 
genes that are highly induced in the majority of the mutants/conditions and are directly 
related to the cell wall integrity pathway and cell wall compensatory responses. Among 
those are BCK1, CHS3, EDE1, PFD1, SLT2 and SLA1 that were also identified in the  SGA. In 
contrast, a specific feature of mutant ccw12Delta is the transcriptional  repression of 
genes involved in mating. Physiological experiments substantiate this finding. Further, we
demonstrate that Ccw12p is present at the cell periphery and highly concentrated at the 
presumptive budding site, around the bud, at the septum and at the tip of the mating 
projection. CONCLUSIONS: The combination of high throughput screenings, phenotypic 
analyses and localization studies provides new insight into the function of Ccw12p. A 
compensatory response, culminating in cell wall remodelling and transport/recycling 
pathways is required to buffer the loss of CCW12. Moreover, the enrichment of Ccw12p in 
bud, septum and mating projection is consistent with a role of Ccw12p in preserving cell 
wall integrity at sites of active growth.The microarray data produced in this analysis 
have been submitted to NCBI GEO database and GSE22649 record was assigned.

PMID: 21320323

GEO ID: GSE22649.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE22649.final.pcl	The genetic interaction network of CCW12, a Saccharomyces cerevisiae gene required for cell wall integrity during budding and formation of mating projections.	4	cell wall organization|cell morphogenesis|mating	1

File last modified: Thu Apr 25 2024

---


# Rintala_2011_PMID_21348598

Citation: Rintala E, Jouhten P, Toivari M, Wiebe MG, Maaheimo H, Penttila M, Ruohonen L. 2011. 
Transcriptional responses of Saccharomyces cerevisiae to shift from respiratory and 
respirofermentative to fully fermentative metabolism.

Full Description: In industrial fermentations of Saccharomyces cerevisiae, transient changes in oxygen 
concentration commonly occur and it is important to understand the behavior of cells 
during these changes. Glucose-limited chemostat cultivations were used to study the time-
dependent effect of sudden oxygen depletion on the transcriptome of S. cerevisiae cells 
initially in fully aerobic or oxygen-limited conditions. The overall responses to 
anaerobic conditions of cells initially in different conditions were very similar. 
Independent of initial culture conditions, transient downregulation of genes related to 
growth and cell proliferation, mitochondrial translation and protein import, and sulphate 
assimilation was seen. In addition, transient or permanent upregulation of genes  related 
to protein degradation, and phosphate and amino acid uptake was observed  in all cultures
. However, only in the initially oxygen-limited cultures was a transient upregulation of 
genes related to fatty acid oxidation, peroxisomal biogenesis, oxidative phosphorylation, 
TCA cycle, response to oxidative stress, and pentose phosphate pathway observed. 
Furthermore, from the initially oxygen-limited conditions, a rapid response around the 
metabolites of upper glycolysis and the pentose phosphate pathway was seen, while from the
initially fully aerobic conditions, a slower response around the pathways for utilization 
of respiratory carbon sources was observed.

PMID: 21348598

GEO ID: GSE22832

PCL filename	short description	# conditions	tags	# channels
GSE22832_final.pcl	Transcriptional responses of Saccharomyces cerevisiae to shift from respiratory and respirofermentative to fully fermentative metabolism.	28	oxygen level alteration|fermentation|respiration	1

File last modified: Thu Apr 25 2024

---


# Rossmann_2011_PMID_21474074

Citation: Rossmann MP, Luo W, Tsaponina O, Chabes A, Stillman B. 2011. A common telomeric gene 
silencing assay is affected by nucleotide metabolism.

Full Description: Telomere-associated position-effect variegation (TPEV) in budding yeast has been  used as 
a model for understanding epigenetic inheritance and gene silencing. A widely used assay 
to identify mutants with improper TPEV employs the URA3 gene at the telomere of chromosome
VII-L that can be counterselected with 5-fluoroorotic  acid (5-FOA). 5-FOA resistance has 
been inferred to represent lack of transcription of URA3 and therefore to represent 
heterochromatin-induced gene silencing. For two genes implicated in telomere silencing, 
POL30 and DOT1, we show that the URA3 telomere reporter assay does not reflect their role 
in heterochromatin formation. Rather, an imbalance in ribonucleotide reductase (RNR), 
which is induced by 5-FOA, and the specific promoter of URA3 fused to ADH4 at telomere VII
-L are jointly responsible for the variegated phenotype. We conclude that metabolic 
changes caused by the drug employed and certain mutants being studied are incompatible 
with the use of certain prototrophic markers for TPEV.

PMID: 21474074

GEO ID: GSE27222

PCL filename	short description	# conditions	tags	# channels
GSE27222_final.pcl	A common telomeric gene silencing assay is affected by nucleotide metabolism.	12	histone modification	1

File last modified: Thu Apr 25 2024

---


# Ruenwai_2011_PMID_21184438

Citation: Ruenwai R, Neiss A, Laoteng K, Vongsangnak W, Dalfard AB, Cheevadhanarak S, Petranovic D, 
Nielsen J. 2011. Heterologous production of polyunsaturated fatty acids in Saccharomyces 
cerevisiae causes a global transcriptional response resulting in reduced proteasomal 
activity and increased oxidative stress.

Full Description: Due to their health benefits there is much interest in developing microbial processes for 
efficient production of polyunsaturated fatty acids (PUFAs). In this study we co-expressed
Mucor rouxii Delta(12) - and Delta(6) -desaturase genes in Saccharomyces cerevisiae, which
resulted in a yeast strain that accumulated linoleic acid and gamma-linolenic acid in the 
different lipid species. Additionally, the strain contained higher levels of phospholipids
and lower levels of ergosterol than the reference strain. Integrated analysis of the  
transcriptome revealed decreased expression of genes involved in ergosterol biosynthesis, 
but more unexpectedly it also pointed towards attenuated activity of the ubiquitin-
proteasome system and a reduced oxidative stress response. In vitro and in vivo 
measurements showed reduced levels of all three proteasomal activities and also increased 
levels of reactive oxidative species in the PUFA-producing strain. Overall our results 
clearly show that PUFAs in yeast can be detrimental for several key cellular pathways, 
such as the oxidative stress response and proteasomal activity, suggesting that the 
membrane composition is of vital importance for these processes.

PMID: 21184438

GEO ID: GSE20155.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE20155.final.pcl	Heterologous production of polyunsaturated fatty acids in Saccharomyces cerevisiae causes a global transcriptional response resulting in reduced proteasomal activity and increased oxidative stress.	9	lipid metabolism	1

File last modified: Thu Apr 25 2024

---


# Sen_2011_PMID_21835776

Citation: Sen P, Ghosh S, Pugh BF, Bartholomew B. 2011. A new, highly conserved domain in Swi2/Snf2 
is required for SWI/SNF remodeling.

Full Description: SWI/SNF is an ATP-dependent remodeler that mobilizes nucleosomes and has important roles 
in gene regulation. The catalytic subunit of SWI/SNF has an ATP-dependent DNA translocase 
domain that is essential for remodeling. Besides the DNA translocase domain there are 
other domains in the catalytic subunit of SWI/SNF that have important roles in mobilizing 
nucleosomes. One of these domains, termed SnAC (Snf2 ATP Coupling), is conserved in all 
eukaryotic SWI/SNF  complexes and is located between the ATPase and A-T hook domains. Here
, we show that the SnAC domain is essential for SWI/SNF activity. The SnAC domain is not 
required for SWI/SNF complex integrity, efficient nucleosome binding, or recruitment by 
acidic transcription activators. The SnAC domain is however required in vivo for 
transcription regulation by SWI/SNF as seen by alternative carbon source growth assays, 
northern analysis, and genome-wide expression profiling. The ATPase and nucleosome 
mobilizing activities of SWI/SNF are severely affected when the SnAC domain is removed or 
mutated. The SnAC domain positively regulates the catalytic activity of the ATPase domain 
of SWI/SNF to hydrolyze ATP without significantly affecting its affinity for ATP.

PMID: 21835776

GEO ID: GSE30675.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE30675.final.pcl	A new, highly conserved domain in Swi2/Snf2 is required for SWI/SNF remodeling.	4	chromatin organization	2

File last modified: Thu Apr 25 2024

---


# Shalem_2011_PMID_21931566

Citation: Shalem O, Groisman B, Choder M, Dahan O, Pilpel Y. 2011. Transcriptome kinetics is 
governed by a genome-wide coupling of mRNA production and degradation: a role for RNA Pol 
II.

Full Description: Transcriptome dynamics is governed by two opposing processes, mRNA production and 
degradation. Recent studies found that changes in these processes are frequently  
coordinated and that the relationship between them shapes transcriptome kinetics. 
Specifically, when transcription changes are counter-acted with changes in mRNA stability
, transient fast-relaxing transcriptome kinetics is observed. A possible molecular 
mechanism underlying such coordinated regulation might lay in two RNA polymerase (Pol II) 
subunits, Rpb4 and Rpb7, which are recruited to mRNAs during  transcription and later 
affect their degradation in the cytoplasm. Here we used a yeast strain carrying a mutant 
Pol II which poorly recruits these subunits. We show that this mutant strain is impaired 
in its ability to modulate mRNA stability in response to stress. The normal negative 
coordinated regulation is lost in the mutant, resulting in abnormal transcriptome profiles
both with respect to magnitude and kinetics of responses. These results reveal an 
important role for Pol II, in regulation of both mRNA synthesis and degradation, and also 
in coordinating between them. We propose a simple model for production-degradation 
coupling that accounts for our observations. The model shows how a simple manipulation of 
the rates of co-transcriptional mRNA imprinting by Pol II may govern genome-wide 
transcriptome kinetics in response to environmental changes.

PMID: 21931566

GEO ID: GSE26829

PCL filename	short description	# conditions	tags	# channels
GSE26829_final.pcl	Transcriptome kinetics is governed by a genome-wide coupling of mRNA production and degradation: a role for RNA Pol II.	58	DNA damage stimulus|oxidative stress|RNA catabolism	1

File last modified: Thu Apr 25 2024

---


# Suzuki_2011_PMID_21648421

Citation: Suzuki T, Iwahashi Y. 2011. Gene expression profiles of yeast Saccharomyces cerevisiae 
sod1 caused by patulin toxicity and evaluation of recovery potential of ascorbic acid.

Full Description: Patulin (PAT) is a fungal secondary metabolite and exhibits various toxicities including 
DNA damage and oxidative stress. These toxicities are eased by ascorbic acid (AsA). 
Although a number of studies regarding the mitigating effect of AsA against PAT toxicity 
have been reported, a comprehensive study about gene expressions is currently underway. 
Here, we carried out a detailed evaluation of  PAT toxicity by co-incubation with AsA 
using the superoxide dismutase (SOD) mutant. DNA microarray results extracted the 
alterations in iron transporter and  Fe/S cluster assembly genes; some of the genes that 
constitute the cellular iron  transporter systems remained dysfunctional even in the 
presence of AsA. Meanwhile, AsA treatment reduced the alterations of G1/S phase cell cycle
regulation genes. These results suggest that oxidative stress-derived DNA damage  still 
exists, although AsA treatment effectively reduces PAT toxicity. This implies that a 
combined condition is required for complete blockade of PAT toxicity.

PMID: 21648421

GEO ID: GSE26171.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE26171.final.pcl	Gene expression profiles of yeast Saccharomyces cerevisiae sod1 caused by patulin toxicity and evaluation of recovery potential of ascorbic acid.	12	chemical stimulus|stress	1

File last modified: Thu Apr 25 2024

---


# Takahashi_2011_PMID_21474073

Citation: Takahashi YH, Schulze JM, Jackson J, Hentrich T, Seidel C, Jaspersen SL, Kobor MS, 
Shilatifard A. 2011. Dot1 and histone H3K79 methylation in natural telomeric and HM 
silencing.

Full Description: The expression of genes residing near telomeres is attenuated through telomere position-
effect variegation (TPEV). By using a URA3 reporter located at TEL-VII-L of Saccharomyces 
cerevisiae, it was proposed that the disruptor of telomeric silencing-1 (Dot1) regulates 
TPEV by catalyzing H3K79 methylation. URA3 reporter  assays also indicated that H3K79 
methylation is required for HM silencing. Surprisingly, a genome-wide expression analysis 
of H3K79 methylation-defective mutants identified only a few telomeric genes, such as 
COS12 at TEL-VII-L, to be  subject to H3K79 methylation-dependent natural silencing. 
Consistently, loss of Dot1 did not globally alter Sir2 or Sir3 occupancy in subtelomeric 
regions, but only led to some telomere-specific changes. Furthermore, H3K79 methylation by
Dot1 did not play a role in the maintenance of natural HML silencing. Therefore,  commonly
used URA3 reporter assays may not report on natural PEV, and therefore,  studies 
concerning the epigenetic mechanism of silencing in yeast should also employ assays 
reporting on natural gene expression patterns.

PMID: 21474073

GEO ID: GSE27234

PCL filename	short description	# conditions	tags	# channels
GSE27234_final.pcl	Dot1 and histone H3K79 methylation in natural telomeric and HM silencing.	8	histone modification	1

File last modified: Thu Apr 25 2024

---


# Tauber_2011_PMID_21044956

Citation: Tauber E, Miller-Fleming L, Mason RP, Kwan W, Clapp J, Butler NJ, Outeiro TF, Muchowski PJ,
Giorgini F. 2011. Functional gene expression profiling in yeast implicates translational 
dysfunction in mutant huntingtin toxicity.

Full Description: Huntington disease (HD) is a neurodegenerative disorder caused by the expansion of a 
polyglutamine tract in the huntingtin (htt) protein. To uncover candidate therapeutic 
targets and networks involved in pathogenesis, we integrated gene expression profiling and
functional genetic screening to identify genes critical  for mutant htt toxicity in yeast
. Using mRNA profiling, we have identified genes  differentially expressed in wild-type 
yeast in response to mutant htt toxicity as well as in three toxicity suppressor strains: 
bna4Delta, mbf1Delta, and ume1Delta. BNA4 encodes the yeast homolog of kynurenine 3-
monooxygenase, a promising drug target for HD. Intriguingly, despite playing diverse 
cellular roles, these three suppressors share common differentially expressed genes 
involved in stress response, translation elongation, and mitochondrial transport. We then 
systematically tested the ability of the differentially expressed genes to suppress mutant
htt toxicity when overexpressed and have thereby identified 12 novel suppressors, 
including genes that play a role in stress response, Golgi to  endosome transport, and 
rRNA processing. Integrating the mRNA profiling data and  the genetic screening data, we 
have generated a robust network that shows enrichment in genes involved in rRNA processing
and ribosome biogenesis. Strikingly, these observations implicate dysfunction of 
translation in the pathology of HD. Recent work has shown that regulation of translation 
is critical for life span extension in Drosophila and that manipulation of this process is
protective in Parkinson disease models. In total, these observations suggest that 
pharmacological manipulation of translation may have therapeutic value in HD.

PMID: 21044956

GEO ID: GSE18644.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE18644.final.pcl	Functional gene expression profiling in yeast implicates translational dysfunction in mutant huntingtin toxicity.	15	disease	1

File last modified: Thu Apr 25 2024

---


# Thibault_2011_PMID_22143797

Citation: Thibault G, Ismail N, Ng DT. 2011. The unfolded protein response supports cellular 
robustness as a broad-spectrum compensatory pathway.

Full Description: Stress pathways monitor intracellular systems and deploy a range of regulatory mechanisms 
in response to stress. One of the best-characterized pathways, the unfolded protein 
response (UPR), is responsible for maintaining endoplasmic reticulum (ER) homeostasis. The
highly conserved Ire1 branch regulates hundreds of gene targets by activating a UPR-
specific transcription factor. To understand  how the UPR manages ER stress, a unique 
genetic approach was applied to reveal how the system corrects disequilibria. The data 
show that the UPR can address a wide range of dysfunctions that are otherwise lethal if 
not for its intervention. Transcriptional profiling of stress-alleviated cells shows that 
the program can be modulated, not just in signal amplitude, but also through differential 
target  gene expression depending on the stress. The breadth of the functions mitigated by
the UPR further supports its role as a major mechanism maintaining systems robustness.

PMID: 22143797

GEO ID: GSE33844

PCL filename	short description	# conditions	tags	# channels
GSE33844_final.pcl	The unfolded protein response supports cellular robustness as a broad-spectrum compensatory pathway.	15	response to unfolded protein	2

File last modified: Thu Apr 25 2024

---


# Tu_2011_PMID_21469193

Citation: Tu WY, Huang YC, Liu LF, Chang LH, Tam MF. 2011. Rpl12p affects the transcription of the 
PHO pathway high-affinity inorganic phosphate transporters and repressible phosphatases.

Full Description: The ribosomal protein Rpl12p of Saccharomyces cerevisiae is encoded by duplicated genes, 
RPL12A and RPL12B. The gene products possess an identical amino acid sequence. Yeast 
strain 6EA1, which lacks both genes, is viable but exhibits a very slow-growth phenotype. 
In this study, 6EA1 cells were transformed with plasmids carrying either RPL12A or RPL12B
, and the transcriptional profiles of wild-type W303, 6EA1 and the transformed cells grown
in synthetic complete medium were examined by microarray analysis. Transcription of PHO84
, a gene encoding a high-affinity phosphate transporter, was drastically suppressed in 
6EA1. PHO84 expression is induced under phosphate-limiting conditions. Therefore, cells 
were  grown in low-phosphate medium and transcripts encoding the PHO pathway proteins were
quantified by qRT-PCR. The high-affinity phosphate transporters and repressible 
phosphatases were suppressed, while PHO4, a PHO pathway transcription activator, was 
upregulated in 6EA1. Accordingly, phosphate transport and acidic phosphatase activities 
were significantly decreased in 6EA1. Addition of RPL12A or RPL12B to 6EA1 largely lessens
these effects. We postulate that RPL12 has an extra-ribosomal function in modulating the 
transcription of genes that need Pho4p activation.

PMID: 21469193

GEO ID: GSE22644.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE22644.final.pcl	Rpl12p affects the transcription of the PHO pathway high-affinity inorganic phosphate transporters and repressible phosphatases.	8	phosphorus utilization	1

File last modified: Thu Apr 25 2024

---


# Uluisik_2011_PMID_22114689

Citation: Uluisik I, Kaya A, Fomenko DE, Karakaya HC, Carlson BA, Gladyshev VN, Koc A. 2011. Boron 
stress activates the general amino acid control mechanism and inhibits protein synthesis.

Full Description: Boron is an essential micronutrient for plants, and it is beneficial for animals. However
, at high concentrations boron is toxic to cells although the mechanism of this toxicity 
is not known. Atr1 has recently been identified as a boron efflux pump whose expression is
upregulated in response to boron treatment. Here, we found that the expression of ATR1 is 
associated with expression of genes involved in amino acid biosynthesis. These mechanisms 
are strictly controlled by the transcription factor Gcn4 in response to boron treatment. 
Further analyses have shown that boron impaired protein synthesis by promoting 
phosphorylation of eIF2alpha in a Gcn2 kinase dependent manner. The uncharged tRNA binding
domain (HisRS) of Gcn2 is necessary for the phosphorylation of eIF2alpha in the presence 
of boron. We postulate that boron exerts its toxic effect through activation of the 
general amino acid control system and inhibition of protein synthesis. Since  the general 
amino acid control pathway is conserved among eukaryotes, this mechanism of boron toxicity
may be of general importance.

PMID: 22114689

GEO ID: GSE29887.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE29887.final.pcl	Boron stress activates the general amino acid control mechanism and inhibits protein synthesis.	3	chemical stimulus|stress	2

File last modified: Thu Apr 25 2024

---


# Verzijlbergen_2011_PMID_21998594

Citation: Verzijlbergen KF, van Welsem T, Sie D, Lenstra TL, Turner DJ, Holstege FC, Kerkhoven RM, 
van Leeuwen F. 2011. A barcode screen for epigenetic regulators reveals a role for the 
NuB4/HAT-B histone acetyltransferase complex in histone turnover.

Full Description: Dynamic modification of histone proteins plays a key role in regulating gene expression. 
However, histones themselves can also be dynamic, which potentially affects the stability 
of histone modifications. To determine the molecular mechanisms of histone turnover, we 
developed a parallel screening method for epigenetic regulators by analyzing chromatin 
states on DNA barcodes. Histone turnover was quantified by employing a genetic pulse-chase
technique called RITE, which was combined with chromatin immunoprecipitation and high-
throughput sequencing. In this screen, the NuB4/HAT-B complex, containing the conserved 
type B histone acetyltransferase Hat1, was found to promote histone turnover. Unexpectedly
, the three members of this complex could be functionally separated from each other as 
well as from the known interacting factor and histone chaperone Asf1. Thus, systematic and
direct interrogation of chromatin structure  on DNA barcodes can lead to the discovery of 
genes and pathways involved in chromatin modification and dynamics.

PMID: 21998594

GEO ID: GSE30168

PCL filename	short description	# conditions	tags	# channels
GSE30168_final.pcl	A barcode screen for epigenetic regulators reveals a role for the NuB4/HAT-B histone acetyltransferase complex in histone turnover.	36	histone modification|chromatin organization	2

File last modified: Thu Apr 25 2024

---


# Wan_2011_PMID_21763693

Citation: Wan Y, Chen W, Xing J, Tan J, Li B, Chen H, Lin Z, Chiang JH, Saleem RA. 2011. 
Transcriptome profiling reveals a novel role for trichostatin A in antagonizing histone 
chaperone Chz1 mediated telomere anti-silencing.

Full Description: The histone chaperones play an important role in chromatin assembly and disassembly during
replication and transcription. We have assessed the global roles of histone chaperones in 
Saccharomyces cerevisiae. Microarray transcriptional analyzes indicate that histone 
chaperones have their own specific target genes, and various histone chaperones have 
partially overlapping functions during transcriptional regulation. The histone deacetylase
inhibitor TSA and histone chaperones Asf1, Vps75 and Rtt106 can function in parallel 
pathways to regulate transcription. Moreover, TSA can specifically antagonize histone 
chaperone Chz1-mediated telomere anti-silencing. This study demonstrates that a mutual 
cross-talk mechanism exists between histone chaperones and histone deacetylation in 
transcriptional regulation.

PMID: 21763693

GEO ID: GSE30228.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE30228.final.pcl	Transcriptome profiling reveals a novel role for trichostatin A in antagonizing histone chaperone Chz1 mediated telomere anti-silencing.	11	chromatin organization	2

File last modified: Thu Apr 25 2024

---


# Wenger_2011_PMID_21829391

Citation: Wenger JW, Piotrowski J, Nagarajan S, Chiotti K, Sherlock G, Rosenzweig F. 2011. Hunger 
artists: yeast adapted to carbon limitation show trade-offs under carbon sufficiency.

Full Description: As organisms adaptively evolve to a new environment, selection results in the improvement 
of certain traits, bringing about an increase in fitness. Trade-offs  may result from this
process if function in other traits is reduced in alternative environments either by the 
adaptive mutations themselves or by the accumulation of neutral mutations elsewhere in the
genome. Though the cost of adaptation has long been a fundamental premise in evolutionary 
biology, the existence of and molecular basis for trade-offs in alternative environments 
are not well-established. Here, we show that yeast evolved under aerobic glucose 
limitation show surprisingly few trade-offs when cultured in other carbon-limited 
environments, under either aerobic or anaerobic conditions. However, while adaptive clones
consistently outperform their common ancestor under carbon limiting conditions, in some 
cases they perform less well than their ancestor in  aerobic, carbon-rich environments, 
indicating that trade-offs can appear when resources are non-limiting. To more deeply 
understand how adaptation to one condition affects performance in others, we determined 
steady-state transcript abundance of adaptive clones grown under diverse conditions and 
performed whole-genome sequencing to identify mutations that distinguish them from one 
another and from their common ancestor. We identified mutations in genes involved in 
glucose sensing, signaling, and transport, which, when considered in the context of the 
expression data, help explain their adaptation to carbon poor environments. However, 
different sets of mutations in each independently evolved  clone indicate that multiple 
mutational paths lead to the adaptive phenotype. We  conclude that yeasts that evolve high
fitness under one resource-limiting condition also become more fit under other resource-
limiting conditions, but may  pay a fitness cost when those same resources are abundant.

PMID: 21829391

GEO ID: GSE25081

PCL filename	short description	# conditions	tags	# channels
GSE25081_final.pcl	Hunger artists: yeast adapted to carbon limitation show trade-offs under carbon sufficiency.	36	oxygen level alteration|carbon utilization|fermentation|respiration	2

File last modified: Thu Apr 25 2024

---


# Zhang_2011_PMID_22068328

Citation: Zhang J, Vaga S, Chumnanpuen P, Kumar R, Vemuri GN, Aebersold R, Nielsen J. 2011. Mapping 
the interaction of Snf1 with TORC1 in Saccharomyces cerevisiae.

Full Description: Nutrient sensing and coordination of metabolic pathways are crucial functions for all 
living cells, but details of the coordination under different environmental conditions 
remain elusive. We therefore undertook a systems biology approach to investigate the 
interactions between the Snf1 and the target of rapamycin complex 1 (TORC1) in 
Saccharomyces cerevisiae. We show that Snf1 regulates a much broader range of biological 
processes compared with TORC1 under both glucose- and ammonium-limited conditions. We also
find that Snf1 has a role in upregulating the NADP(+)-dependent glutamate dehydrogenase (
encoded by GDH3) under derepressing condition, and therefore may also have a role in 
ammonium assimilation and amino-acid biosynthesis, which can be considered as a 
convergence of Snf1 and TORC1 pathways. In addition to the accepted role of Snf1  in 
regulating fatty acid (FA) metabolism, we show that TORC1 also regulates FA metabolism, 
likely through modulating the peroxisome and beta-oxidation. Finally, we conclude that 
direct interactions between Snf1 and TORC1 pathways are unlikely under nutrient-limited 
conditions and propose that TORC1 is repressed in a manner that is independent of Snf1.

PMID: 22068328

GEO ID: GSE24421

PCL filename	short description	# conditions	tags	# channels
GSE24421_final.pcl	Mapping the interaction of Snf1 with TORC1 in Saccharomyces cerevisiae.	24	carbon utilization|nitrogen utilization|signaling	1

File last modified: Thu Apr 25 2024

---


# Zhou_2011_PMID_21700227

Citation: Zhou X, O'Shea EK. 2011. Integrated approaches reveal determinants of genome-wide binding 
and function of  the transcription factor Pho4.

Full Description: DNA sequences with high affinity for transcription factors occur more frequently  in the 
genome than instances of genes bound or regulated by these factors. It is  not clear what 
factors determine the genome-wide pattern of binding or regulation for a given 
transcription factor. We used an integrated approach to study how trans influences shape 
the binding and regulatory landscape of Pho4, a budding yeast transcription factor 
activated in response to phosphate limitation. We find that nucleosomes significantly 
restrict Pho4 binding. At nucleosome-depleted sites, competition from another 
transcription factor, Cbf1, determines Pho4 occupancy, raising the threshold for 
transcriptional activation in phosphate replete conditions and preventing Pho4 activation 
of genes outside the phosphate  regulon during phosphate starvation. Pho4 binding is not 
sufficient for transcriptional activation-a cooperative interaction between Pho2 and Pho4 
specifies genes that are activated. Combining these experimental observations, we are able
to globally predict Pho4 binding and its functionality.

PMID: 21700227

GEO ID: GSE23580

PCL filename	short description	# conditions	tags	# channels
GSE23580_final.pcl	Integrated approaches reveal determinants of genome-wide binding and function of  the transcription factor Pho4.	66	oxygen level alteration|phosphorus utilization|transcription	2

File last modified: Thu Apr 25 2024

---


# de_Kok_2011_PMID_21684346

Citation: de Kok S, Yilmaz D, Suir E, Pronk JT, Daran JM, van Maris AJ. 2011. Increasing free-energy
(ATP) conservation in maltose-grown Saccharomyces cerevisiae by expression of a 
heterologous maltose phosphorylase.

Full Description: Increasing free-energy conservation from the conversion of substrate into product is 
crucial for further development of many biotechnological processes. In theory, replacing 
the hydrolysis of disaccharides by a phosphorolytic cleavage reaction provides an 
opportunity to increase the ATP yield on the disaccharide. To test this concept, we first 
deleted the native maltose metabolism genes in Saccharomyces cerevisiae. The knockout 
strain showed no maltose-transport activity and a very low residual maltase activity (0.03
mumol mg protein(-1)min(-1)). Expression of a maltose phosphorylase gene from 
Lactobacillus sanfranciscensis and the MAL11 maltose-transporter gene resulted in 
relatively slow growth (mu(aerobic) 0.09 +/- 0.03 h(-1)). Co-expression of Lactococcus 
lactis beta-phosphoglucomutase accelerated maltose utilization via this route (mu(aerobic
) 0.21 +/- 0.01 h(-1), mu(anaerobic) 0.10 +/- 0.00 h(-1)).  Replacing maltose hydrolysis 
with phosphorolysis increased the anaerobic biomass  yield on maltose in anaerobic maltose
-limited chemostat cultures by 26%, thus demonstrating the potential of phosphorolysis to 
improve the free-energy conservation of disaccharide metabolism in industrial 
microorganisms.

PMID: 21684346

GEO ID: GSE30535

PCL filename	short description	# conditions	tags	# channels
GSE30535_s13.final.pcl	Increasing free-energy (ATP) conservation in maltose-grown Saccharomyces cerevisiae by expression of a heterologous maltose phosphorylase.	13	fermentation|carbon utilization	1

File last modified: Thu Apr 25 2024

---


# van_Leeuwen_2011_PMID_21724882

Citation: van Leeuwen JS, Vermeulen NP, Vos JC. 2011. Involvement of the pleiotropic drug resistance
response, protein kinase C signaling, and altered zinc homeostasis in resistance of 
Saccharomyces cerevisiae to diclofenac.

Full Description: Diclofenac is a widely used analgesic drug that can cause serious adverse drug reactions. 
We used Saccharomyces cerevisiae as a model eukaryote with which to elucidate the 
molecular mechanisms of diclofenac toxicity and resistance. Although most yeast cells died
during the initial diclofenac treatment, some survived and started growing again. 
Microarray analysis of the adapted cells identified three major processes involved in 
diclofenac detoxification and tolerance. In particular, pleiotropic drug resistance (PDR) 
genes and genes under the control of Rlm1p, a transcription factor in the protein kinase C
(PKC) pathway, were upregulated in diclofenac-adapted cells. We tested if these processes 
or pathways were directly involved in diclofenac toxicity or resistance. Of the 
pleiotropic drug resistance gene products, the multidrug transporter Pdr5p was crucially 
important for diclofenac tolerance. Furthermore,  deletion of components of the cell wall 
stress-responsive PKC pathway increased diclofenac toxicity, whereas incubation of cells 
with the cell wall stressor calcofluor white before the addition of diclofenac decreased 
its toxicity. Also,  diclofenac induced flocculation, which might trigger the cell wall 
alterations. Genes involved in ribosome biogenesis and rRNA processing were downregulated
, as  were zinc-responsive genes. Paradoxically, deletion of the zinc-responsive 
transcription factor Zap1p or addition of the zinc chelator 1,10-phenanthroline 
significantly increased diclofenac toxicity, establishing a regulatory role for zinc in 
diclofenac resistance. In conclusion, we have identified three new pathways involved in 
diclofenac tolerance in yeast, namely, Pdr5p as the main contributor to the PDR response, 
cell wall signaling via the PKC pathway, and zinc homeostasis, regulated by Zap1p.

PMID: 21724882

GEO ID: GSE29331.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE29331.final.pcl	Involvement of the pleiotropic drug resistance response, protein kinase C signaling, and altered zinc homeostasis in resistance of Saccharomyces cerevisiae to diclofenac.	10	chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Albulescu_2012_PMID_22479188

Citation: Albulescu LO, Sabet N, Gudipati M, Stepankiw N, Bergman ZJ, Huffaker TC, Pleiss JA. 2012. 
A quantitative, high-throughput reverse genetic screen reveals novel connections  between 
Pre-mRNA splicing and 5' and 3' end transcript determinants.

Full Description: Here we present the development and implementation of a genome-wide reverse genetic screen
in the budding yeast, Saccharomyces cerevisiae, that couples high-throughput strain growth
, robotic RNA isolation and cDNA synthesis, and quantitative PCR to allow for a robust 
determination of the level of nearly any cellular RNA in the background of ~5,500 
different mutants. As an initial test of this approach, we sought to identify the full 
complement of factors that impact pre-mRNA splicing. Increasing lines of evidence suggest 
a relationship between pre-mRNA splicing and other cellular pathways including chromatin 
remodeling, transcription, and 3' end processing, yet in many cases the specific proteins 
responsible for functionally connecting these pathways remain unclear. Moreover,  it is 
unclear whether all pathways that are coupled to splicing have been identified. As 
expected, our approach sensitively detects pre-mRNA accumulation in the vast majority of 
strains containing mutations in known splicing factors. Remarkably, however, several 
additional candidates were found to cause increases  in pre-mRNA levels similar to that 
seen for canonical splicing mutants, none of which had previously been implicated in the 
splicing pathway. Instead, several of these factors have been previously implicated to 
play roles in chromatin remodeling, 3' end processing, and other novel categories. Further
analysis of these factors using splicing-sensitive microarrays confirms that deletion of 
Bdf1, a factor that links transcription initiation and chromatin remodeling, leads to a 
global splicing defect, providing evidence for a novel connection between pre-mRNA 
splicing and this component of the SWR1 complex. By contrast, mutations in 3' end 
processing factors such as Cft2 and Yth1 also result in pre-mRNA splicing defects, 
although only for a subset of transcripts, suggesting  that spliceosome assembly in S. 
cerevisiae may more closely resemble mammalian models of exon-definition. More broadly, 
our work demonstrates the capacity of this approach to identify novel regulators of 
various cellular RNAs.

PMID: 22479188

GEO ID: GSE34330GPL8154.sfp.pcl

PCL filename	short description	# conditions	tags	# channels
GSE34330GPL8154.sfp.pcl	A quantitative, high-throughput reverse genetic screen reveals novel connections  between Pre-mRNA splicing and 5' and 3' end transcript determinants.	12	mRNA processing|chromatin organization	2

File last modified: Thu Apr 25 2024

---


# Ansari_2012_PMID_21971086

Citation: Ansari SA, Ganapathi M, Benschop JJ, Holstege FC, Wade JT, Morse RH. 2012. Distinct role 
of Mediator tail module in regulation of SAGA-dependent, TATA-containing genes in yeast.

Full Description: The evolutionarily conserved Mediator complex is required for transcription of nearly all 
RNA Pol II-dependent promoters, with the tail module serving to recruit Mediator to active
promoters in current models. However, transcriptional  dependence on tail module subunits 
varies in a gene-specific manner, and the generality of the tail module requirement for 
transcriptional activation has not  been explored. Here, we show that tail module subunits
function redundantly to recruit Mediator to promoters in yeast, and transcriptome analysis
shows stronger effects on genome-wide expression in a double-tail subunit deletion mutant 
than in single-subunit deletion mutants. Unexpectedly, TATA-containing and SAGA-dependent 
genes were much more affected by impairment of tail module function than were TFIID-
dependent genes. Consistent with this finding, Mediator  and preinitiation complex 
association with SAGA-dependent promoters is substantially reduced in gal11/med15Delta 
med3Delta yeast, whereas association of TBP, Pol II, and other Mediator modules with TFIID
-dependent genes is largely independent of the tail module. Thus, we have identified a 
connection between the Mediator tail module and the division of promoter dependence 
between TFIID and SAGA.

PMID: 21971086

GEO ID: GSE31774

PCL filename	short description	# conditions	tags	# channels
GSE31774_final.pcl	Distinct role of Mediator tail module in regulation of SAGA-dependent, TATA-containing genes in yeast.	21	transcription	2

File last modified: Thu Apr 25 2024

---


# Apweiler_2012_PMID_22697265

Citation: Apweiler E, Sameith K, Margaritis T, Brabers N, van de Pasch L, Bakker LV, van Leenen D, 
Holstege FC, Kemmeren P. 2012. Yeast glucose pathways converge on the transcriptional 
regulation of trehalose biosynthesis.

Full Description: BACKGROUND: Cellular glucose availability is crucial for the functioning of most  
biological processes. Our understanding of the glucose regulatory system has been greatly 
advanced by studying the model organism Saccharomyces cerevisiae, but many aspects of this
system remain elusive. To understand the organisation of the glucose regulatory system, we
analysed 91 deletion mutants of the different glucose signalling and metabolic pathways in
Saccharomyces cerevisiae using DNA microarrays. RESULTS: In general, the mutations do not 
induce pathway-specific transcriptional responses. Instead, one main transcriptional 
response is discerned, which varies in direction to mimic either a high or a low glucose 
response. Detailed analysis uncovers established and new relationships within and between 
individual pathways and their members. In contrast to signalling components, metabolic 
components of the glucose regulatory system are transcriptionally more frequently affected
. A new network approach is applied that exposes the hierarchical organisation of the 
glucose regulatory system. CONCLUSIONS: The tight interconnection between the different 
pathways of the glucose regulatory system is reflected by the main transcriptional 
response observed. Tps2 and Tsl1, two enzymes involved in the biosynthesis of the storage
carbohydrate trehalose, are predicted to be the most downstream transcriptional components
. Epistasis analysis of tps2Delta double mutants supports this prediction. Although based 
on transcriptional changes only, these results suggest that all changes in perceived 
glucose levels ultimately lead to a shift in trehalose biosynthesis.

PMID: 22697265

GEO ID: GSE33098

PCL filename	short description	# conditions	tags	# channels
GSE33098_final.pcl	Yeast glucose pathways converge on the transcriptional regulation of trehalose biosynthesis.	18	carbon utilization	2

File last modified: Thu Apr 25 2024
GSE33097_s257_final.pcl	Yeast glucose pathways converge on the transcriptional regulation of trehalose biosynthesis.	257	carbon utilization|signaling	2

File last modified: Thu Apr 25 2024

---


# Arana_2012_PMID_23079308

Citation: Arana ME, Kerns RT, Wharey L, Gerrish KE, Bushel PR, Kunkel TA. 2012. Transcriptional 
responses to loss of RNase H2 in Saccharomyces cerevisiae.

Full Description: We report here the transcriptional responses in Saccharomyces cerevisiae to deletion of 
the RNH201 gene encoding the catalytic subunit of RNase H2. Deleting  RNH201 alters RNA 
expression of 349 genes by >/=1.5-fold (q-value <0.01), of which 123 are upregulated and 
226 are downregulated. Differentially expressed genes (DEGs) include those involved in 
stress responses and genome maintenance, consistent with a role for RNase H2 in removing 
ribonucleotides incorporated into DNA during replication. Upregulated genes include 
several that encode subunits of RNA polymerases I and III, and genes involved in ribosomal
RNA processing, ribosomal biogenesis and tRNA modification and processing, supporting a 
role for  RNase H2 in resolving R-loops formed during transcription of rRNA and tRNA genes
. A role in R-loop resolution is further suggested by a higher average GC-content proximal
to the transcription start site of downregulated as compared to upregulated genes. Several
DEGs are involved in telomere maintenance, supporting  a role for RNase H2 in resolving 
RNA-DNA hybrids formed at telomeres. A large number of DEGs encode nucleases, helicases 
and genes involved in response to dsRNA viruses, observations that could be relevant to 
the nucleic acid species that elicit an innate immune response in RNase H2-defective 
humans.

PMID: 23079308

GEO ID: GSE34668.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE34668.final.pcl	Transcriptional responses to loss of RNase H2 in Saccharomyces cerevisiae.	14	RNA catabolism	1

File last modified: Thu Apr 25 2024

---


# Barreto_2012_PMID_23039231

Citation: Barreto L, Canadell D, Valverde-Saubi D, Casamayor A, Arino J. 2012. The short-term 
response of yeast to potassium starvation.

Full Description: Potassium is the major intracellular cation in most living cells, including yeasts. 
Although K(+) has been demonstrated to be necessary for diverse cellular  functions, such 
as enzyme activation, additional, still uncharacterized cellular  targets may exist. We 
show here that in Saccharomyces cerevisiae short-term potassium deprivation impacts in the
mRNA level of over one thousand genes. Lack  of potassium drastically alters sulfur 
metabolism (mainly Met and Cys metabolism), triggers an oxidative stress response and 
activates the retrograde pathway, possibly due to the ammonium accumulation that occurs 
through the Trk1 potassium transporter. We also observe a remarkable halt in the 
expression of genes required for ribosome biogenesis and translation, a decrease in 
expression  of diverse components (cyclins, protein kinases) required for progression 
through the cell cycle and a blockage in septins assembly. Only specific subsets of these 
changes were observed in a strain deleted for the TRK1 and TRK2 genes growing in  the 
presence of sufficient potassium (50 mM). Therefore, a shortage of potassium  in the 
environment triggers an acute transcriptional response, which covers different aspects of 
the cell biology so far unexplored, and whose investigation  will likely reveal novel 
functional roles for this cation.

PMID: 23039231

GEO ID: GSE24712

PCL filename	short description	# conditions	tags	# channels
GSE24712_final.pcl	The short-term response of yeast to potassium starvation.	18	cellular ion homeostasis|starvation	2

File last modified: Thu Apr 25 2024

---


# Bester_2012_PMID_22384390

Citation: Bester MC, Jacobson D, Bauer FF. 2012. Many Saccharomyces cerevisiae Cell Wall Protein 
Encoding Genes Are Coregulated by Mss11, but Cellular Adhesion Phenotypes Appear Only Flo 
Protein Dependent.

Full Description: The outer cell wall of the yeast Saccharomyces cerevisiae serves as the interface with the
surrounding environment and directly affects cell-cell and cell-surface  interactions. 
Many of these interactions are facilitated by specific adhesins that belong to the Flo 
protein family. Flo mannoproteins have been implicated in  phenotypes such as flocculation
, substrate adhesion, biofilm formation, and pseudohyphal growth. Genetic data strongly 
suggest that individual Flo proteins are responsible for many specific cellular adhesion 
phenotypes. However, it remains unclear whether such phenotypes are determined solely by 
the nature of the expressed FLO genes or rather as the result of a combination of FLO gene
expression and other cell wall properties and cell wall proteins. Mss11 has been  shown to
be a central element of FLO1 and FLO11 gene regulation and acts together with the cAMP-PKA
-dependent transcription factor Flo8. Here we use genome-wide transcription analysis to 
identify genes that are directly or indirectly regulated by Mss11. Interestingly, many of 
these genes encode cell wall mannoproteins, in particular, members of the TIR and DAN 
families. To examine whether these genes play a role in the adhesion properties associated
with Mss11  expression, we assessed deletion mutants of these genes in wild-type and 
flo11Delta genetic backgrounds. This analysis shows that only FLO genes, in particular 
FLO1/10/11, appear to significantly impact on such phenotypes. Thus adhesion-related 
phenotypes are primarily dependent on the balance of FLO gene expression.

PMID: 22384390

GEO ID: GSE29371

PCL filename	short description	# conditions	tags	# channels
GSE29371_final.pcl	Many Saccharomyces cerevisiae Cell Wall Protein Encoding Genes Are Coregulated by Mss11, but Cellular Adhesion Phenotypes Appear Only Flo Protein Dependent.	12	transcription	1

File last modified: Thu Apr 25 2024
GSE17716_final.pcl	Many Saccharomyces cerevisiae Cell Wall Protein Encoding Genes Are Coregulated by Mss11, but Cellular Adhesion Phenotypes Appear Only Flo Protein Dependent.	16	transcription|cell wall organization	1

File last modified: Thu Apr 25 2024

---


# Bose_2012_PMID_22719263

Citation: Bose T, Lee KK, Lu S, Xu B, Harris B, Slaughter B, Unruh J, Garrett A, McDowell W, Box A, 
Li H, Peak A, Ramachandran S, Seidel C, Gerton JL. 2012. Cohesin proteins promote 
ribosomal RNA production and protein translation in yeast and human cells.

Full Description: Cohesin is a protein complex known for its essential role in chromosome segregation. 
However, cohesin and associated factors have additional functions in transcription, DNA 
damage repair, and chromosome condensation. The human cohesinopathy diseases are thought 
to stem not from defects in chromosome segregation but from gene expression. The role of 
cohesin in gene expression is not well understood. We used budding yeast strains bearing 
mutations analogous to the human cohesinopathy disease alleles under control of their 
native promoter to study gene expression. These mutations do not significantly affect 
chromosome segregation. Transcriptional profiling reveals that many targets of the 
transcriptional activator Gcn4 are induced in the eco1-W216G mutant background. The 
upregulation of Gcn4 was observed in many cohesin mutants, and this observation suggested 
protein translation was reduced. We demonstrate that the cohesinopathy mutations eco1-
W216G and smc1-Q843Delta are associated with defects in ribosome biogenesis and a 
reduction in the actively translating fraction of ribosomes, eiF2alpha-phosphorylation, 
and (35)S-methionine incorporation, all of  which indicate a deficit in protein 
translation. Metabolic labeling shows that the eco1-W216G and smc1-Q843Delta mutants 
produce less ribosomal RNA, which is expected to constrain ribosome biogenesis. Further 
analysis shows that the production of rRNA from an individual repeat is reduced while copy
number remains unchanged. Similar defects in rRNA production and protein translation are 
observed in a human Roberts syndrome cell line. In addition, cohesion is defective 
specifically at the rDNA locus in the eco1-W216G mutant, as has been previously reported 
for Roberts syndrome. Collectively, our data suggest that cohesin proteins normally 
facilitate production of ribosomal RNA and protein translation, and this is one way they 
can influence gene expression. Reduced translational capacity could contribute to the 
human cohesinopathies.

PMID: 22719263

GEO ID: GSE27235.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE27235.final.pcl	Cohesin proteins promote ribosomal RNA production and protein translation in yeast and human cells.	36	translational regulation|amino acid utilization	1

File last modified: Thu Apr 25 2024

---


# Busti_2012_PMID_21964263

Citation: Busti S, Gotti L, Balestrieri C, Querin L, Drovandi G, Felici G, Mavelli G, Bertolazzi P, 
Alberghina L, Vanoni M. 2012. Overexpression of Far1, a cyclin-dependent kinase inhibitor
induces a large transcriptional reprogramming in which RNA synthesis senses Far1 in a Sfp1
-mediated way.

Full Description: The FAR1 gene encodes an 830 residue bifunctional protein, whose major function is 
inhibition of cyclin-dependent kinase complexes involved in the G1/S transition. FAR1 
transcription is maximal between mitosis and early G1 phase. Enhanced FAR1 transcription 
is necessary but not sufficient for the pheromone-induced G1 arrest, since FAR1 
overexpression itself does not trigger cell cycle arrest. Besides its well established 
role in the response to pheromone, recent evidences suggest that Far1 may also regulate 
the mitotic cell  cycle progression: in particular, it has been proposed that Far1, 
together with the G1 cyclin Cln3, may be part of a cell sizer mechanism that controls the 
entry into S phase. Far1 is an unstable protein throughout the cell cycle except during G1
phase. Far1 levels peak in newborn cells as a consequence of a burst of synthetic activity
at the end of the previous cycle, and the amounts per cell remain roughly constant during 
the G1 phase. Phosphorylation (at serine 87) by Cdk1-Cln complexes primes Far1 for 
ubiquitin-mediated proteolysis. By coupling a  genome-wide transcriptional analysis of 
FAR1-overexpressing and far1Delta cells grown in ethanol- or glucose-supplemented minimal 
media with a range of phenotypic analysis, we show that FAR1 overexpression not only 
coordinately increases RNA and protein accumulation, but induces strong transcriptional 
remodeling, metabolism being the most affected cellular property, suggesting that the Far1
/Cln3 sizer regulates cell growth either directly or indirectly by affecting metabolism 
and pathways known to modulate ribosome biogenesis. A crucial role in mediating the effect
of Far1 overexpression is played by the Sfp1 protein, a key transcriptional regulator of 
ribosome biogenesis, whose presence is mandatory to allow a coordinated increase in both 
RNA and protein levels in ethanol-grown cells.

PMID: 21964263

GEO ID: GSE31143.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE31143.final.pcl	Overexpression of Far1, a cyclin-dependent kinase inhibitor, induces a large transcriptional reprogramming in which RNA synthesis senses Far1 in a Sfp1-mediated way.	18	mitotic cell cycle|signaling|carbon utilization	1

File last modified: Thu Apr 25 2024

---


# Cain_2012_PMID_22095082

Citation: Cain CW, Lohse MB, Homann OR, Sil A, Johnson AD. 2012. A conserved transcriptional 
regulator governs fungal morphology in widely diverged species.

Full Description: Fungi exhibit a large variety of morphological forms. Here, we examine the functions of a 
deeply conserved regulator of morphology in three fungal species:  Saccharomyces 
cerevisiae, Candida albicans, and Histoplasma capsulatum. We show that, despite an 
estimated 600 million years since those species diverged from a  common ancestor, Wor1 in 
C. albicans, Ryp1 in H. capsulatum, and Mit1 in S. cerevisiae are transcriptional 
regulators that recognize the same DNA sequence. Previous work established that Wor1 
regulates white-opaque switching in C. albicans and that its ortholog Ryp1 regulates the 
yeast to mycelial transition in H. capsulatum. Here we show that the ortholog Mit1 in S. 
cerevisiae is also a master regulator of a morphological transition, in this case 
pseudohyphal growth. Full-genome chromatin immunoprecipitation experiments show that Mit1 
binds to the control regions of the previously known regulators of pseudohyphal growth as 
well as those of many additional genes. Through a comparison of binding sites for Mit1 in 
S. cerevisiae, Wor1 in C. albicans, and Wor1 ectopically expressed in S. cerevisiae, we 
conclude that the genes controlled by the orthologous regulators overlap only slightly 
between these two species despite the fact that the DNA binding specificity of the 
regulators has remained largely unchanged. We suggest  that the ancestral Wor1/Mit1/Ryp1 
protein controlled aspects of cell morphology and that movement of genes in and out of the
Wor1/Mit1/Ryp1 regulon is responsible, in part, for the differences of morphological forms
among these species.

PMID: 22095082

GEO ID: GSE32550.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE32550.final.pcl	A conserved transcriptional regulator governs fungal morphology in widely diverged species.	8	evolution|transcriptional regulation	2

File last modified: Thu Apr 25 2024

---


# Cap_2012_PMID_22560924

Citation: Cap M, Stepanek L, Harant K, Vachova L, Palkova Z. 2012. Cell differentiation within a 
yeast colony: metabolic and regulatory parallels with a tumor-affected organism.

Full Description: Nutrient sensing and metabolic reprogramming are crucial for metazoan cell aging  and 
tumor growth. Here, we identify metabolic and regulatory parallels between a  layered, 
multicellular yeast colony and a tumor-affected organism. During development, a yeast 
colony stratifies into U and L cells occupying the upper and lower colony regions, 
respectively. U cells activate a unique metabolism controlled by the glutamine-induced TOR
pathway, amino acid-sensing systems (SPS  and Gcn4p) and signaling from mitochondria with 
lowered respiration. These systems jointly modulate U cell physiology, which adapts to 
nutrient limitations  and utilize the nutrients released from L cells. Stress-resistant U 
cells share metabolic pathways and other similar characteristics with tumor cells, 
including  the ability to proliferate. L cells behave similarly to stressed and starving 
cells, which activate degradative mechanisms to provide nutrients to U cells. Our data 
suggest a nutrient flow between both cell types, resembling the Cori cycle and glutamine-
NH(4)(+) shuttle between tumor and healthy metazoan cells.

PMID: 22560924

GEO ID: GSE35887.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE35887.final.pcl	Cell differentiation within a yeast colony: metabolic and regulatory parallels with a tumor-affected organism.	7	colony morphology|disease|signaling	2

File last modified: Thu Apr 25 2024

---


# Carter_2012_PMID_23071457

Citation: Carter GW, Hays M, Sherman A, Galitski T. 2012. Use of pleiotropy to model genetic 
interactions in a population.

Full Description: Systems-level genetic studies in humans and model systems increasingly involve both high-
resolution genotyping and multi-dimensional quantitative phenotyping. We present a novel 
method to infer and interpret genetic interactions that exploits the complementary 
information in multiple phenotypes. We applied this approach to a population of yeast 
strains with randomly assorted perturbations of five genes involved in mating. We 
quantified pheromone response at the molecular  level and overall mating efficiency. These
phenotypes were jointly analyzed to derive a network of genetic interactions that mapped 
mating-pathway relationships. To determine the distinct biological processes driving the 
phenotypic complementarity, we analyzed patterns of gene expression to find that  the 
pheromone response phenotype is specific to cellular fusion, whereas mating efficiency was
a combined measure of cellular fusion, cell cycle arrest, and modifications in cellular 
metabolism. We applied our novel method to global gene  expression patterns to derive an 
expression-specific interaction network and demonstrate applicability to global transcript
data. Our approach provides a basis for interpretation of genetic interactions and the 
generation of specific hypotheses from populations assayed for multiple phenotypes.

PMID: 23071457

GEO ID: GSE34787

PCL filename	short description	# conditions	tags	# channels
GSE34787_final.pcl	Use of pleiotropy to model genetic interactions in a population.	96	mating	1

File last modified: Thu Apr 25 2024

---


# Casamayor_2012_PMID_22372618

Citation: Casamayor A, Serrano R, Platara M, Casado C, Ruiz A, Arino J. 2012. The role of the Snf1 
kinase in the adaptive response of Saccharomyces cerevisiae  to alkaline pH stress.

Full Description: Alkaline pH stress invokes a potent and fast transcriptional response in Saccharomyces 
cerevisiae that includes many genes repressed by glucose. Certain mutants in the glucose-
sensing and -response pathways, such as those lacking the  Snf1 kinase, are sensitive to 
alkalinization. In the present study we show that the addition of glucose to the medium 
improves the growth of wild-type cells at high pH, fully abolishes the snf1 alkali-
sensitive phenotype and attenuates high  pH-induced Snf1 phosphorylation at Thr(210). Lack
of Elm1, one of the three upstream Snf1 kinases (Tos3, Elm1 and Sak1), markedly increases 
alkali sensitivity, whereas the phenotype of the triple mutant tos3 elm1 sak1 is even more
pronounced than that of snf1 cells and is poorly rescued by glucose supplementation. DNA 
microarray analysis reveals that about 75% of the genes induced in the short term by high 
pH are also induced by glucose scarcity. Snf1 mediates, in full or in part, the activation
of a significant subset (38%) of short-term alkali-induced genes, including those encoding
high-affinity hexose transporters and phosphorylating enzymes. The induction of genes 
encoding enzymes involved in glycogen, but not trehalose, metabolism is largely dependent 
of the presence of Snf1. Therefore the function of Snf1 in adaptation to glucose scarcity 
appears crucial for alkaline pH tolerance. Incorporation of micromolar amounts of iron and
copper to a glucose-supplemented medium resulted in an additive effect and allows near-
normal growth at high pH, thus indicating that these three nutrients are key limiting 
factors for growth in an alkaline environment.

PMID: 22372618

GEO ID: GSE25697

PCL filename	short description	# conditions	tags	# channels
GSE25697_final.pcl	The role of the Snf1 kinase in the adaptive response of Saccharomyces cerevisiae  to alkaline pH stress.	12	stress|signaling|carbon utilization	2

File last modified: Thu Apr 25 2024

---


# Celton_2012_PMID_22805527

Citation: Celton M, Sanchez I, Goelzer A, Fromion V, Camarasa C, Dequin S. 2012. A comparative 
transcriptomic, fluxomic and metabolomic analysis of the response of Saccharomyces 
cerevisiae to increases in NADPH oxidation.

Full Description: BACKGROUND: Redox homeostasis is essential to sustain metabolism and growth. We recently 
reported that yeast cells meet a gradual increase in imposed NADPH demand by progressively
increasing flux through the pentose phosphate (PP) and acetate pathways and by exchanging 
NADH for NADPH in the cytosol, via a transhydrogenase-like cycle. Here, we studied the 
mechanisms underlying this metabolic response, through a combination of gene expression 
profiling and analyses of extracellular and intracellular metabolites and 13 C-flux 
analysis. RESULTS: NADPH oxidation was increased by reducing acetoin to 2,3-butanediol in 
a strain overexpressing an engineered NADPH-dependent butanediol dehydrogenase cultured in
the presence of acetoin. An increase in NADPH demand to 22 times the  anabolic requirement
for NADPH was accompanied by the intracellular accumulation  of PP pathway metabolites 
consistent with an increase in flux through this pathway. Increases in NADPH demand were 
accompanied by the successive induction of several genes of the PP pathway. NADPH-
consuming pathways, such as amino-acid  biosynthesis, were upregulated as an indirect 
effect of the decrease in NADPH availability. Metabolomic analysis showed that the most 
extreme modification of NADPH demand resulted in an energetic problem. Our results also 
highlight the influence of redox status on aroma production. CONCLUSIONS: Combined 13 C-
flux, intracellular metabolite levels and microarrays analyses revealed that NADPH 
homeostasis, in response to a progressive increase in NADPH demand, was achieved  by the 
regulation, at several levels, of the PP pathway. This pathway is principally under 
metabolic control, but regulation of the transcription of PP pathway genes can exert a 
stronger effect, by redirecting larger amounts of carbon to this pathway to satisfy the 
demand for NADPH. No coordinated response of genes involved in NADPH metabolism was 
observed, suggesting that yeast has no  system for sensing NADPH/NADP+ ratio. Instead, the
induction of NADPH-consuming amino-acid pathways in conditions of NADPH limitation may 
indirectly trigger the  transcription of a set of PP pathway genes.

PMID: 22805527

GEO ID: GSE34808.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE34808.final.pcl	A comparative transcriptomic, fluxomic and metabolomic analysis of the response of Saccharomyces cerevisiae to increases in NADPH oxidation.	3	metabolism and metabolic regulation	2

File last modified: Thu Apr 25 2024
GSE34809.final.pcl	A comparative transcriptomic, fluxomic and metabolomic analysis of the response of Saccharomyces cerevisiae to increases in NADPH oxidation.	9	metabolism and metabolic regulation	2

File last modified: Thu Apr 25 2024

---


# Chin_2012_PMID_22289124

Citation: Chin SL, Marcus IM, Klevecz RR, Li CM. 2012. Dynamics of oscillatory phenotypes in 
Saccharomyces cerevisiae reveal a network of genome-wide transcriptional oscillators.

Full Description: Genetic and environmental factors are well-studied influences on phenotype; however, time 
is a variable that is rarely considered when studying changes in cellular phenotype. Time-
resolved microarray data revealed genome-wide transcriptional oscillation in a yeast 
continuous culture system with approximately 2 and approximately 4 h periods. We mapped 
the global patterns of transcriptional oscillations into a 3D map to represent different 
cellular phenotypes of redox cycles. This map shows the dynamic nature of gene expression
in that transcripts are ordered and coupled to each other through time and concentration 
space. Although cells differed in oscillation periods, transcripts  involved in certain 
processes were conserved in a deterministic way. When oscillation period lengthened, the 
peak to trough ratio of transcripts increased  and the fraction of cells in the unbudded (
G0/G1) phase of the cell division cycle increased. Decreasing the glucose level in the 
culture medium was one way to increase the redox cycle, possibly from changes in metabolic
flux. The period  may be responding to lower glucose levels by increasing the fraction of 
cells in  G1 and reducing S-phase gating so that cells can spend more time in catabolic 
processes. Our results support that gene transcripts are coordinated with metabolic 
functions and the cell division cycle.

PMID: 22289124

GEO ID: GSE30052.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE30052.final.pcl	Dynamics of oscillatory phenotypes in Saccharomyces cerevisiae reveal a network of genome-wide transcriptional oscillators.	49	mitotic cell cycle	1

File last modified: Thu Apr 25 2024
GSE30051.final.pcl	Dynamics of oscillatory phenotypes in Saccharomyces cerevisiae reveal a network of genome-wide transcriptional oscillators.	32	mitotic cell cycle	1

File last modified: Thu Apr 25 2024

---


# Chumnanpuen_2012_PMID_22622761

Citation: Chumnanpuen P, Zhang J, Nookaew I, Nielsen J. 2012. Integrated analysis of transcriptome 
and lipid profiling reveals the co-influences of inositol-choline and Snf1 in controlling 
lipid biosynthesis in yeast.

Full Description: In the yeast Saccharomyces cerevisiae many genes involved in lipid biosynthesis are 
transcriptionally controlled by inositol-choline and the protein kinase Snf1. Here we 
undertook a global study on how inositol-choline and Snf1 interact in controlling lipid 
metabolism in yeast. Using both a reference strain (CEN.PK113-7D) and a snf1Delta strain 
cultured at different nutrient limitations  (carbon and nitrogen), at a fixed specific 
growth rate of 0.1 h(-1), and at different inositol choline concentrations, we quantified 
the expression of genes  involved in lipid biosynthesis and the fluxes towards the 
different lipid components. Through integrated analysis of the transcriptome, the lipid 
profiling and the fluxome, it was possible to obtain a high quality, large-scale dataset 
that could be used to identify correlations and associations between the different 
components. At the transcription level, Snf1 and inositol-choline interact either directly
through the main phospholipid-involving transcription factors (i.e. Ino2, Ino4, and Opi1) 
or through other transcription factors e.g. Gis1, Mga2, and Hac1. However, there seems to 
be flux regulation at the enzyme levels of several lipid involving enzymes. The analysis 
showed the strength of using both transcriptome and lipid profiling analysis for mapping 
the co-influence of inositol-choline and Snf1 on phospholipid metabolism.

PMID: 22622761

GEO ID: GSE32418

PCL filename	short description	# conditions	tags	# channels
GSE32418_final.pcl	Integrated analysis of transcriptome and lipid profiling reveals the co-influences of inositol-choline and Snf1 in controlling lipid biosynthesis in yeast.	24	lipid metabolism|nitrogen utilization|carbon utilization	1

File last modified: Thu Apr 25 2024

---


# Costelloe_2012_PMID_22960744

Citation: Costelloe T, Louge R, Tomimatsu N, Mukherjee B, Martini E, Khadaroo B, Dubois K, Wiegant 
WW, Thierry A, Burma S, van Attikum H, Llorente B. 2012. The yeast Fun30 and human 
SMARCAD1 chromatin remodellers promote DNA end resection.

Full Description: Several homology-dependent pathways can repair potentially lethal DNA double-strand breaks
(DSBs). The first step common to all homologous recombination reactions is the 5'-3' 
degradation of DSB ends that yields the 3' single-stranded DNA required for the loading of
checkpoint and recombination proteins. In yeast, the Mre11-Rad50-Xrs2 complex (Xrs2 is 
known as NBN or NBS1 in humans) and Sae2 (known as RBBP8 or CTIP in humans) initiate end 
resection, whereas long-range resection depends on the exonuclease Exo1, or the helicase-
topoisomerase complex Sgs1-Top3-Rmi1 together with the endonuclease Dna2 (refs 1-6). DSBs 
occur in the context of chromatin, but how the resection machinery navigates through 
nucleosomal DNA is a process that is not well understood. Here we show that the yeast 
Saccharomyces cerevisiae Fun30 protein and its human counterpart SMARCAD1 (ref. 8), two 
poorly characterized ATP-dependent chromatin remodellers of the Snf2 ATPase family, are 
directly involved in the DSB response. Fun30 physically associates with DSB ends and 
directly promotes both Exo1- and Sgs1-dependent end resection through a mechanism 
involving its ATPase activity. The function of Fun30 in resection facilitates the repair 
of camptothecin-induced DNA lesions, although it becomes dispensable when  Exo1 is 
ectopically overexpressed. Interestingly, SMARCAD1 is also recruited to DSBs, and the 
kinetics of recruitment is similar to that of EXO1. The loss of SMARCAD1 impairs end 
resection and recombinational DNA repair, and renders cells  hypersensitive to DNA damage 
resulting from camptothecin or poly(ADP-ribose) polymerase inhibitor treatments. These 
findings unveil an evolutionarily conserved role for the Fun30 and SMARCAD1 chromatin 
remodellers in controlling end resection, homologous recombination and genome stability in
the context of chromatin.

PMID: 22960744

GEO ID: GSE38735.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE38735.final.pcl	The yeast Fun30 and human SMARCAD1 chromatin remodellers promote DNA end resection.	2	chromatin organization	2

File last modified: Thu Apr 25 2024

---


# Dominissini_2012_PMID_22575960

Citation: Dominissini D, Moshitch-Moshkovitz S, Schwartz S, Salmon-Divon M, Ungar L, Osenberg S, 
Cesarkas K, Jacob-Hirsch J, Amariglio N, Kupiec M, Sorek R, Rechavi G. 2012. Topology of 
the human and mouse m6A RNA methylomes revealed by m6A-seq.

Full Description: An extensive repertoire of modifications is known to underlie the versatile coding, 
structural and catalytic functions of RNA, but it remains largely uncharted territory. 
Although biochemical studies indicate that N(6)-methyladenosine (m(6)A) is the most 
prevalent internal modification in messenger RNA, an in-depth study of its distribution 
and functions has been impeded by a lack of robust analytical methods. Here we present the
human and mouse m(6)A modification landscape in a transcriptome-wide manner, using a novel
approach, m(6)A-seq, based on antibody-mediated capture and massively parallel sequencing
. We identify over 12,000 m(6)A sites characterized by a typical consensus in the 
transcripts of more than 7,000 human genes. Sites preferentially appear in two distinct 
landmarks--around stop codons and within long internal exons--and are highly conserved 
between human and mouse. Although most sites are  well preserved across normal and 
cancerous tissues and in response to various stimuli, a subset of stimulus-dependent, 
dynamically modulated sites is identified. Silencing the m(6)A methyltransferase 
significantly affects gene expression and alternative splicing patterns, resulting in 
modulation of the p53  (also known as TP53) signalling pathway and apoptosis. Our findings
therefore suggest that RNA decoration by m(6)A has a fundamental role in regulation of 
gene expression.

PMID: 22575960

GEO ID: GSE36958

PCL filename	short description	# conditions	tags	# channels
GSE36958_final.pcl	Topology of the human and mouse m6A RNA methylomes revealed by m6A-seq.	9	sporulation	1

File last modified: Thu Apr 25 2024

---


# Flom_2012_PMID_22461145

Citation: Flom GA, Langner E, Johnson JL. 2012. Identification of an Hsp90 mutation that selectively
disrupts cAMP/PKA signaling  in Saccharomyces cerevisiae.

Full Description: The molecular chaperone Hsp90 cooperates with multiple cochaperone proteins as it promotes
the folding and activation of diverse client proteins. Some cochaperones regulate the 
ATPase activity of Hsp90, while others appear to promote Hsp90 interaction with specific 
types of client proteins. Through its interaction with  the adenylate cyclase Cyr1, the 
Sgt1 cochaperone modulates the activity of the cAMP pathway in Saccharomyces cerevisiae. A
specific mutation in yeast Hsp90, hsc82-W296A, or a mutation in Sgt1, sgt1-K360E, resulted
in altered transcription patterns genetically linked to the cAMP pathway. Hsp90 interacted
with Cyr1 in vivo and the hsc82-W296A mutation resulted in reduced accumulation of Cyr1. 
Hsp90-Sgt1 interaction was altered by either the hsc82-W296A or sgt1-K360E mutation, 
suggesting defective Hsp90-Sgt1 cooperation leads to reduced Cyr1 activity. Microarray 
analysis of hsc82-W296A cells indicated that over 80 % of all transcriptional changes in 
this strain may be attributed to altered cAMP signaling. This suggests that a majority of 
the cellular defects observed in hsc82-W296A cells are due to altered interaction with one
specific essential cochaperone, Sgt1 and one essential client, Cyr1. Together our results 
indicate that specific interaction of Hsp90 and Sgt1 with Cyr1 plays a key role in 
regulating gene expression, including genes involved in polarized morphogenesis.

PMID: 22461145

GEO ID: GSE34757

PCL filename	short description	# conditions	tags	# channels
GSE34757_final.pcl	Identification of an Hsp90 mutation that selectively disrupts cAMP/PKA signaling  in Saccharomyces cerevisiae.	6	signaling	1

File last modified: Thu Apr 25 2024

---


# Garcia-Rodriguez_2012_PMID_22493290

Citation: Garcia-Rodriguez N, Diaz de la Loza Mdel C, Andreson B, Monje-Casas F, Rothstein R, 
Wellinger RE. 2012. Impaired manganese metabolism causes mitotic misregulation.

Full Description: Manganese is an essential trace element, whose intracellular levels need to be carefully 
regulated. Mn(2+) acts as a cofactor for many enzymes and excess of Mn(2+) is toxic. 
Alterations in Mn(2+) homeostasis affect metabolic functions and mutations in the human Mn
(2+)/Ca(2+) transporter ATP2C1 have been linked to Hailey-Hailey disease. By deletion of 
the yeast orthologue PMR1 we have studied the impact of Mn(2+) on cell cycle progression 
and show that an excess of cytosolic Mn(2+) alters S-phase transit, induces 
transcriptional up-regulation of cell cycle regulators, bypasses the need for S-phase cell
cycle checkpoints and predisposes to genomic instability. On the other hand, we find that 
depletion of  the Golgi Mn(2+) pool requires a functional morphology checkpoint to avoid 
the formation of polyploid cells.

PMID: 22493290

GEO ID: GSE29420

PCL filename	short description	# conditions	tags	# channels
GSE29420_final.pcl	Impaired manganese metabolism causes mitotic misregulation.	6	cellular ion homeostasis	1

File last modified: Thu Apr 25 2024

---


# Geijer_2012_PMID_23066959

Citation: Geijer C, Pirkov I, Vongsangnak W, Ericsson A, Nielsen J, Krantz M, Hohmann S. 2012. Time 
course gene expression profiling of yeast spore germination reveals a network of 
transcription factors orchestrating the global response.

Full Description: BACKGROUND: Spore germination of the yeast Saccharomyces cerevisiae is a multi-step 
developmental path on which dormant spores re-enter the mitotic cell cycle and resume 
vegetative growth. Upon addition of a fermentable carbon source  and nutrients, the outer 
layers of the protective spore wall are locally degraded, the tightly packed spore gains 
volume and an elongated shape, and eventually the germinating spore re-enters the cell 
cycle. The regulatory pathways driving this process are still largely unknown. Here we 
characterize the global gene expression profiles of germinating spores and identify 
potential transcriptional regulators of this process with the aim to increase our 
understanding of the mechanisms that control the transition from cellular dormancy to 
proliferation. RESULTS: Employing detailed gene expression time course data we have 
analysed the reprogramming of dormant spores during the transition to proliferation 
stimulated by a rich growth medium or pure glucose. Exit from dormancy results in rapid 
and global changes consisting of different sequential gene expression subprograms. The 
regulated genes reflect the transition towards glucose metabolism, the resumption of 
growth and the release of stress, similar to cells exiting a stationary growth phase. High
resolution time course analysis during the onset of germination allowed us to identify a 
transient up-regulation of genes involved in protein folding and transport. We also 
identified a network of transcription factors that may be regulating the global response. 
While the expression outputs following stimulation by rich glucose medium or by glucose 
alone are qualitatively similar, the response to rich medium is stronger. Moreover, spores
sense and react to amino acid starvation within the first 30 min after germination 
initiation, and this response can be linked to specific transcription factors. CONCLUSIONS
: Resumption of growth in germinating spores is characterized by a highly synchronized 
temporal organisation of up- and down-regulated genes which reflects the metabolic 
reshaping of the quickening spores.

PMID: 23066959

GEO ID: GSE29960

PCL filename	short description	# conditions	tags	# channels
GSE29960_final.pcl	Time course gene expression profiling of yeast spore germination reveals a network of transcription factors orchestrating the global response.	18	carbon utilization|nutrient utilization|sporulation	2

File last modified: Thu Apr 25 2024

---


# Guan_2012_PMID_22851651

Citation: Guan Q, Haroon S, Bravo DG, Will JL, Gasch AP. 2012. Cellular memory of acquired stress 
resistance in Saccharomyces cerevisiae.

Full Description: Cellular memory of past experiences has been observed in several organisms and across a 
variety of experiences, including bacteria remembering prior nutritional status and amoeba
learning to anticipate future environmental conditions. Here, we show that Saccharomyces 
cerevisiae maintains a multifaceted  memory of prior stress exposure. We previously 
demonstrated that yeast cells exposed to a mild dose of salt acquire subsequent tolerance 
to severe doses of H(2)O(2). We set out to characterize the retention of acquired 
tolerance and in the process uncovered two distinct aspects of cellular memory. First, we 
found that H(2)O(2) resistance persisted for four to five generations after cells were  
removed from the prior salt treatment and was transmitted to daughter cells that  never 
directly experienced the pretreatment. Maintenance of this memory did not require nascent 
protein synthesis after the initial salt pretreatment, but rather required long-lived 
cytosolic catalase Ctt1p that was synthesized during salt exposure and then distributed to
daughter cells during subsequent cell divisions. In addition to and separable from the 
memory of H(2)O(2) resistance, these cells  also displayed a faster gene-expression 
response to subsequent stress at >1000 genes, representing transcriptional memory. The 
faster gene-expression response requires the nuclear pore component Nup42p and serves an 
important function by facilitating faster reacquisition of H(2)O(2) tolerance after a 
second cycle of salt exposure. Memory of prior stress exposure likely provides a 
significant advantage to microbial populations living in ever-changing environments.

PMID: 22851651

GEO ID: GSE32196GPL15778.sfp.pcl

PCL filename	short description	# conditions	tags	# channels
GSE32196GPL15778.sfp.pcl	Cellular memory of acquired stress resistance in Saccharomyces cerevisiae.	11	osmotic stress|oxidative stress	2

File last modified: Thu Apr 25 2024

---


# Haitani_2012_PMID_22841865

Citation: Haitani Y, Tanaka K, Yamamoto M, Nakamura T, Ando A, Ogawa J, Shima J. 2012. 
Identification of an acetate-tolerant strain of Saccharomyces cerevisiae and 
characterization by gene expression analysis.

Full Description: A massive screening was performed to identify an acetate-tolerant strain of Saccharomyces 
cerevisiae. We found that S. cerevisiae ATCC 38555 is acetate-tolerant, with a 
fermentation profile indicating that it has a high level of acetate adaptation. The global
gene expression analysis indicated that AFT1- and HAA1-regulated genes are clearly up-
regulated.

PMID: 22841865

GEO ID: GSE36914.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE36914.final.pcl	Identification of an acetate-tolerant strain of Saccharomyces cerevisiae and characterization by gene expression analysis.	8	carbon utilization|stress	1

File last modified: Thu Apr 25 2024

---


# Hao_2012_PMID_22179789

Citation: Hao N, O'Shea EK. 2012. Signal-dependent dynamics of transcription factor translocation 
controls gene expression.

Full Description: Information about environmental stimuli is often transmitted using common signaling 
molecules, but the mechanisms that ensure signaling specificity are not entirely known. 
Here we show that the identities and intensities of different stresses are transmitted by 
modulation of the amplitude, duration or frequency of nuclear translocation of the 
Saccharomyces cerevisiae general stress response transcription factor Msn2. Through 
artificial control of the dynamics of Msn2 translocation, we reveal how distinct dynamical
schemes differentially affect reporter gene expression. Using a simple model, we predict 
stress-induced reporter gene expression from single-cell translocation dynamics. We then 
demonstrate that the response of natural target genes to dynamical modulation of  Msn2 
translocation is influenced by differences in the kinetics of promoter transitions and 
transcription factor binding properties. Thus, multiple environmental signals can trigger 
qualitatively different dynamics of a single transcription factor and influence gene 
expression patterns.

PMID: 22179789

GEO ID: GSE32703

PCL filename	short description	# conditions	tags	# channels
GSE32703_final.pcl	Signal-dependent dynamics of transcription factor translocation controls gene expression.	56	stress|signaling	2

File last modified: Thu Apr 25 2024

---


# Hong_2012_PMID_22904057

Citation: Hong KK, Nielsen J. 2012. Recovery of phenotypes obtained by adaptive evolution through 
inverse metabolic engineering.

Full Description: In a previous study, system level analysis of adaptively evolved yeast mutants showing 
improved galactose utilization revealed relevant mutations. The governing mutations were 
suggested to be in the Ras/PKA signaling pathway and ergosterol metabolism. Here, site-
directed mutants having one of the mutations RAS2(Lys77),  RAS2(Tyr112), and ERG5(Pro370) 
were constructed and evaluated. The mutants were also combined with overexpression of PGM2
, earlier proved as a beneficial target  for galactose utilization. The constructed 
strains were analyzed for their gross  phenotype, transcriptome and targeted metabolites, 
and the results were compared  to those obtained from reference strains and the evolved 
strains. The RAS2(Lys77) mutation resulted in the highest specific galactose uptake rate 
among all of the  strains with an increased maximum specific growth rate on galactose. The
RAS2(Tyr112) mutation also improved the specific galactose uptake rate and also resulted 
in many transcriptional changes, including ergosterol metabolism. The ERG5(Pro370) 
mutation only showed a small improvement, but when it was combined with PGM2 
overexpression, the phenotype was almost the same as that of the evolved mutants. 
Combination of the RAS2 mutations with PGM2 overexpression also  led to a complete 
recovery of the adaptive phenotype in galactose utilization. Recovery of the gross 
phenotype by the reconstructed mutants was achieved with much fewer changes in the genome 
and transcriptome than for the evolved mutants.  Our study demonstrates how the 
identification of specific mutations by systems biology can direct new metabolic 
engineering strategies for improving galactose utilization by yeast.

PMID: 22904057

GEO ID: GSE36118

PCL filename	short description	# conditions	tags	# channels
GSE36118_final.pcl	Recovery of phenotypes obtained by adaptive evolution through inverse metabolic engineering.	14	carbon utilization|evolution|signaling	1

File last modified: Thu Apr 25 2024

---


# Jaime_2012_PMID_22727066

Citation: Jaime MD, Lopez-Llorca LV, Conesa A, Lee AY, Proctor M, Heisler LE, Gebbia M, Giaever G, 
Westwood JT, Nislow C. 2012. Identification of yeast genes that confer resistance to 
chitosan oligosaccharide  (COS) using chemogenomics.

Full Description: BACKGROUND: Chitosan oligosaccharide (COS), a deacetylated derivative of chitin,  is an 
abundant, and renewable natural polymer. COS has higher antimicrobial properties than 
chitosan and is presumed to act by disrupting/permeabilizing the  cell membranes of 
bacteria, yeast and fungi. COS is relatively non-toxic to mammals. By identifying the 
molecular and genetic targets of COS, we hope to gain a better understanding of the 
antifungal mode of action of COS. RESULTS: Three different chemogenomic fitness assays, 
haploinsufficiency (HIP), homozygous deletion (HOP), and multicopy suppression (MSP) 
profiling were combined with a transcriptomic analysis to gain insight in to the mode of 
action and mechanisms of resistance to chitosan oligosaccharides. The fitness assays 
identified 39 yeast deletion strains sensitive to COS and 21 suppressors of COS 
sensitivity. The genes identified are involved in processes such as RNA biology (
transcription, translation and regulatory mechanisms), membrane functions (e.g.  
signalling, transport and targeting), membrane structural components, cell division, and 
proteasome processes. The transcriptomes of control wild type and 5 suppressor strains 
overexpressing ARL1, BCK2, ERG24, MSG5, or RBA50, were analyzed in the presence and 
absence of COS. Some of the up-regulated transcripts in the suppressor overexpressing 
strains exposed to COS included genes involved in transcription, cell cycle, stress 
response and the Ras signal transduction pathway. Down-regulated transcripts included 
those encoding protein folding components and respiratory chain proteins. The COS-induced 
transcriptional response is distinct from previously described environmental stress 
responses (i.e. thermal, salt, osmotic and oxidative stress) and pre-treatment with these 
well characterized environmental stressors provided little or any resistance to COS. 
CONCLUSIONS: Overexpression of the ARL1 gene, a member of the Ras superfamily that 
regulates membrane trafficking, provides protection against COS-induced cell membrane 
permeability and damage. We found that the ARL1 COS-resistant over-expression strain was 
as sensitive to Amphotericin B, Fluconazole and Terbinafine as the wild type cells and 
that when COS and Fluconazole are used in combination they act in a synergistic fashion. 
The gene targets of COS identified in this study indicate that COS's mechanism of action 
is different from other commonly studied fungicides that target membranes, suggesting that
COS may be an effective fungicide for drug-resistant fungal pathogens.

PMID: 22727066

GEO ID: GSE32888.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE32888.final.pcl	Identification of yeast genes that confer resistance to chitosan oligosaccharide  (COS) using chemogenomics.	36	chemical stimulus|stress|signaling|cell cycle regulation	1

File last modified: Thu Apr 25 2024

---


# Joshi_2012_PMID_22718977

Citation: Joshi RS, Pina B, Roca J. 2012. Topoisomerase II is required for the production of long 
Pol II gene transcripts in yeast.

Full Description: The extent to which the DNA relaxation activities of eukaryotic topoisomerases (topo I and
topo II) are redundant during gene transcription is unclear. Although both enzymes can 
often substitute for each other in vivo, studies in vitro had revealed that the DNA cross-
inversion mechanism of topo II relaxes chromatin more efficiently than the DNA strand-
rotation mechanism of topo I. Here, we show that  the inactivation of topo II in budding 
yeast produces an abrupt decrease of virtually all polyA+ RNA transcripts of length above 
approximately 3 kb, irrespective of their function. This reduction is not related to 
transcription initiation but to the stall of RNA polymerase II (Pol II) during elongation
. This reduction does not occur in topo I mutants; and it is not avoided by overproducing 
yeast topo I or bacterial topo I, which relaxes (-) DNA supercoils. It is rescued by 
catalytically active topo II or a GyrBA enzyme, which relaxes (+) DNA supercoils. These 
findings demonstrate that DNA relaxation activities of  topo I and topo II are not 
interchangeable in vivo. Apparently, only topo II relaxes efficiently the (+) DNA 
supercoils that stall the advancement of Pol II in long genes. A mechanistic model is 
proposed.

PMID: 22718977

GEO ID: GSE18605.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE18605.final.pcl	Topoisomerase II is required for the production of long Pol II gene transcripts in yeast.	6	chromatin organization	2

File last modified: Thu Apr 25 2024

---


# Lanza_2012_PMID_22558379

Citation: Lanza AM, Blazeck JJ, Crook NC, Alper HS. 2012. Linking yeast Gcn5p catalytic function and
gene regulation using a quantitative,  graded dominant mutant approach.

Full Description: Establishing causative links between protein functional domains and global gene regulation
is critical for advancements in genetics, biotechnology, disease treatment, and systems 
biology. This task is challenging for multifunctional proteins when relying on traditional
approaches such as gene deletions since they remove all domains simultaneously. Here, we 
describe a novel approach to extract  quantitative, causative links by modulating the 
expression of a dominant mutant allele to create a function-specific competitive 
inhibition. Using the yeast histone acetyltransferase Gcn5p as a case study, we 
demonstrate the utility of this approach and (1) find evidence that Gcn5p is more involved
in cell-wide gene repression, instead of the accepted gene activation associated with HATs
, (2) identify previously unknown gene targets and interactions for Gcn5p-based 
acetylation, (3) quantify the strength of some Gcn5p-DNA associations, (4) demonstrate 
that this approach can be used to correctly identify canonical chromatin modifications, (5
) establish the role of acetyltransferase activity on  synthetic lethal interactions, and
(6) identify new functional classes of genes regulated by Gcn5p acetyltransferase activity
--all six of these major conclusions were unattainable by using standard gene knockout 
studies alone. We recommend that a graded dominant mutant approach be utilized in 
conjunction with a traditional knockout to study multifunctional proteins and generate 
higher-resolution data that more accurately probes protein domain function and influence.

PMID: 22558379

GEO ID: GSE26923

PCL filename	short description	# conditions	tags	# channels
GSE26923_final.pcl	Linking yeast Gcn5p catalytic function and gene regulation using a quantitative,  graded dominant mutant approach.	15	histone modification	1

File last modified: Thu Apr 25 2024

---


# Lewis_2012_PMID_23275883

Citation: Lewis JA, Gasch AP. 2012. Natural variation in the yeast glucose-signaling network reveals
a new role for the Mig3p transcription factor.

Full Description: The Crabtree effect, in which fermentative metabolism is preferred at the expense of 
respiration, is a hallmark of budding yeast's glucose response and a model for the Warburg
effect in human tumors. While the glucose-responsive transcriptional  repressors Mig1p and
Mig2p play well-characterized roles in the Crabtree effect,  little function for the 
related Mig3p transcription factor has been uncovered, despite numerous investigations of 
laboratory yeast strains. Here we studied a wild isolate of Saccharomyces cerevisiae to 
uncover a critical role for Mig3p that has been lost in S288c-derived laboratory strains. 
We found that Mig3p affects the expression of hundreds of glucose-responsive genes in the 
oak strain  YPS163, both during growth under standard conditions and upon ethanol 
treatment.  Our results suggest that Mig3p may act as a multifunctional activator/
repressor that plays separate roles under standard vs. stress conditions and that this 
function has been largely lost in the lab strains. Population analysis suggests that the 
lab strain and several wild strains harbor mutations that diminish Mig3p function. Thus, 
by expanding our attention to multiple genetic backgrounds, we have uncovered an important
missing link in a key metabolic response.

PMID: 23275883

GEO ID: GSE40153.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE40153.final.pcl	Natural variation in the yeast glucose-signaling network reveals a new role for the Mig3p transcription factor.	19	carbon utilization|signaling	2

File last modified: Thu Apr 25 2024

---


# Lin_2012_PMID_22308403

Citation: Lin L, Chamberlain L, Zhu LJ, Green MR. 2012. Analysis of Gal4-directed transcription 
activation using Tra1 mutants selectively defective for interaction with Gal4.

Full Description: Promoter-specific transcriptional activators (activators) stimulate transcription through 
direct interactions with one or more components of the transcription machinery, termed the
target. The identification of direct in vivo targets of activators has been a major 
challenge. Previous studies have provided evidence that the Tra1 subunit of the yeast SAGA
(Spt-Ada-Gcn5-acetyltransferase) complex  is the target of the yeast activator Gal4. 
However, several other general transcription factors, in particular the mediator complex, 
have also been implicated as Gal4 targets. Here we perform a large-scale genetic screen to
derive and characterize tra1 alleles that are selectively defective for interaction with 
Gal4 in vivo [Gal4 interaction defective (GID) mutants]. In contrast to WT Tra1, Tra1 GID 
mutants are not recruited by Gal4 to the promoter and cannot support Gal4-directed 
transcription, demonstrating the essentiality of the Gal4-Tra1 interaction. In yeast 
strains expressing a Tra1 GID mutant, binding of Gal4 to the promoter is unexpectedly also
diminished, indicating that Gal4 and Tra1 bind cooperatively. Consistent with cooperative 
binding, we demonstrate that the Gal4-Tra1 interaction occurs predominantly on the 
promoter and not off DNA. Finally, we show that although Tra1 is targeted by other 
activators, these interactions are unaffected by GID mutations, revealing an unanticipated
specificity of the Gal4-Tra1 interaction.

PMID: 22308403

GEO ID: GSE31389

PCL filename	short description	# conditions	tags	# channels
GSE31389_final.pcl	Analysis of Gal4-directed transcription activation using Tra1 mutants selectively defective for interaction with Gal4.	6	transcription	1

File last modified: Thu Apr 25 2024
GSE31390_final.pcl	Analysis of Gal4-directed transcription activation using Tra1 mutants selectively defective for interaction with Gal4.	6	transcription	1

File last modified: Thu Apr 25 2024

---


# Marquina_2012_PMID_22750472

Citation: Marquina M, Queralt E, Casamayor A, Arino J. 2012. Lack of the Glc7 phosphatase regulatory
subunit Ypi1 activates the morphogenetic  checkpoint.

Full Description: Ypi1 is an essential regulator of the Saccharomyces cerevisiae Glc7 protein phosphatase. 
Although lack of Ypi1 results in a dramatic blockage in the G2/M cell cycle transition, 
with abnormally shaped large buds and short spindles, the  molecular bases for this 
phenotype are still obscure. We report here that depletion of Ypi1 results in 
stabilization of the Pds1 securin, suggesting the activation of a G2/M checkpoint. 
Depletion of Ypi1 in cells deleted for MAD1/MAD2 or RAD9 still resulted in G2/M blockage, 
in spite that these cells lack key components of the spindle assembly and DNA damage 
checkpoints signaling, respectively. In contrast, deletion of SWE1, which encodes a 
protein kinase required for the morphogenesis checkpoint signaling, allowed passage 
through G2/M and recovery of normal cell morphology, although the cells did not 
proliferate. Depletion of Ypi1 caused stabilization of the Swe1 kinase, persistent 
phosphorylation of protein kinase Cdc28 at Y19, a landmark for morphogenesis checkpoint 
activation, and depletion of the Cdc11 septin, which explains the failure to form properly
assembled septin rings at the bud necks. Deletion of SWE1 restored normal Cdc11 levels in 
the absence of Ypi1. These results demonstrate that Ypi1 plays an important role in the 
morphogenesis checkpoint, possibly by regulating Swe1.

PMID: 22750472

GEO ID: GSE23134.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE23134.final.pcl	Lack of the Glc7 phosphatase regulatory subunit Ypi1 activates the morphogenetic  checkpoint.	6	cell morphogenesis|cell cycle regulation	2

File last modified: Thu Apr 25 2024

---


# Nadal-Ribelles_2012_PMID_23158682

Citation: Nadal-Ribelles M, Conde N, Flores O, Gonzalez-Vallinas J, Eyras E, Orozco M, de Nadal E, 
Posas F. 2012. Hog1 bypasses stress-mediated down-regulation of transcription by RNA 
polymerase  II redistribution and chromatin remodeling.

Full Description: BACKGROUND: Cells are subjected to dramatic changes of gene expression upon environmental 
changes. Stress causes a general down-regulation of gene expression together with the 
induction of a set of stress-responsive genes. The p38-related  stress-activated protein 
kinase Hog1 is an important regulator of transcription upon osmostress in yeast. RESULTS: 
Genome-wide localization studies of RNA polymerase II (RNA Pol II) and Hog1 showed that 
stress induced major changes in RNA Pol II localization, with a shift toward stress-
responsive genes relative to  housekeeping genes. RNA Pol II relocalization required Hog1
, which was also localized to stress-responsive loci. In addition to RNA Pol II-bound 
genes, Hog1  also localized to RNA polymerase III-bound genes, pointing to a wider role 
for Hog1 in transcriptional control than initially expected. Interestingly, an increasing 
association of Hog1 with stress-responsive genes was strongly correlated with chromatin 
remodeling and increased gene expression. Remarkably, MNase-Seq analysis showed that 
although chromatin structure was not significantly altered at a genome-wide level in 
response to stress, there was pronounced chromatin remodeling for those genes that 
displayed Hog1 association. CONCLUSION: Hog1 serves to bypass the general down-regulation 
of gene expression that occurs  in response to osmostress, and does so both by targeting 
RNA Pol II machinery and by inducing chromatin remodeling at stress-responsive loci.

PMID: 23158682

GEO ID: GSE41451.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE41451.final.pcl	Hog1 bypasses stress-mediated down-regulation of transcription by RNA polymerase  II redistribution and chromatin remodeling.	3	osmotic stress|chromatin organization	2

File last modified: Thu Apr 25 2024

---


# Navlakha_2012_PMID_22916002

Citation: Navlakha S, Gitter A, Bar-Joseph Z. 2012. A network-based approach for predicting missing 
pathway interactions.

Full Description: Embedded within large-scale protein interaction networks are signaling pathways that 
encode response cascades in the cell. Unfortunately, even for well-studied species like S
. cerevisiae, only a fraction of all true protein interactions are  known, which makes it 
difficult to reason about the exact flow of signals and the corresponding causal relations
in the network. To help address this problem, we introduce a framework for predicting new 
interactions that aid connectivity between upstream proteins (sources) and downstream 
transcription factors (targets) of a particular pathway. Our algorithms attempt to 
globally minimize the distance between sources and targets by finding a small set of 
shortcut edges to add to the network. Unlike existing algorithms for predicting general 
protein  interactions, by focusing on proteins involved in specific responses our approach
homes-in on pathway-consistent interactions. We applied our method to extend pathways in 
osmotic stress response in yeast and identified several missing interactions, some of 
which are supported by published reports. We also performed experiments that support a 
novel interaction not previously reported. Our framework is general and may be applicable 
to edge prediction problems in other domains.

PMID: 22916002

GEO ID: GSE28213GPL14666.sfp.pcl

PCL filename	short description	# conditions	tags	# channels
GSE28213GPL14666.sfp.pcl	A network-based approach for predicting missing pathway interactions.	12	signaling|osmotic stress	2

File last modified: Thu Apr 25 2024

---


# Niles_2012_PMID_22307609

Citation: Niles BJ, Mogri H, Hill A, Vlahakis A, Powers T. 2012. Plasma membrane recruitment and 
activation of the AGC kinase Ypk1 is mediated by  target of rapamycin complex 2 (TORC2) 
and its effector proteins Slm1 and Slm2.

Full Description: The yeast AGC kinase orthologs Ypk1 and Ypk2 control several important cellular processes
, including actin polarization, endocytosis, and sphingolipid metabolism. Activation of 
Ypk1/2 requires phosphorylation by kinases localized at the plasma membrane (PM), 
including the 3-phosphoinositide-dependent kinase 1 orthologs Pkh1/Pkh2 and the target of 
rapamycin complex 2 (TORC2). Unlike their mammalian counterparts SGK and Akt, Ypk1 and 
Ypk2 lack an identifiable lipid-targeting motif; therefore, how these proteins are 
recruited to the PM has  remained elusive. To explore Ypk1/2 function, we constructed ATP 
analog-sensitive alleles of both kinases and monitored global changes in gene expression 
following their inhibition, where we observed increased expression of stress-responsive 
target genes controlled by Ca(2+)-dependent phosphatase calcineurin. TORC2 has been shown 
previously to negatively regulate calcineurin in part by phosphorylating two related 
proteins, Slm1 and Slm2, which associate with the PM  via plextrin homology domains. We 
therefore investigated the relationship between Slm1 and Ypk1 and discovered that these 
proteins interact physically and that Slm1 recruits Ypk1 to the PM for phosphorylation by 
TORC2. We observed further that these steps facilitate subsequent phosphorylation of Ypk1 
by Pkh1/2. Remarkably, a requirement for Slm1, can be bypassed by fusing the plextrin 
homology domain of Slm1 alone onto Ypk1, demonstrating that the essential function of Slm1
is largely attributable to its role in Ypk1 activation. These findings both extend the 
scope of cellular processes regulated by Ypk1/2 to include negative regulation of 
calcineurin and broaden the repertoire of mechanisms for membrane recruitment and 
activation of a protein kinase.

PMID: 22307609

GEO ID: GSE33185

PCL filename	short description	# conditions	tags	# channels
GSE33185_final.pcl	Plasma membrane recruitment and activation of the AGC kinase Ypk1 is mediated by  target of rapamycin complex 2 (TORC2) and its effector proteins Slm1 and Slm2.	9	protein phosphorylation|signaling	2

File last modified: Thu Apr 25 2024

---


# Ochi_2012_PMID_22679107

Citation: Ochi K, Nishizawa T, Inaoka T, Yamada A, Hashimoto K, Hosaka T, Okamoto S, Ozeki Y. 2012. 
Heterologous expression of a plant RelA-SpoT homologue results in increased stress 
tolerance in Saccharomyces cerevisiae by accumulation of the bacterial alarmone ppGpp.

Full Description: The bacterial alarmone ppGpp is present only in bacteria and the chloroplasts of  plants, 
but not in mammalian cells or eukaryotic micro-organisms such as yeasts and fungi. The 
importance of the ppGpp signalling system in eukaryotes has therefore been largely 
overlooked. Here, we demonstrated that heterologous expression of a relA-spoT homologue (
Sj-RSH) isolated from the halophilic plant Suaeda japonica in the yeast Saccharomyces 
cerevisiae results in accumulation of  ppGpp, accompanied by enhancement of tolerance 
against various stress stimuli, such as osmotic stress, ethanol, hydrogen peroxide, high 
temperature and freezing. Unlike bacterial ppGpp accumulation, ppGpp was accumulated in 
the early growth phase but not in the late growth phase. Moreover, nutritional downshift 
resulted in a decrease in ppGpp level, suggesting that the observed Sj-RSH activity to 
synthesize ppGpp is not starvation-dependent, contrary to our expectations based on 
bacteria. Accumulated ppGpp was found to be present solely  in the cytosolic fraction and 
not in the mitochondrial fraction, perhaps reflecting the ribosome-independent ppGpp 
synthesis in S. cerevisiae cells. Unlike bacterial inosine monophosphate (IMP) 
dehydrogenases, the IMP dehydrogenase of S. cerevisiae was insensitive to ppGpp. 
Microarray analysis showed that ppGpp accumulation gave rise to marked changes in gene 
expression, with both upregulation and downregulation, including changes in mitochondrial 
gene expression. The most prominent upregulation (38-fold) was detected in the 
hypothetical gene YBR072C-A of unknown function, followed by many other known stress-
responsive genes. S. cerevisiae may provide new opportunities to uncover and analyse the 
ppGpp signalling system in eukaryotic cells.

PMID: 22679107

GEO ID: GSE38239.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE38239.final.pcl	Heterologous expression of a plant RelA-SpoT homologue results in increased stress tolerance in Saccharomyces cerevisiae by accumulation of the bacterial alarmone ppGpp.	2	stress|transcriptional regulation	1

File last modified: Thu Apr 25 2024

---


# Oromendia_2012_PMID_23222101

Citation: Oromendia AB, Dodgson SE, Amon A. 2012. Aneuploidy causes proteotoxic stress in yeast.

Full Description: Gains or losses of entire chromosomes lead to aneuploidy, a condition tolerated poorly in 
all eukaryotes analyzed to date. How aneuploidy affects organismal and cellular physiology
is poorly understood. We found that aneuploid budding yeast cells are under proteotoxic 
stress. Aneuploid strains are prone to aggregation of endogenous proteins as well as of 
ectopically expressed hard-to-fold proteins such as those containing polyglutamine (polyQ
) stretches. Protein aggregate formation in aneuploid yeast strains is likely due to 
limiting protein quality-control systems, since the proteasome and at least one chaperone 
family, Hsp90, are compromised in many aneuploid strains. The link between aneuploidy and 
the formation and persistence of protein aggregates could have important implications for 
diseases such as cancer and neurodegeneration.

PMID: 23222101

GEO ID: GSE40073

PCL filename	short description	# conditions	tags	# channels
GSE40073_final.pcl	Aneuploidy causes proteotoxic stress in yeast.	20	heat shock	2

File last modified: Thu Apr 25 2024

---


# Orozco_2012_PMID_22327582

Citation: Orozco H, Matallana E, Aranda A. 2012. Oxidative stress tolerance, adenylate cyclase, and 
autophagy are key players in the chronological life span of Saccharomyces cerevisiae 
during winemaking.

Full Description: Most grape juice fermentation takes place when yeast cells are in a nondividing state 
called the stationary phase. Under such circumstances, we aimed to identify the genetic 
determinants controlling longevity, known as the chronological life span. We identified 
commercial strains with both short (EC1118) and long (CSM) life spans in laboratory growth
medium and compared them under diverse conditions. Strain CSM shows better tolerance to 
stresses, including oxidative stress, in the stationary phase. This is reflected during 
winemaking, when this strain has an increased maximum life span. Compared to EC1118, CSM 
overexpresses  a mitochondrial rhodanese gene-like gene, RDL2, whose deletion leads to 
increased reactive oxygen species production at the end of fermentation and a correlative 
loss of viability at this point. EC1118 shows faster growth and higher expression of 
glycolytic genes, and this is related to greater PKA activity due to the upregulation of 
the adenylate cyclase gene. This phenotype has been linked to the presence of a delta 
element in its promoter, whose removal increases the life span. Finally, EC1118 exhibits a
higher level of protein degradation by autophagy, which might help achieve fast growth at 
the expense of cellular structures and may be relevant for long-term survival under 
winemaking conditions.

PMID: 22327582

GEO ID: GSE33068.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE33068.final.pcl	Oxidative stress tolerance, adenylate cyclase, and autophagy are key players in the chronological life span of Saccharomyces cerevisiae during winemaking.	3	fermentation|cell aging|oxidative stress	2

File last modified: Thu Apr 25 2024

---


# Pedersen_2012_PMID_23284296

Citation: Pedersen JM, Fredsoe J, Roedgaard M, Andreasen L, Mundbjerg K, Kruhoffer M, Brinch M, 
Schierup MH, Bjergbaek L, Andersen AH. 2012. DNA Topoisomerases maintain promoters in a 
state competent for transcriptional activation in Saccharomyces cerevisiae.

Full Description: To investigate the role of DNA topoisomerases in transcription, we have studied global 
gene expression in Saccharomyces cerevisiae cells deficient for topoisomerases I and II 
and performed single-gene analyses to support our findings. The genome-wide studies show a
general transcriptional down-regulation  upon lack of the enzymes, which correlates with 
gene activity but not gene length. Furthermore, our data reveal a distinct subclass of 
genes with a strong requirement for topoisomerases. These genes are characterized by high 
transcriptional plasticity, chromatin regulation, TATA box presence, and enrichment of a 
nucleosome at a critical position in the promoter region, in line with a repressible/
inducible mode of regulation. Single-gene studies with a range of genes belonging to this 
group demonstrate that topoisomerases play an important role during activation of these 
genes. Subsequent in-depth analysis of  the inducible PHO5 gene reveals that 
topoisomerases are essential for binding of  the Pho4p transcription factor to the PHO5 
promoter, which is required for promoter nucleosome removal during activation. In contrast
, topoisomerases are dispensable for constitutive transcription initiation and elongation 
of PHO5, as  well as the nuclear entrance of Pho4p. Finally, we provide evidence that 
topoisomerases are required to maintain the PHO5 promoter in a superhelical state, which 
is competent for proper activation. In conclusion, our results reveal a hitherto unknown 
function of topoisomerases during transcriptional activation of genes with a repressible/
inducible mode of regulation.

PMID: 23284296

GEO ID: GSE22809.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE22809.final.pcl	DNA Topoisomerases maintain promoters in a state competent for transcriptional activation in Saccharomyces cerevisiae.	12	transcription	1

File last modified: Thu Apr 25 2024

---


# Penacho_2012_PMID_22065482

Citation: Penacho V, Blondin B, Valero E, Gonzalez R. 2012. Flocculation and transcriptional 
adaptation to fermentation conditions in a recombinant wine yeast strain defective for 
KNR4/SMI1.

Full Description: KNR4 defective recombinant wine yeast strains were previously shown to oversecrete 
mannoproteins during alcoholic fermentation and, depending on the genetic background, to 
contribute to protein stability of white wines. We have tried to get a deeper insight into
the consequences of KNR4 deletion in a wine yeast strain, from both a biological and an 
enological standpoint, and to understand the mechanisms leading to improved mannoprotein 
release. In fermentation experiments, followed by aging on lees, and compared to the 
parent strain, the recombinant strain shows increased release of mannoproteins during the 
fermentation but little increase during aging. Mannoprotein release by the recombinant 
strain takes place mainly during the fermentation step. In contrast,  autolysis of the 
recombinant strain keeps going after aging for 78 days. In addition, the recombinant 
strain is moderately flocculent, which would be interesting for the production of 
sparkling wines. This might be related to changes in the expression of Flo1p-regulated 
genes. The new biological processes  affected by KNR4 deletion in wine yeasts, as revealed
by this transcriptomic study are flocculation, adaptation to anaerobiosis, oxidative 
stress response, and ethanol tolerance, as well as FKS1 overexpression; but no 
overexpression was  detected for genes coding for major structural mannoproteins of the 
cell wall.

PMID: 22065482

GEO ID: GSE26965.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE26965.final.pcl	Flocculation and transcriptional adaptation to fermentation conditions in a recombinant wine yeast strain defective for KNR4/SMI1.	6	fermentation|cell wall organization	2

File last modified: Thu Apr 25 2024

---


# Prunuske_2012_PMID_22203981

Citation: Prunuske AJ, Waltner JK, Kuhn P, Gu B, Craig EA. 2012. Role for the molecular chaperones 
Zuo1 and Ssz1 in quorum sensing via activation  of the transcription factor Pdr1.

Full Description: Zuo1 functions as a J-protein cochaperone of its partner Hsp70. In addition, the  C 
terminus of Zuo1 and the N terminus of Ssz1, with which Zuo1 forms a heterodimer, can 
independently activate the Saccharomyces cerevisiae transcription factor pleiotropic drug 
resistance 1 (Pdr1). Here we report that activation of Pdr1 by Zuo1 or Ssz1 causes 
premature growth arrest of cells during the diauxic shift, as they adapt to the changing 
environmental conditions. Conversely, cells lacking Zuo1 or Ssz1 overgrow, arresting at a 
higher cell density, an effect overcome by activation of Pdr1. Cells lacking the genes 
encoding plasma membrane transporters Pdr5 and Snq2, two targets of Pdr1, also overgrow at
the diauxic shift. Adding conditioned medium harvested from cultures  of wild-type cells 
attenuated the overgrowth of both zuo1Deltassz1Delta and pdr5Deltasnq2Delta cells, 
suggesting the extracellular presence of molecules that signal growth arrest. In addition
, our yeast two-hybrid analysis revealed an interaction between Pdr1 and both Zuo1 and 
Ssz1. Together, our results support a  model in which (i) membrane transporters, encoded 
by Pdr1 target genes act to promote cell-cell communication by exporting quorum sensing 
molecules, in addition to playing a role in pleiotropic drug resistance; and (ii) 
molecular chaperones function at promoters to regulate this intercellular communication 
through their activation of the transcription factor Pdr1.

PMID: 22203981

GEO ID: GSE31693.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE31693.final.pcl	Role for the molecular chaperones Zuo1 and Ssz1 in quorum sensing via activation  of the transcription factor Pdr1.	3	diauxic shift	1

File last modified: Thu Apr 25 2024

---


# Radman-Livaja_2012_PMID_22807688

Citation: Radman-Livaja M, Quan TK, Valenzuela L, Armstrong JA, van Welsem T, Kim T, Lee LJ, 
Buratowski S, van Leeuwen F, Rando OJ, Hartzog GA. 2012. A key role for Chd1 in histone H3
dynamics at the 3' ends of long genes in yeast.

Full Description: Chd proteins are ATP-dependent chromatin remodeling enzymes implicated in biological 
functions from transcriptional elongation to control of pluripotency.  Previous studies of
the Chd1 subclass of these proteins have implicated them in diverse roles in gene 
expression including functions during initiation, elongation, and termination. Furthermore
, some evidence has suggested a role for  Chd1 in replication-independent histone exchange
or assembly. Here, we examine roles of Chd1 in replication-independent dynamics of histone
H3 in both Drosophila and yeast. We find evidence of a role for Chd1 in H3 dynamics in 
both  organisms. Using genome-wide ChIP-on-chip analysis, we find that Chd1 influences  
histone turnover at the 5' and 3' ends of genes, accelerating H3 replacement at the 5' 
ends of genes while protecting the 3' ends of genes from excessive H3 turnover. Although 
consistent with a direct role for Chd1 in exchange, these results may indicate that Chd1 
stabilizes nucleosomes perturbed by transcription. Curiously, we observe a strong effect 
of gene length on Chd1's effects on H3 turnover. Finally, we show that Chd1 also affects 
histone modification patterns over genes, likely as a consequence of its effects on 
histone replacement. Taken  together, our results emphasize a role for Chd1 in histone 
replacement in both budding yeast and Drosophila melanogaster, and surprisingly they show 
that the major effects of Chd1 on turnover occur at the 3' ends of genes.

PMID: 22807688

GEO ID: GSE38496.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE38496.final.pcl	A key role for Chd1 in histone H3 dynamics at the 3' ends of long genes in yeast.	2	histone modification|chromatin organization	2

File last modified: Thu Apr 25 2024

---


# Rupwate_2012_PMID_22673525

Citation: Rupwate SD, Rupwate PS, Rajasekharan R. 2012. Regulation of lipid biosynthesis by 
phosphatidylinositol-specific phospholipase C through the transcriptional repression of 
upstream activating sequence inositol containing genes.

Full Description: The regulation of phospholipid biosynthesis in Saccharomyces cerevisiae through cis-acting
upstream activating sequence inositol (UAS(ino)) and trans-acting elements, such as the 
INO2-INO4 complex and OPI1 by inositol supplementation in growth is thoroughly studied. In
this study, we provide evidence for the regulation of lipid biosynthesis by 
phosphatidylinositol-specific phospholipase C (PLC) through UAS(ino) and the trans-acting 
elements. Gene expression analysis and radiolabelling experiments demonstrated that the 
overexpression of rice PLC in yeast cells altered phospholipid biosynthesis at the levels 
of transcriptional and enzyme activity. This is the first report implicating PLC in the 
direct regulation of lipid biosynthesis.

PMID: 22673525

GEO ID: GSE35677.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE35677.final.pcl	Regulation of lipid biosynthesis by phosphatidylinositol-specific phospholipase C through the transcriptional repression of upstream activating sequence inositol containing genes.	4	lipid metabolism	1

File last modified: Thu Apr 25 2024

---


# Sanz_2012_PMID_22621902

Citation: Sanz AB, Garcia R, Rodriguez-Pena JM, Diez-Muniz S, Nombela C, Peterson CL, Arroyo J. 2012
 Chromatin remodeling by the SWI/SNF complex is essential for transcription mediated by 
the yeast cell wall integrity MAPK pathway.

Full Description: In Saccharomyces cerevisiae, the transcriptional program triggered by cell wall stress is 
coordinated by Slt2/Mpk1, the mitogen-activated protein kinase (MAPK) of the cell wall 
integrity (CWI) pathway, and is mostly mediated by the transcription factor Rlm1. Here we 
show that the SWI/SNF chromatin-remodeling complex plays a critical role in orchestrating 
the transcriptional response regulated by Rlm1. swi/snf mutants show drastically reduced 
expression of cell wall stress-responsive genes and hypersensitivity to cell wall-
interfering compounds. On stress, binding of RNA Pol II to the promoters of these genes 
depends on Rlm1, Slt2, and SWI/SNF. Rlm1 physically interacts with SWI/SNF to direct its 
association to target promoters. Finally, we observe nucleosome displacement at the CWI-
responsive gene MLP1/KDX1, which relies on the SWI/SNF complex. Taken together, our 
results identify the SWI/SNF complex as a key element of the CWI MAPK pathway that 
mediates the chromatin remodeling necessary  for adequate transcriptional response to cell
wall stress.

PMID: 22621902

GEO ID: GSE31176

PCL filename	short description	# conditions	tags	# channels
GSE31176_final.pcl	Chromatin remodeling by the SWI/SNF complex is essential for transcription mediated by the yeast cell wall integrity MAPK pathway.	18	cell wall organization|stress|chromatin organization	1

File last modified: Thu Apr 25 2024

---


# Sheltzer_2012_PMID_22802626

Citation: Sheltzer JM, Torres EM, Dunham MJ, Amon A. 2012. Transcriptional consequences of 
aneuploidy.

Full Description: Aneuploidy, or an aberrant karyotype, results in developmental disabilities and has been 
implicated in tumorigenesis. However, the causes of aneuploidy-induced phenotypes and the 
consequences of aneuploidy on cell physiology remain poorly understood. We have performed 
a metaanalysis on gene expression data from aneuploid cells in diverse organisms, 
including yeast, plants, mice, and humans.  We found highly related gene expression 
patterns that are conserved between species: genes that were involved in the response to 
stress were consistently upregulated, and genes associated with the cell cycle and cell 
proliferation were downregulated in aneuploid cells. Within species, different 
aneuploidies induced  similar changes in gene expression, independent of the specific 
chromosomal aberrations. Taken together, our results demonstrate that aneuploidies of 
different chromosomes and in different organisms impact similar cellular pathways and 
cause a stereotypical antiproliferative response that must be overcome before 
transformation.

PMID: 22802626

GEO ID: GSE35853

PCL filename	short description	# conditions	tags	# channels
GSE35853_final.pcl	Transcriptional consequences of aneuploidy.	8	stress	2

File last modified: Thu Apr 25 2024

---


# Simmons_Kovacs_2012_PMID_22306294

Citation: Simmons Kovacs LA, Mayhew MB, Orlando DA, Jin Y, Li Q, Huang C, Reed SI, Mukherjee S, 
Haase SB. 2012. Cyclin-dependent kinases are regulators and effectors of oscillations 
driven by a transcription factor network.

Full Description: During embryonic cell cycles, B-cyclin-CDKs function as the core component of an  
autonomous oscillator. Current models for the cell-cycle oscillator in nonembryonic cells 
are slightly more complex, incorporating multiple G1, S phase, and mitotic cyclin-CDK 
complexes. However, periodic events persist in yeast cells lacking all S phase and mitotic
B-cyclin genes, challenging the assertion that cyclin-CDK complexes are essential for 
oscillations. These and other results led  to the proposal that a network of sequentially 
activated transcription factors functions as an underlying cell-cycle oscillator. Here we 
examine the individual  contributions of a transcription factor network and cyclin-CDKs to
the maintenance of cell-cycle oscillations. Our findings suggest that while cyclin-CDKs 
are not required for oscillations, they do contribute to oscillation  robustness. A model 
emerges in which cyclin expression (thereby, CDK activity) is entrained to an autonomous 
transcriptional oscillator. CDKs then modulate oscillator function and serve as effectors 
of the oscillator.

PMID: 22306294

GEO ID: GSE32974

PCL filename	short description	# conditions	tags	# channels
GSE32974_final.pcl	Cyclin-dependent kinases are regulators and effectors of oscillations driven by a transcription factor network.	48	protein phosphorylation|mitotic cell cycle	1

File last modified: Thu Apr 25 2024

---


# Spedale_2012_PMID_21976730

Citation: Spedale G, Meddens CA, Koster MJ, Ko CW, van Hooff SR, Holstege FC, Timmers HT, Pijnappel 
WW. 2012. Tight cooperation between Mot1p and NC2beta in regulating genome-wide 
transcription, repression of transcription following heat shock induction and genetic 
interaction with SAGA.

Full Description: TATA-binding protein (TBP) is central to the regulation of eukaryotic transcription 
initiation. Recruitment of TBP to target genes can be positively regulated by one of two 
basal transcription factor complexes: SAGA or TFIID. Negative regulation of TBP promoter 
association can be performed by Mot1p or the  NC2 complex. Recent evidence suggests that 
Mot1p, NC2 and TBP form a DNA-dependent protein complex. Here, we compare the functions of
Mot1p and NC2betaduring basal and activated transcription using the anchor-away technique 
for conditional nuclear depletion. Genome-wide expression analysis indicates that both 
proteins regulate a highly similar set of genes. Upregulated genes were enriched for SAGA 
occupancy, while downregulated genes preferred TFIID binding. Mot1p and NC2beta depletion 
during heat shock resulted in failure to downregulate gene expression after initial 
activation, which was accompanied by increased TBP  and RNA pol II promoter occupancies. 
Depletion of Mot1p or NC2beta displayed preferential synthetic lethality with the TBP-
interaction module of SAGA. Our results support the model that Mot1p and NC2beta directly 
cooperate in vivo to regulate TBP function, and that they are involved in maintaining 
basal expression levels as well as in resetting gene expression after induction by stress.

PMID: 21976730

GEO ID: GSE30148

PCL filename	short description	# conditions	tags	# channels
GSE30148_final.pcl	Tight cooperation between Mot1p and NC2beta in regulating genome-wide transcription, repression of transcription following heat shock induction and genetic interaction with SAGA.	12	transcription	2

File last modified: Thu Apr 25 2024

---


# Suzuki_2012_PMID_22897823

Citation: Suzuki T, Iwahashi Y. 2012. Comprehensive gene expression analysis of type B 
trichothecenes.

Full Description: Type B trichothecenes, deoxynivalenol (DON) and nivalenol (NIV), are secondary metabolites
of Fusarium species and are major pollutants in food and feed products. Recently, the 
production trend of their derivatives, 3-acetyldeoxynivalenol (3-AcDON), 15-
acetyldeoxynivalenol (15-AcDON), and 4-acetylnivalenol (4-AcNIV or fusarenon-X), has been 
changing in various regions  worldwide. Although in vivo behavior has been reported, it is
necessary to acquire more detailed information about these derivatives. Here, the yeast 
PDR5 mutant was used for toxicity evaluation, and the growth test revealed that DON, 15-
AcDON, and 4-AcNIV had higher toxicity compared to 3-AcDON and NIV. 15-AcDON exerted the 
most significant gene expression changes, and cellular localization clustering exhibited 
repression of mitochondrial ribosomal genes. This study suggests that the toxicity trends 
of both DON products (DON and its derivatives)  and NIV products (NIV and its derivatives
) are similar to those observed in mammalian cells, with a notable toxic response to 15-
AcDON.

PMID: 22897823

GEO ID: GSE36954.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE36954.remapped.final.pcl	Comprehensive gene expression analysis of type B trichothecenes.	18	chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Thibault_2012_PMID_23000174

Citation: Thibault G, Shui G, Kim W, McAlister GC, Ismail N, Gygi SP, Wenk MR, Ng DT. 2012. The 
membrane stress response buffers lethal effects of lipid disequilibrium by reprogramming 
the protein homeostasis network.

Full Description: Lipid composition can differ widely among organelles and even between leaflets of a 
membrane. Lipid homeostasis is critical because disequilibrium can have disease outcomes. 
Despite their importance, mechanisms maintaining lipid homeostasis remain poorly 
understood. Here, we establish a model system to study the global effects of lipid 
imbalance. Quantitative lipid profiling was integral to monitor  changes to lipid 
composition and for system validation. Applying global transcriptional and proteomic 
analyses, a dramatically altered biochemical landscape was revealed from adaptive cells. 
The resulting composite regulation we term the membrane stress response (MSR) confers 
compensation, not through restoration of lipid composition, but by remodeling the protein 
homeostasis network. To validate its physiological significance, we analyzed the unfolded 
protein response (UPR), one facet of the MSR and a key regulator of protein homeostasis. 
We demonstrate that the UPR maintains protein biogenesis, quality control, and membrane 
integrity-functions otherwise lethally compromised in lipid dysregulated cells.

PMID: 23000174

GEO ID: GSE39419

PCL filename	short description	# conditions	tags	# channels
GSE39419_final.pcl	The membrane stress response buffers lethal effects of lipid disequilibrium by reprogramming the protein homeostasis network.	16	lipid metabolism|response to unfolded protein	2

File last modified: Thu Apr 25 2024

---


# Travesa_2012_PMID_22333915

Citation: Travesa A, Kuo D, de Bruin RA, Kalashnikova TI, Guaderrama M, Thai K, Aslanian A, Smolka 
MB, Yates JR 3rd, Ideker T, Wittenberg C. 2012. DNA replication stress differentially 
regulates G1/S genes via Rad53-dependent inactivation of Nrm1.

Full Description: MBF and SBF transcription factors regulate a large family of coordinately expressed G1/S 
genes required for early cell-cycle functions including DNA replication and repair. SBF is
inactivated upon S-phase entry by Clb/CDK whereas  MBF targets are repressed by the co-
repressor, Nrm1. Using genome-wide expression analysis of cells treated with methyl 
methane sulfonate (MMS), hydroxyurea (HU) or camptothecin (CPT), we show that genotoxic 
stress during S phase specifically  induces MBF-regulated genes. This occurs via direct 
phosphorylation of Nrm1 by Rad53, the effector checkpoint kinase, which prevents its 
binding to MBF target promoters. We conclude that MBF-regulated genes are distinguished 
from SBF-regulated genes by their sensitivity to activation by the S-phase checkpoint, 
thereby, providing an effective mechanism for enhancing DNA replication and repair and 
promoting genome stability.

PMID: 22333915

GEO ID: GSE33695

PCL filename	short description	# conditions	tags	# channels
GSE33695_final.pcl	DNA replication stress differentially regulates G1/S genes via Rad53-dependent inactivation of Nrm1.	61	stress|cell cycle regulation|DNA damage stimulus	2

File last modified: Thu Apr 25 2024

---


# Tyo_2012_PMID_22380681

Citation: Tyo KE, Liu Z, Petranovic D, Nielsen J. 2012. Imbalance of heterologous protein folding 
and disulfide bond formation rates yields runaway oxidative stress.

Full Description: BACKGROUND: The protein secretory pathway must process a wide assortment of native 
proteins for eukaryotic cells to function. As well, recombinant protein secretion is used 
extensively to produce many biologics and industrial enzymes. Therefore, secretory pathway
dysfunction can be highly detrimental to the cell and can drastically inhibit product 
titers in biochemical production. Because the secretory pathway is a highly-integrated, 
multi-organelle system, dysfunction can happen at many levels and dissecting the root 
cause can be challenging. In this study, we apply a systems biology approach to analyze 
secretory pathway dysfunctions resulting from heterologous production of a small protein (
insulin precursor) or a larger protein (alpha-amylase). RESULTS: HAC1-dependent and 
independent dysfunctions and cellular responses were apparent across multiple datasets. In
particular, processes involving (a) degradation of protein/recycling amino acids, (b) 
overall transcription/translation repression, and (c) oxidative  stress were broadly 
associated with secretory stress. CONCLUSIONS: Apparent runaway oxidative stress due to 
radical production observed here and elsewhere can be explained by a futile cycle of 
disulfide formation and breaking that consumes reduced glutathione and produces reactive 
oxygen species. The futile cycle is dominating when protein folding rates are low relative
to disulfide bond formation rates. While not strictly conclusive with the present data, 
this insight does provide a molecular interpretation to an, until now, largely empirical 
understanding of optimizing heterologous protein secretion. This molecular insight has 
direct implications on engineering a broad range of recombinant proteins for secretion and
provides potential hypotheses for the root causes of several secretory-associated diseases.

PMID: 22380681

GEO ID: GSE27062

PCL filename	short description	# conditions	tags	# channels
GSE27062_final.pcl	Imbalance of heterologous protein folding and disulfide bond formation rates yields runaway oxidative stress.	18	response to unfolded protein|oxidative stress	1

File last modified: Thu Apr 25 2024

---


# Vizoso-Vazquez_2012_PMID_22189861

Citation: Vizoso-Vazquez A, Lamas-Maceiras M, Becerra M, Gonzalez-Siso MI, Rodriguez-Belmonte E, 
Cerdan ME. 2012. Ixr1p and the control of the Saccharomyces cerevisiae hypoxic response.

Full Description: In Saccharomyces cerevisiae, adaptation to hypoxia/anaerobiosis requires the 
transcriptional induction or derepression of multiple genes organized in regulons 
controlled by specific transcriptional regulators. Ixr1p is a transcriptional regulatory 
factor that causes aerobic repression of several hypoxic genes (COX5B, TIR1, and HEM13) 
and also the activation of HEM13 during hypoxic growth. Analysis of the transcriptome of 
the wild-type strain BY4741 and its isogenic derivative Deltaixr1, grown in aerobic and 
hypoxic conditions, reveals differential regulation of genes related not only to the 
hypoxic and oxidative stress responses but also to the re-adaptation of catabolic and 
anabolic fluxes in response to oxygen limitation. The function of Ixr1p in the 
transcriptional regulation of genes from the sulfate assimilation pathway and other 
pathways producing alpha-keto acids is of biotechnological importance for industries based
on yeast-derived fermentation products.

PMID: 22189861

GEO ID: GSE30046

PCL filename	short description	# conditions	tags	# channels
GSE30046_final.pcl	Ixr1p and the control of the Saccharomyces cerevisiae hypoxic response.	12	oxygen level alteration	2

File last modified: Thu Apr 25 2024

---


# Voordeckers_2012_PMID_22882838

Citation: Voordeckers K, De Maeyer D, van der Zande E, Vinces MD, Meert W, Cloots L, Ryan O, Marchal
K, Verstrepen KJ. 2012. Identification of a complex genetic network underlying 
Saccharomyces cerevisiae colony morphology.

Full Description: When grown on solid substrates, different microorganisms often form colonies with very 
specific morphologies. Whereas the pioneers of microbiology often used colony morphology 
to discriminate between species and strains, the phenomenon has not received much 
attention recently. In this study, we use a genome-wide assay in the model yeast 
Saccharomyces cerevisiae to identify all genes that affect colony morphology. We show that
several major signalling cascades, including the  MAPK, TORC, SNF1 and RIM101 pathways 
play a role, indicating that morphological changes are a reaction to changing environments
. Other genes that affect colony morphology are involved in protein sorting and epigenetic
regulation. Interestingly, the screen reveals only few genes that are likely to play a 
direct role in establishing colony morphology, with one notable example being FLO11, a 
gene encoding a cell-surface adhesin that has already been implicated in colony morphology
, biofilm formation, and invasive and pseudohyphal growth. Using a series of modified 
promoters for fine-tuning FLO11 expression, we confirm the central role of Flo11 and show 
that differences in FLO11 expression result in distinct colony morphologies. Together, our
results provide a first comprehensive look at the complex genetic network that underlies 
the diversity in the morphologies of yeast colonies.

PMID: 22882838

GEO ID: GSE36151.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE36151.final.pcl	Identification of a complex genetic network underlying Saccharomyces cerevisiae colony morphology.	18	colony morphology	1

File last modified: Thu Apr 25 2024

---


# Wuttke_2012_PMID_22912585

Citation: Wuttke D, Connor R, Vora C, Craig T, Li Y, Wood S, Vasieva O, Shmookler Reis R, Tang F, de
Magalhaes JP. 2012. Dissecting the gene network of dietary restriction to identify 
evolutionarily conserved pathways and new functional genes.

Full Description: Dietary restriction (DR), limiting nutrient intake from diet without causing malnutrition
, delays the aging process and extends lifespan in multiple organisms. The conserved life-
extending effect of DR suggests the involvement of  fundamental mechanisms, although these
remain a subject of debate. To help decipher the life-extending mechanisms of DR, we first
compiled a list of genes that if genetically altered disrupt or prevent the life-extending
effects of DR.  We called these DR-essential genes and identified more than 100 in model 
organisms such as yeast, worms, flies, and mice. In order for other researchers to benefit
from this first curated list of genes essential for DR, we established an online database 
called GenDR (http://genomics.senescence.info/diet/). To dissect the interactions of DR-
essential genes and discover the underlying lifespan-extending mechanisms, we then used a 
variety of network and systems biology approaches to analyze the gene network of DR. We 
show that DR-essential genes are more conserved at the molecular level and have more 
molecular interactions than expected by chance. Furthermore, we employed a guilt-by-
association method to predict novel DR-essential genes. In budding yeast, we predicted 
nine genes related to vacuolar functions; we show experimentally that mutations deleting 
eight of those genes prevent the life-extending effects of DR. Three of these mutants (
OPT2, FRE6, and RCR2) had extended lifespan under ad libitum, indicating that the lack of 
further longevity under DR is not caused by a general compromise of fitness. These results
demonstrate how network analyses of DR using GenDR can be used to make phenotypically 
relevant predictions. Moreover, gene-regulatory circuits reveal that the DR-induced 
transcriptional signature in yeast involves nutrient-sensing, stress responses and meiotic
transcription factors. Finally, comparing the influence of gene expression changes during 
DR on the interactomes of multiple organisms led us to suggest that DR commonly suppresses
translation, while stimulating an ancient reproduction-related process.

PMID: 22912585

GEO ID: GSE38635.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE38635.final.pcl	Dissecting the gene network of dietary restriction to identify evolutionarily conserved pathways and new functional genes.	2	carbon utilization|stress|cell aging	1

File last modified: Thu Apr 25 2024

---


# Yona_2012_PMID_23197825

Citation: Yona AH, Manor YS, Herbst RH, Romano GH, Mitchell A, Kupiec M, Pilpel Y, Dahan O. 2012. 
Chromosomal duplication is a transient evolutionary solution to stress.

Full Description: Aneuploidy, an abnormal number of chromosomes, is a widespread phenomenon found in 
unicellulars such as yeast, as well as in plants and in mammalians, especially in cancer. 
Aneuploidy is a genome-scale aberration that imposes a severe burden on the cell, yet 
under stressful conditions specific aneuploidies confer a selective advantage. This dual 
nature of aneuploidy raises the question of whether it can serve as a stable and 
sustainable evolutionary adaptation. To clarify this, we conducted a set of laboratory 
evolution experiments in yeast and followed the long-term dynamics of aneuploidy under 
diverse conditions. Here we show that chromosomal duplications are first acquired as a 
crude solution to stress, yet only as transient solutions that are eliminated and replaced
by more  efficient solutions obtained at the individual gene level. These transient 
dynamics of aneuploidy were repeatedly observed in our laboratory evolution experiments; 
chromosomal duplications gained under stress were eliminated not only when the stress was 
relieved, but even if it persisted. Furthermore, when stress was applied gradually rather 
than abruptly, alternative solutions appear to have emerged, but not aneuploidy. Our 
findings indicate that chromosomal duplication is a first evolutionary line of defense, 
that retains survivability under strong and abrupt selective pressures, yet it merely 
serves as a quick fix, whereas more refined and sustainable solutions take over. Thus, in 
the perspective of genome evolution trajectory, aneuploidy is a useful yet short-lived 
intermediate that facilitates further adaptation.

PMID: 23197825

GEO ID: GSE40817

PCL filename	short description	# conditions	tags	# channels
GSE40817_final.pcl	Chromosomal duplication is a transient evolutionary solution to stress.	40	evolution|heat shock|stress	1

File last modified: Thu Apr 25 2024

---


# de_Castro_2012_PMID_23092287

Citation: de Castro PA, Savoldi M, Bonatto D, Malavazi I, Goldman MH, Berretta AA, Goldman GH. 2012
 Transcriptional profiling of Saccharomyces cerevisiae exposed to propolis.

Full Description: BACKGROUND: Propolis is a natural product of plant resins collected by honeybees  (Apis 
mellifera) from various plant sources. Our previous studies indicated that  propolis 
sensitivity is dependent on the mitochondrial function and that vacuolar acidification and
autophagy are important for yeast cell death caused by propolis. Here, we extended our 
understanding of propolis-mediated cell death in  the yeast Saccharomyces cerevisiae by 
applying systems biology tools to analyze the transcriptional profiling of cells exposed 
to propolis. METHODS: We have used transcriptional profiling of S. cerevisiae exposed to 
propolis. We validated our  findings by using real-time PCR of selected genes. Systems 
biology tools (physical protein-protein interaction [PPPI] network) were applied to 
analyse the propolis-induced transcriptional bevavior, aiming to identify which pathways 
are  modulated by propolis in S. cerevisiae and potentially influencing cell death. 
RESULTS: We were able to observe 1,339 genes modulated in at least one time point when 
compared to the reference time (propolis untreated samples) (t-test, p-value 0.01). 
Enrichment analysis performed by Gene Ontology (GO) Term finder tool showed enrichment for
several biological categories among the genes up-regulated  in the microarray 
hybridization such as transport and transmembrane transport and response to stress. Real-
time RT-PCR analysis of selected genes showed by our microarray hybridization approach was
capable of providing information about S. cerevisiae gene expression modulation with a 
considerably high level of confidence. Finally, a physical protein-protein (PPPI) network 
design and global  topological analysis stressed the importance of these pathways in 
response of S.  cerevisiae to propolis and were correlated with the transcriptional data 
obtained thorough the microarray analysis. CONCLUSIONS: In summary, our data indicate that
propolis is largely affecting several pathways in the eukaryotic cell. However, the most 
prominent pathways are related to oxidative stress, mitochondrial electron transport chain
, vacuolar acidification, regulation of macroautophagy associated with protein target to 
vacuole, cellular response to starvation, and negative regulation of transcription from 
RNA polymerase II promoter. Our work emphasizes again the importance of S. cerevisiae as a
model system to understand  at molecular level the mechanism whereby propolis causes cell 
death in this organism at the concentration herein tested. Our study is the first one that
investigates systematically by using functional genomics how propolis influences  and 
modulates the mRNA abundance of an organism and may stimulate further work on the propolis
-mediated cell death mechanisms in fungi.

PMID: 23092287

GEO ID: GSE33971.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE33971.final.pcl	Transcriptional profiling of Saccharomyces cerevisiae exposed to propolis.	8	chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# de_Kok_2012_PMID_22257278

Citation: de Kok S, Nijkamp JF, Oud B, Roque FC, de Ridder D, Daran JM, Pronk JT, van Maris AJ. 2012
 Laboratory evolution of new lactate transporter genes in a jen1Delta mutant of 
Saccharomyces cerevisiae and their identification as ADY2 alleles by whole-genome 
resequencing and transcriptome analysis.

Full Description: Laboratory evolution is a powerful approach in applied and fundamental yeast research, but
complete elucidation of the molecular basis of evolved phenotypes remains a challenge. In 
this study, DNA microarray-based transcriptome analysis and whole-genome resequencing were
used to investigate evolution of novel lactate transporters in Saccharomyces cerevisiae 
that can replace Jen1p, the only documented S. cerevisiae lactate transporter. To this end
, a jen1Delta mutant was evolved for growth on lactate in serial batch cultures. Two 
independent evolution experiments yielded growth on lactate as sole carbon source (0.14 
and 0.18 h(-1)  , respectively). Transcriptome analysis did not provide leads, but whole-
genome resequencing showed different single-nucleotide changes (C755G/Leu219Val and C655G/
Ala252Gly) in the acetate transporter gene ADY2. Introduction of these ADY2 alleles in a 
jen1Delta ady2Delta strain enabled growth on lactate (0.14 h(-1) for Ady2p(Leu219Val) and 
0.12 h(-1) for Ady2p(Ala252Gly) ), demonstrating that these  alleles of ADY2 encode 
efficient lactate transporters. Depth of coverage of DNA sequencing, combined with 
karyotyping, gene deletions and diagnostic PCR, showed  that an isochromosome III (c. 475 
kb) with two additional copies of ADY2(C755G) had been formed via crossover between 
retrotransposons YCLWDelta15 and YCRCDelta6. The isochromosome formation shows how even 
short periods of selective pressure can cause substantial karyotype changes.

PMID: 22257278

GEO ID: GSE31634.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE31634.final.pcl	Laboratory evolution of new lactate transporter genes in a jen1Delta mutant of Saccharomyces cerevisiae and their identification as ADY2 alleles by whole-genome resequencing and transcriptome analysis.	6	carbon utilization|evolution	1

File last modified: Thu Apr 25 2024

---


# Alkim_2013_PMID_23864114

Citation: Alkim C, Benbadis L, Yilmaz U, Cakar ZP, Francois JM. 2013. Mechanisms other than 
activation of the iron regulon account for the hyper-resistance to cobalt of a 
Saccharomyces cerevisiae strain obtained by evolutionary engineering.

Full Description: Cobalt is an important metal ion with magnetic properties that is widely used for several 
industrial applications. Overexposure to cobalt ions can be highly toxic  for the 
organisms because they usually overwhelm the endogenous physiological system that 
maintains their homeostasis causing (geno)toxic effects. To gain insight into the 
mechanism of cobalt toxicity, we characterized at the molecular  and genetic levels a 
cobalt resistant CI25E Saccharomyces cerevisiae strain previously isolated by an in vivo 
evolutionary engineering strategy, and which was able to grow on 5 to 10 mM CoCl2. This 
evolved strain showed cross-resistance to other metal ions including iron, manganese, 
nickel and zinc, but not to copper. Moreover, the cobalt resistant trait was semi-dominant
, and linked to more than one gene, as indicated by the absence of 2(+):2(-) segregation 
of the cobalt resistance. Genome wide transcriptional profiling revealed a constitutive  
activation of the iron regulon that could be accounted for by a constitutive nuclear 
localization of the transcriptional activator Aft1. However, the presence of Aft1 in the 
nucleus was not a prerequisite for hyper-resistance to cobalt, since a mutant defective in
nuclear monothiol glutaredoxin encoding GRX3 and GRX4 that also leads to nuclear 
localization of Aft1 was cobalt hypersensitive. In addition, the loss of AFT1 only 
partially abolished the cobalt resistance in the  evolved strain, and the deletion of COT1
encoding the major vacuolar transporter  of cobalt had only a minor effect on this trait. 
Paradoxically to the activation  of iron regulon, the evolved strain was hypersensitive to
the iron chelator BPS,  and this hypersensitivity was abrogated by cobalt ions. Taken 
together, this work suggested that cobalt resistance is not merely dependent upon 
activation of AFT1, but it likely implicates other mechanisms including intracellular 
reallocation of iron into important compartments whose function is dependent on this metal
and adaptation of some cellular proteins to use Co(2+) in place of Fe(2+) for their 
catalytic activities.

PMID: 23864114

GEO ID: GSE39185.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE39185.final.pcl	Mechanisms other than activation of the iron regulon account for the hyper-resistance to cobalt of a Saccharomyces cerevisiae strain obtained by evolutionary engineering.	9	evolution|metal or metalloid ion stress	1

File last modified: Thu Apr 25 2024

---


# Almario_2013_PMID_23613173

Citation: Almario MP, Reyes LH, Kao KC. 2013. Evolutionary engineering of Saccharomyces cerevisiae 
for enhanced tolerance to hydrolysates of lignocellulosic biomass.

Full Description: Lignocellulosic biomass has become an important feedstock to mitigate current ethical and 
economical concerns related to the bio-based production of fuels and  chemicals. During 
the pre-treatment and hydrolysis of the lignocellulosic biomass, a complex mixture of 
sugars and inhibitors are formed. The inhibitors interfere with microbial growth and 
product yields. This study uses an adaptive laboratory evolution method called visualizing
evolution in real-time (VERT) to uncover the molecular mechanisms associated with 
tolerance to hydrolysates of lignocellulosic biomass in Saccharomyces cerevisiae. VERT 
enables a more rational scheme for isolating adaptive mutants for characterization and 
molecular analyses. Subsequent growth kinetic analyses of the mutants in individual and 
combinations of common inhibitors present in hydrolysates (acetic acid, furfural, and 
hydroxymethylfurfural) showed differential levels of resistance to different  inhibitors, 
with enhanced growth rates up to 57%, 12%, 22%, and 24% in hydrolysates, acetic acid, HMF 
and furfural, respectively. Interestingly, some of the adaptive mutants exhibited reduced 
fitness in the presence of individual inhibitors, but showed enhanced fitness in the 
presence of combinations of inhibitors compared to the parental strains. Transcriptomic 
analysis revealed different mechanisms for resistance to hydrolysates and a potential 
cross adaptation between oxidative stress and hydrolysates tolerance in several of the  
mutants.

PMID: 23613173

GEO ID: GSE44085.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE44085.remapped.final.pcl	Evolutionary engineering of Saccharomyces cerevisiae for enhanced tolerance to hydrolysates of lignocellulosic biomass.	24	evolution	2

File last modified: Thu Apr 25 2024

---


# Aow_2013_PMID_23558744

Citation: Aow JS, Xue X, Run JQ, Lim GF, Goh WS, Clarke ND. 2013. Differential binding of the 
related transcription factors Pho4 and Cbf1 can tune  the sensitivity of promoters to 
different levels of an induction signal.

Full Description: Transcription factors that belong to the same family typically have similar, but  not 
identical, binding specificities. As such, they can be expected to compete differentially 
for binding to different variants of their binding sites. Pho4 is  a yeast factor whose 
nuclear concentration is up-regulated in low phosphate, while the related factor, Cbf1, is
constitutively expressed. We constructed 16 GFP-reporter genes containing all palindromic 
variants of the motif NNCACGTGNN, and determined their activities at a range of phosphate 
concentrations. Pho4 affinity did not explain expression data well except under fully 
induced conditions. However, reporter activity was quantitatively well explained under all
conditions by a model in which Cbf1 itself has modest activating activity, and Pho4 and 
Cbf1 compete with one another. Chromatin immunoprecipitation and computational analyses of
natural Pho4 target genes, along with the activities of the reporter constructs, indicates
that genes differ in their sensitivity to intermediate induction signals in part because 
of differences in their affinity for Cbf1. The induction sensitivity of both natural Pho4 
target genes and reporter genes was well explained only by a model that assumes a role for
Cbf1 in remodeling chromatin. Our analyses highlight the importance of taking into account
the activities of related transcription factors in explaining system-wide gene expression 
data.

PMID: 23558744

GEO ID: GSE26770.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE26770.final.pcl	Differential binding of the related transcription factors Pho4 and Cbf1 can tune  the sensitivity of promoters to different levels of an induction signal.	5	phosphorus utilization	1

File last modified: Thu Apr 25 2024

---


# Aristizabal_2013_PMID_24009531

Citation: Aristizabal MJ, Negri GL, Benschop JJ, Holstege FC, Krogan NJ, Kobor MS. 2013. High-
throughput genetic and gene expression analysis of the RNAPII-CTD reveals unexpected 
connections to SRB10/CDK8.

Full Description: The C-terminal domain (CTD) of RNA polymerase II (RNAPII) is composed of heptapeptide 
repeats, which play a key regulatory role in gene expression. Using  genetic interaction, 
chromatin immunoprecipitation followed by microarrays (ChIP-on-chip) and mRNA expression 
analysis, we found that truncating the CTD resulted in distinct changes to cellular 
function. Truncating the CTD altered RNAPII occupancy, leading to not only decreases, but 
also increases in mRNA levels. The latter were largely mediated by promoter elements and 
in part were linked to the transcription factor Rpn4. The mediator subunit Cdk8 was 
enriched at promoters of these genes, and its removal not only restored normal mRNA and 
RNAPII occupancy levels, but also reduced the abnormally high cellular amounts of Rpn4. 
This suggested a positive role of Cdk8 in relationship to RNAPII, which contrasted with 
the observed negative role at the activated INO1 gene. Here, loss of CDK8 suppressed the 
reduced mRNA expression and RNAPII occupancy levels of CTD truncation mutants.

PMID: 24009531

GEO ID: GSE43120.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE43120.remapped.final.pcl	High-throughput genetic and gene expression analysis of the RNAPII-CTD reveals unexpected connections to SRB10/CDK8.	24	transcription	2

File last modified: Thu Apr 25 2024

---


# Ask_2013_PMID_24341320

Citation: Ask M, Bettiga M, Duraiswamy VR, Olsson L. 2013. Pulsed addition of HMF and furfural to 
batch-grown xylose-utilizing Saccharomyces cerevisiae results in different physiological 
responses in glucose and xylose consumption phase.

Full Description: BACKGROUND: Pretreatment of lignocellulosic biomass generates a number of undesired 
degradation products that can inhibit microbial metabolism. Two of these compounds, the 
furan aldehydes 5-hydroxymethylfurfural (HMF) and 2-furaldehyde (furfural), have been 
shown to be an impediment for viable ethanol  production. In the present study, HMF and 
furfural were pulse-added during either the glucose or the xylose consumption phase in 
order to dissect the effects of these inhibitors on energy state, redox metabolism, and 
gene expression of xylose-consuming Saccharomyces cerevisiae. RESULTS: Pulsed addition of 
3.9 g L-1  HMF and 1.2 g L-1 furfural during either the glucose or the xylose consumption 
phase resulted in distinct physiological responses. Addition of furan aldehydes in the 
glucose consumption phase was followed by a decrease in the specific growth rate and the 
glycerol yield, whereas the acetate yield increased 7.3-fold, suggesting that NAD(P)H for 
furan aldehyde conversion was generated by acetate synthesis. No change in the 
intracellular levels of NAD(P)H was observed 1 hour after pulsing, whereas the 
intracellular concentration of ATP increased by 58%. An investigation of the response at 
transcriptional level revealed changes known  to be correlated with perturbations in the 
specific growth rate, such as protein  and nucleotide biosynthesis. Addition of furan 
aldehydes during the xylose consumption phase brought about an increase in the glycerol 
and acetate yields, whereas the xylitol yield was severely reduced. The intracellular 
concentrations  of NADH and NADPH decreased by 58 and 85%, respectively, hence suggesting 
that HMF and furfural drained the cells of reducing power. The intracellular concentration
of ATP was reduced by 42% 1 hour after pulsing of inhibitors, suggesting that energy-
requiring repair or maintenance processes were activated.  Transcriptome profiling showed 
that NADPH-requiring processes such as amino acid  biosynthesis and sulfate and nitrogen 
assimilation were induced 1 hour after pulsing. CONCLUSIONS: The redox and energy 
metabolism were found to be more severely affected after pulsing of furan aldehydes during
the xylose consumption  phase than during glucose consumption. Conceivably, this 
discrepancy resulted from the low xylose utilization rate, hence suggesting that xylose 
metabolism is  a feasible target for metabolic engineering of more robust xylose-utilizing
yeast strains.

PMID: 24341320

GEO ID: GSE50182.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE50182.final.pcl	Pulsed addition of HMF and furfural to batch-grown xylose-utilizing Saccharomyces cerevisiae results in different physiological responses in glucose and xylose consumption phase.	12	carbon utilization|chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Batista-Nascimento_2013_PMID_23295455

Citation: Batista-Nascimento L, Toledano MB, Thiele DJ, Rodrigues-Pousada C. 2013. Yeast protective 
response to arsenate involves the repression of the high affinity iron uptake system.

Full Description: Arsenic is a double-edge sword. On the one hand it is powerful carcinogen and on  the 
other it is used therapeutically to treat acute promyelocytic leukemia. Here  we report 
that arsenic activates the iron responsive transcription factor, Aft1,  as a consequence 
of a defective high-affinity iron uptake mediated by Fet3 and Ftr1, whose mRNAs are 
drastically decreased upon arsenic exposure. Moreover, arsenic causes the internalization 
and degradation of Fet3. Most importantly, fet3ftr1 mutant exhibits increased arsenic 
resistance and decreased arsenic accumulation over the wild-type suggesting that Fet3 
plays a role in arsenic toxicity. Finally we provide data suggesting that arsenic also 
disrupts iron uptake in mammals and the link between Fet3, arsenic and iron, can be 
relevant to clinical applications.

PMID: 23295455

GEO ID: GSE33427.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE33427.remapped.final.pcl	Yeast protective response to arsenate involves the repression of the high affinity iron uptake system.	15	chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Bessonov_2013_PMID_24130853

Citation: Bessonov K, Walkey CJ, Shelp BJ, van Vuuren HJ, Chiu D, van der Merwe G. 2013. Functional 
analyses of NSF1 in wine yeast using interconnected correlation clustering and molecular 
analyses.

Full Description: Analyzing time-course expression data captured in microarray datasets is a complex 
undertaking as the vast and complex data space is represented by a relatively low number 
of samples as compared to thousands of available genes. Here, we developed the 
Interdependent Correlation Clustering (ICC) method to analyze relationships that exist 
among genes conditioned on the expression of a specific target gene in microarray data. 
Based on Correlation Clustering, the ICC method analyzes a large set of correlation values
related to gene expression profiles extracted from given microarray datasets. ICC can be 
applied to any microarray dataset and any target gene. We applied this method to 
microarray data generated from wine fermentations and selected NSF1, which encodes a C2H2 
zinc finger-type transcription factor, as the target gene. The validity of the method  was
verified by accurate identifications of the previously known functional roles of NSF1. In 
addition, we identified and verified potential new functions for this gene; specifically, 
NSF1 is a negative regulator for the expression of sulfur metabolism genes, the nuclear 
localization of Nsf1 protein (Nsf1p) is controlled  in a sulfur-dependent manner, and the 
transcription of NSF1 is regulated by Met4p, an important transcriptional activator of 
sulfur metabolism genes. The inter-disciplinary approach adopted here highlighted the 
accuracy and relevancy of the ICC method in mining for novel gene functions using complex 
microarray datasets with a limited number of samples.

PMID: 24130853

GEO ID: GSE34117.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE34117.remapped.final.pcl	Functional analyses of NSF1 in wine yeast using interconnected correlation clustering and molecular analyses.	14	fermentation	1

File last modified: Thu Apr 25 2024

---


# Braberg_2013_PMID_23932120

Citation: Braberg H, Jin H, Moehle EA, Chan YA, Wang S, Shales M, Benschop JJ, Morris JH, Qiu C, Hu 
F, Tang LK, Fraser JS, Holstege FC, Hieter P, Guthrie C, Kaplan CD, Krogan NJ. 2013. From 
structure to systems: high-resolution, quantitative genetic analysis of RNA  polymerase II.

Full Description: RNA polymerase II (RNAPII) lies at the core of dynamic control of gene expression. Using 
53 RNAPII point mutants, we generated a point mutant epistatic  miniarray profile (pE-MAP
) comprising approximately 60,000 quantitative genetic interactions in Saccharomyces 
cerevisiae. This analysis enabled functional assignment of RNAPII subdomains and uncovered
connections between individual regions and other protein complexes. Using splicing 
microarrays and mutants that  alter elongation rates in vitro, we found an inverse 
relationship between RNAPII  speed and in vivo splicing efficiency. Furthermore, the pE-
MAP classified fast and slow mutants that favor upstream and downstream start site 
selection, respectively. The striking coordination of polymerization rate with 
transcription initiation and splicing suggests that transcription rate is tuned to 
regulate multiple gene expression steps. The pE-MAP approach provides a powerful strategy
to understand other multifunctional machines at amino acid resolution.

PMID: 23932120

GEO ID: GSE47429.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE47429.remapped.final.pcl	From structure to systems: high-resolution, quantitative genetic analysis of RNA  polymerase II.	57	transcription	2

File last modified: Thu Apr 25 2024

---


# Braun_2013_PMID_23207903

Citation: Braun KA, Parua PK, Dombek KM, Miner GE, Young ET. 2013. 14-3-3 (Bmh) proteins regulate 
combinatorial transcription following RNA polymerase II recruitment by binding at Adr1-
dependent promoters in Saccharomyces cerevisiae.

Full Description: Adr1 and Cat8 are nutrient-regulated transcription factors in Saccharomyces cerevisiae 
that coactivate genes necessary for growth in the absence of a fermentable carbon source. 
Transcriptional activation by Adr1 is dependent on the AMP-activated protein kinase Snf1 
and is inhibited by binding of Bmh, yeast 14-3-3 proteins, to the phosphorylated Adr1 
regulatory domain. We show here that  Bmh inhibits transcription by binding to Adr1 at 
promoters that contain a preinitiation complex, demonstrating that Bmh-mediated inhibition
is not due to nuclear exclusion, inhibition of DNA binding, or RNA polymerase II (Pol II) 
recruitment. Adr1-dependent mRNA levels under repressing growth conditions are 
synergistically enhanced in a mutant lacking Bmh and the two major histone deacetylases (
HDACs), suggesting that Bmh and HDACs inhibit gene expression independently. The synergism
requires Snf1 and Adr1 but not Cat8. Inactivating Bmh or preventing it from binding to 
Adr1 suppresses the normal requirement for Cat8 at codependent promoters, suggesting that 
Bmh modulates combinatorial control of gene expression in addition to having an inhibitory
role in transcription. Activating Snf1 by deleting Reg1, a Glc7 protein phosphatase 
regulatory subunit, is lethal in combination with defective Bmh in strain W303, suggesting
that Bmh and Snf1 have opposing roles in an essential cellular process.

PMID: 23207903

GEO ID: GSE40116.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE40116.remapped.final.pcl	14-3-3 (Bmh) proteins regulate combinatorial transcription following RNA polymerase II recruitment by binding at Adr1-dependent promoters in Saccharomyces cerevisiae.	12	carbon utilization|transcription|histone modification	1

File last modified: Thu Apr 25 2024
GSE40116_final.pcl	14-3-3 (Bmh) proteins regulate combinatorial transcription following RNA polymerase II recruitment by binding at Adr1-dependent promoters in Saccharomyces cerevisiae.	12	transcription|signaling	1

File last modified: Thu Apr 25 2024

---


# Brion_2013_PMID_24094006

Citation: Brion C, Ambroset C, Sanchez I, Legras JL, Blondin B. 2013. Differential adaptation to 
multi-stressed conditions of wine fermentation revealed by variations in yeast regulatory 
networks.

Full Description: BACKGROUND: Variation of gene expression can lead to phenotypic variation and have 
therefore been assumed to contribute the diversity of wine yeast (Saccharomyces cerevisiae
) properties. However, the molecular bases of this variation of gene expression are 
unknown. We addressed these questions by carrying out an integrated genetical-genomic 
study in fermentation conditions. We report here quantitative trait loci (QTL) mapping 
based on expression profiling in a segregating population generated by a cross between a 
derivative of the popular wine strain EC1118 and the laboratory strain S288c. RESULTS: 
Most of the  fermentation traits studied appeared to be under multi-allelic control. We 
mapped five phenotypic QTLs and 1465 expression QTLs. Several expression QTLs overlapped 
in hotspots. Among the linkages unraveled here, several were associated with metabolic 
processes essential for wine fermentation such as glucose sensing or nitrogen and vitamin 
metabolism. Variations affecting the regulation of drug detoxification and export (TPO1, 
PDR12 or QDR2) were linked to variation in four  genes encoding transcription factors (
PDR8, WAR1, YRR1 and HAP1). We demonstrated that the allelic variation of WAR1 and TPO1 
affected sorbic and octanoic acid resistance, respectively. Moreover, analysis of the 
transcription factors phylogeny suggests they evolved with a specific adaptation of the 
strains to wine fermentation conditions. Unexpectedly, we found that the variation of 
fermentation rates was associated with a partial disomy of chromosome 16. This disomy 
resulted from the well known 8-16 translocation. CONCLUSIONS: This large data set made it 
possible to decipher the effects of genetic variation on gene expression during 
fermentation and certain wine fermentation properties. Our findings shed a new light on 
the adaptation mechanisms required by yeast to cope  with the multiple stresses generated 
by wine fermentation. In this context, the detoxification and export systems appear to be 
of particular importance, probably due to nitrogen starvation. Furthermore, we show that 
the well characterized 8-16 translocation located in SSU1, which is associated with 
sulfite resistance, can lead to a partial chromosomic amplification in the progeny of 
strains that carry  it, greatly improving fermentation kinetics. This amplification has 
been detected among other wine yeasts.

PMID: 24094006

GEO ID: GSE41025.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE41025.remapped.final.pcl	Differential adaptation to multi-stressed conditions of wine fermentation revealed by variations in yeast regulatory networks.	56	fermentation|stationary phase maintenance|stress	1

File last modified: Thu Apr 25 2024

---


# Brown_2013_PMID_23360418

Citation: Brown NA, de Castro PA, de Castro Pimentel Figueiredo B, Savoldi M, Buckeridge MS, Lopes 
ML, de Lima Paullilo SC, Borges EP, Amorim HV, Goldman MH, Bonatto D, Malavazi I, Goldman 
GH. 2013. Transcriptional profiling of Brazilian Saccharomyces cerevisiae strains selected
for semi-continuous fermentation of sugarcane must.

Full Description: Brazil played a pioneering role in the global establishment of the sugarcane bioethanol 
industry. The bioethanol fermentation process currently used in Brazil is unique due to 
the acid wash and recycling of yeast cells. Two, industrially adopted, wild yeast strains
, CAT-1 and PE-2, have become the most widely used in  Brazil. How these strains respond 
to the unique fermentation process is poorly understood. The improved performance of CAT-1
and PE-2 is hypothesised to be related to enhanced stress tolerance. This study presents a
genome-wide analysis  of the CAT-1 and PE-2 transcriptomes during a small-scale 
fermentation process that mimicked the industrial conditions. The common and unique 
transcriptional responses of the two strains to the Brazilian fermentation process were 
identified. Environmental stress response genes were up-regulated postfermenter feeding, 
demonstrating the impact of the prior acid wash and high glucose environment. Cell wall 
and oxidative stress tolerance were subsequently demonstrated to be enhanced for the 
industrial strains. Conversely, numerous genes involved in protein synthesis were down-
regulated at the end of fermentation revealing the later impact of ethanol-induced stress
. Subsequently,  the industrial strains demonstrated a greater tolerance of ethanol and 
the disruption of endoplasmic reticulum homoeostasis. This increased ethanol tolerance was
finally correlated with an increased unfolded protein response and  increased HAC1 
splicing.

PMID: 23360418

GEO ID: GSE26618.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE26618.final.pcl	Saccharomyces cerevisiae bottom of the fermentor vs. fermentation	7	fermentation	1

File last modified: Thu Apr 25 2024
GSE26617.final.pcl	Saccharomyces cerevisiae bottom of the fermentor vs. feeding	7	fermentation	1

File last modified: Thu Apr 25 2024

---


# Buhligen_2013_PMID_24060829

Citation: Buhligen F, Rudinger P, Fetzer I, Stahl F, Scheper T, Harms H, Muller S. 2013. 
Sustainability of industrial yeast serial repitching practice studied by gene expression 
and correlation analysis.

Full Description: Bottom-fermenting Saccharomyces pastorianus strains driving brewing fermentation  
processes are usually reused several times. It is still unclear, whether the number of 
successions may have an impact on cell physiology prompting consequences for brewing 
quality. In this study, fermentation performance of up to twenty consecutive runs in a 
brewery was investigated. For each run mRNA expression levels of cellular marker molecules
, which are known to correlate with metabolism, hexose transport, aging processes, stress 
response mechanisms and flocculation capability was estimated to obtain information on 
changes in cell physiology over the successive runs. Low-density microarrays were used for
this purpose and the resulting gene expression profiles were finally correlated with 
changes in the abiotic micro-environments. A surprising stability of the marker molecule 
expression profiles within each specific serial repitching was stated. Loss of 
flocculation or an advanced aging could not be detected during serial repitching in the 
analyzed brewery. However, certain runs of the serial repitchings showed high variation in
stress response which was found to be caused by perturbations of the abiotic conditions. 
Regardless, the study showed that S.  pastorianus can be used repeatedly in serial 
repitching processes without loss of prominent physiological characteristics.

PMID: 24060829

GEO ID: GSE45950.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE45950.final.pcl	Sustainability of industrial yeast serial repitching practice studied by gene expression and correlation analysis.	8	fermentation|evolution	2

File last modified: Thu Apr 25 2024

---


# Byeon_2013_PMID_23779104

Citation: Byeon B, Wang W, Barski A, Ranallo RT, Bao K, Schones DE, Zhao K, Wu C, Wu WH. 2013. The 
ATP-dependent chromatin remodeling enzyme Fun30 represses transcription by sliding 
promoter-proximal nucleosomes.

Full Description: The evolutionarily conserved ATP-dependent chromatin remodeling enzyme Fun30 has  recently
been shown to play important roles in heterochromatin silencing and DNA  repair. However, 
how Fun30 remodels nucleosomes is not clear. Here we report a nucleosome sliding activity 
of Fun30 and its role in transcriptional repression.  We observed that Fun30 repressed the
expression of genes involved in amino acid and carbohydrate metabolism, the stress 
response, and meiosis. In addition, Fun30 was localized at the 5' and 3' ends of genes and
within the open reading frames of its targets. Consistent with its role in gene repression
, we observed that Fun30 target genes lacked histone modifications often associated with 
gene activation and showed an increased level of ubiquitinated histone H2B. Furthermore, a
genome-wide nucleosome mapping analysis revealed that the length of the nucleosome-free 
region at the 5' end of a subset of genes was changed in Fun30-depleted cells. In addition
, the positions of the -1, +2, and +3 nucleosomes at the 5' end of target genes were 
shifted significantly, whereas the position of the +1 nucleosome remained largely 
unchanged in the fun30Delta mutant. Finally, we demonstrated that affinity-purified, 
single-component Fun30 exhibited a nucleosome sliding activity in an ATP-dependent manner
. These results define a role for Fun30 in the regulation of transcription and indicate 
that Fun30 remodels chromatin at the 5' end of genes by sliding promoter-proximal 
nucleosomes.

PMID: 23779104

GEO ID: GSE48570.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE48570.final.pcl	The ATP-dependent chromatin remodeling enzyme Fun30 represses transcription by sliding promoter-proximal nucleosomes.	3	chromatin organization	2

File last modified: Thu Apr 25 2024

---


# Cai_2013_PMID_24386402

Citation: Cai Y, Futcher B. 2013. Effects of the yeast RNA-binding protein Whi3 on the half-life and
abundance of CLN3 mRNA and other targets.

Full Description: Whi3 is an RNA binding protein known to bind the mRNA of the yeast G1 cyclin gene CLN3. It
inhibits CLN3 function, but the mechanism of this inhibition is unclear; in previous 
studies, Whi3 made no observable difference to CLN3 mRNA levels, translation, or protein 
abundance. Here, we re-approach this issue using microarrays, RNA-Seq, ribosome profiling
, and other methods. By multiple methods, we find that the whi3 mutation causes a small 
but consistent increase in the abundance of hundreds of mRNAs, including the CLN3 mRNA. 
The effect on various mRNAs is roughly in proportion to the density of GCAU or UGCAU 
motifs carried by  these mRNAs, which may be a binding site for Whi3. mRNA instability of 
Whi3 targets may in part depend on a 3' AU rich element (ARE), AUUUUA. In addition, the 
whi3 mutation causes a small increase in the translational efficiency of CLN3 mRNA. The 
increase in CLN3 mRNA half-life and abundance together with the increase in translational 
efficiency is fully sufficient to explain the small-cell phenotype of whi3 mutants. Under 
stress conditions, Whi3 becomes a component of P-bodies or stress granules, but Whi3 also 
acts under non-stress condition, when no P-bodies are visible. We suggest that Whi3 may be
a very broadly-acting, but mild, modulator of mRNA stability. In CLN3, Whi3 may bind to  
the 3' GCAU motifs to attract the Ccr4-Not complex to promote RNA deadenylation and 
turnover, and Whi3 may bind to the 5' GCAU motifs to inhibit translation.

PMID: 24386402

GEO ID: GSE51784.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE51784.final.pcl	Effects of the yeast RNA-binding protein Whi3 on the half-life and abundance of CLN3 mRNA and other targets.	4	cell cycle regulation|chromatin organization	2

File last modified: Thu Apr 25 2024

---


# Chang_2013_PMID_23660188

Citation: Chang M, Kang HJ, Baek IJ, Kang CM, Park YS, Yun CW. 2013. Kdx1 regulates RCK1 gene 
expression by interacting with Rlm1 in Saccharomyces cerevisiae.

Full Description: Kdx1 is known as a stress-responsive protein. To better understand the function of Kdx1, 
we performed microarray analysis in KDX1 overexpressing cells and found  that the 
overexpression of KDX1 dramatically induced the expression of RCK1, a stress-responsive 
gene. This result was confirmed by northern blot analysis. Furthermore, the overexpression
of RCK1 partially rescued the growth defect caused by zymolyase stress. The expression of 
RCK1 was regulated independently by Slt2 and Hog1, and Kdx1 failed to induce the 
expression of RCK1 in a HOG1 deletion strain. The transcriptional factors Smp1, Sko1, Msn2
, Msn4, and Hot1, which are regulated by Hog1, did not affect RCK1 expression, but Rlm1 
did. Furthermore, the mutation of certain phosphorylation sites in RLM1 inhibited the  
induction of RCK1 expression by Kdx1. We found a conserved Rlm1 binding site in the 5' 
untranslated region (UTR) of RCK1, and the mutation of these Rlm1 binding  sites also 
inhibited the induction of RCK1 expression by Kdx1. Finally, we showed that Kdx1 
physically interacts with Rlm1 and that this interaction affects the ability of Rlm1 to 
bind to the RCK1 5' UTR. Taken together, these data suggest that Kdx1 interacts with Rlm1 
to activate RCK1 gene expression in response to stress in Saccharomyces cerevisiae.

PMID: 23660188

GEO ID: GSE45692.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE45692.remapped.final.pcl	Kdx1 regulates RCK1 gene expression by interacting with Rlm1 in Saccharomyces cerevisiae.	1	cell wall organization|stress|signaling	2

File last modified: Thu Apr 25 2024

---


# Chang_2013_PMID_24051094

Citation: Chang M, Kang HJ, Baek IJ, Kang CM, Park YS, Yun CW. 2013. Rck1 up-regulates Hog1 activity
by down-regulating Slt2 activity in Saccharomyces cerevisiae.

Full Description: We previously reported that the over-expression of KDX1 up-regulates RCK1 gene expression
. To further understand the function of Rck1, microarray analysis was performed using a 
RCK1 over-expressing strain. Based on microarray and Northern blot analyses, we determined
that the expression of KDX1 was down-regulated when  RCK1 was over-expressed. Furthermore
, we determined that phosphorylated forms of  Slt2 and Mkk2 were down-regulated by the 
over-expression of RCK1. Ptp2, a phosphatase that is regulated by the Slt2 MAP kinase 
pathway, was down-regulated  by the over-expression of RCK1. Ptp2 is a negative regulator 
of Hog1; thus, the phosphorylated form of Hog1 was up-regulated by RCK1 over-expression. A
point mutation of lysine 152 to arginine resulted in a failure to up-regulate Hog1 and  
the subsequent down-regulation of CTT1, which is a Hog1 pathway target gene. Furthermore, 
using microarray and Northern blot analyses, we determined that genes that are regulated 
by Msn2/Msn4 were up-regulated by Rck1 and that this was the result of Hog1 activation by 
RCK1 over-expression. Together, our results suggest that Rck1 inhibits Slt2 MAP kinase 
pathway activity and then Ptp2, which  subsequently activates Hog1.

PMID: 24051094

GEO ID: GSE49340.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE49340.remapped.final.pcl	Rck1 up-regulates Hog1 activity by down-regulating Slt2 activity in Saccharomyces cerevisiae.	1	cell wall organization|stress	2

File last modified: Thu Apr 25 2024

---


# Chen_2013_PMID_23124522

Citation: Chen K, Wilson MA, Hirsch C, Watson A, Liang S, Lu Y, Li W, Dent SY. 2013. Stabilization 
of the promoter nucleosomes in nucleosome-free regions by the yeast Cyc8-Tup1 corepressor.

Full Description: The yeast Cyc8 (also known as Ssn6)-Tup1 complex regulates gene expression through a 
variety of mechanisms, including positioning of nucleosomes over promoters of some target 
genes to limit accessibility to the transcription machinery. To further define the 
functions of Cyc8-Tup1 in gene regulation and chromatin remodeling, we performed genome-
wide profiling of changes in nucleosome organization and gene expression that occur upon 
loss of CYC8 or TUP1 and observed extensive nucleosome alterations in both promoters and 
gene bodies of derepressed genes. Our improved nucleosome profiling and analysis 
approaches revealed low-occupancy promoter nucleosomes (P nucleosomes) at locations 
previously defined as nucleosome-free regions. In the absence of CYC8 or TUP1, this P 
nucleosome is frequently lost, whereas nucleosomes are gained at -1 and +1 positions, 
accompanying up-regulation of downstream genes. Our analysis of public ChIP-seq data 
revealed that Cyc8 and Tup1 preferentially bind TATA-containing promoters, which are also 
enriched in genes derepressed upon loss of CYC8 or TUP1. These results suggest that 
stabilization of the P nucleosome on TATA-containing promoters may be a central feature of
the repressive chromatin architecture created by the Cyc8-Tup1 corepressor, and that 
releasing the P nucleosome contributes to gene activation.

PMID: 23124522

GEO ID: GSE37466.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE37466.final.pcl	Stabilization of the promoter nucleosomes in nucleosome-free regions by the yeast Cyc8-Tup1 corepressor.	7	chromatin organization	1

File last modified: Thu Apr 25 2024

---


# Chu_2013_PMID_23979602

Citation: Chu HY, Hopper AK. 2013. Genome-wide investigation of the role of the tRNA nuclear-
cytoplasmic trafficking pathway in regulation of the yeast Saccharomyces cerevisiae 
transcriptome and proteome.

Full Description: In eukaryotic cells, tRNAs are transcribed and partially processed in the nucleus before 
they are exported to the cytoplasm, where they have an essential role in protein synthesis
. Surprisingly, mature cytoplasmic tRNAs shuttle between nucleus and cytoplasm, and tRNA 
subcellular distribution is nutrient dependent. At least  three members of the beta-
importin family, Los1, Mtr10, and Msn5, function in tRNA nuclear-cytoplasmic intracellular
movement. To test the hypothesis that the  tRNA retrograde pathway regulates the 
translation of particular transcripts, we compared the expression profiles from 
nontranslating mRNAs and polyribosome-associated translating mRNAs collected from 
msn5Delta, mtr10Delta, and wild-type cells under fed or acute amino acid depletion 
conditions. Our microarray data revealed that the methionine, arginine, and leucine 
biosynthesis  pathways are targets of the tRNA retrograde process. We confirmed the 
microarray  data by Northern and Western blot analyses. The levels of some of the 
particular  target mRNAs were reduced, while others appeared not to be affected. However, 
the protein levels of all tested targets in these pathways were greatly decreased when 
tRNA nuclear import or reexport to the cytoplasm was disrupted. This study provides 
information that tRNA nuclear-cytoplasmic dynamics is connected to the biogenesis of 
proteins involved in amino acid biosynthesis.

PMID: 23979602

GEO ID: GSE47713.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE47713.final.pcl	Genome-wide investigation of the role of the tRNA nuclear-cytoplasmic trafficking pathway in regulation of the yeast Saccharomyces cerevisiae transcriptome and proteome.	36	translational regulation|amino acid utilization|starvation	1

File last modified: Thu Apr 25 2024

---


# Chumnanpuen_2013_PMID_24456840

Citation: Chumnanpuen P, Nookaew I, Nielsen J. 2013. Integrated analysis, transcriptome-lipidome, 
reveals the effects of INO-level (INO2 and INO4) on lipid metabolism in yeast.

Full Description: BACKGROUND: In the yeast Saccharomyces cerevisiae, genes containing UASINO sequences are 
regulated by the Ino2/Ino4 and Opi1 transcription factors, and this regulation controls 
lipid biosynthesis. The expression level of INO2 and INO4 genes (INO-level) at different 
nutrient limited conditions might lead to various  responses in yeast lipid metabolism. 
METHODS: In this study, we undertook a global study on how INO-levels (transcription level
of INO2 and INO4) affect lipid metabolism in yeast and we also studied the effects of 
single and double deletions of the two INO-genes (deficient effect). Using 2 types of 
nutrient limitations (carbon and nitrogen) in chemostat cultures operated at a fixed 
specific growth rate of 0.1 h-1 and strains having different INO-level, we were able to 
see the effect on expression level of the genes involved in lipid biosynthesis and the 
fluxes towards the different lipid components. Through combined measurements of the 
transcriptome, metabolome, and lipidome it was possible to obtain a large dataset that 
could be used to identify how the INO-level controls lipid metabolism and also establish 
correlations between the different components. RESULTS: In this study, we undertook a 
global study on how  INO-levels (transcription level of INO2 and INO4) affect lipid 
metabolism in yeast and we also studied the effects of single and double deletions of the 
two INO-genes (deficient effect). Using 2 types of nutrient limitations (carbon and 
nitrogen) in chemostat cultures operated at a fixed specific growth rate of 0.1 h-1 and 
strains having different INO-level, we were able to see the effect on expression level of 
the genes involved in lipid biosynthesis and the fluxes towards the different lipid 
components. Through combined measurements of the transcriptome, metabolome, and lipidome 
it was possible to obtain a large dataset that could be used to identify how the INO-level
controls lipid metabolism and also establish correlations between the different components
. CONCLUSIONS: Our analysis showed the strength of using a combination of transcriptome 
and lipidome analysis to illustrate the effect of INO-levels on phospholipid metabolism 
and based on our analysis we established a global regulatory map.

PMID: 24456840

GEO ID: GSE36298.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE36298.final.pcl	Integrated analysis, transcriptome-lipidome, reveals the effects of INO-level (INO2 and INO4) on lipid metabolism in yeast.	33	lipid metabolism|nutrient utilization	1

File last modified: Thu Apr 25 2024

---


# Cook_2013_PMID_23826764

Citation: Cook R, Hennell JR, Lee S, Khoo CS, Carles MC, Higgins VJ, Govindaraghavan S, Sucher NJ. 
2013. The Saccharomyces cerevisiae transcriptome as a mirror of phytochemical variation in
complex extracts of Equisetum arvense from America, China, Europe and India.

Full Description: BACKGROUND: Pattern-oriented chemical profiling is increasingly being used to characterize
the phytochemical composition of herbal medicines for quality control purposes. Ideally, a
fingerprint of the biological effects should complement the chemical fingerprint. For 
ethical and practical reasons it is not  possible to test each herbal extract in 
laboratory animals or humans. What is needed is a test system consisting of an organism 
with relevant biology and complexity that can serve as a surrogate in vitro system. The 
purpose of this study was to test the hypothesis that the Saccharomyces cerevisiae 
transcriptome  might be used as an indicator of phytochemical variation of closely-related
yet distinctly different extracts prepared from a single species of a phytogeographically 
widely distributed medicinal plant. We combined phytochemical profiling using 
chromatographic methods (HPTLC, HPLC-PDA-MS/MS) and gene expression studies using 
Affymetrix Yeast 2.0 gene chip with principal component  analysis and k-nearest neighbor 
clustering analysis to test this hypothesis using extracts prepared from the 
phytogeographically widely distributed medicinal plant Equisetum arvense as a test case. 
RESULTS: We found that the Equisetum arvense extracts exhibited qualitative and 
quantitative differences in their phytochemical composition grouped along their 
phytogeographical origin. Exposure  of yeast to the extracts led to changes in gene 
expression that reflected both the similarities and differences in the phytochemical 
composition of the extracts. The Equisetum arvense extracts elicited changes in the 
expression of genes involved in mRNA translation, drug transport, metabolism of energy 
reserves, phospholipid metabolism, and the cellular stress response. CONCLUSIONS: Our data
show that functional genomics in S. cerevisiae may be developed as a sensitive bioassay 
for the scientific investigation of the interplay between phytochemical composition and 
transcriptional effects of complex mixtures of chemical compounds. S. cerevisiae 
transcriptomics may also be developed for testing of mixtures of conventional drugs (
polypills) to discover novel antagonistic or synergistic effects of those drug 
combinations.

PMID: 23826764

GEO ID: GSE24888.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE24888.final.pcl	The Saccharomyces cerevisiae transcriptome as a mirror of phytochemical variation in complex extracts of Equisetum arvense from America, China, Europe and India.	18	chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Garcia-Oliver_2013_PMID_23599000

Citation: Garcia-Oliver E, Pascual-Garcia P, Garcia-Molinero V, Lenstra TL, Holstege FC, Rodriguez-
Navarro S. 2013. A novel role for Sem1 and TREX-2 in transcription involves their impact 
on recruitment and H2B deubiquitylation activity of SAGA.

Full Description: Transcription and mRNA export are linked processes. However, the molecular mechanisms of 
this coordination are not clear. Sus1 (hENY2) participates in this  coordination as part 
of two protein complexes: SAGA, a transcriptional co-activator; TREX-2, which functions in
mRNA biogenesis and export. Here, we investigate the coordinated action of SAGA and TREX-2
required for gene expression. We demonstrate that TREX-2 subunit Sem1 also participates in
transcription activation. Like Sus1, Sem1 is required for the induction of ARG1 and GAL1, 
these being SAGA-regulated genes. Chromatin immunoprecipitations show that proper 
recruitment of certain SAGA subunits to the GAL1 promoter depends on  Sem1. Notably, both 
in vivo and in vitro analyses reveal that Sem1 influences SAGA-dependent histone H2B 
deubiquitylation. Most of these phenotypes are also found to depend on another TREX-2 
subunit, Thp1. These results unveil a new role  for Sem1 in the activation of the SAGA-
dependent gene GAL1 and influencing H2B deubiquitylation. Our work provides insights into 
a novel functional relationship between Sem1 and the SAGA complex.

PMID: 23599000

GEO ID: GSE39861.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE39861.remapped.final.pcl	A novel role for Sem1 and TREX-2 in transcription involves their impact on recruitment and H2B deubiquitylation activity of SAGA.	32	histone modification|transcription	2

File last modified: Thu Apr 25 2024

---


# Geijer_2013_PMID_23758973

Citation: Geijer C, Medrala-Klein D, Petelenz-Kurdziel E, Ericsson A, Smedh M, Andersson M, Goksor M,
Nadal-Ribelles M, Posas F, Krantz M, Nordlander B, Hohmann S. 2013. Initiation of the 
transcriptional response to hyperosmotic shock correlates with  the potential for volume 
recovery.

Full Description: The control of activity and localization of transcription factors is critical for 
appropriate transcriptional responses. In eukaryotes, signal transduction components such 
as mitogen-activated protein kinase (MAPK) shuttle into the nucleus to activate 
transcription. It is not known in detail how different amounts of nuclear MAPK over time 
affect the transcriptional response. In the present study, we aimed to address this issue 
by studying the high osmolarity glycerol (HOG) system in Saccharomyces cerevisiae. We 
employed a conditional osmotic system, which changes the period of the MAPK Hog1 signal 
independent of the initial stress level. We determined the dynamics of the Hog1 nuclear 
localization and cell volume by single-cell analysis in well-controlled microfluidics 
systems and compared the responses with the global transcriptional  output of cell 
populations. We discovered that the onset of the initial transcriptional response 
correlates with the potential of cells for rapid adaptation; cells that are capable of 
recovering quickly initiate the transcriptional responses immediately, whereas cells that 
require longer time to  adapt also respond later. This is reflected by Hog1 nuclear 
localization, Hog1 promoter association and the transcriptional response, but not Hog1 
phosphorylation, suggesting that a presently uncharacterized rapid adaptive mechanism 
precedes the Hog1 nuclear response. Furthermore, we found that the period of Hog1 nuclear 
residence affects the amplitude of the transcriptional response rather than the spectrum 
of responsive genes.

PMID: 23758973

GEO ID: GSE29716.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE29716.final.pcl	Initiation of the transcriptional response to hyperosmotic shock correlates with  the potential for volume recovery.	16	osmotic stress|signaling	2

File last modified: Thu Apr 25 2024

---


# Gitter_2013_PMID_23064748

Citation: Gitter A, Carmi M, Barkai N, Bar-Joseph Z. 2013. Linking the signaling cascades and 
dynamic regulatory networks controlling stress responses.

Full Description: Accurate models of the cross-talk between signaling pathways and transcriptional  
regulatory networks within cells are essential to understand complex response programs. We
present a new computational method that combines condition-specific  time-series 
expression data with general protein interaction data to reconstruct  dynamic and causal 
stress response networks. These networks characterize the pathways involved in the 
response, their time of activation, and the affected genes. The signaling and regulatory 
components of our networks are linked via a set of common transcription factors that serve
as targets in the signaling network and as regulators of the transcriptional response 
network. Detailed case  studies of stress responses in budding yeast demonstrate the 
predictive power of  our method. Our method correctly identifies the core signaling 
proteins and transcription factors of the response programs. It further predicts the 
involvement of additional transcription factors and other proteins not previously 
implicated in the response pathways. We experimentally verify several of these predictions
for the osmotic stress response network. Our approach requires little condition-specific 
data: only a partial set of upstream initiators and time-series gene expression data, 
which are readily available for many conditions and species. Consequently, our method is 
widely applicable and can be used to derive accurate, dynamic response models in several 
species.

PMID: 23064748

GEO ID: GSE28213GPL13340.sfp.pcl

PCL filename	short description	# conditions	tags	# channels
GSE28213GPL13340.sfp.pcl	Linking the signaling cascades and dynamic regulatory networks controlling stress responses.	16	osmotic stress	1

File last modified: Thu Apr 25 2024

---


# Gonzalez_2013_PMID_23704987

Citation: Gonzalez A, Casado C, Arino J, Casamayor A. 2013. Ptc6 is required for proper rapamycin-
induced down-regulation of the genes coding for ribosomal and rRNA processing proteins in 
S. cerevisiae.

Full Description: Ptc6 is one of the seven components (Ptc1-Ptc7) of the protein phosphatase 2C family in 
the yeast Saccharomyces cerevisiae. In contrast to other type 2C phosphatases, the 
cellular role of this isoform is poorly understood. We present  here a comprehensive 
characterization of this gene product. Cells lacking Ptc6 are sensitive to zinc ions, and 
somewhat tolerant to cell-wall damaging agents and to Li(plus). Ptc6 mutants are sensitive
to rapamycin, albeit to lesser extent than ptc1 cells. This phenotype is not rescued by 
overexpression of PTC1 and mutation of ptc6 does not reproduce the characteristic genetic 
interactions of the ptc1 mutation with components of the TOR pathway, thus suggesting 
different cellular roles for both isoforms. We show here that the rapamycin-sensitive 
phenotype of ptc6 cells is unrelated to the reported role of Pt6 in controlling pyruvate 
dehydrogenase activity. Lack of Ptc6 results in substantial attenuation  of the 
transcriptional response to rapamycin, particularly in the subset of repressed genes 
encoding ribosomal proteins or involved in rRNA processing. In contrast, repressed genes 
involved in translation are Ptc6-independent. These effects cannot be attributed to the 
regulation of the Sch9 kinase, but they could involve modulation of the binding of the 
Ifh1 co-activator to specific gene promoters.

PMID: 23704987

GEO ID: GSE38260.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE38260.remapped.final.pcl	Ptc6 is required for proper rapamycin-induced down-regulation of the genes coding for ribosomal and rRNA processing proteins in S. cerevisiae.	14	chemical stimulus|protein phosphorylation|signaling	2

File last modified: Thu Apr 25 2024

---


# Hong_2013_PMID_23376593

Citation: Hong KK, Nielsen J. 2013. Adaptively evolved yeast mutants on galactose show trade-offs in
carbon utilization on glucose.

Full Description: Adaptive evolution offers many opportunities in metabolic engineering; however, several 
constraints still exist as evolutionary trade-offs may impose collateral  cost to obtain 
new traits. The application of adaptive evolution for strains development could be further
improved by elucidating the molecular mechanisms. In this study, adaptively evolved yeast 
mutants with improved galactose utilization  ability showed impaired glucose utilization. 
The molecular genetic basis of this  trade-off was investigated using a systems biology 
approach. Transcriptional and  metabolic changes resulting from the improvement of 
galactose utilization were found maintained during growth on glucose. Moreover, glucose 
repression related genes showed conserved expression patterns during growth on both sugars
. Mutations in the RAS2 gene that were identified as beneficial for galactose utilization 
in evolved mutants exhibited significant correlation with attenuation of glucose 
utilization. These results indicate that antagonistic pleiotropy is the dominant mechanism
in the observed trade-off, and it is likely realized by changes in glucose signaling.

PMID: 23376593

GEO ID: GSE39065.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE39065.final.pcl	Adaptively evolved yeast mutants on galactose show trade-offs in carbon utilization on glucose.	6	carbon utilization|evolution	1

File last modified: Thu Apr 25 2024

---


# Jaehnig_2013_PMID_23810556

Citation: Jaehnig EJ, Kuo D, Hombauer H, Ideker TG, Kolodner RD. 2013. Checkpoint kinases regulate a
global network of transcription factors in response to DNA damage.

Full Description: DNA damage activates checkpoint kinases that induce several downstream events, including 
widespread changes in transcription. However, the specific connections  between the 
checkpoint kinases and downstream transcription factors (TFs) are not well understood. 
Here, we integrate kinase mutant expression profiles, transcriptional regulatory 
interactions, and phosphoproteomics to map kinases and downstream TFs to transcriptional 
regulatory networks. Specifically, we investigate the role of the Saccharomyces cerevisiae
checkpoint kinases (Mec1, Tel1, Chk1, Rad53, and Dun1) in the transcriptional response to 
DNA damage caused by methyl methanesulfonate. The result is a global kinase-TF regulatory 
network in which Mec1 and Tel1 signal through Rad53 to synergistically regulate the 
expression of more than 600 genes. This network involves at least nine TFs, many  of which
have Rad53-dependent phosphorylation sites, as regulators of checkpoint-kinase-dependent 
genes. We also identify a major DNA damage-induced transcriptional network that regulates 
stress response genes independently of the checkpoint kinases.

PMID: 23810556

GEO ID: GSE40351.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE40351.remapped.final.pcl	Checkpoint kinases regulate a global network of transcription factors in response to DNA damage.	72	DNA damage stimulus|protein phosphorylation|transcription	2

File last modified: Thu Apr 25 2024

---


# Judeh_2013_PMID_23268448

Citation: Judeh T, Johnson C, Kumar A, Zhu D. 2013. TEAK: topology enrichment analysis framework for
detecting activated biological subpathways.

Full Description: To mine gene expression data sets effectively, analysis frameworks need to incorporate 
methods that identify intergenic relationships within enriched biologically relevant 
subpathways. For this purpose, we developed the Topology Enrichment Analysis frameworK (
TEAK). TEAK employs a novel in-house algorithm and a tailor-made Clique Percolation Method
to extract linear and nonlinear KEGG subpathways, respectively. TEAK scores subpathways 
using the Bayesian Information Criterion for context specific data and the Kullback-
Leibler divergence for case-control data. In this article, we utilized TEAK with 
experimental studies to analyze microarray data sets profiling stress responses in the 
model eukaryote Saccharomyces cerevisiae. Using a public microarray data set, we 
identified via TEAK linear sphingolipid metabolic subpathways activated during the yeast 
response to nitrogen stress, and phenotypic analyses of the corresponding deletion strains
indicated previously unreported fitness defects for the dpl1Delta and lag1Delta mutants 
under conditions of nitrogen limitation. In addition, we studied the yeast filamentous 
response to nitrogen stress by profiling changes in transcript levels upon deletion of two
key filamentous growth transcription factors, FLO8 and MSS11. Via TEAK we identified a 
nonlinear  glycerophospholipid metabolism subpathway involving the SLC1 gene, which we 
found via mutational analysis to be required for yeast filamentous growth.

PMID: 23268448

GEO ID: GSE40530.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE40530.final.pcl	TEAK: topology enrichment analysis framework for detecting activated biological subpathways.	6	filamentous growth|nitrogen utilization	1

File last modified: Thu Apr 25 2024

---


# Kucukgoze_2013_PMID_23992612

Citation: Kucukgoze G, Alkim C, Yilmaz U, Kisakesen HI, Gunduz S, Akman S, Cakar ZP. 2013. 
Evolutionary engineering and transcriptomic analysis of nickel-resistant Saccharomyces 
cerevisiae.

Full Description: Increased exposure to nickel compounds and alloys due to industrial development has 
resulted in nickel pollution and many pathological effects on human health. However, there
is very limited information about nickel response, transport, and  tolerance in eukaryotes
. To investigate nickel resistance in the model eukaryote  Saccharomyces cerevisiae, 
evolutionary engineering by batch selection under gradually increasing nickel stress 
levels was performed. Nickel hyper-resistant mutants that could resist up to 5.3 mM NiCl2
, a lethal level for the reference strain, were selected. The mutants were also cross-
resistant against iron, cobalt, zinc, and manganese stresses and accumulated more than 
twofold higher nickel than the reference strain. Global transcriptomic analysis revealed 
that 640 upregulated genes were related to iron homeostasis, stress response, and 
oxidative damage, implying that nickel resistance may share common mechanisms with iron 
and cobalt resistance, general stress response, and oxidative damage.

PMID: 23992612

GEO ID: GSE50985.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE50985.final.pcl	Evolutionary engineering and transcriptomic analysis of nickel-resistant Saccharomyces cerevisiae.	6	evolution|metal or metalloid ion stress	2

File last modified: Thu Apr 25 2024

---


# Lai_2013_PMID_23893744

Citation: Lai X, Beilharz T, Au WC, Hammet A, Preiss T, Basrai MA, Heierhorst J. 2013. Yeast hEST1A/
B (SMG5/6)-like proteins contribute to environment-sensing adaptive  gene expression 
responses.

Full Description: During its natural life cycle, budding yeast (Saccharomyces cerevisiae) has to adapt to 
drastically changing environments, but how environmental-sensing pathways are linked to 
adaptive gene expression changes remains incompletely understood. Here, we describe two 
closely related yeast hEST1A-B (SMG5-6)-like proteins termed Esl1 and Esl2 that contain a 
14-3-3-like domain and a putative PilT N-terminus ribonuclease domain. We found that, 
unlike their metazoan orthologs, Esl1 and Esl2 were not involved in nonsense-mediated mRNA
decay or telomere maintenance pathways. However, in genome-wide expression array analyses
, absence of Esl1 and Esl2 led to more than two-fold deregulation of approximately  50 
transcripts, most of which were expressed inversely to the appropriate metabolic response 
to environmental nutrient supply; for instance, normally glucose-repressed genes were 
derepressed in esl1Delta esl2Delta double mutants during growth in a high-glucose 
environment. Likewise, in a genome-wide synthetic gene array screen, esl1Delta esl2Delta 
double mutants were synthetic sick with null mutations for Rim8 and Dfg16, which form the 
environmental-sensing complex of the Rim101 pH response gene expression pathway. Overall, 
these results suggest that Esl1 and Esl2 contribute to the regulation of adaptive gene 
expression responses of environmental sensing pathways.

PMID: 23893744

GEO ID: GSE48956.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE48956.remapped.final.pcl	Yeast hEST1A/B (SMG5/6)-like proteins contribute to environment-sensing adaptive  gene expression responses.	4	environmental-sensing|metabolism	2

File last modified: Thu Apr 25 2024

---


# Larsson_2013_PMID_23991176

Citation: Larsson M, Uvell H, Sandstrom J, Ryden P, Selth LA, Bjorklund S. 2013. Functional studies 
of the yeast med5, med15 and med16 mediator tail subunits.

Full Description: The yeast Mediator complex can be divided into three modules, designated Head, Middle and 
Tail. Tail comprises the Med2, Med3, Med5, Med15 and Med16 protein subunits, which are all
encoded by genes that are individually non-essential for  viability. In cells lacking 
Med16, Tail is displaced from Head and Middle. However, inactivation of MED5/MED15 and 
MED15/MED16 are synthetically lethal, indicating that Tail performs essential functions as
a separate complex even when it is not bound to Middle and Head. We have used the N-Degron
method to create temperature-sensitive (ts) mutants in the Mediator tail subunits Med5, 
Med15 and  Med16 to study the immediate effects on global gene expression when each 
subunit  is individually inactivated, and when Med5/15 or Med15/16 are inactivated 
together. We identify 25 genes in each double mutant that show a significant change in 
expression when compared to the corresponding single mutants and to the wild type strain. 
Importantly, 13 of the 25 identified genes are common for both  double mutants. We also 
find that all strains in which MED15 is inactivated show  down-regulation of genes that 
have been identified as targets for the Ace2 transcriptional activator protein, which is 
important for progression through the G1 phase of the cell cycle. Supporting this 
observation, we demonstrate that loss of Med15 leads to a G1 arrest phenotype. 
Collectively, these findings provide insight into the function of the Mediator Tail module.

PMID: 23991176

GEO ID: GSE47712.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE47712.remapped.final.pcl	Functional studies of the yeast med5, med15 and med16 mediator tail subunits.	12	transcription	1

File last modified: Thu Apr 25 2024

---


# Lenstra_2013_PMID_24324601

Citation: Lenstra TL, Tudek A, Clauder S, Xu Z, Pachis ST, van Leenen D, Kemmeren P, Steinmetz LM, 
Libri D, Holstege FC. 2013. The role of Ctk1 kinase in termination of small non-coding 
RNAs.

Full Description: Transcription termination in Saccharomyces cerevisiae can be performed by at least two 
distinct pathways and is influenced by the phosphorylation status of the carboxy-terminal 
domain (CTD) of RNA polymerase II (Pol II). Late termination of mRNAs is performed by the 
CPF/CF complex, the recruitment of which is dependent on CTD-Ser2 phosphorylation (Ser2P
). Early termination of shorter cryptic unstable transcripts (CUTs) and small nucleolar/
nuclear RNAs (sno/snRNAs) is performed by the Nrd1-Nab3-Sen1 (NNS) complex that binds 
phosphorylated CTD-Ser5 (Ser5P) via the CTD-interacting domain (CID) of Nrd1p. In this 
study, mutants of the different termination pathways were compared by genome-wide 
expression analysis. Surprisingly, the expression changes observed upon loss of the CTD-
Ser2 kinase Ctk1p are more similar to those derived from alterations in the Ser5P-
dependent NNS pathway, than from loss of CTD-Ser2P binding factors. Tiling array analysis 
of ctk1Delta cells reveals readthrough at snoRNAs, at many  cryptic unstable transcripts (
CUTs) and stable uncharacterized transcripts (SUTs), but only at some mRNAs. Despite the 
suggested predominant role in termination of mRNAs, we observed that a CTK1 deletion or a 
Pol II CTD mutant lacking all Ser2 positions does not result in a global mRNA termination 
defect. Rather, termination defects in these strains are widely observed at NNS-dependent 
genes. These results indicate that Ctk1p and Ser2 CTD phosphorylation have a wide impact 
in termination of small non-coding RNAs but only affect a subset of mRNA coding genes.

PMID: 24324601

GEO ID: GSE40254.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE40254.remapped.final.pcl	The role of Ctk1 kinase in termination of small non-coding RNAs.	22	mRNA processing|transcription	2

File last modified: Thu Apr 25 2024

---


# Liang_2013_PMID_23985319

Citation: Liang CY, Wang LC, Lo WS. 2013. Dissociation of the H3K36 demethylase Rph1 from chromatin 
mediates derepression of environmental stress-response genes under genotoxic stress in 
Saccharomyces cerevisiae.

Full Description: Cells respond to environmental signals by altering gene expression through transcription 
factors. Rph1 is a histone demethylase containing a Jumonji C (JmjC) domain and belongs to
the C(2)H(2) zinc-finger protein family. Here we investigate the regulatory network of 
Rph1 in yeast by expression microarray analysis. More than 75% of Rph1-regulated genes 
showed increased expression in the rph1-deletion mutant, suggesting that Rph1 is mainly a 
transcriptional repressor. The binding motif 5'-CCCCTWA-3', which resembles the stress 
response element, is overrepresented in the promoters of Rph1-repressed genes. A 
significant proportion of Rph1-regulated genes respond to DNA damage and environmental 
stress. Rph1 is a labile protein, and Rad53 negatively modulates Rph1 protein level. We 
find that the JmjN domain is important in maintaining protein stability and the repressive
effect of Rph1. Rph1 is directly associated  with the promoter region of targeted genes 
and dissociated from chromatin before  transcriptional derepression on DNA damage and 
oxidative stress. Of interest, the master stress-activated regulator Msn2 also regulates a
subset of Rph1-repressed  genes under oxidative stress. Our findings confirm the 
regulatory role of Rph1 as a transcriptional repressor and reveal that Rph1 might be a 
regulatory node connecting different signaling pathways responding to environmental 
stresses.

PMID: 23985319

GEO ID: GSE39950.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE39950.remapped.final.pcl	Dissociation of the H3K36 demethylase Rph1 from chromatin mediates derepression of environmental stress-response genes under genotoxic stress in Saccharomyces cerevisiae.	6	transcription|histone modification	1

File last modified: Thu Apr 25 2024

---


# Ling_2013_PMID_23826995

Citation: Ling H, Chen B, Kang A, Lee JM, Chang MW. 2013. Transcriptome response to alkane biofuels 
in Saccharomyces cerevisiae: identification of efflux pumps involved in alkane tolerance.

Full Description: BACKGROUND: Hydrocarbon alkanes have been recently considered as important next-generation
biofuels because microbial production of alkane biofuels was demonstrated. However, the 
toxicity of alkanes to microbial hosts can possibly be a bottleneck for high productivity 
of alkane biofuels. To tackle this toxicity issue, it is essential to understand molecular
mechanisms of interactions between alkanes and microbial hosts, and to harness these 
mechanisms to develop microbial host strains with improved tolerance against alkanes. In 
this study, we aimed to  improve the tolerance of Saccharomyces cerevisiae, a model 
eukaryotic host of industrial significance, to alkane biofuels by exploiting cellular 
mechanisms underlying alkane response. RESULTS: To this end, we first confirmed that 
nonane  (C9), decane (C10), and undecane (C11) were significantly toxic and accumulated in
S. cerevisiae. Transcriptome analyses suggested that C9 and C10 induced a range of 
cellular mechanisms such as efflux pumps, membrane modification, radical detoxification, 
and energy supply. Since efflux pumps could possibly aid in alkane secretion, thereby 
reducing the cytotoxicity, we formed the hypothesis that those induced efflux pumps could 
contribute to alkane export and tolerance.  In support of this hypothesis, we demonstrated
the roles of the efflux pumps Snq2p and Pdr5p in reducing intracellular levels of C10 and 
C11, as well as enhancing tolerance levels against C10 and C11. This result provided the 
evidence that Snq2p and Pdr5p were associated with alkane export and tolerance in S. 
cerevisiae. CONCLUSIONS: Here, we investigated the cellular mechanisms of S. cerevisiae 
response to alkane biofuels at a systems level through transcriptome analyses. Based on 
these mechanisms, we identified efflux pumps involved in alkane export and tolerance in S
. cerevisiae. We believe that the results here provide valuable insights into designing 
microbial engineering strategies to improve cellular tolerance for highly efficient alkane
biofuel production.

PMID: 23826995

GEO ID: GSE38653.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE38653.final.pcl	Transcriptome response to alkane biofuels in Saccharomyces cerevisiae: identification of efflux pumps involved in alkane tolerance.	8	chemical stimulus|stress	2

File last modified: Thu Apr 25 2024

---


# Liu_2013_PMID_23392765

Citation: Liu Z, Hou J, Martinez JL, Petranovic D, Nielsen J. 2013. Correlation of cell growth and 
heterologous protein production by Saccharomyces cerevisiae.

Full Description: With the increasing demand for biopharmaceutical proteins and industrial enzymes, it is 
necessary to optimize the production by microbial fermentation or cell cultures. Yeasts 
are well established for the production of a wide range of recombinant proteins, but there
are also some limitations; e.g., metabolic and cellular stresses have a strong impact on 
recombinant protein production. In this work, we investigated the effect of the specific 
growth rate on the production of two different recombinant proteins. Our results show that
human insulin precursor is produced in a growth-associated manner, whereas alpha-amylase 
tends to have a  higher yield on substrate at low specific growth rates. Based on 
transcriptional  analysis, we found that the difference in the production of the two 
proteins as function of the specific growth rate is mainly due to differences in 
endoplasmic  reticulum processing, protein turnover, cell cycle, and global stress 
response. We also found that there is a shift at a specific growth rate of 0.1 h(-1) that 
influences protein production. Thus, for lower specific growth rates, the alpha-amylase 
and insulin precursor-producing strains present similar cell responses and phenotypes, 
whereas for higher specific growth rates, the two strains respond differently to changes 
in the specific growth rate.

PMID: 23392765

GEO ID: GSE40934.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE40934.final.pcl	Correlation of cell growth and heterologous protein production by Saccharomyces cerevisiae.	18	diauxic shift|stress|metabolism	1

File last modified: Thu Apr 25 2024

---


# Liu_2013_PMID_23435897

Citation: Liu Z, Osterlund T, Hou J, Petranovic D, Nielsen J. 2013. Anaerobic alpha-amylase 
production and secretion with fumarate as the final electron acceptor in Saccharomyces 
cerevisiae.

Full Description: In this study, we focus on production of heterologous alpha-amylase in the yeast  
Saccharomyces cerevisiae under anaerobic conditions. We compare the metabolic fluxes and 
transcriptional regulation under aerobic and anaerobic conditions, with the objective of 
identifying the final electron acceptor for protein folding under anaerobic conditions. We
find that yeast produces more amylase under anaerobic conditions than under aerobic 
conditions, and we propose a model for electron transfer under anaerobic conditions. 
According to our model, during protein folding the electrons from the endoplasmic 
reticulum are transferred to fumarate as the final electron acceptor. This model is 
supported by findings that the addition of fumarate under anaerobic (but not aerobic) 
conditions improves cell growth, specifically in the alpha-amylase-producing strain, in 
which it is not used as a carbon source. Our results provide a model for the molecular 
mechanism of anaerobic protein secretion using fumarate as the final electron acceptor, 
which may allow for further engineering of yeast for improved protein secretion under 
anaerobic growth conditions.

PMID: 23435897

GEO ID: GSE38848.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE38848.remapped.final.pcl	Anaerobic alpha-amylase production and secretion with fumarate as the final electron acceptor in Saccharomyces cerevisiae.	18	oxygen level alteration	1

File last modified: Thu Apr 25 2024

---


# Liu_2013_PMID_23725375

Citation: Liu J, Huang X, Withers BR, Blalock E, Liu K, Dickson RC. 2013. Reducing sphingolipid 
synthesis orchestrates global changes to extend yeast lifespan.

Full Description: Studies of aging and longevity are revealing how diseases that shorten life can be 
controlled to improve the quality of life and lifespan itself. Two strategies  under 
intense study to accomplish these goals are rapamycin treatment and calorie restriction. 
New strategies are being discovered including one that uses low-dose myriocin treatment. 
Myriocin inhibits the first enzyme in sphingolipid synthesis  in all eukaryotes, and we 
showed recently that low-dose myriocin treatment increases yeast lifespan at least in part
by down-regulating the sphingolipid-controlled Pkh1/2-Sch9 (ortholog of mammalian S6 
kinase) signaling pathway. Here we show that myriocin treatment induces global effects and
changes  expression of approximately forty percent of the yeast genome with 1252 genes up-
regulated and 1497 down-regulated (P < 0.05) compared with untreated cells. These changes 
are due to modulation of evolutionarily conserved signaling pathways including activation 
of the Snf1/AMPK pathway and down-regulation of the protein kinase A (PKA) and target of 
rapamycin complex 1 (TORC1) pathways. Many processes that enhance lifespan are regulated 
by these pathways in response to myriocin treatment including respiration, carbon 
metabolism, stress resistance, protein synthesis, and autophagy. These extensive effects 
of myriocin match those of rapamycin and calorie restriction. Our studies in yeast 
together with other studies in mammals reveal the potential of myriocin or related 
compounds to lower the incidence of age-related diseases in humans and improve health span.

PMID: 23725375

GEO ID: GSE41362.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE41362.final.pcl	Reducing sphingolipid synthesis orchestrates global changes to extend yeast lifespan.	6	cell aging|carbon utilization|chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Lopez-Serra_2013_PMID_23219725

Citation: Lopez-Serra L, Lengronne A, Borges V, Kelly G, Uhlmann F. 2013. Budding yeast Wapl 
controls sister chromatid cohesion maintenance and chromosome  condensation.

Full Description: The establishment of stable sister chromatid cohesion during DNA replication requires 
acetylation of the chromosomal cohesin complex by the replication fork-associated 
acetyltransferase Eco1. Cohesin acetylation is thought to facilitate replication fork 
progression by counteracting an as yet ill-defined cohesion antiestablishment activity 
imposed by the Wapl protein. Here, using budding yeast, we find no evidence that cohesin 
acetylation must overcome Wapl during replication fork progression. Instead, Wapl emerges 
as a negative regulator of cohesion maintenance in G2, a function that it likely exerts 
through its role as destabilizer of unacetylated, chromosome-bound cohesin. Our results 
suggest that acetylation renders cohesin Wapl-resistant from S phase onward until mitosis
. In the absence of Wapl, sister chromatid cohesion functions well, suggesting that Wapl 
partakes in a cohesin function outside of sister chromatid cohesion. We find that Wapl is 
not required for cohesin's known role in transcriptional regulation. Rather, cells lacking
Wapl display increased chromosome condensation in both interphase and mitosis. Thus, as a 
conserved regulator of cohesin dynamics on chromosomes, Wapl controls cohesion maintenance
after its establishment in S phase and adjusts the chromosome condensation status.

PMID: 23219725

GEO ID: GSE42458.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE42458.final.pcl	Budding yeast Wapl controls sister chromatid cohesion maintenance and chromosome  condensation.	4	chromosome segregation	1

File last modified: Thu Apr 25 2024

---


# Mendes_2013_PMID_23872557

Citation: Mendes F, Sieuwerts S, de Hulster E, Almering MJ, Luttik MA, Pronk JT, Smid EJ, Bron PA, 
Daran-Lapujade P. 2013. Transcriptome-based characterization of interactions between 
Saccharomyces cerevisiae and Lactobacillus delbrueckii subsp. bulgaricus in lactose-grown 
chemostat cocultures.

Full Description: Mixed populations of Saccharomyces cerevisiae yeasts and lactic acid bacteria occur in 
many dairy, food, and beverage fermentations, but knowledge about their  interactions is 
incomplete. In the present study, interactions between Saccharomyces cerevisiae and 
Lactobacillus delbrueckii subsp. bulgaricus, two microorganisms that co-occur in kefir 
fermentations, were studied during anaerobic growth on lactose. By combining physiological
and transcriptome analysis of the two strains in the cocultures, five mechanisms of 
interaction were identified. (i) Lb. delbrueckii subsp. bulgaricus hydrolyzes lactose, 
which  cannot be metabolized by S. cerevisiae, to galactose and glucose. Subsequently, 
galactose, which cannot be metabolized by Lb. delbrueckii subsp. bulgaricus, is excreted 
and provides a carbon source for yeast. (ii) In pure cultures, Lb. delbrueckii subsp. 
bulgaricus grows only in the presence of increased CO2 concentrations. In anaerobic mixed 
cultures, the yeast provides this CO2 via alcoholic fermentation. (iii) Analysis of amino 
acid consumption from the defined medium indicated that S. cerevisiae supplied alanine to 
the bacterium. (iv) A mild but significant low-iron response in the yeast transcriptome, 
identified by  DNA microarray analysis, was consistent with the chelation of iron by the 
lactate produced by Lb. delbrueckii subsp. bulgaricus. (v) Transcriptome analysis of Lb.  
delbrueckii subsp. bulgaricus in mixed cultures showed an overrepresentation of 
transcripts involved in lipid metabolism, suggesting either a competition of the  two 
microorganisms for fatty acids or a response to the ethanol produced by S. cerevisiae. 
This study demonstrates that chemostat-based transcriptome analysis is a powerful tool to 
investigate microbial interactions in mixed populations.

PMID: 23872557

GEO ID: GSE45776.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE45776.remapped.final.pcl	Transcriptome-based characterization of interactions between Saccharomyces cerevisiae and Lactobacillus delbrueckii subsp. bulgaricus in lactose-grown chemostat cocultures.	6	carbon utilization	1

File last modified: Thu Apr 25 2024

---


# Montefusco_2013_PMID_24170935

Citation: Montefusco DJ, Chen L, Matmati N, Lu S, Newcomb B, Cooper GF, Hannun YA, Lu X. 2013. 
Distinct signaling roles of ceramide species in yeast revealed through systematic 
perturbation and systems biology analyses.

Full Description: Ceramide, the central molecule of sphingolipid metabolism, is an important bioactive 
molecule that participates in various cellular regulatory events and that has been 
implicated in disease. Deciphering ceramide signaling is challenging because multiple 
ceramide species exist, and many of them may have distinct functions. We applied systems 
biology and molecular approaches to perturb ceramide metabolism in the yeast Saccharomyces
cerevisiae and inferred causal relationships between ceramide species and their potential 
targets by combining lipidomic, genomic, and transcriptomic analyses. We found that during
heat stress, distinct metabolic mechanisms controlled the abundance of different  groups 
of ceramide species and provided experimental support for the importance of the 
dihydroceramidase Ydc1 in mediating the decrease in dihydroceramides during heat stress. 
Additionally, distinct groups of ceramide species, with different N-acyl chains and 
hydroxylations, regulated different sets of functionally related genes, indicating that 
the structural complexity of these lipids produces functional diversity. The 
transcriptional modules that we identified provide a resource to begin to dissect the 
specific functions of ceramides.

PMID: 24170935

GEO ID: GSE40399.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE40399.final.pcl	Distinct signaling roles of ceramide species in yeast revealed through systematic perturbation and systems biology analyses.	21	lipid metabolism|chemical stimulus|stress	1

File last modified: Thu Apr 25 2024

---


# Moreno-Cermeno_2013_PMID_24100161

Citation: Moreno-Cermeno A, Alsina D, Cabiscol E, Tamarit J, Ros J. 2013. Metabolic remodeling in 
frataxin-deficient yeast is mediated by Cth2 and Adr1.

Full Description: Frataxin is a mitochondrial protein involved in iron metabolism whose deficiency  in 
humans causes Friedreich ataxia. We performed transcriptomic and proteomic analyses of 
conditional Yeast Frataxin Homologue (Yfh1) mutants (tetO7-YFH1) to investigate metabolic 
remodeling upon Yfh1 depletion. These studies revealed that Yfh1 depletion leads to 
downregulation of many glucose-repressed genes. Most of them were Adr1 targets, a key 
transcription factor required for growth in non-fermentable carbon sources. Using a GFP-
tagged Adr1, we observed that Yfh1 depletion promotes the export of Adr1 from the nucleus 
to the cytosol without affecting its protein levels. This effect was also observed upon 
H2O2 treatment,  but not by iron overload/starvation, indicating the presence of a 
regulatory pathway involved in Adr1 export and inactivation upon stress conditions. We 
also  observed that CTH2, a gene involved in the mRNA degradation of several iron-
containing enzymes, was induced upon Yfh1 depletion. Accordingly, decreased  levels of 
aconitase and succinate dehydrogenase were observed. Nevertheless, their levels were 
maintained in a Deltacth2 mutant even in the absence of Yfh1. From these results we can 
conclude that, in addition to altering iron homeostasis, frataxin depletion involves 
drastic metabolic remodeling governed by Adr1 and Cth2 that finally leads to 
downregulation of iron-sulfur proteins and other proteins involved in respiratory 
metabolism.

PMID: 24100161

GEO ID: GSE44871.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE44871.remapped.final.pcl	Metabolic remodeling in frataxin-deficient yeast is mediated by Cth2 and Adr1.	6	oxidative stress|stress|fermentation|respiration	2

File last modified: Thu Apr 25 2024

---


# Ng_2013_PMID_24037263

Citation: Ng TM, Lenstra TL, Duggan N, Jiang S, Ceto S, Holstege FC, Dai J, Boeke JD, Biggins S. 
2013. Kinetochore function and chromosome segregation rely on critical residues in 
histones H3 and H4 in budding yeast.

Full Description: Accurate chromosome segregation requires that sister kinetochores biorient and attach to 
microtubules from opposite poles. Kinetochore biorientation relies on the underlying 
centromeric chromatin, which provides a platform to assemble the kinetochore and to 
recruit the regulatory factors that ensure the high fidelity of this process. To identify 
the centromeric chromatin determinants that contribute to chromosome segregation, we 
performed two complementary unbiased genetic screens using a library of budding yeast 
mutants in every residue of histone H3 and H4. In one screen, we identified mutants that 
lead to increased loss of a nonessential chromosome. In the second screen, we isolated 
mutants whose viability depends on a key regulator of biorientation, the Aurora B protein 
kinase. Nine mutants were common to both screens and exhibited kinetochore biorientation 
defects. Four of the mutants map near the unstructured nucleosome entry site, and their 
genetic interaction with reduced IPL1 can be suppressed by  increasing the dosage of SGO1
, a key regulator of biorientation. In addition, the composition of purified kinetochores 
was altered in six of the mutants. Together, this work identifies previously unknown 
histone residues involved in chromosome segregation and lays the foundation for future 
studies on the role of the underlying chromatin structure in chromosome segregation.

PMID: 24037263

GEO ID: GSE39903.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE39903.remapped.final.pcl	Kinetochore function and chromosome segregation rely on critical residues in histones H3 and H4 in budding yeast.	22	chromatin organization	2

File last modified: Thu Apr 25 2024

---


# O'Doherty_2013_PMID_24074273

Citation: O'Doherty PJ, Lyons V, Higgins VJ, Rogers PJ, Bailey TD, Wu MJ. 2013. Transcriptomic 
insights into the molecular response of Saccharomyces cerevisiae to linoleic acid 
hydroperoxide.

Full Description: Eukaryotic microorganisms are constantly challenged by reactive oxygen species derived 
endogenously or encountered in their environment. Such adversity is particularly applied 
to Saccharomyces cerevisiae under harsh industrial conditions. One of the major oxidants 
to challenge S. cerevisiae is linoleic acid hydroperoxide (LoaOOH). This study, which used
genome-wide microarray analysis in conjunction with deletion mutant screening, uncovered 
the molecular pathways of S. cerevisiae that were altered by an arresting concentration of
LoaOOH (75 muM). The oxidative stress response, iron homeostasis, detoxification through 
PDR transport and direct lipid beta-oxidation were evident through the induction of the 
genes encoding for peroxiredoxins (GPX2, TSA2), the NADPH:oxidoreductase (OYE3), iron 
uptake (FIT2, ARN2, FET3), PDR transporters (PDR5, PDR15, SNQ2) and  beta-oxidation 
machinery (FAA2, POX1). Further, we discovered that Gpx3p, the dual redox sensor and 
peroxidase, is required for protection against LoaOOH, indicated by the sensitivity of 
gpx3Delta to a mild dose of LoaOOH (37.5 muM). Deletion of GPX3 conferred a greater 
sensitivity to LoaOOH than the loss of its signalling partner YAP1. Deletion of either of 
the iron homeostasis regulators AFT1 or AFT2 also resulted in sensitivity to LoaOOH. These
novel findings for Gpx3p, Aft1p and Aft2p point to their distinct roles in response to the
lipid peroxide. Finally, the expression of 89 previously uncharacterised genes was 
significantly altered against LoaOOH, which will contribute to their eventual annotation.

PMID: 24074273

GEO ID: GSE47820.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE47820.remapped.final.pcl	Transcriptomic insights into the molecular response of Saccharomyces cerevisiae to linoleic acid hydroperoxide.	6	oxidative stress|stress	1

File last modified: Thu Apr 25 2024

---


# Park_2013_PMID_24349523

Citation: Park D, Lee Y, Bhupindersingh G, Iyer VR. 2013. Widespread misinterpretable ChIP-seq bias 
in yeast.

Full Description: Chromatin immunoprecipitation followed by sequencing (ChIP-seq) is widely used to detect 
genome-wide interactions between a protein of interest and DNA in vivo. Loci showing 
strong enrichment over adjacent background regions are typically considered to be sites of
binding. Insufficient attention has been given to systematic artifacts inherent to the 
ChIP-seq procedure that might generate a misleading picture of protein binding to certain 
loci. We show here that unrelated transcription factors appear to consistently bind to the
gene bodies of highly transcribed genes in yeast. Strikingly, several types of negative 
control  experiments, including a protein that is not expected to bind chromatin, also 
showed similar patterns of strong binding within gene bodies. These false positive signals
were evident across sequencing platforms and immunoprecipitation protocols, as well as in 
previously published datasets from other labs. We show that these false positive signals 
derive from high rates of transcription, and are inherent to the ChIP procedure, although 
they are exacerbated by sequencing library construction procedures. This expression bias 
is strong enough that a known transcriptional repressor like Tup1 can erroneously appear 
to be an activator. Another type of background bias stems from the inherent nucleosomal 
structure of chromatin, and can potentially make it seem like certain factors bind 
nucleosomes even when they don't. Our analysis suggests that a mock ChIP sample offers a 
better normalization control for the expression bias, whereas the ChIP input is more 
appropriate for the nucleosomal periodicity bias. While these  controls alleviate the 
effect of the biases to some extent, they are unable to eliminate it completely. Caution 
is therefore warranted regarding the interpretation of data that seemingly show the 
association of various transcription and chromatin factors with highly transcribed genes 
in yeast.

PMID: 24349523

GEO ID: GSE51376.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE51376.final.pcl	Widespread misinterpretable ChIP-seq bias in yeast.	12	transcription|cytoskeleton and molecular motors|chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Rachfall_2013_PMID_23071099

Citation: Rachfall N, Schmitt K, Bandau S, Smolinski N, Ehrenreich A, Valerius O, Braus GH. 2013. 
RACK1/Asc1p, a ribosomal node in cellular signaling.

Full Description: RACK1/Asc1p and its essential orthologues in higher eukaryotes, such as RACK1 in  metazoa
, are involved in several distinct cellular signaling processes. The implications of a 
total deletion have never been assessed in a comprehensive manner. This study reveals the 
major cellular processes affected in a Saccharomyces cerevisiae Deltaasc1 deletion 
background via de novo proteome and transcriptome analysis, as well as subsequent 
phenotypical characterizations. The deletion of ASC1 reduces iron uptake and causes 
nitrosative stress, both known indicators for hypoxia, which manifests in a shift of 
energy metabolism from respiration to fermentation in the Deltaasc1 strain. Asc1p further 
impacts cellular metabolism through its regulative role in the MAP kinase signal 
transduction pathways of invasive/filamentous growth and cell wall integrity. In  the 
Deltaasc1 mutant strain, aberrations from the expected cellular response, mediated by 
these pathways, can be observed and are linked to changes in protein  abundances of 
pathway-targeted transcription factors. Evidence of the translational regulation of such 
transcription factors suggests that ribosomal Asc1p is involved in signal transduction 
pathways and controls the biosynthesis of the respective final transcriptional regulators.

PMID: 23071099

GEO ID: GSE34286

PCL filename	short description	# conditions	tags	# channels
GSE34286_final.pcl	RACK1/Asc1p, a ribosomal node in cellular signaling.	12	signaling	2

File last modified: Thu Apr 25 2024

---


# Reiter_2013_PMID_23275436

Citation: Reiter W, Klopf E, De Wever V, Anrather D, Petryshyn A, Roetzer A, Niederacher G, 
Roitinger E, Dohnal I, Gorner W, Mechtler K, Brocard C, Schuller C, Ammerer G. 2013. Yeast
protein phosphatase 2A-Cdc55 regulates the transcriptional response to hyperosmolarity 
stress by regulating Msn2 and Msn4 chromatin recruitment.

Full Description: We have identified Cdc55, a regulatory B subunit of protein phosphatase 2A (PP2A), as an 
essential activating factor for stress gene transcription in Saccharomyces cerevisiae. The
presence of PP2A-Cdc55 is required for full activation of the environmental stress 
response mediated by the transcription factors Msn2 and Msn4. We show that PP2A-Cdc55 
contributes to sustained nuclear accumulation of Msn2 and Msn4 during hyperosmolarity 
stress. PP2A-Cdc55 also enhances Msn2-dependent transactivation, required for extended 
chromatin recruitment of the transcription factor. We analyzed a possible direct 
regulatory role for PP2A-Cdc55 on the phosphorylation status of Msn2. Detailed mass 
spectrometric and genetic analysis of Msn2 showed that stress exposure causes immediate 
transient dephosphorylation of Msn2 which is not dependent on PP2A-Cdc55 activity. 
Furthermore, the Hog1 mitogen-activated protein kinase pathway activity is not influenced 
by PP2A-Cdc55. We therefore propose that the PP2A-Cdc55 phosphatase is not involved in 
cytosolic stress signal perception but  is involved in a specific intranuclear mechanism 
to regulate Msn2 and Msn4 nuclear accumulation and chromatin association under stress 
conditions.

PMID: 23275436

GEO ID: GSE38565.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE38565.final.pcl	Yeast protein phosphatase 2A-Cdc55 regulates the transcriptional response to hyperosmolarity stress by regulating Msn2 and Msn4 chromatin recruitment.	16	osmotic stress|transcriptional regulation	1

File last modified: Thu Apr 25 2024

---


# Roy_2013_PMID_23640720

Citation: Roy S, Wapinski I, Pfiffner J, French C, Socha A, Konieczka J, Habib N, Kellis M, Thompson
D, Regev A. 2013. Arboretum: reconstruction and analysis of the evolutionary history of 
condition-specific transcriptional modules.

Full Description: Comparative functional genomics studies the evolution of biological processes by  
analyzing functional data, such as gene expression profiles, across species. A major 
challenge is to compare profiles collected in a complex phylogeny. Here, we present 
Arboretum, a novel scalable computational algorithm that integrates expression data from 
multiple species with species and gene phylogenies to infer  modules of coexpressed genes 
in extant species and their evolutionary histories.  We also develop new, generally 
applicable measures of conservation and divergence in gene regulatory modules to assess 
the impact of changes in gene content and expression on module evolution. We used 
Arboretum to study the evolution of the transcriptional response to heat shock in eight 
species of Ascomycota fungi and to reconstruct modules of the ancestral environmental 
stress response (ESR). We found substantial conservation in the stress response across 
species and in the reconstructed components of the ancestral ESR modules. The greatest 
divergence was in the most induced stress, primarily through module expansion. The 
divergence of the heat stress response exceeds that observed in the response to glucose 
depletion in the same species. Arboretum and its associated analyses provide a 
comprehensive framework to systematically study regulatory evolution of condition-specific
responses.

PMID: 23640720

GEO ID: GSE38478GPL9294.remapped.sfp.pcl

PCL filename	short description	# conditions	tags	# channels
GSE38478GPL9294.remapped.sfp.pcl	Arboretum: reconstruction and analysis of the evolutionary history of condition-specific transcriptional modules.	32	heat shock|oxidative stress|osmotic stress|stress	2

File last modified: Thu Apr 25 2024

---


# Santos-Pereira_2013_PMID_24240235

Citation: Santos-Pereira JM, Herrero AB, Garcia-Rubio ML, Marin A, Moreno S, Aguilera A. 2013. The 
Npl3 hnRNP prevents R-loop-mediated transcription-replication conflicts and genome 
instability.

Full Description: Transcription is a major obstacle for replication fork (RF) progression and a cause of 
genome instability. Part of this instability is mediated by cotranscriptional R loops, 
which are believed to increase by suboptimal assembly  of the nascent messenger 
ribonucleoprotein particle (mRNP). However, no clear evidence exists that heterogeneous 
nuclear RNPs (hnRNPs), the basic mRNP components, prevent R-loop stabilization. Here we 
show that yeast Npl3, the most  abundant RNA-binding hnRNP, prevents R-loop-mediated 
genome instability. npl3Delta cells show transcription-dependent and R-loop-dependent 
hyperrecombination and genome-wide replication obstacles as determined by accumulation of 
the Rrm3 helicase. Such obstacles preferentially occur at long and highly expressed genes
, to which Npl3 is preferentially bound in wild-type cells, and are reduced by RNase H1 
overexpression. The resulting replication stress confers hypersensitivity to double-strand
break-inducing agents. Therefore, our work demonstrates that mRNP factors are critical for
genome integrity and opens the option of using them as therapeutic targets in anti-cancer 
treatment.

PMID: 24240235

GEO ID: GSE50186.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE50186.remapped.final.pcl	The Npl3 hnRNP prevents R-loop-mediated transcription-replication conflicts and genome instability.	6	mRNA processing|transcription	1

File last modified: Thu Apr 25 2024

---


# Schuller_2013_PMID_23576094

Citation: Schuller A, Auffermann G, Zoschke K, Schmidt U, Ostermann K, Rodel G. 2013. Overexpression
of ctr1Delta300, a high-affinity copper transporter with deletion  of the cytosolic C-
terminus in Saccharomyces cerevisiae under excess copper, leads to disruption of 
transition metal homeostasis and transcriptional remodelling of cellular processes.

Full Description: In an approach to generating Saccharomyces cerevisiae strains with increased intracellular
copper amounts for technical applications, we overexpressed the copper transporter CTR1 
and a variant of CTR1 with a truncation in the C-terminus after the 300th amino acid (
ctr1Delta300). We determined the copper sensitivity of the generated strains and used 
inductively coupled plasma spectrometry analysis (ICP-OES and ICP-MS) to investigate the 
effects of overexpression of both constructs under excess copper on the cellular content 
of different elements in S. cerevisiae. In addition, we performed DNA microarray analysis 
to obtain the gene expression profile under the changed element contents. Overexpression 
of CTR1 increased the copper content in the cells to 160% and 78 genes were differentially
regulated. Overexpression of the truncated ctr1Delta300 resulted in an increased copper, 
iron and zinc content of > 200% and 980 genes showed differential expression. We found 
that transition metal ion homeostasis was disrupted in ctr1Delta300-overexpressing strains
under excess copper and that this was combined with a transcriptional remodelling of 
cellular processes.

PMID: 23576094

GEO ID: GSE44043.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE44043.final.pcl	Overexpression of ctr1Delta300, a high-affinity copper transporter with deletion  of the cytosolic C-terminus in Saccharomyces cerevisiae under excess copper, leads to disruption of transition metal homeostasis and transcriptional remodelling of cellular processes.	12	metal or metalloid ion stress	1

File last modified: Thu Apr 25 2024

---


# Sha_2013_PMID_24073228

Citation: Sha W, Martins AM, Laubenbacher R, Mendes P, Shulaev V. 2013. The genome-wide early 
temporal response of Saccharomyces cerevisiae to oxidative  stress induced by cumene 
hydroperoxide.

Full Description: Oxidative stress is a well-known biological process that occurs in all respiring  cells 
and is involved in pathophysiological processes such as aging and apoptosis. Oxidative 
stress agents include peroxides such as hydrogen peroxide, cumene hydroperoxide, and 
linoleic acid hydroperoxide, the thiol oxidant diamide, and menadione, a generator of 
superoxide, amongst others. The present study analyzed the early temporal genome-wide 
transcriptional response of Saccharomyces cerevisiae to oxidative stress induced by the 
aromatic peroxide cumene hydroperoxide. The accurate dataset obtained, supported by the 
use of temporal controls, biological replicates and well controlled growth conditions, 
provided a detailed picture of the early dynamics of the process. We identified a set of 
genes previously not implicated in the oxidative stress response, including several 
transcriptional regulators showing a fast transient response, suggesting  a coordinated 
process in the transcriptional reprogramming. We discuss the role of the glutathione, 
thioredoxin and reactive oxygen species-removing systems, the proteasome and the pentose 
phosphate pathway. A data-driven clustering of the expression patterns identified one 
specific cluster that mostly consisted of genes known to be regulated by the Yap1p and 
Skn7p transcription factors, emphasizing their mediator role in the transcriptional 
response to oxidants. Comparison of our results with data reported for hydrogen peroxide 
identified 664 genes that specifically respond to cumene hydroperoxide, suggesting 
distinct transcriptional responses to these two peroxides. Genes up-regulated only by 
cumene hydroperoxide are mainly related to the cell membrane and cell wall, and 
proteolysis process, while those down-regulated only by this aromatic peroxide are 
involved in mitochondrial function.

PMID: 24073228

GEO ID: GSE7645.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE7645.remapped.final.pcl	The genome-wide early temporal response of Saccharomyces cerevisiae to oxidative  stress induced by cumene hydroperoxide.	48	oxidative stress|stress	1

File last modified: Thu Apr 25 2024

---


# Suzuki_2013_PMID_23325148

Citation: Suzuki T, Iwahashi Y. 2013. RNA preparation of Saccharomyces cerevisiae using the 
digestion method may give misleading results.

Full Description: Zymolyase (lyticase) is used for cell wall digestion in yeast experiments and is  needed 
for incubation processes under moderate experimental conditions. This has  been thought to
cause unfavorable effects, and many researchers are aware that the enzyme method is 
unsuitable for RNA preparation following several reports of  stress responses to the 
enzyme process. However, RNA preparation with enzyme digestion continues to be used. This 
may be because there have been insufficient  data directly comparing RNA preparation 
conditions with previous studies. We investigated the influence of enzyme processes in RNA
preparation using a DNA microarray, and compared superoxide dismutase (SOD) activities 
with a non-treated control and the results of previous research. Gene expressions were 
commonly changed by enzyme processes, and SOD activities increased only during short-term
incubation. Meanwhile, both SOD gene expressions and SOD activity during RNA preparation 
indicated different results than gained under conditions of long-term incubation. These 
results suggest that zymolyase treatment surely influences gene expressions and enzyme 
activity, although the effect assumed by previous studies  is not necessarily in agreement
with that of RNA preparation.

PMID: 23325148

GEO ID: GSE36955.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE36955.remapped.final.pcl	RNA preparation of Saccharomyces cerevisiae using the digestion method may give misleading results.	6	cell wall organization	1

File last modified: Thu Apr 25 2024

---


# Tesniere_2013_PMID_23658613

Citation: Tesniere C, Delobel P, Pradal M, Blondin B. 2013. Impact of nutrient imbalance on wine 
alcoholic fermentations: nitrogen excess enhances yeast cell death in lipid-limited must.

Full Description: We evaluated the consequences of nutritional imbalances, particularly lipid/nitrogen 
imbalances, on wine yeast survival during alcoholic fermentation.  We report that lipid 
limitation (ergosterol limitation in our model) led to a rapid loss of viability during 
the stationary phase of fermentation and that the  cell death rate is strongly modulated 
by nitrogen availability and nature. Yeast  survival was reduced in the presence of excess
nitrogen in lipid-limited fermentations. The rapidly dying yeast cells in fermentations in
high nitrogen and lipid-limited conditions displayed a lower storage of the carbohydrates 
trehalose and glycogen than observed in nitrogen-limited cells. We studied the cell stress
response using HSP12 promoter-driven GFP expression as a marker, and  found that lipid 
limitation triggered a weaker stress response than nitrogen limitation. We used a SCH9-
deleted strain to assess the involvement of nitrogen signalling pathways in the triggering
of cell death. Deletion of SCH9 increased yeast viability in the presence of excess 
nitrogen, indicating that a signalling  pathway acting through Sch9p is involved in this 
nitrogen-triggered cell death. We also show that various nitrogen sources, but not 
histidine or proline, provoked cell death. Our various findings indicate that lipid 
limitation does not elicit a transcriptional programme that leads to a stress response 
protecting yeast cells and that nitrogen excess triggers cell death by modulating this 
stress response, but not through HSP12. These results reveal a possibly negative  role of 
nitrogen in fermentation, with reported effects referring to ergosterol limitation 
conditions. These effects should be taken into account in the management of alcoholic 
fermentations.

PMID: 23658613

GEO ID: GSE42027.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE42027.remapped.final.pcl	Impact of nutrient imbalance on wine alcoholic fermentations: nitrogen excess enhances yeast cell death in lipid-limited must.	4	lipid metabolism|stress	1

File last modified: Thu Apr 25 2024

---


# Van_de_Vosse_2013_PMID_23452847

Citation: Van de Vosse DW, Wan Y, Lapetina DL, Chen WM, Chiang JH, Aitchison JD, Wozniak RW. 2013. A,
role for the nucleoporin Nup170p in chromatin structure and gene silencing.

Full Description: Embedded in the nuclear envelope, nuclear pore complexes (NPCs) not only regulate nuclear 
transport but also interface with transcriptionally active euchromatin, largely silenced 
heterochromatin, as well as the boundaries between these regions. It is unclear what 
functional role NPCs play in establishing or maintaining these distinct chromatin domains
. We report that the yeast NPC protein Nup170p interacts with regions of the genome that 
contain ribosomal protein and subtelomeric genes, where it functions in nucleosome 
positioning and  as a repressor of transcription. We show that the role of Nup170p in 
subtelomeric gene silencing is linked to its association with the RSC chromatin-remodeling
complex and the silencing factor Sir4p, and that the binding of Nup170p and Sir4p to 
subtelomeric chromatin is cooperative and necessary for the association of telomeres with 
the nuclear envelope. Our results establish the NPC as an active participant in silencing 
and the formation of peripheral heterochromatin.

PMID: 23452847

GEO ID: GSE36793.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE36793.final.pcl	A role for the nucleoporin Nup170p in chromatin structure and gene silencing.	6	transcription|chromatin organization	2

File last modified: Thu Apr 25 2024

---


# Worley_2013_PMID_23643537

Citation: Worley J, Luo X, Capaldi AP. 2013. Inositol pyrophosphates regulate cell growth and the 
environmental stress response by activating the HDAC Rpd3L.

Full Description: Cells respond to stress and starvation by adjusting their growth rate and enacting stress 
defense programs. In eukaryotes this involves inactivation of TORC1, which in turn 
triggers downregulation of ribosome and protein synthesis genes and upregulation of stress
response genes. Here we report that the highly conserved inositol pyrophosphate (PP-IP) 
second messengers (including 1-PP-IP5, 5-PP-IP4, and 5-PP-IP5) are also critical 
regulators of cell growth and the general stress response, acting in parallel with the 
TORC1 pathway to control the activity of the class I histone deacetylase Rpd3L. In fact, 
yeast cells that cannot synthesize any of the PP-IPs mount little to no transcriptional 
response to osmotic, heat, or oxidative stress. Furthermore, PP-IP-dependent regulation of
Rpd3L occurs independently of the role individual PP-IPs (such as 5-PP-IP5) play  in 
activating specialized stress/starvation response pathways. Thus, the PP-IP second 
messengers simultaneously activate and tune the global response to stress  and starvation 
signals.

PMID: 23643537

GEO ID: GSE45370.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE45370.remapped.final.pcl	Inositol pyrophosphates regulate cell growth and the environmental stress response by activating the HDAC Rpd3L.	55	heat shock|oxidative stress|osmotic stress|stress|signaling	2

File last modified: Thu Apr 25 2024

---


# Xue-Franzen_2013_PMID_23865462

Citation: Xue-Franzen Y, Henriksson J, Burglin TR, Wright AP. 2013. Distinct roles of the Gcn5 
histone acetyltransferase revealed during transient stress-induced reprogramming of the 
genome.

Full Description: BACKGROUND: Gcn5 belongs to a family of histone acetyltransferases (HATs) that regulate 
protein function by acetylation. Gcn5 plays several different roles in gene transcription 
throughout the genome but their characterisation by classical  mutation approaches is 
hampered by the high degree of apparent functional redundancy between HAT proteins. 
RESULTS: Here we utilise the reduced redundancy  associated with the transiently high 
levels of genomic reprogramming during stress adaptation as a complementary approach to 
understand the functions of redundant protein families like HATs. We show genome-wide 
evidence for two functionally distinct roles of Gcn5. First, Gcn5 transiently re-localises
to the  ORFs of long genes during stress adaptation. Taken together with earlier 
mechanistic studies, our data suggests that Gcn5 plays a genome- wide role in specifically
increasing the transcriptional elongation of long genes, thus increasing the production 
efficiency of complete long transcripts. Second, we suggest that Gcn5 transiently 
interacts with histones close to the transcription  start site of the many genes that it 
activates during stress adaptation by acetylation of histone H3K18, leading to histone 
depletion, probably as a result  of nucleosome loss as has been described previously. 
CONCLUSIONS: We show that stress adaptation can be used to elucidate the functions of 
otherwise redundant proteins, like Gcn5, in gene transcription. Further, we show that 
normalization of chromatin-associated protein levels in ChIP experiments in relation to 
the histone levels may provide a useful complement to standard approaches. In the present 
study analysis of data in this way provides an alternative explanation for previously 
indicated repressive role of Gcn5 in gene transcription.

PMID: 23865462

GEO ID: GSE36599.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE36599.remapped.final.pcl	Distinct roles of the Gcn5 histone acetyltransferase revealed during transient stress-induced reprogramming of the genome.	10	osmotic stress|stress	1

File last modified: Thu Apr 25 2024

---


# van_de_Pasch_2013_PMID_23785440

Citation: van de Pasch LA, Miles AJ, Nijenhuis W, Brabers NA, van Leenen D, Lijnzaad P, Brown MK, 
Ouellet J, Barral Y, Kops GJ, Holstege FC. 2013. Centromere binding and a conserved role 
in chromosome stability for SUMO-dependent ubiquitin ligases.

Full Description: The Saccharomyces cerevisiae Slx5/8 complex is the founding member of a recently  defined 
class of SUMO-targeted ubiquitin ligases (STUbLs). Slx5/8 has been implicated in genome 
stability and transcription, but the precise contribution is unclear. To characterise Slx5
/8 function, we determined genome-wide changes in gene expression upon loss of either 
subunit. The majority of mRNA changes are part of a general stress response, also 
exhibited by mutants of other genome integrity pathways and therefore indicative of an 
indirect effect on transcription. Genome-wide binding analysis reveals a uniquely 
centromeric location for Slx5. Detailed phenotype analyses of slx5Delta and slx8Delta 
mutants show severe mitotic defects that include aneuploidy, spindle mispositioning, fish 
hooks and aberrant spindle kinetics. This is associated with accumulation of the  PP2A 
regulatory subunit Rts1 at centromeres prior to entry into anaphase. Knockdown of the 
human STUbL orthologue RNF4 also results in chromosome segregation errors due to 
chromosome bridges. The study shows that STUbLs have a  conserved role in maintenance of 
chromosome stability and links SUMO-dependent ubiquitination to a centromere-specific 
function during mitosis.

PMID: 23785440

GEO ID: GSE33929.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE33929.remapped.final.pcl	Centromere binding and a conserved role in chromosome stability for SUMO-dependent ubiquitin ligases.	34	stress	2

File last modified: Thu Apr 25 2024

---


# Bisschops_2014_PMID_24487068

Citation: Bisschops MM, Zwartjens P, Keuter SG, Pronk JT, Daran-Lapujade P. 2014. To divide or not 
to divide: a key role of Rim15 in calorie-restricted yeast cultures.

Full Description: The PAS kinase Rim15 is proposed to integrate signals from different nutrient-sensing 
pathways and to control transcriptional reprogramming of Saccharomyces cerevisiae upon 
nutrient depletion. Despite this proposed role, previous transcriptome analyses of rim15 
mutants solely focused on growing cultures. In the present work, retentostat cultivation 
enabled analysis of the role of Rim15 under severely calorie-restricted, virtually non-
growing conditions. Under these conditions, deletion of RIM15 affected transcription of 
over 10-fold more genes than in growing cultures. Transcriptional responses, metabolic 
rates and cellular morphology indicated a key role of Rim15 in controlled cell-cycle 
arrest upon nutrient depletion. Moreover, deletion of rim15 reduced heat-shock tolerance 
in non-growing, but not in growing cultures. The failure of rim15 cells to adapt to 
calorie restriction by entering a robust post-mitotic state resembles cancer cell 
physiology and shows that retentostat cultivation of yeast strains can provide relevant 
models for healthy post-mitotic and transformed human cells.

PMID: 24487068

GEO ID: GSE46853.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE46853.final.pcl	To divide or not to divide: a key role of Rim15 in calorie-restricted yeast cultures.	11	cell aging|carbon utilization|cell aging|stationary phase entry	1

File last modified: Thu Apr 25 2024

---


# Bloom-Ackermann_2014_PMID_24453985

Citation: Bloom-Ackermann Z, Navon S, Gingold H, Towers R, Pilpel Y, Dahan O. 2014. A comprehensive 
tRNA deletion library unravels the genetic architecture of the tRNA pool.

Full Description: Deciphering the architecture of the tRNA pool is a prime challenge in translation research
, as tRNAs govern the efficiency and accuracy of the process. Towards this challenge, we 
created a systematic tRNA deletion library in Saccharomyces cerevisiae, aimed at 
dissecting the specific contribution of each tRNA gene to the tRNA pool and to the cell's 
fitness. By harnessing this resource, we observed that the majority of tRNA deletions show
no appreciable phenotype in rich medium, yet under more challenging conditions, additional
phenotypes were observed. Robustness to tRNA gene deletion was often facilitated through 
extensive backup compensation within and between tRNA families. Interestingly, we found 
that within tRNA families, genes carrying identical anti-codons can contribute differently
to the cellular fitness, suggesting the importance of the genomic surrounding to tRNA 
expression. Characterization of the transcriptome response to deletions of tRNA genes 
exposed two disparate patterns: in single-copy families,  deletions elicited a stress 
response; in deletions of genes from multi-copy families, expression of the translation 
machinery increased. Our results uncover  the complex architecture of the tRNA pool and 
pave the way towards complete understanding of their role in cell physiology.

PMID: 24453985

GEO ID: GSE47050.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE47050.final.pcl	A comprehensive tRNA deletion library unravels the genetic architecture of the tRNA pool.	13	translational regulation	1

File last modified: Thu Apr 25 2024

---


# Brice_2014_PMID_24334661

Citation: Brice C, Sanchez I, Tesniere C, Blondin B. 2014. Assessing the mechanisms responsible for 
differences between nitrogen requirements of saccharomyces cerevisiae wine yeasts in 
alcoholic fermentation.

Full Description: Nitrogen is an essential nutrient for Saccharomyces cerevisiae wine yeasts during 
alcoholic fermentation, and its abundance determines the fermentation rate and duration. 
The capacity to ferment under conditions of nitrogen deficiency differs between yeasts. A 
characterization of the nitrogen requirements of a set of 23 strains revealed large 
differences in their fermentative performances under nitrogen deficiency, and these 
differences reflect the nitrogen requirements of the strains. We selected and compared two
groups of strains, one with low nitrogen requirements (LNRs) and the other with high 
nitrogen requirements (HNRs). A comparison of various physiological traits indicated that 
the differences are not related to the ability to store nitrogen or the protein content. 
No differences in protein synthesis activity were detected between strains with different 
nitrogen requirements. Transcriptomic analysis revealed expression patterns specific to 
each of the two groups of strains, with an overexpression of stress genes in HNR strains 
and a stronger expression of biosynthetic genes in LNR strains. Our data suggest that 
differences in glycolytic flux may originate from variations in nitrogen sensing and 
signaling under conditions of starvation.

PMID: 24334661

GEO ID: GSE50626.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE50626.final.pcl	Assessing the mechanisms responsible for differences between nitrogen requirements of saccharomyces cerevisiae wine yeasts in alcoholic fermentation.	16	nitrogen utilization|fermentation	1

File last modified: Thu Apr 25 2024

---


# Bristow_2014_PMID_25200947

Citation: Bristow SL, Leman AR, Simmons Kovacs LA, Deckard A, Harer J, Haase SB. 2014. Checkpoints 
couple transcription network oscillator dynamics to cell-cycle progression.

Full Description: BackgroundThe coupling of cyclin dependent kinases (CDKs) to an intrinsically oscillating 
network of transcription factors has been proposed to control progression through the cell
cycle in budding yeast, Saccharomyces cerevisiae. The transcription network regulates the 
temporal expression of many genes, including cyclins, and drives cell-cycle progression, 
in part, by generating successive waves of distinct CDK activities that trigger the 
ordered program of cell-cycle events. Network oscillations continue autonomously in mutant
cells arrested by depletion of CDK activities, suggesting the oscillator can be uncoupled 
from cell-cycle progression. It is not clear what mechanisms, if any, ensure that the 
network oscillator is restrained when progression in normal cells is delayed or arrested. 
A recent proposal suggests CDK acts as a master regulator of cell-cycle processes that 
have the potential for autonomous oscillatory behavior.ResultsHere we find that mitotic 
CDK is not sufficient for fully inhibiting transcript oscillations in arrested cells. We 
do find that activation  of the DNA replication and spindle assembly checkpoints can fully
arrest the network oscillator via overlapping, but distinct mechanisms. Further, we 
demonstrate that the DNA replication checkpoint effector protein, Rad53, acts to  arrest a
portion of transcript oscillations in addition to its role in halting cell-cycle 
progression.ConclusionsOur findings indicate that checkpoint mechanisms, likely via 
phosphorylation of network transcription factors, maintain coupling of the network 
oscillator to progression during cell-cycle arrest.

PMID: 25200947

GEO ID: GSE49650.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE49650.remapped.final.pcl	Checkpoints couple transcription network oscillator dynamics to cell-cycle progression.	160	cell cycle regulation|protein phosphorylation	1

File last modified: Thu Apr 25 2024

---


# Chasman_2014_PMID_25411400

Citation: Chasman D, Ho YH, Berry DB, Nemec CM, MacGilvray ME, Hose J, Merrill AE, Lee MV, Will JL, 
Coon JJ, Ansari AZ, Craven M, Gasch AP. 2014. Pathway connectivity and signaling 
coordination in the yeast stress-activated signaling network.

Full Description: Stressed cells coordinate a multi-faceted response spanning many levels of physiology. Yet
knowledge of the complete stress-activated regulatory network as  well as design 
principles for signal integration remains incomplete. We developed an experimental and 
computational approach to integrate available protein interaction data with gene fitness 
contributions, mutant transcriptome profiles,  and phospho-proteome changes in cells 
responding to salt stress, to infer the salt-responsive signaling network in yeast. The 
inferred subnetwork presented many novel predictions by implicating new regulators, 
uncovering unrecognized crosstalk between known pathways, and pointing to previously 
unknown 'hubs' of signal integration. We exploited these predictions to show that Cdc14 
phosphatase is a central hub in the network and that modification of RNA polymerase II 
coordinates induction of stress-defense genes with reduction of growth-related transcripts
. We find that the orthologous human network is enriched for cancer-causing genes, 
underscoring the importance of the subnetwork's predictions in understanding stress 
biology.

PMID: 25411400

GEO ID: GSE56995.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE56995.final.pcl	Pathway connectivity and signaling coordination; Snf1 is involved in the aberrant cell cycle induction in cdc14-3 mutant upon NaCl stress in S. cerevisiae	8	osmotic stress|signaling	2

File last modified: Thu Apr 25 2024
GSE60700.final.pcl	Pathway connectivity and signaling coordination; Modification of C-terminal Domain (CTD) of RPB1 plays a role in transcriptome change upon salt stress	9	osmotic stress|signaling	2

File last modified: Thu Apr 25 2024

---


# Chiasson_2014_PMID_24707045

Citation: Chiasson DM, Loughlin PC, Mazurkiewicz D, Mohammadidehcheshmeh M, Fedorova EE, Okamoto M, 
McLean E, Glass AD, Smith SE, Bisseling T, Tyerman SD, Day DA, Kaiser BN. 2014. Soybean 
SAT1 (Symbiotic Ammonium Transporter 1) encodes a bHLH transcription factor involved in 
nodule growth and NH4+ transport.

Full Description: Glycine max symbiotic ammonium transporter 1 was first documented as a putative ammonium (
NH4(+)) channel localized to the symbiosome membrane of soybean root nodules. We show that
Glycine max symbiotic ammonium transporter 1 is actually a  membrane-localized basic helix
-loop-helix (bHLH) DNA-binding transcription factor now renamed Glycine max bHLH membrane 
1 (GmbHLHm1). In yeast, GmbHLHm1 enters the nucleus and transcriptionally activates a 
unique plasma membrane NH4(+) channel Saccharomyces cerevisiae ammonium facilitator 1. 
Ammonium facilitator 1 homologs  are present in soybean and other plant species, where 
they often share chromosomal microsynteny with bHLHm1 loci. GmbHLHm1 is important to the 
soybean rhizobium symbiosis because loss of activity results in a reduction of nodule 
fitness and growth. Transcriptional changes in nodules highlight downstream signaling 
pathways involving circadian clock regulation, nutrient transport, hormone signaling, and 
cell wall modification. Collectively, these results show that GmbHLHm1 influences nodule 
development and activity and is linked to a novel mechanism for NH4(+) transport common to
both yeast and plants.

PMID: 24707045

GEO ID: GSE55804.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE55804.final.pcl	Soybean SAT1 (Symbiotic Ammonium Transporter 1) encodes a bHLH transcription factor involved in nodule growth and NH4+ transport.	6	synthetic biology	1

File last modified: Thu Apr 25 2024

---


# Drozdova_2014_PMID_25227157

Citation: Drozdova P, Rogoza T, Radchenko E, Lipaeva P, Mironova L. 2014. Transcriptional response 
to the [ISP ] prion of Saccharomyces cerevisiae differs  from that induced by the deletion
of its structural gene, SFP1.

Full Description: Currently, several protein-based genetic determinants, or prions, are described in yeast, 
and several hundred prion candidates have been predicted. Importantly,  many known and 
potential prion proteins regulate transcription; therefore, prion  induction should affect
gene expression. While it is generally believed that the  prion phenotype should mimic the
deletion phenotype, this rule has exceptions. Formed by the transcription factor Sfp1p, [
ISP+ ] is one such exception as the [ISP+ ] and sfp1Delta strains differ in many 
phenotypic traits. These data suggest that effects of prion formation by a transcription 
factor and its absence may affect gene expression in a different way. However, studies 
addressing this issue are practically absent. Here, we explore how [ISP+ ] affects gene 
expression and how these changes correspond to the effect of SFP1 deletion. Our data 
indicate that the [ISP+ ]-related expression changes cannot be explained by  the 
inactivation of Sfp1p. Remarkably, most Sfp1p targets are not affected in the [ISP+ ] 
strain; instead, the genes upregulated in the [ISP+ ] strain are enriched in Gcn4p and 
Aft1p targets. We propose that Sfp1p serves as a part of a regulatory complex, and the 
activity of this complex may be modulated differently by the absence or prionization of 
Sfp1p.

PMID: 25227157

GEO ID: GSE52189.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE52189.remapped.final.pcl	Transcriptional response to the [ISP ] prion of Saccharomyces cerevisiae differs  from that induced by the deletion of its structural gene, SFP1.	12	transcription	1

File last modified: Thu Apr 25 2024

---


# Gaillard_2014_PMID_24603480

Citation: Gaillard H, Aguilera A. 2014. Cleavage factor I links transcription termination to DNA 
damage response and genome integrity maintenance in Saccharomyces cerevisiae.

Full Description: During transcription, the nascent pre-mRNA undergoes a series of processing steps before 
being exported to the cytoplasm. The 3'-end processing machinery involves  different 
proteins, this function being crucial to cell growth and viability in eukaryotes. Here, we
found that the rna14-1, rna15-1, and hrp1-5 alleles of the cleavage factor I (CFI) cause 
sensitivity to UV-light in the absence of global genome repair in Saccharomyces cerevisiae
. Unexpectedly, CFI mutants were proficient in UV-lesion repair in a transcribed gene. DNA
damage checkpoint activation and RNA polymerase II (RNAPII) degradation in response to UV 
were delayed in CFI-deficient cells, indicating that CFI participates in the DNA damage 
response (DDR). This is further sustained by the synthetic growth defects  observed 
between rna14-1 and mutants of different repair pathways. Additionally,  we found that 
rna14-1 suffers severe replication progression defects and that a functional G1/S 
checkpoint becomes essential in avoiding genetic instability in those cells. Thus, CFI 
function is required to maintain genome integrity and to prevent replication hindrance. 
These findings reveal a new function for CFI in the DDR and underscore the importance of 
coordinating transcription termination with replication in the maintenance of genomic 
stability.

PMID: 24603480

GEO ID: GSE50947.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE50947.remapped.final.pcl	Cleavage factor I links transcription termination to DNA damage response and genome integrity maintenance in Saccharomyces cerevisiae.	11	mRNA processing|transcription	1

File last modified: Thu Apr 25 2024

---


# Gamero_2014_PMID_24854353

Citation: Gamero A, Belloch C, Ibanez C, Querol A. 2014. Molecular analysis of the genes involved in
aroma synthesis in the species S. cerevisiae, S. kudriavzevii and S. bayanus var. uvarum 
in winemaking conditions.

Full Description: The Saccharomyces genus is the main yeast involved in wine fermentations to play  a 
crucial role in the production and release of aromatic compounds. Despite the several 
studies done into the genome-wide expression analysis using DNA microarray technology in 
wine S. cerevisiae strains, this is the first to investigate other species of the 
Saccharomyces genus. This research work investigates the expression of the genes involved 
in flavor compound production in three different Saccharomyces species (S. cerevisiae, S. 
bayanus var. uvarum and S. kudriavzevii) under low (12 degrees C) and moderate 
fermentation temperatures (28 degrees C). The global genes analysis showed that 30% of 
genes appeared to be differently expressed in the three cryophilic strains if compared  to
the reference strain (mesophilic S. cerevisiae), suggesting a very close cold  adaptation 
response. Remarkable differences in the gene expression level were observed when comparing
the three species, S. cerevisiae, S. bayanus var. uvarum  and S. kudriavzevii, which will 
result in different aroma profiles. Knowledge of  these differences in the transcriptome 
can be a tool to help modulate aroma to create wines with the desired aromatic traits.

PMID: 24854353

GEO ID: GSE30778.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE30778.final.pcl	Molecular analysis of the genes involved in aroma synthesis in the species S. cerevisiae, S. kudriavzevii and S. bayanus var. uvarum in winemaking conditions.	18	heat shock|fermentation	2

File last modified: Thu Apr 25 2024

---


# Gil_2014_PMID_24835131

Citation: Gil FN, Becker JD, Viegas CA. 2014. Potential mechanisms underlying response to effects of
the fungicide pyrimethanil from gene expression profiling in Saccharomyces cerevisiae.

Full Description: Pyrimethanil is a fungicide mostly applied in vineyards. When misused, residue levels 
detected in grape must or in the environment may be of concern. The present work aimed to 
analyze mechanisms underlying response to deleterious effects of pyrimethanil in the 
eukaryotic model Saccharomyces cerevisiae. Pyrimethanil concentration-dependent effects at
phenotypic (inhibition of growth) and transcriptomic levels were examined. For 
transcriptional profiling, analysis  focused on two sublethal exposure conditions that 
inhibited yeast growth by 20% or 50% compared with control cells not exposed to the 
fungicide. Gene expression  modifications increased with the magnitude of growth 
inhibition, in numbers and fold-change of differentially expressed genes and in diversity 
of over-represented functional categories. These included mostly biosynthesis of arginine 
and sulfur amino acids metabolism, as well as energy conservation, antioxidant response, 
and multidrug transport. Several pyrimethanil-responsive genes encoded proteins sharing 
significant homology with proteins from phytopathogenic fungi and ecologically relevant 
higher eukaryotes.

PMID: 24835131

GEO ID: GSE53126.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE53126.final.pcl	Potential mechanisms underlying response to effects of the fungicide pyrimethanil from gene expression profiling in Saccharomyces cerevisiae.	9	chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Gopinath_2014_PMID_25316598

Citation: Gopinath RK, You ST, Chien KY, Swamy KB, Yu JS, Schuyler SC, Leu JY. 2014. The Hsp90-
dependent proteome is conserved and enriched for hub proteins with high levels of protein-
protein connectivity.

Full Description: Hsp90 is one of the most abundant and conserved proteins in the cell. Reduced levels or 
activity of Hsp90 causes defects in many cellular processes and also reveals genetic and 
nongenetic variation within a population. Despite information about Hsp90 protein-protein 
interactions, a global view of the Hsp90-regulated proteome in yeast is unavailable. To 
investigate the degree of dependency of individual yeast proteins on Hsp90, we used the 
stable isotope labeling by amino acids in cell culture method coupled with mass 
spectrometry to quantify around 4,000 proteins in low-Hsp90 cells. We observed that 904 
proteins changed in their abundance by more than 1.5-fold. When compared with the 
transcriptome of the same population of cells, two-thirds of the misregulated proteins 
were observed to be  affected posttranscriptionally, of which the majority were 
downregulated. Further analyses indicated that the downregulated proteins are highly 
conserved and assume central roles in cellular networks with a high number of protein 
interacting partners, suggesting that Hsp90 buffers genetic and nongenetic variation 
through regulating protein network hubs. The downregulated proteins were enriched for 
essential proteins previously not known to be Hsp90-dependent.  Finally, we observed that 
downregulation of transcription factors and mating pathway components by attenuating Hsp90
function led to decreased target gene expression and pheromone response, respectively, 
providing a direct link between  observed proteome regulation and cellular phenotypes.

PMID: 25316598

GEO ID: GSE56186.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE56186.final.pcl	The Hsp90-dependent proteome is conserved and enriched for hub proteins with high levels of protein-protein connectivity.	6	heat shock	2

File last modified: Thu Apr 25 2024

---


# Hackmann_2014_PMID_24452287

Citation: Hackmann A, Wu H, Schneider UM, Meyer K, Jung K, Krebber H. 2014. Quality control of 
spliced mRNAs requires the shuttling SR proteins Gbp2 and Hrb1.

Full Description: Eukaryotic cells have to prevent the export of unspliced pre-mRNAs until intron removal is
completed to avoid the expression of aberrant and potentially harmful  proteins. Only 
mature mRNAs associate with the export receptor Mex67/TAP and enter the cytoplasm. Here we
show that the two shuttling serine/arginine (SR)-proteins Gbp2 and Hrb1 are key 
surveillance factors for the selective export of spliced mRNAs in yeast. Their absence 
leads to the significant leakage of unspliced pre-mRNAs into the cytoplasm. They bind to 
pre-mRNAs and the spliceosome during splicing, where they are necessary for the 
surveillance of splicing and the stable binding of the TRAMP complex to spliceosome-bound 
transcripts. Faulty transcripts are marked for their degradation at the nuclear exosome. 
On correct mRNAs the SR proteins recruit Mex67 upon completion of splicing to allow a 
quality controlled nuclear export. Altogether, these data identify a role for shuttling SR
proteins in mRNA surveillance and nuclear mRNA quality control.

PMID: 24452287

GEO ID: GSE52808.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE52808.final.pcl	Quality control of spliced mRNAs requires the shuttling SR proteins Gbp2 and Hrb1.	6	mRNA processing	2

File last modified: Thu Apr 25 2024

---


# Hasin_2014_PMID_24628813

Citation: Hasin N, Cusack SA, Ali SS, Fitzpatrick DA, Jones GW. 2014. Global transcript and 
phenotypic analysis of yeast cells expressing Ssa1, Ssa2, Ssa3 or Ssa4 as sole source of 
cytosolic Hsp70-Ssa chaperone activity.

Full Description: BACKGROUND: Cytosolic Hsp70 is a ubiquitous molecular chaperone that is involved  in 
responding to a variety of cellular stresses. A major function of Hsp70 is to  prevent the
aggregation of denatured proteins by binding to exposed hydrophobic regions and preventing
the accumulation of amorphous aggregates. To gain further  insight into the functional 
redundancy and specialisation of the highly homologous yeast Hsp70-Ssa family we expressed
each of the individual Ssa proteins as the sole source of Hsp70 in the cell and assessed 
phenotypic differences in prion propagation and stress resistance. Additionally we also 
analysed the global gene expression patterns in yeast strains expressing individual Ssa 
proteins, using microarray and RT-qPCR analysis. RESULTS: We confirm and extend previous 
studies demonstrating that cells expressing different Hsp70-Ssa isoforms vary in their 
ability to propagate the yeast [PSI+] prion, with Ssa3 being the most proficient. Of the 
four Ssa family members the heat inducible isoforms are more proficient in acquiring 
thermotolerance and we show a greater requirement than was previously thought, for 
cellular processes in addition to the traditional Hsp104 protein disaggregase machinery, 
in acquiring such thermotolerance. Cells expressing different Hsp70-Ssa isoforms also 
display  differences in phenotypic response to exposure to cell wall damaging and 
oxidative stress agents, again with the heat inducible isoforms providing better  
protection than constitutive isoforms. We assessed global transcriptome profiles  for 
cells expressing individual Hsp70-Ssa isoforms as the sole source of cytosolic Hsp70, and 
identified a significant difference in cellular gene expression between these strains. 
Differences in gene expression profiles provide a rationale for some phenotypic 
differences we observed in this study. We also demonstrate a high degree of correlation 
between microarray data and RT-qPCR analysis for a selection of genes. CONCLUSIONS: The 
Hsp70-Ssa family provide both redundant and variant-specific functions within the yeast 
cell. Yeast cells expressing individual members of the Hsp70-Ssa family as the sole source
of Ssa protein display differences in global gene expression profiles. These changes in  
global gene expression may contribute significantly to the phenotypic differences observed
between the Hsp70-Ssa family members.

PMID: 24628813

GEO ID: GSE32433.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE32433.final.pcl	Global transcript and phenotypic analysis of yeast cells expressing Ssa1, Ssa2, Ssa3 or Ssa4 as sole source of cytosolic Hsp70-Ssa chaperone activity.	4	heat shock|response to unfolded protein|stress	1

File last modified: Thu Apr 25 2024

---


# Hasunuma_2014_PMID_23916856

Citation: Hasunuma T, Ismail KS, Nambu Y, Kondo A. 2014. Co-expression of TAL1 and ADH1 in 
recombinant xylose-fermenting Saccharomyces cerevisiae improves ethanol production from 
lignocellulosic hydrolysates in the presence of furfural.

Full Description: Lignocellulosic biomass dedicated to bioethanol production usually contains pentoses and 
inhibitory compounds such as furfural that are not well tolerated by Saccharomyces 
cerevisiae. Thus, S. cerevisiae strains with the capability of utilizing both glucose and 
xylose in the presence of inhibitors such as furfural  are very important in industrial 
ethanol production. Under the synergistic conditions of transaldolase (TAL) and alcohol 
dehydrogenase (ADH) overexpression, S. cerevisiae MT8-1X/TAL-ADH was able to produce 1.3-
fold and 2.3-fold more ethanol in the presence of 70 mM furfural than a TAL-expressing 
strain and a control strain, respectively. We also tested the strains' ability by 
mimicking industrial ethanol production from hemicellulosic hydrolysate containing 
fermentation inhibitors, and ethanol production was further improved by 16% when  using 
MT8-1X/TAL-ADH compared to the control strain. Transcript analysis further  revealed that 
besides the pentose phosphate pathway genes TKL1 and TAL1, ADH7 was also upregulated in 
response to furfural stress, which resulted in higher ethanol production compared to the 
TAL-expressing strain. The improved capability of our  modified strain was based on its 
capacity to more quickly reduce furfural in situ resulting in higher ethanol production. 
The co-expression of TAL/ADH genes is one crucial strategy to fully utilize undetoxified 
lignocellulosic hydrolysate, leading to cost-competitive ethanol production.

PMID: 23916856

GEO ID: GSE45273.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE45273.remapped.final.pcl	Co-expression of TAL1 and ADH1 in recombinant xylose-fermenting Saccharomyces cerevisiae improves ethanol production from lignocellulosic hydrolysates in the presence of furfural.	12	chemical stimulus	2

File last modified: Thu Apr 25 2024

---


# Hebly_2014_PMID_24814792

Citation: Hebly M, de Ridder D, de Hulster EA, de la Torre Cortes P, Pronk JT, Daran-Lapujade P. 
2014. Physiological and transcriptional responses of anaerobic chemostat cultures of 
Saccharomyces cerevisiae subjected to diurnal temperature cycles.

Full Description: Diurnal temperature cycling is an intrinsic characteristic of many exposed microbial 
ecosystems. However, its influence on yeast physiology and the yeast transcriptome has not
been studied in detail. In this study, 24-h sinusoidal temperature cycles, oscillating 
between 12 degrees C and 30 degrees C, were imposed on anaerobic, glucose-limited 
chemostat cultures of Saccharomyces cerevisiae. After three diurnal temperature cycles (
DTC), concentrations of glucose and extracellular metabolites as well as CO2 production 
rates showed regular, reproducible circadian rhythms. DTC also led to waves of 
transcriptional activation and repression, which involved one-sixth of the yeast genome. A
substantial fraction of these DTC-responsive genes appeared to respond primarily  to 
changes in the glucose concentration. Elimination of known glucose-responsive  genes 
revealed an overrepresentation of previously identified temperature-responsive genes as 
well as genes involved in the cell cycle and de novo purine biosynthesis. In-depth 
analysis demonstrated that DTC led to a partial synchronization of the cell cycle of the 
yeast populations in chemostat cultures, which was lost upon release from DTC. Comparison 
of DTC results with data from steady-state cultures showed that the 24-h DTC was 
sufficiently slow to allow S. cerevisiae chemostat cultures to acclimate their 
transcriptome and physiology at the DTC temperature maximum and to approach acclimation at
the DTC  temperature minimum. Furthermore, this comparison and literature data on growth 
rate-dependent cell cycle phase distribution indicated that cell cycle synchronization was
most likely an effect of imposed fluctuations of the relative growth rate (mu/mumax) 
rather than a direct effect of temperature.

PMID: 24814792

GEO ID: GSE55372.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE55372.remapped.final.pcl	Physiological and transcriptional responses of anaerobic chemostat cultures of Saccharomyces cerevisiae subjected to diurnal temperature cycles.	17	heat shock|stress	1

File last modified: Thu Apr 25 2024

---


# Herrera-Moyano_2014_PMID_24636987

Citation: Herrera-Moyano E, Mergui X, Garcia-Rubio ML, Barroso S, Aguilera A. 2014. The yeast and 
human FACT chromatin-reorganizing complexes solve R-loop-mediated transcription-
replication conflicts.

Full Description: FACT (facilitates chromatin transcription) is a chromatin-reorganizing complex that swaps 
nucleosomes around the RNA polymerase during transcription elongation  and has a role in 
replication that is not fully understood yet. Here we show that recombination factors are 
required for the survival of yeast FACT mutants, consistent with an accumulation of DNA 
breaks that we detected by Rad52 foci and  transcription-dependent hyperrecombination. 
Breaks also accumulate in FACT-depleted human cells, as shown by gammaH2AX foci and single
-cell electrophoresis. Furthermore, FACT-deficient yeast and human cells show replication 
impairment, which in yeast we demonstrate by ChIP-chip (chromatin immunoprecipitation [
ChIP] coupled with microarray analysis) of Rrm3 to occur genome-wide but preferentially at
highly transcribed regions. Strikingly, in yeast FACT mutants, high levels of Rad52 foci 
are suppressed by RNH1 overexpression; R loops accumulate at high levels, and replication 
becomes normal when global RNA synthesis is inhibited in FACT-depleted human cells. The 
results  demonstrate a key function of FACT in the resolution of R-loop-mediated 
transcription-replication conflicts, likely associated with a specific chromatin  
organization.

PMID: 24636987

GEO ID: GSE54340.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE54340.remapped.final.pcl	The yeast and human FACT chromatin-reorganizing complexes solve R-loop-mediated transcription-replication conflicts.	9	chromatin organization	1

File last modified: Thu Apr 25 2024

---


# Hou_2014_PMID_24237754

Citation: Hou J, Tang H, Liu Z, Osterlund T, Nielsen J, Petranovic D. 2014. Management of the 
endoplasmic reticulum stress by activation of the heat shock response in yeast.

Full Description: In yeast Saccharomyces cerevisiae, accumulation of misfolded proteins in the endoplasmic 
reticulum (ER) causes ER stress and activates the unfolded protein response (UPR), which 
is mediated by Hac1p. The heat shock response (HSR) mediated by Hsf1p, mainly regulates 
cytosolic processes and protects the cell from stresses. Here, we find that a constitutive
activation of the HSR could increase ER stress resistance in both wild-type and UPR-
deficient cells. Activation of HSR decreased UPR activation in the WT (as shown by the 
decreased HAC1 mRNA splicing). We analyzed the genome-wide transcriptional response in 
order to propose regulatory mechanisms that govern the interplay between UPR and  HSR and 
followed up for the hypotheses by experiments in vivo and in vitro. Interestingly, we 
found that the regulation of ER stress response via HSR is (1)  only partially dependent 
on over-expression of Kar2p (ER resident chaperone induced by ER stress); (2) does not 
involve the increase in protein turnover via  the proteasome activity; (3) is related to 
the oxidative stress response. From the transcription data, we also propose that HSR 
enhances ER stress resistance mainly through facilitation of protein folding and secretion
. We also find that HSR coordinates multiple stress-response pathways, including the 
repression of the overall transcription and translation.

PMID: 24237754

GEO ID: GSE39311.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE39311.remapped.final.pcl	Management of the endoplasmic reticulum stress by activation of the heat shock response in yeast.	12	ubiquitin or ULP modification	1

File last modified: Thu Apr 25 2024

---


# Hughes_Hallett_2014_PMID_25085507

Citation: Hughes Hallett JE, Luo X, Capaldi AP. 2014. State transitions in the TORC1 signaling 
pathway and information processing in Saccharomyces cerevisiae.

Full Description: TOR kinase complex I (TORC1) is a key regulator of cell growth and metabolism in  all 
eukaryotes. Previous studies in yeast have shown that three GTPases-Gtr1, Gtr2, and Rho1-
bind to TORC1 in nitrogen and amino acid starvation conditions to  block phosphorylation 
of the S6 kinase Sch9 and activate protein phosphatase 2A (PP2A). This leads to 
downregulation of 450 Sch9-dependent protein and ribosome synthesis genes and upregulation
of 100 PP2A-dependent nitrogen assimilation and  amino acid synthesis genes. Here, using 
bandshift assays and microarray measurements, we show that the TORC1 pathway also 
populates three other stress/starvation states. First, in glucose starvation conditions, 
the AMP-activated protein kinase (AMPK/Snf1) and at least one other factor push the TORC1 
pathway into an off state, in which Sch9-branch signaling and PP2A-branch signaling are 
both inhibited. Remarkably, the TORC1 pathway remains in the glucose starvation (PP2A 
inhibited) state even when cells are simultaneously starved for nitrogen and glucose. 
Second, in osmotic stress, the MAPK Hog1/p38 drives the TORC1 pathway into a different 
state, in which Sch9 signaling and PP2A-branch signaling are inhibited, but PP2A-branch 
signaling can still be activated by nitrogen starvation. Third, in oxidative stress and 
heat stress, TORC1-Sch9 signaling is blocked while weak PP2A-branch signaling occurs. 
Together, our data show that the TORC1 pathway acts as an information-processing  hub, 
activating different genes in different conditions to ensure that available  energy is 
allocated to drive growth, amino acid synthesis, or a stress response,  depending on the 
needs of the cell.

PMID: 25085507

GEO ID: GSE58992.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE58992.final.pcl	State transitions in the TORC1 signaling pathway and information processing in Saccharomyces cerevisiae.	16	starvation|oxidative stress|osmotic stress|heat shock|signaling	2

File last modified: Thu Apr 25 2024

---


# Johnson_2014_PMID_24830393

Citation: Johnson JE, Johnson FB. 2014. Methionine restriction activates the retrograde response and
confers both stress  tolerance and lifespan extension to yeast, mouse and human cells.

Full Description: A methionine-restricted diet robustly improves healthspan in key model organisms. For 
example, methionine restriction reduces age-related pathologies and extends lifespan up to
45% in rodents. However, the mechanisms underlying these benefits  remain largely unknown
. We tested whether the yeast chronological aging assay could model the benefits of 
methionine restriction, and found that this intervention extends lifespan when enforced by
either dietary or genetic approaches, and furthermore, that the observed lifespan 
extension is due primarily to reduced acid accumulation. In addition, methionine 
restriction-induced lifespan extension requires the activity of the retrograde response, 
which regulates nuclear gene expression in response to changes in mitochondrial function. 
Consistent with an involvement of stress-responsive retrograde signaling, we also found 
that methionine-restricted yeast are more stress tolerant than control cells. Prompted by 
these findings in yeast, we tested the effects of genetic methionine restriction on the 
stress tolerance and  replicative lifespans of cultured mouse and human fibroblasts. We 
found that such methionine-restricted mammalian cells are resistant to numerous cytotoxic 
stresses, and are substantially longer-lived than control cells. In addition, similar to 
yeast, the extended lifespan of methionine-restricted mammalian cells  is associated with 
NFkappaB-mediated retrograde signaling. Overall, our data suggest that improved stress 
tolerance and extension of replicative lifespan may  contribute to the improved healthspan
observed in methionine-restricted rodents,  and also support the possibility that 
manipulation of the pathways engaged by methionine restriction may improve healthspan in 
humans.

PMID: 24830393

GEO ID: GSE56641.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE56641.final.pcl	Methionine restriction activates the retrograde response and confers both stress  tolerance and lifespan extension to yeast, mouse and human cells.	3	amino acid utilization|signaling|cell aging	1

File last modified: Thu Apr 25 2024

---


# Kamei_2014_PMID_25294875

Citation: Kamei Y, Tamada Y, Nakayama Y, Fukusaki E, Mukai Y. 2014. Changes in transcription and 
metabolism during the early stage of replicative cellular senescence in budding yeast.

Full Description: Age-related damage accumulates and a variety of biological activities and functions 
deteriorate in senescent cells. However, little is known about when cellular aging 
behaviors begin and what cellular aging processes change. Previous research demonstrated 
age-related mRNA changes in budding yeast by the 18th to 20th generation, which is the 
average replicative lifespan of yeast (i.e. about half of the population is dead by this 
time point). Here, we performed transcriptional and metabolic profiling for yeast at early
stages of senescence (4th, 7th, and 11th generation), that is, for populations in which 
most cells are still alive. Transcriptional profiles showed up- and down-regulation for 
approximately 20% of the genes profiled after the first four generations, few further 
changes by the 7th generation, and an additional 12% of the genes were up- and down-
regulated after 11 generations. Pathway analysis revealed that these 11th generation cells
had accumulated transcripts coding for enzymes involved in  sugar metabolism, the TCA 
cycle, and amino acid degradation and showed decreased  levels of mRNAs coding for enzymes
involved in amino acid biosynthetic pathways.  These observations were consistent with the
metabolomic profiles of aging cells:  an accumulation of pyruvic acid and TCA cycle 
intermediates and depletion of most amino acids, especially branched-chain amino acids. 
Stationary phase-induced genes were highly expressed after 11 generations even though the 
growth medium contained adequate levels of nutrients, indicating deterioration of the 
nutrient  sensing and/or signaling pathways by the 11th generation. These changes are 
presumably early indications of replicative senescence.

PMID: 25294875

GEO ID: GSE59797.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE59797.final.pcl	Changes in transcription and metabolism during the early stage of replicative cellular senescence in budding yeast.	3	cell aging	2

File last modified: Thu Apr 25 2024

---


# Kang_2014_PMID_24206186

Citation: Kang HJ, Jeong SJ, Kim KN, Baek IJ, Chang M, Kang CM, Park YS, Yun CW. 2014. A novel 
protein, Pho92, has a conserved YTH domain and regulates phosphate metabolism by 
decreasing the mRNA stability of PHO4 in Saccharomyces cerevisiae.

Full Description: The homologue of human YTHDF2, Ydr374c (Pho92), is the only protein that has a YTH (YT521-
B homology) domain in Saccharomyces cerevisiae. Based on microarray analysis, genes 
involved in the phosphate signal transduction (PHO) pathway were  up-regulated in the 
Deltapho92 strain, as were genes regulated by Pho4, which is  an important transcription 
factor in the PHO pathway. To identify the exact mechanism of Pho92 action with respect to
phosphate metabolism, we investigated the effect of Pho92 on PHO4 expression. The half-
life of PHO4 mRNA was increased  in the Deltapho92 strain; this phenotype was also 
observed in the deletion mutants UPF1 and POP2, which are components of the NMD (nonsense-
mediated decay)  pathway and the Pop2-Ccr4-Not deadenylase complex respectively. Pho92 
interacts physically with Pop2 of the Pop2-Ccr4-Not deadenylase complex. Furthermore, 
Pho92 binding to the 3'-UTR of PHO4 was dependent on the phosphate concentration. Deletion
of the PHO4 3'-UTR resulted in PHO4 mRNA resistance to Pho92-dependent degradation. The 
results of the present study indicate that Pho92 regulates Pho4  expression at the post-
transcriptional level via the regulation of mRNA stability. Taken together, Pho92 
participates in cellular phosphate metabolism, specifically via the regulation of PHO4 
mRNA stability by binding to the 3'-UTR in a phosphate-dependent manner.

PMID: 24206186

GEO ID: GSE50086.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE50086.final.pcl	A novel protein, Pho92, has a conserved YTH domain and regulates phosphate metabolism by decreasing the mRNA stability of PHO4 in Saccharomyces cerevisiae.	2	phosphorus utilization	2

File last modified: Thu Apr 25 2024

---


# Kemmeren_2014_PMID_24766815

Citation: Kemmeren P, Sameith K, van de Pasch LA, Benschop JJ, Lenstra TL, Margaritis T, O'Duibhir E,
Apweiler E, van Wageningen S, Ko CW, van Heesch S, Kashani MM, Ampatziadis-Michailidis G, 
Brok MO, Brabers NA, Miles AJ, Bouwmeester D, van Hooff SR, van Bakel H, Sluiters E, 
Bakker LV, Snel B, Lijnzaad P, van Leenen D, Groot Koerkamp MJ, Holstege FC. 2014. Large-
scale genetic perturbations reveal regulatory networks and an abundance of  gene-specific 
repressors.

Full Description: To understand regulatory systems, it would be useful to uniformly determine how different 
components contribute to the expression of all other genes. We therefore monitored mRNA 
expression genome-wide, for individual deletions of one-quarter of yeast genes, focusing 
on (putative) regulators. The resulting genetic perturbation signatures reflect many 
different properties. These include  the architecture of protein complexes and pathways, 
identification of expression  changes compatible with viability, and the varying 
responsiveness to genetic perturbation. The data are assembled into a genetic perturbation
network that shows different connectivities for different classes of regulators. Four feed
-forward loop (FFL) types are overrepresented, including incoherent type 2 FFLs that 
likely represent feedback. Systematic transcription factor classification shows a 
surprisingly high abundance of gene-specific repressors, suggesting that yeast chromatin 
is not as generally restrictive to transcription  as is often assumed. The data set is 
useful for studying individual genes and for discovering properties of an entire 
regulatory system.

PMID: 24766815

GEO ID: GSE42241.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE42241.final.pcl	Large-scale genetic perturbations reveal regulatory networks and an abundance of  gene-specific repressors; yeast wt-matA pool - plate reader	20	genome variation	2

File last modified: Thu Apr 25 2024
GSE45115_s244.final.pcl	Large-scale genetic perturbations reveal regulatory networks and an abundance of  gene-specific repressors; Expression profiling of 376 wildtypes to assess day-to-day variance	244	genome variation	2

File last modified: Thu Apr 25 2024
GSE42240.final.pcl	Large-scale genetic perturbations reveal regulatory networks and an abundance of  gene-specific repressors; yeast wt-matA pool	8	genome variation	2

File last modified: Thu Apr 25 2024
GSE42217.final.pcl	Large-scale genetic perturbations reveal regulatory networks and an abundance of  gene-specific repressors; yeast wt pool - plate reader	200	genome variation	2

File last modified: Thu Apr 25 2024
GSE42215.final.pcl	Large-scale genetic perturbations reveal regulatory networks and an abundance of  gene-specific repressors; yeast wt pool	200	genome variation	2

File last modified: Thu Apr 25 2024

---


# Kozak_2014_PMID_24269999

Citation: Kozak BU, van Rossum HM, Benjamin KR, Wu L, Daran JM, Pronk JT, van Maris AJ. 2014. 
Replacement of the Saccharomyces cerevisiae acetyl-CoA synthetases by alternative pathways
for cytosolic acetyl-CoA synthesis.

Full Description: Cytosolic acetyl-coenzyme A is a precursor for many biotechnologically relevant compounds 
produced by Saccharomyces cerevisiae. In this yeast, cytosolic acetyl-CoA synthesis and 
growth strictly depend on expression of either the Acs1  or Acs2 isoenzyme of acetyl-CoA 
synthetase (ACS). Since hydrolysis of ATP to AMP  and pyrophosphate in the ACS reaction 
constrains maximum yields of acetyl-CoA-derived products, this study explores replacement 
of ACS by two ATP-independent pathways for acetyl-CoA synthesis. After evaluating 
expression of different bacterial genes encoding acetylating acetaldehyde dehydrogenase (A
-ALD) and pyruvate-formate lyase (PFL), acs1Delta acs2Delta S. cerevisiae strains were  
constructed in which A-ALD or PFL successfully replaced ACS. In A-ALD-dependent strains, 
aerobic growth rates of up to 0.27 h(-1) were observed, while anaerobic  growth rates of 
PFL-dependent S. cerevisiae (0.20 h(-1)) were stoichiometrically  coupled to formate 
production. In glucose-limited chemostat cultures, intracellular metabolite analysis did 
not reveal major differences between A-ALD-dependent and reference strains. However, 
biomass yields on glucose of A-ALD- and PFL-dependent strains were lower than those of the
reference strain. Transcriptome analysis suggested that reduced biomass yields were caused
by acetaldehyde and formate in A-ALD- and PFL-dependent strains, respectively. Transcript 
profiles also indicated that a previously proposed role of Acs2 in histone acetylation is 
probably linked to cytosolic acetyl-CoA levels rather than to direct involvement of Acs2 
in histone acetylation. While demonstrating that yeast ACS can be fully replaced, this 
study demonstrates that further modifications are needed to achieve optimal in vivo 
performance of the alternative reactions for supply of cytosolic acetyl-CoA as a product 
precursor.

PMID: 24269999

GEO ID: GSE47983.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE47983.final.pcl	Replacement of the Saccharomyces cerevisiae acetyl-CoA synthetases by alternative pathways for cytosolic acetyl-CoA synthesis.	12	diauxic shift	1

File last modified: Thu Apr 25 2024

---


# Landry_2014_PMID_24714560

Citation: Landry BD, Mapa CE, Arsenault HE, Poti KE, Benanti JA. 2014. Regulation of a transcription
factor network by Cdk1 coordinates late cell cycle  gene expression.

Full Description: To maintain genome stability, regulators of chromosome segregation must be expressed in 
coordination with mitotic events. Expression of these late cell cycle genes is regulated 
by cyclin-dependent kinase (Cdk1), which phosphorylates  a network of conserved 
transcription factors (TFs). However, the effects of Cdk1  phosphorylation on many key TFs
are not known. We find that elimination of Cdk1-mediated phosphorylation of four S-phase 
TFs decreases expression of many late cell cycle genes, delays mitotic progression, and 
reduces fitness in budding yeast. Blocking phosphorylation impairs degradation of all four
TFs. Consequently, phosphorylation-deficient mutants of the repressors Yox1 and Yhp1 
exhibit increased promoter occupancy and decreased expression of their target genes. 
Interestingly, although phosphorylation of the transcriptional activator Hcm1 on its N-
terminus promotes its degradation, phosphorylation on its C-terminus is required for its 
activity, indicating that Cdk1 both activates and  inhibits a single TF. We conclude that 
Cdk1 promotes gene expression by both activating transcriptional activators and 
inactivating transcriptional repressors. Furthermore, our data suggest that coordinated 
regulation of the TF network by Cdk1 is necessary for faithful cell division.

PMID: 24714560

GEO ID: GSE55121.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE55121.remapped.final.pcl	Regulation of a transcription factor network by Cdk1 coordinates late cell cycle  gene expression.	32	cell cycle regulation|mitotic cell cycle	2

File last modified: Thu Apr 25 2024

---


# Lewis_2014_PMID_24970865

Citation: Lewis JA, Broman AT, Will J, Gasch AP. 2014. Genetic architecture of ethanol-responsive 
transcriptome variation in Saccharomyces cerevisiae strains.

Full Description: Natural variation in gene expression is pervasive within and between species, and it 
likely explains a significant fraction of phenotypic variation between individuals. 
Phenotypic variation in acute systemic responses can also be leveraged to reveal 
physiological differences in how individuals perceive and respond to environmental 
perturbations. We previously found extensive variation in the transcriptomic response to 
acute ethanol exposure in two wild isolates and a common laboratory strain of 
Saccharomyces cerevisiae. Many expression differences persisted across several modules of 
coregulated genes, implicating trans-acting systemic differences in ethanol sensing and/or
response. Here, we conducted expression QTL mapping of the ethanol response in two strain 
crosses to identify the genetic basis for these differences. To understand systemic 
differences, we focused on hotspot loci that affect many transcripts in trans.  Candidate 
causal regulators contained within hotspots implicate upstream regulators as well as 
downstream effectors of the ethanol response. Overlap in hotspot targets revealed additive
genetic effects of trans-acting loci as well as epi-hotspots, in which epistatic 
interactions between two loci affected the same suites of downstream targets. One epi-
hotspot implicated interactions between Mkt1p and proteins linked to translational 
regulation, prompting us to show that Mkt1p localizes to P bodies upon ethanol stress in a
strain-specific manner. Our results provide a glimpse into the genetic architecture 
underlying natural variation in a stress response and present new details on how yeast 
respond to ethanol stress.

PMID: 24970865

GEO ID: GSE54196.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE54196.remapped.final.pcl	Genetic architecture of ethanol-responsive transcriptome variation in Saccharomyces cerevisiae strains.	196	stress	2

File last modified: Thu Apr 25 2024

---


# Licursi_2014_PMID_24164706

Citation: Licursi V, Salvi C, De Cesare V, Rinaldi T, Mattei B, Fabbri C, Serino G, Bramasole L, 
Zimbler JZ, Pick E, Barnes BM, Bard M, Negri R. 2014. The COP9 signalosome is involved in 
the regulation of lipid metabolism and of transition metals uptake in Saccharomyces 
cerevisiae.

Full Description: The COP9 signalosome (CSN) is a highly conserved eukaryotic protein complex which 
regulates the cullin RING family of ubiquitin ligases and carries out a deneddylase 
activity that resides in subunit 5 (CSN5). Whereas CSN activity is essential for the 
development of higher eukaryotes, several unicellular fungi including the budding yeast 
Saccharomyces cerevisiae can survive without a functional CSN. Nevertheless, the budding 
yeast CSN is biochemically active and deletion mutants of each of its subunits exhibit 
deficiency in cullins deneddylation, although the biological context of this activity is 
still unknown  in this organism. To further characterize CSN function in budding yeast, we
present here a transcriptomic and proteomic analysis of a S. cerevisiae strain deleted in 
the CSN5/RRI1 gene (hereafter referred to as CSN5), coding for the only canonical subunit 
of the complex. We show that Csn5 is involved in modulation of the genes controlling amino
acid and lipid metabolism and especially ergosterol biosynthesis. These alterations in 
gene expression correlate with the lower ergosterol levels and increased intracellular 
zinc content which we observed in csn5 null mutant cells. We show that some of these 
regulatory effects of Csn5, in particular the control of isoprenoid biosynthesis, are 
conserved through evolution, since similar transcriptomic and/or proteomic effects of csn5
mutation were previously observed in other eukaryotic organisms such as Aspergillus 
nidulans, Arabidopsis thaliana and Drosophila melanogaster. Our results suggest that the 
diverged budding yeast CSN is more conserved than was previously thought.

PMID: 24164706

GEO ID: GSE51563.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE51563.remapped.final.pcl	The COP9 signalosome is involved in the regulation of lipid metabolism and of transition metals uptake in Saccharomyces cerevisiae.	2	ubiquitin or ULP modification	2

File last modified: Thu Apr 25 2024

---


# Lindgren_2014_PMID_25483075

Citation: Lindgren E, Hagg S, Giordano F, Bjorkegren J, Strom L. 2014. Inactivation of the budding 
yeast cohesin loader Scc2 alters gene expression both globally and in response to a single
DNA double strand break.

Full Description: Genome integrity is fundamental for cell survival and cell cycle progression. Important 
mechanisms for keeping the genome intact are proper sister chromatid segregation, correct 
gene regulation and efficient repair of damaged DNA. Cohesin and its DNA loader, the Scc2/
4 complex have been implicated in all these cellular actions. The gene regulation role has
been described in several organisms. In yeast it has been suggested that the proteins in 
the cohesin network would effect transcription based on its role as insulator. More 
recently, data are emerging indicating direct roles for gene regulation also in yeast. 
Here we extend these studies by investigating whether the cohesin loader Scc2 is involved 
in regulation of gene expression. We performed global gene expression profiling in the 
absence and presence of DNA damage, in wild type and Scc2 deficient G2/M arrested cells, 
when it is known that Scc2 is important for DNA double strand break repair and formation 
of damage induced cohesion. We found that not only the DNA damage specific transcriptional
response is distorted after inactivation of Scc2 but also the overall transcription 
profile. Interestingly, these alterations did not correlate with changes in cohesin 
binding.

PMID: 25483075

GEO ID: GSE61530.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE61530.final.pcl	Inactivation of the budding yeast cohesin loader Scc2 alters gene expression both globally and in response to a single DNA double strand break.	27	DNA damage stimulus|DNA replication, recombination and repair	1

File last modified: Thu Apr 25 2024

---


# Lopez-Serra_2014_PMID_25173104

Citation: Lopez-Serra L, Kelly G, Patel H, Stewart A, Uhlmann F. 2014. The Scc2-Scc4 complex acts in
sister chromatid cohesion and transcriptional regulation by maintaining nucleosome-free 
regions.

Full Description: The cohesin complex is at the heart of many chromosomal activities, including sister 
chromatid cohesion and transcriptional regulation. Cohesin loading onto chromosomes 
depends on the Scc2-Scc4 cohesin loader complex, but the chromatin features that form 
cohesin loading sites remain poorly understood. Here we show that the RSC chromatin 
remodeling complex recruits budding yeast Scc2-Scc4 to broad nucleosome-free regions, 
which the cohesin loader helps to maintain. Consequently, inactivation of either the 
cohesin loader or the RSC complex has similar effects on nucleosome positioning, gene 
expression and sister chromatid cohesion. These results show an intimate link between 
local chromatin structure and higher-order chromosome architecture. Our findings pertain 
to the similarities between two severe human disorders, Cornelia de Lange syndrome, which 
is caused by alterations in the human cohesin loader, and Coffin-Siris syndrome, which 
results from alterations in human RSC complex components. Both syndromes can arise from 
gene misregulation due to related changes in the nucleosome landscape.

PMID: 25173104

GEO ID: GSE56991.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE56991.final.pcl	The Scc2-Scc4 complex acts in sister chromatid cohesion and transcriptional regulation by maintaining nucleosome-free regions.	7	chromatin organization	1

File last modified: Thu Apr 25 2024

---


# Lu_2014_PMID_24631914

Citation: Lu S, Lee KK, Harris B, Xiong B, Bose T, Saraf A, Hattem G, Florens L, Seidel C, Gerton JL,
 2014. The cohesin acetyltransferase Eco1 coordinates rDNA replication and transcription.

Full Description: Eco1 is the acetyltransferase that establishes sister-chromatid cohesion during DNA 
replication. A budding yeast strain with an eco1 mutation that genocopies Roberts syndrome
has reduced ribosomal DNA (rDNA) transcription and a transcriptional signature of 
starvation. We show that deleting FOB1--a gene that  encodes a replication fork-blocking 
protein specific for the rDNA region--rescues rRNA production and partially rescues 
transcription genome-wide. Further studies  show that deletion of FOB1 corrects the genome
-wide replication defects, nucleolar structure, and rDNA segregation that occur in the 
eco1 mutant. Our study highlights that the presence of cohesin at the rDNA locus has a 
central role in controlling global DNA replication and gene expression.

PMID: 24631914

GEO ID: GSE54741.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE54741.remapped.final.pcl	The cohesin acetyltransferase Eco1 coordinates rDNA replication and transcription.	27	mitotic cell cycle	1

File last modified: Thu Apr 25 2024

---


# Macadangdang_2014_PMID_24939988

Citation: Macadangdang BR, Oberai A, Spektor T, Campos OA, Sheng F, Carey MF, Vogelauer M, 
Kurdistani SK. 2014. Evolution of histone 2A for chromatin compaction in eukaryotes.

Full Description: During eukaryotic evolution, genome size has increased disproportionately to nuclear 
volume, necessitating greater degrees of chromatin compaction in higher eukaryotes, which 
have evolved several mechanisms for genome compaction. However, it is unknown whether 
histones themselves have evolved to regulate chromatin compaction. Analysis of histone 
sequences from 160 eukaryotes revealed that the H2A N-terminus has systematically acquired
arginines as genomes expanded. Insertion of arginines into their evolutionarily conserved 
position in H2A of a small-genome organism increased linear compaction by as much as 40%, 
while their  absence markedly diminished compaction in cells with large genomes. This 
effect was recapitulated in vitro with nucleosomal arrays using unmodified histones, 
indicating that the H2A N-terminus directly modulates the chromatin fiber likely  through 
intra- and inter-nucleosomal arginine-DNA contacts to enable tighter nucleosomal packing. 
Our findings reveal a novel evolutionary mechanism for regulation of chromatin compaction 
and may explain the frequent mutations of the  H2A N-terminus in cancer.

PMID: 24939988

GEO ID: GSE50440.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE50440.remapped.final.pcl	Evolution of histone 2A for chromatin compaction in eukaryotes.	11	chromatin organization	1

File last modified: Thu Apr 25 2024

---


# Merrill_2014_PMID_24938287

Citation: Merrill AE, Hebert AS, MacGilvray ME, Rose CM, Bailey DJ, Bradley JC, Wood WW, El Masri M,
Westphall MS, Gasch AP, Coon JJ. 2014. NeuCode labels for relative protein quantification.

Full Description: We describe a synthesis strategy for the preparation of lysine isotopologues that differ 
in mass by as little as 6 mDa. We demonstrate that incorporation of these  molecules into 
the proteomes of actively growing cells does not affect cellular proliferation, and we 
discuss how to use the embedded mass signatures (neutron encoding (NeuCode)) for 
multiplexed proteome quantification by means of high-resolution mass spectrometry. NeuCode
SILAC amalgamates the quantitative accuracy of SILAC with the multiplexing of isobaric 
tags and, in doing so, offers up new opportunities for biological investigation. We 
applied NeuCode SILAC to examine the relationship between transcript and protein levels in
yeast cells responding to environmental stress. Finally, we monitored the time-resolved 
responses of five signaling mutants in a single 18-plex experiment.

PMID: 24938287

GEO ID: GSE53549.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE53549.final.pcl	NeuCode labels for relative protein quantification.	2	osmotic stress	2

File last modified: Thu Apr 25 2024

---


# Miller-Fleming_2014_PMID_24706893

Citation: Miller-Fleming L, Antas P, Pais TF, Smalley JL, Giorgini F, Outeiro TF. 2014. Yeast DJ-1 
superfamily members are required for diauxic-shift reprogramming and cell survival in 
stationary phase.

Full Description: The yeast Hsp31 minifamily proteins (Hsp31, Hsp32, Hsp33, Hsp34) belong to the highly 
conserved DJ-1 superfamily. The human DJ-1 protein is associated with cancer and 
neurodegenerative disorders, such as Parkinson disease. However, the precise function of 
human and yeast DJ-1 proteins is unclear. Here we show that the yeast DJ-1 homologs have a
role in diauxic-shift (DS), characterized by metabolic reprogramming because of glucose 
limitation. We find that the Hsp31 genes are strongly induced in DS and in stationary 
phase (SP), and that deletion  of these genes reduces chronological lifespan, impairs 
transcriptional reprogramming at DS, and impairs the acquisition of several typical 
characteristics of SP, including autophagy induction. In addition, under carbon starvation
, the HSP31 family gene-deletion strains display impaired autophagy, disrupted target of 
rapamycin complex 1 (TORC1) localization to P-bodies, and caused abnormal TORC1-mediated 
Atg13 phosphorylation. Repression of TORC1 by rapamycin in the gene-deletion strains 
completely reversed their sensitivity to heat shock. Taken together, our data indicate 
that Hsp31 minifamily is required for DS reprogramming and cell survival in SP, and plays 
a role upstream of TORC1. The enhanced understanding of the cellular function of these 
genes sheds light into the biological role of other members of the superfamily, including 
DJ-1, which is an attractive target for therapeutic intervention in cancer and in 
Parkinson disease.

PMID: 24706893

GEO ID: GSE56124.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE56124.remapped.final.pcl	Yeast DJ-1 superfamily members are required for diauxic-shift reprogramming and cell survival in stationary phase.	12	diauxic shift	1

File last modified: Thu Apr 25 2024

---


# Mojzita_2014_PMID_25192596

Citation: Mojzita D, Oja M, Rintala E, Wiebe M, Penttila M, Ruohonen L. 2014. Transcriptome of 
Saccharomyces cerevisiae during production of D-xylonate.

Full Description: BACKGROUND: Production of D-xylonate by the yeast S. cerevisiae provides an example of 
bioprocess development for sustainable production of value-added chemicals from cheap raw 
materials or side streams. Production of D-xylonate may  lead to considerable 
intracellular accumulation of D-xylonate and to loss of viability during the production 
process. In order to understand the physiological responses associated with D-xylonate 
production, we performed transcriptome analyses during D-xylonate production by a robust 
recombinant strain of S. cerevisiae which produces up to 50 g/L D-xylonate. RESULTS: 
Comparison of the transcriptomes of the D-xylonate producing and the control strain showed
considerably higher expression of the genes controlled by the cell wall integrity (CWI) 
pathway and of some genes previously identified as up-regulated in response to other 
organic acids in the D-xylonate producing strain. Increased phosphorylation of Slt2 kinase
in the D-xylonate producing strain also indicated  that D-xylonate production caused 
stress to the cell wall. Surprisingly, genes encoding proteins involved in translation, 
ribosome structure and RNA metabolism, processes which are commonly down-regulated under 
conditions causing cellular stress, were up-regulated during D-xylonate production, 
compared to the control.  The overall transcriptional responses were, therefore, very 
dissimilar to those previously reported as being associated with stress, including stress 
induced by  organic acid treatment or production. Quantitative PCR analyses of selected 
genes supported the observations made in the transcriptomic analysis. In addition, 
consumption of ethanol was slower and the level of trehalose was lower in the D-xylonate 
producing strain, compared to the control. CONCLUSIONS: The production of organic acids 
has a major impact on the physiology of yeast cells, but the transcriptional responses to 
presence or production of different acids differs considerably, being much more diverse 
than responses to other stresses. D-Xylonate production apparently imposed considerable 
stress on the cell wall. Transcriptional data also indicated that activation of the PKA 
pathway occurred during D-xylonate production, leaving cells unable to adapt normally to 
stationary phase. This, together with intracellular acidification, probably contributes to
cell death.

PMID: 25192596

GEO ID: GSE52736.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE52736.final.pcl	Transcriptome of Saccharomyces cerevisiae during production of D-xylonate.	36	chemical stimulus|stress	1

File last modified: Thu Apr 25 2024

---


# Nagarajan_2014_PMID_24706810

Citation: Nagarajan S, Kruckeberg AL, Schmidt KH, Kroll E, Hamilton M, McInnerney K, Summers R, 
Taylor T, Rosenzweig F. 2014. Uncoupling reproduction from metabolism extends 
chronological lifespan in yeast.

Full Description: Studies of replicative and chronological lifespan in Saccharomyces cerevisiae have 
advanced understanding of longevity in all eukaryotes. Chronological lifespan in this 
species is defined as the age-dependent viability of nondividing cells. To date this 
parameter has only been estimated under calorie restriction,  mimicked by starvation. 
Because postmitotic cells in higher eukaryotes often do not starve, we developed a model 
yeast system to study cells as they age in the absence of calorie restriction. Yeast cells
were encapsulated in a matrix consisting of calcium alginate to form approximately 3 mm 
beads that were packed  into bioreactors and fed ad libitum. Under these conditions cells 
ceased to divide, became heat shock and zymolyase resistant, yet retained high 
fermentative capacity. Over the course of 17 d, immobilized yeast cells maintained >95% 
viability, whereas the viability of starving, freely suspended (planktonic) cells 
decreased to <10%. Immobilized cells exhibited a stable pattern of gene expression that 
differed markedly from growing or starving planktonic cells, highly expressing genes in 
glycolysis, cell wall remodeling, and stress resistance, but decreasing transcription of 
genes in the tricarboxylic acid cycle, and genes that regulate the cell cycle, including 
master cyclins CDC28 and CLN1. Stress resistance transcription factor MSN4 and its 
upstream effector RIM15 are conspicuously up-regulated in the immobilized state, and an 
immobilized rim15 knockout strain fails to exhibit the long-lived, growth-arrested 
phenotype, suggesting that altered regulation of the Rim15-mediated nutrient-sensing 
pathway plays an important role in extending yeast chronological lifespan under calorie-
unrestricted conditions.

PMID: 24706810

GEO ID: GSE21187.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE21187.final.pcl	Uncoupling reproduction from metabolism extends chronological lifespan in yeast.	26	cell aging	2

File last modified: Thu Apr 25 2024

---


# Nakamura_2014_PMID_25177541

Citation: Nakamura T, Yamamoto M, Saito K, Ando A, Shima J. 2014. Identification of a gene, FMP21, 
whose expression levels are involved in thermotolerance in Saccharomyces cerevisiae.

Full Description: Elucidation of the mechanism of high temperature tolerance in yeasts is important for the 
molecular breeding of high temperature-tolerant yeasts that can be used in bioethanol 
production. We identified genes whose expression is correlated with the degree of 
thermotolerance in Saccharomyces cerevisiae by DNA microarray analysis. Gene expression 
profiles of three S. cerevisiae strains showing different levels of thermotolerance were 
compared, and we chose three of them as  candidate genes. Among these genes, FMP21 was 
investigated as a thermotolerance-related gene in S. cerevisiae by comparing the growth at
high temperature with the gene expression in eight strains. The expression ratio of FMP21 
at 37 degrees C was correlated with the doubling time ratio at a coefficient of 
determination of 0.787. The potential involvement of the Fmp21 in  the thermotolerance of 
yeasts was evaluated. The FMP21 deletion variant showed a  decreased respiratory growth 
rate and increased thermosensitivity. Furthermore, the overexpression of FMP21 improved 
thermotolerance in yeasts. In conclusion, the function of Fmp21 is important for 
thermotolerance in yeasts.

PMID: 25177541

GEO ID: GSE33276.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE33276.final.pcl	Identification of a gene, FMP21, whose expression levels are involved in thermotolerance in Saccharomyces cerevisiae.	14	heat shock	1

File last modified: Thu Apr 25 2024

---


# Nasuno_2014_PMID_25423296

Citation: Nasuno R, Aitoku M, Manago Y, Nishimura A, Sasano Y, Takagi H. 2014. Nitric oxide-mediated
antioxidative mechanism in yeast through the activation of  the transcription factor Mac1.

Full Description: The budding yeast Saccharomyces cerevisiae possesses various defense mechanisms against 
environmental stresses that generate reactive oxygen species, leading to  growth 
inhibition or cell death. Our recent study showed a novel antioxidative mechanism mediated
by nitric oxide (NO) in yeast cells, but the mechanism underlying the oxidative stress 
tolerance remained unclear. We report here one of the downstream pathways of NO involved 
in stress-tolerance mechanism in yeast. Our microarray and real-time quantitative PCR 
analyses revealed that exogenous NO treatment induced the expression of genes responsible 
for copper metabolism under the control of the transcription factor Mac1, including the 
CTR1 gene encoding high-affinity copper transporter. Our ChIP analysis also demonstrated 
that exogenous NO enhances the binding of Mac1 to the promoter region of target genes. 
Interestingly, we found that NO produced under high-temperature stress conditions 
increased the transcription level of the CTR1 gene. Furthermore, NO produced during 
exposure to high temperature also increased intracellular copper content,  the activity of
Cu,Zn-superoxide dismutase Sod1, and cell viability after exposure to high-temperature in 
a manner dependent on Mac1. NO did not affect the expression of the MAC1 gene, indicating 
that NO activates Mac1 through its post-translational modification. Based on the results 
shown here, we propose a novel NO-mediated antioxidative mechanism that Mac1 activated by 
NO induces the CTR1 gene, leading to an increase in cellular copper level, and then Cu(I) 
activates Sod1. This is the first report to unveil the mechanism of NO-dependent  
antioxidative system in yeast.

PMID: 25423296

GEO ID: GSE61836.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE61836.final.pcl	Nitric oxide-mediated antioxidative mechanism in yeast through the activation of  the transcription factor Mac1.	4	oxidative stress	1

File last modified: Thu Apr 25 2024

---


# O'Duibhir_2014_PMID_24952590

Citation: O'Duibhir E, Lijnzaad P, Benschop JJ, Lenstra TL, van Leenen D, Groot Koerkamp MJ, 
Margaritis T, Brok MO, Kemmeren P, Holstege FC. 2014. Cell cycle population effects in 
perturbation studies.

Full Description: Growth condition perturbation or gene function disruption are commonly used strategies to 
study cellular systems. Although it is widely appreciated that such experiments may 
involve indirect effects, these frequently remain uncharacterized. Here, analysis of 
functionally unrelated Saccharyomyces cerevisiae deletion strains reveals a common gene 
expression signature. One property shared by these strains is slower growth, with 
increased presence of the signature in more slowly growing strains. The slow growth 
signature is highly similar to the environmental stress response (ESR), an expression 
response common to diverse environmental perturbations. Both environmental and genetic 
perturbations result in growth rate changes. These are accompanied by a change in the 
distribution of cells over different cell cycle phases. Rather than representing a direct 
expression response in single cells, both the slow growth signature and ESR mainly reflect
a redistribution of cells over different cell cycle phases, primarily characterized by an 
increase in the G1 population. The findings have implications for any study of 
perturbation that is accompanied by growth rate changes. Strategies to counter these 
effects are presented and discussed.

PMID: 24952590

GEO ID: GSE54539.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE54539.remapped.final.pcl	Cell cycle population effects in perturbation studies.	12	nutrient utilization	2

File last modified: Thu Apr 25 2024
GSE54527.remapped.final.pcl	Cell cycle population effects in perturbation studies.	4	phosphorus utilization	2

File last modified: Thu Apr 25 2024
GSE54528.remapped.final.pcl	Cell cycle population effects in perturbation studies.	22	heat shock|stress	2

File last modified: Thu Apr 25 2024

---


# Reyes_2014_PMID_24262517

Citation: Reyes LH, Gomez JM, Kao KC. 2014. Improving carotenoids production in yeast via adaptive 
laboratory evolution.

Full Description: Adaptive laboratory evolution is an important tool for the engineering of strains for 
industrially relevant phenotypes. Traditionally, adaptive laboratory evolution has been 
implemented to improve robustness of industrial strains under  diverse operational 
conditions; however due to the required coupling between growth and survival, its 
application for increased production of secondary metabolites generally results in 
decreased production due to the metabolic burden imposed by, or toxicity of, the produced 
compound. In this study, adaptive laboratory evolution was successfully applied to improve
carotenoids production in an engineered Saccharomyces cerevisiae producer strain by 
exploiting the antioxidant properties of carotenoids. Short-term evolution experiment 
using periodic hydrogen peroxide shocking schemes resulted in a 3-fold increase in 
carotenoids production (from 6 mg/g dry cell weight to up to 18 mg/g dry cell weight). 
Subsequent transcriptome analysis was used to elucidate the molecular mechanisms for 
increased carotenoids production. Upregulation of genes related with lipid biosynthesis 
and mevalonate biosynthesis pathways were commonly observed in the carotenoids hyper-
producers analyzed.

PMID: 24262517

GEO ID: GSE51613.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE51613.remapped.final.pcl	Improving carotenoids production in yeast via adaptive laboratory evolution.	13	evolution|oxidative stress|stress	2

File last modified: Thu Apr 25 2024

---


# Rodriguez-Lombardero_2014_PMID_24763424

Citation: Rodriguez-Lombardero S, Vizoso-Vazquez A, Lombardia LJ, Becerra M, Gonzalez-Siso MI, 
Cerdan ME. 2014. Sky1 regulates the expression of sulfur metabolism genes in response to 
cisplatin.

Full Description: Cisplatin is commonly used in cancer therapy and yeast cells are also sensitive to this 
compound. We present a transcriptome analysis discriminating between RNA  changes induced 
by cisplatin treatment, which are dependent on or independent of  SKY1 function--a gene 
whose deletion increases resistance to the drug. Gene expression changes produced by 
addition of cisplatin to W303 and W303-Deltasky1 cells were recorded using DNA microarrays
. The data, validated by quantitative PCR, revealed 122 differentially expressed genes: 69
upregulated and 53 downregulated. Among the upregulated genes, those related to sulfur 
metabolism were over-represented and partially dependent on Sky1. Deletions of MET4 or 
other genes encoding co-regulators of the expression of sulfur-metabolism-related genes, 
with the exception of MET28, did not modify the cisplatin sensitivity of yeast cells. One 
of the genes with the highest cisplatin-induced upregulation was SEO1, encoding a putative
permease of sulfur compounds. We also measured the platinum, sulfur and glutathione 
content in W303, W303-Deltasky1 and W303-Deltaseo1 cells after cisplatin treatment, and 
integration of the data suggested that these transcriptional changes might represent a 
cellular response  that allowed chelation of cisplatin with sulfur-containing amino acids 
and also helped DNA repair by stimulating purine biosynthesis. The transcription pattern 
of stimulation of sulfur-containing amino acids and purine synthesis decreased, or even 
disappeared, in the W303-Deltasky1 strain.

PMID: 24763424

GEO ID: GSE41094.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE41094.remapped.final.pcl	Sky1 regulates the expression of sulfur metabolism genes in response to cisplatin.	18	DNA damage stimulus|stress	1

File last modified: Thu Apr 25 2024

---


# Romagnoli_2014_PMID_24733517

Citation: Romagnoli G, Knijnenburg TA, Liti G, Louis EJ, Pronk JT, Daran JM. 2014. Deletion of the 
Saccharomyces cerevisiae ARO8 gene, encoding an aromatic amino acid transaminase, enhances
phenylethanol production from glucose.

Full Description: Phenylethanol has a characteristic rose-like aroma that makes it a popular ingredient in 
foods, beverages and cosmetics. Microbial production of phenylethanol currently relies on 
whole-cell bioconversion of phenylalanine with  yeasts that harbour an Ehrlich pathway for
phenylalanine catabolism. Complete biosynthesis of phenylethanol from a cheap carbon 
source, such as glucose, provides an economically attractive alternative for phenylalanine
bioconversion.  In this study, synthetic genetic array (SGA) screening was applied to 
identify genes involved in regulation of phenylethanol synthesis in Saccharomyces 
cerevisiae. The screen focused on transcriptional regulation of ARO10, which encodes the 
major decarboxylase involved in conversion of phenylpyruvate to phenylethanol. A deletion 
in ARO8, which encodes an aromatic amino acid transaminase, was found to underlie the 
transcriptional upregulation of ARO10 during growth, with ammonium sulphate as the sole 
nitrogen source. Physiological  characterization revealed that the aro8Delta mutation led 
to substantial changes  in the absolute and relative intracellular concentrations of amino
acids. Moreover, deletion of ARO8 led to de novo production of phenylethanol during growth
on a glucose synthetic medium with ammonium as the sole nitrogen source. The aro8Delta 
mutation also stimulated phenylethanol production when combined with other, previously 
documented, mutations that deregulate aromatic amino acid  biosynthesis in S. cerevisiae. 
The resulting engineered S. cerevisiae strain produced >3 mm phenylethanol from glucose 
during growth on a simple synthetic medium. The strong impact of a transaminase deletion 
on intracellular amino acid  concentrations opens new possibilities for yeast-based 
production of amino acid-derived products. Copyright (c) 2014 John Wiley & Sons, Ltd.

PMID: 24733517

GEO ID: GSE52256.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE52256.remapped.final.pcl	Deletion of the Saccharomyces cerevisiae ARO8 gene, encoding an aromatic amino acid transaminase, enhances phenylethanol production from glucose.	4	amino acid metabolism	1

File last modified: Thu Apr 25 2024

---


# Saint_2014_PMID_24550006

Citation: Saint M, Sawhney S, Sinha I, Singh RP, Dahiya R, Thakur A, Siddharthan R, Natarajan K. 
2014. The TAF9 C-terminal conserved region domain is required for SAGA and TFIID promoter 
occupancy to promote transcriptional activation.

Full Description: A common function of the TFIID and SAGA complexes, which are recruited by transcriptional 
activators, is to deliver TBP to promoters to stimulate transcription. Neither the 
relative contributions of the five shared TBP-associated factor (TAF) subunits in TFIID 
and SAGA nor the requirement for different domains in shared TAFs for transcriptional 
activation is well understood. In this study, we uncovered the essential requirement for 
the highly  conserved C-terminal region (CRD) of Taf9, a shared TAF, for transcriptional 
activation in yeast. Transcriptome profiling performed under Gcn4-activating conditions 
showed that the Taf9 CRD is required for induced expression of approximately 9% of the 
yeast genome. The CRD was not essential for the Taf9-Taf6 interaction, TFIID or SAGA 
integrity, or Gcn4 interaction with SAGA in cell extracts. Microarray profiling of a SAGA 
mutant (spt20Delta) yielded a common set of genes induced by Spt20 and the Taf9 CRD. 
Chromatin immunoprecipitation (ChIP)  assays showed that, although the Taf9 CRD mutation 
did not impair Gcn4 occupancy, the occupancies of TFIID, SAGA, and the preinitiation 
complex were severely impaired at several promoters. These results suggest a crucial role 
for the Taf9  CRD in genome-wide transcription and highlight the importance of conserved 
domains, other than histone fold domains, as a common determinant for TFIID and SAGA 
functions.

PMID: 24550006

GEO ID: GSE44544.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE44544.remapped.final.pcl	The TAF9 C-terminal conserved region domain is required for SAGA and TFIID promoter occupancy to promote transcriptional activation.	14	starvation|transcription	2

File last modified: Thu Apr 25 2024

---


# Santos-Pereira_2014_PMID_25294824

Citation: Santos-Pereira JM, Garcia-Rubio ML, Gonzalez-Aguilera C, Luna R, Aguilera A. 2014. A 
genome-wide function of THSC/TREX-2 at active genes prevents transcription-replication 
collisions.

Full Description: The THSC/TREX-2 complex of Saccharomyces cerevisiae mediates the anchoring of transcribed 
genes to the nuclear pore, linking transcription elongation with mRNA export and genome 
stability, as shown for specific reporters. However, it is still unknown whether the 
function of TREX-2 is global and the reason for its relevant role in genome integrity. 
Here, by studying two TREX-2 representative subunits, Thp1 and Sac3, we show that TREX-2 
has a genome-wide role in gene expression. Both proteins show similar distributions along 
the genome, with a gradient disposition at active genes that increases towards the 3' end
. Thp1 and  Sac3 have a relevant impact on the expression of long, G+C-rich and highly 
transcribed genes. Interestingly, replication impairment detected by the genome-wide 
accumulation of the replicative Rrm3 helicase is increased preferentially at highly 
expressed genes in the thp1Delta and sac3Delta mutants analyzed. Therefore, our work 
provides evidence of a function of TREX-2 at the genome-wide level and suggests a role for
TREX-2 in preventing transcription-replication conflicts, as a source of genome 
instability derived from a defective messenger ribonucleoprotein particle (mRNP) 
biogenesis.

PMID: 25294824

GEO ID: GSE56702.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE56702.final.pcl	A genome-wide function of THSC/TREX-2 at active genes prevents transcription-replication collisions.	15	transcription|mRNA processing	1

File last modified: Thu Apr 25 2024

---


# Slavov_2014_PMID_24767987

Citation: Slavov N, Budnik BA, Schwab D, Airoldi EM, van Oudenaarden A. 2014. Constant growth rate 
can be supported by decreasing energy flux and increasing aerobic glycolysis.

Full Description: Fermenting glucose in the presence of enough oxygen to support respiration, known as 
aerobic glycolysis, is believed to maximize growth rate. We observed increasing aerobic 
glycolysis during exponential growth, suggesting additional physiological roles for 
aerobic glycolysis. We investigated such roles in yeast batch cultures by quantifying O2 
consumption, CO2 production, amino acids, mRNAs, proteins, posttranslational modifications
, and stress sensitivity in the course of nine doublings at constant rate. During this 
course, the cells support a constant biomass-production rate with decreasing rates of 
respiration and ATP production but also decrease their stress resistance. As the 
respiration rate decreases, so do the levels of enzymes catalyzing rate-determining 
reactions of the tricarboxylic-acid cycle (providing NADH for respiration) and of 
mitochondrial folate-mediated NADPH production (required for oxidative defense).  The 
findings demonstrate that exponential growth can represent not a single metabolic/
physiological state but a continuum of changing states and that aerobic glycolysis can 
reduce the energy demands associated with respiratory metabolism and stress survival.

PMID: 24767987

GEO ID: GSE56773.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE56773.remapped.final.pcl	Constant growth rate can be supported by decreasing energy flux and increasing aerobic glycolysis.	8	fermentation|respiration	2

File last modified: Thu Apr 25 2024

---


# Stovicek_2014_PMID_24533484

Citation: Stovicek V, Vachova L, Begany M, Wilkinson D, Palkova Z. 2014. Global changes in gene 
expression associated with phenotypic switching of wild yeast.

Full Description: BACKGROUND: Saccharomyces cerevisiae strains isolated from natural settings form  
structured biofilm colonies that are equipped with intricate protective mechanisms. These 
wild strains are able to reprogram themselves with a certain frequency during cultivation 
in plentiful laboratory conditions. The resulting domesticated strains switch off certain 
protective mechanisms and form smooth colonies that resemble those of common laboratory 
strains. RESULTS: Here, we show that domestication can be reversed when a domesticated 
strain is challenged by various adverse conditions; the resulting feral strain restores 
its ability to form structured biofilm colonies. Phenotypic, microscopic and 
transcriptomic analyses show that phenotypic transition is a complex process that affects 
various aspects of feral strain physiology; it leads to a phenotype that resembles the 
original wild strain in some aspects and the domesticated derivative in others. We specify
the genetic determinants that are likely involved in the formation of a structured biofilm
colonies. In addition to FLO11, these determinants include genes that affect the cell wall
and membrane composition. We also identify changes occurring during phenotypic transitions
that affect other properties of phenotypic strain-variants, such as resistance to the 
impact of environmental stress. Here we document the regulatory role of the histone 
deacetylase Hda1p in developing such a resistance. CONCLUSIONS: We provide detailed 
analysis of transcriptomic and phenotypic modulations of three related S. cerevisiae 
strains that arose by phenotypic switching under diverse environmental conditions. We 
identify changes specifically related to a strain's  ability to create complex structured 
colonies; we also show that other changes, such as genome rearrangement(s), are unrelated 
to this ability. Finally, we identify the importance of histone deacetylase Hda1p in 
strain resistance to stresses.

PMID: 24533484

GEO ID: GSE40625.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE40625.remapped.final.pcl	Global changes in gene expression associated with phenotypic switching of wild yeast.	18	evolution	2

File last modified: Thu Apr 25 2024

---


# Tsang_2014_PMID_24647101

Citation: Tsang CK, Liu Y, Thomas J, Zhang Y, Zheng XF. 2014. Superoxide dismutase 1 acts as a 
nuclear transcription factor to regulate oxidative stress resistance.

Full Description: Superoxide dismutase 1 (Sod1) has been known for nearly half a century for catalysis of 
superoxide to hydrogen peroxide. Here we report a new Sod1 function  in oxidative 
signalling: in response to elevated endogenous and exogenous reactive oxygen species (ROS
), Sod1 rapidly relocates into the nucleus, which is  important for maintaining genomic 
stability. Interestingly, H2O2 is sufficient to promote Sod1 nuclear localization, 
indicating that it is responding to general ROS rather than Sod1 substrate superoxide. ROS
signalling is mediated by Mec1/ATM and its effector Dun1/Cds1 kinase, through Dun1 
interaction with Sod1 and regulation of Sod1 by phosphorylation at S60, 99. In the nucleus
, Sod1 binds to promoters and regulates the expression of oxidative resistance and repair 
genes.  Altogether, our study unravels an unorthodox function of Sod1 as a transcription  
factor and elucidates the regulatory mechanism for its localization.

PMID: 24647101

GEO ID: GSE55081.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE55081.remapped.final.pcl	Superoxide dismutase 1 acts as a nuclear transcription factor to regulate oxidative stress resistance.	4	oxidative stress|stress	1

File last modified: Thu Apr 25 2024

---


# Wang_2014_PMID_24682482

Citation: Wang Y, Chen H, Yu O. 2014. A plant malonyl-CoA synthetase enhances lipid content and 
polyketide yield in yeast cells.

Full Description: Malonyl-CoA is the essential building block of natural products such as fatty acids, 
polyketides, and flavonoids. Engineering the biosynthesis of fatty acids is important for 
biofuel production while that of polyketides provides precursors of medicines and 
nutritional supplements. However, microorganisms maintain a small amount of cellular 
malonyl-CoA, which could limit production of lipid and polyketides under certain 
conditions. Malonyl-CoA concentration is regulated by multiple pathways and signals, and 
changes in intracellular malonyl-CoA often lead to complex alterations in metabolism. In 
the present work, overexpression of a plant malonyl-CoA synthetase gene (AAE13) in 
Saccharomyces cerevisiae resulted  in 1.6- and 2.4-fold increases in lipid and resveratrol
accumulation simultaneously. We also demonstrated that AAE13 partially complemented the 
temperature-sensitive acc1 mutant, replacing this key enzyme in central metabolism. 
Mechanistic analysis by CoA quantification and transcriptomic measurement suggested that 
increases in malonyl-CoA concentration were coupled with drastic reductions in other major
CoA compounds and clear suppression of tricarboxylic acid cycle-related genes. These 
results suggest that malonyl-CoA is a critical target for fatty acid and polyketide 
engineering and that overexpression of malonyl-CoA synthetic enzymes needs to be combined 
with upregulation of CoA synthesis to maintain metastasis of central metabolism.

PMID: 24682482

GEO ID: GSE52701.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE52701.final.pcl	A plant malonyl-CoA synthetase enhances lipid content and polyketide yield in yeast cells.	4	lipid metabolism|synthetic biology	2

File last modified: Thu Apr 25 2024

---


# Yamane-Sando_2014_PMID_24510621

Citation: Yamane-Sando Y, Shimobayashi E, Shimobayashi M, Kozutsumi Y, Oka S, Takematsu H. 2014. 
Fpk1/2 kinases regulate cellular sphingoid long-chain base abundance and alter cellular 
resistance to LCB elevation or depletion.

Full Description: Sphingolipids are a family of eukaryotic lipids biosynthesized from sphingoid long-chain 
bases (LCBs). Sphingolipids are an essential class of lipids, as their depletion results 
in cell death. However, acute LCB supplementation is also toxic; thus, proper cellular LCB
levels should be maintained. To characterize the sphingolipid-signaling intercross, we 
performed a kinome screening assay in which budding yeast protein kinase-knockout strains 
were screened for resistance  to ISP-1, a potent inhibitor of LCB biosynthesis. Here, one 
pair of such DIR (deletion-mediated ISP-1 resistance) genes, FPK1 and FPK2, was further 
characterized. Cellular LCB levels increased in the fpk1/2 strain, which was 
hypersensitive to phytosphingosine (PHS), a major LCB species of yeast cells. 
Concomitantly, this strain acquired resistance to ISP-1. Fpk1 and Fpk2 were involved in 
two downstream events; that is, ISP-1 uptake due to aminophospholipid flippase and LCB 
degradation due to LCB4 expression. RSK3, which belongs to the p90-S6K subfamily, was 
identified as a functional counterpart of Fpk1/2 in mammalian cells as the RSK3 gene 
functionally complemented the ISP-1-resistant phenotype of fpk1/2 cells.

PMID: 24510621

GEO ID: GSE42083.remapped.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE42083.remapped.final.pcl	Fpk1/2 kinases regulate cellular sphingoid long-chain base abundance and alter cellular resistance to LCB elevation or depletion.	1	lipid metabolism	1

File last modified: Thu Apr 25 2024

---


# Barbosa_2015_PMID_25884705

Citation: Barbosa C, Garcia-Martinez J, Perez-Ortin JE, Mendes-Ferreira A. 2015. Comparative 
transcriptomic analysis reveals similarities and dissimilarities in Saccharomyces 
cerevisiae wine strains response to nitrogen availability.

Full Description: Nitrogen levels in grape-juices are of major importance in winemaking ensuring adequate 
yeast growth and fermentation performance. Here we used a comparative transcriptome 
analysis to uncover wine yeasts responses to nitrogen availability  during fermentation. 
Gene expression was assessed in three genetically and phenotypically divergent commercial 
wine strains (CEG, VL1 and QA23), under low (67 mg/L) and high nitrogen (670 mg/L) regimes
, at three time points during fermentation (12 h, 24 h and 96 h). Two-way ANOVA analysis 
of each fermentation condition led to the identification of genes whose expression was 
dependent on strain, fermentation stage and on the interaction of both factors. The high 
fermenter yeast strain QA23 was more clearly distinct from the other two strains, by 
differential expression of genes involved in flocculation, mitochondrial functions, energy
generation and protein folding and stabilization. For all strains, higher transcriptional 
variability due to fermentation stage was seen in the high nitrogen fermentations. A 
positive correlation between maximum fermentation rate and the expression of genes 
involved in stress response was observed. The finding of common genes correlated with both
fermentation activity  and nitrogen up-take underlies the role of nitrogen on yeast 
fermentative fitness. The comparative analysis of genes differentially expressed between 
both  fermentation conditions at 12 h, where the main difference was the level of nitrogen
available, showed the highest variability amongst strains revealing strain-specific 
responses. Nevertheless, we were able to identify a small set of  genes whose expression 
profiles can quantitatively assess the common response of  the yeast strains to varying 
nitrogen conditions. The use of three contrasting yeast strains in gene expression 
analysis prompts the identification of more reliable, accurate and reproducible biomarkers
that will facilitate the diagnosis of deficiency of this nutrient in the grape-musts and 
the development of strategies to optimize yeast performance in industrial fermentations.

PMID: 25884705

GEO ID: GSE63187.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE63187.final.pcl	Comparative transcriptomic analysis reveals similarities and dissimilarities in Saccharomyces cerevisiae wine strains response to nitrogen availability.	25	nitrogen utilization	1

File last modified: Thu Apr 25 2024

---


# Barbosa_2015_PMID_26314747

Citation:  Barbosa C, Mendes-Faia A, Lage P, Mira NP, Mendes-Ferreira A . 2015. Genomic expression 
program of Saccharomyces cerevisiae along a mixed-culture wine fermentation with 
Hanseniaspora guilliermondii.

Full Description:  Natural grape-juice fermentations involve the sequential development of different yeast 
species which strongly influence the chemical and sensorial traits of the final product
.   In the present study,we aimed to examine the transcriptomic response of Saccharomyces 
cerevisiae to the presence of Hanseniaspora guilliermondii wine fermentation


 .

PMID: 26314747

GEO ID: GSE66521.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE66521.mapped.pcl	Transcriptomic response of Saccharomyces cerevisiae in mixed-culture wine fermentation with Hanseniaspora guilliermondii	16	fermentation|genome variation	1

File last modified: Thu Apr 25 2024

---


# Buck_2015_PMID_25759377

Citation: Buck TM, Jordan R, Lyons-Weiler J, Adelman JL, Needham PG, Kleyman TR, Brodsky JL. 2015. 
Expression of three topologically distinct membrane proteins elicits unique stress 
response pathways in the yeast Saccharomyces cerevisiae.

Full Description: Misfolded membrane proteins are retained in the endoplasmic reticulum (ER) and are subject
to ER-associated degradation, which clears the secretory pathway of potentially toxic 
species. While the transcriptional response to environmental stressors has been 
extensively studied, limited data exist describing the cellular response to misfolded 
membrane proteins. To this end, we expressed and then compared the transcriptional 
profiles elicited by the synthesis of three ER  retained, misfolded ion channels: The 
alpha-subunit of the epithelial sodium channel, ENaC, the cystic fibrosis transmembrane 
conductance regulator, CFTR, and an inwardly rectifying potassium channel, Kir2.1, which 
vary in their mass, membrane topologies, and quaternary structures. To examine 
transcriptional profiles in a null background, the proteins were expressed in yeast, which
was previously used to examine the degradation requirements for each substrate. 
Surprisingly, the proteins failed to induce a canonical unfolded protein response or heat 
shock response, although messages encoding several cytosolic and ER lumenal protein 
folding factors rose when alphaENaC or CFTR was expressed. In contrast, the levels of 
these genes were unaltered by Kir2.1 expression; instead, the yeast iron regulon was 
activated. Nevertheless, a significant number of genes that respond to various 
environmental stressors were upregulated by all three substrates, and compared with 
previous microarray data we deduced the existence of a group of genes that reflect a novel
misfolded membrane protein response. These data indicate that aberrant proteins in the ER 
elicit profound yet unique cellular responses.

PMID: 25759377

GEO ID: GSE61244.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE61244.final.pcl	Expression data from yeast heterologously expressing _ENaC, CFTR or Kir2.1 [CFTR]	11	disease|stress	1

File last modified: Thu Apr 25 2024
GSE61245.final.pcl	Expression data from yeast heterologously expressing _ENaC, CFTR or Kir2.1 [Kir2.1]	8	disease|stress	1

File last modified: Thu Apr 25 2024
GSE61243.final.pcl	Expression data from yeast heterologously expressing _ENaC, CFTR or Kir2.1 [_ENaC]	10	disease|stress	1

File last modified: Thu Apr 25 2024

---


# Canadell_2015_PMID_25425491

Citation: Canadell D, Gonzalez A, Casado C, Arino J. 2015. Functional interactions between potassium
and phosphate homeostasis in Saccharomyces cerevisiae.

Full Description: Maintenance of ion homeostatic mechanisms is essential for living cells, including the 
budding yeast Saccharomyces cerevisiae. Whereas the impact of changes in phosphate 
metabolism on metal ion homeostasis has been recently examined, the inverse effect is 
still largely unexplored. We show here that depletion of potassium from the medium or 
alteration of diverse regulatory pathways controlling potassium uptake, such as the Trk 
potassium transporters or  the Pma1 H(+) -ATPase, triggers a response that mimics that of 
phosphate (Pi) deprivation, exemplified by accumulation of the high-affinity Pi 
transporter Pho84. This response is mediated by and requires the integrity of the PHO 
signaling pathway. Removal of potassium from the medium does not alter the amount of total
or free intracellular Pi, but is accompanied by decreased ATP and ADP levels and rapid 
depletion of cellular polyphosphates. Therefore, our data do not support the notion of Pi 
being the major signaling molecule triggering phosphate-starvation responses. We also 
observe that cells with compromised potassium uptake cannot grow under limiting Pi 
conditions. The link between potassium and phosphate homeostasis reported here could 
explain the invasive phenotype, characteristic of nutrient deprivation, observed in 
potassium-deficient yeast cells.

PMID: 25425491

GEO ID: GSE58985.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE58985.final.pcl	Functional interactions between potassium and phosphate homeostasis in Saccharomyces cerevisiae.	3	cellular ion homeostasis|phosphorus utilization	2

File last modified: Thu Apr 25 2024

---


# Cary_2015_PMID_26392412

Citation: Cary GA, Vinh DB, May P, Kuestner R, Dudley AM. 2015. Proteomic Analysis of Dhh1 Complexes
Reveals a Role for Hsp40 Chaperone Ydj1 in Yeast P-Body Assembly.

Full Description: P-bodies (PB) are ribonucleoprotein (RNP) complexes that aggregate into cytoplasmic foci 
when cells are exposed to stress. Although the conserved mRNA decay and translational 
repression machineries are known components of PB, how and why cells assemble RNP 
complexes into large foci remain unclear. Using mass spectrometry to analyze proteins 
immunoisolated with the core PB protein Dhh1, we show that a considerable number of 
proteins contain low-complexity sequences, similar to proteins highly represented in 
mammalian RNP granules. We also show that the Hsp40 chaperone Ydj1, which contains an low-
complexity domain and controls prion protein aggregation, is required for the formation of
Dhh1-GFP foci on glucose depletion. New classes of proteins that reproducibly coenrich 
with Dhh1-GFP during PB induction include proteins involved in nucleotide or amino acid 
metabolism, glycolysis, transfer RNA aminoacylation, and protein folding. Many of these 
proteins have been shown to form foci in response to other stresses. Finally, analysis of 
RNA associated with Dhh1-GFP shows enrichment of mRNA encoding the PB protein Pat1 and 
catalytic RNAs along with their associated  mitochondrial RNA-binding proteins. Thus, 
global characterization of PB composition has uncovered proteins important for PB assembly
and evidence suggesting an active role for RNA in PB function.

PMID: 26392412

GEO ID: GSE65989.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE65989.final.pcl	Proteomic Analysis of Dhh1 Complexes Reveals a Role for Hsp40 Chaperone Ydj1 in Yeast P-Body Assembly.	14	stress	1

File last modified: Thu Apr 25 2024

---


# Chujo_2015_PMID_25304510

Citation: Chujo M, Yoshida S, Ota A, Murata K, Kawai S. 2015. Acquisition of the ability to 
assimilate mannitol by Saccharomyces cerevisiae through dysfunction of the general 
corepressor Tup1-Cyc8.

Full Description: Saccharomyces cerevisiae normally cannot assimilate mannitol, a promising brown macroalgal
carbon source for bioethanol production. The molecular basis of this inability remains 
unknown. We found that cells capable of assimilating mannitol arose spontaneously from 
wild-type S. cerevisiae during prolonged culture in mannitol-containing medium. Based on 
microarray data, complementation analysis, and cell growth data, we demonstrated that 
acquisition of mannitol-assimilating ability was due to spontaneous mutations in the genes
encoding Tup1 or Cyc8, which constitute a general corepressor complex that regulates many 
kinds of genes. We also showed that an S. cerevisiae strain carrying a mutant allele of 
CYC8 exhibited superior salt tolerance relative to other ethanologenic microorganisms; 
this characteristic would be highly beneficial for the production of bioethanol from 
marine biomass. Thus, we succeeded in conferring the ability to assimilate mannitol on S. 
cerevisiae through dysfunction of Tup1-Cyc8, facilitating production of ethanol from 
mannitol.

PMID: 25304510

GEO ID: GSE55755.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE55755.final.pcl	Acquisition of the ability to assimilate mannitol by Saccharomyces cerevisiae through dysfunction of the general corepressor Tup1-Cyc8.	4	carbon utilization	1

File last modified: Thu Apr 25 2024

---


# Crosas_2015_PMID_25715111

Citation: Crosas E, Sumoy L, Gonzalez E, Diaz M, Bartolome S, Farres J, Pares X, Biosca JA, 
Fernandez MR. 2015. The yeast zeta-crystallin/NADPH:quinone oxidoreductase (Zta1p) is 
under nutritional control by the target of rapamycin pathway and is involved in the 
regulation of argininosuccinate lyase mRNA half-life.

Full Description: The yeast zeta-crystallin (Zta1p) is a quinone oxidoreductase belonging to the zeta-
crystallin family, with activity in the reduction of alkenal/alkenone compounds. Various 
biological functions have been ascribed to the members of this protein family, such as 
their ability to interact specifically with AU-rich sequences in mRNA, and thus they have 
been proposed to act as AU-rich element-binding proteins (AREBPs). In this study, we 
evaluated the specificity of Zta1p for RNA versus DNA by means of a novel nonisotopic 
method for the in vitro  quantitative detection of protein . RNA complexes. Through 
comparative transcriptomic analysis, we found that the lack of Zta1p negatively affects 
the expression of a group of genes involved in amino acid biosynthesis, the 
argininosuccinate lyase (ARG4) gene being one of them. Here, we propose that Zta1p 
participates in the post-transcriptional regulation of ARG4 expression by increasing the 
ARG4 mRNA half-life. In addition, expression of the zeta-crystallin gene (ZTA1) is itself 
regulated by nutrient availability through  the general amino acid control and target of 
rapamycin pathways. Our results shed new light on the zeta-crystallin family members from 
yeast to humans as stress response proteins with a bifunctional role in the detoxification
of alkenal and alkenone compounds, and the regulation of gene expression.

PMID: 25715111

GEO ID: GSE45297.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE45297.final.pcl	The yeast zeta-crystallin/NADPH:quinone oxidoreductase (Zta1p) is under nutritional control by the target of rapamycin pathway and is involved in the regulation of argininosuccinate lyase mRNA half-life.	12	chemical stimulus|stress	2

File last modified: Thu Apr 25 2024

---


# Felipe-Abrio_2015_PMID_25452497

Citation: Felipe-Abrio I, Lafuente-Barquero J, Garcia-Rubio ML, Aguilera A. 2015. RNA polymerase II 
contributes to preventing transcription-mediated replication fork stalls.

Full Description: Transcription is a major contributor to genome instability. A main cause of transcription-
associated instability relies on the capacity of transcription to stall replication. 
However, we know little of the possible role, if any, of the RNA polymerase (RNAP) in this
process. Here, we analyzed 4 specific yeast RNAPII  mutants that show different phenotypes
of genetic instability including hyper-recombination, DNA damage sensitivity and/or a 
strong dependency on double-strand break repair functions for viability. Three specific 
alleles of the RNAPII core, rpb1-1, rpb1-S751F and rpb9, cause a defect in replication 
fork progression, compensated for by additional origin firing, as the main action 
responsible for instability. The transcription elongation defects of rpb1-S751F and rpb9 
plus our observation that rpb1-1 causes RNAPII retention on chromatin suggest that RNAPII 
could participate in facilitating fork progression upon a transcription-replication 
encounter. Our results imply that the RNAPII or ancillary factors actively help prevent 
transcription-associated genome instability.

PMID: 25452497

GEO ID: GSE55223.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE55223.final.pcl	RNA polymerase II contributes to preventing transcription-mediated replication fork stalls.	15	transcription|DNA replication, recombination and repair	1

File last modified: Thu Apr 25 2024

---


# Fernandez-Acero_2015_PMID_26261079

Citation: Fernandez-Acero T, Rodriguez-Escudero I, Molina M, Cid VJ. 2015. The yeast cell wall 
integrity pathway signals from recycling endosomes upon elimination of 
phosphatidylinositol (4,5)-bisphosphate by mammalian phosphatidylinositol 3-kinase.

Full Description: Phosphatidylinositol (4,5)-bisphosphate [PtdIns(4,5)P(2)] is essential for recognition of 
the plasma membrane inner leaf by protein complexes. We expressed  mammalian class I 
phosphoinositide 3-kinase (PI3K) in Saccharomyces cerevisiae to eliminate PtdIns(4,5)P(2) 
by its conversion into PtdIns(3,4,5)P(3), a lipid naturally missing in this yeast. This 
led to loss of actin function and endocytosis defects, causing a blockage in polarized 
secretion. Also, the cell wall integrity (CWI) mitogen-activated protein kinase (MAPK) 
pathway was activated, triggering a typical transcriptional response. In the absence of 
PtdIns(4,5)P(2) at the plasma membrane, the Pkc1 protein kinase upstream the CWI  MAPK 
module localized to post-Golgi endosomes marked by SNARE Snc1 and Rab GTPases Ypt31 and 
Ypt32. Other components at the head of the pathway, like the mechanosensor Wsc1, the 
GTPase Rho1 and its activator the GDP/GTP exchange factor Rom2, co-localized with Pkc1 in 
these compartments. Chemical inhibition of PI3K proved that both CWI activation and Pkc1 
relocation to endosomes are reversible.  These results suggest that the CWI pathway is 
able to respond to loss of plasma membrane identity from recycling endosomes.

PMID: 26261079

GEO ID: GSE59596.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE59596.final.pcl	The yeast cell wall integrity pathway signals from recycling endosomes upon elimination of phosphatidylinositol (4,5)-bisphosphate by mammalian phosphatidylinositol 3-kinase.	6	signaling|cell wall organization	1

File last modified: Thu Apr 25 2024

---


# Gamero_2015_PMID_26336982

Citation: Gamero A, Belloch C, Querol A. 2015. Genomic and transcriptomic analysis of aroma 
synthesis in two hybrids between Saccharomyces cerevisiae and S. kudriavzevii in 
winemaking conditions.

Full Description: BACKGROUND: Aroma is one of the most important attributes defining wine quality in which 
yeasts play a crucial role, synthesizing aromatic compounds or releasing odourless 
conjugates. A present-day trend in winemaking consists of lowering fermentation 
temperature to achieve higher aroma production and retention. S. cerevisiae x S. 
kudriavzevii hybrids seem to have inherited beneficial traits from their parental species
, like fermenting efficiently at low temperature or producing higher amounts of certain 
aromatic compounds. In this study, allelic composition and gene expression of the genes 
related to aroma synthesis in two genetically and phenotypically different S. cerevisiae x
S. kudriavzevii hybrids, Lalvin W27 and VIN7, were compared and related to aroma 
production in microvinifications at 12 and 28 degrees C. In addition, the contribution of 
the allele coming from each parental to the overall expression was explored by RT-PCR. 
RESULTS: The results indicated large differences in allele composition, gene expression 
and the contribution of each parental to the overall expression at the fermentation 
temperatures tested. Results obtained by RT-PCR showed that in ARO1 and ATF2 genes the S. 
kudriavzevii allele was more expressed than that of S. cerevisiae particularly at 12 
degrees C. CONCLUSIONS: This study revealed high differences regarding allele composition 
and gene expression in two S. cerevisiae x S. kudriavzevii hybrids, which may have led to 
different aroma profiles in winemaking conditions. The contribution of the alleles coming 
from each parental  to the overall expression has proved to differently influence aroma 
synthesis. Besides, the quantitative contribution to the overall gene expression of the 
alleles coming from one parental strain or the other was clearly determined by the 
fermentation temperature for some genes.

PMID: 26336982

GEO ID: GSE30779.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE30779.final.pcl	Genomic and transcriptomic analysis of aroma synthesis in two hybrids between Saccharomyces cerevisiae and S. kudriavzevii in winemaking conditions.	12	heat shock|fermentation|synthetic biology	2

File last modified: Thu Apr 25 2024

---


# Garca-Rodrguez_2015_PMID_25713143

Citation:  Garca-Rodrguez N, Manzano-Lpez J, Muoz-Bravo M, Fernndez-Garca E, Muiz M, Wellinger RE . 
2015. Manganese redistribution by calcium-stimulated vesicle trafficking bypasses the need
for P-type ATPase function.

Full Description: Pmr1 is a cis-Golgi Mn/Ca transporter with a key role in protein glycosylation and 
manganese detoxification. We used microarray analysis to analyze the impact of CaCl2 
supplementation on gene expression in pmr1 mutants and found that extracellular CaCl2 
suppresses transcriptional misregulation in pmr mutants.

PMID: 25713143

GEO ID: GSE62161.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE62161.mapped.pcl	Expression profile from Saccharomyces cerevisiae strains deleted for PMR1 treated with 5mM CaCl2	6	metal or metalloid ion utilization	1

File last modified: Thu Apr 25 2024

---


# Golla_2015_PMID_25919230

Citation: Golla U, Bandi G, Tomar RS. 2015. Molecular cytotoxicity mechanisms of allyl alcohol (
acrolein) in budding yeast.

Full Description: Allyl alcohol (AA) is one of the environmental pollutants used as a herbicide and 
industrial chemical. AA undergoes enzymatic oxidation in vivo to form Acrolein (Acr), a 
highly reactive and ubiquitous environmental toxicant. The exposure to AA/Acr has 
detrimental effects on cells and is highly fatal. In corroboration to  the current 
literature describing AA/Acr toxicity, this study aimed to investigate the molecular 
cytotoxicity mechanisms of AA/Acr using budding yeast as a eukaryotic model organism. 
Genome-wide transcriptome analysis of cells treated with a sublethal dose of AA (0.4 mM) 
showed differential regulation of approximately 30% of the yeast genome. Functional 
enrichment analysis of the AA transcriptome revealed that genes belong to diverse cellular
processes including  the cell cycle, DNA damage repair, metal homeostasis, stress response
genes, ribosomal biogenesis, metabolism, meiosis, ubiquitination, cell morphogenesis, and 
transport. Moreover, we have identified novel molecular targets of AA/Acr through genetic 
screening, which belongs to oxidative stress, DNA damage repair,  iron homeostasis, and 
cell wall integrity. This study also demonstrated the epigenetic basis of AA/Acr toxicity 
mediated through histone tails and chromatin  modifiers. Interestingly, our study 
disclosed the use of pyrazole and ethanol as  probable antidotes for AA intoxication. For 
the first time, this study also demonstrated the reproductive toxicity of AA/Acr using the
yeast gametogenesis (spermatogenesis) model. Altogether, this study unravels the molecular
mechanisms of AA/Acr cytotoxicity and facilitates the prediction of biomarkers for 
toxicity  assessment and therapeutic approaches.

PMID: 25919230

GEO ID: GSE62399.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE62399.final.pcl	Molecular cytotoxicity mechanisms of allyl alcohol (acrolein) in budding yeast.	4	chemical stimulus|stress|oxidative stress|DNA damage stimulus	1

File last modified: Thu Apr 25 2024

---


# Gomar-Alba_2015_PMID_26048894

Citation: Gomar-Alba M, Morcillo-Parra MA, Olmo ML. 2015. Response of yeast cells to high glucose 
involves molecular and physiological differences when compared to other osmostress 
conditions.

Full Description: Yeast cells can be affected by several causes of osmotic stress, such as high salt, 
sorbitol or glucose concentrations. The last condition is particularly interesting during 
natural processes where this microorganism participates. Response to osmostress requires 
the HOG (High Osmolarity Glycerol) pathway and several transcription factors, including 
Hot1, which plays a key role in high glucose concentrations. In this work, we describe how
the yeast response to osmotic stress shows differences in accordance with the stress agent
responsible  for it. Compared with other conditions, under high glucose stress, 
delocalization of MAPK (Mitogen-Activated Protein Kinase) Hog1 is slower, induction of 
HOT1 expression is higher and Msn2/4 transcription factors are involved to a lesser extent
. The transcriptomic analyses carried out with samples incubated for 30 min in the 
presence of high glucose or sorbitol reveal the presence of two functional categories with
a differential expression between these conditions: glycogen biosynthesis and mobilization
, and membrane-anchored proteins. We present data to demonstrate that the cells treated 
with 20% (w/v) (1.11 M) glucose contain higher chitin levels and are more sensitive to 
calcofluor white and ethanol.

PMID: 26048894

GEO ID: GSE59659.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE59659.final.pcl	Response of yeast cells to high glucose involves molecular and physiological differences when compared to other osmostress conditions.	4	osmotic stress|carbon utilization	2

File last modified: Thu Apr 25 2024
GSE68816.mapped.pcl	20% (w/v) Glucose osmotic stress (YP20) vs 1M Sorbitol osmotic stress (YPS) in a hot1delta mutant	4	osmotic stress	2

File last modified: Thu Apr 25 2024

---


# Goossens_2015_PMID_25873380

Citation: Goossens KV, Ielasi FS, Nookaew I, Stals I, Alonso-Sarduy L, Daenen L, Van Mulders SE, 
Stassen C, van Eijsden RG, Siewers V, Delvaux FR, Kasas S, Nielsen J, Devreese B, Willaert
RG. 2015. Molecular mechanism of flocculation self-recognition in yeast and its role in 
mating and survival.

Full Description: We studied the flocculation mechanism at the molecular level by determining the atomic 
structures of N-Flo1p and N-Lg-Flo1p in complex with their ligands. We show that they have
similar ligand binding mechanisms but distinct carbohydrate specificities and affinities, 
which are determined by the compactness of the binding site. We characterized the glycans 
of Flo1p and their role in this binding process and demonstrate that glycan-glycan 
interactions significantly contribute to the cell-cell adhesion mechanism. Therefore, the 
extended flocculation mechanism is based on the self-interaction of Flo proteins and this
interaction is established in two stages, involving both glycan-glycan and protein-glycan 
interactions. The crucial role of calcium in both types of interaction was demonstrated: 
Ca(2+) takes part in the binding of the carbohydrate to the protein, and the glycans 
aggregate only in the presence of Ca(2+). These results unify the generally accepted 
lectin hypothesis with the historically first-proposed Ca(2+)-bridge hypothesis. 
Additionally, a new role  of cell flocculation is demonstrated; i.e., flocculation is 
linked to cell conjugation and mating, and survival chances consequently increase 
significantly  by spore formation and by introduction of genetic variability. The role of 
Flo1p  in mating was demonstrated by showing that mating efficiency is increased when 
cells flocculate and by differential transcriptome analysis of flocculating versus 
nonflocculating cells in a low-shear environment (microgravity). The results show that a 
multicellular clump (floc) provides a uniquely organized multicellular ultrastructure that
provides a suitable microenvironment to induce  and perform cell conjugation and mating. 
IMPORTANCE: Yeast cells can form multicellular clumps under adverse growth conditions that
protect cells from harsh environmental stresses. The floc formation is based on the self-
interaction of Flo proteins via an N-terminal PA14 lectin domain. We have focused on the 
flocculation mechanism and its role. We found that carbohydrate specificity and affinity 
are determined by the accessibility of the binding site of the Flo proteins where the 
external loops in the ligand-binding domains are involved in glycan recognition 
specificity. We demonstrated that, in addition to the Flo lectin-glycan interaction, 
glycan-glycan interactions also contribute significantly to cell-cell recognition and 
interaction. Additionally, we show that flocculation provides a uniquely organized 
multicellular ultrastructure that is suitable to induce and accomplish cell mating. 
Therefore, flocculation is an important mechanism to enhance long-term yeast survival.

PMID: 25873380

GEO ID: GSE64468.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE64468.final.pcl	Molecular mechanism of flocculation self-recognition in yeast and its role in mating and survival.	17	flocculation	1

File last modified: Thu Apr 25 2024

---


# Hodges_2015_PMID_25971662

Citation: Hodges AJ, Gallegos IJ, Laughery MF, Meas R, Tran L, Wyrick JJ. 2015. Histone Sprocket 
Arginine Residues Are Important for Gene Expression, DNA Repair, and Cell Viability in 
Saccharomyces cerevisiae.

Full Description: A critical feature of the intermolecular contacts that bind DNA to the histone octamer is 
the series of histone arginine residues that insert into the DNA minor groove at each 
superhelical location where the minor groove faces the histone octamer. One of these 
sprocket arginine residues, histone H4 R45, significantly affects chromatin structure in 
vivo and is lethal when mutated to alanine or cysteine in Saccharomyces cerevisiae (
budding yeast). However, the roles of the remaining sprocket arginine residues (H3 R63, H3
R83, H2A R43, H2B R36, H2A R78,  H3 R49) in chromatin structure and other cellular 
processes have not been well characterized. We have genetically characterized mutations in
each of these histone residues when introduced either singly or in combination to yeast 
cells.  We find that pairs of arginine residues that bind DNA adjacent to the DNA exit/
entry sites in the nucleosome are lethal in yeast when mutated in combination and cause a 
defect in histone occupancy. Furthermore, mutations in individual residues compromise 
repair of UV-induced DNA lesions and affect gene expression and cryptic transcription. 
This study reveals simple rules for how the location and structural mode of DNA binding 
influence the biological function of  each histone sprocket arginine residue.

PMID: 25971662

GEO ID: GSE68159.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE68159.final.pcl	Histone Sprocket Arginine Residues Are Important for Gene Expression, DNA Repair, and Cell Viability in Saccharomyces cerevisiae.	6	chromatin organization|transcription	1

File last modified: Thu Apr 25 2024

---


# Hoffman_2015_PMID_25609572

Citation: Hoffman EA, McCulley A, Haarer B, Arnak R, Feng W. 2015. Break-seq reveals hydroxyurea-
induced chromosome fragility as a result of unscheduled conflict between DNA replication 
and transcription.

Full Description: We have previously demonstrated that in Saccharomyces cerevisiae replication, checkpoint 
inactivation via a mec1 mutation leads to chromosome breakage at replication forks 
initiated from virtually all origins after transient exposure to hydroxyurea (HU), an 
inhibitor of ribonucleotide reductase. Here we sought to  determine whether all 
replication forks containing single-stranded DNA gaps have  equal probability of producing
double-strand breaks (DSBs) when cells attempt to  recover from HU exposure. We devised a 
new methodology, Break-seq, that combines  our previously described DSB labeling with next
generation sequencing to map chromosome breaks with improved sensitivity and resolution. 
We show that DSBs preferentially occur at genes transcriptionally induced by HU. Notably, 
different subsets of the HU-induced genes produced DSBs in MEC1 and mec1 cells as 
replication forks traversed a greater distance in MEC1 cells than in mec1 cells during 
recovery from HU. Specifically, while MEC1 cells exhibited chromosome breakage at stress-
response transcription factors, mec1 cells predominantly suffered chromosome breakage at 
transporter genes, many of which are the substrates of those transcription factors. We 
propose that HU-induced chromosome  fragility arises at higher frequency near HU-induced 
genes as a result of destabilized replication forks encountering transcription factor 
binding and/or the act of transcription. We further propose that replication inhibitors 
can induce unscheduled encounters between replication and transcription and give rise to 
distinct patterns of chromosome fragile sites.

PMID: 25609572

GEO ID: GSE64446.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE64446.final.pcl	Break-seq reveals hydroxyurea-induced chromosome fragility as a result of unscheduled conflict between DNA replication and transcription.	2	DNA replication, recombination and repair|transcription	1

File last modified: Thu Apr 25 2024

---


# Jung_2015_PMID_26602082

Citation: Jung I, Seo J, Lee HS, Stanton LW, Kim D, Choi JK. 2015. Global mapping of the regulatory 
interactions of histone residues.

Full Description: Histone residues can serve as platforms for specific regulatory function. Here we 
constructed a map of regulatory associations between histone residues and a wide  spectrum
of chromatin regulation factors based on gene expression changes by histone point 
mutations in Saccharomyces cerevisiae. Detailed analyses of this map revealed novel 
associations. Regarding the modulation of H3K4 and K36 methylation by Set1, Set2, or Jhd2
, we proposed a role for H4K91 acetylation in early Pol II elongation, and for H4K16 
deacetylation in late elongation and crosstalk with H3K4 demethylation for gene silencing
. The association of H3K56 with nucleosome positioning suggested that this lysine residue 
and its acetylation might contribute to nucleosome mobility for transcription activation. 
Further insights into chromatin regulation are expected from this approach.

PMID: 26602082

GEO ID: GSE29059.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE29059.final.pcl	Global mapping of the regulatory interactions of histone residues.	127	histone modification	1

File last modified: Thu Apr 25 2024

---


# Khakhina_2015_PMID_25724885

Citation: Khakhina S, Johnson SS, Manoharlal R, Russo SB, Blugeon C, Lemoine S, Sunshine AB, Dunham 
MJ, Cowart LA, Devaux F, Moye-Rowley WS. 2015. Control of Plasma Membrane Permeability by 
ABC Transporters.

Full Description: ATP-binding cassette transporters Pdr5 and Yor1 from Saccharomyces cerevisiae control the 
asymmetric distribution of phospholipids across the plasma membrane as well as serving as 
ATP-dependent drug efflux pumps. Mutant strains lacking these transporter proteins were 
found to exhibit very different resistance phenotypes to two inhibitors of sphingolipid 
biosynthesis that act either late (aureobasidin A [AbA]) or early (myriocin [Myr]) in the 
pathway leading to production of these important plasma membrane lipids. These pdr5Delta 
yor1 strains were highly AbA resistant but extremely sensitive to Myr. We provide evidence
that these phenotypic changes are likely due to modulation of the plasma membrane flippase
complexes, Dnf1/Lem3 and Dnf2/Lem3. Flippases act to move phospholipids from the outer to 
the inner leaflet of the plasma membrane. Genetic analyses indicate that lem3Delta mutant 
strains are highly AbA sensitive and Myr  resistant. These phenotypes are fully epistatic 
to those seen in pdr5Delta yor1 strains. Direct analysis of AbA-induced signaling 
demonstrated that loss of Pdr5  and Yor1 inhibited the AbA-triggered phosphorylation of 
the AGC kinase Ypk1 and its substrate Orm1. Microarray experiments found that a pdr5Delta 
yor1 strain induced a Pdr1-dependent induction of the entire Pdr regulon. Our data support
the view that Pdr5/Yor1 negatively regulate flippase function and activity of the nuclear 
Pdr1 transcription factor. Together, these data argue that the interaction of the ABC 
transporters Pdr5 and Yor1 with the Lem3-dependent flippases regulates permeability of AbA
via control of plasma membrane protein function as seen for the high-affinity tryptophan 
permease Tat2.

PMID: 25724885

GEO ID: GSE65264.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE65264.final.pcl	Control of Plasma Membrane Permeability by ABC Transporters.	8	lipid metabolism|cell wall organization	2

File last modified: Thu Apr 25 2024

---


# Klecker_2015_PMID_26669658

Citation:  Klecker T, Wemmer M, Haag M, Weig A, Bockler S, Langer T, Nunnari J, Westermann B . 2015
 Interaction of MDM33 with mitochondrial inner membrane homeostasis pathways in yeast.

Full Description:  We performed a genome-wide screen for genetic MDM33 interaction partners. Using the 
opposite approach of the well known synthetic dosage lethality (screening for deletion 
strains in which a non-toxic overexpression is associated with a synthetic growth arrest) 
we screened the whole genome for genes that are essential for the growth arrest associated
with MDM33 overexpression. In a pooled collection of viable haploid gene deletion mutants
, of which each is tagged with a unique identifying molecular   bar-code   sequence, the 
overexpression of MDM33 was induced using a galactose responsive promotor. more


.. .

PMID: 26669658

GEO ID: GSE49580.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE49580.mapped.pcl	MDM33 overexpression suppressor screen	24	organelles, biogenesis, structure, and function	1

File last modified: Thu Apr 25 2024

---


# Lopez-Malo_2015_PMID_26194190

Citation: Lopez-Malo M, Garcia-Rios E, Melgar B, Sanchez MR, Dunham MJ, Guillamon JM. 2015. 
Evolutionary engineering of a wine yeast strain revealed a key role of inositol and 
mannoprotein metabolism during low-temperature fermentation.

Full Description: BACKGROUND: Wine produced at low temperature is often considered to improve sensory 
qualities. However, there are certain drawbacks to low temperature fermentations: e.g. low
growth rate, long lag phase, and sluggish or stuck fermentations. Selection and 
development of new Saccharomyces cerevisiae strains  well adapted at low temperature is 
interesting for future biotechnological applications. This study aimed to select and 
develop wine yeast strains that well adapt to ferment at low temperature through 
evolutionary engineering, and to decipher the process underlying the obtained phenotypes. 
RESULTS: We used a pool  of 27 commercial yeast strains and set up batch serial dilution 
experiments to mimic wine fermentation conditions at 12 degrees C. Evolutionary 
engineering was  accomplished by using the natural yeast mutation rate and mutagenesis 
procedures. One strain (P5) outcompeted the others under both experimental conditions and 
was able to impose after 200 generations. The evolved strains showed improved growth  and 
low-temperature fermentation performance compared to the ancestral strain. This 
improvement was acquired only under inositol limitation. The transcriptomic  comparison 
between the evolved and parental strains showed the greatest up-regulation in four 
mannoprotein coding genes, which belong to the DAN/TIR family (DAN1, TIR1, TIR4 and TIR3
). Genome sequencing of the evolved strain revealed the presence of a SNP in the GAA1 gene
and the construction of a site-directed mutant (GAA1 (Thr108)) in a derivative haploid of 
the ancestral strain resulted in improved fermentation performance. GAA1 encodes a GPI 
transamidase complex subunit that adds GPI, which is required for inositol synthesis, to 
newly synthesized proteins, including mannoproteins. CONCLUSIONS: In this study we 
demonstrate the importance of inositol and mannoproteins in yeast adaptation at low 
temperature and the central role of the GAA1 gene by linking both metabolisms.

PMID: 26194190

GEO ID: GSE67428.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE67428.final.pcl	Evolutionary engineering of a wine yeast strain revealed a key role of inositol and mannoprotein metabolism during low-temperature fermentation.	6	evolution	1

File last modified: Thu Apr 25 2024

---


# Mith_2015_PMID_25755096

Citation: Mith O, Benhamdi A, Castillo T, Berge M, MacDiarmid CW, Steffen J, Eide DJ, Perrier V, 
Subileau M, Gosti F, Berthomieu P, Marques L. 2015. The antifungal plant defensin AhPDF1.
1b is a beneficial factor involved in adaptive response to zinc overload when it is 
expressed in yeast cells.

Full Description: Antimicrobial peptides represent an expanding family of peptides involved in innate 
immunity of many living organisms. They show an amazing diversity in their sequence, 
structure, and mechanism of action. Among them, plant defensins are renowned for their 
antifungal activity but various side activities have also been described. Usually, a new 
biological role is reported along with the discovery of a new defensin and it is thus not 
clear if this multifunctionality exists at the  family level or at the peptide level. We 
previously showed that the plant defensin AhPDF1.1b exhibits an unexpected role by 
conferring zinc tolerance to yeast and plant cells. In this paper, we further explored 
this activity using different yeast genetic backgrounds: especially the zrc1 mutant and an
UPRE-GFP reporter yeast strain. We showed that AhPDF1.1b interferes with adaptive cell 
response in the endoplasmic reticulum to confer cellular zinc tolerance. We thus  
highlighted that, depending on its cellular localization, AhPDF1.1b exerts quite  separate
activities: when it is applied exogenously, it is a toxin against fungal and also root 
cells, but when it is expressed in yeast cells, it is a peptide that modulates the 
cellular adaptive response to zinc overload.

PMID: 25755096

GEO ID: GSE63395.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE63395.final.pcl	The antifungal plant defensin AhPDF1.1b is a beneficial factor involved in adaptive response to zinc overload when it is expressed in yeast cells.	8	chemical stimulus|oxidative stress	1

File last modified: Thu Apr 25 2024

---


# Noble_2015_PMID_25947166

Citation: Noble J, Sanchez I, Blondin B. 2015. Identification of new Saccharomyces cerevisiae 
variants of the MET2 and SKP2 genes controlling the sulfur assimilation pathway and the 
production of undesirable sulfur compounds during alcoholic fermentation.

Full Description: BACKGROUND: Wine yeasts can produce undesirable sulfur compounds during alcoholic 
fermentation, such as SO2 and H2S, in variable amounts depending mostly on the yeast 
strain but also on the conditions. However, although sulfur metabolism has  been widely 
studied, some of the genetic determinants of differences in sulfite and/or sulfide 
production between wine yeast strains remain to be identified. In  this study, we used an 
integrated approach to decipher the genetic determinants of variation in the production of
undesirable sulfur compounds. RESULTS: We examined the kinetics of SO2 production by two 
parental strains, one high and one low sulfite producer. These strains displayed similar 
production profiles but only the high-sulfite producer strain continued to produce SO2 in 
the stationary  phase. Transcriptomic analysis revealed that the low-sulfite producer 
strain overexpressed genes of the sulfur assimilation pathway, which is the mark of a 
lower flux through the pathway consistent with a lower intracellular concentration in 
cysteine. A QTL mapping strategy then enabled us to identify MET2 and SKP2 as the genes 
responsible for these phenotypic differences between strains and we identified new 
variants of these genes in the low-sulfite producer strain. MET2 influences the 
availability of a metabolic intermediate, O-acetylhomoserine, whereas SKP2 affects the 
activity of a key enzyme of the sulfur assimilation branch of the pathway, the APS kinase
, encoded by MET14. Furthermore, these genes also affected the production of propanol and 
acetaldehyde. These pleiotropic effects are probably linked to the influence of these 
genes on interconnected pathways and to the chemical reactivity of sulfite  with other 
metabolites. CONCLUSIONS: This study provides new insight into the regulation of sulfur 
metabolism in wine yeasts and identifies variants of MET2 and SKP2 genes, that control the
activity of both branches of the sulfur amino acid synthesis pathway and modulate sulfite/
sulfide production and other related  phenotypes. These results provide novel targets for 
the improvement of wine yeast strains.

PMID: 25947166

GEO ID: GSE55083.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE55083.final.pcl	Identification of new Saccharomyces cerevisiae variants of the MET2 and SKP2 genes controlling the sulfur assimilation pathway and the production of undesirable sulfur compounds during alcoholic fermentation.	2	fermentation|stationary phase entry	2

File last modified: Thu Apr 25 2024

---


# Oliveira_2015_PMID_25888284

Citation: Oliveira AP, Dimopoulos S, Busetto AG, Christen S, Dechant R, Falter L, Haghir Chehreghani
M, Jozefczuk S, Ludwig C, Rudroff F, Schulz JC, Gonzalez A, Soulard A, Stracka D, 
Aebersold R, Buhmann JM, Hall MN, Peter M, Sauer U, Stelling J. 2015. Inferring causal 
metabolic signals that regulate the dynamic TORC1-dependent transcriptome.

Full Description: Cells react to nutritional cues in changing environments via the integrated action of 
signaling, transcriptional, and metabolic networks. Mechanistic insight into signaling 
processes is often complicated because ubiquitous feedback loops obscure causal 
relationships. Consequently, the endogenous inputs of many nutrient signaling pathways 
remain unknown. Recent advances for system-wide experimental data generation have 
facilitated the quantification of signaling systems, but the integration of multi-level 
dynamic data remains challenging. Here, we co-designed dynamic experiments and a 
probabilistic, model-based method  to infer causal relationships between metabolism, 
signaling, and gene regulation. We analyzed the dynamic regulation of nitrogen metabolism 
by the target of rapamycin complex 1 (TORC1) pathway in budding yeast. Dynamic 
transcriptomic, proteomic, and metabolomic measurements along shifts in nitrogen quality 
yielded  a consistent dataset that demonstrated extensive re-wiring of cellular networks 
during adaptation. Our inference method identified putative downstream targets of TORC1 
and putative metabolic inputs of TORC1, including the hypothesized glutamine signal. The 
work provides a basis for further mechanistic studies of nitrogen metabolism and a general
computational framework to study cellular processes.

PMID: 25888284

GEO ID: GSE54850.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE54850.final.pcl	Regulation of TORC1-dependent transcriptome; Dynamic mRNA gene expression during a nutritional downshift from glutamine to proline	15	nitrogen utilization	1

File last modified: Thu Apr 25 2024
GSE54851.final.pcl	Regulation of TORC1-dependent transcriptome; Dynamic mRNA gene expression following rapamycin treatment	14	chemical stimulus|signaling	1

File last modified: Thu Apr 25 2024
GSE54844.final.pcl	Regulation of TORC1-dependent transcriptome; Dynamic mRNA gene expression during a nutritional upshift from proline to glutamine	14	nitrogen utilization	1

File last modified: Thu Apr 25 2024

---


# Pastor-Flores_2015_PMID_26391581

Citation: Pastor-Flores D, Ferrer-Dalmau J, Bahi A, Boleda M, Biondi RM, Casamayor A. 2015. 
Depletion of yeast PDK1 orthologs triggers a stress-like transcriptional response.

Full Description: BACKGROUND: Pkh proteins are the PDK1 orthologs in S. cerevisiae. They have redundant and 
essential activity and are responsible for the phosphorylation of several members of the 
AGC family of protein kinases. Pkh proteins have been involved in several cellular 
functions, including cell wall integrity and endocytosis. However the global expression 
changes caused by their depletion are  still unknown. RESULTS: A doxycycline-repressible 
tetO7 promoter driving the expression of PKH2 in cells carrying deletions of the PKH1 and 
PKH3 genes allowed us to progressively deplete cells from Pkh proteins when treated with 
doxycycline. Global gene expression analysis indicate that depletion of Pkh results in the
up-regulation of genes involved in the accumulation of glycogen and also of those related 
to stress responses. Moreover, genes involved in the ion transport were quickly down-
regulated when the levels of Pkh decreased. The reduction in the mRNA levels required for 
protein translation, however, was only  observed after longer doxycycline treatment (24 h
). We uncovered that Pkh is important for the proper transcriptional response to heat 
shock, and is mostly required for the effects driven by the transcription factors Hsf1 and
Msn2/Msn4,  but is not required for down-regulation of the mRNA coding for ribosomal 
proteins. CONCLUSIONS: By using the tetO7 promoter we elucidated for the first time the 
transcriptomic changes directly or indirectly caused by progressive depletion of Pkh. 
Furthermore, this system enabled the characterization of the transcriptional response 
triggered by heat shock in wild-type and Pkh-depleted cells, showing that about 40 % of 
the observed expression changes were, to some degree, dependent on Pkh.

PMID: 26391581

GEO ID: GSE32623.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE32623.final.pcl	Depletion of yeast PDK1 orthologs triggers a stress-like transcriptional response.	8	protein phosphorylation|heat shock	2

File last modified: Thu Apr 25 2024

---


# Renaud-Young_2015_PMID_25701288

Citation: Renaud-Young M, Lloyd DC, Chatfield-Reed K, George I, Chua G, Cobb J. 2015. The NuA4 
complex promotes translesion synthesis (TLS)-mediated DNA damage tolerance.

Full Description: Lesions in DNA can block replication fork progression, leading to its collapse and gross 
chromosomal rearrangements. To circumvent such outcomes, the DNA damage tolerance (DDT) 
pathway becomes engaged, allowing the replisome to bypass a lesion and complete S phase. 
Chromatin remodeling complexes have been implicated  in the DDT pathways, and here we 
identify the NuA4 remodeler, which is a histone  acetyltransferase, to function on the 
translesion synthesis (TLS) branch of DDT.  Genetic analyses in Saccharomyces cerevisiae 
showed synergistic sensitivity to MMS when NuA4 alleles, esa1-L254P and yng2Delta, were 
combined with the error-free bypass mutant ubc13Delta. The loss of viability was less 
pronounced when NuA4 complex mutants were disrupted in combination with error-prone/TLS 
factors, such as rev3Delta, suggesting an epistatic relationship between NuA4 and error-
prone bypass. Consistent with cellular viability measurements, replication  profiles after
exposure to MMS indicated that small regions of unreplicated DNA or damage were present to
a greater extent in esa1-L254P/ubc13Delta mutants, which persist beyond the completion of 
bulk replication compared to esa1-L254P/rev3Delta. The critical role of NuA4 in error-
prone bypass is functional even after the bulk of replication is complete. Underscoring 
this observation, when Yng2 expression is restricted specifically to G2/M of the cell  
cycle, viability and TLS-dependent mutagenesis rates were restored. Lastly, disruption of 
HTZ1, which is a target of NuA4, also resulted in mutagenic rates of reversion on level 
with esa1-L254P and yng2Delta mutants, indicating that the  histone variant H2A.Z 
functions in vivo on the TLS branch of DDT.

PMID: 25701288

GEO ID: GSE66176.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE66176.final.pcl	The NuA4 complex promotes translesion synthesis (TLS)-mediated DNA damage tolerance.	16	DNA replication, recombination and repair|DNA damage stimulus	2

File last modified: Thu Apr 25 2024

---


# Robinson_2015_PMID_26130709

Citation: Robinson DG, Wang JY, Storey JD. 2015. A nested parallel experiment demonstrates 
differences in intensity-dependence between RNA-seq and microarrays.

Full Description: Understanding the differences between microarray and RNA-Seq technologies for measuring 
gene expression is necessary for informed design of experiments and choice of data 
analysis methods. Previous comparisons have come to sometimes contradictory conclusions, 
which we suggest result from a lack of attention to the intensity-dependent nature of 
variation generated by the technologies. To examine this trend, we carried out a parallel 
nested experiment performed simultaneously on the two technologies that systematically 
split variation into four stages (treatment, biological variation, library preparation and
chip/lane noise), allowing a separation and comparison of the sources of variation in a 
well-controlled cellular system, Saccharomyces cerevisiae. With this novel dataset, we 
demonstrate that power and accuracy are more dependent on per-gene read depth in RNA-Seq 
than they are on fluorescence intensity in microarrays. However, we carried out 
quantitative PCR validations which indicate that microarrays may demonstrate greater 
systematic bias in low-intensity genes than in RNA-seq.

PMID: 26130709

GEO ID: GSE65365.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE65365.final.pcl	A nested parallel experiment demonstrates differences in intensity-dependence between RNA-seq and microarrays.	16	carbon utilization	2

File last modified: Thu Apr 25 2024

---


# Sameith_2015_PMID_26700642

Citation:  Sameith K, Amini S, Groot Koerkamp MJ, van Leenen D, Brok M, Brabers N, Lijnzaad P, van 
Hooff SR, Benschop JJ, Lenstra TL, Apweiler E, van Wageningen S,  Snel B, Holstege FC, 
Kemmeren P . 2015. A high-resolution gene expression atlas of epistasis between gene-
specific transcription factors exposes potential mechanisms for genetic interactions.

Full Description:  Contains gene expression profiles of yeast single and double deletion mutants of gene-
specific transcription factors. Genetic interactions were studied by comparing gene 
expression changes of double mutants with gene expression changes in the respective single
mutants. Pairs of gene-specific transcription factors were chosen based on previous 
evidence for epistasis, including synthetic genetic interactions as well as common DNA 
binding


 .

PMID: 26700642

GEO ID: GSE42536.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE42536.mapped.pcl	Transcription profiling by array of yeast single and double deletion mutants of gene-specific transcription factors	287	genetic interaction	2

File last modified: Thu Apr 25 2024

---


# Schosserer_2015_PMID_25635753

Citation: Schosserer M, Minois N, Angerer TB, Amring M, Dellago H, Harreither E, Calle-Perez A, 
Pircher A, Gerstl MP, Pfeifenberger S, Brandl C, Sonntagbauer M, Kriegner A, Linder A, 
Weinhausel A, Mohr T, Steiger M, Mattanovich D, Rinnerthaler M, Karl T, Sharma S, Entian 
KD, Kos M, Breitenbach M, Wilson IB, Polacek N, Grillari-Voglauer R, Breitenbach-Koller L,
Grillari J. 2015. Methylation of ribosomal RNA by NSUN5 is a conserved mechanism 
modulating organismal lifespan.

Full Description: Several pathways modulating longevity and stress resistance converge on translation by 
targeting ribosomal proteins or initiation factors, but whether this involves 
modifications of ribosomal RNA is unclear. Here, we show that reduced levels of the 
conserved RNA methyltransferase NSUN5 increase the lifespan and stress resistance in yeast
, worms and flies. Rcm1, the yeast homologue of NSUN5, methylates C2278 within a conserved
region of 25S rRNA. Loss of Rcm1 alters the structural conformation of the ribosome in 
close proximity to C2278, as well as translational fidelity, and favours recruitment of a 
distinct subset of oxidative stress-responsive mRNAs into polysomes. Thus, rather than 
merely being a static molecular machine executing translation, the ribosome exhibits 
functional diversity by modification of just a single rRNA nucleotide, resulting  in an 
alteration of organismal physiological behaviour, and linking rRNA-mediated translational 
regulation to modulation of lifespan, and differential stress response.

PMID: 25635753

GEO ID: GSE63030.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE63030.final.pcl	Methylation of ribosomal RNA by NSUN5 is a conserved mechanism modulating organismal lifespan.	24	cell aging|stress	2

File last modified: Thu Apr 25 2024

---


# Suzuki_2015_PMID_25609182

Citation: Suzuki T, Iwahashi Y. 2015. Low toxicity of deoxynivalenol-3-glucoside in microbial cells.

Full Description: Host plants excrete a glucosylation enzyme onto the plant surface that changes mycotoxins 
derived from fungal secondary metabolites to glucosylated products. Deoxynivalenol-3-
glucoside (DON3G) is synthesized by grain uridine diphosphate-glucosyltransferase, and is 
found worldwide, although information on  its toxicity is lacking. Here, we conducted 
growth tests and DNA microarray analysis to elucidate the characteristics of DON3G. The 
Saccharomyces cerevisiae  PDR5 mutant strain exposed to DON3G demonstrated similar growth 
to the dimethyl sulfoxide control, and DNA microarray analysis revealed limited 
differences. Only 10 genes were extracted, and the expression profile of stress response 
genes was  similar to that of DON, in contrast to metabolism genes like SER3, which 
encodes  3-phosphoglycerate dehydrogenase. Growth tests with Chlamydomonas reinhardtii 
also showed a similar growth rate to the control sample. These results suggest that DON3G 
has extremely low toxicity to these cells, and the glucosylation of mycotoxins is a useful
protective mechanism not only for host plants, but also for other species.

PMID: 25609182

GEO ID: GSE63663.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE63663.final.pcl	Low toxicity of deoxynivalenol-3-glucoside in microbial cells.	6	chemical stimulus|stress	1

File last modified: Thu Apr 25 2024

---


# Thierry_2015_PMID_25635677

Citation: Thierry A, Khanna V, Creno S, Lafontaine I, Ma L, Bouchier C, Dujon B. 2015. Macrotene 
chromosomes provide insights to a new mechanism of high-order gene amplification in 
eukaryotes.

Full Description: Copy number variation of chromosomal segments is now recognized as a major source of 
genetic polymorphism within natural populations of eukaryotes, as well as a possible cause
of genetic diseases in humans, including cancer, but its molecular bases remain 
incompletely understood. In the baker's yeast Saccharomyces cerevisiae, a variety of low-
order amplifications (segmental duplications) were observed after adaptation to limiting 
environmental conditions or recovery from gene dosage imbalance, and interpreted in terms 
of replication-based mechanisms associated or not with homologous recombination. Here we 
show the emergence of novel high-order amplification structures, with corresponding 
overexpression of embedded genes, during evolution under favourable growth conditions of 
severely unfit yeast cells bearing genetically disabled genomes. Such events form 
massively extended chromosomes, which we propose to call macrotene, whose characteristics 
suggest the products of intrachromosomal rolling-circle type of replication structures, 
probably initiated by increased accidental template switches under important cellular 
stress conditions.

PMID: 25635677

GEO ID: GSE64431.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE64431.final.pcl	Macrotene chromosomes provide insights to a new mechanism of high-order gene amplification in eukaryotes.	4	genome variation	2

File last modified: Thu Apr 25 2024

---


# Vos_2015_PMID_26369953

Citation:  Vos T, de la Torre Cortes P, van Gulik WM, Pronk JT, Daran-Lapujade P . 2015. Growth-rate
dependency of de novo resveratrol production in chemostat cultures of an engineered 
Saccharomyces cerevisiae strain.

Full Description:  Saccharomyces cerevisiae has become a popular host for production of non-native compounds
. The metabolic pathways involved generally require a net input of energy. To maximize the
ATP yield on sugar in S. cerevisiae, industrial cultivation is typically performed in 
aerobic, sugar-limited fed-batch reactors which, due to constraints in oxygen transfer and
cooling capacities, have to be operated at low specific growth rates. more


.. .

PMID: 26369953

GEO ID: GSE65942.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE65942.mapped.pcl	Growth-rate dependency of de novo resveratrol production in chemostat cultures of an engineered Saccharomyces cerevisiae strain	16	stress|synthetic biology	1

File last modified: Thu Apr 25 2024

---


# Wanichthanarak_2015_PMID_26079307

Citation: Wanichthanarak K, Wongtosrad N, Petranovic D. 2015. Genome-wide expression analyses of the
stationary phase model of ageing in yeast.

Full Description: Ageing processes involved in replicative lifespan (RLS) and chronological lifespan (CLS) 
have been found to be conserved among many organisms, including in unicellular Eukarya 
such as yeast Saccharomyces cerevisiae. Here we performed an  integrated approach of 
genome wide expression profiles of yeast at different time points, during growth and 
starvation. The aim of the study was to identify transcriptional changes in those 
conditions by using several different computational analyses in order to propose 
transcription factors, biological networks and metabolic pathways that seem to be relevant
during the process of chronological ageing in yeast. Specifically, we performed 
differential gene expression analysis, gene-set enrichment analysis and network-based 
analysis, and we identified pathways affected in the stationary phase and specific 
transcription factors driving transcriptional adaptations. The results indicate signal 
propagation from G protein-coupled receptors through signaling pathway components and 
other stress and nutrient-induced transcription factors resulting  in adaptation of yeast 
cells to the lack of nutrients by activating metabolism associated with aerobic metabolism
of carbon sources such as ethanol, glycerol and fatty acids. In addition, we found STE12, 
XBP1 and TOS8 as highly connected nodes in the subnetworks of ageing yeast.

PMID: 26079307

GEO ID: GSE55508.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE55508.final.pcl	Genome-wide expression analyses of the stationary phase model of ageing in yeast.	12	cell aging	1

File last modified: Thu Apr 25 2024

---


# Zhang_2015_PMID_25897127

Citation: Zhang Y, Liu ZL, Song M. 2015. ChiNet uncovers rewired transcription subnetworks in 
tolerant yeast for advanced  biofuels conversion.

Full Description: Analysis of rewired upstream subnetworks impacting downstream differential gene expression
aids the delineation of evolving molecular mechanisms. Cumulative statistics based on 
conventional differential correlation are limited for subnetwork rewiring analysis since 
rewiring is not necessarily equivalent to change in correlation coefficients. Here we 
present a computational method ChiNet to quantify subnetwork rewiring by statistical 
heterogeneity that enables detection of potential genotype changes causing altered 
transcription regulation  in evolving organisms. Given a differentially expressed 
downstream gene set, ChiNet backtracks a rewired upstream subnetwork from a super-network 
including gene interactions known to occur under various molecular contexts. We 
benchmarked ChiNet for its high accuracy in distinguishing rewired artificial subnetworks
, in silico yeast transcription-metabolic subnetworks, and rewired transcription 
subnetworks for Candida albicans versus Saccharomyces cerevisiae, against two differential
-correlation based subnetwork rewiring approaches. Then, using transcriptome data from 
tolerant S. cerevisiae strain NRRL Y-50049 and a wild-type intolerant strain, ChiNet 
identified 44 metabolic pathways affected by  rewired transcription subnetworks anchored 
to major adaptively activated transcription factor genes YAP1, RPN4, SFP1 and ROX1, in 
response to toxic chemical challenges involved in lignocellulose-to-biofuels conversion. 
These findings support the use of ChiNet in rewiring analysis of subnetworks where 
differential interaction patterns resulting from divergent nonlinear dynamics abound.

PMID: 25897127

GEO ID: GSE50492.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE50492.final.pcl	ChiNet uncovers rewired transcription subnetworks in tolerant yeast for advanced  biofuels conversion.	10	chemical stimulus|evolution	2

File last modified: Thu Apr 25 2024

---


# Zhou_2015_PMID_26150418

Citation: Zhou Z, Humphryes N, van Eijk P, Waters R, Yu S, Kraehenbuehl R, Hartsuiker E, Reed SH. 
2015. UV induced ubiquitination of the yeast Rad4-Rad23 complex promotes survival by 
regulating cellular dNTP pools.

Full Description: Regulating gene expression programmes is a central facet of the DNA damage response. The 
Dun1 kinase protein controls expression of many DNA damage induced  genes, including the 
ribonucleotide reductase genes, which regulate cellular dNTP pools. Using a combination of
gene expression profiling and chromatin immunoprecipitation, we demonstrate that in the 
absence of DNA damage the yeast Rad4-Rad23 nucleotide excision repair complex binds to the
promoters of certain DNA damage response genes including DUN1, inhibiting their expression
. UV radiation promotes the loss of occupancy of the Rad4-Rad23 complex from the 
regulatory regions of these genes, enabling their induction and thereby controlling the 
production of dNTPs. We demonstrate that this regulatory mechanism, which is dependent on 
the ubiquitination of Rad4 by the GG-NER E3 ligase, promotes UV survival in yeast cells. 
These results support an unanticipated regulatory mechanism that integrates ubiquitination
of NER DNA repair factors with the regulation of the transcriptional response controlling 
dNTP production and cellular survival after UV damage.

PMID: 26150418

GEO ID: GSE23204.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE23204.final.pcl	UV induced ubiquitination of the yeast Rad4-Rad23 complex promotes survival by regulating cellular dNTP pools.	30	DNA damage stimulus|DNA replication, recombination and repair	1

File last modified: Thu Apr 25 2024

---


# Zhu_2015_PMID_26138482

Citation: Zhu X, Chen L, Carlsten JO, Liu Q, Yang J, Liu B, Gustafsson CM. 2015. Mediator tail 
subunits can form amyloid-like aggregates in vivo and affect stress response in yeast.

Full Description: The Med2, Med3 and Med15 proteins form a heterotrimeric subdomain in the budding  yeast 
Mediator complex. This Med15 module is an important target for many gene specific 
transcription activators. A previous proteome wide screen in yeast identified Med3 as a 
protein with priogenic potential. In the present work, we have extended this observation 
and demonstrate that both Med3 and Med15 form amyloid-like protein aggregates under H2O2 
stress conditions. Amyloid formation can also be stimulated by overexpression of Med3 or 
of a glutamine-rich domain present in Med15, which in turn leads to loss of the entire 
Med15 module from Mediator and a change in stress response. In combination with genome 
wide transcription analysis, our data demonstrate that amyloid formation can change the 
subunit composition of Mediator and thereby influence transcriptional output  in budding 
yeast.

PMID: 26138482

GEO ID: GSE65666.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE65666.final.pcl	Mediator tail subunits can form amyloid-like aggregates in vivo and affect stress response in yeast.	6	transcription	1

File last modified: Thu Apr 25 2024

---


# Alff-Tuomala_2016_PMID_26454869

Citation:  Alff-Tuomala S, Salusjarvi L, Barth D, Oja M, Penttila M, Pitkanen JP, Ruohonen L, 
Jouhten P . 2016. Xylose-induced dynamic effects on metabolism and gene expression in 
engineered Saccharomyces cerevisiae in anaerobic glucose-xylose cultures.

Full Description:  Xylose induced effects on metabolism and gene expression during anaerobic growth of an 
engineered Saccharomyces cerevisiae on mixed glucose-xylose medium were quantified. Gene 
expression of S. cerevisiae harbouring an XR-XDH pathway for xylose utilisation was 
analysed from early cultivation when mainly glucose was metabolised, to times when xylose 
was co-consumed in the presence of low glucose concentrations, and finally, to glucose 
depletion and solely xylose being consumed. more


.. .

PMID: 26454869

GEO ID: GSE69966.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE69966.mapped.pcl	Xylose induced dynamic effects on metabolism and gene expression in engineered Saccharomyces cerevisiae in anaerobic glucose-xylose cultures	30	carbon utilization	1

File last modified: Thu Apr 25 2024

---


# Belanger_2016_PMID_26888869

Citation: Belanger KD, Larson N, Kahn J, Tkachev D, Ay A. 2016. Microarray Analysis of Gene 
Expression in Saccharomyces cerevisiae kap108Delta Mutants upon Addition of Oxidative 
Stress.

Full Description: Protein transport between the nucleus and cytoplasm of eukaryotic cells is tightly 
regulated, providing a mechanism for controlling intracellular localization of proteins, 
and regulating gene expression. In this study, we have  investigated the importance of 
nucleocytoplasmic transport mediated by the karyopherin Kap108 in regulating cellular 
responses to oxidative stress in Saccharomyces cerevisiae We carried out microarray 
analyses on wild-type and kap108 mutant cells grown under normal conditions, shortly after
introduction of  oxidative stress, after 1 hr of oxidative stress, and 1 hr after 
oxidative stress was removed. We observe more than 500 genes that undergo a 40% or greater
change  in differential expression between wild-type and kap108Delta cells under at least 
one of these conditions. Genes undergoing changes in expression can be categorized in two 
general groups: 1) those that are differentially expressed between wild-type and 
kap108Delta cells, no matter the oxidative stress conditions; and 2) those that have 
patterns of response dependent upon both the absence of Kap108, and introduction or 
removal of oxidative stress. Gene ontology analysis reveals that, among the genes whose 
expression is reduced in the absence of Kap108 are those involved in stress response and 
intracellular transport, while those overexpressed are largely involved in mating and 
pheromone response.  We also identified 25 clusters of genes that undergo similar patterns
of change in gene expression when oxidative stresses are added and subsequently removed, 
including genes involved in stress response, oxidation-reduction processing, iron 
homeostasis, ascospore wall assembly, transmembrane transport, and cell fusion during 
mating. These data suggest that Kap108 is important for regulating expression of genes 
involved in a variety of specific cell functions.

PMID: 26888869

GEO ID: GSE71068.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE71068.final.pcl	Microarray Analysis of Gene Expression in Saccharomyces cerevisiae kap108Delta Mutants upon Addition of Oxidative Stress.	12	oxidative stress	2

File last modified: Thu Apr 25 2024

---


# Dorfel_2016_PMID_27668839

Citation:  Dorfel MJ, Fang H, Crain J, Klingener M, Weiser J, Lyon GJ . 2016. Proteomic and genomic 
characterization of a yeast model for Ogden syndrome.

Full Description:  Gene expression microarrays accompanying Proteomic and genomic characterization of a 
yeast model for Ogden syndrome by Doerfel et al 2016 in press at Yeast.  Naa10 is a Na-
terminal acetyltransferase that, in a complex with its auxiliary subunit Naa15, co-
translationally acetylates the a-amino group of newly synthetized proteins as they emerge 
from the ribosome. Roughly 40-50% of the human proteome is acetylated by Naa10, rendering 
this an enzyme with one of the most broad substrate ranges known. more


.. .

PMID: 27668839

GEO ID: GSE86358.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE86358.mapped.pcl	Proteomic and genomic characterization of a yeast model for Ogden syndrome [microarray]	16	disease|heat shock	2

File last modified: Thu Apr 25 2024

---


# Garcia-Martinez_2016_PMID_26717982

Citation:  Garcia-Martinez J, Delgado-Ramos L, Ayala G, Pelechano V, Medina DA, Carrasco F, Gonzalez
R, Andres-Leon E, Steinmetz L, Warringer J, Chavez S, Perez-Ortin JE . 2016.  The cellular
growth rate controls overall mRNA turnover, and modulates either transcription or 
degradation rates of particular gene regulons.

Full Description:  We analyzed the effect of PUB1 and NGR1 on the transcription rates, mRNA stabilities and 
mRNA levels by doing GRO experiments in a deletion mutants comparing with a wild type in 
both glucose (YPD) and galactose (YPGal) media. Some data for the wt in YPGal are not 
included because were already in GEO database:  GSE1002 accession number


 .

PMID: 26717982

GEO ID: GSE63729.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE63729.mapped.pcl	Genomic Run On (GRO): determination of the nascent transcriptional rates and mRNA levels in several yeast mutants (PUB1 and NGR1)	8	RNA catabolism|transcriptional regulation	2

File last modified: Thu Apr 25 2024
GSE65225.mapped.pcl	Genomic Run On (GRO): determination of the nascent transcriptional rates in a series of Hrp1 mutants.	6	RNA catabolism|transcriptional regulation	1

File last modified: Thu Apr 25 2024
GSE57467.mapped.pcl	Genomic Run On (GRO): determination of the nascent transcriptional rates and mRNA levels in several yeast mutants.	18	RNA catabolism|transcriptional regulation	1

File last modified: Thu Apr 25 2024

---


# Garcia-Martinez_2016_PMID_27648972

Citation:  Garcia-Martinez J, Troule K, Chavez S, Perez-Ortin JE . 2016. Growth rate controls mRNA 
turnover in steady and non-steady states.

Full Description: We analyzed the effect of the elimination of the cap-binding protein Cbc1 on transcription
rate and mRNA amount during the response to osmotic stress using the Genomic Run-on assay.

PMID: 27648972

GEO ID: GSE72356.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE72356.mapped.pcl	Determination of transcriptional rates and mRNA levels during osmotic stress  in S. cerevisiae wild type and cbc1/sto1 mutant cells using Genomic Run On (GRO)	20	osmotic stress	2

File last modified: Thu Apr 25 2024

---


# Garrido-Godino_2016_PMID_27001033

Citation:  Garrido-Godino AI, Garcia-Lopez MC, Garcia-Martinez J, Pelechano V, Medina DA, Perez-
Ortin JE, Navarro F . 2016. Rpb1 foot mutations demonstrate a major role of Rpb4 in mRNA 
stability during stress situations in yeast.

Full Description: The goal of the project was to study the effects on transctiptome of some Rpb1 foot 
mutations that block Rpb4/7 assembly.

PMID: 27001033

GEO ID: GSE65283.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE65283.mapped.pcl	Transcriptomic study in RNA pol II foot mutants	18	RNA catabolism|stress	1

File last modified: Thu Apr 25 2024

---


# Gavalda_2016_PMID_27035147

Citation:  Gavalda S, Santos-Pereira JM, Garcia-Rubio ML, Luna R, Aguilera A . 2016.  Excess of Yra1
RNA-Binding Factor Causes Transcription-Dependent Genome Instability, Replication 
Impairment and Telomere Shortening.

Full Description:  Transcription is a major obstacle for replication fork progression and a cause of genome 
instability. Such instability increases in mutants with an imbalance proportion of Yra1, a
component of THO/TREX. We used microarrays to the global impact of Yra1 overexpression and
found that no general impact was observed


 .

PMID: 27035147

GEO ID: GSE68487.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE68487.mapped.pcl	Expression data from Saccharomyces cerevisae strains overexpressing RNA-binding hnRNP Yra1	6	transcriptional regulation	1

File last modified: Thu Apr 25 2024

---


# Golla_2016_PMID_27734932

Citation:  Golla U, Joseph D, Tomar RS . 2016.  Combined Transcriptomics and Chemical-Genetics 
Reveal Molecular Mode of Action of Valproic acid, an Anticancer Molecule using Budding 
Yeast Model
 .

Full Description:  Valproic acid (VA) is a small-chain branched fatty acid, widely used as anticonvulsant, 
and mood stabilizer to treat psychiatric illness. Valproic acid is also known to inhibit 
the histone deacetylases (HDACs), which makes it as a potent antitumor agent in alone or 
in combination with other cytotoxic drugs. Beside its conventional activities, valproic 
acid reported to have much broader, complicated effects and affect many complex 
physiological processes. more


.. .

PMID: 27734932

GEO ID: GSE62400.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE62400.mapped.pcl	Genome-wide expression profiling after Valproic acid treatment in Saccharomyces cerevisiae	4	chemical stimulus|disease	1

File last modified: Thu Apr 25 2024

---


# Hinkle_2016_PMID_26606276

Citation: Hinkle KL, Anderson CC, Forkey B, Griffin J, Cone K, Vitzthum C, Olsen D. 2016. Exposure 
to the lampricide 3-trifluoromethyl-4-nitrophenol results in increased expression of 
carbohydrate transporters in Saccharomyces cerevisiae.

Full Description: The lampricide 3-trifluoromethyl-4-nitrophenol (TFM) is used to control sea lamprey (
Petromyzon marinus) populations in freshwater lakes. Although TFM can have sublethal and 
lethal effects, little is known about gene expression changes  with TFM exposure. 
Microarray analysis was used to determine differential gene expression over 4 h of 
exposure in Saccharomyces cerevisiae. Among the most significantly up-regulated genes were
regulators of carbohydrate transport, including HXT1, HXT3, HXT4, IMA5, MIG2, and YKR075C
. Environ Toxicol Chem 2016;35:1727-1732. (c) 2015 SETAC.

PMID: 26606276

GEO ID: GSE74259.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE74259.final.pcl	Exposure to the lampricide 3-trifluoromethyl-4-nitrophenol results in increased expression of carbohydrate transporters in Saccharomyces cerevisiae.	13	chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Kugler_2016_PMID_27245696

Citation:  Kugler KG, Jandric Z, Beyer R, Klopf E, Glaser W, Lemmens M, Shams M, Mayer K, Adam G, 
Schuller C . 2016. Ribosome quality control is a central protection mechanism for yeast 
exposed to deoxynivalenol and trichothecin.

Full Description:  Trichothecenes, like deoxynivalenol (DON) are secondary metabolites of Fusarium species, 
and are major pollutants in cereal food and feed products. They inhibit eukaryotic protein
synthesis, however with only partly resolved consequences for the cellular homeostasis. 
Here we report a systematic functional investigation of the effects of DON on exponential 
phase growing yeast cells using either short doses of DON or continuos treatment with DON


 .

PMID: 27245696

GEO ID: GSE75462.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE75462.mapped.pcl	Ribosome quality control is a central protection mechanism for yeast exposed to Deoxynivalenol and Trichothecin	23	chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Lu_2016_PMID_27812096

Citation:  Lu YJ, Swamy KB, Leu JY . 2016. Experimental Evolution Reveals Interplay between Sch9 and
Polyploid Stability in Yeast.

Full Description:  Transcriptional change by profiling yeast cells, comparing polyploid S. cerevisiae cells 
at 1000 generations with ancestral cells. Transcriptional change by profiling yeast cells
, comparing both ancestral and evolved cells with diploid cells, stands for control


 .

PMID: 27812096

GEO ID: GSE76596.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE76596.mapped.pcl	Whole genome gene expression analysis of evolved polyploid Saccharomyces cerevisiae cell lines at 36 degrees Celsius	30	evolution|ploidy	2

File last modified: Thu Apr 25 2024
GSE76597.mapped.pcl	Karyotype analysis and whole genome gene expression analysis of evolved polyploid Saccharomyces cerevisiae cell lines	63	evolution|ploidy	2

File last modified: Thu Apr 25 2024

---


# Marsit_2016_PMID_26549518

Citation:  Marsit S, Sanchez I, Galeote V, Dequin S . 2016. Horizontally acquired oligopeptide 
transporters favour adaptation of Saccharomyces cerevisiae wine yeast to oenological 
environment.

Full Description:  To assess the impact of a higher oligopeptide assimilation mediated by Fot1/2, 
oligopeptides transporters acquired by HGT on wine yeast cell metabolism in winemaking 
conditions,we carried out a comparison of transcriptomic profiles of the wine wild type 
strain 59A and the deletion mutant of FOT1/2 genes during exponential growth


 .

PMID: 26549518

GEO ID: GSE62446.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE62446.mapped.pcl	Transcriptomic analysis of the metabolic impact of higher oligopeptide assimilation mediated by horizontally acquired oligopeptide transporters Fot1/2 in wine strains.	15	fermentation|protein trafficking, localization and degradation	1

File last modified: Thu Apr 25 2024

---


# Navarro-Tapia_2016_PMID_26925053

Citation:  Navarro-Tapia E, Nana RK, Querol A, Perez-Torrado R . 2016. Ethanol Cellular Defense 
Induce Unfolded Protein Response in Yeast.

Full Description:  Ethanol is a valuable industrial product and a common metabolite used by many cell types
. However, this molecule produces high levels of cytotoxicity affecting cellular 
performance at several levels. In the presence of ethanol, cells must adjust some of their
components, such as the membrane lipids to maintain homeostasis. In the case of 
microorganism as Saccharomyces cerevisiae, ethanol is one of the principal products of 
their metabolism and is the main stress factor during fermentation. more


.. .

PMID: 26925053

GEO ID: GSE44863.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE44863.mapped.pcl	Ethanol cellular defense induce unfolded protein response in yeast	8	fermentation|stress	2

File last modified: Thu Apr 25 2024

---


# Oeser_2016_PMID_26800527

Citation:  Oeser ML, Amen T, Nadel CM, Bradley AI, Reed BJ, Jones RD, Gopalan J, Kaganovich D, 
Gardner RG . 2016. Dynamic Sumoylation of a Conserved Transcription Corepressor Prevents 
Persistent Inclusion Formation during Hyperosmotic Stress.

Full Description:  Cells are often exposed to physical or chemical stresses that can damage the structures 
of essential biomolecules. Stress-induced cellular damage can become deleterious if not 
managed appropriately. Rapid and adaptive responses to stresses are therefore crucial for 
cell survival. In eukaryotic cells, different stresses trigger post-translational 
modification of proteins with the small ubiquitin-like modifier SUMO. more


.. .

PMID: 26800527

GEO ID: GSE57476.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE57476.mapped.pcl	Dynamic Sumoylation of a Conserved Transcription Corepressor Prevents Persistent Inclusion Formation during Hyperosmotic Stress	23	osmotic stress|protein modification	2

File last modified: Thu Apr 25 2024

---


# Ogawa_2016_PMID_27698120

Citation: Ogawa T, Tsubakiyama R, Kanai M, Koyama T, Fujii T, Iefuji H, Soga T, Kume K, Miyakawa T, 
Hirata D, Mizunuma M. 2016. Stimulating S-adenosyl-l-methionine synthesis extends lifespan
via activation of  AMPK.

Full Description: Dietary restriction (DR), such as calorie restriction (CR) or methionine (Met) restriction
, extends the lifespan of diverse model organisms. Although studies have identified 
several metabolites that contribute to the beneficial effects of  DR, the molecular 
mechanism underlying the key metabolites responsible for DR regimens is not fully 
understood. Here we show that stimulating S-adenosyl-l-methionine (AdoMet) synthesis 
extended the lifespan of the budding yeast Saccharomyces cerevisiae The AdoMet synthesis-
mediated beneficial metabolic effects, which resulted from consuming both Met and ATP, 
mimicked CR. Indeed, stimulating AdoMet synthesis activated the universal energy-sensing 
regulator Snf1, which is the S. cerevisiae ortholog of AMP-activated protein kinase (AMPK
), resulting in lifespan extension. Furthermore, our findings revealed that S-adenosyl-l-
homocysteine contributed to longevity with a higher accumulation of  AdoMet only under the
severe CR (0.05% glucose) conditions. Thus, our data uncovered molecular links between Met
metabolites and lifespan, suggesting a unique function of AdoMet as a reservoir of Met and
ATP for cell survival.

PMID: 27698120

GEO ID: GSE76206.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE76206.final.pcl	Stimulating S-adenosyl-l-methionine synthesis extends lifespan via activation of  AMPK.	4	cell aging	1

File last modified: Thu Apr 25 2024

---


# Rollero_2016_PMID_26861624

Citation:  Rollero S, Mouret JR, Sanchez I, Camarasa C, Ortiz-Julien A, Sablayrolles JM, Dequin S . 
2016. Key role of lipid management in nitrogen and aroma metabolism in an evolved wine 
yeast strain.

Full Description:  An evolved strain (ECA5) presented two successive yields of bioconversion of higher 
alcohols to acetate esters, while its parental strain (EC1118) had a constant yield 
through the fermentation. Transcriptomic analysis was performed during wine fermentation 
in SM330 containing 8mg/L phytosterols, at  35 g/l and at 70 g/l of CO2 released. For ECA5
, this corresponds to a sample before and one after the change of bioconversion yield. 
more


.. .

PMID: 26861624

GEO ID: GSE68354.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE68354.mapped.pcl	Comparative transcriptomic analysis of an evolved and an ancestral  wine yeast strains presenting differences in the bioconversion of higher alcohols to acetate esters	10	evolution|fermentation	1

File last modified: Thu Apr 25 2024

---


# Sariki_2016_PMID_27718307

Citation:  Sariki SK, Sahu PK, Golla U, Singh V, Azad GK, Tomar RS . 2016.  Sen1, the homolog of 
human Senataxin, is critical for cell survival through regulation of redox homeostasis, 
mitochondrial function, and the TOR pathway in Saccharomyces cerevisiae.

Full Description:  In Saccharomyces cerevisiae, Sen1 is a 252-kDa, nuclear superfamily-1 RNA/DNA helicase 
that encoded by an essential gene SEN1 (Senataxin). It is an important component of the 
Nrd1p-Nab3p-Sen1p (NRD1) complex that regulates the transcriptional termination of most 
non-coding and some coding transcripts at RNA polymerase pause sites. Sen1 specifically 
interacts with Rnt1p (RNase III), an endoribonuclease, and with Rpb1p (Rpo21p), a subunit 
of RNA polymerase II, through its N-terminal domain (NTD), which is a critical element of 
the RNA-processing machinery. more


.. .

PMID: 27718307

GEO ID: GSE75447.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE75447.mapped.pcl	Comparative transcriptome analysis of basal gene expression in Wild-type and Sen1_N mutant of Saccharomyces cerevisiae	4	disease|DNA replication, recombination and repair	1

File last modified: Thu Apr 25 2024

---


# Tarrant_2016_PMID_27807005

Citation: Tarrant DJ, Stirpe M, Rowe M, Howard MJ, von der Haar T, Gourlay CW. 2016. Inappropriate 
expression of the translation elongation factor 1A disrupts genome  stability and 
metabolism.

Full Description: The translation elongation factor eEF1A is one of the most abundant proteins found within 
cells, and its role within protein synthesis is well documented. Levels of eEF1A are 
tightly controlled, with inappropriate expression linked to oncogenesis. However, the 
mechanisms by which increased eEF1A expression alters cell behaviour are unknown. Our 
analyses in yeast suggest that elevation of eEF1A levels leads to stabilisation of the 
spindle pole body and changes in nuclear organisation. Elevation of the eEF1A2 isoform 
also leads to altered nuclear morphology in cultured human cells, suggesting a conserved 
role in maintaining genome stability. Gene expression and metabolomic analyses reveal that
the level  of eEF1A is crucial for the maintenance of metabolism and amino acid levels in 
yeast, most likely because of its role in the control of vacuole function. Increased 
eEF1A2 levels trigger lysosome biogenesis in cultured human cells, also suggesting a 
conserved role within metabolic control mechanisms. Taken together,  our data suggest that
the control of eEF1A levels is important for the maintenance of a number of cell functions
beyond translation and that its de-regulation might contribute to its oncogenic properties.

PMID: 27807005

GEO ID: GSE89286.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE89286.final.pcl	Inappropriate expression of the translation elongation factor 1A disrupts genome  stability and metabolism.	6	translational regulation|carbon utilization|amino acid utilization	1

File last modified: Thu Apr 25 2024

---


# Yadav_2016_PMID_27222810

Citation: Yadav KK, Rajasekharan R. 2016. Microarray data analyses of yeast RNA Pol I subunit RPA12 
deletion strain.

Full Description: The ribosomal RNA (rRNA) biosynthesis is the most energy consuming process in all living 
cells and the majority of total transcription activity is dedicated for synthesizing rRNA
. The cells may adjust the synthesis of rRNA with the availability of resources. rRNA is 
mainly synthesized by RNA polymerase I that is composed of 14 subunits. Deletion of RPA12
, 14, 39 and 49 are viable. RPA12 is a  very small protein (13.6 kDa), and the amount of 
protein in the cells is very high (12,000 molecules per cell), but the role of this 
protein is unknown in other cellular metabolic processes (Kulak et al., 2014 [1]). RPA12 
consists of two zinc-binding domains and it is required for the termination of rRNA 
synthesis (Mullem et al., 2002 [2]). Deletions of RPA12 in Saccharomyces cerevisiae and 
Schizosaccharomyces pombe cause a conditional growth defect (Nogi et al., 1993 [3]). In S
. pombe, C-terminal deletion behaves like wild-type (Imazawa et al., 2001 [4]). This 
prompted us to investigate in detail the physiological role of RPA12 in S. cerevisiae, we 
performed the microarray of rpa12 strain and deposited into Gene Expression Omnibus under 
GSE68731. The analysis of microarray data revealed that the expression of major cellular 
metabolism genes is high. The amino acid biosynthesis, nonpolar lipid biosynthesis and 
glucose metabolic genes  are highly expressed. The analyses also revealed that the rpa12 
cells have an uncontrolled synthesis of cell metabolites, so RPA12 could be a master 
regulator  for whole cellular metabolism.

PMID: 27222810

GEO ID: GSE68731.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE68731.final.pcl	Microarray data analyses of yeast RNA Pol I subunit RPA12 deletion strain.	3	transcription	1

File last modified: Thu Apr 25 2024

---


# Zeng_2016_PMID_27485516

Citation:  Zeng WY, Tang YQ, Gou M, Xia ZY, Kida K . 2016. Transcriptomes of a xylose-utilizing 
industrial flocculating Saccharomyces cerevisiae strain cultured in media containing 
different sugar sources.

Full Description: The molecular basis for glucose and xylose fermentation by industrial Saccharomyces 
cerevisiae is of interest to promote bioethanol production We used microarrays to 
investigate the transcriptional difference of a industrial strain cultured in both single 
sugar media and a mixed sugar medium of glucose and xylose.

PMID: 27485516

GEO ID: GSE80748.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE80748.mapped.pcl	Transcriptomes of a xylose-utilizing industrial flocculating Saccharomyces cerevisiae strain cultured in media containing different sugar sources	8	fermentation|flocculation	1

File last modified: Thu Apr 25 2024

---


# Baptista_2017_PMID_28918903

Citation:  Baptista T, Grnberg S, Minoungou N, Koster MJE, Timmers HTM, Hahn S, Devys D, Tora L . 
2017. SAGA Is a General Cofactor for RNA Polymerase II Transcription.

Full Description:  The SAGA co-activator has been implicated in the regulation of a smal subset of genes in 
budding yeast in transcriptomic analyses performed in steady-state levels of RNA We used 
microarrays to analyse newly-synthesized RNA in several mutants for the SAGA complex to 
disclose and readdress the impact of this complex in RNA Polymerase II transcription.

PMID: 28918903

GEO ID: GSE96849.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE96849.mapped.pcl	SAGA Is a General Cofactor for RNA Polymerase II Transcription	68	transcriptional regulation	1

File last modified: Thu Apr 25 2024

---


# Celaj_2017_PMID_28705884

Citation:  Celaj A, Schlecht U, Smith JD, Xu W, Suresh S, Miranda M, Aparicio AM, Proctor M, Davis 
RW, Roth FP, St Onge RP . 2017. Quantitative analysis of protein interaction network 
dynamics in yeast
 .

Full Description:  To understand the principles underlying protein-protein interaction (PPI) complex changes
in response to external perturbations, we created a highly multiplexed version of the 
murine dihydrofolate reductase protein complementation assay (mDHFR PCA) in Saccharomyces 
cerevisiae, allowing quantitative PPI complex profiling in vivo. We investigated the 
effects of 14 different conditions (including small molecules, abiotic stress factors, and
nutrient composition) on a total of 1383 PPIs. More than half of PPIs (758) were found to 
be variable, and their Gene Ontology (GO) annotations were found to be informative of both
the nature of the perturbation within each condition, as well as the overall variability 
of the interactions across conditions. Many perturbations triggered network changes 
characterized by large connected modules centered around highly connected proteins ('hubs
'), suggesting that cellular control of a few proteins (e.g., by mRNA levels) can induce 
widespread PPI remodeling. Under a diauxic shift from glucose to ethanol as the main 
carbon source, we found a striking relationship between PPI changes measured by our assay 
and those predicted by mRNA expression under a simple law of mass action based model


 .

PMID: 28705884

GEO ID: GSE72423.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE72423.mapped.pcl	Systematic identification of changes in the yeast protein interaction network in response to environmental, chemical, and genetic perturbation [transcriptome data] 	10	diauxic shift	1

File last modified: Thu Apr 25 2024

---


# Chen_2017_PMID_27387517

Citation:  Chen L, Wang M, Hou J, Fu J, Shen Y, Liu F, Zhang Z, Bao X . 2017. HAL2 overexpression 
induces iron acquisition in bdf1_ cells and enhances their salt resistance.

Full Description:  The strain bdf1_+HAL2 improved salt resistance of bdf1. To gain further insight into the 
mechanism of bdf1 salt sensitivity, DNA microarray analysis was performed to determine the
reason for the salt sensitivity of bdf1 cells and the process of how HAL2 overexpression 
and HDA1 deletion improves salt resistance. Transcriptomic analysis under salt treatment (
0.5 mol.L-1 NaCl for 45 min) was performed using four different strains: bdf1, W303, bdf1_
+HAL2 and bdf1hda1. The transcription of 1,393 genes were significantly changed( > 2-fold
) in bdf1 compared with W303 upon NaCl stress (0.5 mol.L-1 NaCl for 45 min). Only 108 
genes were significantly changed ( > 2-fold) in bdf1_+HAL2 compared with bdf1 upon NaCl 
stress (0.5 mol.L-1 NaCl for 45 min). Only 88 genes were significantly changed ( > 2-fold
) in bdf1_+HAL2 compared with bdf1 upon NaCl stress (0.5 mol.L-1 NaCl for 45 min

). .

PMID: 27387517

GEO ID: GSE75828.sfp.pcl

PCL filename	short description	# conditions	tags	# channels
GSE75828.sfp.pcl	HAL2 overexpression induces iron acquisition and enhanced salt resistance in bdf1delta	5	osmotic stress	1

File last modified: Thu Apr 25 2024

---


# Chen_2017_PMID_28798664

Citation:  Chen X, Bisschops MMM, Agarwal NR, Ji B, Shanmugavel KP, Petranovic D . 2017. Interplay 
of Energetics and ER Stress Exacerbates Alzheimer's Amyloid-_ (A_) Toxicity in Yeast
 .

Full Description:  Alzheimers disease (AD) is a progressive neurodegenerative disorder. Oligomers of Amyloid
-_ peptides (A_) are thought to play a pivotal role in AD pathogenesis, yet the mechanisms
involved remain unclear. Two major isoforms of A_ associated with AD are A_40 and A_42, 
the latter being more prone to form oligomers and toxic. Humanized yeast models are 
currently applied to unravel the cellular mechanisms behind A_ toxicity. Here, we took a 
systems biology approach to study two yeast AD models which expressed either A_40 or A_42 
in bioreactor cultures. Strict control of oxygen availability and culture pH, strongly 
affected the chronological lifespan and reduced confounding effects of variations during 
cell growth. Reduced growth rates and biomass yields were observed upon expression of A_42
, indicating a redirection of energy from growth to maintenance. Quantitative physiology 
analyses furthermore revealed reduced mitochondrial functionality and ATP generation in 
A_42 expressing cells, which matched with observed aberrant fragmented mitochondrial 
structures. Genome-wide expression levels analysis showed that A_42 expression triggers 
strong ER stress and unfolded protein responses (UPR). Expression of A_40 induced only 
mild ER stress, leading to activation of UPR target genes that cope with misfolded 
proteins, which resulted in hardly affected physiology. The combination of well-controlled
cultures and AD yeast models strengthen our understanding of how cells translate different
levels of A_ toxicity signals into particular cell fate programs, and further enhance 
their role as a discovery platform to identify potential therapies


 .

PMID: 28798664

GEO ID: GSE94793.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE94793.mapped.pcl	Expression data from Saccharomyces cerevisiae	24	disease	1

File last modified: Thu Apr 25 2024

---


# Chirichella_2017_PMID_28092690

Citation:  Chirichella M, Lisi S, Fantini M, Goracci M, Calvello M, Brandi R, Arisi I, D'Onofrio M, 
Di Primio C, Cattaneo A . 2017. Post-translational selective intracellular silencing of 
acetylated proteins with de novo selected intrabodies
 .

Full Description:  Post-translational modifications (PTMs) of proteins, such as acetylation or 
phosphorylation, represent a huge untapped source of targets of great biological and 
therapeutic potential, but their validation can only be indirect. PTMs cannot be 
selectively hit by the powerful nucleic-acid based interference approaches. Antibodies 
represent a class of molecules that could be exploited to selectively target PTMs. However
, currently it is not possible to systematically derive antibodies that selectively bind 
PTMs of a native protein and work intracellularly to achieve a native PTM-selective 
interference. We developed the Post-translational Intracellular Silencing Antibody 
technology (P.I.S.A.), a new method for the selection of intrabodies targeting the native 
PTM version of proteins and their use for PTM targeting in cells. We used PISA to select 
an intrabody against acetylated-K9-H3 Histone (ScFv-58F). Microarray study is aimed at the
investigation of ScFv-58F-specific changes in gene expression in yeast, in comparison to 
yeast expressing a control intrabody (ScFv-112A, anti-acetyl-Integrase, selected with PISA
), and to a wild type yeast


 .

PMID: 28092690

GEO ID: GSE86954.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE86954.mapped.pcl	Transcriptomic study on yeast expressing an intracellular antibody against acetylated-K9-H3 Histone	12	protein modification|protein trafficking, localization and degradation	1

File last modified: Thu Apr 25 2024

---


# Cho_2017_PMID_28934013

Citation:  Cho CY, Motta FC, Kelliher CM, Deckard A, Haase SB . 2017. Reconciling conflicting models
for global control of cell-cycle transcription.

Full Description: Transcript dynamics in mitotic exit mutants in the S. cerevisiae BF264-15D strain 
background. We examined the extent to which periodic cell-cycle transcription persisted in
cells arrested in anaphase with intermediate level of B-cyclins.

PMID: 28934013

GEO ID: GSE96997.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE96997.mapped.pcl	Investigating global transcript dynamics in mitotically arrested budding yeast cells	144	transcriptional regulation	1

File last modified: Thu Apr 25 2024

---


# Choi_2017_PMID_28329151

Citation:  Choi KM, Hong SJ, van Deursen JM, Kim S, Kim KH, Lee CK . 2017. Caloric Restriction and 
Rapamycin Differentially Alter Energy Metabolism in Yeast
 .

Full Description:  Rapamycin (RM), a drug that inhibits the mechanistic target of rapamycin (mTOR) pathway 
and responds to nutrient availability, seemingly mimics the effects of caloric restriction
(CR) on healthy life span. However, the extent of the mechanistic overlap between RM and 
CR remains incompletely understood. Here, we compared the impact of CR and RM on cellular 
metabolic status. Both regimens maintained intracellular ATP through the chronological 
aging process and showed enhanced mitochondrial capacity. Comparative transcriptome 
analysis showed that CR had a stronger impact on global gene expression than RM. We 
observed a like impact on the metabolome and identified distinct metabolites affected by 
CR and RM. CR severely reduced the level of energy storage molecules including glycogen 
and lipid droplets, whereas RM did not. RM boosted the production of enzymes responsible 
for the breakdown of glycogen and lipid droplets. Collectively, these results provide 
insights into the distinct energy metabolism mechanisms induced by CR and RM, suggesting 
that these two anti-aging regimens might extend life span through distinctive pathways


 .

PMID: 28329151

GEO ID: GSE82324.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE82324.mapped.pcl	Expression data from yeasts under non-caloric restriction (NR), caloric restriction (CR) and rapamycin treatment (RM) 	9	chemical stimulus|metabolism	1

File last modified: Thu Apr 25 2024

---


# Deshpande_2017_PMID_28435838

Citation:  Deshpande AA, Bhatia M, Laxman S, Bachhawat AK . 2017. Thiol trapping and metabolic 
redistribution of sulfur metabolites enable cells to overcome cysteine overload.

Full Description:  Excess cysteine (and cystine) is known to be toxic in organisms. Due to the absence of 
cysteine dioxygenase (involved in degradation of excess cysteine in humans and pathogenic 
fungi) in non pathogenic fungi such as S. cerevisiae, mechanism of cysteine (and cystine) 
tolerance is yet to be addressed


 .

PMID: 28435838

GEO ID: GSE87794.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE87794.mapped.pcl	Physiological response of S. cerevisiae excess cysteine or cystine	8	sulfur utilization|amino acid metabolism	2

File last modified: Thu Apr 25 2024

---


# Driscoll_2017_PMID_28560281

Citation:  Driscoll HE, Murry JM, English EL, Hunter TC, Pivarski K, Dolci ED . 2017. Microarray 
data and gene expression statistics for <i>Saccharomyces cerevisiae</i> exposed to 
simulated asbestos mine drainage.

Full Description: Investigation of global gene expression changes in Saccharomyces cerevisiae strain NRRL Y-
12632 (ATCC 18824) grown in media made with asbestos mine tailings-laden water compared to
the control grown in media made with double distilled water.

PMID: 28560281

GEO ID: GSE89875.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE89875.mapped.pcl	Expression data from budding yeast exposed to simulated asbestos mine drainage	7	chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Duc_2017_PMID_28922393

Citation:  Duc C, Pradal M, Sanchez I, Noble J, Tesnire C, Blondin B . 2017. A set of nutrient 
limitations trigger yeast cell death in a nitrogen-dependent manner during wine alcoholic 
fermentation.

Full Description:  In order to asses yeast EC1118 strain expression changes during wine alcoholic 
fermentation triggered by various nutrient starvations, this experiment describes the gene
expression under micronutrient starvations that lead to yeast cell death (oleic acid 
starvation, ergosterol starvation, pantothenic acid starvation and nicotinic starvation) 
or allow the maintenance of yeast viability (nitrogen starvation

) .

PMID: 28922393

GEO ID: GSE95152.sfp.pcl

PCL filename	short description	# conditions	tags	# channels
GSE95152.sfp.pcl	Transcriptomic response of the yeast Saccharomyces cerevisiae EC1118 during wine alcoholic fermentation under unusual nutrient starvations	72	fermentation|stress|starvation	1

File last modified: Thu Apr 25 2024

---


# Golla_2017_PMID_29228701

Citation:  Golla U, Swagatika S, Chauhan S, Tomar RS . 2017.  A systematic assessment of chemical, 
genetic, and epigenetic factors influencing the activity of anticancer drug KP1019 (FFC14A,

). .

Full Description:  KP1019 (trans-[tetrachlorobis(1H-indazole) ruthenate(III)]) is a ruthenium complex that 
exhibited anti-cancer activity in several in vitro and in vivo studies. KP1019 was even 
efficient against cancer cells that were resistant to other chemotherapeutic agents and 
thus emerged as a promising anti-cancer drug without dose-limiting cytotoxicity. However, 
the molecular mechanisms of its action are elusive. So, the present study was designed to 
understand the cellular processes that were affected by KP1019 treatment using budding 
yeast as a model organism through the transcriptomics approach. The genome-wide 
transcriptome was analyzed in wild-type yeast cells upon treatment with a sub-lethal dose 
of KP1019 (50g/ml). The functional enrichment analysis of KP1019 transcriptome revealed 
many of cellular processes that were compromised and thus probable mechanisms of action


 .

PMID: 29228701

GEO ID: GSE76985.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE76985.mapped.pcl	Genome-wide monitoring of gene expression in Saccharomyces cerevisiae upon KP1019 treatment	4	chemical stimulus|metabolism and metabolic regulation	1

File last modified: Thu Apr 25 2024

---


# Klopf_2017_PMID_28025392

Citation: Klopf E, Schmidt HA, Clauder-Munster S, Steinmetz LM, Schuller C. 2017. INO80 represses 
osmostress induced gene expression by resetting promoter proximal nucleosomes.

Full Description: The conserved INO80 chromatin remodeling complex is involved in regulation of DNA damage 
repair, replication and transcription. It is commonly recruited to the transcription start
region and contributes to the establishment of promoter-proximal nucleosomes. We find a 
substantial influence of INO80 on nucleosome dynamics and gene expression during stress 
induced transcription. Transcription induced by osmotic stress leads to genome-wide 
remodeling of promoter proximal nucleosomes. INO80 function is required for timely return 
of evicted nucleosomes to the 5 end of induced genes. Reduced INO80 function in Arp8-
deficient cells leads to correlated prolonged transcription and nucleosome eviction. INO80
and the related complex SWR1 regulate incorporation of the H2A.Z  isoform at promoter 
proximal nucleosomes. However, H2A.Z seems not to influence osmotic stress induced gene 
regulation. Furthermore, we show that high rates of transcription promote INO80 
recruitment to promoter regions, suggesting a connection between active transcription and 
promoter proximal nucleosome remodeling. In addition, we find that absence of INO80 
enhances bidirectional promoter activity at highly induced genes and expression of a 
number of stress induced transcripts. We suggest that INO80 has a direct repressive role 
via promoter proximal nucleosome remodeling to limit high levels of transcription in  
yeast.

PMID: 28025392

GEO ID: GSE78766.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE78766.final.pcl	INO80 represses osmostress induced gene expression by resetting promoter proximal nucleosomes.	8	chromatin organization|osmotic stress	1

File last modified: Thu Apr 25 2024

---


# Plateau_2017_PMID_28303947

Citation:  Plateau P, Saveanu C, Lestini R, Dauplais M, Decourty L, Jacquier A, Blanquet S, Lazard M,
. 2017. Exposure to selenomethionine causes selenocysteine misincorporation and protein 
aggregation in Saccharomyces cerevisiae.

Full Description: Growth assay in the presence of Selenomethionine that uses the barcoded collections of 
yeast gene modification (deletion or DamP) to identify strains that are hypersensitive to 
the presence of the aminoacid.

PMID: 28303947

GEO ID: GSE84340.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE84340.mapped.pcl	Saccharomyces cerevisiae deletion and DAmP collections chemogenomic screen with Selenomethionine	8	chemical stimulus|stress	2

File last modified: Thu Apr 25 2024

---


# Schiavone_2017_PMID_29085340

Citation:  Schiavone M, Djean S, Sieczkowski N, Castex M, Dague E, Franois JM . 2017.  Integration 
of Biochemical, Biophysical and Transcriptomics Data for Investigating the Structural and 
Nanomechanical Properties of the Yeast Cell Wall.

Full Description:  The yeast cell is surrounded by a cell wall conferring protection and resistance to 
environmental conditions that can be harmful. Identify the molecular cues (genes) which 
shape the biochemical composition and the nanomechanical properties of the cell wall and 
the links between these two parameters represent a major issue in the understanding of the
biogenesis and the molecular assembly of this essential cellular structure, which may have
consequences in diverse biotechnological applications. We addressed this question in two 
ways. Firstly, we compared the biochemical and biophysical properties using atomic force 
microscopy (AFM) methods of 4 industrial strains with the laboratory sequenced strain 
BY4743 and used transcriptome data of these strains to infer biological hypothesis about 
differences of these properties between strains. This comparative approach showed a 4-6-
fold higher hydrophobicity of industrial strains that was correlated to higher expression 
of genes encoding adhesin and adhesin-like proteins and not to their higher mannans 
content. The second approach was to employ a multivariate statistical analysis to identify
highly correlated variables among biochemical, biophysical and genes expression data. 
Accordingly, we found a tight association between hydrophobicity and adhesion events that 
positively correlated with a set of 22 genes in which the main enriched GO function was 
the sterol metabolic process. We also identified a strong association of beta-1,3-glucans 
with contour length that corresponds to the extension of mannans chains upon pulling the 
mannosyl units with the lectin-coated AFM tips. This association was positively correlated
with a group of 27 genes in which the seripauperin multigene family was highly documented 
and  negatively connected with a set of 23 genes whose main GO biological process was  
sulfur assimilation/cysteine biosynthetic process. On the other hand, the elasticity 
modulus was found weakly associated with levels of beta-1,6-glucans, and this biophysical 
variable was positively correlated with a set of genes implicated in microtubules 
polymerization, tubulin folding and mitotic organization


 .

PMID: 29085340

GEO ID: GSE103392.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE103392.mapped.pcl	 Integration of biochemical, biophysical and transcriptomics data for investigating the structural and nanomechanical properties of the yeast cell wall 	16	cell wall organization	1

File last modified: Thu Apr 25 2024

---


# Tai_2017_PMID_28495531

Citation:  Tai A, Kamei Y, Mukai Y . 2017. The forkhead-like transcription factor (Fhl1p) maintains 
yeast replicative lifespan by regulating ribonucleotide reductase 1 (RNR1) gene 
transcription.

Full Description:  Forkhead transcription factor FHL1 gene was required for replicative lifespan as well as 
cell proliferation in yeast. In this study, to see how Fhl1p determines the lifespan, we 
performed a DNA microarray analysis of heterozygous diploid strain deleted for FHL1 and, 
from the Fhl1p-target genes, screened for a lifespan-related gene. Transcriptomic profiles
showed large increases and moderate decreases in transcripts in the fhl1/FHL1 mutant. 
Pathway analysis suggested upregulation of most of carbon metabolisms (glycolysis, 
gluconeogenesis, and TCA cycle) and downregulation of translation (ribosomal proteins and 
translation factors) in the FHL1-deficient mutant. We found a remarkable downregulation of
a set of phosphate metabolism genes and ribonucleotide reductase large subunit genes, RNR1
and RNR3. We demonstrated that Fhl1p regulates RNR1 gene transcription to maintain dNTP 
levels, modulating longevity against replication stress and genomic instability


 .

PMID: 28495531

GEO ID: GSE95591.sfp.pcl

PCL filename	short description	# conditions	tags	# channels
GSE95591.sfp.pcl	Yeast heterozygous FHL1 deletion mutant	1	cell aging	2

File last modified: Thu Apr 25 2024

---


# Warfield_2017_PMID_28918900

Citation: Warfield L, Ramachandran S, Baptista T, Devys D, Tora L, Hahn S. 2017. Transcription of 
Nearly All Yeast RNA Polymerase II-Transcribed Genes Is Dependent on Transcription Factor 
TFIID.

Full Description: Previous studies suggested that expression of most yeast mRNAs is dominated by either 
transcription factor TFIID or SAGA. We re-examined the role of TFIID by rapid depletion of
S. cerevisiae TFIID subunits and measurement of changes in nascent transcription. We find 
that transcription of nearly all mRNAs is strongly dependent on TFIID function. Degron-
dependent depletion of Taf1, Taf2, Taf7, Taf11, and Taf13 showed similar transcription 
decreases for genes in the Taf1-depleted, Taf1-enriched, TATA-containing, and TATA-less 
gene classes. The magnitude of TFIID dependence varies with growth conditions, although 
this variation is similar genome-wide. Many studies have suggested differences in gene-
regulatory mechanisms between TATA and TATA-less genes, and these differences have been 
attributed in part to differential dependence on SAGA or TFIID. Our work indicates that 
TFIID participates in expression of nearly all yeast mRNAs and that differences in 
regulation between these two gene categories  is due to other properties.

PMID: 28918900

GEO ID: GSE96830.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE96830.final.pcl	Transcription of Nearly All Yeast RNA Polymerase II-Transcribed Genes Is Dependent on Transcription Factor TFIID.	16	transcription	1

File last modified: Thu Apr 25 2024

---


# Yeh_2017_PMID_28934473

Citation: Yeh CS, Chang SL, Chen JH, Wang HK, Chou YC, Wang CH, Huang SH, Larson A, Pleiss JA, Chang
WH, Chang TH. 2017. The conserved AU dinucleotide at the 5' end of nascent U1 snRNA is 
optimized for  the interaction with nuclear cap-binding-complex.

Full Description: Splicing is initiated by a productive interaction between the pre-mRNA and the U1 snRNP, 
in which a short RNA duplex is established between the 5' splice site of a pre-mRNA and 
the 5' end of the U1 snRNA. A long-standing puzzle has been why the  AU dincucleotide at 
the 5'-end of the U1 snRNA is highly conserved, despite the absence of an apparent role in
the formation of the duplex. To explore this conundrum, we varied this AU dinucleotide 
into all possible permutations and analyzed the resulting molecular consequences. This led
to the unexpected findings that the AU dinucleotide dictates the optimal binding of cap-
binding complex (CBC) to the 5' end of the nascent U1 snRNA, which ultimately influences  
the utilization of U1 snRNP in splicing. Our data also provide a structural interpretation
as to why the AU dinucleotide is conserved during evolution.

PMID: 28934473

GEO ID: GSE97015.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE97015.final.pcl	The conserved AU dinucleotide at the 5' end of nascent U1 snRNA is optimized for  the interaction with nuclear cap-binding-complex.	2	mRNA processing	2

File last modified: Thu Apr 25 2024

---


# Zeng_2017_PMID_28004152

Citation: Zeng WY, Tang YQ, Gou M, Sun ZY, Xia ZY, Kida K. 2017. Comparative transcriptomes reveal 
novel evolutionary strategies adopted by Saccharomyces cerevisiae with improved xylose 
utilization capability.

Full Description: Production of ethanol from xylose by recombinant Saccharomyces cerevisiae is suboptimal 
with slow fermentation rate, compared with that from glucose. In this  study, a strain-
expressing Scheffersomyces stipitis xylose reductase-xylitol dehydrogenase (XR-XDH) 
pathway was subjected to adaptive evolution on xylose; this approach generated populations
with the significantly improved cell growth and ethanol production rate. Mutants were 
isolated, and the best one was used for sporulation to generate eight stable mutant 
strains with improved xylose fermentation ability. They were used in a microarray assay to
study the molecular basis of the enhanced phenotype. The enriched transcriptional 
differences among the eight mutant strains and the native strain revealed novel responses 
to xylose, which likely contributes to the improved xylose utilization. The upregulated 
vitamin B1 and B6 biosynthesis indicated that thiamine served as an important cofactor in 
xylose metabolism and may alleviate the redox stress. The increased expression of genes 
involved in sulfur amino acid biosynthesis and the  decreased expression of genes related 
to Fe(II) transport may alleviate redox stress as well. Meanwhile, it was remarkable that 
several glucose-repressible genes, including genes of the galactose metabolism, 
gluconeogenesis, and ethanol  catabolism, had a lower expression level after adaptive 
evolution. Concomitantly, the expression levels of two regulators of the glucose signaling
pathway, Rgs2 and Sip4, decreased, indicating a reshaped signaling pathway to xylose after
adaptive evolution. Our findings provide new targets for construction of a superior 
bioethanol producing strain through inverse metabolic engineering.

PMID: 28004152

GEO ID: GSE80822.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE80822.final.pcl	Comparative transcriptomes reveal novel evolutionary strategies adopted by Saccharomyces cerevisiae with improved xylose utilization capability.	12	carbon utilization|evolution|fermentation	1

File last modified: Thu Apr 25 2024

---


# Zhu_2017_PMID_28499955

Citation:  Zhu X, Zou S, Li Y, Liang Y. . 2017. Transcriptomic analysis of Saccharomyces cerevisiae 
upon honokiol treatment.

Full Description:  Honokiol (HNK), one of the main medicinal components in Magnolia officinalis, possesses 
antimicrobial activity against a variety of pathogenic bacteria and fungi.S. cerevisiae is
a model eukaryote used for investigating the cellular and molecular mechanisms of anti-
fungal drugs. To explore the molecular mechanism of its anti-fungal activity, we 
determined the effects of HNK on the mRNA expression profile of Saccharomyces cerevisiae 
using a DNA microarray approach


 .

PMID: 28499955

GEO ID: GSE94945.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE94945.mapped.pcl	Expression data from Saccharomyces cerevisiae upon honokiol treatment	4	chemical stimulus|metal or metalloid ion stress	1

File last modified: Thu Apr 25 2024

---


# de_Jonge_2017_PMID_27979920

Citation:  de Jonge WJ, O'Duibhir E, Lijnzaad P, van Leenen D, Groot Koerkamp MJ, Kemmeren P, 
Holstege FC . 2017. Molecular mechanisms that distinguish TFIID housekeeping from 
regulatable SAGA promoters.

Full Description:  An important distinction is frequently made between constitutively expressed housekeeping
genes versus regulated genes. Although generally characterized by different DNA elements, 
chromatin architecture and cofactors, it is not known to what degree promoter classes 
strictly follow regulatability rules and which molecular mechanisms dictate such 
differences. We show that SAGA-dominated/TATA-box promoters are more responsive to changes
in the amount of activator, even compared to TFIID/TATA-like promoters that depend on the 
same activator Hsf1. Regulatability is therefore an inherent property of promoter class. 
Further analyses show that SAGA/TATA-box promoters are more dynamic because TBP 
recruitment through SAGA is susceptible to removal by Mot1. In addition, the nucleosome 
configuration upon activator depletion shifts on SAGA/TATA-box promoters and seems less 
amenable to preinitiation complex formation. The results explain the fundamental 
difference between housekeeping and regulatable genes, revealing an additional facet of 
combinatorial control: an activator can elicit a different response dependent on core 
promoter class


 .

PMID: 27979920

GEO ID: GSE81480.mapped.pcl

PCL filename	short description	# conditions	tags	# channels
GSE81480.mapped.pcl	MOT1-expression: Molecular mechanisms that distinguish TFIID housekeeping from regulatable SAGA promoters	16	transcriptional regulation	2

File last modified: Thu Apr 25 2024
GSE81479.mapped.pcl	HSF1-expression: Molecular mechanisms that distinguish TFIID housekeeping from regulatable SAGA promoters	22	transcriptional regulation	2

File last modified: Thu Apr 25 2024

---


# Chia_2018_PMID_30206180

Citation: Chia SZ, Lai YW, Yagoub D, Lev S, Hamey JJ, Pang CNI, Desmarini D, Chen Z, Djordjevic JT, 
Erce MA, Hart-Smith G, Wilkins MR. 2018. Knockout of the Hmt1p Arginine Methyltransferase 
in Saccharomyces cerevisiae Leads to the Dysregulation of Phosphate-associated Genes and 
Processes.

Full Description: Hmt1p is the predominant arginine methyltransferase in Saccharomyces cerevisiae Its 
substrate proteins are involved in transcription, transcriptional regulation, 
nucleocytoplasmic transport and RNA splicing. Hmt1p-catalyzed methylation can also 
modulate protein-protein interactions. Hmt1p is conserved from unicellular eukaryotes 
through to mammals where its ortholog, PRMT1, is lethal upon knockout. In yeast, however, 
the effect of knockout on the transcriptome and proteome has not been described. 
Transcriptome analysis revealed downregulation of phosphate-responsive genes in hmt1Delta
, including acid phosphatases PHO5, PHO11, and PHO12, phosphate transporters PHO84 and 
PHO89 and the vacuolar transporter chaperone VTC3 Analysis of the hmt1Delta proteome 
revealed decreased abundance of phosphate-associated proteins including phosphate 
transporter Pho84p, vacuolar alkaline phosphatase Pho8p, acid phosphatase Pho3p and 
subunits of the vacuolar transporter chaperone complex Vtc1p, Vtc3p and Vtc4p. Consistent 
with this, phosphate homeostasis was dysregulated in hmt1Delta cells, showing decreased 
extracellular phosphatase levels and decreased total Pi in phosphate-depleted medium. In 
vitro, we showed that transcription factor Pho4p can be methylated at  Arg-241, which 
could explain phosphate dysregulation in hmt1Delta if interplay exists with 
phosphorylation at Ser-242 or Ser-243, or if Arg-241 methylation affects the capacity of 
Pho4p to homodimerize or interact with Pho2p. However, the Arg-241 methylation site was 
not validated in vivo and the localization of a  Pho4p-GFP fusion in hmt1Delta was not 
different from wild type. To our knowledge, this is the first study to reveal an 
association between Hmt1p and phosphate homeostasis and one which suggests a regulatory 
link between S-adenosyl methionine and intracellular phosphate.

PMID: 30206180

GEO ID: GSE99869.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE99869.final.pcl	Knockout of the Hmt1p Arginine Methyltransferase in Saccharomyces cerevisiae Leads to the Dysregulation of Phosphate-associated Genes and Processes.	7	phosphorus utilization	1

File last modified: Thu Apr 25 2024

---


# Gil_2018_PMID_29611082

Citation: Gil FN, Goncalves AC, Becker JD, Viegas CA. 2018. Comparative analysis of transcriptomic 
responses to sub-lethal levels of six environmentally relevant pesticides in Saccharomyces
cerevisiae.

Full Description: Accidental spills and misuse of pesticides may lead to current and/or legacy environmental
contamination and may pose concerns regarding possible risks towards non-target microbes 
and higher eukaryotes in ecosystems. The present study was aimed at comparing 
transcriptomic responses to effects of sub-lethal levels of six environmentally relevant 
pesticide active substances in the Saccharomyces cerevisiae eukaryotic model. The 
insecticide carbofuran, the fungicide pyrimethanil and the herbicides alachlor, S-
metolachlor, diuron and methyl(4-chloro-2-methylphenoxy)acetate were studied. Some are 
currently used agricultural pesticides, while others are under restricted utilization or 
banned  in Europe and/or North America albeit being used in other geographical locations. 
In the present work transcriptional profiles representing genome-wide responses in a 
standardized yeast population upon 2 h of exposure to concentrations of each compound 
exerting equivalent toxic effects, i.e., inhibition of growth by 20% relative to the 
untreated control cells, were examined. Hierarchical clustering and Venn analyses of the 
datasets of differentially expressed genes pointed out transcriptional patterns 
distinguishable between the six active substances. Functional enrichment analyses allowed 
predicting mechanisms of pesticide toxicity and response to pesticide stress in the yeast 
model. In general, variations in transcript numbers of selected genes assessed by Real-
Time quantitative reverse transcription polymerase chain reaction confirmed microarray 
data and correlated well with growth inhibitory effects. A possible biological relevance 
of mechanistic predictions arising from these comparative transcriptomic analyses is 
discussed in the context of better understanding potential modes of action and adverse 
side-effects of pesticides.

PMID: 29611082

GEO ID: GSE53125

PCL filename	short description	# conditions	tags	# channels
GSE53125_s30.final.pcl	Comparative analysis of transcriptomic responses to sub-lethal levels of six environmentally relevant pesticides in Saccharomyces cerevisiae.	30	stress|chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Ahmed_2019_PMID_31144305

Citation: Ahmed K, Carter DE, Lajoie P. 2019. Hyperactive TORC1 sensitizes yeast cells to 
endoplasmic reticulum stress by compromising cell wall integrity.

Full Description: Accumulation of misfolded proteins in the endoplasmic reticulum (ER) activates the 
unfolded protein response (UPR). Here, we investigated how the target of rapamycin complex
1 (TORC1) signaling cascade acts in parallel with the UPR to regulate ER stress 
sensitivity. Using Saccharomyces cerevisiae, we found that TORC1 signaling is attenuated 
during ER stress and constitutive activation of TORC1 increases sensitivity to ER 
stressors independently of the UPR. Transcriptome analysis revealed that TORC1 
hyperactivation results in cell wall remodelling. Conversely, hyperactive TORC1 sensitizes
cells to cell wall stressors, including the antifungal caspofungin. Elucidating the 
crosstalk between the UPR, cell wall integrity, and TORC1 signaling may uncover new 
paradigms through which the response to protein misfolding is regulated.

PMID: 31144305

GEO ID: GSE129200.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE129200.final.pcl	Hyperactive TORC1 sensitizes yeast cells to endoplasmic reticulum stress by compromising cell wall integrity.	8	stress|protein structure and folding	1

File last modified: Thu Apr 25 2024

---


# Blasco-Moreno_2019_PMID_30899024

Citation: Blasco-Moreno B, de Campos-Mata L, Bottcher R, Garcia-Martinez J, Jungfleisch J, 
Nedialkova DD, Chattopadhyay S, Gas ME, Oliva B, Perez-Ortin JE, Leidel SA, Choder M, Diez
J. 2019. The exonuclease Xrn1 activates transcription and translation of mRNAs encoding 
membrane proteins.

Full Description: The highly conserved 5'-3' exonuclease Xrn1 regulates gene expression in eukaryotes by 
coupling nuclear DNA transcription to cytosolic mRNA decay. By integrating transcriptome-
wide analyses of translation with biochemical and functional studies, we demonstrate an 
unanticipated regulatory role of Xrn1 in protein synthesis. Xrn1 promotes translation of a
specific group of transcripts encoding membrane proteins. Xrn1-dependence for translation 
is linked to poor structural RNA contexts for translation initiation, is mediated by 
interactions with components of the translation initiation machinery and correlates with 
an Xrn1-dependence for mRNA localization at the endoplasmic reticulum, the translation 
compartment of membrane proteins. Importantly, for this group of mRNAs, Xrn1 stimulates 
transcription, mRNA translation and decay. Our results uncover a crosstalk between the 
three major stages of gene expression coordinated by Xrn1 to maintain appropriate levels 
of membrane proteins.

PMID: 30899024

GEO ID: GSE123326.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE123326.final.pcl	The exonuclease Xrn1 activates transcription and translation of mRNAs encoding membrane proteins.	4	RNA processing and metabolism|transcriptional regulation	1

File last modified: Thu Apr 25 2024

---


# Cho_2019_PMID_30668223

Citation: Cho CY, Kelliher CM, Haase SB. 2019. The cell-cycle transcriptional network generates and 
transmits a pulse of transcription once each cell cycle.

Full Description: Multiple studies have suggested the critical roles of cyclin-dependent kinases (CDKs) as 
well as a transcription factor (TF) network in generating the robust cell-cycle 
transcriptional program. However, the precise mechanisms by which these components 
function together in the gene regulatory network remain unclear. Here we show that the TF 
network can generate and transmit a "pulse" of transcription independently of CDK 
oscillations. The premature firing of the transcriptional pulse is prevented by early G1 
inhibitors, including transcriptional corepressors and the E3 ubiquitin ligase complex APC
(Cdh1). We demonstrate that G1 cyclin-CDKs facilitate the activation and accumulation of 
TF  proteins in S/G2/M phases through inhibiting G1 transcriptional corepressors (Whi5 and
Stb1) and APC(Cdh1), thereby promoting the initiation and propagation of the pulse by the 
TF network. These findings suggest a unique oscillatory mechanism in which global phase-
specific transcription emerges from a pulse-generating network that fires once-and-only-
once at the start of the cycle.

PMID: 30668223

GEO ID: GSE75694.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE75694.final.pcl	The cell-cycle transcriptional network generates and transmits a pulse of transcription once each cell cycle.	270	cell cycle regulation	1

File last modified: Thu Apr 25 2024

---


# Farrugia_2019_PMID_30992471

Citation: Farrugia G, Azzopardi M, Saliba C, Grech G, Gross AS, Pistolic J, Benes V, Vassallo N, 
Borg J, Madeo F, Eisenberg T, Balzan R. 2019. Aspirin impairs acetyl-coenzyme A metabolism
in redox-compromised yeast cells.

Full Description: Aspirin is a widely used anti-inflammatory and antithrombotic drug also known in  recent 
years for its promising chemopreventive antineoplastic properties, thought to be mediated 
in part by its ability to induce apoptotic cell death. However, the full range of 
mechanisms underlying aspirin's cancer-preventive properties is still elusive. In this 
study, we observed that aspirin impaired both the synthesis and transport of acetyl-
coenzyme A (acetyl-CoA) into the mitochondria of manganese superoxide dismutase (MnSOD)-
deficient Saccharomyces cerevisiae EG110 yeast cells, but not of the wild-type cells, 
grown aerobically in ethanol medium. This occurred at both the gene level, as indicated by
microarray and qRT-PCR analyses, and at the protein level as indicated by enzyme assays. 
These results show that in redox-compromised MnSOD-deficient yeast cells, but not in wild-
type cells, aspirin starves the mitochondria of acetyl-CoA and likely causes energy 
failure linked to mitochondrial damage, resulting in cell death. Since acetyl-CoA is one 
of the least-studied targets of aspirin in terms of the latter's propensity to prevent 
cancer, this work may provide further mechanistic  insight into aspirin's chemopreventive 
behavior with respect to early stage cancer cells, which tend to have downregulated MnSOD 
and are also redox-compromised.

PMID: 30992471

GEO ID: GSE115660.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE115660.final.pcl	Aspirin impairs acetyl-coenzyme A metabolism in redox-compromised yeast cells.	12	respiration|chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Gaillard_2019_PMID_30715474

Citation: Gaillard H, Santos-Pereira JM, Aguilera A. 2019. The Nup84 complex coordinates the DNA 
damage response to warrant genome integrity.

Full Description: DNA lesions interfere with cellular processes such as transcription and replication and 
need to be adequately resolved to warrant genome integrity. Beyond their primary role in 
molecule transport, nuclear pore complexes (NPCs) function in other processes such as 
transcription, nuclear organization and DNA double strand break (DSB) repair. Here we 
found that the removal of UV-induced DNA lesions by nucleotide excision repair (NER) is 
compromised in the absence of  the Nup84 nuclear pore component. Importantly, nup84Delta 
cells show an exacerbated sensitivity to UV in early S phase and delayed replication fork 
progression, suggesting that unrepaired spontaneous DNA lesions persist during S  phase. 
In addition, nup84Delta cells are defective in the repair of replication-born DSBs by 
sister chromatid recombination (SCR) and rely on post-replicative repair functions for 
normal proliferation, indicating dysfunctions in the cellular pathways that enable 
replication on damaged DNA templates. Altogether, our data reveal a central role of the 
NPC in the DNA damage response to facilitate replication progression through damaged DNA 
templates by promoting efficient NER and SCR and preventing chromosomal rearrangements.

PMID: 30715474

GEO ID: GSE113118.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE113118.final.pcl	The Nup84 complex coordinates the DNA damage response to warrant genome integrity.	6	DNA damage stimulus|DNA replication, recombination and repair	1

File last modified: Thu Apr 25 2024

---


# Hopfler_2019_PMID_31015336

Citation: Hopfler M, Kern MJ, Straub T, Prytuliak R, Habermann BH, Pfander B, Jentsch S. 2019. Slx5/
Slx8-dependent ubiquitin hotspots on chromatin contribute to stress tolerance.

Full Description: Chromatin is a highly regulated environment, and protein association with chromatin is 
often controlled by post-translational modifications and the corresponding enzymatic 
machinery. Specifically, SUMO-targeted ubiquitin ligases  (STUbLs) have emerged as key 
players in nuclear quality control, genome maintenance, and transcription. However, how 
STUbLs select specific substrates among myriads of SUMOylated proteins on chromatin 
remains unclear. Here, we reveal a remarkable co-localization of the budding yeast STUbL 
Slx5/Slx8 and ubiquitin at seven genomic loci that we term "ubiquitin hotspots". 
Ubiquitylation at these sites depends on Slx5/Slx8 and protein turnover on the Cdc48 
segregase.  We identify the transcription factor-like Ymr111c/Euc1 to associate with these
sites and to be a critical determinant of ubiquitylation. Euc1 specifically targets Slx5/
Slx8 to ubiquitin hotspots via bipartite binding of Slx5 that involves the Slx5 SUMO-
interacting motifs and an additional, novel substrate recognition domain. Interestingly, 
the Euc1-ubiquitin hotspot pathway acts redundantly with chromatin modifiers of the H2A.Z 
and Rpd3L pathways in specific  stress responses. Thus, our data suggest that STUbL-
dependent ubiquitin hotspots  shape chromatin during stress adaptation.

PMID: 31015336

GEO ID: GSE118817.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE118817.final.pcl	Slx5/Slx8-dependent ubiquitin hotspots on chromatin contribute to stress tolerance.	12	stress|chromatin organization	1

File last modified: Thu Apr 25 2024

---


# Kunkel_2019_PMID_31395866

Citation: Kunkel J, Luo X, Capaldi AP. 2019. Integrated TORC1 and PKA signaling control the temporal
activation of glucose-induced gene expression in yeast.

Full Description: The growth rate of a yeast cell is controlled by the target of rapamycin kinase complex I
(TORC1) and cAMP-dependent protein kinase (PKA) pathways. To determine  how TORC1 and PKA 
cooperate to regulate cell growth, we performed temporal analysis of gene expression in 
yeast switched from a non-fermentable substrate, to glucose, in the presence and absence 
of TORC1 and PKA inhibitors. Quantitative analysis of these data reveals that PKA drives 
the expression of key cell growth  genes during transitions into, and out of, the rapid 
growth state in glucose, while TORC1 is important for the steady-state expression of the 
same genes. This  circuit design may enable yeast to set an exact growth rate based on the
abundance of internal metabolites such as amino acids, via TORC1, but also adapt  rapidly 
to changes in external nutrients, such as glucose, via PKA.

PMID: 31395866

GEO ID: GSE133591.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE133591.final.pcl	Integrated TORC1 and PKA signaling control the temporal activation of glucose-induced gene expression in yeast.	21	carbon utilization|stress	2

File last modified: Thu Apr 25 2024

---


# Lien_2019_PMID_30973945

Citation: Lien PTK, Viet NTM, Mizuno T, Suda Y, Irie K. 2019. Pop2 phosphorylation at S39 
contributes to the glucose repression of stress response genes, HSP12 and HSP26.

Full Description: The S. cerevisiae Pop2 protein is an exonuclease in the Ccr4-Not complex that is  a 
conserved regulator of gene expression. Pop2 regulates gene expression post-
transcriptionally by shortening the poly(A) tail of mRNA. A previous study has shown that 
Pop2 is phosphorylated at threonine 97 (T97) by Yak1 protein kinase in response to glucose
limitation. However, the physiological importance of Pop2 phosphorylation remains unknown
. In this study, we found that Pop2 is phosphorylated at serine 39 (S39) under unstressed 
conditions. The dephosphorylation of S39 was occurred rapidly after glucose depletion, and
the addition of glucose to the glucose-deprived culture recovered this phosphorylation, 
suggesting that Pop2 phosphorylation at S39 is regulated by glucose. This glucose-
regulated phosphorylation of Pop2 at S39 is dependent on Pho85 kinase. We previously 
reported that Pop2 takes a part in the cell wall integrity pathway by regulating LRG1 mRNA
; however, S39 phosphorylation of Pop2 is not involved in LRG1 expression. On the other 
hand, Pop2 phosphorylation at S39 is involved in the expression of HSP12 and HSP26, which 
encode a small heat shock protein. In the medium supplemented with glucose, Pop2 might be 
phosphorylated at S39 by Pho85 kinase, and this phosphorylation contributes to repress the
expression of HSP12 and HSP26. Glucose starvation inactivated Pho85,  which resulted in 
the derepression of HSP12 and HSP26, together with other glucose sensing mechanisms. Our 
results suggest that Pho85-dependent phosphorylation of Pop2 is a part of the glucose 
sensing system in yeast.

PMID: 30973945

GEO ID: GSE124908.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE124908.final.pcl	Pop2 phosphorylation at S39 contributes to the glucose repression of stress response genes, HSP12 and HSP26.	8	stress|RNA processing and metabolism	1

File last modified: Thu Apr 25 2024

---


# Moreno_2019_PMID_31439886

Citation: Moreno AD, Gonzalez-Fernandez C, Ballesteros M, Tomas-Pejo E. 2019. Insoluble solids at 
high concentrations repress yeast's response against stress and increase intracellular ROS,
levels.

Full Description: Lignocellulosic ethanol production requires high substrate concentrations for its cost-
competitiveness. This implies the presence of high concentrations of insoluble solids (IS
) at the initial stages of the process, which may limit the fermentation performance of 
the corresponding microorganism. The presence of 40-60% IS (w/w) resulted in lower glucose
consumption rates and reduced ethanol volumetric productivities of Saccharomyces 
cerevisiae F12. Yeast cells exposed to IS exhibited a wrinkled cell surface and a reduced 
mean cell size due to cavity formation. In addition, the intracellular levels of reactive 
oxygen species (ROS) increased up to 40%. These ROS levels increased up to 70% when both 
lignocellulose-derived inhibitors and IS were simultaneously present. The general stress 
response mechanisms (e.g. DDR2, TPS1 or ZWF1 genes, trehalose and glycogen biosynthesis, 
and DNA repair mechanisms) were found repressed, and ROS formation  could not be 
counteracted by the induction of the genes involved in repairing the oxidative damage such
as glutathione, thioredoxin and methionine scavenging systems (e.g. CTA1, GRX4, MXR1, and 
TSA1; and the repression of cell cycle progression, CLN3). Overall, these results clearly 
show the role of IS as an important microbial stress factor that affect yeast cells at 
physical, physiological, and molecular levels.

PMID: 31439886

GEO ID: GSE115460.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE115460.final.pcl	Insoluble solids at high concentrations repress yeast's response against stress and increase intracellular ROS levels.	6	oxidative stress	1

File last modified: Thu Apr 25 2024

---


# Pagano_2019_PMID_30986968

Citation: Pagano L, Caldara M, Villani M, Zappettini A, Marmiroli N, Marmiroli M. 2019. In Vivo-In 
Vitro Comparative Toxicology of Cadmium Sulphide Quantum Dots in the Model Organism 
Saccharomyces cerevisiae.

Full Description: The aim of this work was to use the yeast Saccharomyces cerevisiae as a tool for  
toxicogenomic studies of Engineered Nanomaterials (ENMs) risk assessment, in particular 
focusing on cadmium based quantum dots (CdS QDs). This model has been  exploited for its 
peculiar features: a short replication time, growth on both fermentable and oxidizable 
carbon sources, and for the contextual availability of genome wide information in the form
of genetic maps, DNA microarray, and collections of barcoded mutants. The comparison of 
the whole genome analysis with the microarray experiments (99.9% coverage) and with the 
phenotypic analysis of 4688 barcoded haploid mutants (80.2% coverage), shed light on the 
genes involved  in the response to CdS QDs, both in vivo and in vitro. The results have 
clarified the mechanisms involved in the exposure to CdS QDs, and whether these ENMs and 
Cd(2+) exploited different pathways of response, in particular related to oxidative stress
and to the maintenance of mitochondrial integrity and function.  Saccharomyces cerevisiae 
remains a versatile and robust alternative for organismal toxicological studies, with a 
high level of heuristic insights into the toxicology of more complex eukaryotes, including
mammals.

PMID: 30986968

GEO ID: GSE125759.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE125759.final.pcl	In Vivo-In Vitro Comparative Toxicology of Cadmium Sulphide Quantum Dots in the Model Organism Saccharomyces cerevisiae.	6	oxidative stress	1

File last modified: Thu Apr 25 2024

---


# Romero_2019_PMID_31394264

Citation: Romero AM, Ramos-Alonso L, Montella-Manuel S, Garcia-Martinez J, de la Torre-Ruiz MA, 
Perez-Ortin JE, Martinez-Pastor MT, Puig S. 2019. A genome-wide transcriptional study 
reveals that iron deficiency inhibits the yeast TORC1 pathway.

Full Description: Iron is an essential micronutrient that participates as a cofactor in a broad range of 
metabolic processes including mitochondrial respiration, DNA replication, protein 
translation and lipid biosynthesis. Adaptation to iron deficiency requires the global 
reorganization of cellular metabolism directed to  optimize iron utilization. The budding 
yeast Saccharomyces cerevisiae has been widely used to characterize the responses of 
eukaryotic microorganisms to iron depletion. In this report, we used a genomic approach to
investigate the contribution of transcription rates to the modulation of mRNA levels 
during adaptation of yeast cells to iron starvation. We reveal that a decrease in the 
activity of all RNA polymerases contributes to the down-regulation of many mRNAs, tRNAs 
and rRNAs. Opposite to the general expression pattern, many genes including components of 
the iron deficiency response, the mitochondrial retrograde pathway  and the general stress
response display a remarkable increase in both transcription rates and mRNA levels upon 
iron limitation, whereas genes encoding  ribosomal proteins or implicated in ribosome 
biogenesis exhibit a pronounced fall. This expression profile is consistent with an 
activation of the environmental stress response. The phosphorylation stage of multiple 
regulatory factors strongly suggests that the conserved nutrient signaling pathway TORC1 
is  inhibited during the progress of iron deficiency. These results suggest an intricate 
crosstalk between iron metabolism and the TORC1 pathway that should be  considered in many
disorders.

PMID: 31394264

GEO ID: GSE127875.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE127875.final.pcl	A genome-wide transcriptional study reveals that iron deficiency inhibits the yeast TORC1 pathway.	11	metal or metalloid ion stress	1

File last modified: Thu Apr 25 2024

---


# Sunyer-Figueres_2020_PMID_33019712

Citation: Sunyer-Figueres M, Vazquez J, Mas A, Torija MJ, Beltran G. 2020. Transcriptomic Insights 
into the Effect of Melatonin in Saccharomyces cerevisiae  in the Presence and Absence of 
Oxidative Stress.

Full Description: Melatonin is a ubiquitous indolamine that plays important roles in various aspects of 
biological processes in mammals. In Saccharomyces cerevisiae, melatonin has been reported 
to exhibit antioxidant properties and to modulate the expression of some genes involved in
endogenous defense systems. The aim of this  study was to elucidate the role of 
supplemented melatonin at the transcriptional  level in S. cerevisiae in the presence and 
absence of oxidative stress. This was  achieved by exposing yeast cells pretreated with 
different melatonin concentrations to hydrogen peroxide and assessing the entry of 
melatonin into the cell and the yeast response at the transcriptional level (by microarray
and qPCR  analyses) and the physiological level (by analyzing changes in the lipid 
composition and mitochondrial activity). We found that exogenous melatonin crossed 
cellular membranes at nanomolar concentrations and modulated the expression of many genes
, mainly downregulating the expression of mitochondrial genes in the absence of oxidative 
stress, triggering a hypoxia-like response, and upregulating them under stress, mainly the
cytochrome complex and electron transport chain. Other categories that were enriched by 
the effect of melatonin were related to transport, antioxidant activity, signaling, and 
carbohydrate and  lipid metabolism. The overall results suggest that melatonin is able to 
reprogram the cellular machinery to achieve tolerance to oxidative stress.

PMID: 33019712

GEO ID: GSE154702.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE154702.final.pcl	Transcriptomic Insights into the Effect of Melatonin in Saccharomyces cerevisiae  in the Presence and Absence of Oxidative Stress.	15	transcriptional regulation	2

File last modified: Thu Apr 25 2024

---


# Bao_2021_PMID_34520050

Citation: Bao X, Koorengevel MC, Groot Koerkamp MJA, Homavar A, Weijn A, Crielaard S, Renne MF, 
Lorent JH, Geerts WJ, Surma MA, Mari M, Holstege FCP, Klose C, de Kroon AIPM. 2021. 
Shortening of membrane lipid acyl chains compensates for phosphatidylcholine deficiency in
choline-auxotroph yeast.

Full Description: Phosphatidylcholine (PC) is an abundant membrane lipid component in most eukaryotes, 
including yeast, and has been assigned multiple functions in addition to acting as 
building block of the lipid bilayer. Here, by isolating S. cerevisiae suppressor mutants 
that exhibit robust growth in the absence of PC, we show that PC essentiality is subject 
to cellular evolvability in yeast. The requirement for PC is suppressed by monosomy of 
chromosome XV or by a point mutation in the ACC1 gene encoding acetyl-CoA carboxylase. 
Although these two genetic adaptations rewire lipid biosynthesis in different ways, both 
decrease Acc1 activity, thereby reducing average acyl chain length. Consistently, soraphen
A, a specific inhibitor of Acc1, rescues a yeast mutant with deficient PC synthesis. In 
the aneuploid suppressor, feedback inhibition of Acc1 through acyl-CoA produced by fatty 
acid synthase (FAS) results from upregulation of lipid synthesis. The results show that 
budding yeast regulates acyl chain length by fine-tuning the activities of Acc1 and FAS 
and indicate that PC evolved by benefitting the maintenance of membrane fluidity.

PMID: 34520050

GEO ID: GSE75725.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE75725.final.pcl	Shortening of membrane lipid acyl chains compensates for phosphatidylcholine deficiency in choline-auxotroph yeast.	10	lipid metabolism	2

File last modified: Thu Apr 25 2024

---


# Garcia-Martinez_2021_PMID_33258404

Citation: Garcia-Martinez J, Perez-Martinez ME, Perez-Ortin JE, Alepuz P. 2021. Recruitment of Xrn1 
to stress-induced genes allows efficient transcription by controlling RNA polymerase II 
backtracking.

Full Description: A new paradigm has emerged proposing that the crosstalk between nuclear transcription and 
cytoplasmic mRNA stability keeps robust mRNA levels in cells under steady-state conditions
. A key piece in this crosstalk is the highly conserved 5'-3' RNA exonuclease Xrn1, which 
degrades most cytoplasmic mRNAs but also associates with nuclear chromatin to activate 
transcription by not well-understood mechanisms. Here, we investigated the role of Xrn1 in
the transcriptional response of Saccharomyces cerevisiae cells to osmotic stress. We  show
that a lack of Xrn1 results in much lower transcriptional induction of the upregulated 
genes but in similar high levels of their transcripts because of parallel mRNA 
stabilization. Unexpectedly, lower transcription in xrn1 occurs with a higher accumulation
of RNA polymerase II (RNAPII) at stress-inducible genes, suggesting that this polymerase 
remains inactive backtracked. Xrn1 seems to be directly implicated in the formation of a 
competent elongation complex because Xrn1 is recruited to the osmotic stress-upregulated 
genes in parallel with the RNAPII complex, and both are dependent on the mitogen-activated
protein  kinase Hog1. Our findings extend the role of Xrn1 in preventing the accumulation
of inactive RNAPII at highly induced genes to other situations of rapid and strong 
transcriptional upregulation.

PMID: 33258404

GEO ID: GSE151736.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE151736.final.pcl	Recruitment of Xrn1 to stress-induced genes allows efficient transcription by controlling RNA polymerase II backtracking.	10	transcriptional regulation	2

File last modified: Thu Apr 25 2024

---


# Klim_2021_PMID_34039270

Citation: Klim J, Zielenkiewicz U, Skoneczny M, Skoneczna A, Kurlandzka A, Kaczanowski S. 2021. 
Genetic interaction network has a very limited impact on the evolutionary trajectories in 
continuous culture-grown populations of yeast.

Full Description: BACKGROUND: The impact of genetic interaction networks on evolution is a fundamental issue
. Previous studies have demonstrated that the topology of the network is determined by the
properties of the cellular machinery. Functionally related genes frequently interact with 
one another, and they establish modules, e.g., modules of protein complexes and 
biochemical pathways. In this study, we experimentally tested the hypothesis that 
compensatory evolutionary modifications, such as mutations and transcriptional changes, 
occur frequently in genes from perturbed modules of interacting genes. RESULTS: Using 
Saccharomyces cerevisiae haploid deletion mutants as a model, we investigated two modules 
lacking COG7 or NUP133, which are evolutionarily conserved genes with many interactions. 
We performed laboratory evolution experiments with these strains in two genetic 
backgrounds (with or without additional deletion of MSH2), subjecting them to continuous 
culture in a non-limiting minimal medium. Next, the evolved yeast populations were 
characterized through whole-genome sequencing and transcriptome analyses. No obvious 
compensatory changes resulting from inactivation of genes already included in modules were
identified. The supposedly compensatory inactivation of genes in the evolved strains was 
only rarely observed to be in accordance with the established fitness effect of the 
genetic interaction network. In fact, a substantial majority of the gene inactivations 
were predicted to be neutral in the experimental conditions used to determine the 
interaction network. Similarly, transcriptome changes during continuous culture mostly 
signified adaptation to growth conditions rather than compensation of the  absence of the 
COG7, NUP133 or MSH2 genes. However, we noticed that for genes whose inactivation was 
deleterious an upregulation of transcription was more common than downregulation. 
CONCLUSIONS: Our findings demonstrate that the genetic interactions and the modular 
structure of the network described by others have very limited effects on the evolutionary
trajectory following gene deletion  of module elements in our experimental conditions and 
has no significant impact on short-term compensatory evolution. However, we observed 
likely compensatory evolution in functionally related (albeit non-interacting) genes.

PMID: 34039270

GEO ID: GSE167397.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE167397.final.pcl	Genetic interaction network has a very limited impact on the evolutionary trajectories in continuous culture-grown populations of yeast.	17	genetic interaction	2

File last modified: Thu Apr 25 2024

---


# Alhaj_Sulaiman_2022_PMID_35705668

Citation: Alhaj Sulaiman A, Ali R, Aouida M, Moovarkumudalvan B, Ramotar D. 2022. The histone H2B 
Arg95 residue links the pheromone response pathway to  rapamycin-induced G(1) arrest in 
yeast.

Full Description: Rapamycin is an immunosuppressant used for treating many types of diseases such  as kidney
carcinomas. In yeast, rapamycin inhibits the TORC1 kinase signaling  pathway causing rapid
alteration in gene expression and ultimately cell cycle  arrest in G(1) through mechanisms
that are not fully understood. Herein, we  screened a histone mutant collection and report
that one of the mutants, H2B  R95A, is strikingly resistant to rapamycin due to a 
defective cell cycle arrest.  We show that the H2B R95A causes defects in the expression 
of a subset of genes  of the pheromone pathway required for alpha factor-induced G(1) 
arrest. The  expression of the STE5 gene and its encoded scaffold protein Ste5, required 
for  the sequential activation of the MAPKs of the pheromone pathway, is greatly  reduced 
in the H2B R95A mutant. Similar to the H2B R95A mutant, cells devoid of  Ste5 are also 
resistant to rapamycin. Rapamycin-induced G(1) arrest does not  involve detectable 
phosphorylation of the MAPKs, Kss1, and Fus3, as reported for  alpha factor-induced G(1) 
arrest. However, we observed a sharp induction of the G(1)  cyclin Cln2 (~ 3- to 4-fold) 
in the ste5Delta mutant within 30 min of exposure to  rapamycin. Our data provide a new 
insight whereby rapamycin signaling via the  Torc1 kinase may exploit the pheromone 
pathway to arrest cells in the G(1) phase.

PMID: 35705668

GEO ID: GSE201413.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE201413.final.pcl	The histone H2B Arg95 residue links the pheromone response pathway to  rapamycin-induced G(1) arrest in yeast.	12	chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Chattopadhyay_2022_PMID_36418294

Citation: Chattopadhyay S, Garcia-Martinez J, Haimovich G, Fischer J, Khwaja A, Barkai O, Chuartzman
SG, Schuldiner M, Elran R, Rosenberg MI, Urim S, Deshmukh S, Bohnsack KE, Bohnsack MT, 
Perez-Ortin JE, Choder M. 2022. RNA-controlled nucleocytoplasmic shuttling of mRNA decay 
factors regulates mRNA  synthesis and a novel mRNA decay pathway.

Full Description: mRNA level is controlled by factors that mediate both mRNA synthesis and decay,  including
the 5' to 3' exonuclease Xrn1. Here we show that nucleocytoplasmic  shuttling of several 
yeast mRNA decay factors plays a key role in determining  both mRNA synthesis and decay. 
Shuttling is regulated by RNA-controlled binding  of the karyopherin Kap120 to two nuclear
localization sequences (NLSs) in Xrn1,  location of one of which is conserved from yeast 
to human. The decaying RNA binds  and masks NLS1, establishing a link between mRNA decay 
and Xrn1 shuttling.  Preventing Xrn1 import, either by deleting KAP120 or mutating the two
Xrn1 NLSs,  compromises transcription and, unexpectedly, also cytoplasmic decay, 
uncovering a  cytoplasmic decay pathway that initiates in the nucleus. Most mRNAs are 
degraded  by both pathways - the ratio between them represents a full spectrum.  
Importantly, Xrn1 shuttling is required for proper responses to environmental  changes, e.
g., fluctuating temperatures, involving proper changes in mRNA  abundance and in cell 
proliferation rate.

PMID: 36418294

GEO ID: GSE204991.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE204991.final.pcl	RNA-controlled nucleocytoplasmic shuttling of mRNA decay factors regulates mRNA  synthesis and a novel mRNA decay pathway.	4	RNA catabolism	1

File last modified: Thu Apr 25 2024

---


# Godillot_2022_PMID_35273585

Citation: Godillot J, Sanchez I, Perez M, Picou C, Galeote V, Sablayrolles JM, Farines V, Mouret JR,
 2022. The Timing of Nitrogen Addition Impacts Yeast Genes Expression and the Production  
of Aroma Compounds During Wine Fermentation.

Full Description: Among the different compounds present in the must, nitrogen is an essential  nutrient for 
the management of fermentation kinetics but also plays an important  role in the synthesis
of fermentative aromas. To address the problems related to  nitrogen deficiencies, 
nitrogen additions during alcoholic fermentation have been  implemented. The consequences 
of such additions on the main reaction are well  known. However, their impact on aromas 
synthesis is still poorly understood. So,  the main objective of this study was to 
determine the impact of nitrogen addition  during the stationary phase on both the 
fermentation kinetics and aroma  synthesis. To reach this goal, we used a 
transdisciplinary approach combining  statistical modeling (Box-Behnken design and 
response surface modeling) and gene  expression study (transcriptomic analysis). Our 
results indicated that nitrogen  metabolism, central carbon metabolism (CCM), fermentation
kinetics and aroma  production were significantly impacted by nitrogen addition. The most 
remarkable  point was the different regulation of the bioconversion of higher alcohols 
into  acetate esters on one hand and of fatty acids into ethyl esters on the other  hand. 
We highlighted that the conversion of higher alcohols into acetate esters  was maximum 
when nitrogen was added at the beginning of the stationary phase.  Conversely, the highest
conversion of acids into ethyl esters was reached when  nitrogen was added close to the 
end of the stationary phase. Moreover, even if  the key element in the production of these
two ester families appeared to be the  enzymatic activity responsible for their production
, rather than the availability  of the corresponding precursors, these enzymatic 
activities were differently  regulated. For acetate esters, the regulation occurred at 
gene level: the ATF2  gene was overexpressed following nitrogen addition during the 
stationary phase.  On the opposite, no induction of gene expression was noted for ethyl 
esters; it  seemed that there was an allosteric regulation.

PMID: 35273585

GEO ID: GSE189010.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE189010.final.pcl	The Timing of Nitrogen Addition Impacts Yeast Genes Expression and the Production  of Aroma Compounds During Wine Fermentation.	24	nitrogen utilization	1

File last modified: Thu Apr 25 2024

---


# Jimenez-Gutierrez_2022_PMID_36232332

Citation: Jimenez-Gutierrez E, Fernandez-Acero T, Alonso-Rodriguez E, Molina M, Martin H. 2022. 
Neomycin Interferes with Phosphatidylinositol-4,5-Bisphosphate at the Yeast  Plasma 
Membrane and Activates the Cell Wall Integrity Pathway.

Full Description: The cell wall integrity pathway (CWI) is a MAPK-mediated signaling route  essential for 
yeast cell response to cell wall damage, regulating distinct  aspects of fungal physiology
. We have recently proven that the incorporation of a  genetic circuit that operates as a 
signal amplifier into this pathway allows for  the identification of novel elements 
involved in CWI signaling. Here, we show  that the strong growth inhibition triggered by 
pathway hyperactivation in cells  carrying the "Integrity Pathway Activation Circuit" (
IPAC) also allows the easy  identification of new stimuli. By using the IPAC, we have 
found various chemical  agents that activate the CWI pathway, including the aminoglycoside
neomycin.  Cells lacking key components of this pathway are sensitive to this antibiotic
,  due to the disruption of signaling upon neomycin stimulation. Neomycin reduces  both 
phosphatidylinositol-4,5-bisphosphate (PIP(2)) availability at the plasma  membrane and 
myriocin-induced TORC2-dependent Ypk1 phosphorylation, suggesting a  strong interference 
with plasma membrane homeostasis, specifically with PIP(2).  The neomycin-induced 
transcriptional profile involves not only genes related to  stress and cell wall 
biogenesis, but also to amino acid metabolism, reflecting  the action of this antibiotic 
on the yeast ribosome.

PMID: 36232332

GEO ID: GSE210785.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE210785.final.pcl	Neomycin Interferes with Phosphatidylinositol-4,5-Bisphosphate at the Yeast  Plasma Membrane and Activates the Cell Wall Integrity Pathway.	6	chemical stimulus	1

File last modified: Thu Apr 25 2024

---


# Lu_2022_PMID_36066422

Citation: Lu PYT, Kirlin AC, Aristizabal MJ, Brewis HT, Levesque N, Setiaputra DT, Avvakumov N, 
Benschop JJ, Groot Koerkamp M, Holstege FCP, Krogan NJ, Yip CK, Cote J, Kobor MS. 2022. A 
balancing act: interactions within NuA4/TIP60 regulate picNuA4 function in  Saccharomyces 
cerevisiae and humans.

Full Description: The NuA4 lysine acetyltransferase complex acetylates histone and nonhistone  proteins and 
functions in transcription regulation, cell cycle progression, and  DNA repair. NuA4 
harbors an interesting duality in that its catalytic module can  function independently 
and distinctly as picNuA4. At the molecular level, picNuA4  anchors to its bigger brother 
via physical interactions between the C-terminus of  Epl1 and the HSA domain of Eaf1, the 
NuA4 central scaffolding subunit. This is  reflected at the regulatory level, as picNuA4 
can be liberated genetically from  NuA4 by disrupting the Epl1-Eaf1 interaction. As such, 
removal of either Eaf1 or  the Epl1 C-terminus offers a unique opportunity to elucidate 
the contributions of  Eaf1 and Epl1 to NuA4 biology and in turn their roles in balancing 
picNuA4 and  NuA4 activities. Using high-throughput genetic and gene expression profiling
, and  targeted functional assays to compare eaf1Delta and epl1-CDelta mutants, we found 
that  EAF1 and EPL1 had both overlapping and distinct roles. Strikingly, loss of EAF1  or 
its HSA domain led to a significant decrease in the amount of picNuA4, while  loss of the 
Epl1 C-terminus increased picNuA4 levels, suggesting starkly opposing  effects on picNuA4 
regulation. The eaf1Delta epl1-CDelta double mutants resembled the  epl1-CDelta single 
mutants, indicating that Eaf1's role in picNuA4 regulation  depended on the Epl1 C-
terminus. Key aspects of this regulation were  evolutionarily conserved, as truncating an 
Epl1 homolog in human cells increased  the levels of other picNuA4 subunits. Our findings 
suggested a model in which  distinct aspects of the Epl1-Eaf1 interaction regulated 
picNuA4 amount and  activity.

PMID: 36066422

GEO ID: GSE137261.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE137261.final.pcl	A balancing act: interactions within NuA4/TIP60 regulate picNuA4 function in  Saccharomyces cerevisiae and humans.	10	histone modification	2

File last modified: Thu Apr 25 2024

---


# Moreno_2022_PMID_35017613

Citation: Moreno AD, Gonzalez-Fernandez C, Tomas-Pejo E. 2022. Insights into cell robustness against
lignocellulosic inhibitors and insoluble  solids in bioethanol production processes.

Full Description: Increasing yeast robustness against lignocellulosic-derived inhibitors and  insoluble 
solids in bioethanol production is essential for the transition to a  bio-based economy. 
This work evaluates the effect exerted by insoluble solids on  yeast tolerance to 
inhibitory compounds, which is crucial in high gravity  processes. Adaptive laboratory 
evolution (ALE) was applied on a xylose-fermenting  Saccharomyces cerevisiae strain to 
simultaneously increase the tolerance to  lignocellulosic inhibitors and insoluble solids
. The evolved strain gave rise to  a fivefold increase in bioethanol yield in fermentation
experiments with high  concentration of inhibitors and 10% (w/v) of water insoluble solids
. This strain  also produced 5% (P > 0.01) more ethanol than the parental in simultaneous
saccharification and fermentation of steam-exploded wheat straw, mainly due to an  
increased xylose consumption. In response to the stress conditions (solids and  inhibitors
) imposed in ALE, cells induced the expression of genes related to cell  wall integrity (
SRL1, CWP2, WSC2 and WSC4) and general stress response (e.g.,  CDC5, DUN1, CTT1, GRE1), 
simultaneously repressing genes related to protein  synthesis and iron transport and 
homeostasis (e.g., FTR1, ARN1, FRE1), ultimately  leading to the improved phenotype. These
results contribute towards understanding  molecular mechanisms that cells might use to 
convert lignocellulosic substrates  effectively.

PMID: 35017613

GEO ID: GSE159167.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE159167.final.pcl	Insights into cell robustness against lignocellulosic inhibitors and insoluble  solids in bioethanol production processes.	6	fermentation	1

File last modified: Thu Apr 25 2024

---


# Revilleza_2022_PMID_35522675

Citation: Revilleza JEC, Sato M, Irie K, Suda Y, Mizuno T, Irie K. 2022. Regulation of CLB6 
expression by the cytoplasmic deadenylase Ccr4 through its  coding and 3' UTR regions.

Full Description: RNA stability control contributes to the proper expression of gene products.  Messenger 
RNAs (mRNAs) in eukaryotic cells possess a 5' cap structure and the 3'  poly(A) tail which
are important for mRNA stability and efficient translation.  The Ccr4-Not complex is a 
major cytoplasmic deadenylase and functions in mRNA  degradation. The CLB1-6 genes in 
Saccharomyces cerevisiae encode B-type cyclins  which are involved in the cell cycle 
progression together with the  cyclin-dependent kinase Cdc28. The CLB genes consist of 
CLB1/2, CLB3/4, and  CLB5/6 whose gene products accumulate at the G2-M, S-G2, and late G1 
phase,  respectively. These Clb protein levels are thought to be mainly regulated by the  
transcriptional control and the protein stability control. Here we investigated  
regulation of CLB1-6 expression by Ccr4. Our results show that all CLB1-6 mRNA  levels 
were significantly increased in the ccr4Delta mutant compared to those in  wild-type cells
. Clb1, Clb4, and Clb6 protein levels were slightly increased in  the ccr4Delta mutant, 
but the Clb2, Clb3, and Clb5 protein levels were similar to  those in wild-type cells. 
Since both CLB6 mRNA and Clb6 protein levels were most  significantly increased in the 
ccr4Delta mutant, we further analyzed the cis-elements  for the Ccr4-mediated regulation 
within CLB6 mRNA. We found that there were  destabilizing sequences in both coding 
sequence and 3' untranslated region (3'  UTR). The destabilizing sequences in the coding 
region were found to be both  within and outside the sequences corresponding the cyclin 
domain. The CLB6 3' UTR  was sufficient for mRNA destabilization and decrease of the 
reporter GFP gene and  this destabilization involved Ccr4. Our results suggest that CLB6 
expression is  regulated by Ccr4 through the coding sequence and 3' UTR of CLB6 mRNA.

PMID: 35522675

GEO ID: GSE198743.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE198743.final.pcl	Regulation of CLB6 expression by the cytoplasmic deadenylase Ccr4 through its  coding and 3' UTR regions.	6	transcriptional regulation	1

File last modified: Thu Apr 25 2024

---


# Wu_2022_PMID_36419096

Citation: Wu Y, Li B, Miao B, Xie C, Tang YQ. 2022. Saccharomyces cerevisiae employs complex 
regulation strategies to tolerate low pH  stress during ethanol production.

Full Description: BACKGROUND: Industrial bioethanol production may involve a low pH environment  caused by 
inorganic acids, improving the tolerance of Saccharomyces cerevisiae to  a low pH 
environment is of industrial importance to increase ethanol yield,  control bacterial 
contamination, and reduce production cost. In our previous  study, acid tolerance of a 
diploid industrial Saccharomyces cerevisiae strain  KF-7 was chronically acclimatized by 
continuous ethanol fermentation under  gradually increasing low-pH stress conditions. Two 
haploid strains B3 and C3  having excellent low pH tolerance were derived through the 
sporulation of an  isolated mutant. Diploid strain BC3 was obtained by mating these two 
haploids. In  this study, B3, C3, BC3, and the original strain KF-7 were subjected to  
comparison transcriptome analysis to investigate the molecular mechanism of the  enhanced 
phenotype. RESULT: The comparison transcriptome analysis results  suggested that the 
upregulated vitamin B1 and B6 biosynthesis contributed to the  low pH tolerance. Amino 
acid metabolism, DNA repairment, and general stress  response might also alleviate low pH 
stress. CONCLUSION: Saccharomyces cerevisiae  seems to employ complex regulation 
strategies to tolerate low pH during ethanol  production. The findings provide guides for 
the construction of low pH-tolerant  industrial strains that can be used in industrial 
fermentation processes.

PMID: 36419096

GEO ID: GSE210964.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE210964.final.pcl	Saccharomyces cerevisiae employs complex regulation strategies to tolerate low pH  stress during ethanol production.	16	stress	1

File last modified: Thu Apr 25 2024

---


# Kamata_2023_PMID_37189367

Citation: Kamata K, Ayano T, Oki M. 2023. Spt3 and Spt8 Are Involved in the Formation of a Silencing
Boundary by  Interacting with TATA-Binding Protein.

Full Description: In Saccharomyces cerevisiae, a heterochromatin-like chromatin structure called  the 
silencing region is present at the telomere as a complex of Sir2, Sir3, and  Sir4. 
Although spreading of the silencing region is blocked by histone  acetylase-mediated 
boundary formation, the details of the factors and mechanisms  involved in the spread and 
formation of the boundary at each telomere are  unknown. Here, we show that Spt3 and Spt8 
block the spread of the silencing  regions. Spt3 and Spt8 are members of the Spt-Ada-Gcn5-
acetyltransferase (SAGA)  complex, which has histone acetyltransferase activity. We 
performed microarray  analysis of the transcriptome of spt3Delta and spt8Delta strains and
RT-qPCR analysis of  the transcript levels of genes from the subtelomeric region in 
mutants in which  the interaction of Spt3 with TATA-binding protein (TBP) is altered. The 
results  not only indicated that both Spt3 and Spt8 are involved in TBP-mediated boundary
formation on the right arm of chromosome III, but also that boundary formation in  this 
region is DNA sequence independent. Although both Spt3 and Spt8 interact  with TBP, Spt3 
had a greater effect on genome-wide transcription. Mutant analysis  showed that the 
interaction between Spt3 and TBP plays an important role in the  boundary formation.

PMID: 37189367

GEO ID: GSE220290.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE220290.final.pcl	Spt3 and Spt8 Are Involved in the Formation of a Silencing Boundary by  Interacting with TATA-Binding Protein.	3	transcription	1

File last modified: Thu Apr 25 2024

---


# Park_2023_PMID_37639444

Citation: Park ZM, Remillard M, Belnap E, Rose MD. 2023. Kar4 is required for the normal pattern of 
meiotic gene expression.

Full Description: Kar4p, the yeast homolog of the mammalian methyltransferase subunit METTL14, is  required 
for efficient mRNA m6A methylation, which regulates meiotic entry. Kar4p  is also required
for a second seemingly non-catalytic function during meiosis.  Overexpression of the early
meiotic transcription factor, IME1, can bypass the  requirement for Kar4p in meiotic entry
but the additional overexpression of the  translational regulator, RIM4, is required to 
permit sporulation in kar4Delta/Delta.  Using microarray analysis and RNA sequencing, we 
sought to determine the impact  of removing Kar4p and consequently mRNA methylation on the
early meiotic  transcriptome in a strain background (S288c) that is sensitive to the loss 
of  early meiotic regulators. We found that kar4Delta/Delta mutants have a largely wild 
type  transcriptional profile with the exception of two groups of genes that show  delayed
and reduced expression: (1) a set of Ime1p-dependent early genes as well  as IME1, and (2
) a set of late genes dependent on the mid-meiotic transcription  factor, Ndt80p. The 
early gene expression defect is likely the result of the loss  of mRNA methylation and is 
rescued by overexpressing IME1, but the late defect is  only suppressed by overexpression 
of both IME1 and RIM4. The requirement for RIM4  led us to predict that the non-catalytic 
function of Kar4p, like  methyltransferase complex orthologs in other systems, may 
function at the level  of translation. Mass spectrometry analysis identified several genes
involved in  meiotic recombination with strongly reduced protein levels, but with little 
to no  reduction in transcript levels in kar4Delta/Delta after IME1 overexpression. The 
low  levels of these proteins were rescued by overexpression of RIM4 and IME1, but not  by
the overexpression of IME1 alone. These data expand our understanding of the  role of 
Kar4p in regulating meiosis and provide key insights into a potential  mechanism of Kar4p'
s later meiotic function that is independent of mRNA  methylation.

PMID: 37639444

GEO ID: GSE220125.final.pcl

PCL filename	short description	# conditions	tags	# channels
GSE220125.final.pcl	Kar4 is required for the normal pattern of meiotic gene expression.	175	RNA processing and metabolism|sporulation	2

File last modified: Thu Apr 25 2024

---

