<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>torchcell.sequence.data &mdash; pytorch_geometric  documentation</title>
      <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../../../_static/css/mytheme.css" type="text/css" />
    <link rel="shortcut icon" href="https://raw.githubusercontent.com/pyg-team/pyg_sphinx_theme/master/pyg_sphinx_theme/static/img/favicon.png"/>
  <!--[if lt IE 9]>
    <script src="../../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../../../" id="documentation_options" src="../../../_static/documentation_options.js"></script>
        <script src="../../../_static/jquery.js"></script>
        <script src="../../../_static/underscore.js"></script>
        <script src="../../../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../../../_static/doctools.js"></script>
        <script src="../../../_static/js/on_pyg_load.js"></script>
        <script src="../../../_static/js/version_alert.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script src="../../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../../index.html">
            
              <img src="https://raw.githubusercontent.com/pyg-team/pyg_sphinx_theme/master/pyg_sphinx_theme/static/img/pyg_logo.png" class="logo" alt="Logo"/>
          </a>
              <div class="version">
                0.0.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Package Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../modules/sequence.html">torchcell.sequence</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../index.html">pytorch_geometric</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../index.html">Module code</a></li>
      <li class="breadcrumb-item active">torchcell.sequence.data</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <h1>Source code for torchcell.sequence.data</h1><div class="highlight"><pre>
<span></span><span class="c1"># src/torchcell/sequence/data.py</span>
<span class="c1"># [[src.torchcell.sequence.data]]</span>
<span class="c1"># https://github.com/Mjvolk3/torchcell/tree/main/src/torchcell/sequence/data.py</span>
<span class="c1"># Test file: /src/torchcell/sequence/test_data.py</span>

<span class="kn">import</span> <span class="nn">logging</span>
<span class="kn">from</span> <span class="nn">abc</span> <span class="kn">import</span> <span class="n">ABC</span><span class="p">,</span> <span class="n">abstractmethod</span>
<span class="kn">from</span> <span class="nn">turtle</span> <span class="kn">import</span> <span class="n">st</span>
<span class="kn">from</span> <span class="nn">typing</span> <span class="kn">import</span> <span class="n">Set</span>

<span class="kn">import</span> <span class="nn">gffutils</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">from</span> <span class="nn">attrs</span> <span class="kn">import</span> <span class="n">define</span><span class="p">,</span> <span class="n">field</span>
<span class="kn">from</span> <span class="nn">Bio</span> <span class="kn">import</span> <span class="n">SeqIO</span>
<span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">Seq</span>
<span class="kn">from</span> <span class="nn">gffutils</span> <span class="kn">import</span> <span class="n">Feature</span><span class="p">,</span> <span class="n">FeatureDB</span>
<span class="kn">from</span> <span class="nn">gffutils.biopython_integration</span> <span class="kn">import</span> <span class="n">to_seqfeature</span>
<span class="kn">from</span> <span class="nn">matplotlib</span> <span class="kn">import</span> <span class="n">pyplot</span> <span class="k">as</span> <span class="n">plt</span>
<span class="kn">from</span> <span class="nn">pydantic</span> <span class="kn">import</span> <span class="n">BaseModel</span><span class="p">,</span> <span class="n">ConfigDict</span><span class="p">,</span> <span class="n">Field</span><span class="p">,</span> <span class="n">ValidationError</span><span class="p">,</span> <span class="n">validator</span>
<span class="kn">from</span> <span class="nn">sortedcontainers</span> <span class="kn">import</span> <span class="n">SortedSet</span>
<span class="kn">from</span> <span class="nn">sympy</span> <span class="kn">import</span> <span class="n">sequence</span>

<span class="kn">from</span> <span class="nn">torchcell.datamodels</span> <span class="kn">import</span> <span class="n">ModelStrict</span>
<span class="kn">from</span> <span class="nn">torchcell.models.constants</span> <span class="kn">import</span> <span class="n">DNA_LLM_MAX_TOKEN_SIZE</span>

<span class="n">log</span> <span class="o">=</span> <span class="n">logging</span><span class="o">.</span><span class="n">getLogger</span><span class="p">(</span><span class="vm">__name__</span><span class="p">)</span>
<span class="n">log</span><span class="o">.</span><span class="n">setLevel</span><span class="p">(</span><span class="n">logging</span><span class="o">.</span><span class="n">INFO</span><span class="p">)</span>


<div class="viewcode-block" id="DnaSelectionResult"><a class="viewcode-back" href="../../../generated/torchcell.sequence.DnaSelectionResult.html#torchcell.sequence.DnaSelectionResult">[docs]</a><span class="k">class</span> <span class="nc">DnaSelectionResult</span><span class="p">(</span><span class="n">ModelStrict</span><span class="p">):</span>
    <span class="nb">id</span><span class="p">:</span> <span class="nb">str</span>
    <span class="n">chromosome</span><span class="p">:</span> <span class="nb">int</span>
    <span class="n">strand</span><span class="p">:</span> <span class="nb">str</span>
    <span class="n">start</span><span class="p">:</span> <span class="nb">int</span>
    <span class="n">end</span><span class="p">:</span> <span class="nb">int</span>
    <span class="n">seq</span><span class="p">:</span> <span class="nb">str</span>

    <span class="k">def</span> <span class="fm">__len__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
        <span class="k">return</span> <span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">seq</span><span class="p">)</span>

    <span class="k">def</span> <span class="fm">__ge__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">other</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
        <span class="k">if</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">other</span><span class="p">,</span> <span class="n">DnaSelectionResult</span><span class="p">):</span>
            <span class="k">return</span> <span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">seq</span><span class="p">)</span> <span class="o">&gt;=</span> <span class="nb">len</span><span class="p">(</span><span class="n">other</span><span class="o">.</span><span class="n">seq</span><span class="p">)</span>
        <span class="k">return</span> <span class="bp">NotImplemented</span>

    <span class="k">def</span> <span class="fm">__le__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">other</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
        <span class="k">if</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">other</span><span class="p">,</span> <span class="n">DnaSelectionResult</span><span class="p">):</span>
            <span class="k">return</span> <span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">seq</span><span class="p">)</span> <span class="o">&lt;=</span> <span class="nb">len</span><span class="p">(</span><span class="n">other</span><span class="o">.</span><span class="n">seq</span><span class="p">)</span>
        <span class="k">return</span> <span class="bp">NotImplemented</span>

    <span class="nd">@validator</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">,</span> <span class="n">pre</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">always</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    <span class="k">def</span> <span class="nf">end_leq_start</span><span class="p">(</span><span class="bp">cls</span><span class="p">,</span> <span class="n">v</span><span class="p">,</span> <span class="n">values</span><span class="p">):</span>
        <span class="k">if</span> <span class="s2">&quot;start&quot;</span> <span class="ow">in</span> <span class="n">values</span> <span class="ow">and</span> <span class="n">v</span> <span class="o">&lt;=</span> <span class="n">values</span><span class="p">[</span><span class="s2">&quot;start&quot;</span><span class="p">]:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;Start must be less than end&quot;</span><span class="p">)</span>
        <span class="k">return</span> <span class="n">v</span>

    <span class="nd">@validator</span><span class="p">(</span><span class="s2">&quot;strand&quot;</span><span class="p">,</span> <span class="n">pre</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">always</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    <span class="k">def</span> <span class="nf">check_strand</span><span class="p">(</span><span class="bp">cls</span><span class="p">,</span> <span class="n">v</span><span class="p">):</span>
        <span class="k">if</span> <span class="n">v</span> <span class="ow">not</span> <span class="ow">in</span> <span class="p">[</span><span class="s2">&quot;+&quot;</span><span class="p">,</span> <span class="s2">&quot;-&quot;</span><span class="p">]:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;Strand must be either &#39;+&#39; or &#39;-&#39;&quot;</span><span class="p">)</span>
        <span class="k">return</span> <span class="n">v</span>

    <span class="nd">@validator</span><span class="p">(</span><span class="s2">&quot;chromosome&quot;</span><span class="p">,</span> <span class="s2">&quot;start&quot;</span><span class="p">,</span> <span class="s2">&quot;end&quot;</span><span class="p">,</span> <span class="n">pre</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">always</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    <span class="k">def</span> <span class="nf">check_positive</span><span class="p">(</span><span class="bp">cls</span><span class="p">,</span> <span class="n">v</span><span class="p">):</span>
        <span class="k">if</span> <span class="n">v</span> <span class="o">&lt;</span> <span class="mi">0</span><span class="p">:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">v</span><span class="si">}</span><span class="s2"> must be positive&quot;</span><span class="p">)</span>
        <span class="k">return</span> <span class="n">v</span>

    <span class="nd">@validator</span><span class="p">(</span><span class="s2">&quot;seq&quot;</span><span class="p">,</span> <span class="n">pre</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">always</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    <span class="k">def</span> <span class="nf">check_seq_len</span><span class="p">(</span><span class="bp">cls</span><span class="p">,</span> <span class="n">v</span><span class="p">):</span>
        <span class="n">sequence_length</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">v</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">sequence_length</span> <span class="o">&lt;</span> <span class="mi">0</span> <span class="ow">or</span> <span class="n">sequence_length</span> <span class="o">&gt;</span> <span class="n">DNA_LLM_MAX_TOKEN_SIZE</span><span class="p">:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
                <span class="sa">f</span><span class="s2">&quot;Sequence length (</span><span class="si">{</span><span class="n">sequence_length</span><span class="si">}</span><span class="s2">) not geq 0 and leq </span><span class="si">{</span><span class="n">DNA_LLM_MAX_TOKEN_SIZE</span><span class="si">}</span><span class="s2">&quot;</span>
            <span class="p">)</span>
        <span class="k">return</span> <span class="n">v</span></div>


<div class="viewcode-block" id="DnaWindowResult"><a class="viewcode-back" href="../../../generated/torchcell.sequence.DnaWindowResult.html#torchcell.sequence.DnaWindowResult">[docs]</a><span class="k">class</span> <span class="nc">DnaWindowResult</span><span class="p">(</span><span class="n">DnaSelectionResult</span><span class="p">):</span>
    <span class="n">start_window</span><span class="p">:</span> <span class="nb">int</span>
    <span class="n">end_window</span><span class="p">:</span> <span class="nb">int</span>

    <span class="k">def</span> <span class="fm">__repr__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
        <span class="c1"># Use f-string to create a formatted string</span>
        <span class="k">return</span> <span class="sa">f</span><span class="s2">&quot;DnaWindowResult(id=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">id</span><span class="si">!r}</span><span class="s2">, chromosome=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">chromosome</span><span class="si">!r}</span><span class="s2">, strand=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">strand</span><span class="si">!r}</span><span class="s2">, start_window=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">start_window</span><span class="si">!r}</span><span class="s2">, end_window=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">end_window</span><span class="si">!r}</span><span class="s2">, seq=</span><span class="si">{</span><span class="bp">self</span><span class="o">.</span><span class="n">seq</span><span class="si">!r}</span><span class="s2">)&quot;</span>

    <span class="nd">@validator</span><span class="p">(</span><span class="s2">&quot;start_window&quot;</span><span class="p">,</span> <span class="s2">&quot;end_window&quot;</span><span class="p">,</span> <span class="n">pre</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">always</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    <span class="k">def</span> <span class="nf">check_window</span><span class="p">(</span><span class="bp">cls</span><span class="p">,</span> <span class="n">v</span><span class="p">):</span>
        <span class="k">if</span> <span class="n">v</span> <span class="o">&lt;</span> <span class="mi">0</span><span class="p">:</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">v</span><span class="si">}</span><span class="s2"> must be positive&quot;</span><span class="p">)</span>
        <span class="k">return</span> <span class="n">v</span></div>


<div class="viewcode-block" id="GeneSet"><a class="viewcode-back" href="../../../generated/torchcell.sequence.GeneSet.html#torchcell.sequence.GeneSet">[docs]</a><span class="k">class</span> <span class="nc">GeneSet</span><span class="p">(</span><span class="n">SortedSet</span><span class="p">):</span>
    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">iterable</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">key</span><span class="o">=</span><span class="kc">None</span><span class="p">):</span>
        <span class="nb">super</span><span class="p">()</span><span class="o">.</span><span class="fm">__init__</span><span class="p">(</span><span class="n">iterable</span><span class="p">,</span> <span class="n">key</span><span class="p">)</span>
        <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="bp">self</span><span class="p">:</span>
            <span class="k">if</span> <span class="ow">not</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">item</span><span class="p">,</span> <span class="nb">str</span><span class="p">):</span>
                <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
                    <span class="sa">f</span><span class="s2">&quot;All items in gene_set must be str, got </span><span class="si">{</span><span class="nb">type</span><span class="p">(</span><span class="n">item</span><span class="p">)</span><span class="o">.</span><span class="vm">__name__</span><span class="si">}</span><span class="s2">&quot;</span>
                <span class="p">)</span>

    <span class="k">def</span> <span class="fm">__repr__</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="n">n</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span>
        <span class="n">limited_items</span> <span class="o">=</span> <span class="p">(</span><span class="bp">self</span><span class="p">)[:</span><span class="mi">3</span><span class="p">]</span>
        <span class="k">return</span> <span class="sa">f</span><span class="s2">&quot;GeneSet(size=</span><span class="si">{</span><span class="n">n</span><span class="si">}</span><span class="s2">, items=</span><span class="si">{</span><span class="n">limited_items</span><span class="si">}</span><span class="s2">...)&quot;</span></div>


<span class="c1">###########</span>
<span class="c1"># Abstract Base Class for structure</span>


<span class="c1">##########</span>
<span class="c1"># Class holding gene</span>
<div class="viewcode-block" id="Gene"><a class="viewcode-back" href="../../../generated/torchcell.sequence.Gene.html#torchcell.sequence.Gene">[docs]</a><span class="k">class</span> <span class="nc">Gene</span><span class="p">(</span><span class="n">ABC</span><span class="p">):</span>
    <span class="n">model_config</span> <span class="o">=</span> <span class="n">ConfigDict</span><span class="p">(</span><span class="n">frozen</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">extra</span><span class="o">=</span><span class="s2">&quot;forbid&quot;</span><span class="p">)</span>

    <span class="nd">@abstractmethod</span>
    <span class="k">def</span> <span class="nf">window</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">window_size</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">is_max_size</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">DnaWindowResult</span><span class="p">:</span>
        <span class="k">pass</span>

    <span class="nd">@abstractmethod</span>
    <span class="k">def</span> <span class="nf">window_five_prime</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span> <span class="n">window_size</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">allow_undersize</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">DnaWindowResult</span><span class="p">:</span>
        <span class="k">pass</span>

    <span class="nd">@abstractmethod</span>
    <span class="k">def</span> <span class="nf">window_three_prime</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span> <span class="n">window_size</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">allow_undersize</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">DnaWindowResult</span><span class="p">:</span>
        <span class="k">pass</span></div>

    <span class="c1"># name: str</span>
    <span class="c1"># seq: str</span>
    <span class="c1"># chromosome: int</span>
    <span class="c1"># start: int</span>
    <span class="c1"># end: int</span>
    <span class="c1"># strand: str</span>
    <span class="c1"># five_utr: str</span>
    <span class="c1"># three_utr: str</span>


<div class="viewcode-block" id="Genome"><a class="viewcode-back" href="../../../generated/torchcell.sequence.Genome.html#torchcell.sequence.Genome">[docs]</a><span class="k">class</span> <span class="nc">Genome</span><span class="p">(</span><span class="n">ABC</span><span class="p">):</span>
    <span class="c1"># Used elsewhere [[src/torchcell/sgd/validation/valid_models.py]]</span>
    <span class="c1"># CHECK IF THIS IS NEEDED.. I think this is a pydantic thing</span>
    <span class="c1"># model_config = ConfigDict(frozen=True, extra=&quot;forbid&quot;)</span>
    <span class="c1"># TODO not sure if we need to specify all vars in the __init__</span>
    <span class="c1"># TODO do we need to set data_root like this?</span>
    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">data_root</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span><span class="p">):</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">data_root</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="n">data_root</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">db</span><span class="p">:</span> <span class="n">FeatureDB</span> <span class="o">=</span> <span class="kc">None</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">fasta_sequences</span><span class="p">:</span> <span class="nb">dict</span> <span class="o">=</span> <span class="kc">None</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">chr_to_nc</span><span class="p">:</span> <span class="nb">dict</span> <span class="o">=</span> <span class="kc">None</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">nc_to_chr</span><span class="p">:</span> <span class="nb">dict</span> <span class="o">=</span> <span class="kc">None</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">chr_to_len</span><span class="p">:</span> <span class="nb">dict</span> <span class="o">=</span> <span class="kc">None</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_gene_set</span><span class="p">:</span> <span class="nb">set</span><span class="p">[</span><span class="nb">str</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_fasta_path</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_gff_path</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span>

    <span class="nd">@property</span>
    <span class="k">def</span> <span class="nf">gene_set</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">GeneSet</span><span class="p">:</span>
        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">_gene_set</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">_gene_set</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">compute_gene_set</span><span class="p">()</span>
        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_gene_set</span>

    <span class="nd">@gene_set</span><span class="o">.</span><span class="n">setter</span>
    <span class="k">def</span> <span class="nf">gene_set</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">value</span><span class="p">:</span> <span class="nb">set</span><span class="p">[</span><span class="nb">str</span><span class="p">]):</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">_gene_set</span> <span class="o">=</span> <span class="n">value</span>

    <span class="nd">@abstractmethod</span>
    <span class="k">def</span> <span class="nf">compute_gene_set</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">set</span><span class="p">[</span><span class="nb">str</span><span class="p">]:</span>
        <span class="k">pass</span>  <span class="c1"># Abstract methods don&#39;t have a body</span>

    <span class="nd">@abstractmethod</span>
    <span class="k">def</span> <span class="nf">get_seq</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span> <span class="nb">chr</span><span class="p">:</span> <span class="nb">int</span> <span class="o">|</span> <span class="nb">str</span><span class="p">,</span> <span class="n">start</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">end</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">strand</span><span class="p">:</span> <span class="nb">str</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">DnaSelectionResult</span><span class="p">:</span>
        <span class="k">pass</span>

    <span class="nd">@abstractmethod</span>
    <span class="k">def</span> <span class="nf">select_feature_window</span><span class="p">(</span>
        <span class="bp">self</span><span class="p">,</span> <span class="n">feature</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">window_size</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">is_max_size</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span>
    <span class="p">)</span> <span class="o">-&gt;</span> <span class="n">DnaWindowResult</span><span class="p">:</span>
        <span class="k">pass</span>

    <span class="nd">@property</span>
    <span class="nd">@abstractmethod</span>
    <span class="k">def</span> <span class="nf">gene_attribute_table</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">:</span>
        <span class="k">pass</span>

    <span class="nd">@property</span>
    <span class="nd">@abstractmethod</span>
    <span class="k">def</span> <span class="nf">feature_types</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">list</span><span class="p">[</span><span class="nb">str</span><span class="p">]:</span>
        <span class="k">pass</span>

    <span class="nd">@abstractmethod</span>
    <span class="k">def</span> <span class="fm">__getitem__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">item</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Gene</span> <span class="o">|</span> <span class="kc">None</span><span class="p">:</span>
        <span class="k">pass</span>

    <span class="c1"># Not sure if it makes more sense to have the number of genes be the length or the sum bp over all chromosomes.</span>
    <span class="k">def</span> <span class="fm">__len__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
        <span class="k">return</span> <span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">gene_set</span><span class="p">)</span></div>


<span class="c1">############</span>
<span class="c1"># Helper functions</span>
<div class="viewcode-block" id="mismatch_positions"><a class="viewcode-back" href="../../../generated/torchcell.sequence.mismatch_positions.html#torchcell.sequence.mismatch_positions">[docs]</a><span class="k">def</span> <span class="nf">mismatch_positions</span><span class="p">(</span><span class="n">seq1</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">seq2</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">list</span><span class="p">[</span><span class="nb">int</span><span class="p">]:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Computes the positions at which two sequences differ.</span>

<span class="sd">    This function takes two sequences, seq1 and seq2, represented as strings</span>
<span class="sd">    and returns a list of positions at which the two sequences have different</span>
<span class="sd">    characters. The sequences must be of the same length, else a ValueError is raised.</span>

<span class="sd">    Args:</span>
<span class="sd">        seq1 (str): The first sequence to compare.</span>
<span class="sd">        seq2 (str): The second sequence to compare.</span>

<span class="sd">    Returns:</span>
<span class="sd">        list[str]: A list containing the positions at</span>
<span class="sd">        which the two sequences differ.</span>
<span class="sd">        An empty list is returned if the sequences are identical.</span>

<span class="sd">    Raises:</span>
<span class="sd">        ValueError: If the lengths of seq1 and seq2 are not equal.</span>

<span class="sd">    Example:</span>
<span class="sd">        &gt;&gt;&gt; mismatch_positions(&quot;ATGC&quot;, &quot;ATCC&quot;)</span>
<span class="sd">        [2]</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">seq1</span><span class="p">)</span> <span class="o">!=</span> <span class="nb">len</span><span class="p">(</span><span class="n">seq2</span><span class="p">):</span>
        <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;Sequences must be the same length&quot;</span><span class="p">)</span>
    <span class="n">mismatches</span> <span class="o">=</span> <span class="p">[</span><span class="n">i</span> <span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="p">(</span><span class="n">n1</span><span class="p">,</span> <span class="n">n2</span><span class="p">)</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="n">seq1</span><span class="p">,</span> <span class="n">seq2</span><span class="p">))</span> <span class="k">if</span> <span class="n">n1</span> <span class="o">!=</span> <span class="n">n2</span><span class="p">]</span>
    <span class="k">return</span> <span class="n">mismatches</span></div>


<div class="viewcode-block" id="get_chr_from_description"><a class="viewcode-back" href="../../../generated/torchcell.sequence.get_chr_from_description.html#torchcell.sequence.get_chr_from_description">[docs]</a><span class="k">def</span> <span class="nf">get_chr_from_description</span><span class="p">(</span><span class="n">description</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Extracts the chromosome number from a given description string.</span>

<span class="sd">    Processes a description string containing either a chromosome</span>
<span class="sd">    in Roman numeral (e.g., &quot;[chromosome=IX]&quot;).</span>
<span class="sd">    Or a location (e.g., &quot;[location=mitochondrion]&quot;).</span>
<span class="sd">    If a chromosome is found, it converts the Roman numeral to an integer.</span>
<span class="sd">    If the location is a mitochondrion, it returns 0.</span>
<span class="sd">    Handles descriptions containing chromosome information in a predefined format.</span>

<span class="sd">    Args:</span>
<span class="sd">        description (str): Description string containing chromosome number</span>

<span class="sd">    Returns:</span>
<span class="sd">        int: Chromosome number.  Returns 0 if the location is mitochondrion.</span>

<span class="sd">    Raises:</span>
<span class="sd">        ValueError: If the Roman numeral conversion fails due to invalid format.</span>

<span class="sd">    Example:</span>
<span class="sd">        &gt;&gt;&gt; get_chr_from_description(&quot;[chromosome=IX] some other info&quot;)</span>
<span class="sd">        9</span>
<span class="sd">        &gt;&gt;&gt; get_chr_from_description(&quot;[location=mitochondrion] some other info&quot;)</span>
<span class="sd">        0</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="c1"># CHECK - format might be specific yeast s288c genome</span>
    <span class="n">desc_split</span> <span class="o">=</span> <span class="n">description</span><span class="o">.</span><span class="n">split</span><span class="p">()</span>
    <span class="k">for</span> <span class="n">part</span> <span class="ow">in</span> <span class="n">desc_split</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">part</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s2">&quot;[chromosome=&quot;</span><span class="p">):</span>
            <span class="n">roman_num</span> <span class="o">=</span> <span class="n">part</span><span class="p">[</span>
                <span class="nb">len</span><span class="p">(</span><span class="s2">&quot;[chromosome=&quot;</span><span class="p">)</span> <span class="p">:</span> <span class="o">-</span><span class="mi">1</span>
            <span class="p">]</span>  <span class="c1"># Exclude the last char, which is &quot;]&quot;</span>
            <span class="k">return</span> <span class="n">roman_to_int</span><span class="p">(</span><span class="n">roman_num</span><span class="p">)</span>
        <span class="k">elif</span> <span class="n">part</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s2">&quot;[location=&quot;</span><span class="p">):</span>
            <span class="c1"># we assign mitochondiral DNA to chromosome 0 as convenience</span>
            <span class="k">if</span> <span class="n">part</span><span class="p">[</span><span class="nb">len</span><span class="p">(</span><span class="s2">&quot;[location=&quot;</span><span class="p">)</span> <span class="p">:</span> <span class="o">-</span><span class="mi">1</span><span class="p">]</span> <span class="o">==</span> <span class="s2">&quot;mitochondrion&quot;</span><span class="p">:</span>
                <span class="k">return</span> <span class="mi">0</span></div>


<div class="viewcode-block" id="roman_to_int"><a class="viewcode-back" href="../../../generated/torchcell.sequence.roman_to_int.html#torchcell.sequence.roman_to_int">[docs]</a><span class="k">def</span> <span class="nf">roman_to_int</span><span class="p">(</span><span class="n">s</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Converts a Roman numeral string to an integer.</span>

<span class="sd">    This function interprets the given string `s` as a Roman numeral and</span>
<span class="sd">    returns its value as an integer. It handles the standard Roman numeral</span>
<span class="sd">    symbols (I, V, X, L, C, D, M) and uses the subtractive notation rule,</span>
<span class="sd">    where placing a smaller numeral to the left of a larger numeral</span>
<span class="sd">    represents subtraction (e.g., IV for 4).</span>

<span class="sd">    Args:</span>
<span class="sd">        s (str): The Roman numeral string to convert, consisting of the characters</span>
<span class="sd">                 I, V, X, L, C, D, M. It is assumed to be a valid Roman numeral.</span>

<span class="sd">    Returns:</span>
<span class="sd">        int: The integer value of the Roman numeral.</span>

<span class="sd">    Raises:</span>
<span class="sd">        KeyError: If input string contains characters without valid Roman numeral.</span>

<span class="sd">    Example:</span>
<span class="sd">        &gt;&gt;&gt; roman_to_int(&quot;IV&quot;)</span>
<span class="sd">        4</span>
<span class="sd">        &gt;&gt;&gt; roman_to_int(&quot;IX&quot;)</span>
<span class="sd">        9</span>
<span class="sd">        &gt;&gt;&gt; roman_to_int(&quot;XIII&quot;)</span>
<span class="sd">        13</span>

<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">roman_to_int_mapping</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s2">&quot;I&quot;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
        <span class="s2">&quot;V&quot;</span><span class="p">:</span> <span class="mi">5</span><span class="p">,</span>
        <span class="s2">&quot;X&quot;</span><span class="p">:</span> <span class="mi">10</span><span class="p">,</span>
        <span class="s2">&quot;L&quot;</span><span class="p">:</span> <span class="mi">50</span><span class="p">,</span>
        <span class="s2">&quot;C&quot;</span><span class="p">:</span> <span class="mi">100</span><span class="p">,</span>
        <span class="s2">&quot;D&quot;</span><span class="p">:</span> <span class="mi">500</span><span class="p">,</span>
        <span class="s2">&quot;M&quot;</span><span class="p">:</span> <span class="mi">1000</span><span class="p">,</span>
    <span class="p">}</span>
    <span class="n">result</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">s</span><span class="p">)):</span>
        <span class="k">if</span> <span class="n">i</span> <span class="o">&gt;</span> <span class="mi">0</span> <span class="ow">and</span> <span class="n">roman_to_int_mapping</span><span class="p">[</span><span class="n">s</span><span class="p">[</span><span class="n">i</span><span class="p">]]</span> <span class="o">&gt;</span> <span class="n">roman_to_int_mapping</span><span class="p">[</span><span class="n">s</span><span class="p">[</span><span class="n">i</span> <span class="o">-</span> <span class="mi">1</span><span class="p">]]:</span>
            <span class="n">result</span> <span class="o">+=</span> <span class="n">roman_to_int_mapping</span><span class="p">[</span><span class="n">s</span><span class="p">[</span><span class="n">i</span><span class="p">]]</span> <span class="o">-</span> <span class="mi">2</span> <span class="o">*</span> <span class="n">roman_to_int_mapping</span><span class="p">[</span><span class="n">s</span><span class="p">[</span><span class="n">i</span> <span class="o">-</span> <span class="mi">1</span><span class="p">]]</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">result</span> <span class="o">+=</span> <span class="n">roman_to_int_mapping</span><span class="p">[</span><span class="n">s</span><span class="p">[</span><span class="n">i</span><span class="p">]]</span>
    <span class="k">return</span> <span class="n">result</span></div>


<span class="c1"># Selection Window functions</span>


<span class="k">def</span> <span class="nf">calculate_window_undersized</span><span class="p">(</span>
    <span class="n">start</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">end</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">strand</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">window_size</span><span class="p">:</span> <span class="nb">int</span>
<span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">[</span><span class="nb">int</span><span class="p">,</span> <span class="nb">int</span><span class="p">]:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Calculate the start and end points of a genomic window, respecting the given strand.</span>

<span class="sd">    For &quot;+&quot; strand, the window is created from the `start` point, and for &quot;-&quot; strand,</span>
<span class="sd">    the window is created from the `end` point. This method ensures that the resulting</span>
<span class="sd">    window is of the specified `window_size`, handling undersized windows in the</span>
<span class="sd">    process. This method is particularly useful when selecting from the start of a gene.</span>

<span class="sd">    Args:</span>
<span class="sd">        start (int): The start point of the gene on the genome.</span>
<span class="sd">        end (int): The end point of the gene on the genome.</span>
<span class="sd">        strand (str): The strand of the gene, either &quot;+&quot; or &quot;-&quot;.</span>
<span class="sd">        window_size (int): The desired size of the window.</span>

<span class="sd">    Returns:</span>
<span class="sd">        tuple[int, int]: The calculated start and end points of the window.</span>

<span class="sd">    Raises:</span>
<span class="sd">        AssertionError: If the resulting window size doesn&#39;t match the</span>
<span class="sd">            specified `window_size`.</span>

<span class="sd">    Example:</span>
<span class="sd">        &gt;&gt;&gt; calculate_window_undersized(10, 50, &quot;+&quot;, 20)</span>
<span class="sd">        (10, 30)</span>

<span class="sd">        &gt;&gt;&gt; calculate_window_undersized(10, 50, &quot;-&quot;, 20)</span>
<span class="sd">        (30, 50)</span>

<span class="sd">    &quot;&quot;&quot;</span>
    <span class="c1"># select from start of gene, since this is such a strong signal for function</span>
    <span class="k">if</span> <span class="n">strand</span> <span class="o">==</span> <span class="s2">&quot;+&quot;</span><span class="p">:</span>
        <span class="n">start_window</span> <span class="o">=</span> <span class="n">start</span>
        <span class="n">end_window</span> <span class="o">=</span> <span class="n">start</span> <span class="o">+</span> <span class="n">window_size</span>
    <span class="k">elif</span> <span class="n">strand</span> <span class="o">==</span> <span class="s2">&quot;-&quot;</span><span class="p">:</span>
        <span class="n">start_window</span> <span class="o">=</span> <span class="n">end</span> <span class="o">-</span> <span class="n">window_size</span>
        <span class="n">end_window</span> <span class="o">=</span> <span class="n">end</span>
    <span class="k">assert</span> <span class="p">(</span>
        <span class="n">end_window</span> <span class="o">-</span> <span class="n">start_window</span>
    <span class="p">)</span> <span class="o">==</span> <span class="n">window_size</span><span class="p">,</span> <span class="p">(</span>
        <span class="sa">f</span><span class="s2">&quot;Window sizing is incorrect. Window is larger than </span><span class="si">{</span><span class="n">window_size</span><span class="si">}</span><span class="s2">bp&quot;</span>
    <span class="p">)</span>
    <span class="k">return</span> <span class="n">start_window</span><span class="p">,</span> <span class="n">end_window</span>


<div class="viewcode-block" id="calculate_window_bounds"><a class="viewcode-back" href="../../../generated/torchcell.sequence.calculate_window_bounds.html#torchcell.sequence.calculate_window_bounds">[docs]</a><span class="k">def</span> <span class="nf">calculate_window_bounds</span><span class="p">(</span>
    <span class="n">start</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">end</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">strand</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">window_size</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">chromosome_length</span><span class="p">:</span> <span class="nb">int</span>
<span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">[</span><span class="nb">int</span><span class="p">,</span> <span class="nb">int</span><span class="p">]:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Calculate the window bounds for genomic sequences.</span>

<span class="sd">    This function calculates window bounds for genomic sequences while considering</span>
<span class="sd">    strand direction. It ensures the window does not exceed the chromosome length.</span>

<span class="sd">    Args:</span>
<span class="sd">        start (int): The start point of the sequence on the genome.</span>
<span class="sd">        end (int): The end point of the sequence on the genome.</span>
<span class="sd">        strand (str): The strand of the gene, either &quot;+&quot; or &quot;-&quot;.</span>
<span class="sd">        window_size (int): Desired window size.</span>
<span class="sd">        chromosome_length (int): Length of the chromosome.</span>

<span class="sd">    Returns:</span>
<span class="sd">        tuple[int, int]: The calculated start and end points of the window.</span>

<span class="sd">    Raises:</span>
<span class="sd">        ValueError: If the end position is out of bounds of the chromosome,</span>
<span class="sd">            if the start position is greater than or equal to the end position,</span>
<span class="sd">            or if the window size is greater than the chromosome length.</span>

<span class="sd">    Examples:</span>
<span class="sd">        &gt;&gt;&gt; calculate_window_bounds(0, 20, &quot;+&quot;, 40, 100)</span>
<span class="sd">        (0, 40)</span>
<span class="sd">        &gt;&gt;&gt; calculate_window_bounds(5, 25, &quot;+&quot;, 50, 100)</span>
<span class="sd">        (0, 50)</span>
<span class="sd">        &gt;&gt;&gt; calculate_window_bounds(75, 95, &quot;+&quot;, 50, 100)</span>
<span class="sd">        (50, 100)</span>
<span class="sd">        &gt;&gt;&gt; calculate_window_bounds(0, 20, &quot;-&quot;, 40, 100)</span>
<span class="sd">        (0, 40)</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">if</span> <span class="n">end</span> <span class="o">&gt;</span> <span class="n">chromosome_length</span><span class="p">:</span>
        <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;End position is out of bounds of chromosome&quot;</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">start</span> <span class="o">&gt;=</span> <span class="n">end</span><span class="p">:</span>
        <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;Start position must be less than end position&quot;</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">window_size</span> <span class="o">&gt;</span> <span class="n">chromosome_length</span><span class="p">:</span>
        <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;Window size should never be greater than chromosome length&quot;</span><span class="p">)</span>

    <span class="n">seq_length</span> <span class="o">=</span> <span class="n">end</span> <span class="o">-</span> <span class="n">start</span>
    <span class="k">if</span> <span class="n">window_size</span> <span class="o">&lt;</span> <span class="n">seq_length</span><span class="p">:</span>
        <span class="c1"># log info that the window size is smaller than the sequence</span>
        <span class="n">log</span><span class="o">.</span><span class="n">info</span><span class="p">(</span>
            <span class="sa">f</span><span class="s2">&quot;Window size </span><span class="si">{</span><span class="n">window_size</span><span class="si">}</span><span class="s2"> is smaller than sequence length </span><span class="si">{</span><span class="n">end</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">start</span><span class="si">}</span><span class="s2">.&quot;</span>
        <span class="p">)</span>
        <span class="n">start_window</span><span class="p">,</span> <span class="n">end_window</span> <span class="o">=</span> <span class="n">calculate_window_undersized</span><span class="p">(</span>
            <span class="n">start</span><span class="p">,</span> <span class="n">end</span><span class="p">,</span> <span class="n">strand</span><span class="p">,</span> <span class="n">window_size</span>
        <span class="p">)</span>
        <span class="k">return</span> <span class="n">start_window</span><span class="p">,</span> <span class="n">end_window</span>

    <span class="n">flank_seq_length</span> <span class="o">=</span> <span class="p">(</span><span class="n">window_size</span> <span class="o">-</span> <span class="n">seq_length</span><span class="p">)</span> <span class="o">//</span> <span class="mi">2</span>
    <span class="n">start_window</span> <span class="o">=</span> <span class="n">start</span> <span class="o">-</span> <span class="n">flank_seq_length</span>
    <span class="n">end_window</span> <span class="o">=</span> <span class="n">end</span> <span class="o">+</span> <span class="n">flank_seq_length</span>

    <span class="c1"># Case where start of window is out of bounds</span>
    <span class="k">if</span> <span class="n">start_window</span> <span class="o">&lt;</span> <span class="mi">0</span><span class="p">:</span>
        <span class="n">start_window</span> <span class="o">=</span> <span class="mi">0</span>  <span class="c1"># Set start of window to beginning of sequence</span>
        <span class="n">end_window</span> <span class="o">=</span> <span class="n">window_size</span>

    <span class="c1"># Case where end of window is out of bounds</span>
    <span class="k">if</span> <span class="n">end_window</span> <span class="o">&gt;</span> <span class="n">chromosome_length</span><span class="p">:</span>
        <span class="n">end_window</span> <span class="o">=</span> <span class="n">chromosome_length</span>  <span class="c1"># Set end of window to end of sequence</span>
        <span class="n">start_window</span> <span class="o">=</span> <span class="n">chromosome_length</span> <span class="o">-</span> <span class="n">window_size</span>

    <span class="c1"># Edge case if the adjusted window does not match the window size</span>
    <span class="c1"># and is up against chromosome ends</span>
    <span class="k">if</span> <span class="p">(</span>
        <span class="nb">abs</span><span class="p">((</span><span class="n">end_window</span> <span class="o">-</span> <span class="n">start_window</span><span class="p">)</span> <span class="o">-</span> <span class="n">window_size</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span>
        <span class="ow">and</span> <span class="n">end_window</span> <span class="o">==</span> <span class="n">chromosome_length</span>
    <span class="p">):</span>
        <span class="n">start_window</span> <span class="o">-=</span> <span class="mi">1</span>
    <span class="k">elif</span> <span class="nb">abs</span><span class="p">((</span><span class="n">end_window</span> <span class="o">-</span> <span class="n">start_window</span><span class="p">)</span> <span class="o">-</span> <span class="n">window_size</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span> <span class="ow">and</span> <span class="n">start_window</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
        <span class="n">end_window</span> <span class="o">+=</span> <span class="mi">1</span>
    <span class="c1"># Select more of the upstream</span>
    <span class="k">elif</span> <span class="nb">abs</span><span class="p">((</span><span class="n">end_window</span> <span class="o">-</span> <span class="n">start_window</span><span class="p">)</span> <span class="o">-</span> <span class="n">window_size</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span> <span class="ow">and</span> <span class="n">strand</span> <span class="o">==</span> <span class="s2">&quot;+&quot;</span><span class="p">:</span>
        <span class="n">start_window</span> <span class="o">-=</span> <span class="mi">1</span>
    <span class="k">elif</span> <span class="nb">abs</span><span class="p">((</span><span class="n">end_window</span> <span class="o">-</span> <span class="n">start_window</span><span class="p">)</span> <span class="o">-</span> <span class="n">window_size</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span> <span class="ow">and</span> <span class="n">strand</span> <span class="o">==</span> <span class="s2">&quot;-&quot;</span><span class="p">:</span>
        <span class="n">end_window</span> <span class="o">+=</span> <span class="mi">1</span>
    <span class="k">assert</span> <span class="n">start_window</span> <span class="o">&lt;=</span> <span class="n">start</span><span class="p">,</span> <span class="s2">&quot;Start window must be leq start.&quot;</span>
    <span class="k">assert</span> <span class="n">end_window</span> <span class="o">&gt;=</span> <span class="n">end</span><span class="p">,</span> <span class="s2">&quot;End window must be geq end.&quot;</span>
    <span class="k">return</span> <span class="n">start_window</span><span class="p">,</span> <span class="n">end_window</span></div>


<span class="k">def</span> <span class="nf">calculate_window_undersized_symmetric</span><span class="p">(</span>
    <span class="n">start</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">end</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">window_size</span><span class="p">:</span> <span class="nb">int</span>
<span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">[</span><span class="nb">int</span><span class="p">,</span> <span class="nb">int</span><span class="p">]:</span>
<span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Calculate symmetric window bounds for genomic sequences.</span>

<span class="sd">    This function calculates symmetric window bounds for genomic sequences and</span>
<span class="sd">    ensures that the window size is valid and the start and end are not equal.</span>

<span class="sd">    Args:</span>
<span class="sd">        start (int): The start point of the sequence on the genome.</span>
<span class="sd">        end (int): The end point of the sequence on the genome.</span>
<span class="sd">        window_size (int): Desired window size.</span>

<span class="sd">    Returns:</span>
<span class="sd">        tuple[int, int]: The calculated start and end points of the window.</span>

<span class="sd">    Raises:</span>
<span class="sd">        ValueError: If the start and end positions are the same, or if the</span>
<span class="sd">            window size is less than 2.</span>

<span class="sd">    Examples:</span>
<span class="sd">        &gt;&gt;&gt; calculate_window_undersized_symmetric(10, 20, 4)</span>
<span class="sd">        (13, 17)</span>
<span class="sd">        &gt;&gt;&gt; calculate_window_undersized_symmetric(10, 20, 5)</span>
<span class="sd">        (13, 17)</span>

<span class="sd">    Note:</span>
<span class="sd">        For odd window sizes, the result will be adjusted to keep the window</span>
<span class="sd">        symmetric around the middle of the start and end points.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">if</span> <span class="n">start</span> <span class="o">==</span> <span class="n">end</span><span class="p">:</span>
        <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;Start and end positions are the same&quot;</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">window_size</span> <span class="o">&lt;</span> <span class="mi">2</span><span class="p">:</span>
        <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;Window size must be at least 2&quot;</span><span class="p">)</span>
    <span class="c1"># find the middle</span>
    <span class="n">middle</span> <span class="o">=</span> <span class="p">(</span><span class="n">start</span> <span class="o">+</span> <span class="n">end</span><span class="p">)</span> <span class="o">//</span> <span class="mi">2</span>
    <span class="c1"># take half above and half below</span>
    <span class="n">flank_size</span> <span class="o">=</span> <span class="n">window_size</span> <span class="o">//</span> <span class="mi">2</span>
    <span class="n">start_window</span> <span class="o">=</span> <span class="n">middle</span> <span class="o">-</span> <span class="n">flank_size</span>
    <span class="n">end_window</span> <span class="o">=</span> <span class="n">middle</span> <span class="o">+</span> <span class="n">flank_size</span>
    <span class="c1"># Minus one for odd.</span>
    <span class="k">assert</span> <span class="p">(</span>
        <span class="n">end_window</span> <span class="o">-</span> <span class="n">start_window</span>
    <span class="p">)</span> <span class="o">&gt;=</span> <span class="n">window_size</span> <span class="o">-</span> <span class="mi">1</span><span class="p">,</span> <span class="p">(</span>
        <span class="sa">f</span><span class="s2">&quot;Window sizing is incorrect. Window is larger than </span><span class="si">{</span><span class="n">window_size</span><span class="si">}</span><span class="s2">bp&quot;</span>
    <span class="p">)</span>
    <span class="k">return</span> <span class="n">start_window</span><span class="p">,</span> <span class="n">end_window</span>


<div class="viewcode-block" id="calculate_window_bounds_symmetric"><a class="viewcode-back" href="../../../generated/torchcell.sequence.calculate_window_bounds_symmetric.html#torchcell.sequence.calculate_window_bounds_symmetric">[docs]</a><span class="k">def</span> <span class="nf">calculate_window_bounds_symmetric</span><span class="p">(</span>
    <span class="n">start</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">end</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">window_size</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">chromosome_length</span><span class="p">:</span> <span class="nb">int</span>
<span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">[</span><span class="nb">int</span><span class="p">,</span> <span class="nb">int</span><span class="p">]:</span>
<span class="w">    </span><span class="sa">r</span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Calculate symmetric window bounds considering chromosome limits.</span>

<span class="sd">    This function calculates symmetric window bounds for genomic sequences, taking</span>
<span class="sd">    the chromosome length into account to avoid exceeding it, and also ensuring</span>
<span class="sd">    that the window size, start and end positions are valid within these limits.</span>

<span class="sd">    Args:</span>
<span class="sd">        start (int): The start point of the sequence on the genome.</span>
<span class="sd">        end (int): The end point of the sequence on the genome.</span>
<span class="sd">        window_size (int): The desired window size.</span>
<span class="sd">        chromosome_length (int): The total length of the chromosome.</span>

<span class="sd">    Returns:</span>
<span class="sd">        tuple[int, int]: The calculated start and end points of the symmetric window.</span>

<span class="sd">    Raises:</span>
<span class="sd">        ValueError: If the end position is out of chromosome bounds, start position</span>
<span class="sd">            is not less than end position, or window size is greater than chromosome</span>
<span class="sd">            length.</span>

<span class="sd">    Examples:</span>
<span class="sd">        &gt;&gt;&gt; calculate_window_bounds_symmetric(5, 15, 30, 100)</span>
<span class="sd">        (0, 20)</span>
<span class="sd">        &gt;&gt;&gt; calculate_window_bounds_symmetric(80, 95, 30, 100)</span>
<span class="sd">        (75, 100)</span>
<span class="sd">        &gt;&gt;&gt; calculate_window_bounds_symmetric(45, 55, 30, 100)</span>
<span class="sd">        (35, 65)</span>

<span class="sd">    Note:</span>
<span class="sd">        The function also asserts that the calculated window is within the desired</span>
<span class="sd">        limits, symmetric around the middle of the sequence, and the start and end</span>
<span class="sd">        of the window are appropriately bounded within the sequence limits.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">if</span> <span class="n">end</span> <span class="o">&gt;</span> <span class="n">chromosome_length</span><span class="p">:</span>
        <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;End position is out of bounds of chromosome&quot;</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">start</span> <span class="o">&gt;=</span> <span class="n">end</span><span class="p">:</span>
        <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;Start position must be less than end position&quot;</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">window_size</span> <span class="o">&gt;</span> <span class="n">chromosome_length</span><span class="p">:</span>
        <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;Window size should never be greater than chromosome length&quot;</span><span class="p">)</span>

    <span class="k">if</span> <span class="n">window_size</span> <span class="o">&lt;</span> <span class="n">end</span> <span class="o">-</span> <span class="n">start</span><span class="p">:</span>
        <span class="n">start_window</span><span class="p">,</span> <span class="n">end_window</span> <span class="o">=</span> <span class="n">calculate_window_undersized_symmetric</span><span class="p">(</span>
            <span class="n">start</span><span class="p">,</span> <span class="n">end</span><span class="p">,</span> <span class="n">window_size</span>
        <span class="p">)</span>
        <span class="k">return</span> <span class="n">start_window</span><span class="p">,</span> <span class="n">end_window</span>

    <span class="n">seq_length</span> <span class="o">=</span> <span class="n">end</span> <span class="o">-</span> <span class="n">start</span>
    <span class="c1"># Find limiting window size</span>
    <span class="n">flank_size</span> <span class="o">=</span> <span class="p">(</span><span class="n">window_size</span> <span class="o">-</span> <span class="n">seq_length</span><span class="p">)</span> <span class="o">//</span> <span class="mi">2</span>
    <span class="n">start_flank_pos</span> <span class="o">=</span> <span class="n">start</span> <span class="o">-</span> <span class="n">flank_size</span>
    <span class="n">end_flank_pos</span> <span class="o">=</span> <span class="n">end</span> <span class="o">+</span> <span class="n">flank_size</span>
    <span class="k">if</span> <span class="n">start_flank_pos</span> <span class="o">&gt;=</span> <span class="mi">0</span><span class="p">:</span>
        <span class="n">start_window</span> <span class="o">=</span> <span class="n">start_flank_pos</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">start_window</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="k">if</span> <span class="n">end_flank_pos</span> <span class="o">&lt;=</span> <span class="n">chromosome_length</span><span class="p">:</span>
        <span class="n">end_window</span> <span class="o">=</span> <span class="n">end_flank_pos</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">end_window</span> <span class="o">=</span> <span class="n">chromosome_length</span>

    <span class="c1"># Based on limiting window size take symmetric window</span>
    <span class="n">sym_flank_len</span> <span class="o">=</span> <span class="nb">min</span><span class="p">((</span><span class="n">start</span> <span class="o">-</span> <span class="n">start_window</span><span class="p">),</span> <span class="p">(</span><span class="n">end_window</span> <span class="o">-</span> <span class="n">end</span><span class="p">))</span>
    <span class="n">start_window</span> <span class="o">=</span> <span class="n">start</span> <span class="o">-</span> <span class="n">sym_flank_len</span>
    <span class="n">end_window</span> <span class="o">=</span> <span class="n">end</span> <span class="o">+</span> <span class="n">sym_flank_len</span>

    <span class="k">assert</span> <span class="p">(</span>
        <span class="n">end_window</span> <span class="o">-</span> <span class="n">start_window</span>
    <span class="p">)</span> <span class="o">&lt;=</span> <span class="n">window_size</span><span class="p">,</span> <span class="p">(</span>
        <span class="sa">f</span><span class="s2">&quot;Window sizing is incorrect. Window is larger than </span><span class="si">{</span><span class="n">window_size</span><span class="si">}</span><span class="s2">bp&quot;</span>
    <span class="p">)</span>
    <span class="k">assert</span> <span class="p">(</span>
        <span class="n">end_window</span> <span class="o">-</span> <span class="n">start_window</span>
    <span class="p">)</span> <span class="o">&gt;=</span> <span class="n">end</span> <span class="o">-</span> <span class="n">start</span><span class="p">,</span> <span class="p">(</span>
        <span class="sa">f</span><span class="s2">&quot;Window sizing is incorrect. Window is smaller than sequence </span><span class="si">{</span><span class="n">end</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">start</span><span class="si">}</span><span class="s2">bp&quot;</span>
    <span class="p">)</span>
    <span class="k">assert</span> <span class="p">(</span><span class="n">start</span> <span class="o">-</span> <span class="n">start_window</span><span class="p">)</span> <span class="o">==</span> <span class="p">(</span><span class="n">end_window</span> <span class="o">-</span> <span class="n">end</span><span class="p">),</span> <span class="s2">&quot;sequences are not symmetric&quot;</span>
    <span class="k">assert</span> <span class="n">start_window</span> <span class="o">&lt;=</span> <span class="n">start</span><span class="p">,</span> <span class="s2">&quot;Start window must be leq start.&quot;</span>
    <span class="k">assert</span> <span class="n">end_window</span> <span class="o">&gt;=</span> <span class="n">end</span><span class="p">,</span> <span class="s2">&quot;End window must be geq end.&quot;</span>

    <span class="k">return</span> <span class="n">start_window</span><span class="p">,</span> <span class="n">end_window</span></div>
</pre></div>

           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, PyG Team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>